<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />


<title>ASH_DE_RIN</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ASH_DE_RIN</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>June 7, 2017</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#pairwise-comparisons">Pairwise comparisons</a><ul>
<li><a href="#fit-the-linear-model">Fit the linear model</a></li>
</ul></li>
<li><a href="#make-venn-diagrams-for-directionality-using-ash">Make Venn Diagrams for directionality using ASH</a><ul>
<li><a href="#run-ash">Run ASH</a></li>
<li><a href="#find-where-the-genes-are-upregulated-between-tissues">Find where the genes are upregulated between tissues</a></li>
</ul></li>
<li><a href="#upregulated-and-downregulated-pairwise-figure-supplementary-figure">Upregulated and downregulated pairwise figure (Supplementary Figure)</a></li>
<li><a href="#upregulated-and-downregulated-pairwise-species-figure-supplementary-figure">Upregulated and downregulated pairwise species figure (Supplementary Figure)</a><ul>
<li><a href="#downregulated">Downregulated</a></li>
<li><a href="#find-where-the-genes-are-upregulated-between-species">Find where the genes are upregulated between species</a></li>
<li><a href="#find-where-the-genes-are-downregulated-between-species">Find where the genes are downregulated between species</a></li>
</ul></li>
<li><a href="#find-genes-with-a-tissue-specific-direction-significant-and-upregulated-in-1-versus-2-3-4-but-not-significant-in-2-v-3-4-and-3-v-4-with-fsr-supplementary-tables">Find genes with a tissue-specific direction (significant and upregulated in 1 versus 2, 3, 4 but not significant in 2 v 3, 4 and 3 v 4) with FSR (supplementary tables)</a></li>
<li><a href="#find-genes-with-a-tissue-specific-direction-significant-and-upregulated-in-1-versus-2-3-4-but-not-significant-in-2-v-3-4-and-3-v-4-with-fdr">Find genes with a tissue-specific direction (significant and upregulated in 1 versus 2, 3, 4 but not significant in 2 v 3, 4 and 3 v 4) with FDR</a></li>
<li><a href="#species-specific-1-direction-supplementary-table">Species specific 1 direction (supplementary table)</a></li>
<li><a href="#experimenting-with-interactions-3-species-1-tissue-versus-a-group-of-3-tissues">Experimenting with interactions (3 species, 1 tissue versus a group of 3 tissues)</a><ul>
<li><a href="#interactions-in-all-3-species-hearts-versus-all-other-tissues">Interactions in all 3 species’ hearts versus all other tissues</a></li>
<li><a href="#adaptive-shrinkage-on-interactions-in-all-3-species-hearts-versus-all-other-tissues">Adaptive shrinkage on interactions in all 3 species’ hearts versus all other tissues</a></li>
<li><a href="#interactions-in-all-3-species-livers-versus-all-other-tissues">Interactions in all 3 species’ livers versus all other tissues</a></li>
<li><a href="#adaptive-shrinkage-on-interactions-in-all-3-species-livers-versus-all-other-tissues">Adaptive shrinkage on interactions in all 3 species’ livers versus all other tissues</a></li>
<li><a href="#interactions-in-all-3-species-lungs-versus-all-other-tissues">Interactions in all 3 species’ lungs versus all other tissues</a></li>
<li><a href="#adaptive-shrinkage-on-interactions-in-all-3-species-lungs-versus-all-other-tissues">Adaptive shrinkage on interactions in all 3 species’ lungs versus all other tissues</a></li>
<li><a href="#interactions-in-all-3-species-kidneys-versus-all-other-tissues">Interactions in all 3 species’ kidneys versus all other tissues</a></li>
<li><a href="#tissue-specific-interactions-fdr">Tissue-specific interactions (FDR)</a></li>
<li><a href="#adaptive-shrinkage-on-interactions-in-all-3-species-kidneys-versus-all-other-tissues">Adaptive shrinkage on interactions in all 3 species’ kidneys versus all other tissues</a></li>
<li><a href="#number-of-interactions-using-fsr">Number of interactions using FSR</a></li>
</ul></li>
<li><a href="#combining-humans-and-chimps-to-have-a-great-ape-versus-rhesus-tissues-set-of-interactions-figure-3">Combining humans and chimps to have a Great Ape versus Rhesus tissues set of interactions (Figure 3)</a><ul>
<li><a href="#hearts">Hearts</a></li>
<li><a href="#livers">Livers</a></li>
<li><a href="#lungs">Lungs</a></li>
<li><a href="#kidneys">Kidneys</a></li>
<li><a href="#great-ape-tissue-specific-using-fdr">Great Ape Tissue Specific (using FDR)</a></li>
</ul></li>
<li><a href="#combining-chimps-and-rhesus-to-have-a-human-versus-non-human-tissues-set-of-interactions">Combining chimps and rhesus to have a human versus non-human tissues set of interactions</a><ul>
<li><a href="#hearts-1">Hearts</a></li>
<li><a href="#livers-1">Livers</a></li>
<li><a href="#lungs-1">Lungs</a></li>
<li><a href="#kidneys-1">Kidneys</a></li>
<li><a href="#humannon-human-tissue-specific-using-fdr">Human/Non-Human Tissue Specific (using FDR)</a></li>
</ul></li>
</ul>
</div>

<pre class="r"><code># Load libraries

library(&quot;gplots&quot;)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>library(&quot;RColorBrewer&quot;)
library(&quot;scales&quot;)</code></pre>
<pre><code>Warning: package &#39;scales&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;R.utils&quot;)</code></pre>
<pre><code>Warning: package &#39;R.utils&#39; was built under R version 3.4.4</code></pre>
<pre><code>Loading required package: R.oo</code></pre>
<pre><code>Warning: package &#39;R.oo&#39; was built under R version 3.4.4</code></pre>
<pre><code>Loading required package: R.methodsS3</code></pre>
<pre><code>R.methodsS3 v1.7.1 (2016-02-15) successfully loaded. See ?R.methodsS3 for help.</code></pre>
<pre><code>R.oo v1.22.0 (2018-04-21) successfully loaded. See ?R.oo for help.</code></pre>
<pre><code>
Attaching package: &#39;R.oo&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:methods&#39;:

    getClasses, getMethods</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    attach, detach, gc, load, save</code></pre>
<pre><code>R.utils v2.7.0 successfully loaded. See ?R.utils for help.</code></pre>
<pre><code>
Attaching package: &#39;R.utils&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    timestamp</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    cat, commandArgs, getOption, inherits, isOpen, parse, warnings</code></pre>
<pre class="r"><code>library(&quot;plyr&quot;)
library(&quot;limma&quot;)
library(&quot;gridExtra&quot;)
library(&quot;VennDiagram&quot;)</code></pre>
<pre><code>Warning: package &#39;VennDiagram&#39; was built under R version 3.4.4</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>Loading required package: futile.logger</code></pre>
<pre class="r"><code>source(&quot;functions.R&quot;)
library(ashr)
library(cowplot)</code></pre>
<pre><code>Warning: package &#39;cowplot&#39; was built under R version 3.4.4</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code># Add function for making ggplot2 figures look good (from Bryan Pavlovic)

bjp&lt;-
theme(
  panel.border = element_rect(colour = &quot;black&quot;, fill = NA, size = 2),
  plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
  axis.text.y =  element_text(size = 14,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.text.x =  element_text(size = 14,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.title.y = element_text(size = 14,face = &quot;bold&quot;),
  axis.title.x = element_text(size = 14,face = &quot;bold&quot;),
  legend.text = element_text(size = 14,face = &quot;bold&quot;),
  legend.title = element_text(size = 14,face = &quot;bold&quot;),
  strip.text.x = element_text(size = 14,face = &quot;bold&quot;),
  strip.text.y = element_text(size = 14,face = &quot;bold&quot;),
  strip.background = element_rect(colour = &quot;black&quot;, size = 2))

# Set directory to save the data

data_dir &lt;- &quot;../data&quot;

# Load colors 

colors &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[1],brewer.pal(9, &quot;Blues&quot;)[9]))(100)
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))

# Retrieve RIN score for each sample
RNA_seq_info &lt;- read.csv(&quot;../../../Reg_Evo_Primates/data/RNA_seq_info.csv&quot;)
RIN &lt;- as.data.frame(RNA_seq_info[,22])
RIN &lt;- as.matrix(RIN)
#RIN &lt;- as.data.frame(RIN[-1,])
#RIN &lt;- as.matrix(RIN)
colnames(RIN) &lt;- c(&quot;RIN&quot;)

# Retrieve sample information
samples &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/Sample_info_RNAseq_limma.txt&quot;)

# Eliminate H1H
samples &lt;- samples[-17,]
#samples &lt;- samples[-1,]
dim(samples)</code></pre>
<pre><code>[1] 47  4</code></pre>
<pre class="r"><code># Make labels

labels &lt;- paste(samples$Species, samples$Tissue, sep=&quot;.&quot;)</code></pre>
<div id="pairwise-comparisons" class="section level2">
<h2>Pairwise comparisons</h2>
<pre class="r"><code># Load count data

counts_genes_in_cutoff &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/counts_12184.txt&quot;)
cpm_12184 &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/cpm_12184.txt&quot;)

# TMM 

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)</code></pre>
<pre><code>                     C1H      C1K      C1Li     C1Lu      C2H      C2K
ENSG00000000003 4.569101 6.484481  8.260731 5.481561 4.686636 6.076562
ENSG00000000419 5.842023 5.217972  5.937465 5.478545 5.681016 5.100404
ENSG00000000457 4.560130 5.214732  5.902494 4.972557 4.834031 5.289413
ENSG00000000460 1.506846 1.869887  2.080244 2.308985 1.660573 1.968249
ENSG00000000938 5.611783 3.819613  5.091152 7.550720 2.533135 4.178135
ENSG00000000971 6.877100 4.451824 11.368082 6.100181 6.135730 4.887383
                     C2Li     C2Lu      C3H      C3K      C3Li     C3Lu
ENSG00000000003  8.029471 4.564496 4.915377 6.406310  7.784365 5.875983
ENSG00000000419  5.813444 5.199855 5.675979 5.179418  6.413682 5.596709
ENSG00000000457  6.545270 4.985922 4.618657 5.204247  6.498053 5.168988
ENSG00000000460  2.324903 2.023533 1.580465 1.461635  2.344190 2.124699
ENSG00000000938  5.388459 8.083442 4.965147 4.223500  5.204433 7.160345
ENSG00000000971 11.387090 6.246512 5.606820 4.941061 11.420166 5.990777
                     C4H      C4K      C4Li     C4Lu      H1K      H1Li
ENSG00000000003 4.235754 6.503717  8.453727 5.430223 6.864660  6.576082
ENSG00000000419 5.785414 5.257938  5.881536 5.321782 5.588152  6.082997
ENSG00000000457 4.645293 5.023223  6.597499 5.263806 4.285007  4.953825
ENSG00000000460 1.456629 1.826787  2.206829 2.476664 2.766766  4.989335
ENSG00000000938 3.638952 3.621239  4.580376 7.717763 4.059344  4.479943
ENSG00000000971 6.845219 5.957838 11.330910 6.421417 6.585546 11.216641
                    H1Lu      H2H      H2K     H2Li     H2Lu         H3H
ENSG00000000003 5.099004 3.681088 7.205567 6.638944 4.181104  3.54360583
ENSG00000000419 5.810855 5.606326 5.461678 5.838444 5.313450  5.75090978
ENSG00000000457 4.502116 3.406682 4.158467 4.450840 4.201852  4.44063190
ENSG00000000460 3.318021 1.892216 1.978501 2.657920 2.553081 -0.07576077
ENSG00000000938 7.878166 5.560041 3.740312 5.990299 6.968892  4.09684184
ENSG00000000971 7.561408 6.363288 4.736443 9.409472 7.310814  6.22507411
                     H3K      H3Li     H3Lu      H4H      H4K      H4Li
ENSG00000000003 7.091569  7.735945 5.290097 4.284175 6.371782  6.590115
ENSG00000000419 5.854940  6.216818 5.009845 6.244527 5.608088  5.834153
ENSG00000000457 4.722790  4.993719 3.999170 3.312369 4.087480  5.176119
ENSG00000000460 2.990603  3.237751 2.457261 1.629959 1.983141  3.137731
ENSG00000000938 2.643134  5.741066 6.912746 4.918491 3.820852  6.899117
ENSG00000000971 5.313255 10.346500 7.124250 6.927089 6.032612 10.197598
                    H4Lu      R1H      R1K       R1Li     R1Lu      R2H
ENSG00000000003 4.463456 4.356369 6.932932  8.3343252 5.915547 4.625348
ENSG00000000419 5.350423 5.464082 5.391834  5.8259430 4.887057 5.295517
ENSG00000000457 4.173647 4.285211 5.024853  5.1435415 4.837373 4.311664
ENSG00000000460 2.604439 1.284359 1.555258 -0.1735364 2.480549 1.415808
ENSG00000000938 8.249002 1.837210 2.564210  3.8041639 6.515323 2.327061
ENSG00000000971 8.277236 4.027912 6.576019 12.1322643 7.445976 5.571685
                     R2K      R2Li     R2Lu      R3H      R3K      R3Li
ENSG00000000003 7.134183  8.640291 5.654663 4.469039 7.166047  8.202424
ENSG00000000419 5.043831  5.723120 4.881450 5.462958 5.367362  5.990303
ENSG00000000457 5.154832  5.511121 5.265617 4.241449 5.139792  5.359535
ENSG00000000460 1.320003  1.496941 2.394895 1.656614 1.874414  1.413727
ENSG00000000938 2.681748  3.672247 6.583375 2.870792 2.267214  3.636318
ENSG00000000971 6.778700 11.778253 7.231996 5.150488 6.054215 12.087241
                    R3Lu      R4H      R4K      R4Li     R4Lu
ENSG00000000003 5.453490 4.892515 7.094406  7.705335 5.361237
ENSG00000000419 4.956114 5.427374 5.215706  5.464168 5.041757
ENSG00000000457 5.165431 4.509684 5.144266  5.318210 4.828352
ENSG00000000460 2.577251 1.088179 1.509418  1.534185 2.847559
ENSG00000000938 6.845883 2.474777 2.310571  4.509304 6.834845
ENSG00000000971 7.258258 6.408881 6.469038 11.695479 7.298202</code></pre>
<pre class="r"><code>hist(cpm_in_cutoff, xlab = &quot;Log2(CPM)&quot;, main = &quot;Log2(CPM) values for genes meeting the filtering criteria&quot;, breaks = 100 )</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Make a design matrix

tissue &lt;- samples$Tissue
species &lt;- samples$Species
design &lt;- model.matrix(~ tissue*species + RIN)
colnames(design)[1] &lt;- &quot;Intercept&quot;

# Rename columns of the contrast matrix

colnames(design)[2] &lt;- &quot;kidney&quot;
colnames(design)[3] &lt;- &quot;liver&quot;
colnames(design)[4] &lt;- &quot;lung&quot;
colnames(design)[5] &lt;- &quot;human&quot;
colnames(design)[6] &lt;- &quot;rhesus&quot;

colnames(design)[8] &lt;- &quot;kidneyHuman&quot;
colnames(design)[9] &lt;- &quot;liverHuman&quot;
colnames(design)[10] &lt;- &quot;lungHuman&quot;
colnames(design)[11] &lt;- &quot;kidneyRhesus&quot;
colnames(design)[12] &lt;- &quot;liverRhesus&quot;
colnames(design)[13] &lt;- &quot;lungRhesus&quot;



# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept       kidney        liver         lung        human 
        47.0         12.0         12.0         12.0         15.0 
      rhesus          RIN  kidneyHuman   liverHuman    lungHuman 
        16.0        368.2          4.0          4.0          4.0 
kidneyRhesus  liverRhesus   lungRhesus 
         4.0          4.0          4.0 </code></pre>
<pre class="r"><code># Voom with individual as a random variable

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=T)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus &lt;- 0.2197275

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit.consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save

update_gene_pcs &lt;- prcomp(t(cpm.voom.cyclic$E), scale = T)

pc_values &lt;- update_gene_pcs$x

summary(lm(pc_values[,1]~as.factor(RNA_seq_info$Tissue)))</code></pre>
<pre><code>
Call:
lm(formula = pc_values[, 1] ~ as.factor(RNA_seq_info$Tissue))

Residuals:
    Min      1Q  Median      3Q     Max 
-57.834 -12.655  -2.726  16.612  39.530 

Coefficients:
                                     Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)                           -29.426      7.000  -4.204  0.00013
as.factor(RNA_seq_info$Tissue)kidney   37.050      9.691   3.823  0.00042
as.factor(RNA_seq_info$Tissue)liver    99.239      9.691  10.241 4.17e-13
as.factor(RNA_seq_info$Tissue)lung    -21.037      9.691  -2.171  0.03551
                                        
(Intercept)                          ***
as.factor(RNA_seq_info$Tissue)kidney ***
as.factor(RNA_seq_info$Tissue)liver  ***
as.factor(RNA_seq_info$Tissue)lung   *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 23.22 on 43 degrees of freedom
Multiple R-squared:  0.8107,    Adjusted R-squared:  0.7975 
F-statistic:  61.4 on 3 and 43 DF,  p-value: 1.381e-15</code></pre>
<pre class="r"><code>summary(lm(pc_values[,2]~as.factor(RNA_seq_info$Species)))</code></pre>
<pre><code>
Call:
lm(formula = pc_values[, 2] ~ as.factor(RNA_seq_info$Species))

Residuals:
    Min      1Q  Median      3Q     Max 
-82.003 -11.884   8.865  26.961  48.082 

Coefficients:
                                              Estimate Std. Error t value
(Intercept)                                      8.829     10.090   0.875
as.factor(RNA_seq_info$Species)human            15.606     14.505   1.076
as.factor(RNA_seq_info$Species)rhesus macaque  -40.565     14.269  -2.843
                                              Pr(&gt;|t|)   
(Intercept)                                    0.38633   
as.factor(RNA_seq_info$Species)human           0.28783   
as.factor(RNA_seq_info$Species)rhesus macaque  0.00676 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 40.36 on 44 degrees of freedom
Multiple R-squared:  0.2686,    Adjusted R-squared:  0.2353 
F-statistic: 8.078 on 2 and 44 DF,  p-value: 0.001027</code></pre>
<pre class="r"><code>summary(lm(pc_values[,2]~as.factor(RNA_seq_info$Tissue)))</code></pre>
<pre><code>
Call:
lm(formula = pc_values[, 2] ~ as.factor(RNA_seq_info$Tissue))

Residuals:
    Min      1Q  Median      3Q     Max 
-47.685 -22.275   6.966  14.536  52.703 

Coefficients:
                                     Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)                           -66.054      7.831  -8.435 1.16e-10
as.factor(RNA_seq_info$Tissue)kidney   68.289     10.841   6.299 1.34e-07
as.factor(RNA_seq_info$Tissue)liver    90.522     10.841   8.350 1.52e-10
as.factor(RNA_seq_info$Tissue)lung     99.901     10.841   9.215 9.72e-12
                                        
(Intercept)                          ***
as.factor(RNA_seq_info$Tissue)kidney ***
as.factor(RNA_seq_info$Tissue)liver  ***
as.factor(RNA_seq_info$Tissue)lung   ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 25.97 on 43 degrees of freedom
Multiple R-squared:  0.704, Adjusted R-squared:  0.6834 
F-statistic: 34.09 on 3 and 43 DF,  p-value: 1.934e-11</code></pre>
<div id="fit-the-linear-model" class="section level3">
<h3>Fit the linear model</h3>
<pre class="r"><code>fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit.consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# MA Plots
## If &#39;MA&#39; is an &#39;MArrayLM&#39; object, then the plot is a fitted model MA-plot in which the estimated coefficient is on the y-axis and the average A-value is on the x-axis.                

#limma::plotMA(fit.cyclic.norm, array=1, xlab=&quot;average coefficient&quot;, ylab=&quot;estimated coefficient&quot;)

## - Potential caveat: variances could be different between human, chimp and rhesus (see Gordon Smyth email, 7 June 2013).                                                               
##  We look at the standard error for each condition                                                    
hist(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma, breaks=100)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma), breaks=100)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma))</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## This seems to be pretty comparable between conditions. The human heart is higher, probably because of H1H missing and H3H with a bit strange behavior                                     
stderror &lt;- log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma)
boxplot(list(stderror[,1:4], stderror[,5:8], stderror[,9:12]))</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## A bit higher for human, and a bit lower for rhesus                                                                                                                                    
boxplot(list(stderror[,2:4], stderror[,6:8], stderror[,8:12])) ## excluding heart samples  </code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm1 &lt;- makeContrasts(HvC_Liver = human + liverHuman, 
                     HvC_Lung = human + lungHuman, 
                     HvC_Heart = human, 
                     HvC_Kidney = human + kidneyHuman, 
                     HvR_Liver = human + liverHuman - (rhesus + liverRhesus), 
                     HvR_Lung = human + lungHuman - (rhesus + lungRhesus), 
                     HvR_Heart = human - rhesus, 
                     HvR_Kidney = human + kidneyHuman - (rhesus + kidneyRhesus), 
                     CvR_Liver = -(rhesus + liverRhesus),  
                     CvR_Lung = -(rhesus + lungRhesus), 
                     CvR_Heart = -(rhesus), 
                     CvR_Kidney = -(rhesus + kidneyRhesus), 
                     H_HeartvLi = (human) - (human + liver + liverHuman), 
                     H_HeartvLu = (human) - (human + lung + lungHuman), 
                     H_HeartvK = (human) - (human + kidney + kidneyHuman), 
                     H_LivLu = (human + liver + liverHuman) - (human + lung + lungHuman), 
                     H_LivK = (human + liver + liverHuman) - (human + kidney + kidneyHuman),  
                     H_LuvK = (human + lung + lungHuman) - (human + kidney + kidneyHuman), 
                     C_HeartvLi = -(liver), 
                     C_HeartvLu = -(lung), 
                     C_HeartvK = -(kidney), 
                     C_LivLu = liver - lung, 
                     C_LivK = liver - kidney,  
                     C_LuvK = lung - kidney,  
                     R_HeartvLi = (rhesus) - (rhesus + liver + liverRhesus), 
                     R_HeartvLu = (rhesus) - (rhesus + lung + lungRhesus), 
                     R_HeartvK = (rhesus) - (rhesus + kidney + kidneyRhesus), 
                     R_LivLu = (rhesus + liver + liverRhesus) - (rhesus + lung + lungRhesus), 
                     R_LivK = (rhesus + liver + liverRhesus) - (rhesus + kidney + kidneyRhesus),  
                     R_LuvK = (rhesus + lung + lungRhesus) - (rhesus + kidney + kidneyRhesus),
                     HK_inter = kidneyHuman,
                     HLi_inter = liverHuman, 
                     HLu_inter = lungHuman,
                     RK_inter = kidneyRhesus,
                     RLi_inter = liverRhesus, 
                     RLu_inter = lungRhesus,
                     HRK_inter = kidneyHuman - kidneyRhesus,
                     HRLi_inter = liverHuman -liverRhesus, 
                     HRLu_inter = lungHuman - lungRhesus,
                     levels = design)

# Implement contrasts

contrasts_each_species &lt;- contrasts.fit(fit.cyclic.norm, cm1)
fit1 &lt;- eBayes(contrasts_each_species)

top3 &lt;- list(HvC_Liver =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HvC_Lung =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             HvC_Heart =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             HvC_Kidney =topTable(fit1, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             
             HvR_Liver =topTable(fit1, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HvR_Lung =topTable(fit1, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             HvR_Heart =topTable(fit1, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HvR_Kidney =topTable(fit1, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             
             CvR_Liver =topTable(fit1, coef=9, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CvR_Lung =topTable(fit1, coef=10, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             CvR_Heart =topTable(fit1, coef=11, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             CvR_Kidney =topTable(fit1, coef=12, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             H_HeartvLi =topTable(fit1, coef=13, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             H_HeartvLu =topTable(fit1, coef=14, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             H_HeartvK =topTable(fit1, coef=15, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             H_LivLu =topTable(fit1, coef=16, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             H_LivK =topTable(fit1, coef=17, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             H_LuvK =topTable(fit1, coef=18, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             C_HeartvLi =topTable(fit1, coef=19, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             C_HeartvLu =topTable(fit1, coef=20, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             C_HeartvK =topTable(fit1, coef=21, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             C_LivLu =topTable(fit1, coef=22, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             C_LivK =topTable(fit1, coef=23, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             C_LuvK =topTable(fit1, coef=24, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             R_HeartvLi =topTable(fit1, coef=25, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             R_HeartvLu =topTable(fit1, coef=26, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             R_HeartvK =topTable(fit1, coef=27, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             R_LivLu =topTable(fit1, coef=28, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             R_LivK =topTable(fit1, coef=29, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             R_LuvK =topTable(fit1, coef=30, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            HK_inter =topTable(fit1, coef=31, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
            HLi_inter =topTable(fit1, coef=32, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            HLu_inter =topTable(fit1, coef=33, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            RK_inter =topTable(fit1, coef=34, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
            RLi_inter =topTable(fit1, coef=35, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            RLu_inter =topTable(fit1, coef=36, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            HRK_inter =topTable(fit1, coef=37, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            HRLi_inter =topTable(fit1, coef=38, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
            HRLu_inter =topTable(fit1, coef=39, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))


# Set FDR level at 1% 

FDR_level &lt;- 0.01

## DE between HvC

HvC_Liver =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HvC_Liver[which(HvC_Liver$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 1364    7</code></pre>
<pre class="r"><code>HvC_Lung =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HvC_Lung[which(HvC_Lung$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 805   7</code></pre>
<pre class="r"><code>HvC_Heart =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(HvC_Heart[which(HvC_Heart$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 2195    7</code></pre>
<pre class="r"><code>HvC_Kidney =topTable(fit1, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(HvC_Kidney[which(HvC_Kidney$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 799   7</code></pre>
<pre class="r"><code>mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Humans and Chimps (FDR 1%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;, cat.cex = 1.2, main.cex = 1.5)
grid.draw(Four_comp)

## DE between HvR

HvR_Liver =topTable(fit1, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HvR_Liver[which(HvR_Liver$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 2868    7</code></pre>
<pre class="r"><code>HvR_Lung =topTable(fit1, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HvR_Lung[which(HvR_Lung$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 2833    7</code></pre>
<pre class="r"><code>HvR_Heart =topTable(fit1, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(HvR_Heart[which(HvR_Heart$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4098    7</code></pre>
<pre class="r"><code>HvR_Kidney =topTable(fit1, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)  
dim(HvR_Kidney[which(HvR_Kidney$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 2347    7</code></pre>
<pre class="r"><code>mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[8]]])[top3[[names(top3)[8]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Humans and Rhesus (FDR 1%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between CvR

CvR_Liver =topTable(fit1, coef=9, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CvR_Liver[which(CvR_Liver$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 3139    7</code></pre>
<pre class="r"><code>CvR_Lung =topTable(fit1, coef=10, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CvR_Lung[which(CvR_Lung$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 1917    7</code></pre>
<pre class="r"><code>CvR_Heart =topTable(fit1, coef=11, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(CvR_Heart[which(CvR_Heart$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 3218    7</code></pre>
<pre class="r"><code>CvR_Kidney =topTable(fit1, coef=12, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CvR_Kidney[which(CvR_Kidney$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 2695    7</code></pre>
<pre class="r"><code>mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[9]]])[top3[[names(top3)[9]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[10]]])[top3[[names(top3)[10]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[11]]])[top3[[names(top3)[11]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[12]]])[top3[[names(top3)[12]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Chimps and Rhesus (FDR 1%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

# DE between tissues within a species

  # Within humans
H_HeartvLi =topTable(fit1, coef=13, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(H_HeartvLi[which(H_HeartvLi$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4776    7</code></pre>
<pre class="r"><code>H_HeartvLu =topTable(fit1, coef=14, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)  
dim(H_HeartvLu[which(H_HeartvLu$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4037    7</code></pre>
<pre class="r"><code>H_HeartvK =topTable(fit1, coef=15, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)  
dim(H_HeartvK[which(H_HeartvK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4224    7</code></pre>
<pre class="r"><code>H_LivLu =topTable(fit1, coef=16, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(H_LivLu[which(H_LivLu$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4701    7</code></pre>
<pre class="r"><code>H_LivK =topTable(fit1, coef=17, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(H_LivK[which(H_LivK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4248    7</code></pre>
<pre class="r"><code>H_LuvK =topTable(fit1, coef=18, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(H_LuvK[which(H_LuvK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 3695    7</code></pre>
<pre class="r"><code>  # Within chimps

C_HeartvLi =topTable(fit1, coef=19, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(C_HeartvLi[which(C_HeartvLi$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 6295    7</code></pre>
<pre class="r"><code>C_HeartvLu =topTable(fit1, coef=20, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)  
dim(C_HeartvLu[which(C_HeartvLu$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 5365    7</code></pre>
<pre class="r"><code>C_HeartvK =topTable(fit1, coef=21, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)  
dim(C_HeartvK[which(C_HeartvK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4971    7</code></pre>
<pre class="r"><code>C_LivLu =topTable(fit1, coef=22, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(C_LivLu[which(C_LivLu$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 6247    7</code></pre>
<pre class="r"><code>C_LivK =topTable(fit1, coef=23, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(C_LivK[which(C_LivK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4625    7</code></pre>
<pre class="r"><code>C_LuvK =topTable(fit1, coef=24, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(C_LuvK[which(C_LuvK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 4623    7</code></pre>
<pre class="r"><code>  # Within rhesus

R_HeartvLi =topTable(fit1, coef=25, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(R_HeartvLi[which(R_HeartvLi$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 6933    7</code></pre>
<pre class="r"><code>R_HeartvLu =topTable(fit1, coef=26, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(R_HeartvLu[which(R_HeartvLu$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 6814    7</code></pre>
<pre class="r"><code>R_HeartvK =topTable(fit1, coef=27, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)  
dim(R_HeartvK[which(R_HeartvK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 5980    7</code></pre>
<pre class="r"><code>R_LivLu =topTable(fit1, coef=28, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(R_LivLu[which(R_LivLu$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 7027    7</code></pre>
<pre class="r"><code>R_LivK =topTable(fit1, coef=29, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
dim(R_LivK[which(R_LivK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 5126    7</code></pre>
<pre class="r"><code>R_LuvK =topTable(fit1, coef=30, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(R_LuvK[which(R_LuvK$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 5721    7</code></pre>
<pre class="r"><code>## DE between Heart and Liver

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Heart and Liver (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Heart and Lung

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Heart and Lung (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## DE between Heart and Kidney

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Heart and Kidney (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## DE between Liver and Lung

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Liver and Lung (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Liver and Kidney

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Liver and Kidney (FDR 1%)&quot;, cex=1.5, lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Lung and Kidney

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Lung and Kidney (FDR 1%)&quot;, cex=1.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
<pre class="r"><code>  # Within humans
#H_HeartvLi =topTable(fit1, coef=13, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) 
#dim(H_HeartvLi[which(H_HeartvLi$adj.P.Val &lt; FDR_level), ]) 

human_h_li &lt;- H_HeartvLi[which(H_HeartvLi$adj.P.Val &lt; FDR_level), ]
human_h_lu &lt;- H_HeartvLu[which(H_HeartvLu$adj.P.Val &lt; FDR_level), ]
human_h_k &lt;- H_HeartvK[which(H_HeartvK$adj.P.Val &lt; FDR_level), ]

human_li_lu &lt;- H_LivLu[which(H_LivLu$adj.P.Val &lt; FDR_level), ]
human_li_k &lt;- H_LivK[which(H_LivK$adj.P.Val &lt; FDR_level), ]
human_lu_k &lt;- H_LuvK[which(H_LuvK$adj.P.Val &lt; FDR_level), ]

heart_only &lt;- intersect(intersect(rownames(human_h_li), rownames(human_h_lu)), rownames(human_h_k))
not_heart &lt;- union(union(rownames(human_li_lu), rownames(human_li_k)), rownames(human_lu_k))

human_heart_only &lt;- setdiff(heart_only, not_heart)




chimp_h_li &lt;- C_HeartvLi[which(C_HeartvLi$adj.P.Val &lt; FDR_level), ]
chimp_h_lu &lt;- C_HeartvLu[which(C_HeartvLu$adj.P.Val &lt; FDR_level), ]
chimp_h_k &lt;- C_HeartvK[which(C_HeartvK$adj.P.Val &lt; FDR_level), ]

chimp_li_lu &lt;- C_LivLu[which(C_LivLu$adj.P.Val &lt; FDR_level), ]
chimp_li_k &lt;- C_LivK[which(C_LivK$adj.P.Val &lt; FDR_level), ]
chimp_lu_k &lt;- C_LuvK[which(C_LuvK$adj.P.Val &lt; FDR_level), ]

heart_only &lt;- intersect(intersect(rownames(chimp_h_li), rownames(chimp_h_lu)), rownames(chimp_h_k))
not_heart &lt;- union(union(rownames(chimp_li_lu), rownames(chimp_li_k)), rownames(chimp_lu_k))

chimp_heart_only &lt;- setdiff(heart_only, not_heart)


rhesus_h_li &lt;- R_HeartvLi[which(R_HeartvLi$adj.P.Val &lt; FDR_level), ]
rhesus_h_lu &lt;- R_HeartvLu[which(R_HeartvLu$adj.P.Val &lt; FDR_level), ]
rhesus_h_k &lt;- R_HeartvK[which(R_HeartvK$adj.P.Val &lt; FDR_level), ]

rhesus_li_lu &lt;- R_LivLu[which(R_LivLu$adj.P.Val &lt; FDR_level), ]
rhesus_li_k &lt;- R_LivK[which(R_LivK$adj.P.Val &lt; FDR_level), ]
rhesus_lu_k &lt;- R_LuvK[which(R_LuvK$adj.P.Val &lt; FDR_level), ]

heart_only &lt;- intersect(intersect(rownames(rhesus_h_li), rownames(rhesus_h_lu)), rownames(rhesus_h_k))
not_heart &lt;- union(union(rownames(rhesus_li_lu), rownames(rhesus_li_k)), rownames(rhesus_lu_k))

rhesus_heart_only &lt;- setdiff(heart_only, not_heart)</code></pre>
</div>
</div>
<div id="make-venn-diagrams-for-directionality-using-ash" class="section level2">
<h2>Make Venn Diagrams for directionality using ASH</h2>
<p>In this section, we are going to detect genes that have differences in the same direction (false sign rate = 1%).</p>
<div id="run-ash" class="section level3">
<h3>Run ASH</h3>
<pre class="r"><code># Prepare the data for ASH
tests &lt;- colnames(fit1$coefficients)
results &lt;- vector(length = length(tests), mode = &quot;list&quot;)
names(results) &lt;- tests

# Perform multiple testing correction with adaptive shrinkage (ASH) 
 #
 # x - object MArrayLM from eBayes output
 # coef - coefficient tested by eBayes

run_ash &lt;- function(x, coef){
  #stopifnot(class(x) == &quot;MArrayLM&quot;, coef %in% colnames(x$coefficients),
  #             length(unique(x$df.total) == 1))
  result &lt;- ash(betahat = x$coefficients[, coef], sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post), df = x$df.total[1])
  return(result)
}

get_results &lt;- function(x, number = nrow(x$coefficients), sort.by = &quot;none&quot;,
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == &quot;MArrayLM&quot;)
  results &lt;- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] &lt;- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit1, coef = test)
  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code># Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, &quot;combined-limma.rds&quot;))

# Get the FSR for all genes in all comparisons

#results &lt;- readRDS(&quot;../data/combined-limma.rds&quot;)
#write.csv(results, file = &quot;../data/pairwise_FSR.csv&quot;)</code></pre>
</div>
<div id="find-where-the-genes-are-upregulated-between-tissues" class="section level3">
<h3>Find where the genes are upregulated between tissues</h3>
<pre class="r"><code># Set false sign rate
FDR_level &lt;- 0.05
FSR_level &lt;- 0.005

# FDR upregulated / downregulated

test &lt;- &quot;HvC_Liver&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 1323</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Liver&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 1261</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Lung&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 802</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Lung&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 703</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Heart&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 1691</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Heart&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 1797</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 789</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 671</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2161</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2256</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2121</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2133</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2561</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3048</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 1929</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 1979</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2155</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2262</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 1579</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 1612</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2100</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2523</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2032</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2086</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3046</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3142</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2774</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2751</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2880</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2796</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3124</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2992</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2879</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2782</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2389</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2727</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3724</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3762</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3274</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3460</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3174</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3042</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3777</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3744</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3064</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 2960</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 2930</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3096</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 4073</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 4029</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 4048</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3938</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3808</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3369</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 4304</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3941</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3382</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3104</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &gt; 0])</code></pre>
<pre><code>[1] 3373</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$logFC &lt; 0])</code></pre>
<pre><code>[1] 3672</code></pre>
<pre class="r"><code># FSR

FSR_level &lt;- 0.01


test &lt;- &quot;HvC_Liver&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 1127</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Liver&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 1039</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Lung&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 654</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Lung&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 601</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Heart&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 1751</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Heart&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 1860</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 650</code></pre>
<pre class="r"><code>test &lt;- &quot;HvC_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 483</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2098</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2265</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2169</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2226</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2855</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3340</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 1955</code></pre>
<pre class="r"><code>test &lt;- &quot;HvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2036</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2104</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Liver&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2290</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 1449</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Lung&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 1566</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2140</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Heart&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2666</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2146</code></pre>
<pre class="r"><code>test &lt;- &quot;CvR_Kidney&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2302</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3252</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3344</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3039</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3079</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3049</code></pre>
<pre class="r"><code>test &lt;- &quot;H_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2974</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3334</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3172</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2988</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LivK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2903</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 2405</code></pre>
<pre class="r"><code>test &lt;- &quot;H_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 2740</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 4075</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 4174</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3663</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3898</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3364</code></pre>
<pre class="r"><code>test &lt;- &quot;C_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3280</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 4129</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 4121</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3168</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LivK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3045</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3139</code></pre>
<pre class="r"><code>test &lt;- &quot;C_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3289</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 4452</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLi&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 4486</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 4420</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 4362</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 4263</code></pre>
<pre class="r"><code>test &lt;- &quot;R_HeartvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3769</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 4750</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivLu&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 4353</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3702</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LivK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 3354</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0])</code></pre>
<pre><code>[1] 3697</code></pre>
<pre class="r"><code>test &lt;- &quot;R_LuvK&quot;
length((results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0])</code></pre>
<pre><code>[1] 4035</code></pre>
<pre class="r"><code># Heart versus kidney FSR

mylist &lt;- list()

test &lt;- &quot;H_HeartvK&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_HeartvK&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_HeartvK&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]


# Find ones that are DE in hearts and kidneys in chimpanzees and rhesus but not humans

setdiff(intersect(mylist[[&quot;Rhesus Macaque&quot;]], mylist[[&quot;Chimpanzee&quot;]]), mylist[[&quot;Human&quot;]])</code></pre>
<pre><code>  [1] &quot;ENSG00000003147&quot; &quot;ENSG00000004660&quot; &quot;ENSG00000004700&quot;
  [4] &quot;ENSG00000004897&quot; &quot;ENSG00000004961&quot; &quot;ENSG00000005175&quot;
  [7] &quot;ENSG00000005238&quot; &quot;ENSG00000005249&quot; &quot;ENSG00000005700&quot;
 [10] &quot;ENSG00000006576&quot; &quot;ENSG00000006740&quot; &quot;ENSG00000006837&quot;
 [13] &quot;ENSG00000007168&quot; &quot;ENSG00000008128&quot; &quot;ENSG00000008516&quot;
 [16] &quot;ENSG00000009335&quot; &quot;ENSG00000009954&quot; &quot;ENSG00000010017&quot;
 [19] &quot;ENSG00000011105&quot; &quot;ENSG00000011198&quot; &quot;ENSG00000011275&quot;
 [22] &quot;ENSG00000011347&quot; &quot;ENSG00000011454&quot; &quot;ENSG00000011523&quot;
 [25] &quot;ENSG00000012822&quot; &quot;ENSG00000013374&quot; &quot;ENSG00000013561&quot;
 [28] &quot;ENSG00000014138&quot; &quot;ENSG00000015479&quot; &quot;ENSG00000020129&quot;
 [31] &quot;ENSG00000023287&quot; &quot;ENSG00000023330&quot; &quot;ENSG00000023734&quot;
 [34] &quot;ENSG00000025039&quot; &quot;ENSG00000025293&quot; &quot;ENSG00000025800&quot;
 [37] &quot;ENSG00000028116&quot; &quot;ENSG00000028203&quot; &quot;ENSG00000029725&quot;
 [40] &quot;ENSG00000033327&quot; &quot;ENSG00000037241&quot; &quot;ENSG00000037474&quot;
 [43] &quot;ENSG00000040341&quot; &quot;ENSG00000040933&quot; &quot;ENSG00000041357&quot;
 [46] &quot;ENSG00000046653&quot; &quot;ENSG00000047346&quot; &quot;ENSG00000047621&quot;
 [49] &quot;ENSG00000047932&quot; &quot;ENSG00000048140&quot; &quot;ENSG00000049167&quot;
 [52] &quot;ENSG00000049239&quot; &quot;ENSG00000049245&quot; &quot;ENSG00000049883&quot;
 [55] &quot;ENSG00000050748&quot; &quot;ENSG00000051825&quot; &quot;ENSG00000052126&quot;
 [58] &quot;ENSG00000053747&quot; &quot;ENSG00000055070&quot; &quot;ENSG00000056097&quot;
 [61] &quot;ENSG00000057019&quot; &quot;ENSG00000060762&quot; &quot;ENSG00000061676&quot;
 [64] &quot;ENSG00000062485&quot; &quot;ENSG00000065150&quot; &quot;ENSG00000065833&quot;
 [67] &quot;ENSG00000067057&quot; &quot;ENSG00000069248&quot; &quot;ENSG00000069275&quot;
 [70] &quot;ENSG00000069329&quot; &quot;ENSG00000069345&quot; &quot;ENSG00000069424&quot;
 [73] &quot;ENSG00000070061&quot; &quot;ENSG00000070159&quot; &quot;ENSG00000070404&quot;
 [76] &quot;ENSG00000070718&quot; &quot;ENSG00000070778&quot; &quot;ENSG00000072694&quot;
 [79] &quot;ENSG00000072954&quot; &quot;ENSG00000073464&quot; &quot;ENSG00000073711&quot;
 [82] &quot;ENSG00000073712&quot; &quot;ENSG00000073910&quot; &quot;ENSG00000074054&quot;
 [85] &quot;ENSG00000075188&quot; &quot;ENSG00000075223&quot; &quot;ENSG00000075568&quot;
 [88] &quot;ENSG00000075945&quot; &quot;ENSG00000076248&quot; &quot;ENSG00000076555&quot;
 [91] &quot;ENSG00000077380&quot; &quot;ENSG00000077549&quot; &quot;ENSG00000077721&quot;
 [94] &quot;ENSG00000078674&quot; &quot;ENSG00000079156&quot; &quot;ENSG00000079277&quot;
 [97] &quot;ENSG00000079308&quot; &quot;ENSG00000079332&quot; &quot;ENSG00000079785&quot;
[100] &quot;ENSG00000082213&quot; &quot;ENSG00000082701&quot; &quot;ENSG00000082781&quot;
[103] &quot;ENSG00000083720&quot; &quot;ENSG00000083799&quot; &quot;ENSG00000084090&quot;
[106] &quot;ENSG00000084463&quot; &quot;ENSG00000084764&quot; &quot;ENSG00000085117&quot;
[109] &quot;ENSG00000085377&quot; &quot;ENSG00000085415&quot; &quot;ENSG00000085491&quot;
[112] &quot;ENSG00000085662&quot; &quot;ENSG00000086189&quot; &quot;ENSG00000086200&quot;
[115] &quot;ENSG00000086289&quot; &quot;ENSG00000086666&quot; &quot;ENSG00000087074&quot;
[118] &quot;ENSG00000087470&quot; &quot;ENSG00000088256&quot; &quot;ENSG00000088356&quot;
[121] &quot;ENSG00000088888&quot; &quot;ENSG00000089335&quot; &quot;ENSG00000089876&quot;
[124] &quot;ENSG00000090615&quot; &quot;ENSG00000091140&quot; &quot;ENSG00000091527&quot;
[127] &quot;ENSG00000092203&quot; &quot;ENSG00000092929&quot; &quot;ENSG00000095321&quot;
[130] &quot;ENSG00000096696&quot; &quot;ENSG00000096968&quot; &quot;ENSG00000099246&quot;
[133] &quot;ENSG00000099284&quot; &quot;ENSG00000099968&quot; &quot;ENSG00000100196&quot;
[136] &quot;ENSG00000100216&quot; &quot;ENSG00000100281&quot; &quot;ENSG00000100354&quot;
[139] &quot;ENSG00000100519&quot; &quot;ENSG00000100567&quot; &quot;ENSG00000100605&quot;
[142] &quot;ENSG00000100711&quot; &quot;ENSG00000100749&quot; &quot;ENSG00000100784&quot;
[145] &quot;ENSG00000100916&quot; &quot;ENSG00000100934&quot; &quot;ENSG00000100994&quot;
[148] &quot;ENSG00000101190&quot; &quot;ENSG00000101333&quot; &quot;ENSG00000101413&quot;
[151] &quot;ENSG00000101452&quot; &quot;ENSG00000101464&quot; &quot;ENSG00000101546&quot;
[154] &quot;ENSG00000101928&quot; &quot;ENSG00000101986&quot; &quot;ENSG00000102007&quot;
[157] &quot;ENSG00000102054&quot; &quot;ENSG00000102144&quot; &quot;ENSG00000102181&quot;
[160] &quot;ENSG00000102241&quot; &quot;ENSG00000102878&quot; &quot;ENSG00000102893&quot;
[163] &quot;ENSG00000102900&quot; &quot;ENSG00000102901&quot; &quot;ENSG00000103005&quot;
[166] &quot;ENSG00000103319&quot; &quot;ENSG00000103423&quot; &quot;ENSG00000103429&quot;
[169] &quot;ENSG00000103707&quot; &quot;ENSG00000103966&quot; &quot;ENSG00000104131&quot;
[172] &quot;ENSG00000104331&quot; &quot;ENSG00000104341&quot; &quot;ENSG00000104442&quot;
[175] &quot;ENSG00000104635&quot; &quot;ENSG00000104671&quot; &quot;ENSG00000105132&quot;
[178] &quot;ENSG00000105220&quot; &quot;ENSG00000105290&quot; &quot;ENSG00000105357&quot;
[181] &quot;ENSG00000105429&quot; &quot;ENSG00000105552&quot; &quot;ENSG00000105649&quot;
[184] &quot;ENSG00000105677&quot; &quot;ENSG00000105849&quot; &quot;ENSG00000105866&quot;
[187] &quot;ENSG00000105879&quot; &quot;ENSG00000105993&quot; &quot;ENSG00000106052&quot;
[190] &quot;ENSG00000106078&quot; &quot;ENSG00000106153&quot; &quot;ENSG00000106244&quot;
[193] &quot;ENSG00000106524&quot; &quot;ENSG00000106591&quot; &quot;ENSG00000106682&quot;
[196] &quot;ENSG00000107020&quot; &quot;ENSG00000107295&quot; &quot;ENSG00000107518&quot;
[199] &quot;ENSG00000107581&quot; &quot;ENSG00000107937&quot; &quot;ENSG00000107960&quot;
[202] &quot;ENSG00000108010&quot; &quot;ENSG00000108061&quot; &quot;ENSG00000108094&quot;
[205] &quot;ENSG00000108179&quot; &quot;ENSG00000108264&quot; &quot;ENSG00000108443&quot;
[208] &quot;ENSG00000108821&quot; &quot;ENSG00000108861&quot; &quot;ENSG00000108953&quot;
[211] &quot;ENSG00000109685&quot; &quot;ENSG00000109738&quot; &quot;ENSG00000110422&quot;
[214] &quot;ENSG00000110429&quot; &quot;ENSG00000110651&quot; &quot;ENSG00000110696&quot;
[217] &quot;ENSG00000110841&quot; &quot;ENSG00000110871&quot; &quot;ENSG00000110888&quot;
[220] &quot;ENSG00000111142&quot; &quot;ENSG00000111371&quot; &quot;ENSG00000111450&quot;
[223] &quot;ENSG00000111652&quot; &quot;ENSG00000111716&quot; &quot;ENSG00000111726&quot;
[226] &quot;ENSG00000111850&quot; &quot;ENSG00000111875&quot; &quot;ENSG00000111880&quot;
[229] &quot;ENSG00000111911&quot; &quot;ENSG00000112130&quot; &quot;ENSG00000112293&quot;
[232] &quot;ENSG00000112305&quot; &quot;ENSG00000112335&quot; &quot;ENSG00000112339&quot;
[235] &quot;ENSG00000112378&quot; &quot;ENSG00000112530&quot; &quot;ENSG00000112782&quot;
[238] &quot;ENSG00000112855&quot; &quot;ENSG00000112992&quot; &quot;ENSG00000112996&quot;
[241] &quot;ENSG00000113013&quot; &quot;ENSG00000113068&quot; &quot;ENSG00000113360&quot;
[244] &quot;ENSG00000113504&quot; &quot;ENSG00000113578&quot; &quot;ENSG00000113595&quot;
[247] &quot;ENSG00000113621&quot; &quot;ENSG00000113638&quot; &quot;ENSG00000113712&quot;
[250] &quot;ENSG00000113851&quot; &quot;ENSG00000114013&quot; &quot;ENSG00000114062&quot;
[253] &quot;ENSG00000114098&quot; &quot;ENSG00000114302&quot; &quot;ENSG00000114316&quot;
[256] &quot;ENSG00000114405&quot; &quot;ENSG00000114861&quot; &quot;ENSG00000114999&quot;
[259] &quot;ENSG00000115073&quot; &quot;ENSG00000115233&quot; &quot;ENSG00000115295&quot;
[262] &quot;ENSG00000115520&quot; &quot;ENSG00000115540&quot; &quot;ENSG00000115561&quot;
[265] &quot;ENSG00000115806&quot; &quot;ENSG00000115944&quot; &quot;ENSG00000116199&quot;
[268] &quot;ENSG00000116670&quot; &quot;ENSG00000117114&quot; &quot;ENSG00000117143&quot;
[271] &quot;ENSG00000118418&quot; &quot;ENSG00000118579&quot; &quot;ENSG00000118596&quot;
[274] &quot;ENSG00000119185&quot; &quot;ENSG00000119318&quot; &quot;ENSG00000119326&quot;
[277] &quot;ENSG00000119328&quot; &quot;ENSG00000119471&quot; &quot;ENSG00000119638&quot;
[280] &quot;ENSG00000119689&quot; &quot;ENSG00000119699&quot; &quot;ENSG00000119812&quot;
[283] &quot;ENSG00000120029&quot; &quot;ENSG00000120327&quot; &quot;ENSG00000120333&quot;
[286] &quot;ENSG00000120509&quot; &quot;ENSG00000120647&quot; &quot;ENSG00000120705&quot;
[289] &quot;ENSG00000120727&quot; &quot;ENSG00000121057&quot; &quot;ENSG00000121316&quot;
[292] &quot;ENSG00000121413&quot; &quot;ENSG00000121417&quot; &quot;ENSG00000121671&quot;
[295] &quot;ENSG00000121897&quot; &quot;ENSG00000121903&quot; &quot;ENSG00000121940&quot;
[298] &quot;ENSG00000122574&quot; &quot;ENSG00000122873&quot; &quot;ENSG00000123095&quot;
[301] &quot;ENSG00000123106&quot; &quot;ENSG00000123124&quot; &quot;ENSG00000123146&quot;
[304] &quot;ENSG00000123178&quot; &quot;ENSG00000124006&quot; &quot;ENSG00000124207&quot;
[307] &quot;ENSG00000124257&quot; &quot;ENSG00000124406&quot; &quot;ENSG00000124422&quot;
[310] &quot;ENSG00000124486&quot; &quot;ENSG00000124532&quot; &quot;ENSG00000124535&quot;
[313] &quot;ENSG00000125388&quot; &quot;ENSG00000125827&quot; &quot;ENSG00000125977&quot;
[316] &quot;ENSG00000126261&quot; &quot;ENSG00000126351&quot; &quot;ENSG00000126581&quot;
[319] &quot;ENSG00000126705&quot; &quot;ENSG00000126777&quot; &quot;ENSG00000126947&quot;
[322] &quot;ENSG00000126953&quot; &quot;ENSG00000127804&quot; &quot;ENSG00000127922&quot;
[325] &quot;ENSG00000128245&quot; &quot;ENSG00000128254&quot; &quot;ENSG00000128335&quot;
[328] &quot;ENSG00000128463&quot; &quot;ENSG00000128609&quot; &quot;ENSG00000128654&quot;
[331] &quot;ENSG00000128708&quot; &quot;ENSG00000128944&quot; &quot;ENSG00000129270&quot;
[334] &quot;ENSG00000129465&quot; &quot;ENSG00000129625&quot; &quot;ENSG00000129691&quot;
[337] &quot;ENSG00000130024&quot; &quot;ENSG00000130347&quot; &quot;ENSG00000130413&quot;
[340] &quot;ENSG00000130529&quot; &quot;ENSG00000130559&quot; &quot;ENSG00000130638&quot;
[343] &quot;ENSG00000130703&quot; &quot;ENSG00000130779&quot; &quot;ENSG00000130818&quot;
[346] &quot;ENSG00000130940&quot; &quot;ENSG00000130956&quot; &quot;ENSG00000131508&quot;
[349] &quot;ENSG00000131725&quot; &quot;ENSG00000131788&quot; &quot;ENSG00000131791&quot;
[352] &quot;ENSG00000131844&quot; &quot;ENSG00000131871&quot; &quot;ENSG00000131966&quot;
[355] &quot;ENSG00000132300&quot; &quot;ENSG00000132305&quot; &quot;ENSG00000132359&quot;
[358] &quot;ENSG00000132383&quot; &quot;ENSG00000132432&quot; &quot;ENSG00000132463&quot;
[361] &quot;ENSG00000132535&quot; &quot;ENSG00000132570&quot; &quot;ENSG00000132604&quot;
[364] &quot;ENSG00000132676&quot; &quot;ENSG00000132763&quot; &quot;ENSG00000132912&quot;
[367] &quot;ENSG00000132953&quot; &quot;ENSG00000132964&quot; &quot;ENSG00000133111&quot;
[370] &quot;ENSG00000133302&quot; &quot;ENSG00000133393&quot; &quot;ENSG00000133606&quot;
[373] &quot;ENSG00000133627&quot; &quot;ENSG00000133731&quot; &quot;ENSG00000134014&quot;
[376] &quot;ENSG00000134077&quot; &quot;ENSG00000134253&quot; &quot;ENSG00000134375&quot;
[379] &quot;ENSG00000134717&quot; &quot;ENSG00000134755&quot; &quot;ENSG00000134982&quot;
[382] &quot;ENSG00000135040&quot; &quot;ENSG00000135049&quot; &quot;ENSG00000135083&quot;
[385] &quot;ENSG00000135249&quot; &quot;ENSG00000135250&quot; &quot;ENSG00000135336&quot;
[388] &quot;ENSG00000135341&quot; &quot;ENSG00000135655&quot; &quot;ENSG00000135709&quot;
[391] &quot;ENSG00000135723&quot; &quot;ENSG00000135778&quot; &quot;ENSG00000135919&quot;
[394] &quot;ENSG00000135932&quot; &quot;ENSG00000136143&quot; &quot;ENSG00000136161&quot;
[397] &quot;ENSG00000136504&quot; &quot;ENSG00000136521&quot; &quot;ENSG00000136536&quot;
[400] &quot;ENSG00000136628&quot; &quot;ENSG00000136643&quot; &quot;ENSG00000136653&quot;
[403] &quot;ENSG00000136758&quot; &quot;ENSG00000136783&quot; &quot;ENSG00000136891&quot;
[406] &quot;ENSG00000137055&quot; &quot;ENSG00000137200&quot; &quot;ENSG00000137210&quot;
[409] &quot;ENSG00000137815&quot; &quot;ENSG00000137941&quot; &quot;ENSG00000138032&quot;
[412] &quot;ENSG00000138095&quot; &quot;ENSG00000138138&quot; &quot;ENSG00000138293&quot;
[415] &quot;ENSG00000138442&quot; &quot;ENSG00000138463&quot; &quot;ENSG00000138641&quot;
[418] &quot;ENSG00000138663&quot; &quot;ENSG00000138768&quot; &quot;ENSG00000138785&quot;
[421] &quot;ENSG00000139173&quot; &quot;ENSG00000139174&quot; &quot;ENSG00000139233&quot;
[424] &quot;ENSG00000139436&quot; &quot;ENSG00000139617&quot; &quot;ENSG00000139679&quot;
[427] &quot;ENSG00000139719&quot; &quot;ENSG00000139842&quot; &quot;ENSG00000139977&quot;
[430] &quot;ENSG00000140299&quot; &quot;ENSG00000140307&quot; &quot;ENSG00000140386&quot;
[433] &quot;ENSG00000140403&quot; &quot;ENSG00000140526&quot; &quot;ENSG00000140543&quot;
[436] &quot;ENSG00000140548&quot; &quot;ENSG00000140577&quot; &quot;ENSG00000141219&quot;
[439] &quot;ENSG00000141337&quot; &quot;ENSG00000141429&quot; &quot;ENSG00000141582&quot;
[442] &quot;ENSG00000142002&quot; &quot;ENSG00000142082&quot; &quot;ENSG00000142208&quot;
[445] &quot;ENSG00000142453&quot; &quot;ENSG00000142552&quot; &quot;ENSG00000142856&quot;
[448] &quot;ENSG00000143179&quot; &quot;ENSG00000143324&quot; &quot;ENSG00000143337&quot;
[451] &quot;ENSG00000143374&quot; &quot;ENSG00000143641&quot; &quot;ENSG00000143653&quot;
[454] &quot;ENSG00000143740&quot; &quot;ENSG00000143742&quot; &quot;ENSG00000143761&quot;
[457] &quot;ENSG00000143801&quot; &quot;ENSG00000143847&quot; &quot;ENSG00000143862&quot;
[460] &quot;ENSG00000144161&quot; &quot;ENSG00000144306&quot; &quot;ENSG00000144320&quot;
[463] &quot;ENSG00000144566&quot; &quot;ENSG00000144674&quot; &quot;ENSG00000144791&quot;
[466] &quot;ENSG00000144824&quot; &quot;ENSG00000144935&quot; &quot;ENSG00000144959&quot;
[469] &quot;ENSG00000145335&quot; &quot;ENSG00000145365&quot; &quot;ENSG00000145675&quot;
[472] &quot;ENSG00000145743&quot; &quot;ENSG00000145868&quot; &quot;ENSG00000145936&quot;
[475] &quot;ENSG00000146592&quot; &quot;ENSG00000146729&quot; &quot;ENSG00000147586&quot;
[478] &quot;ENSG00000148187&quot; &quot;ENSG00000148411&quot; &quot;ENSG00000149485&quot;
[481] &quot;ENSG00000149639&quot; &quot;ENSG00000150768&quot; &quot;ENSG00000150773&quot;
[484] &quot;ENSG00000151093&quot; &quot;ENSG00000151376&quot; &quot;ENSG00000151413&quot;
[487] &quot;ENSG00000151748&quot; &quot;ENSG00000152256&quot; &quot;ENSG00000152268&quot;
[490] &quot;ENSG00000152700&quot; &quot;ENSG00000153015&quot; &quot;ENSG00000153956&quot;
[493] &quot;ENSG00000154001&quot; &quot;ENSG00000154065&quot; &quot;ENSG00000154080&quot;
[496] &quot;ENSG00000154114&quot; &quot;ENSG00000154153&quot; &quot;ENSG00000154310&quot;
[499] &quot;ENSG00000154813&quot; &quot;ENSG00000155313&quot; &quot;ENSG00000155393&quot;
[502] &quot;ENSG00000155755&quot; &quot;ENSG00000155959&quot; &quot;ENSG00000155975&quot;
[505] &quot;ENSG00000155984&quot; &quot;ENSG00000156256&quot; &quot;ENSG00000156261&quot;
[508] &quot;ENSG00000156642&quot; &quot;ENSG00000157350&quot; &quot;ENSG00000157500&quot;
[511] &quot;ENSG00000157540&quot; &quot;ENSG00000158019&quot; &quot;ENSG00000158286&quot;
[514] &quot;ENSG00000158528&quot; &quot;ENSG00000158714&quot; &quot;ENSG00000158863&quot;
[517] &quot;ENSG00000159110&quot; &quot;ENSG00000160305&quot; &quot;ENSG00000160883&quot;
[520] &quot;ENSG00000161013&quot; &quot;ENSG00000161217&quot; &quot;ENSG00000161267&quot;
[523] &quot;ENSG00000161298&quot; &quot;ENSG00000161642&quot; &quot;ENSG00000162139&quot;
[526] &quot;ENSG00000162409&quot; &quot;ENSG00000162434&quot; &quot;ENSG00000162437&quot;
[529] &quot;ENSG00000162512&quot; &quot;ENSG00000162616&quot; &quot;ENSG00000162623&quot;
[532] &quot;ENSG00000162688&quot; &quot;ENSG00000162885&quot; &quot;ENSG00000162894&quot;
[535] &quot;ENSG00000163026&quot; &quot;ENSG00000163069&quot; &quot;ENSG00000163166&quot;
[538] &quot;ENSG00000163171&quot; &quot;ENSG00000163539&quot; &quot;ENSG00000163607&quot;
[541] &quot;ENSG00000163626&quot; &quot;ENSG00000163636&quot; &quot;ENSG00000163710&quot;
[544] &quot;ENSG00000163785&quot; &quot;ENSG00000163806&quot; &quot;ENSG00000163918&quot;
[547] &quot;ENSG00000163935&quot; &quot;ENSG00000163995&quot; &quot;ENSG00000164022&quot;
[550] &quot;ENSG00000164074&quot; &quot;ENSG00000164209&quot; &quot;ENSG00000164221&quot;
[553] &quot;ENSG00000164292&quot; &quot;ENSG00000164307&quot; &quot;ENSG00000164338&quot;
[556] &quot;ENSG00000164398&quot; &quot;ENSG00000164649&quot; &quot;ENSG00000164659&quot;
[559] &quot;ENSG00000164754&quot; &quot;ENSG00000164949&quot; &quot;ENSG00000164951&quot;
[562] &quot;ENSG00000164975&quot; &quot;ENSG00000165244&quot; &quot;ENSG00000165338&quot;
[565] &quot;ENSG00000165495&quot; &quot;ENSG00000165516&quot; &quot;ENSG00000165526&quot;
[568] &quot;ENSG00000165660&quot; &quot;ENSG00000165672&quot; &quot;ENSG00000165675&quot;
[571] &quot;ENSG00000165678&quot; &quot;ENSG00000165914&quot; &quot;ENSG00000166478&quot;
[574] &quot;ENSG00000166483&quot; &quot;ENSG00000166750&quot; &quot;ENSG00000166788&quot;
[577] &quot;ENSG00000166946&quot; &quot;ENSG00000167191&quot; &quot;ENSG00000167325&quot;
[580] &quot;ENSG00000167555&quot; &quot;ENSG00000167635&quot; &quot;ENSG00000167721&quot;
[583] &quot;ENSG00000167964&quot; &quot;ENSG00000168118&quot; &quot;ENSG00000168273&quot;
[586] &quot;ENSG00000168288&quot; &quot;ENSG00000168461&quot; &quot;ENSG00000168522&quot;
[589] &quot;ENSG00000168615&quot; &quot;ENSG00000168724&quot; &quot;ENSG00000168803&quot;
[592] &quot;ENSG00000168938&quot; &quot;ENSG00000168958&quot; &quot;ENSG00000169032&quot;
[595] &quot;ENSG00000169085&quot; &quot;ENSG00000169116&quot; &quot;ENSG00000169251&quot;
[598] &quot;ENSG00000169288&quot; &quot;ENSG00000169372&quot; &quot;ENSG00000169499&quot;
[601] &quot;ENSG00000169515&quot; &quot;ENSG00000169851&quot; &quot;ENSG00000169925&quot;
[604] &quot;ENSG00000170027&quot; &quot;ENSG00000170142&quot; &quot;ENSG00000170153&quot;
[607] &quot;ENSG00000170365&quot; &quot;ENSG00000170542&quot; &quot;ENSG00000170606&quot;
[610] &quot;ENSG00000170653&quot; &quot;ENSG00000170791&quot; &quot;ENSG00000170846&quot;
[613] &quot;ENSG00000171130&quot; &quot;ENSG00000171262&quot; &quot;ENSG00000171316&quot;
[616] &quot;ENSG00000171649&quot; &quot;ENSG00000171819&quot; &quot;ENSG00000171861&quot;
[619] &quot;ENSG00000171865&quot; &quot;ENSG00000171885&quot; &quot;ENSG00000172172&quot;
[622] &quot;ENSG00000172765&quot; &quot;ENSG00000172915&quot; &quot;ENSG00000172939&quot;
[625] &quot;ENSG00000172985&quot; &quot;ENSG00000173011&quot; &quot;ENSG00000173041&quot;
[628] &quot;ENSG00000173614&quot; &quot;ENSG00000173681&quot; &quot;ENSG00000173692&quot;
[631] &quot;ENSG00000173706&quot; &quot;ENSG00000173744&quot; &quot;ENSG00000174032&quot;
[634] &quot;ENSG00000174111&quot; &quot;ENSG00000174282&quot; &quot;ENSG00000174374&quot;
[637] &quot;ENSG00000174446&quot; &quot;ENSG00000174684&quot; &quot;ENSG00000174842&quot;
[640] &quot;ENSG00000175048&quot; &quot;ENSG00000175166&quot; &quot;ENSG00000175193&quot;
[643] &quot;ENSG00000175203&quot; &quot;ENSG00000175582&quot; &quot;ENSG00000176563&quot;
[646] &quot;ENSG00000176623&quot; &quot;ENSG00000176842&quot; &quot;ENSG00000177427&quot;
[649] &quot;ENSG00000177692&quot; &quot;ENSG00000177889&quot; &quot;ENSG00000178802&quot;
[652] &quot;ENSG00000178988&quot; &quot;ENSG00000179476&quot; &quot;ENSG00000179912&quot;
[655] &quot;ENSG00000180228&quot; &quot;ENSG00000180263&quot; &quot;ENSG00000180304&quot;
[658] &quot;ENSG00000180329&quot; &quot;ENSG00000180336&quot; &quot;ENSG00000180628&quot;
[661] &quot;ENSG00000180891&quot; &quot;ENSG00000180957&quot; &quot;ENSG00000181381&quot;
[664] &quot;ENSG00000181754&quot; &quot;ENSG00000181873&quot; &quot;ENSG00000182831&quot;
[667] &quot;ENSG00000182923&quot; &quot;ENSG00000183605&quot; &quot;ENSG00000183726&quot;
[670] &quot;ENSG00000183955&quot; &quot;ENSG00000184185&quot; &quot;ENSG00000184313&quot;
[673] &quot;ENSG00000184349&quot; &quot;ENSG00000184481&quot; &quot;ENSG00000184787&quot;
[676] &quot;ENSG00000184792&quot; &quot;ENSG00000184831&quot; &quot;ENSG00000184900&quot;
[679] &quot;ENSG00000185055&quot; &quot;ENSG00000185274&quot; &quot;ENSG00000185345&quot;
[682] &quot;ENSG00000185728&quot; &quot;ENSG00000185973&quot; &quot;ENSG00000186205&quot;
[685] &quot;ENSG00000186462&quot; &quot;ENSG00000186532&quot; &quot;ENSG00000186591&quot;
[688] &quot;ENSG00000186687&quot; &quot;ENSG00000187098&quot; &quot;ENSG00000187109&quot;
[691] &quot;ENSG00000187325&quot; &quot;ENSG00000187626&quot; &quot;ENSG00000187676&quot;
[694] &quot;ENSG00000187955&quot; &quot;ENSG00000188004&quot; &quot;ENSG00000188283&quot;
[697] &quot;ENSG00000188352&quot; &quot;ENSG00000188739&quot; &quot;ENSG00000188931&quot;
[700] &quot;ENSG00000189221&quot; &quot;ENSG00000196208&quot; &quot;ENSG00000196437&quot;
[703] &quot;ENSG00000196526&quot; &quot;ENSG00000196642&quot; &quot;ENSG00000196743&quot;
[706] &quot;ENSG00000196792&quot; &quot;ENSG00000196850&quot; &quot;ENSG00000196968&quot;
[709] &quot;ENSG00000197050&quot; &quot;ENSG00000197128&quot; &quot;ENSG00000197429&quot;
[712] &quot;ENSG00000197496&quot; &quot;ENSG00000197563&quot; &quot;ENSG00000197747&quot;
[715] &quot;ENSG00000197780&quot; &quot;ENSG00000197885&quot; &quot;ENSG00000197912&quot;
[718] &quot;ENSG00000197928&quot; &quot;ENSG00000198015&quot; &quot;ENSG00000198060&quot;
[721] &quot;ENSG00000198252&quot; &quot;ENSG00000198585&quot; &quot;ENSG00000198668&quot;
[724] &quot;ENSG00000198689&quot; &quot;ENSG00000198836&quot; &quot;ENSG00000198894&quot;
[727] &quot;ENSG00000198898&quot; &quot;ENSG00000198952&quot; &quot;ENSG00000198954&quot;
[730] &quot;ENSG00000203666&quot; &quot;ENSG00000203667&quot; &quot;ENSG00000204396&quot;
[733] &quot;ENSG00000204568&quot; &quot;ENSG00000204634&quot; &quot;ENSG00000205089&quot;
[736] &quot;ENSG00000205339&quot; &quot;ENSG00000205476&quot; &quot;ENSG00000205707&quot;
[739] &quot;ENSG00000206561&quot; &quot;ENSG00000206989&quot; &quot;ENSG00000207741&quot;
[742] &quot;ENSG00000212127&quot; &quot;ENSG00000213585&quot; &quot;ENSG00000213600&quot;
[745] &quot;ENSG00000213639&quot; &quot;ENSG00000213702&quot; &quot;ENSG00000213762&quot;
[748] &quot;ENSG00000214194&quot; &quot;ENSG00000214456&quot; &quot;ENSG00000215018&quot;
[751] &quot;ENSG00000215712&quot; &quot;ENSG00000219758&quot; &quot;ENSG00000221420&quot;
[754] &quot;ENSG00000221914&quot; &quot;ENSG00000223134&quot; &quot;ENSG00000223960&quot;
[757] &quot;ENSG00000224078&quot; &quot;ENSG00000225712&quot; &quot;ENSG00000225996&quot;
[760] &quot;ENSG00000227372&quot; &quot;ENSG00000227719&quot; &quot;ENSG00000228275&quot;
[763] &quot;ENSG00000228606&quot; &quot;ENSG00000233511&quot; &quot;ENSG00000234055&quot;
[766] &quot;ENSG00000234983&quot; &quot;ENSG00000235477&quot; &quot;ENSG00000235865&quot;
[769] &quot;ENSG00000235945&quot; &quot;ENSG00000238622&quot; &quot;ENSG00000238942&quot;
[772] &quot;ENSG00000239483&quot; &quot;ENSG00000239593&quot; &quot;ENSG00000240291&quot;
[775] &quot;ENSG00000240404&quot; &quot;ENSG00000242261&quot; &quot;ENSG00000243156&quot;
[778] &quot;ENSG00000243317&quot; &quot;ENSG00000244202&quot; &quot;ENSG00000244286&quot;
[781] &quot;ENSG00000244687&quot; &quot;ENSG00000245060&quot; &quot;ENSG00000247614&quot;
[784] &quot;ENSG00000247828&quot; &quot;ENSG00000248092&quot; &quot;ENSG00000249035&quot;
[787] &quot;ENSG00000249087&quot; &quot;ENSG00000249669&quot; &quot;ENSG00000249758&quot;
[790] &quot;ENSG00000249823&quot; &quot;ENSG00000249947&quot; &quot;ENSG00000250569&quot;
[793] &quot;ENSG00000250902&quot; &quot;ENSG00000250988&quot; &quot;ENSG00000251369&quot;
[796] &quot;ENSG00000251540&quot; &quot;ENSG00000251775&quot; &quot;ENSG00000252115&quot;
[799] &quot;ENSG00000253645&quot; &quot;ENSG00000254099&quot; &quot;ENSG00000254693&quot;
[802] &quot;ENSG00000255054&quot; &quot;ENSG00000255092&quot; &quot;ENSG00000255302&quot;
[805] &quot;ENSG00000255639&quot; &quot;ENSG00000255729&quot; &quot;ENSG00000256206&quot;
[808] &quot;ENSG00000256646&quot; &quot;ENSG00000257076&quot; &quot;ENSG00000257151&quot;
[811] &quot;ENSG00000257225&quot; &quot;ENSG00000257303&quot; &quot;ENSG00000257337&quot;
[814] &quot;ENSG00000257365&quot; &quot;ENSG00000258017&quot; &quot;ENSG00000258465&quot;
[817] &quot;ENSG00000258604&quot; &quot;ENSG00000258629&quot; &quot;ENSG00000258830&quot;
[820] &quot;ENSG00000258950&quot; &quot;ENSG00000259007&quot;</code></pre>
<pre class="r"><code># Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Heart_kidney &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Heart compared to Kidney (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_kidney)

# ENSG00000004468

big_list &lt;- union(mylist[[&quot;Chimpanzee&quot;]], mylist[[&quot;Rhesus Macaque&quot;]])
mylist[[&quot;Human&quot;]][1:200] %in% big_list</code></pre>
<pre><code>  [1] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
 [12]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE
 [23]  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE
 [34]  TRUE FALSE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE FALSE  TRUE
 [45]  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
 [56]  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE FALSE  TRUE FALSE
 [67]  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
 [78]  TRUE  TRUE  TRUE  TRUE FALSE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE
 [89]  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[100]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[111] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE
[122]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[133]  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[144]  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE
[155]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE
[166] FALSE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE FALSE  TRUE
[177]  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE
[188]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE
[199]  TRUE  TRUE</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(mylist[[&quot;Human&quot;]][176], rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
make_exp_df &lt;- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
make_exp_df[,2] &lt;- as.character(make_exp_df[,2])

#levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus macaque&quot;)

make_exp_df$species[make_exp_df$species ==&quot;1&quot;] &lt;- &quot;Chimpanzee&quot;
make_exp_df$species[make_exp_df$species ==&quot;2&quot;] &lt;- &quot;Human&quot;
make_exp_df$species[make_exp_df$species ==&quot;3&quot;] &lt;-  &quot;Rhesus macaque&quot;


colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) +  theme_bw() + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;PQBP1 (ENSG00000102103)&quot;) 




exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000006740&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
#make_exp_df &lt;- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
make_exp_df[,2] &lt;- as.character(make_exp_df[,2])

#levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus macaque&quot;)

make_exp_df$species[make_exp_df$species ==&quot;1&quot;] &lt;- &quot;Chimpanzee&quot;
make_exp_df$species[make_exp_df$species ==&quot;2&quot;] &lt;- &quot;Human&quot;
make_exp_df$species[make_exp_df$species ==&quot;3&quot;] &lt;-  &quot;Rhesus macaque&quot;


colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) +  theme_bw() + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;TNNT2 (ENSG00000118194)&quot;) 





exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000118194&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
#make_exp_df &lt;- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
make_exp_df[,2] &lt;- as.character(make_exp_df[,2])

#levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus macaque&quot;)

make_exp_df$species[make_exp_df$species ==&quot;1&quot;] &lt;- &quot;Chimpanzee&quot;
make_exp_df$species[make_exp_df$species ==&quot;2&quot;] &lt;- &quot;Human&quot;
make_exp_df$species[make_exp_df$species ==&quot;3&quot;] &lt;-  &quot;Rhesus macaque&quot;


colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) +  theme_bw() + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;TNNT2 (ENSG00000118194)&quot;) 




# Heart versus lung

mylist &lt;- list()

test &lt;- &quot;H_HeartvLu&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_HeartvLu&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_HeartvLu&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Heart_lung &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Heart compared to Lung (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_lung)


# Liver versus lung

mylist &lt;- list()

test &lt;- &quot;H_LivLu&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_LivLu&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_LivLu&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Liver_lung &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Liver compared to Lung (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_lung)


# Liver versus kidney

mylist &lt;- list()

test &lt;- &quot;H_LivK&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_LivK&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_LivK&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Liver_kidney &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Liver compared to Kidney (FSR 1%)&quot;,  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_kidney)


# Lung versus Kidney

mylist &lt;- list()

test &lt;- &quot;H_LuvK&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_LuvK&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_LuvK&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Lung_kidney &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Lung compared to Kidney (FSR 1%)&quot;,  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Lung_kidney)</code></pre>
</div>
</div>
<div id="upregulated-and-downregulated-pairwise-figure-supplementary-figure" class="section level2">
<h2>Upregulated and downregulated pairwise figure (Supplementary Figure)</h2>
<pre class="r"><code>upreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]
  return(upreg_two)
}

FSR_level &lt;- 0.01
upreg_human_heart_kidney &lt;- upreg_pairwise_comp(&quot;H_HeartvK&quot;, FSR_level)
upreg_chimp_heart_kidney &lt;- upreg_pairwise_comp(&quot;C_HeartvK&quot;, FSR_level)
upreg_rhesus_heart_kidney &lt;- upreg_pairwise_comp(&quot;R_HeartvK&quot;, FSR_level)

upreg_human_heart_liver &lt;- upreg_pairwise_comp(&quot;H_HeartvLi&quot;, FSR_level)
upreg_chimp_heart_liver &lt;- upreg_pairwise_comp(&quot;C_HeartvLi&quot;, FSR_level)
upreg_rhesus_heart_liver &lt;- upreg_pairwise_comp(&quot;R_HeartvLi&quot;, FSR_level)

upreg_human_heart_lung &lt;- upreg_pairwise_comp(&quot;H_HeartvLu&quot;, FSR_level)
upreg_chimp_heart_lung &lt;- upreg_pairwise_comp(&quot;C_HeartvLu&quot;, FSR_level)
upreg_rhesus_heart_lung &lt;- upreg_pairwise_comp(&quot;R_HeartvLu&quot;, FSR_level)

upreg_human_liver_lung &lt;- upreg_pairwise_comp(&quot;H_LivLu&quot;, FSR_level)
upreg_chimp_liver_lung &lt;- upreg_pairwise_comp(&quot;C_LivLu&quot;, FSR_level)
upreg_rhesus_liver_lung &lt;- upreg_pairwise_comp(&quot;R_LivLu&quot;, FSR_level)

upreg_human_liver_kidney &lt;- upreg_pairwise_comp(&quot;H_LivK&quot;, FSR_level)
upreg_chimp_liver_kidney &lt;- upreg_pairwise_comp(&quot;C_LivK&quot;, FSR_level)
upreg_rhesus_liver_kidney &lt;- upreg_pairwise_comp(&quot;R_LivK&quot;, FSR_level)

upreg_human_lung_kidney &lt;- upreg_pairwise_comp(&quot;H_LuvK&quot;, FSR_level)
upreg_chimp_lung_kidney &lt;- upreg_pairwise_comp(&quot;C_LuvK&quot;, FSR_level)
upreg_rhesus_lung_kidney &lt;- upreg_pairwise_comp(&quot;R_LuvK&quot;, FSR_level)


downreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]
  return(upreg_two)
}


downreg_human_heart_kidney &lt;- downreg_pairwise_comp(&quot;H_HeartvK&quot;, FSR_level)
downreg_chimp_heart_kidney &lt;- downreg_pairwise_comp(&quot;C_HeartvK&quot;, FSR_level)
downreg_rhesus_heart_kidney &lt;- downreg_pairwise_comp(&quot;R_HeartvK&quot;, FSR_level)

downreg_human_heart_liver &lt;- downreg_pairwise_comp(&quot;H_HeartvLi&quot;, FSR_level)
downreg_chimp_heart_liver &lt;- downreg_pairwise_comp(&quot;C_HeartvLi&quot;, FSR_level)
downreg_rhesus_heart_liver &lt;- downreg_pairwise_comp(&quot;R_HeartvLi&quot;, FSR_level)

downreg_human_heart_lung &lt;- downreg_pairwise_comp(&quot;H_HeartvLu&quot;, FSR_level)
downreg_chimp_heart_lung &lt;- downreg_pairwise_comp(&quot;C_HeartvLu&quot;, FSR_level)
downreg_rhesus_heart_lung &lt;- downreg_pairwise_comp(&quot;R_HeartvLu&quot;, FSR_level)

downreg_human_liver_lung &lt;- downreg_pairwise_comp(&quot;H_LivLu&quot;, FSR_level)
downreg_chimp_liver_lung &lt;- downreg_pairwise_comp(&quot;C_LivLu&quot;, FSR_level)
downreg_rhesus_liver_lung &lt;- downreg_pairwise_comp(&quot;R_LivLu&quot;, FSR_level)

downreg_human_liver_kidney &lt;- downreg_pairwise_comp(&quot;H_LivK&quot;, FSR_level)
downreg_chimp_liver_kidney &lt;- downreg_pairwise_comp(&quot;C_LivK&quot;, FSR_level)
downreg_rhesus_liver_kidney &lt;- downreg_pairwise_comp(&quot;R_LivK&quot;, FSR_level)

downreg_human_lung_kidney &lt;- downreg_pairwise_comp(&quot;H_LuvK&quot;, FSR_level)
downreg_chimp_lung_kidney &lt;- downreg_pairwise_comp(&quot;C_LuvK&quot;, FSR_level)
downreg_rhesus_lung_kidney &lt;- downreg_pairwise_comp(&quot;R_LuvK&quot;, FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram &lt;- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist &lt;- list()
mylist[[&quot;Hu&quot;]] &lt;- upreg_human_tissue
mylist[[&quot;Ch&quot;]] &lt;- upreg_chimp_tissue
mylist[[&quot;Rh&quot;]] &lt;- upreg_rhesus_tissue


# Make Venn Diagram
#dev.off()
Heart_specific_up &lt;- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;, cat.dist = 0.05)
grid.draw(Heart_specific_up)

}



svg(&quot;../data/Upregulated_heart_liver.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_liver, upreg_chimp_heart_liver, upreg_rhesus_heart_liver, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_heart_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_lung, upreg_chimp_heart_lung, upreg_rhesus_heart_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_heart_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_kidney, upreg_chimp_heart_kidney, upreg_rhesus_heart_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_liver_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_lung, upreg_chimp_liver_lung, upreg_rhesus_liver_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_liver_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_kidney, upreg_chimp_liver_kidney, upreg_rhesus_liver_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_lung_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung_kidney, upreg_chimp_lung_kidney, upreg_rhesus_lung_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_heart_liver.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_liver, downreg_chimp_heart_liver, downreg_rhesus_heart_liver, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_heart_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_lung, downreg_chimp_heart_lung, downreg_rhesus_heart_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_heart_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_kidney, downreg_chimp_heart_kidney, downreg_rhesus_heart_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_liver_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_lung, downreg_chimp_liver_lung, downreg_rhesus_liver_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_liver_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_kidney, downreg_chimp_liver_kidney, downreg_rhesus_liver_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_lung_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung_kidney, downreg_chimp_lung_kidney, downreg_rhesus_lung_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>#title &lt;- ggdraw() + draw_label(&quot;Pairwise DE genes (FSR 1%)&quot;, fontface=&#39;bold&#39;, size = 20)
p1 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_heart_liver.svg&quot;)+draw_figure_label(&quot;4A.&quot;, size = 16, fontface = &quot;bold&quot;)
p2 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_heart_liver.svg&quot;)+draw_figure_label(&quot;4B.&quot;, size = 16, fontface = &quot;bold&quot;)

p3 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_heart_lung.svg&quot;)+draw_figure_label(&quot;4C.&quot;, size = 16, fontface = &quot;bold&quot;)
p4 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_heart_lung.svg&quot;)+draw_figure_label(&quot;4D.&quot;, size = 16, fontface = &quot;bold&quot;)

p5 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_heart_kidney.svg&quot;)+draw_figure_label(&quot;4E.&quot;, size = 16, fontface = &quot;bold&quot;)
p6 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_heart_kidney.svg&quot;)+draw_figure_label(&quot;4F.&quot;, size = 16, fontface = &quot;bold&quot;)

p7 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_liver_lung.svg&quot;)+draw_figure_label(&quot;4G.&quot;, size = 16, fontface = &quot;bold&quot;)
p8 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_liver_lung.svg&quot;)+draw_figure_label(&quot;4H.&quot;, size = 16, fontface = &quot;bold&quot;)

p9 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_liver_kidney.svg&quot;)+draw_figure_label(&quot;4I.&quot;, size = 16, fontface = &quot;bold&quot;)
p10 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_liver_kidney.svg&quot;)+draw_figure_label(&quot;4J.&quot;, size = 16, fontface = &quot;bold&quot;)

p11 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_lung_kidney.svg&quot;)+draw_figure_label(&quot;4K.&quot;, size = 16, fontface = &quot;bold&quot;)
p12 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_lung_kidney.svg&quot;)+draw_figure_label(&quot;4L.&quot;, size = 16, fontface = &quot;bold&quot;)


#p2 &lt;- ggdraw()+draw_image(&quot;Methyl_heatmap.tiff&quot;)
eight_plots &lt;- plot_grid(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, ncol = 4)
plot_supp_fig &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot(&quot;../data/test.png&quot;, plot_supp_fig,
          ncol = 4, # we&#39;re saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

#save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Supp_figures/Supplementary_Figure_4_common_pairwise.png&quot;, plot_supp_fig,
#          ncol = 4, # we&#39;re saving a grid plot of 2 columns
#          nrow = 3, # and 2 rows
#          # each individual subplot should have an aspect ratio of 1.3
#          base_aspect_ratio = 0.75
#          )

upreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FSR_level &amp; results[[test]]$logFC &gt; 0]
  return(upreg_two)
}

FSR_level &lt;- 0.01
upreg_human_heart_kidney &lt;- upreg_pairwise_comp(&quot;H_HeartvK&quot;, FSR_level)
upreg_chimp_heart_kidney &lt;- upreg_pairwise_comp(&quot;C_HeartvK&quot;, FSR_level)
upreg_rhesus_heart_kidney &lt;- upreg_pairwise_comp(&quot;R_HeartvK&quot;, FSR_level)

upreg_human_heart_liver &lt;- upreg_pairwise_comp(&quot;H_HeartvLi&quot;, FSR_level)
upreg_chimp_heart_liver &lt;- upreg_pairwise_comp(&quot;C_HeartvLi&quot;, FSR_level)
upreg_rhesus_heart_liver &lt;- upreg_pairwise_comp(&quot;R_HeartvLi&quot;, FSR_level)

upreg_human_heart_lung &lt;- upreg_pairwise_comp(&quot;H_HeartvLu&quot;, FSR_level)
upreg_chimp_heart_lung &lt;- upreg_pairwise_comp(&quot;C_HeartvLu&quot;, FSR_level)
upreg_rhesus_heart_lung &lt;- upreg_pairwise_comp(&quot;R_HeartvLu&quot;, FSR_level)

upreg_human_liver_lung &lt;- upreg_pairwise_comp(&quot;H_LivLu&quot;, FSR_level)
upreg_chimp_liver_lung &lt;- upreg_pairwise_comp(&quot;C_LivLu&quot;, FSR_level)
upreg_rhesus_liver_lung &lt;- upreg_pairwise_comp(&quot;R_LivLu&quot;, FSR_level)

upreg_human_liver_kidney &lt;- upreg_pairwise_comp(&quot;H_LivK&quot;, FSR_level)
upreg_chimp_liver_kidney &lt;- upreg_pairwise_comp(&quot;C_LivK&quot;, FSR_level)
upreg_rhesus_liver_kidney &lt;- upreg_pairwise_comp(&quot;R_LivK&quot;, FSR_level)

upreg_human_lung_kidney &lt;- upreg_pairwise_comp(&quot;H_LuvK&quot;, FSR_level)
upreg_chimp_lung_kidney &lt;- upreg_pairwise_comp(&quot;C_LuvK&quot;, FSR_level)
upreg_rhesus_lung_kidney &lt;- upreg_pairwise_comp(&quot;R_LuvK&quot;, FSR_level)


downreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FSR_level &amp; results[[test]]$logFC &lt; 0]
  return(upreg_two)
}


downreg_human_heart_kidney &lt;- downreg_pairwise_comp(&quot;H_HeartvK&quot;, FSR_level)
downreg_chimp_heart_kidney &lt;- downreg_pairwise_comp(&quot;C_HeartvK&quot;, FSR_level)
downreg_rhesus_heart_kidney &lt;- downreg_pairwise_comp(&quot;R_HeartvK&quot;, FSR_level)

downreg_human_heart_liver &lt;- downreg_pairwise_comp(&quot;H_HeartvLi&quot;, FSR_level)
downreg_chimp_heart_liver &lt;- downreg_pairwise_comp(&quot;C_HeartvLi&quot;, FSR_level)
downreg_rhesus_heart_liver &lt;- downreg_pairwise_comp(&quot;R_HeartvLi&quot;, FSR_level)

downreg_human_heart_lung &lt;- downreg_pairwise_comp(&quot;H_HeartvLu&quot;, FSR_level)
downreg_chimp_heart_lung &lt;- downreg_pairwise_comp(&quot;C_HeartvLu&quot;, FSR_level)
downreg_rhesus_heart_lung &lt;- downreg_pairwise_comp(&quot;R_HeartvLu&quot;, FSR_level)

downreg_human_liver_lung &lt;- downreg_pairwise_comp(&quot;H_LivLu&quot;, FSR_level)
downreg_chimp_liver_lung &lt;- downreg_pairwise_comp(&quot;C_LivLu&quot;, FSR_level)
downreg_rhesus_liver_lung &lt;- downreg_pairwise_comp(&quot;R_LivLu&quot;, FSR_level)

downreg_human_liver_kidney &lt;- downreg_pairwise_comp(&quot;H_LivK&quot;, FSR_level)
downreg_chimp_liver_kidney &lt;- downreg_pairwise_comp(&quot;C_LivK&quot;, FSR_level)
downreg_rhesus_liver_kidney &lt;- downreg_pairwise_comp(&quot;R_LivK&quot;, FSR_level)

downreg_human_lung_kidney &lt;- downreg_pairwise_comp(&quot;H_LuvK&quot;, FSR_level)
downreg_chimp_lung_kidney &lt;- downreg_pairwise_comp(&quot;C_LuvK&quot;, FSR_level)
downreg_rhesus_lung_kidney &lt;- downreg_pairwise_comp(&quot;R_LuvK&quot;, FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram &lt;- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist &lt;- list()
mylist[[&quot;Hu&quot;]] &lt;- upreg_human_tissue
mylist[[&quot;Ch&quot;]] &lt;- upreg_chimp_tissue
mylist[[&quot;Rh&quot;]] &lt;- upreg_rhesus_tissue


# Make Venn Diagram
#dev.off()
Heart_specific_up &lt;- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;, cat.dist = 0.05)
grid.draw(Heart_specific_up)

}



svg(&quot;../data/Upregulated_heart_liver.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_liver, upreg_chimp_heart_liver, upreg_rhesus_heart_liver, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_heart_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_lung, upreg_chimp_heart_lung, upreg_rhesus_heart_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_heart_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_kidney, upreg_chimp_heart_kidney, upreg_rhesus_heart_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_liver_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_lung, upreg_chimp_liver_lung, upreg_rhesus_liver_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_liver_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_kidney, upreg_chimp_liver_kidney, upreg_rhesus_liver_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_lung_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung_kidney, upreg_chimp_lung_kidney, upreg_rhesus_lung_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_heart_liver.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_liver, downreg_chimp_heart_liver, downreg_rhesus_heart_liver, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_heart_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_lung, downreg_chimp_heart_lung, downreg_rhesus_heart_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_heart_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_kidney, downreg_chimp_heart_kidney, downreg_rhesus_heart_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_liver_lung.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_lung, downreg_chimp_liver_lung, downreg_rhesus_liver_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_liver_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_kidney, downreg_chimp_liver_kidney, downreg_rhesus_liver_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_lung_kidney.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung_kidney, downreg_chimp_lung_kidney, downreg_rhesus_lung_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>#title &lt;- ggdraw() + draw_label(&quot;Pairwise DE genes (FSR 1%)&quot;, fontface=&#39;bold&#39;, size = 20)
p1 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_heart_liver.svg&quot;)+draw_figure_label(&quot;4A.&quot;, size = 16, fontface = &quot;bold&quot;)
p2 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_heart_liver.svg&quot;)+draw_figure_label(&quot;4B.&quot;, size = 16, fontface = &quot;bold&quot;)

p3 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_heart_lung.svg&quot;)+draw_figure_label(&quot;4C.&quot;, size = 16, fontface = &quot;bold&quot;)
p4 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_heart_lung.svg&quot;)+draw_figure_label(&quot;4D.&quot;, size = 16, fontface = &quot;bold&quot;)

p5 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_heart_kidney.svg&quot;)+draw_figure_label(&quot;4E.&quot;, size = 16, fontface = &quot;bold&quot;)
p6 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_heart_kidney.svg&quot;)+draw_figure_label(&quot;4F.&quot;, size = 16, fontface = &quot;bold&quot;)

p7 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_liver_lung.svg&quot;)+draw_figure_label(&quot;4G.&quot;, size = 16, fontface = &quot;bold&quot;)
p8 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_liver_lung.svg&quot;)+draw_figure_label(&quot;4H.&quot;, size = 16, fontface = &quot;bold&quot;)

p9 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_liver_kidney.svg&quot;)+draw_figure_label(&quot;4I.&quot;, size = 16, fontface = &quot;bold&quot;)
p10 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_liver_kidney.svg&quot;)+draw_figure_label(&quot;4J.&quot;, size = 16, fontface = &quot;bold&quot;)

p11 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_lung_kidney.svg&quot;)+draw_figure_label(&quot;4K.&quot;, size = 16, fontface = &quot;bold&quot;)
p12 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_lung_kidney.svg&quot;)+draw_figure_label(&quot;4L.&quot;, size = 16, fontface = &quot;bold&quot;)


#p2 &lt;- ggdraw()+draw_image(&quot;Methyl_heatmap.tiff&quot;)
eight_plots &lt;- plot_grid(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, ncol = 4)
plot_supp_fig &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot(&quot;../data/test.png&quot;, plot_supp_fig,
          ncol = 4, # we&#39;re saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Supp_figures/Supplementary_Figure_4_common_pairwise.png&quot;, plot_supp_fig,
          ncol = 4, # we&#39;re saving a grid plot of 2 columns
         nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )</code></pre>
</div>
<div id="upregulated-and-downregulated-pairwise-species-figure-supplementary-figure" class="section level2">
<h2>Upregulated and downregulated pairwise species figure (Supplementary Figure)</h2>
<pre class="r"><code># With FDR
upreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FSR_level &amp; results[[test]]$logFC &gt; 0]
  return(upreg_two)
}

FSR_level &lt;- 0.05
upreg_hc_heart &lt;- upreg_pairwise_comp(&quot;HvC_Heart&quot;, FSR_level)
upreg_hr_heart &lt;- upreg_pairwise_comp(&quot;HvR_Heart&quot;, FSR_level)
upreg_cr_heart &lt;- upreg_pairwise_comp(&quot;CvR_Heart&quot;, FSR_level)

upreg_hc_kidney &lt;- upreg_pairwise_comp(&quot;HvC_Kidney&quot;, FSR_level)
upreg_hr_kidney &lt;- upreg_pairwise_comp(&quot;HvR_Kidney&quot;, FSR_level)
upreg_cr_kidney &lt;- upreg_pairwise_comp(&quot;CvR_Kidney&quot;, FSR_level)

upreg_hc_liver &lt;- upreg_pairwise_comp(&quot;HvC_Liver&quot;, FSR_level)
upreg_hr_liver &lt;- upreg_pairwise_comp(&quot;HvR_Liver&quot;, FSR_level)
upreg_cr_liver &lt;- upreg_pairwise_comp(&quot;CvR_Liver&quot;, FSR_level)

upreg_hc_lung &lt;- upreg_pairwise_comp(&quot;HvC_Lung&quot;, FSR_level)
upreg_hr_lung &lt;- upreg_pairwise_comp(&quot;HvR_Lung&quot;, FSR_level)
upreg_cr_lung &lt;- upreg_pairwise_comp(&quot;CvR_Lung&quot;, FSR_level)


downreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FSR_level &amp; results[[test]]$logFC &lt; 0]
  return(upreg_two)
}


downreg_hc_heart &lt;- downreg_pairwise_comp(&quot;HvC_Heart&quot;, FSR_level)
downreg_hr_heart &lt;- downreg_pairwise_comp(&quot;HvR_Heart&quot;, FSR_level)
downreg_cr_heart &lt;- downreg_pairwise_comp(&quot;CvR_Heart&quot;, FSR_level)

downreg_hc_kidney &lt;- downreg_pairwise_comp(&quot;HvC_Kidney&quot;, FSR_level)
downreg_hr_kidney &lt;- downreg_pairwise_comp(&quot;HvR_Kidney&quot;, FSR_level)
downreg_cr_kidney &lt;- downreg_pairwise_comp(&quot;CvR_Kidney&quot;, FSR_level)

downreg_hc_liver &lt;- downreg_pairwise_comp(&quot;HvC_Liver&quot;, FSR_level)
downreg_hr_liver &lt;- downreg_pairwise_comp(&quot;HvR_Liver&quot;, FSR_level)
downreg_cr_liver &lt;- downreg_pairwise_comp(&quot;CvR_Liver&quot;, FSR_level)

downreg_hc_lung &lt;- downreg_pairwise_comp(&quot;HvC_Lung&quot;, FSR_level)
downreg_hr_lung &lt;- downreg_pairwise_comp(&quot;HvR_Lung&quot;, FSR_level)
downreg_cr_lung &lt;- downreg_pairwise_comp(&quot;CvR_Lung&quot;, FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram &lt;- function(heart, liver, lung, kidney, name_file){
mylist &lt;- list()
mylist[[&quot;Heart&quot;]] &lt;- heart
mylist[[&quot;Liver&quot;]] &lt;- liver
mylist[[&quot;Lung&quot;]] &lt;- lung
mylist[[&quot;Kidney&quot;]] &lt;- kidney

# Make Venn Diagram
#dev.off()
Heart_specific_up &lt;- venn.diagram(mylist,  filename = NULL, fill = pal[1:4], main.cex = 3, cat.cex = 2.7, cex=2.7, lty=1, height=2000, width=2000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;)
grid.draw(Heart_specific_up)

}



svg(&quot;../data/Upregulated_hc.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_hc_heart, upreg_hc_liver, upreg_hc_lung, upreg_hc_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_hr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_hr_heart, upreg_hr_liver, upreg_hr_lung, upreg_hr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_cr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_cr_heart, upreg_cr_liver, upreg_cr_lung, upreg_cr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_hc.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_hc_heart, downreg_hc_liver, downreg_hc_lung, downreg_hc_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_hr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_hr_heart, downreg_hr_liver, downreg_hr_lung, downreg_hr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_cr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_cr_heart, downreg_cr_liver, downreg_cr_lung, downreg_cr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>#title &lt;- ggdraw() + draw_label(&quot;Pairwise DE genes (FSR 1%)&quot;, fontface=&#39;bold&#39;, size = 20)
p1 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_hc.svg&quot;)+draw_figure_label(&quot;5A.&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)
p2 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_hc.svg&quot;)+draw_figure_label(&quot;5B.&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)

p3 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_hr.svg&quot;)+draw_figure_label(&quot;5C.&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)
p4 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_hr.svg&quot;)+draw_figure_label(&quot;5D.&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)

p5 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_cr.svg&quot;)+draw_figure_label(&quot;5E.&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)
p6 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_cr.svg&quot;)+draw_figure_label(&quot;5F.&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)



#p2 &lt;- ggdraw()+draw_image(&quot;Methyl_heatmap.tiff&quot;)
eight_plots &lt;- plot_grid(p1, p2, p3, p4, p5, p6, ncol = 2)
plot_supp_fig &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot(&quot;../data/test.png&quot;, plot_supp_fig,
          ncol = 3, # we&#39;re saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Supp_figures/Supplementary_Figure_5_species_pairwise.png&quot;, plot_supp_fig,
          ncol = 4, # we&#39;re saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

# With FSR
upreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]
  return(upreg_two)
}

FSR_level &lt;- 0.05
upreg_hc_heart &lt;- upreg_pairwise_comp(&quot;HvC_Heart&quot;, FSR_level)
upreg_hr_heart &lt;- upreg_pairwise_comp(&quot;HvR_Heart&quot;, FSR_level)
upreg_cr_heart &lt;- upreg_pairwise_comp(&quot;CvR_Heart&quot;, FSR_level)

upreg_hc_kidney &lt;- upreg_pairwise_comp(&quot;HvC_Kidney&quot;, FSR_level)
upreg_hr_kidney &lt;- upreg_pairwise_comp(&quot;HvR_Kidney&quot;, FSR_level)
upreg_cr_kidney &lt;- upreg_pairwise_comp(&quot;CvR_Kidney&quot;, FSR_level)

upreg_hc_liver &lt;- upreg_pairwise_comp(&quot;HvC_Liver&quot;, FSR_level)
upreg_hr_liver &lt;- upreg_pairwise_comp(&quot;HvR_Liver&quot;, FSR_level)
upreg_cr_liver &lt;- upreg_pairwise_comp(&quot;CvR_Liver&quot;, FSR_level)

upreg_hc_lung &lt;- upreg_pairwise_comp(&quot;HvC_Lung&quot;, FSR_level)
upreg_hr_lung &lt;- upreg_pairwise_comp(&quot;HvR_Lung&quot;, FSR_level)
upreg_cr_lung &lt;- upreg_pairwise_comp(&quot;CvR_Lung&quot;, FSR_level)


downreg_pairwise_comp &lt;- function(testname, FSR_level){
  test &lt;- testname
  upreg_two &lt;- (results[[test]]$genes)[results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]
  return(upreg_two)
}


downreg_hc_heart &lt;- downreg_pairwise_comp(&quot;HvC_Heart&quot;, FSR_level)
downreg_hr_heart &lt;- downreg_pairwise_comp(&quot;HvR_Heart&quot;, FSR_level)
downreg_cr_heart &lt;- downreg_pairwise_comp(&quot;CvR_Heart&quot;, FSR_level)

downreg_hc_kidney &lt;- downreg_pairwise_comp(&quot;HvC_Kidney&quot;, FSR_level)
downreg_hr_kidney &lt;- downreg_pairwise_comp(&quot;HvR_Kidney&quot;, FSR_level)
downreg_cr_kidney &lt;- downreg_pairwise_comp(&quot;CvR_Kidney&quot;, FSR_level)

downreg_hc_liver &lt;- downreg_pairwise_comp(&quot;HvC_Liver&quot;, FSR_level)
downreg_hr_liver &lt;- downreg_pairwise_comp(&quot;HvR_Liver&quot;, FSR_level)
downreg_cr_liver &lt;- downreg_pairwise_comp(&quot;CvR_Liver&quot;, FSR_level)

downreg_hc_lung &lt;- downreg_pairwise_comp(&quot;HvC_Lung&quot;, FSR_level)
downreg_hr_lung &lt;- downreg_pairwise_comp(&quot;HvR_Lung&quot;, FSR_level)
downreg_cr_lung &lt;- downreg_pairwise_comp(&quot;CvR_Lung&quot;, FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram &lt;- function(heart, liver, lung, kidney, name_file){
mylist &lt;- list()
mylist[[&quot;Heart&quot;]] &lt;- heart
mylist[[&quot;Liver&quot;]] &lt;- liver
mylist[[&quot;Lung&quot;]] &lt;- lung
mylist[[&quot;Kidney&quot;]] &lt;- kidney

# Make Venn Diagram
#dev.off()
Heart_specific_up &lt;- venn.diagram(mylist,  filename = NULL, fill = pal[1:4], main.cex = 3, cat.cex = 2.7, cex=2.7, lty=1, height=2000, width=2000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;)
grid.draw(Heart_specific_up)

}



svg(&quot;../data/Upregulated_hc.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_hc_heart, upreg_hc_liver, upreg_hc_lung, upreg_hc_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_hr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_hr_heart, upreg_hr_liver, upreg_hr_lung, upreg_hr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Upregulated_cr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_cr_heart, upreg_cr_liver, upreg_cr_lung, upreg_cr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_hc.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_hc_heart, downreg_hc_liver, downreg_hc_lung, downreg_hc_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_hr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_hr_heart, downreg_hr_liver, downreg_hr_lung, downreg_hr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;../data/Downregulated_cr.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_cr_heart, downreg_cr_liver, downreg_cr_lung, downreg_cr_kidney, &quot;Upregulated heart rel. to liver&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>#title &lt;- ggdraw() + draw_label(&quot;Pairwise DE genes (FSR 1%)&quot;, fontface=&#39;bold&#39;, size = 20)
p1 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_hc.svg&quot;)+draw_figure_label(&quot;A. Upregulated Human v Chimpanzee&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)
p2 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_hc.svg&quot;)+draw_figure_label(&quot;B. Downregulated Human v Chimpanzee&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)

p3 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_hr.svg&quot;)+draw_figure_label(&quot;C. Upregulated Human v Rhesus&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)
p4 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_hr.svg&quot;)+draw_figure_label(&quot;D. Downregulated Human v Rhesus&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)

p5 &lt;- ggdraw()+draw_image(&quot;../data/Upregulated_cr.svg&quot;)+draw_figure_label(&quot;E. Upregulated Chimpanzee v Rhesus&quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)
p6 &lt;- ggdraw()+draw_image(&quot;../data/Downregulated_cr.svg&quot;)+draw_figure_label(&quot;F. Downregulated Chimpanzee v Rhesus &quot;, size = 15, fontface = &quot;bold&quot;, position = &quot;top.left&quot;)



#p2 &lt;- ggdraw()+draw_image(&quot;Methyl_heatmap.tiff&quot;)
eight_plots &lt;- plot_grid(p1, p2, p3, p4, p5, p6, ncol = 2)
plot_supp_fig &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot(&quot;../data/test.png&quot;, plot_supp_fig,
          ncol = 3, # we&#39;re saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )</code></pre>
<div id="downregulated" class="section level3">
<h3>Downregulated</h3>
<pre class="r"><code># Set false sign rate
FDR_level &lt;- 0.01
FSR_level &lt;- 0.01

# Heart versus kidney

mylist &lt;- list()

test &lt;- &quot;H_HeartvK&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;C_HeartvK&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;R_HeartvK&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Heart_kidney &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Downregulated in Heart compared to Kidney (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_kidney)


# Heart versus liver

mylist &lt;- list()

test &lt;- &quot;H_HeartvLi&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_HeartvLi&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_HeartvLi&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Heart_liver &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Heart compared to Liver (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_liver)


# Heart versus lung

mylist &lt;- list()

test &lt;- &quot;H_HeartvLu&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_HeartvLu&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_HeartvLu&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Heart_lung &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Heart compared to Lung (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_lung)


# Liver versus lung

mylist &lt;- list()

test &lt;- &quot;H_LivLu&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_LivLu&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_LivLu&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Liver_lung &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Liver compared to Lung (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_lung)


# Liver versus kidney

mylist &lt;- list()

test &lt;- &quot;H_LivK&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_LivK&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_LivK&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Liver_kidney &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Liver compared to Kidney (FSR 1%)&quot;,  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_kidney)


# Lung versus Kidney

mylist &lt;- list()

test &lt;- &quot;H_LuvK&quot;
mylist[[&quot;Human&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;C_LuvK&quot;
mylist[[&quot;Chimpanzee&quot;]] &lt;-   (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;R_LuvK&quot;
mylist[[&quot;Rhesus Macaque&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Lung_kidney &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Lung compared to Kidney (FSR 1%)&quot;,  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Lung_kidney)</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000000938&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species &lt;- as.factor(make_exp_df$species)

keep &lt;- c(1,2,5,6,9,10,13,14,17,20,21,24,25,28,29, 32, 33, 36, 37, 40, 41, 44, 45)
make_exp_df &lt;- make_exp_df[keep,]

levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus&quot;)
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Species&quot;, &quot;Tissue&quot;)
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;TNNT2 (ENSG00000104951)&quot;) </code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="find-where-the-genes-are-upregulated-between-species" class="section level3">
<h3>Find where the genes are upregulated between species</h3>
<pre class="r"><code># Set false sign rate

FSR_level &lt;- 0.01

  # DE between humans and chimpanzees

mylist &lt;- list()

test &lt;- &quot;HvC_Liver&quot;
mylist[[&quot;Liver&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;HvC_Lung&quot;
mylist[[&quot;Lung&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;HvC_Heart&quot;
mylist[[&quot;Heart&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;HvC_Kidney&quot;
mylist[[&quot;Kidney&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Human_chimp_up &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Humans compared to Chimps (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_chimp_up)


  # DE between humans and rhesus
mylist &lt;- list()
test &lt;- &quot;HvR_Liver&quot;
mylist[[&quot;Liver&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;HvR_Lung&quot;
mylist[[&quot;Lung&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;HvR_Heart&quot;
mylist[[&quot;Heart&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;HvR_Kidney&quot;
mylist[[&quot;Kidney&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Human_rhesus_up &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Humans compared to Rhesus (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_rhesus_up)



  # DE between chimpanzees and rhesus
mylist &lt;- list()
test &lt;- &quot;CvR_Liver&quot;
mylist[[&quot;Liver&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;CvR_Lung&quot;
mylist[[&quot;Lung&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;CvR_Heart&quot;
mylist[[&quot;Heart&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

test &lt;- &quot;CvR_Kidney&quot;
mylist[[&quot;Kidney&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &gt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Chimp_rhesus_up &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Upregulated in Chimps relative to Rhesus (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Chimp_rhesus_up)</code></pre>
</div>
<div id="find-where-the-genes-are-downregulated-between-species" class="section level3">
<h3>Find where the genes are downregulated between species</h3>
<pre class="r"><code># Set false sign rate

FSR_level &lt;- 0.01

  # DE between humans and chimpanzees

mylist &lt;- list()

test &lt;- &quot;HvC_Liver&quot;
mylist[[&quot;Liver&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;HvC_Lung&quot;
mylist[[&quot;Lung&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;HvC_Heart&quot;
mylist[[&quot;Heart&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;HvC_Kidney&quot;
mylist[[&quot;Kidney&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Human_chimp_down &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Downregulated in Humans compared to Chimps (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_chimp_down)


  # DE between humans and rhesus
mylist &lt;- list()
test &lt;- &quot;HvR_Liver&quot;
mylist[[&quot;Liver&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;HvR_Lung&quot;
mylist[[&quot;Lung&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;HvR_Heart&quot;
mylist[[&quot;Heart&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;HvR_Kidney&quot;
mylist[[&quot;Kidney&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Human_rhesus_down &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Downregulated in Humans compared to Rhesus (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_rhesus_down)



  # DE between chimpanzees and rhesus
mylist &lt;- list()
test &lt;- &quot;CvR_Liver&quot;
mylist[[&quot;Liver&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;CvR_Lung&quot;
mylist[[&quot;Lung&quot;]] &lt;-  (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;CvR_Heart&quot;
mylist[[&quot;Heart&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

test &lt;- &quot;CvR_Kidney&quot;
mylist[[&quot;Kidney&quot;]] &lt;- (results[[test]]$genes)[results[[test]]$adj.P.Val &lt; FDR_level &amp; results[[test]]$svalue &lt; FSR_level &amp; results[[test]]$beta_est &lt; 0]

# Make Venn Diagram
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Chimp_rhesus_down &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Downregulated in Chimps relative to Rhesus (FSR 1%)&quot;, main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Chimp_rhesus_down)</code></pre>
</div>
</div>
<div id="find-genes-with-a-tissue-specific-direction-significant-and-upregulated-in-1-versus-2-3-4-but-not-significant-in-2-v-3-4-and-3-v-4-with-fsr-supplementary-tables" class="section level2">
<h2>Find genes with a tissue-specific direction (significant and upregulated in 1 versus 2, 3, 4 but not significant in 2 v 3, 4 and 3 v 4) with FSR (supplementary tables)</h2>
<pre class="r"><code># Combine the human ASH results
combined_human &lt;- cbind(results[[&quot;H_HeartvLi&quot;]]$svalue, results[[&quot;H_HeartvLi&quot;]]$beta_est, results[[&quot;H_HeartvLu&quot;]]$svalue, results[[&quot;H_HeartvLu&quot;]]$beta_est, results[[&quot;H_HeartvK&quot;]]$svalue, results[[&quot;H_HeartvK&quot;]]$beta_est, results[[&quot;H_LivLu&quot;]]$svalue, results[[&quot;H_LivLu&quot;]]$beta_est, results[[&quot;H_LivK&quot;]]$svalue, results[[&quot;H_LivK&quot;]]$beta_est, results[[&quot;H_LuvK&quot;]]$svalue, results[[&quot;H_LuvK&quot;]]$beta_est)


#combined_human &lt;- cbind(results[[&quot;H_HeartvLi&quot;]]$qvalue, results[[&quot;H_HeartvLi&quot;]]$beta_est, results[[&quot;H_HeartvLu&quot;]]$qvalue, results[[&quot;H_HeartvLu&quot;]]$beta_est, results[[&quot;H_HeartvK&quot;]]$qvalue, results[[&quot;H_HeartvK&quot;]]$beta_est, results[[&quot;H_LivLu&quot;]]$qvalue, results[[&quot;H_LivLu&quot;]]$beta_est, results[[&quot;H_LivK&quot;]]$qvalue, results[[&quot;H_LivK&quot;]]$beta_est, results[[&quot;H_LuvK&quot;]]$qvalue, results[[&quot;H_LuvK&quot;]]$beta_est)


dim(combined_human)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code>rownames(combined_human) &lt;- rownames(cpm_12184)
  
# Combine the chimp ASH results
combined_chimp &lt;- cbind(results[[&quot;C_HeartvLi&quot;]]$svalue, results[[&quot;C_HeartvLi&quot;]]$beta_est, results[[&quot;C_HeartvLu&quot;]]$svalue, results[[&quot;C_HeartvLu&quot;]]$beta_est, results[[&quot;C_HeartvK&quot;]]$svalue, results[[&quot;C_HeartvK&quot;]]$beta_est, results[[&quot;C_LivLu&quot;]]$svalue, results[[&quot;C_LivLu&quot;]]$beta_est, results[[&quot;C_LivK&quot;]]$svalue, results[[&quot;C_LivK&quot;]]$beta_est, results[[&quot;C_LuvK&quot;]]$svalue, results[[&quot;C_LuvK&quot;]]$beta_est)

#combined_chimp &lt;- cbind(results[[&quot;C_HeartvLi&quot;]]$qvalue, results[[&quot;C_HeartvLi&quot;]]$beta_est, results[[&quot;C_HeartvLu&quot;]]$qvalue, results[[&quot;C_HeartvLu&quot;]]$beta_est, results[[&quot;C_HeartvK&quot;]]$qvalue, results[[&quot;C_HeartvK&quot;]]$beta_est, results[[&quot;C_LivLu&quot;]]$qvalue, results[[&quot;C_LivLu&quot;]]$beta_est, results[[&quot;C_LivK&quot;]]$qvalue, results[[&quot;C_LivK&quot;]]$beta_est, results[[&quot;C_LuvK&quot;]]$qvalue, results[[&quot;C_LuvK&quot;]]$beta_est)

dim(combined_chimp)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code>rownames(combined_chimp) &lt;- rownames(cpm_12184)

# Combine the rhesus ASH results
combined_rhesus &lt;- cbind(results[[&quot;R_HeartvLi&quot;]]$svalue, results[[&quot;R_HeartvLi&quot;]]$beta_est, results[[&quot;R_HeartvLu&quot;]]$svalue, results[[&quot;R_HeartvLu&quot;]]$beta_est, results[[&quot;R_HeartvK&quot;]]$svalue, results[[&quot;R_HeartvK&quot;]]$beta_est, results[[&quot;R_LivLu&quot;]]$svalue, results[[&quot;R_LivLu&quot;]]$beta_est, results[[&quot;R_LivK&quot;]]$svalue, results[[&quot;R_LivK&quot;]]$beta_est, results[[&quot;R_LuvK&quot;]]$svalue, results[[&quot;R_LuvK&quot;]]$beta_est)

#combined_rhesus &lt;- cbind(results[[&quot;R_HeartvLi&quot;]]$qvalue, results[[&quot;R_HeartvLi&quot;]]$beta_est, results[[&quot;R_HeartvLu&quot;]]$qvalue, results[[&quot;R_HeartvLu&quot;]]$beta_est, results[[&quot;R_HeartvK&quot;]]$qvalue, results[[&quot;R_HeartvK&quot;]]$beta_est, results[[&quot;R_LivLu&quot;]]$qvalue, results[[&quot;R_LivLu&quot;]]$beta_est, results[[&quot;R_LivK&quot;]]$qvalue, results[[&quot;R_LivK&quot;]]$beta_est, results[[&quot;R_LuvK&quot;]]$qvalue, results[[&quot;R_LuvK&quot;]]$beta_est)

dim(combined_rhesus)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code>rownames(combined_rhesus) &lt;- rownames(cpm_12184)

FSR_level &lt;- 0.01

# Upregulated consistently in one human tissue

upreg_human_heart &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &gt; 0 &amp; combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &gt; 0 &amp; combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &gt; 0 &amp; combined_human[,7] &gt; FSR_level &amp; combined_human[,9] &gt; FSR_level &amp;  combined_human[,11] &gt; FSR_level),])

upreg_human_kidney &lt;- as.data.frame(combined_human[which(combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &lt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &lt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &lt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,3] &gt; FSR_level &amp; combined_human[,7] &gt; FSR_level),])

upreg_human_liver &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &lt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &gt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &gt; 0 &amp; combined_human[,3] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp; combined_human[,11] &gt; FSR_level),])

upreg_human_lung &lt;- as.data.frame(combined_human[which(combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &lt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &lt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &gt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp; combined_human[,9] &gt; FSR_level),])

# Upregulated in only one human tissue

upreg_human_heart_only &lt;- setdiff(upreg_human_heart, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_lung)))
upreg_human_kidney_only &lt;- setdiff(upreg_human_kidney, (union(union(upreg_human_heart, upreg_human_liver), upreg_human_lung)))
upreg_human_liver_only &lt;- setdiff(upreg_human_liver, (union(union(upreg_human_kidney, upreg_human_heart), upreg_human_lung)))
upreg_human_lung_only &lt;- setdiff(upreg_human_lung, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_heart)))

nrow(upreg_human_heart)</code></pre>
<pre><code>[1] 390</code></pre>
<pre class="r"><code>nrow(upreg_human_kidney)</code></pre>
<pre><code>[1] 266</code></pre>
<pre class="r"><code>nrow(upreg_human_liver)</code></pre>
<pre><code>[1] 541</code></pre>
<pre class="r"><code>nrow(upreg_human_lung)</code></pre>
<pre><code>[1] 305</code></pre>
<pre class="r"><code>nrow(upreg_human_heart_only)</code></pre>
<pre><code>[1] 390</code></pre>
<pre class="r"><code>nrow(upreg_human_kidney_only)</code></pre>
<pre><code>[1] 266</code></pre>
<pre class="r"><code>nrow(upreg_human_liver_only)</code></pre>
<pre><code>[1] 541</code></pre>
<pre class="r"><code>nrow(upreg_human_lung_only)</code></pre>
<pre><code>[1] 305</code></pre>
<pre class="r"><code># Upregulated consistently in one chimp tissue

upreg_chimp_heart &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &gt; 0 &amp; combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &gt; 0 &amp; combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &gt; 0 &amp; combined_chimp[,7] &gt; FSR_level &amp; combined_chimp[,9] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

upreg_chimp_kidney &lt;- as.data.frame(combined_chimp[which(combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &lt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &lt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &lt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,3] &gt; FSR_level &amp;  combined_chimp[,7] &gt; FSR_level),])

upreg_chimp_liver &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &lt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &gt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &gt; 0 &amp; combined_chimp[,3] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

upreg_chimp_lung &lt;- as.data.frame(combined_chimp[which(combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &lt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &lt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &gt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,9] &gt; FSR_level),])

nrow(upreg_chimp_heart)</code></pre>
<pre><code>[1] 384</code></pre>
<pre class="r"><code>nrow(upreg_chimp_kidney)</code></pre>
<pre><code>[1] 152</code></pre>
<pre class="r"><code>nrow(upreg_chimp_liver)</code></pre>
<pre><code>[1] 546</code></pre>
<pre class="r"><code>nrow(upreg_chimp_lung)</code></pre>
<pre><code>[1] 464</code></pre>
<pre class="r"><code># Upregulated consistently in one rhesus tissue

upreg_rhesus_heart &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &gt; 0 &amp; combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &gt; 0 &amp; combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &gt; 0 &amp; combined_rhesus[,7] &gt; FSR_level &amp; combined_rhesus[,9] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

upreg_rhesus_kidney &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &lt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &lt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &lt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,3] &gt; FSR_level &amp;  combined_rhesus[,7] &gt; FSR_level),])

upreg_rhesus_liver &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &lt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &gt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &gt; 0 &amp; combined_rhesus[,3] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

upreg_rhesus_lung &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &lt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &lt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &gt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,9] &gt; FSR_level),])

nrow(upreg_rhesus_heart)</code></pre>
<pre><code>[1] 345</code></pre>
<pre class="r"><code>nrow(upreg_rhesus_kidney)</code></pre>
<pre><code>[1] 133</code></pre>
<pre class="r"><code>nrow(upreg_rhesus_liver)</code></pre>
<pre><code>[1] 416</code></pre>
<pre class="r"><code>nrow(upreg_rhesus_lung)</code></pre>
<pre><code>[1] 363</code></pre>
<pre class="r"><code># Downregulated consistently in one human tissue

downreg_human_heart &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &lt; 0 &amp; combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &lt; 0 &amp; combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &lt; 0 &amp;  combined_human[,7] &gt; FSR_level &amp; combined_human[,9] &gt; FSR_level &amp;  combined_human[,11] &gt; FSR_level),])

downreg_human_kidney &lt;- as.data.frame(combined_human[which(combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &gt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &gt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &gt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,3] &gt; FSR_level &amp;  combined_human[,7] &gt; FSR_level),])

downreg_human_liver &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &gt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &lt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &lt; 0 &amp; combined_human[,3] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp;  combined_human[,11] &gt; FSR_level),])

downreg_human_lung &lt;- as.data.frame(combined_human[which(combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &gt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &gt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &lt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp;  combined_human[,9] &gt; FSR_level),])

# Downregulated in only one human tissue

nrow(downreg_human_heart)</code></pre>
<pre><code>[1] 354</code></pre>
<pre class="r"><code>nrow(downreg_human_kidney)</code></pre>
<pre><code>[1] 141</code></pre>
<pre class="r"><code>nrow(downreg_human_liver)</code></pre>
<pre><code>[1] 324</code></pre>
<pre class="r"><code>nrow(downreg_human_lung)</code></pre>
<pre><code>[1] 234</code></pre>
<pre class="r"><code># Downregulated consistently in one chimp tissue

downreg_chimp_heart &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &lt; 0 &amp; combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &lt; 0 &amp; combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &lt; 0 &amp; combined_chimp[,7] &gt; FSR_level &amp; combined_chimp[,9] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

downreg_chimp_kidney &lt;- as.data.frame(combined_chimp[which(combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &gt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &gt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &gt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,3] &gt; FSR_level &amp;  combined_chimp[,7] &gt; FSR_level),])

downreg_chimp_liver &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &gt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &lt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &lt; 0 &amp; combined_chimp[,3] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

downreg_chimp_lung &lt;- as.data.frame(combined_chimp[which(combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &gt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &gt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &lt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,9] &gt; FSR_level),])

nrow(downreg_chimp_heart)</code></pre>
<pre><code>[1] 397</code></pre>
<pre class="r"><code>nrow(downreg_chimp_kidney)</code></pre>
<pre><code>[1] 90</code></pre>
<pre class="r"><code>nrow(downreg_chimp_liver)</code></pre>
<pre><code>[1] 301</code></pre>
<pre class="r"><code>nrow(downreg_chimp_lung)</code></pre>
<pre><code>[1] 228</code></pre>
<pre class="r"><code># Downregulated consistently in one rhesus tissue

downreg_rhesus_heart &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &lt; 0 &amp; combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &lt; 0 &amp; combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &lt; 0 &amp; combined_rhesus[,7] &gt; FSR_level &amp; combined_rhesus[,9] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

downreg_rhesus_kidney &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &gt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &gt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &gt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,3] &gt; FSR_level &amp;  combined_rhesus[,7] &gt; FSR_level),])

downreg_rhesus_liver &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &gt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &lt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &lt; 0 &amp; combined_rhesus[,3] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

downreg_rhesus_lung &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &gt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &gt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &lt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,9] &gt; FSR_level),])

nrow(downreg_rhesus_heart)</code></pre>
<pre><code>[1] 294</code></pre>
<pre class="r"><code>nrow(downreg_rhesus_kidney)</code></pre>
<pre><code>[1] 86</code></pre>
<pre class="r"><code>nrow(downreg_rhesus_liver)</code></pre>
<pre><code>[1] 202</code></pre>
<pre class="r"><code>nrow(downreg_rhesus_lung)</code></pre>
<pre><code>[1] 240</code></pre>
<pre class="r"><code># Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram &lt;- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist &lt;- list()
mylist[[&quot;Hu&quot;]] &lt;- rownames(upreg_human_tissue)
mylist[[&quot;Ch&quot;]] &lt;- rownames(upreg_chimp_tissue)
mylist[[&quot;Rh&quot;]] &lt;- rownames(upreg_rhesus_tissue)


# Make Venn Diagram
#dev.off()
Heart_specific_up &lt;- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;, cat.dist = 0.05)
grid.draw(Heart_specific_up)
}

svg(&quot;Upregulated_heart_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart, &quot;Upregulated in heart only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>intersect(intersect(upreg_human_heart, upreg_chimp_heart), upreg_rhesus_heart)</code></pre>
<pre><code>data frame with 0 columns and 0 rows</code></pre>
<pre class="r"><code>#dev.off()
#draw.triple.venn(323+183+626+158, 521+183+626+618, 158+626+618+1182, 183+626, 618+626, 158+626, 626,  main= &quot;Heart #versus kidney (hypomethylated)&quot;,  category = c(&quot;Human&quot;, &quot;Chimp&quot;, &quot;Rhesus macaque&quot;), height=3000, width=3000, cex=2 , #fill = pal[1:3], lty=1, cat.cex = 2, main.cex = 2)



svg(&quot;Upregulated_kidney_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Upregulated_liver_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Upregulated_lung_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_heart_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_kidney_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_liver_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_lung_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung, NULL)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>#title &lt;- ggdraw() + draw_label(&quot;Genes regulated in a tissue specific direction (FSR 1%)&quot;, fontface=&#39;bold&#39;)
p1 &lt;- ggdraw()+draw_image(&quot;Upregulated_heart_specific.svg&quot;) 
p2 &lt;- ggdraw()+draw_image(&quot;Downregulated_heart_specific.svg&quot;) 
p3 &lt;- ggdraw()+draw_image(&quot;Upregulated_liver_specific.svg&quot;)  
p4 &lt;- ggdraw()+draw_image(&quot;Downregulated_liver_specific.svg&quot;) 

p5 &lt;- ggdraw()+draw_image(&quot;Upregulated_lung_specific.svg&quot;)  
p6 &lt;- ggdraw()+draw_image(&quot;Downregulated_lung_specific.svg&quot;)  
p7 &lt;- ggdraw()+draw_image(&quot;Upregulated_kidney_specific.svg&quot;)  
p8 &lt;- ggdraw()+draw_image(&quot;Downregulated_kidney_specific.svg&quot;)  

eight_plots &lt;- plot_grid(p1, p2, p3, p4, p5, p6, p7, p8, labels = c(&quot;6A.&quot;, &quot;6B.&quot;, &quot;6C.&quot;, &quot;6D.&quot;, &quot;6E.&quot;, &quot;6F.&quot;, &quot;6G.&quot;, &quot;6H.&quot;), ncol = 2)
plot_fig2 &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))

save_plot(&quot;../data/test.png&quot;, plot_fig2,
           ncol = 2, # we&#39;re saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
           # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 0.75
           )

save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Figures/Figure_6_Tissue_specific.png&quot;, plot_fig2,
           ncol = 2, # we&#39;re saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
         # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 0.75
           )



# Find all conserved up/downregulated tissue specific

conserved_tissue_specific &lt;- function(human_tissue_name, chimp_tissue_name, rhesus_tissue_name){

  upreg_human_tissue &lt;- rownames(human_tissue_name)
  upreg_chimp_tissue &lt;- rownames(chimp_tissue_name)
  upreg_rhesus_tissue &lt;- rownames(rhesus_tissue_name)
  
all_three &lt;- intersect(intersect(upreg_human_tissue, upreg_chimp_tissue), upreg_rhesus_tissue)

return(all_three)
}

heart_upregulated &lt;- conserved_tissue_specific(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart)
write.csv(heart_upregulated, &quot;../data/specific_upreg_heart_genes.csv&quot;, row.names = FALSE)
heart_downregulated &lt;- conserved_tissue_specific(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart)
write.csv(heart_downregulated, &quot;../data/specific_downreg_heart_genes.csv&quot;, row.names = FALSE)

liver_upregulated &lt;- conserved_tissue_specific(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver)
write.csv(liver_upregulated, &quot;../data/specific_upreg_liver_genes.csv&quot;, row.names = FALSE)
liver_downregulated &lt;- conserved_tissue_specific(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver)
write.csv(liver_downregulated, &quot;../data/specific_downreg_liver_genes.csv&quot;, row.names = FALSE)

lung_upregulated &lt;- conserved_tissue_specific(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung)
write.csv(lung_upregulated, &quot;../data/specific_upreg_lung_genes.csv&quot;, row.names = FALSE)
lung_downregulated &lt;- conserved_tissue_specific(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung)
write.csv(lung_downregulated, &quot;../data/specific_downreg_lung_genes.csv&quot;, row.names = FALSE)

kidney_upregulated &lt;- conserved_tissue_specific(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney)
write.csv(kidney_upregulated, &quot;../data/specific_upreg_kidney_genes.csv&quot;, row.names = FALSE)
kidney_downregulated &lt;- conserved_tissue_specific(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney)
write.csv(kidney_downregulated, &quot;../data/specific_downreg_kidney_genes.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="find-genes-with-a-tissue-specific-direction-significant-and-upregulated-in-1-versus-2-3-4-but-not-significant-in-2-v-3-4-and-3-v-4-with-fdr" class="section level2">
<h2>Find genes with a tissue-specific direction (significant and upregulated in 1 versus 2, 3, 4 but not significant in 2 v 3, 4 and 3 v 4) with FDR</h2>
<pre class="r"><code># Combine the human ASH results
combined_human &lt;- cbind(results[[&quot;H_HeartvLi&quot;]]$adj.P.Val, results[[&quot;H_HeartvLi&quot;]]$logFC, results[[&quot;H_HeartvLu&quot;]]$adj.P.Val, results[[&quot;H_HeartvLu&quot;]]$logFC, results[[&quot;H_HeartvK&quot;]]$adj.P.Val, results[[&quot;H_HeartvK&quot;]]$logFC, results[[&quot;H_LivLu&quot;]]$adj.P.Val, results[[&quot;H_LivLu&quot;]]$logFC, results[[&quot;H_LivK&quot;]]$adj.P.Val, results[[&quot;H_LivK&quot;]]$logFC, results[[&quot;H_LuvK&quot;]]$adj.P.Val, results[[&quot;H_LuvK&quot;]]$logFC)

dim(combined_human)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code>rownames(combined_human) &lt;- rownames(cpm_12184)
  
# Combine the chimp ASH results
combined_chimp &lt;- cbind(results[[&quot;C_HeartvLi&quot;]]$adj.P.Val, results[[&quot;C_HeartvLi&quot;]]$logFC, results[[&quot;C_HeartvLu&quot;]]$adj.P.Val, results[[&quot;C_HeartvLu&quot;]]$logFC, results[[&quot;C_HeartvK&quot;]]$adj.P.Val, results[[&quot;C_HeartvK&quot;]]$logFC, results[[&quot;C_LivLu&quot;]]$adj.P.Val, results[[&quot;C_LivLu&quot;]]$logFC, results[[&quot;C_LivK&quot;]]$adj.P.Val, results[[&quot;C_LivK&quot;]]$logFC, results[[&quot;C_LuvK&quot;]]$adj.P.Val, results[[&quot;C_LuvK&quot;]]$logFC)

dim(combined_chimp)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code>rownames(combined_chimp) &lt;- rownames(cpm_12184)

# Combine the rhesus ASH results
combined_rhesus &lt;- cbind(results[[&quot;R_HeartvLi&quot;]]$adj.P.Val, results[[&quot;R_HeartvLi&quot;]]$logFC, results[[&quot;R_HeartvLu&quot;]]$adj.P.Val, results[[&quot;R_HeartvLu&quot;]]$logFC, results[[&quot;R_HeartvK&quot;]]$adj.P.Val, results[[&quot;R_HeartvK&quot;]]$logFC, results[[&quot;R_LivLu&quot;]]$adj.P.Val, results[[&quot;R_LivLu&quot;]]$logFC, results[[&quot;R_LivK&quot;]]$adj.P.Val, results[[&quot;R_LivK&quot;]]$logFC, results[[&quot;R_LuvK&quot;]]$adj.P.Val, results[[&quot;R_LuvK&quot;]]$logFC)

dim(combined_rhesus)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code>rownames(combined_rhesus) &lt;- rownames(cpm_12184)


FSR_level &lt;-  0.01

# Upregulated consistently in one human tissue

upreg_human_heart &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &gt; 0 &amp; combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &gt; 0 &amp; combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &gt; 0 &amp; combined_human[,7] &gt; FSR_level &amp; combined_human[,9] &gt; FSR_level &amp;  combined_human[,11] &gt; FSR_level),])

upreg_human_kidney &lt;- as.data.frame(combined_human[which(combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &lt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &lt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &lt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,3] &gt; FSR_level &amp; combined_human[,7] &gt; FSR_level),])

upreg_human_liver &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &lt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &gt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &gt; 0 &amp; combined_human[,3] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp; combined_human[,11] &gt; FSR_level),])

upreg_human_lung &lt;- as.data.frame(combined_human[which(combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &lt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &lt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &gt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp; combined_human[,9] &gt; FSR_level),])

# Upregulated in only one human tissue

#upreg_human_heart_only &lt;- setdiff(upreg_human_heart, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_lung)))
#upreg_human_kidney_only &lt;- setdiff(upreg_human_kidney, (union(union(upreg_human_heart, upreg_human_liver), upreg_human_lung)))
#upreg_human_liver_only &lt;- setdiff(upreg_human_liver, (union(union(upreg_human_kidney, upreg_human_heart), upreg_human_lung)))
#upreg_human_lung_only &lt;- setdiff(upreg_human_lung, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_heart)))

nrow(upreg_human_heart)</code></pre>
<pre><code>[1] 375</code></pre>
<pre class="r"><code>nrow(upreg_human_kidney)</code></pre>
<pre><code>[1] 226</code></pre>
<pre class="r"><code>nrow(upreg_human_liver)</code></pre>
<pre><code>[1] 501</code></pre>
<pre class="r"><code>nrow(upreg_human_lung)</code></pre>
<pre><code>[1] 211</code></pre>
<pre class="r"><code>nrow(upreg_human_heart_only)</code></pre>
<pre><code>[1] 390</code></pre>
<pre class="r"><code>nrow(upreg_human_kidney_only)</code></pre>
<pre><code>[1] 266</code></pre>
<pre class="r"><code>nrow(upreg_human_liver_only)</code></pre>
<pre><code>[1] 541</code></pre>
<pre class="r"><code>nrow(upreg_human_lung_only)</code></pre>
<pre><code>[1] 305</code></pre>
<pre class="r"><code># Upregulated consistently in one chimp tissue

upreg_chimp_heart &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &gt; 0 &amp; combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &gt; 0 &amp; combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &gt; 0 &amp; combined_chimp[,7] &gt; FSR_level &amp; combined_chimp[,9] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

upreg_chimp_kidney &lt;- as.data.frame(combined_chimp[which(combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &lt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &lt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &lt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,3] &gt; FSR_level &amp;  combined_chimp[,7] &gt; FSR_level),])

upreg_chimp_liver &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &lt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &gt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &gt; 0 &amp; combined_chimp[,3] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

upreg_chimp_lung &lt;- as.data.frame(combined_chimp[which(combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &lt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &lt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &gt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,9] &gt; FSR_level),])

nrow(upreg_chimp_heart)</code></pre>
<pre><code>[1] 438</code></pre>
<pre class="r"><code>nrow(upreg_chimp_kidney)</code></pre>
<pre><code>[1] 206</code></pre>
<pre class="r"><code>nrow(upreg_chimp_liver)</code></pre>
<pre><code>[1] 599</code></pre>
<pre class="r"><code>nrow(upreg_chimp_lung)</code></pre>
<pre><code>[1] 440</code></pre>
<pre class="r"><code># Upregulated consistently in one rhesus tissue

upreg_rhesus_heart &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &gt; 0 &amp; combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &gt; 0 &amp; combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &gt; 0 &amp; combined_rhesus[,7] &gt; FSR_level &amp; combined_rhesus[,9] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

upreg_rhesus_kidney &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &lt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &lt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &lt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,3] &gt; FSR_level &amp;  combined_rhesus[,7] &gt; FSR_level),])

upreg_rhesus_liver &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &lt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &gt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &gt; 0 &amp; combined_rhesus[,3] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

upreg_rhesus_lung &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &lt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &lt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &gt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,9] &gt; FSR_level),])

nrow(upreg_rhesus_heart)</code></pre>
<pre><code>[1] 472</code></pre>
<pre class="r"><code>nrow(upreg_rhesus_kidney)</code></pre>
<pre><code>[1] 181</code></pre>
<pre class="r"><code>nrow(upreg_rhesus_liver)</code></pre>
<pre><code>[1] 507</code></pre>
<pre class="r"><code>nrow(upreg_rhesus_lung)</code></pre>
<pre><code>[1] 476</code></pre>
<pre class="r"><code># Downregulated consistently in one human tissue

downreg_human_heart &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &lt; 0 &amp; combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &lt; 0 &amp; combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &lt; 0 &amp;  combined_human[,7] &gt; FSR_level &amp; combined_human[,9] &gt; FSR_level &amp;  combined_human[,11] &gt; FSR_level),])

downreg_human_kidney &lt;- as.data.frame(combined_human[which(combined_human[,5] &lt; FSR_level &amp; combined_human[,6] &gt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &gt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &gt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,3] &gt; FSR_level &amp;  combined_human[,7] &gt; FSR_level),])

downreg_human_liver &lt;- as.data.frame(combined_human[which(combined_human[,1] &lt; FSR_level &amp; combined_human[,2] &gt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &lt; 0 &amp; combined_human[,9] &lt; FSR_level &amp; combined_human[,10] &lt; 0 &amp; combined_human[,3] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp;  combined_human[,11] &gt; FSR_level),])

downreg_human_lung &lt;- as.data.frame(combined_human[which(combined_human[,3] &lt; FSR_level &amp; combined_human[,4] &gt; 0 &amp; combined_human[,7] &lt; FSR_level &amp; combined_human[,8] &gt; 0 &amp; combined_human[,11] &lt; FSR_level &amp; combined_human[,12] &lt; 0 &amp; combined_human[,1] &gt; FSR_level &amp; combined_human[,5] &gt; FSR_level &amp;  combined_human[,9] &gt; FSR_level),])

# Downregulated in only one human tissue

nrow(downreg_human_heart)</code></pre>
<pre><code>[1] 242</code></pre>
<pre class="r"><code>nrow(downreg_human_kidney)</code></pre>
<pre><code>[1] 100</code></pre>
<pre class="r"><code>nrow(downreg_human_liver)</code></pre>
<pre><code>[1] 306</code></pre>
<pre class="r"><code>nrow(downreg_human_lung)</code></pre>
<pre><code>[1] 208</code></pre>
<pre class="r"><code># Downregulated consistently in one chimp tissue

downreg_chimp_heart &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &lt; 0 &amp; combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &lt; 0 &amp; combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &lt; 0 &amp; combined_chimp[,7] &gt; FSR_level &amp; combined_chimp[,9] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

downreg_chimp_kidney &lt;- as.data.frame(combined_chimp[which(combined_chimp[,5] &lt; FSR_level &amp; combined_chimp[,6] &gt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &gt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &gt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,3] &gt; FSR_level &amp;  combined_chimp[,7] &gt; FSR_level),])

downreg_chimp_liver &lt;- as.data.frame(combined_chimp[which(combined_chimp[,1] &lt; FSR_level &amp; combined_chimp[,2] &gt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &lt; 0 &amp; combined_chimp[,9] &lt; FSR_level &amp; combined_chimp[,10] &lt; 0 &amp; combined_chimp[,3] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,11] &gt; FSR_level),])

downreg_chimp_lung &lt;- as.data.frame(combined_chimp[which(combined_chimp[,3] &lt; FSR_level &amp; combined_chimp[,4] &gt; 0 &amp; combined_chimp[,7] &lt; FSR_level &amp; combined_chimp[,8] &gt; 0 &amp; combined_chimp[,11] &lt; FSR_level &amp; combined_chimp[,12] &lt; 0 &amp; combined_chimp[,1] &gt; FSR_level &amp; combined_chimp[,5] &gt; FSR_level &amp;  combined_chimp[,9] &gt; FSR_level),])

nrow(downreg_chimp_heart)</code></pre>
<pre><code>[1] 334</code></pre>
<pre class="r"><code>nrow(downreg_chimp_kidney)</code></pre>
<pre><code>[1] 61</code></pre>
<pre class="r"><code>nrow(downreg_chimp_liver)</code></pre>
<pre><code>[1] 350</code></pre>
<pre class="r"><code>nrow(downreg_chimp_lung)</code></pre>
<pre><code>[1] 239</code></pre>
<pre class="r"><code># Downregulated consistently in one rhesus tissue

downreg_rhesus_heart &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &lt; 0 &amp; combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &lt; 0 &amp; combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &lt; 0 &amp; combined_rhesus[,7] &gt; FSR_level &amp; combined_rhesus[,9] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

downreg_rhesus_kidney &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,5] &lt; FSR_level &amp; combined_rhesus[,6] &gt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &gt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &gt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,3] &gt; FSR_level &amp;  combined_rhesus[,7] &gt; FSR_level),])

downreg_rhesus_liver &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,1] &lt; FSR_level &amp; combined_rhesus[,2] &gt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &lt; 0 &amp; combined_rhesus[,9] &lt; FSR_level &amp; combined_rhesus[,10] &lt; 0 &amp; combined_rhesus[,3] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,11] &gt; FSR_level),])

downreg_rhesus_lung &lt;- as.data.frame(combined_rhesus[which(combined_rhesus[,3] &lt; FSR_level &amp; combined_rhesus[,4] &gt; 0 &amp; combined_rhesus[,7] &lt; FSR_level &amp; combined_rhesus[,8] &gt; 0 &amp; combined_rhesus[,11] &lt; FSR_level &amp; combined_rhesus[,12] &lt; 0 &amp; combined_rhesus[,1] &gt; FSR_level &amp; combined_rhesus[,5] &gt; FSR_level &amp;  combined_rhesus[,9] &gt; FSR_level),])

nrow(downreg_rhesus_heart)</code></pre>
<pre><code>[1] 381</code></pre>
<pre class="r"><code>nrow(downreg_rhesus_kidney)</code></pre>
<pre><code>[1] 71</code></pre>
<pre class="r"><code>nrow(downreg_rhesus_liver)</code></pre>
<pre><code>[1] 304</code></pre>
<pre class="r"><code>nrow(downreg_rhesus_lung)</code></pre>
<pre><code>[1] 329</code></pre>
<pre class="r"><code># Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram &lt;- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist &lt;- list()
mylist[[&quot;Hu&quot;]] &lt;- rownames(upreg_human_tissue)
mylist[[&quot;Ch&quot;]] &lt;- rownames(upreg_chimp_tissue)
mylist[[&quot;Rh&quot;]] &lt;- rownames(upreg_rhesus_tissue)


# Make Venn Diagram
#dev.off()
Heart_specific_up &lt;- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = &quot;sans&quot;, cat.fontfamily = &quot;sans&quot;, main.fontfamily = &quot;sans&quot;, cat.fontface = &quot;bold&quot;, main.fontface = &quot;bold&quot;, cat.dist = 0.05)
grid.draw(Heart_specific_up)
}


svg(&quot;Upregulated_heart_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart, &quot;Upregulated in heart only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Upregulated_kidney_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney, &quot;Upregulated in kidney only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Upregulated_liver_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver, &quot;Upregulated in liver only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Upregulated_lung_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung, &quot;Upregulated in lung only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_heart_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart, &quot;Downregulated in heart only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_kidney_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney, &quot;Downregulated in kidney only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_liver_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver, &quot;Downregulated in liver only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>svg(&quot;Downregulated_lung_specific.svg&quot;, width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung, &quot;Downregulated in lung only&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>#title &lt;- ggdraw() + draw_label(&quot;Genes regulated in a tissue specific direction (FDR 1%)&quot;, fontface=&#39;bold&#39;)
p1 &lt;- ggdraw()+draw_image(&quot;Upregulated_heart_specific.svg&quot;)
p2 &lt;- ggdraw()+draw_image(&quot;Downregulated_heart_specific.svg&quot;)
p3 &lt;- ggdraw()+draw_image(&quot;Upregulated_kidney_specific.svg&quot;)
p4 &lt;- ggdraw()+draw_image(&quot;Downregulated_kidney_specific.svg&quot;)

p5 &lt;- ggdraw()+draw_image(&quot;Upregulated_liver_specific.svg&quot;)
p6 &lt;- ggdraw()+draw_image(&quot;Downregulated_liver_specific.svg&quot;)
p7 &lt;- ggdraw()+draw_image(&quot;Upregulated_lung_specific.svg&quot;)
p8 &lt;- ggdraw()+draw_image(&quot;Downregulated_lung_specific.svg&quot;)

eight_plots &lt;- plot_grid(p1, p2,  p5, p6, p7, p8, p3, p4, labels = c(&quot;2A.&quot;, &quot;2B.&quot;, &quot;2C.&quot;, &quot;2D.&quot;, &quot;2E.&quot;, &quot;2F.&quot;, &quot;2G.&quot;, &quot;2H.&quot;), ncol = 2)
plot_fig2 &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))

save_plot(&quot;../data/test.png&quot;, plot_fig2,
           ncol = 2, # we&#39;re saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
           # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 0.75
           )

save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Figures/Figure_2_Tissue_specific.png&quot;, plot_fig2,
           ncol = 2, # we&#39;re saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
       # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
           )

heart_upregulated &lt;- conserved_tissue_specific(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart)
#write.csv(heart_upregulated, &quot;../data/specific_upreg_heart_genes.csv&quot;, row.names = FALSE)
heart_downregulated &lt;- conserved_tissue_specific(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart)
#write.csv(heart_downregulated, &quot;../data/specific_downreg_heart_genes.csv&quot;, row.names = FALSE)

liver_upregulated &lt;- conserved_tissue_specific(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver)
#write.csv(liver_upregulated, &quot;../data/specific_upreg_liver_genes.csv&quot;, row.names = FALSE)
liver_downregulated &lt;- conserved_tissue_specific(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver)
#write.csv(liver_downregulated, &quot;../data/specific_downreg_liver_genes.csv&quot;, row.names = FALSE)

lung_upregulated &lt;- conserved_tissue_specific(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung)
#write.csv(lung_upregulated, &quot;../data/specific_upreg_lung_genes.csv&quot;, row.names = FALSE)
lung_downregulated &lt;- conserved_tissue_specific(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung)
#write.csv(lung_downregulated, &quot;../data/specific_downreg_lung_genes.csv&quot;, row.names = FALSE)

kidney_upregulated &lt;- conserved_tissue_specific(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney)
#write.csv(kidney_upregulated, &quot;../data/specific_upreg_kidney_genes.csv&quot;, row.names = FALSE)
kidney_downregulated &lt;- conserved_tissue_specific(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney)
#write.csv(kidney_downregulated, &quot;../data/specific_downreg_kidney_genes.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="species-specific-1-direction-supplementary-table" class="section level2">
<h2>Species specific 1 direction (supplementary table)</h2>
<pre class="r"><code>FDR_level &lt;- 0.01
combined_hcr &lt;- cbind(results[[&quot;HvC_Heart&quot;]]$adj.P.Val, results[[&quot;HvC_Heart&quot;]]$logFC, results[[&quot;HvR_Heart&quot;]]$adj.P.Val, results[[&quot;HvR_Heart&quot;]]$logFC, results[[&quot;CvR_Heart&quot;]]$adj.P.Val, results[[&quot;CvR_Heart&quot;]]$logFC, results[[&quot;HvC_Kidney&quot;]]$adj.P.Val, results[[&quot;HvC_Kidney&quot;]]$logFC, results[[&quot;HvR_Kidney&quot;]]$adj.P.Val, results[[&quot;HvR_Kidney&quot;]]$logFC, results[[&quot;CvR_Kidney&quot;]]$adj.P.Val, results[[&quot;CvR_Kidney&quot;]]$logFC, results[[&quot;HvC_Liver&quot;]]$adj.P.Val, results[[&quot;HvC_Liver&quot;]]$logFC, results[[&quot;HvR_Liver&quot;]]$adj.P.Val, results[[&quot;HvR_Liver&quot;]]$logFC, results[[&quot;CvR_Liver&quot;]]$adj.P.Val, results[[&quot;CvR_Liver&quot;]]$logFC, results[[&quot;HvC_Lung&quot;]]$adj.P.Val, results[[&quot;HvC_Lung&quot;]]$logFC, results[[&quot;HvR_Lung&quot;]]$adj.P.Val, results[[&quot;HvR_Lung&quot;]]$logFC, results[[&quot;CvR_Lung&quot;]]$adj.P.Val, results[[&quot;CvR_Lung&quot;]]$logFC)

dim(combined_hcr)</code></pre>
<pre><code>[1] 12184    24</code></pre>
<pre class="r"><code>rownames(combined_hcr) &lt;- rownames(cpm_12184)

# Human specific and upregulated

combined_human_upregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &gt; 0 &amp; combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &gt; 0 &amp; combined_hcr[,5] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &gt; 0 &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &gt; 0 &amp; combined_hcr[,11] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &gt; 0 &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &gt; 0 &amp; combined_hcr[,17] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &gt; 0 &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &gt; 0 &amp; combined_hcr[,23] &gt; FDR_level),]


# Human specific and downregulated

combined_human_downregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &lt; 0 &amp; combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &lt; 0 &amp; combined_hcr[,5] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &lt; 0 &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &lt; 0 &amp; combined_hcr[,11] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &lt; 0 &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &lt; 0 &amp; combined_hcr[,17] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &lt; 0 &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &lt; 0 &amp; combined_hcr[,23] &gt; FDR_level),]

# Chimp specific and upregulated

combined_chimp_upregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &lt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &gt; 0 &amp; combined_hcr[,3] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &lt; 0 &amp; combined_hcr[,10] &lt; FDR_level &amp; combined_hcr[,11] &gt; 0 &amp; combined_hcr[,9] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &lt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &gt; 0 &amp; combined_hcr[,15] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &lt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &gt; 0 &amp; combined_hcr[,21] &gt; FDR_level),]


# Chimp specific and downregulated

combined_chimp_downregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &gt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &lt; 0 &amp; combined_hcr[,3] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &gt; 0 &amp; combined_hcr[,10] &lt; FDR_level &amp; combined_hcr[,11] &lt; 0 &amp; combined_hcr[,9] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &gt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &lt; 0 &amp; combined_hcr[,15] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &gt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &lt; 0 &amp; combined_hcr[,21] &gt; FDR_level),]

# Rhesus specific and upregulated

combined_rhesus_upregulated &lt;- combined_hcr[which(combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &lt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &lt; 0 &amp; combined_hcr[,1] &gt; FDR_level &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &lt; 0 &amp; combined_hcr[,11] &lt; FDR_level &amp; combined_hcr[,12] &lt; 0 &amp; combined_hcr[,7] &gt; FDR_level &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &lt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &lt; 0 &amp; combined_hcr[,13] &gt; FDR_level &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &lt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &lt; 0 &amp; combined_hcr[,19] &gt; FDR_level),]

combined_rhesus_downregulated &lt;- combined_hcr[which(combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &gt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &gt; 0 &amp; combined_hcr[,1] &gt; FDR_level &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &gt; 0 &amp; combined_hcr[,11] &lt; FDR_level &amp; combined_hcr[,12] &gt; 0 &amp; combined_hcr[,7] &gt; FDR_level &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &gt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &gt; 0 &amp; combined_hcr[,13] &gt; FDR_level &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &gt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &gt; 0 &amp; combined_hcr[,19] &gt; FDR_level),]


length(combined_human_upregulated)</code></pre>
<pre><code>[1] 264</code></pre>
<pre class="r"><code>length(combined_human_downregulated)</code></pre>
<pre><code>[1] 192</code></pre>
<pre class="r"><code>length(combined_chimp_upregulated)</code></pre>
<pre><code>[1] 48</code></pre>
<pre class="r"><code>length(combined_chimp_downregulated)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(combined_rhesus_upregulated)</code></pre>
<pre><code>[1] 2304</code></pre>
<pre class="r"><code>length(combined_rhesus_downregulated)</code></pre>
<pre><code>[1] 2376</code></pre>
<pre class="r"><code># Use adaptive shrinkage/FSR instead of FDR

FDR_level &lt;- 0.05
combined_hcr &lt;- cbind(results[[&quot;HvC_Heart&quot;]]$svalue, results[[&quot;HvC_Heart&quot;]]$beta_est, results[[&quot;HvR_Heart&quot;]]$svalue, results[[&quot;HvR_Heart&quot;]]$beta_est, results[[&quot;CvR_Heart&quot;]]$svalue, results[[&quot;CvR_Heart&quot;]]$beta_est, results[[&quot;HvC_Kidney&quot;]]$svalue, results[[&quot;HvC_Kidney&quot;]]$beta_est, results[[&quot;HvR_Kidney&quot;]]$svalue, results[[&quot;HvR_Kidney&quot;]]$beta_est, results[[&quot;CvR_Kidney&quot;]]$svalue, results[[&quot;CvR_Kidney&quot;]]$beta_est, results[[&quot;HvC_Liver&quot;]]$svalue, results[[&quot;HvC_Liver&quot;]]$beta_est, results[[&quot;HvR_Liver&quot;]]$svalue, results[[&quot;HvR_Liver&quot;]]$beta_est, results[[&quot;CvR_Liver&quot;]]$svalue, results[[&quot;CvR_Liver&quot;]]$beta_est, results[[&quot;HvC_Lung&quot;]]$svalue, results[[&quot;HvC_Lung&quot;]]$beta_est, results[[&quot;HvR_Lung&quot;]]$svalue, results[[&quot;HvR_Lung&quot;]]$beta_est, results[[&quot;CvR_Lung&quot;]]$svalue, results[[&quot;CvR_Lung&quot;]]$beta_est)

dim(combined_hcr)</code></pre>
<pre><code>[1] 12184    24</code></pre>
<pre class="r"><code>rownames(combined_hcr) &lt;- rownames(cpm_12184)

# Human specific and upregulated

combined_human_upregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &gt; 0 &amp; combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &gt; 0 &amp; combined_hcr[,5] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &gt; 0 &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &gt; 0 &amp; combined_hcr[,11] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &gt; 0 &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &gt; 0 &amp; combined_hcr[,17] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &gt; 0 &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &gt; 0 &amp; combined_hcr[,23] &gt; FDR_level),]


# Human specific and downregulated

combined_human_downregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &lt; 0 &amp; combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &lt; 0 &amp; combined_hcr[,5] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &lt; 0 &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &lt; 0 &amp; combined_hcr[,11] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &lt; 0 &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &lt; 0 &amp; combined_hcr[,17] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &lt; 0 &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &lt; 0 &amp; combined_hcr[,23] &gt; FDR_level),]

# Chimp specific and upregulated

combined_chimp_upregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &lt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &gt; 0 &amp; combined_hcr[,3] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &lt; 0 &amp; combined_hcr[,10] &lt; FDR_level &amp; combined_hcr[,11] &gt; 0 &amp; combined_hcr[,9] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &lt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &gt; 0 &amp; combined_hcr[,15] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &lt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &gt; 0 &amp; combined_hcr[,21] &gt; FDR_level),]


# Chimp specific and downregulated

combined_chimp_downregulated &lt;- combined_hcr[which(combined_hcr[,1] &lt; FDR_level &amp; combined_hcr[,2] &gt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &lt; 0 &amp; combined_hcr[,3] &gt; FDR_level &amp; combined_hcr[,7] &lt; FDR_level &amp; combined_hcr[,8] &gt; 0 &amp; combined_hcr[,10] &lt; FDR_level &amp; combined_hcr[,11] &lt; 0 &amp; combined_hcr[,9] &gt; FDR_level &amp; combined_hcr[,13] &lt; FDR_level &amp; combined_hcr[,14] &gt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &lt; 0 &amp; combined_hcr[,15] &gt; FDR_level &amp; combined_hcr[,19] &lt; FDR_level &amp; combined_hcr[,20] &gt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &lt; 0 &amp; combined_hcr[,21] &gt; FDR_level),]

# Rhesus specific and upregulated

combined_rhesus_upregulated &lt;- combined_hcr[which(combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &lt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &lt; 0 &amp; combined_hcr[,1] &gt; FDR_level &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &lt; 0 &amp; combined_hcr[,11] &lt; FDR_level &amp; combined_hcr[,12] &lt; 0 &amp; combined_hcr[,7] &gt; FDR_level &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &lt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &lt; 0 &amp; combined_hcr[,13] &gt; FDR_level &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &lt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &lt; 0 &amp; combined_hcr[,19] &gt; FDR_level),]

combined_rhesus_downregulated &lt;- combined_hcr[which(combined_hcr[,3] &lt; FDR_level &amp; combined_hcr[,4] &gt; 0 &amp; combined_hcr[,5] &lt; FDR_level &amp; combined_hcr[,6] &gt; 0 &amp; combined_hcr[,1] &gt; FDR_level &amp; combined_hcr[,9] &lt; FDR_level &amp; combined_hcr[,10] &gt; 0 &amp; combined_hcr[,11] &lt; FDR_level &amp; combined_hcr[,12] &gt; 0 &amp; combined_hcr[,7] &gt; FDR_level &amp; combined_hcr[,15] &lt; FDR_level &amp; combined_hcr[,16] &gt; 0 &amp; combined_hcr[,17] &lt; FDR_level &amp; combined_hcr[,18] &gt; 0 &amp; combined_hcr[,13] &gt; FDR_level &amp; combined_hcr[,21] &lt; FDR_level &amp; combined_hcr[,22] &gt; 0 &amp; combined_hcr[,23] &lt; FDR_level &amp; combined_hcr[,24] &gt; 0 &amp; combined_hcr[,19] &gt; FDR_level),]


length(combined_human_upregulated)</code></pre>
<pre><code>[1] 288</code></pre>
<pre class="r"><code>length(combined_human_downregulated)</code></pre>
<pre><code>[1] 144</code></pre>
<pre class="r"><code>length(combined_chimp_upregulated)</code></pre>
<pre><code>[1] 24</code></pre>
<pre class="r"><code>length(combined_chimp_downregulated)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(combined_rhesus_upregulated)</code></pre>
<pre><code>[1] 5112</code></pre>
<pre class="r"><code>length(combined_rhesus_downregulated)</code></pre>
<pre><code>[1] 3792</code></pre>
</div>
<div id="experimenting-with-interactions-3-species-1-tissue-versus-a-group-of-3-tissues" class="section level2">
<h2>Experimenting with interactions (3 species, 1 tissue versus a group of 3 tissues)</h2>
<p>There are many possible interactions (e.g. Human and Chimp heart versus liver). To minimize the number of comparisons, we are going to compare 1 tissue to a group of the other 3 tissues (e.g. Human and Chimp heart versus non-heart).</p>
<div id="interactions-in-all-3-species-hearts-versus-all-other-tissues" class="section level3">
<h3>Interactions in all 3 species’ hearts versus all other tissues</h3>
<pre class="r"><code># Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue_group) 

## Make the contrast matrix and rename columns of the contrast matrix

design &lt;- model.matrix(~ species*tissue_group + RIN)

# Rename columns of the contrast matrix
colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Human&quot;
colnames(design)[3] &lt;- &quot;Rhesus&quot;
colnames(design)[4] &lt;- &quot;tissue_heart&quot; # Heart versus non-heart
colnames(design)[5] &lt;- &quot;RIN&quot;
colnames(design)[6] &lt;- &quot;Human_heart&quot;
colnames(design)[7] &lt;- &quot;Rhesus_heart&quot;


# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept        Human       Rhesus tissue_heart          RIN 
        47.0         15.0         16.0         11.0        368.2 
 Human_heart Rhesus_heart 
         3.0          4.0 </code></pre>
<pre class="r"><code># Run linear model with new design matrix
cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;- -0.1009512
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm2 &lt;- makeContrasts(HC_HG = Human_heart - Human, 
                     HR_HG = (Human_heart - Rhesus_heart) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_heart,
                     levels = design)
# Implement contrasts

contrasts_interactions &lt;- contrasts.fit(fit.cyclic.norm, cm2)
fit1 &lt;- eBayes(contrasts_interactions)



# Pull interactions

top3 &lt;- list(HC_HG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HR_HG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CR_HG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Determine the number of statistically signficant interactions

HC_HG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HC_HG[which(HC_HG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 281   7</code></pre>
<pre class="r"><code>HR_HG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HR_HG[which(HR_HG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 1212    7</code></pre>
<pre class="r"><code>CR_HG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CR_HG[which(CR_HG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 874   7</code></pre>
</div>
<div id="adaptive-shrinkage-on-interactions-in-all-3-species-hearts-versus-all-other-tissues" class="section level3">
<h3>Adaptive shrinkage on interactions in all 3 species’ hearts versus all other tissues</h3>
<pre class="r"><code># Prepare the data for ASH
tests &lt;- colnames(fit1$coefficients)
tests</code></pre>
<pre><code>[1] &quot;HC_HG&quot; &quot;HR_HG&quot; &quot;CR_HG&quot;</code></pre>
<pre class="r"><code>results &lt;- vector(length = length(tests), mode = &quot;list&quot;)
names(results) &lt;- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] &lt;- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit1, coef = test)
  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code># Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, &quot;results-limma-voom-ash-interactions.rds&quot;))

heart_combined_interactions &lt;- as.data.frame(cbind(results[[&quot;HC_HG&quot;]]$svalue, 
                               results[[&quot;HR_HG&quot;]]$svalue, 
                               results[[&quot;CR_HG&quot;]]$svalue))

dim(heart_combined_interactions)</code></pre>
<pre><code>[1] 12184     3</code></pre>
<pre class="r"><code>colnames(heart_combined_interactions) &lt;- c(&quot;HC_HG&quot;, &quot;HR_HG&quot;, &quot;CR_HG&quot;)</code></pre>
</div>
<div id="interactions-in-all-3-species-livers-versus-all-other-tissues" class="section level3">
<h3>Interactions in all 3 species’ livers versus all other tissues</h3>
<pre class="r"><code># Reassign tissues to be liver versus non-livers

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue_group) 


# Rename columns of the contrast matrix

## Make the contrast matrix and rename columns of the contrast matrix

design &lt;- model.matrix(~ species*tissue_group + RIN)

# Rename columns of the contrast matrix
colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Human&quot;
colnames(design)[3] &lt;- &quot;Rhesus&quot;
colnames(design)[4] &lt;- &quot;tissue_livers&quot; # Heart versus non-livers
colnames(design)[5] &lt;- &quot;RIN&quot;
colnames(design)[6] &lt;- &quot;Human_liver&quot;
colnames(design)[7] &lt;- &quot;Rhesus_liver&quot;


# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>    Intercept         Human        Rhesus tissue_livers           RIN 
         47.0          15.0          16.0          12.0         368.2 
  Human_liver  Rhesus_liver 
          4.0           4.0 </code></pre>
<pre class="r"><code># Run linear model with new design matrix
cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;- -0.0912331
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm2 &lt;- makeContrasts(HC_HG = Human_liver - Human, 
                     HR_HG = (Human_liver - Rhesus_liver) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_liver,
                     levels = design)

# Implement contrasts

contrasts_interactions &lt;- contrasts.fit(fit.cyclic.norm, cm2)
fit1 &lt;- eBayes(contrasts_interactions)

# Pull interactions

top3 &lt;- list(HC_LiG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HR_LiG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CR_LiG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Determine the number of statistically signficant interactions

HC_LiG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HC_LiG[which(HC_LiG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 167   7</code></pre>
<pre class="r"><code>HR_LiG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HR_LiG[which(HR_LiG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 1073    7</code></pre>
<pre class="r"><code>CR_LiG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CR_LiG[which(CR_LiG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 784   7</code></pre>
</div>
<div id="adaptive-shrinkage-on-interactions-in-all-3-species-livers-versus-all-other-tissues" class="section level3">
<h3>Adaptive shrinkage on interactions in all 3 species’ livers versus all other tissues</h3>
<pre class="r"><code># Prepare the data for ASH
tests &lt;- colnames(fit1$coefficients)
tests</code></pre>
<pre><code>[1] &quot;HC_HG&quot; &quot;HR_HG&quot; &quot;CR_HG&quot;</code></pre>
<pre class="r"><code>results &lt;- vector(length = length(tests), mode = &quot;list&quot;)
names(results) &lt;- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] &lt;- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit1, coef = test)
  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code># Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, &quot;results-limma-voom-ash-interactions.rds&quot;))

liver_combined_interactions &lt;- as.data.frame(cbind(results[[&quot;HC_HG&quot;]]$svalue, 
                               results[[&quot;HR_HG&quot;]]$svalue, 
                               results[[&quot;CR_HG&quot;]]$svalue))

dim(liver_combined_interactions)</code></pre>
<pre><code>[1] 12184     3</code></pre>
<pre class="r"><code>colnames(liver_combined_interactions) &lt;- c(&quot;HC_LiG&quot;, &quot;HR_LiG&quot;, &quot;CR_LiG&quot;)</code></pre>
</div>
<div id="interactions-in-all-3-species-lungs-versus-all-other-tissues" class="section level3">
<h3>Interactions in all 3 species’ lungs versus all other tissues</h3>
<pre class="r"><code># Reassign tissues to be lung versus non-lung

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 

## Make the contrast matrix and rename columns of the contrast matrix

design &lt;- model.matrix(~ species*tissue_group + RIN)

# Rename columns of the contrast matrix
colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Human&quot;
colnames(design)[3] &lt;- &quot;Rhesus&quot;
colnames(design)[4] &lt;- &quot;tissue_lungs&quot; # Heart versus non-livers
colnames(design)[5] &lt;- &quot;RIN&quot;
colnames(design)[6] &lt;- &quot;Human_lung&quot;
colnames(design)[7] &lt;- &quot;Rhesus_lung&quot;


# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept        Human       Rhesus tissue_lungs          RIN 
        47.0         15.0         16.0         12.0        368.2 
  Human_lung  Rhesus_lung 
         4.0          4.0 </code></pre>
<pre class="r"><code># Run linear model with new design matrix
cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;-  -0.09711687
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm4 &lt;-  makeContrasts(HC_HG = Human_lung - Human, 
                     HR_HG = (Human_lung - Rhesus_lung) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_lung,
                     levels = design)

# Implement contrasts

contrasts_interactions &lt;- contrasts.fit(fit.cyclic.norm, cm4)
fit1 &lt;- eBayes(contrasts_interactions)

# Pull interactions

top3 &lt;- list(HC_LuG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HR_LuG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CR_LuG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Determine the number of statistically signficant interactions

HC_LuG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HC_LuG[which(HC_LuG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 168   7</code></pre>
<pre class="r"><code>HR_LuG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HR_LuG[which(HR_LuG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 1105    7</code></pre>
<pre class="r"><code>CR_LuG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CR_LuG[which(CR_LuG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 1168    7</code></pre>
</div>
<div id="adaptive-shrinkage-on-interactions-in-all-3-species-lungs-versus-all-other-tissues" class="section level3">
<h3>Adaptive shrinkage on interactions in all 3 species’ lungs versus all other tissues</h3>
<pre class="r"><code># Prepare the data for ASH
tests &lt;- colnames(fit1$coefficients)
tests</code></pre>
<pre><code>[1] &quot;HC_HG&quot; &quot;HR_HG&quot; &quot;CR_HG&quot;</code></pre>
<pre class="r"><code>results &lt;- vector(length = length(tests), mode = &quot;list&quot;)
names(results) &lt;- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] &lt;- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit1, coef = test)
  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code># Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, &quot;results-limma-voom-ash-interactions.rds&quot;))

lung_combined_interactions &lt;- as.data.frame(cbind(results[[&quot;HC_HG&quot;]]$svalue, 
                               results[[&quot;HR_HG&quot;]]$svalue, 
                               results[[&quot;CR_HG&quot;]]$svalue))

dim(lung_combined_interactions)</code></pre>
<pre><code>[1] 12184     3</code></pre>
<pre class="r"><code>colnames(lung_combined_interactions) &lt;- c(&quot;HC_LuG&quot;, &quot;HR_LuG&quot;, &quot;CR_LuG&quot;)

#check_lung_combined_interactions &lt;- as.data.frame(cbind(results[[&quot;HC_LuG&quot;]]$adj.P.Val, results[[&quot;HC_LuG&quot;]]$svalue))
#colnames(check_lung_combined_interactions) &lt;- c(&quot;BH-adj P value&quot;, &quot;s value&quot;)
#dim(check_lung_combined_interactions[which(check_lung_combined_interactions[,1] &lt; 0.01) ,])
#check_lung_combined_interactions[which(check_lung_combined_interactions[,1] &lt; 0.01) ,]
#dim(check_lung_combined_interactions[which(check_lung_combined_interactions[,2] &lt; 0.01) ,])</code></pre>
</div>
<div id="interactions-in-all-3-species-kidneys-versus-all-other-tissues" class="section level3">
<h3>Interactions in all 3 species’ kidneys versus all other tissues</h3>
<pre class="r"><code># Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 

design &lt;- model.matrix(~ species*tissue_group + RIN)

## Make the contrast matrix and rename columns of the contrast matrix

# Rename columns of the contrast matrix
colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Human&quot;
colnames(design)[3] &lt;- &quot;Rhesus&quot;
colnames(design)[4] &lt;- &quot;tissue_kidneys&quot; # Heart versus non-livers
colnames(design)[5] &lt;- &quot;RIN&quot;
colnames(design)[6] &lt;- &quot;Human_kidney&quot;
colnames(design)[7] &lt;- &quot;Rhesus_kidney&quot;


# Implement contrasts

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>     Intercept          Human         Rhesus tissue_kidneys            RIN 
          47.0           15.0           16.0           12.0          368.2 
  Human_kidney  Rhesus_kidney 
           4.0            4.0 </code></pre>
<pre class="r"><code># Run linear model with new design matrix
cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;-  -0.128052
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm5 &lt;- makeContrasts(HC_HG = Human_kidney - Human, 
                     HR_HG = (Human_kidney - Rhesus_kidney) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_kidney,
                     levels = design)

# Implement contrasts

contrasts_interactions &lt;- contrasts.fit(fit.cyclic.norm, cm5)
fit1 &lt;- eBayes(contrasts_interactions)

# Pull interactions

top3 &lt;- list(HC_KG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HR_KG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CR_KG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Determine the number of statistically signficant interactions

HC_KG =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HC_KG[which(HC_KG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 151   7</code></pre>
<pre class="r"><code>HR_KG =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(HR_KG[which(HR_KG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 728   7</code></pre>
<pre class="r"><code>CR_KG =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(CR_KG[which(CR_KG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 472   7</code></pre>
</div>
<div id="tissue-specific-interactions-fdr" class="section level3">
<h3>Tissue-specific interactions (FDR)</h3>
<pre class="r"><code># Set FDR level

FDR_level &lt;- 0.01

# Interactions between 3 species hearts and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &lt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &lt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &lt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &gt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &gt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &gt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &gt; FDR_level), ]))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code># Interactions between 3 species livers and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &lt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &lt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &lt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &gt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &gt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &gt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &gt; FDR_level), ]))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code># Interactions between 3 species lungs and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &lt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &lt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &lt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &gt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &gt; FDR_level), ]))</code></pre>
<pre><code>[1] &quot;ENSG00000099326&quot;</code></pre>
<pre class="r"><code># Interactions between 3 species kidneus and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &gt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &gt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &lt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &lt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &lt; FDR_level), ]))</code></pre>
<pre><code>[1] &quot;ENSG00000121236&quot; &quot;ENSG00000258232&quot;</code></pre>
<pre class="r"><code># Interactions between 3 species hearts and other tissues (Human/Chimp versus Rhesus)

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &lt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &lt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &gt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &gt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &gt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &gt; FDR_level), ]))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code># Interactions between 3 species livers and other tissues (Human/Chimp versus Rhesus)

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &lt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &lt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &gt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &gt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &gt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &gt; FDR_level), ]))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code># Interactions between 3 species lungs and other tissues (Human/Chimp versus Rhesus)

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &lt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &lt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &gt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &gt; FDR_level), ]))</code></pre>
<pre><code>  [1] &quot;ENSG00000005469&quot; &quot;ENSG00000008516&quot; &quot;ENSG00000011422&quot;
  [4] &quot;ENSG00000013503&quot; &quot;ENSG00000042445&quot; &quot;ENSG00000048540&quot;
  [7] &quot;ENSG00000064961&quot; &quot;ENSG00000071859&quot; &quot;ENSG00000075303&quot;
 [10] &quot;ENSG00000076924&quot; &quot;ENSG00000078124&quot; &quot;ENSG00000079616&quot;
 [13] &quot;ENSG00000082146&quot; &quot;ENSG00000083838&quot; &quot;ENSG00000085514&quot;
 [16] &quot;ENSG00000085831&quot; &quot;ENSG00000087206&quot; &quot;ENSG00000088826&quot;
 [19] &quot;ENSG00000089127&quot; &quot;ENSG00000092036&quot; &quot;ENSG00000092529&quot;
 [22] &quot;ENSG00000093072&quot; &quot;ENSG00000100297&quot; &quot;ENSG00000100359&quot;
 [25] &quot;ENSG00000100376&quot; &quot;ENSG00000101224&quot; &quot;ENSG00000102034&quot;
 [28] &quot;ENSG00000104518&quot; &quot;ENSG00000105287&quot; &quot;ENSG00000105819&quot;
 [31] &quot;ENSG00000107672&quot; &quot;ENSG00000107949&quot; &quot;ENSG00000108064&quot;
 [34] &quot;ENSG00000108219&quot; &quot;ENSG00000108433&quot; &quot;ENSG00000108551&quot;
 [37] &quot;ENSG00000108587&quot; &quot;ENSG00000108797&quot; &quot;ENSG00000110274&quot;
 [40] &quot;ENSG00000111110&quot; &quot;ENSG00000113391&quot; &quot;ENSG00000114270&quot;
 [43] &quot;ENSG00000116898&quot; &quot;ENSG00000117245&quot; &quot;ENSG00000117360&quot;
 [46] &quot;ENSG00000119801&quot; &quot;ENSG00000120280&quot; &quot;ENSG00000120333&quot;
 [49] &quot;ENSG00000120992&quot; &quot;ENSG00000122335&quot; &quot;ENSG00000122386&quot;
 [52] &quot;ENSG00000124767&quot; &quot;ENSG00000125846&quot; &quot;ENSG00000125966&quot;
 [55] &quot;ENSG00000127054&quot; &quot;ENSG00000128928&quot; &quot;ENSG00000129219&quot;
 [58] &quot;ENSG00000129317&quot; &quot;ENSG00000129933&quot; &quot;ENSG00000130731&quot;
 [61] &quot;ENSG00000132017&quot; &quot;ENSG00000132530&quot; &quot;ENSG00000132591&quot;
 [64] &quot;ENSG00000132640&quot; &quot;ENSG00000132781&quot; &quot;ENSG00000133789&quot;
 [67] &quot;ENSG00000133983&quot; &quot;ENSG00000134627&quot; &quot;ENSG00000135249&quot;
 [70] &quot;ENSG00000136710&quot; &quot;ENSG00000136811&quot; &quot;ENSG00000137185&quot;
 [73] &quot;ENSG00000137936&quot; &quot;ENSG00000138095&quot; &quot;ENSG00000138378&quot;
 [76] &quot;ENSG00000139117&quot; &quot;ENSG00000139132&quot; &quot;ENSG00000139193&quot;
 [79] &quot;ENSG00000139874&quot; &quot;ENSG00000140043&quot; &quot;ENSG00000143067&quot;
 [82] &quot;ENSG00000143554&quot; &quot;ENSG00000143753&quot; &quot;ENSG00000144362&quot;
 [85] &quot;ENSG00000144857&quot; &quot;ENSG00000146477&quot; &quot;ENSG00000149489&quot;
 [88] &quot;ENSG00000149499&quot; &quot;ENSG00000150977&quot; &quot;ENSG00000151893&quot;
 [91] &quot;ENSG00000152219&quot; &quot;ENSG00000154146&quot; &quot;ENSG00000154814&quot;
 [94] &quot;ENSG00000155393&quot; &quot;ENSG00000155926&quot; &quot;ENSG00000156787&quot;
 [97] &quot;ENSG00000158423&quot; &quot;ENSG00000159128&quot; &quot;ENSG00000159314&quot;
[100] &quot;ENSG00000159445&quot; &quot;ENSG00000159674&quot; &quot;ENSG00000160401&quot;
[103] &quot;ENSG00000160767&quot; &quot;ENSG00000160783&quot; &quot;ENSG00000160972&quot;
[106] &quot;ENSG00000161526&quot; &quot;ENSG00000162591&quot; &quot;ENSG00000162650&quot;
[109] &quot;ENSG00000162714&quot; &quot;ENSG00000162878&quot; &quot;ENSG00000164054&quot;
[112] &quot;ENSG00000164088&quot; &quot;ENSG00000164674&quot; &quot;ENSG00000165792&quot;
[115] &quot;ENSG00000166508&quot; &quot;ENSG00000166833&quot; &quot;ENSG00000166839&quot;
[118] &quot;ENSG00000167077&quot; &quot;ENSG00000167720&quot; &quot;ENSG00000168589&quot;
[121] &quot;ENSG00000169992&quot; &quot;ENSG00000170456&quot; &quot;ENSG00000170906&quot;
[124] &quot;ENSG00000171703&quot; &quot;ENSG00000172264&quot; &quot;ENSG00000172660&quot;
[127] &quot;ENSG00000172878&quot; &quot;ENSG00000174175&quot; &quot;ENSG00000174516&quot;
[130] &quot;ENSG00000175130&quot; &quot;ENSG00000175264&quot; &quot;ENSG00000175581&quot;
[133] &quot;ENSG00000177084&quot; &quot;ENSG00000178038&quot; &quot;ENSG00000182141&quot;
[136] &quot;ENSG00000184083&quot; &quot;ENSG00000184857&quot; &quot;ENSG00000185187&quot;
[139] &quot;ENSG00000185238&quot; &quot;ENSG00000188647&quot; &quot;ENSG00000188786&quot;
[142] &quot;ENSG00000188938&quot; &quot;ENSG00000196526&quot; &quot;ENSG00000198015&quot;
[145] &quot;ENSG00000198843&quot; &quot;ENSG00000198862&quot; &quot;ENSG00000205707&quot;
[148] &quot;ENSG00000206538&quot; &quot;ENSG00000213516&quot; &quot;ENSG00000213903&quot;
[151] &quot;ENSG00000215039&quot; &quot;ENSG00000222046&quot; &quot;ENSG00000223609&quot;
[154] &quot;ENSG00000227671&quot; &quot;ENSG00000230055&quot; &quot;ENSG00000237883&quot;
[157] &quot;ENSG00000241852&quot; &quot;ENSG00000242500&quot; &quot;ENSG00000243056&quot;
[160] &quot;ENSG00000245573&quot; &quot;ENSG00000249124&quot; &quot;ENSG00000253368&quot;
[163] &quot;ENSG00000253982&quot; &quot;ENSG00000254445&quot; &quot;ENSG00000258210&quot;
[166] &quot;ENSG00000258461&quot; &quot;ENSG00000258644&quot; &quot;ENSG00000258653&quot;</code></pre>
<pre class="r"><code># Interactions between 3 species kidneys and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val &gt; FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val &gt; FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val &gt; FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val &gt; FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val &gt; FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val &gt; FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val &gt; FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val &lt; FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val &lt; FDR_level), ]))</code></pre>
<pre><code> [1] &quot;ENSG00000066923&quot; &quot;ENSG00000070371&quot; &quot;ENSG00000074356&quot;
 [4] &quot;ENSG00000095066&quot; &quot;ENSG00000100170&quot; &quot;ENSG00000100490&quot;
 [7] &quot;ENSG00000100612&quot; &quot;ENSG00000102595&quot; &quot;ENSG00000105825&quot;
[10] &quot;ENSG00000105928&quot; &quot;ENSG00000106123&quot; &quot;ENSG00000107611&quot;
[13] &quot;ENSG00000110079&quot; &quot;ENSG00000110675&quot; &quot;ENSG00000112667&quot;
[16] &quot;ENSG00000115556&quot; &quot;ENSG00000120594&quot; &quot;ENSG00000122121&quot;
[19] &quot;ENSG00000123352&quot; &quot;ENSG00000125895&quot; &quot;ENSG00000126562&quot;
[22] &quot;ENSG00000130558&quot; &quot;ENSG00000130997&quot; &quot;ENSG00000133059&quot;
[25] &quot;ENSG00000133250&quot; &quot;ENSG00000137960&quot; &quot;ENSG00000138134&quot;
[28] &quot;ENSG00000139289&quot; &quot;ENSG00000143801&quot; &quot;ENSG00000144136&quot;
[31] &quot;ENSG00000148175&quot; &quot;ENSG00000150672&quot; &quot;ENSG00000151491&quot;
[34] &quot;ENSG00000154265&quot; &quot;ENSG00000161574&quot; &quot;ENSG00000162241&quot;
[37] &quot;ENSG00000166860&quot; &quot;ENSG00000167778&quot; &quot;ENSG00000168395&quot;
[40] &quot;ENSG00000172296&quot; &quot;ENSG00000172331&quot; &quot;ENSG00000173567&quot;
[43] &quot;ENSG00000174799&quot; &quot;ENSG00000176046&quot; &quot;ENSG00000177082&quot;
[46] &quot;ENSG00000181396&quot; &quot;ENSG00000182983&quot; &quot;ENSG00000185015&quot;
[49] &quot;ENSG00000187231&quot; &quot;ENSG00000187741&quot; &quot;ENSG00000196456&quot;
[52] &quot;ENSG00000205517&quot; &quot;ENSG00000213494&quot; &quot;ENSG00000214128&quot;
[55] &quot;ENSG00000239521&quot; &quot;ENSG00000254528&quot; &quot;ENSG00000257453&quot;</code></pre>
</div>
<div id="adaptive-shrinkage-on-interactions-in-all-3-species-kidneys-versus-all-other-tissues" class="section level3">
<h3>Adaptive shrinkage on interactions in all 3 species’ kidneys versus all other tissues</h3>
<pre class="r"><code># Prepare the data for ASH
tests &lt;- colnames(fit1$coefficients)
tests</code></pre>
<pre><code>[1] &quot;HC_HG&quot; &quot;HR_HG&quot; &quot;CR_HG&quot;</code></pre>
<pre class="r"><code>results &lt;- vector(length = length(tests), mode = &quot;list&quot;)
names(results) &lt;- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] &lt;- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit1, coef = test)
  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code># Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, &quot;results-limma-voom-ash-interactions.rds&quot;))

kidney_combined_interactions &lt;- as.data.frame(cbind(results[[&quot;HC_HG&quot;]]$svalue, 
                               results[[&quot;HR_HG&quot;]]$svalue, 
                               results[[&quot;CR_HG&quot;]]$svalue))

dim(kidney_combined_interactions)</code></pre>
<pre><code>[1] 12184     3</code></pre>
<pre class="r"><code>colnames(kidney_combined_interactions) &lt;- c(&quot;HC_KG&quot;, &quot;HR_KG&quot;, &quot;CR_KG&quot;)</code></pre>
</div>
<div id="number-of-interactions-using-fsr" class="section level3">
<h3>Number of interactions using FSR</h3>
<pre class="r"><code># Heart
# Human versus Chimp

dim(heart_combined_interactions[which(heart_combined_interactions[,1] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 203   3</code></pre>
<pre class="r"><code># Human versus Rhesus
dim(heart_combined_interactions[which(heart_combined_interactions[,2] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 1165    3</code></pre>
<pre class="r"><code># Chimp versus Rhesus
dim(heart_combined_interactions[which(heart_combined_interactions[,3] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 705   3</code></pre>
<pre class="r"><code># Liver
# Human versus Chimp

dim(liver_combined_interactions[which(liver_combined_interactions[,1] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 107   3</code></pre>
<pre class="r"><code># Human versus Rhesus
dim(liver_combined_interactions[which(liver_combined_interactions[,2] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 910   3</code></pre>
<pre class="r"><code># Chimp versus Rhesus
dim(liver_combined_interactions[which(liver_combined_interactions[,3] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 547   3</code></pre>
<pre class="r"><code># Lung
# Human versus Chimp

dim(lung_combined_interactions[which(lung_combined_interactions[,1] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 113   3</code></pre>
<pre class="r"><code># Human versus Rhesus
dim(lung_combined_interactions[which(lung_combined_interactions[,2] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 1049    3</code></pre>
<pre class="r"><code># Chimp versus Rhesus
dim(lung_combined_interactions[which(lung_combined_interactions[,3] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 994   3</code></pre>
<pre class="r"><code>#  Kidney
# Human versus Chimp

dim(kidney_combined_interactions[which(kidney_combined_interactions[,1] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 113   3</code></pre>
<pre class="r"><code># Human versus Rhesus
dim(kidney_combined_interactions[which(kidney_combined_interactions[,2] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 470   3</code></pre>
<pre class="r"><code># Chimp versus Rhesus
dim(kidney_combined_interactions[which(kidney_combined_interactions[,3] &lt; FSR_level), ]) </code></pre>
<pre><code>[1] 361   3</code></pre>
<pre class="r"><code># Combine the svalue from interactions
three_species_all_tissue_interactions &lt;- cbind(heart_combined_interactions, liver_combined_interactions, lung_combined_interactions, kidney_combined_interactions)

rownames(three_species_all_tissue_interactions) &lt;- rownames(cpm.voom.cyclic$E)
dim(three_species_all_tissue_interactions)</code></pre>
<pre><code>[1] 12184    12</code></pre>
<pre class="r"><code># Set FDR

FDR_level &lt;- 0.01
FSR_level &lt;- 0.01

# Interactions between 3 species hearts and other tissues

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &lt; FSR_level &amp; three_species_all_tissue_interactions[,2] &lt; FSR_level &amp; three_species_all_tissue_interactions[,3] &lt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &gt; FSR_level &amp; three_species_all_tissue_interactions[,6] &gt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &gt; FSR_level &amp; three_species_all_tissue_interactions[,9] &gt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &gt; FSR_level &amp; three_species_all_tissue_interactions[,12] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000182923&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species &lt;- as.factor(make_exp_df$species)
levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus&quot;)
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Species&quot;, &quot;Tissue&quot;)
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species) + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Interactions between 3 species livers and other tissues

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &gt; FSR_level &amp; three_species_all_tissue_interactions[,3] &gt; FSR_level &amp; three_species_all_tissue_interactions[,4] &lt; FSR_level &amp; three_species_all_tissue_interactions[,5] &lt; FSR_level &amp; three_species_all_tissue_interactions[,6] &lt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &gt; FSR_level &amp; three_species_all_tissue_interactions[,9] &gt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &gt; FSR_level &amp; three_species_all_tissue_interactions[,12] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000183077&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species &lt;- as.factor(make_exp_df$species)
levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus&quot;)
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Species&quot;, &quot;Tissue&quot;)
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species) + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-29-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Interactions between 3 species lungs and other tissues


nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &gt; FSR_level &amp; three_species_all_tissue_interactions[,3] &gt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &gt; FSR_level &amp; three_species_all_tissue_interactions[,6] &gt; FSR_level &amp; three_species_all_tissue_interactions[,7] &lt; FSR_level &amp; three_species_all_tissue_interactions[,8] &lt; FSR_level &amp; three_species_all_tissue_interactions[,9] &lt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &gt; FSR_level &amp; three_species_all_tissue_interactions[,12] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code># Interactions between 3 species kidneys and other tissues


nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &gt; FSR_level &amp; three_species_all_tissue_interactions[,3] &gt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &gt; FSR_level &amp; three_species_all_tissue_interactions[,6] &gt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &gt; FSR_level &amp; three_species_all_tissue_interactions[,9] &gt; FSR_level &amp; three_species_all_tissue_interactions[,10] &lt; FSR_level &amp; three_species_all_tissue_interactions[,11] &lt; FSR_level &amp; three_species_all_tissue_interactions[,12] &lt; FSR_level), ])</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Interactions between 3 species hearts and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &lt; FSR_level &amp; three_species_all_tissue_interactions[,3] &lt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &gt; FSR_level &amp; three_species_all_tissue_interactions[,6] &gt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &gt; FSR_level &amp; three_species_all_tissue_interactions[,9] &gt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &gt; FSR_level &amp; three_species_all_tissue_interactions[,12] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 251</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000004487&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species &lt;- as.factor(make_exp_df$species)
levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus&quot;)
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Species&quot;, &quot;Tissue&quot;)
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species) + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-29-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Interactions between 3 species livers and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &gt; FSR_level &amp; three_species_all_tissue_interactions[,3] &gt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &lt; FSR_level &amp; three_species_all_tissue_interactions[,6] &lt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &gt; FSR_level &amp; three_species_all_tissue_interactions[,9] &gt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &gt; FSR_level &amp; three_species_all_tissue_interactions[,12] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 182</code></pre>
<pre class="r"><code># Interactions between 3 species lungs and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &gt; FSR_level &amp; three_species_all_tissue_interactions[,3] &gt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &gt; FSR_level &amp; three_species_all_tissue_interactions[,6] &gt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &lt; FSR_level &amp; three_species_all_tissue_interactions[,9] &lt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &gt; FSR_level &amp; three_species_all_tissue_interactions[,12] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 291</code></pre>
<pre class="r"><code># Interactions between 3 species kidneys and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] &gt; FSR_level &amp; three_species_all_tissue_interactions[,2] &gt; FSR_level &amp; three_species_all_tissue_interactions[,3] &gt; FSR_level &amp; three_species_all_tissue_interactions[,4] &gt; FSR_level &amp; three_species_all_tissue_interactions[,5] &gt; FSR_level &amp; three_species_all_tissue_interactions[,6] &gt; FSR_level &amp; three_species_all_tissue_interactions[,7] &gt; FSR_level &amp; three_species_all_tissue_interactions[,8] &gt; FSR_level &amp; three_species_all_tissue_interactions[,9] &gt; FSR_level &amp; three_species_all_tissue_interactions[,10] &gt; FSR_level &amp; three_species_all_tissue_interactions[,11] &lt; FSR_level &amp; three_species_all_tissue_interactions[,12] &lt; FSR_level), ])</code></pre>
<pre><code>[1] 78</code></pre>
</div>
</div>
<div id="combining-humans-and-chimps-to-have-a-great-ape-versus-rhesus-tissues-set-of-interactions-figure-3" class="section level2">
<h2>Combining humans and chimps to have a Great Ape versus Rhesus tissues set of interactions (Figure 3)</h2>
<div id="hearts" class="section level3">
<h3>Hearts</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Human&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Heart&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_heart&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept       Rhesus        Heart          RIN Rhesus_heart 
        47.0         16.0         11.0        368.2          4.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;-  -0.02032154
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

GA_HG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(GA_HG[which(GA_HG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 233   7</code></pre>
<pre class="r"><code>dim(GA_HG[which(GA_HG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 168   7</code></pre>
<pre class="r"><code>dim(GA_HG[which(GA_HG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 233   7</code></pre>
<pre class="r"><code>dim(GA_HG[which(GA_HG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 509   7</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000007384&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
#levels(make_exp_df$Species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus&quot;)
colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~make_exp_df[,2])  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;) </code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-30-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # Extract limma results
  results[[&quot;GA_Heart&quot;]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[&quot;GA_Heart&quot;]] &lt;- cbind(results[[&quot;GA_Heart&quot;]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

GA_HG_svalue &lt;- results[[&quot;GA_Heart&quot;]]
nrow(GA_HG_svalue[which(GA_HG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 405</code></pre>
<pre class="r"><code>nrow(GA_HG_svalue[which(GA_HG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 293</code></pre>
<pre class="r"><code>nrow(GA_HG_svalue[which(GA_HG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 405</code></pre>
<pre class="r"><code>nrow(GA_HG_svalue[which(GA_HG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 1050</code></pre>
<pre class="r"><code>write.csv(results[[&quot;GA_Heart&quot;]], &quot;../data/GA_heart.csv&quot;)</code></pre>
</div>
<div id="livers" class="section level3">
<h3>Livers</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Human&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be liver versus non-liver

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Liver&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_liver&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept       Rhesus        Liver          RIN Rhesus_liver 
        47.0         16.0         12.0        368.2          4.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;-  -0.008945556
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)


# Determine the number of statistically signficant interactions

GA_LiG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(GA_LiG[which(GA_LiG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 264   7</code></pre>
<pre class="r"><code>dim(GA_LiG[which(GA_LiG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 180   7</code></pre>
<pre class="r"><code>dim(GA_LiG[which(GA_LiG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 264   7</code></pre>
<pre class="r"><code>dim(GA_LiG[which(GA_LiG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 545   7</code></pre>
<pre class="r"><code>  # Extract limma results
  results[[&quot;GA_Liver&quot;]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[&quot;GA_Liver&quot;]] &lt;- cbind(results[[&quot;GA_Liver&quot;]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


GA_LiG_svalue &lt;- results[[&quot;GA_Liver&quot;]]
nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 309</code></pre>
<pre class="r"><code>nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 227</code></pre>
<pre class="r"><code>nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 309</code></pre>
<pre class="r"><code>nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 703</code></pre>
</div>
<div id="lungs" class="section level3">
<h3>Lungs</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Human&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Lung&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_lung&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>  Intercept      Rhesus        Lung         RIN Rhesus_lung 
       47.0        16.0        12.0       368.2         4.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;-  -0.01548285
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)


# Determine the number of statistically signficant interactions

GA_LuG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(GA_LuG[which(GA_LuG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 82  7</code></pre>
<pre class="r"><code>dim(GA_LuG[which(GA_LuG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 46  7</code></pre>
<pre class="r"><code>dim(GA_LuG[which(GA_LuG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 82  7</code></pre>
<pre class="r"><code>dim(GA_LuG[which(GA_LuG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 270   7</code></pre>
<pre class="r"><code>  # Extract limma results
  results[[&quot;GA_Lung&quot;]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[&quot;GA_Lung&quot;]] &lt;- cbind(results[[&quot;GA_Lung&quot;]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

GA_LuG_svalue &lt;- results[[&quot;GA_Lung&quot;]]
nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 174</code></pre>
<pre class="r"><code>nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 105</code></pre>
<pre class="r"><code>nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 174</code></pre>
<pre class="r"><code>nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 590</code></pre>
</div>
<div id="kidneys" class="section level3">
<h3>Kidneys</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Human&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Kidney&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_kidney&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>    Intercept        Rhesus        Kidney           RIN Rhesus_kidney 
         47.0          16.0          12.0         368.2           4.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;-  -0.05037098
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

GA_KG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(GA_KG[which(GA_KG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 56  7</code></pre>
<pre class="r"><code>dim(GA_KG[which(GA_KG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 44  7</code></pre>
<pre class="r"><code>dim(GA_KG[which(GA_KG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 56  7</code></pre>
<pre class="r"><code>dim(GA_KG[which(GA_KG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 127   7</code></pre>
<pre class="r"><code>  # Extract limma results
  results[[&quot;GA_Kidney&quot;]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[&quot;GA_Kidney&quot;]] &lt;- cbind(results[[&quot;GA_Kidney&quot;]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

GA_KG_svalue &lt;- results[[&quot;GA_Kidney&quot;]]

nrow(GA_KG_svalue[which(GA_KG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 58</code></pre>
<pre class="r"><code>nrow(GA_KG_svalue[which(GA_KG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 47</code></pre>
<pre class="r"><code>nrow(GA_KG_svalue[which(GA_KG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 58</code></pre>
<pre class="r"><code>nrow(GA_KG_svalue[which(GA_KG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 104</code></pre>
<pre class="r"><code>GA_values &lt;- cbind(GA_HG_svalue, GA_LiG_svalue, GA_LuG_svalue, GA_KG_svalue)</code></pre>
<pre class="r"><code># Find tissue specific

# bind s values together

GA_svalue &lt;- cbind(GA_HG_svalue$svalue, GA_LiG_svalue$svalue, GA_LuG_svalue$svalue, GA_KG_svalue$svalue)
rownames(GA_svalue) &lt;- rownames(cpm_12184)
# Heart tissue specific
nrow(GA_svalue[which(GA_svalue[,1] &lt; FSR_level &amp; GA_svalue[,2] &gt; FSR_level &amp; GA_svalue[,3] &gt; FSR_level &amp; GA_svalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 377</code></pre>
<pre class="r"><code># Liver tissue specific
nrow(GA_svalue[which(GA_svalue[,1] &gt; FSR_level &amp; GA_svalue[,2] &lt; FSR_level &amp; GA_svalue[,3] &gt; FSR_level &amp; GA_svalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 283</code></pre>
<pre class="r"><code># Lung tissue specific
nrow(GA_svalue[which(GA_svalue[,1] &gt; FSR_level &amp; GA_svalue[,2] &gt; FSR_level &amp; GA_svalue[,3] &lt; FSR_level &amp; GA_svalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 161</code></pre>
<pre class="r"><code># Kidney tissue specific
nrow(GA_svalue[which(GA_svalue[,1] &gt; FSR_level &amp; GA_svalue[,2] &gt; FSR_level &amp; GA_svalue[,3] &gt; FSR_level &amp; GA_svalue[,4] &lt; FSR_level), ])</code></pre>
<pre><code>[1] 57</code></pre>
<pre class="r"><code># 377, 283, 161, 57

exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000001561&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
#levels(make_exp_df$species) &lt;- c(&quot;Great Ape&quot;, &quot;Rhesus&quot;)
colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~make_exp_df[,2])  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;) </code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="great-ape-tissue-specific-using-fdr" class="section level3">
<h3>Great Ape Tissue Specific (using FDR)</h3>
<pre class="r"><code># Set FDR level

FDR_level &lt;- 0.05

GA_BHpvalue &lt;- cbind(GA_HG[,6], GA_LiG[,6], GA_LuG[,6], GA_KG[,6])

# Interactions between GA hearts and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &lt; FDR_level &amp; GA_BHpvalue[,2] &gt; FDR_level &amp; GA_BHpvalue[,3] &gt; FDR_level &amp; GA_BHpvalue[,4] &gt; FDR_level), ])</code></pre>
<pre><code>[1] 447</code></pre>
<pre class="r"><code># Interactions between GA livers and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &gt; FDR_level &amp; GA_BHpvalue[,2] &lt; FDR_level &amp; GA_BHpvalue[,3] &gt; FDR_level &amp; GA_BHpvalue[,4] &gt; FDR_level), ])</code></pre>
<pre><code>[1] 475</code></pre>
<pre class="r"><code># Interactions between GA lungs and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &gt; FDR_level &amp; GA_BHpvalue[,2] &gt; FDR_level &amp; GA_BHpvalue[,3] &lt; FDR_level &amp; GA_BHpvalue[,4] &gt; FDR_level), ])</code></pre>
<pre><code>[1] 231</code></pre>
<pre class="r"><code># Interactions between GA kidneys and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &gt; FDR_level &amp; GA_BHpvalue[,2] &gt; FDR_level &amp; GA_BHpvalue[,3] &gt; FDR_level &amp; GA_BHpvalue[,4] &lt; FDR_level), ])</code></pre>
<pre><code>[1] 102</code></pre>
</div>
</div>
<div id="combining-chimps-and-rhesus-to-have-a-human-versus-non-human-tissues-set-of-interactions" class="section level2">
<h2>Combining chimps and rhesus to have a human versus non-human tissues set of interactions</h2>
<div id="hearts-1" class="section level3">
<h3>Hearts</h3>
<pre class="r"><code># Reassign species to be human versus non-human (GA)

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Rhesus&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Heart&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_heart&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept       Rhesus        Heart          RIN Rhesus_heart 
        47.0         15.0         11.0        368.2          3.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus &lt;-  -0.01741244
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit.consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit.consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

NH_HG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(NH_HG[which(NH_HG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 211   7</code></pre>
<pre class="r"><code>dim(NH_HG[which(NH_HG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 27  7</code></pre>
<pre class="r"><code>dim(NH_HG[which(NH_HG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 57  7</code></pre>
<pre class="r"><code>dim(NH_HG[which(NH_HG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 211   7</code></pre>
<pre class="r"><code>  # Extract limma results
  results[[test]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

NH_HG_svalue &lt;- results[[test]]
nrow(NH_HG_svalue[which(NH_HG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 200</code></pre>
<pre class="r"><code>nrow(NH_HG_svalue[which(NH_HG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 117</code></pre>
<pre class="r"><code>nrow(NH_HG_svalue[which(NH_HG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 200</code></pre>
<pre class="r"><code>nrow(NH_HG_svalue[which(NH_HG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 733</code></pre>
</div>
<div id="livers-1" class="section level3">
<h3>Livers</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Rhesus&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart

# Reassign tissues to be liver versus non-liver

tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Liver&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_liver&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>   Intercept       Rhesus        Liver          RIN Rhesus_liver 
        47.0         15.0         12.0        368.2          4.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus &lt;-  -0.01239498
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit.consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit.consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)


# Determine the number of statistically signficant interactions

NH_LiG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(NH_LiG[which(NH_LiG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 61  7</code></pre>
<pre class="r"><code>dim(NH_LiG[which(NH_LiG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 18  7</code></pre>
<pre class="r"><code>dim(NH_LiG[which(NH_LiG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 19  7</code></pre>
<pre class="r"><code>dim(NH_LiG[which(NH_LiG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 61  7</code></pre>
<pre class="r"><code>  # Extract limma results
  results[[test]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


NH_LiG_svalue &lt;- results[[test]]
nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 29</code></pre>
<pre class="r"><code>nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 19</code></pre>
<pre class="r"><code>nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 29</code></pre>
<pre class="r"><code>nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 86</code></pre>
</div>
<div id="lungs-1" class="section level3">
<h3>Lungs</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species) 
GA_species &lt;- gsub(&quot;Rhesus&quot;, &quot;GA&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart


tissue_group &lt;- gsub(&quot;kidney&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Lung&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_lung&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>  Intercept      Rhesus        Lung         RIN Rhesus_lung 
       47.0        15.0        12.0       368.2         4.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
#corfit$consensus &lt;-  -0.5891937
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

#cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

NH_LuG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(NH_LuG[which(NH_LuG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 20  7</code></pre>
<pre class="r"><code>dim(NH_LuG[which(NH_LuG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 2 7</code></pre>
<pre class="r"><code>dim(NH_LuG[which(NH_LuG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 2 7</code></pre>
<pre class="r"><code>dim(NH_LuG[which(NH_LuG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 20  7</code></pre>
<pre class="r"><code>  results[[test]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

NH_LuG_svalue &lt;- results[[test]]
nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 173</code></pre>
</div>
<div id="kidneys-1" class="section level3">
<h3>Kidneys</h3>
<pre class="r"><code># Reassign species to be GA versus not GA

GA_species &lt;- gsub(&quot;Chimp&quot;, &quot;Non human&quot;, species) 
GA_species &lt;- gsub(&quot;Rhesus&quot;, &quot;Non human&quot;, GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group &lt;- gsub(&quot;lung&quot;, &quot;Grouped&quot;, tissue) 
tissue_group &lt;- gsub(&quot;heart&quot;, &quot;Grouped&quot;, tissue_group) 
tissue_group &lt;- gsub(&quot;liver&quot;, &quot;Grouped&quot;, tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design &lt;- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] &lt;- &quot;Intercept&quot;
colnames(design)[2] &lt;- &quot;Rhesus&quot;
colnames(design)[3] &lt;- &quot;Kidney&quot;
colnames(design)[4] &lt;- &quot;RIN&quot;
colnames(design)[5] &lt;- &quot;Rhesus_kidney&quot;

# Look at the number of samples in each column 
colSums(design)</code></pre>
<pre><code>    Intercept        Rhesus        Kidney           RIN Rhesus_kidney 
         47.0          32.0          12.0         368.2           8.0 </code></pre>
<pre class="r"><code>cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;)

#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus &lt;- -0.05638723
#corfit$consensus &lt;- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit.consensus)</code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run linear model with new design matrix

fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit.consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

NH_KG =topTable(fit.cyclic.norm, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
dim(NH_KG[which(NH_KG$adj.P.Val &lt; FDR_level), ]) </code></pre>
<pre><code>[1] 11  7</code></pre>
<pre class="r"><code>dim(NH_KG[which(NH_KG$adj.P.Val &lt; 0.005), ]) </code></pre>
<pre><code>[1] 6 7</code></pre>
<pre class="r"><code>dim(NH_KG[which(NH_KG$adj.P.Val &lt; 0.01), ]) </code></pre>
<pre><code>[1] 8 7</code></pre>
<pre class="r"><code>dim(NH_KG[which(NH_KG$adj.P.Val &lt; 0.05), ]) </code></pre>
<pre><code>[1] 11  7</code></pre>
<pre class="r"><code>  # Extract limma results
  results[[test]] &lt;- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash &lt;- run_ash(fit.cyclic.norm, coef = 5)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>  results[[test]] &lt;- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

NH_KG_svalue &lt;- results[[test]]

nrow(NH_KG_svalue[which(NH_KG_svalue$svalue &lt; FSR_level),])</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>nrow(NH_KG_svalue[which(NH_KG_svalue$svalue &lt; 0.005),])</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code>nrow(NH_KG_svalue[which(NH_KG_svalue$svalue &lt; 0.01),])</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>nrow(NH_KG_svalue[which(NH_KG_svalue$svalue &lt; 0.05),])</code></pre>
<pre><code>[1] 10</code></pre>
<pre class="r"><code>exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000240230&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, GA_species, tissue), stringsAsFactors = F)
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
levels(make_exp_df$GA_species) &lt;- c(&quot;Human&quot;, &quot;Non-Human&quot;)
colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;Heart&quot;,&quot;Kidney&quot;,&quot;Liver&quot;, &quot;Lung&quot;)) + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;COX19 (ENSG00000240230)&quot;) </code></pre>
<p><img src="figure/DE_RIN_Plus_ASH.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="humannon-human-tissue-specific-using-fdr" class="section level3">
<h3>Human/Non-Human Tissue Specific (using FDR)</h3>
<pre class="r"><code># Set FDR level

FDR_level &lt;- 0.01
FSR_level &lt;- 0.01

GA_BHpvalue &lt;- cbind(NH_HG[,6], NH_LiG[,6], NH_LuG[,6], NH_KG[,6])

# Interactions between GA hearts and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &lt; FSR_level &amp; GA_BHpvalue[,2] &gt; FSR_level &amp; GA_BHpvalue[,3] &gt; FSR_level &amp; GA_BHpvalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 57</code></pre>
<pre class="r"><code># Interactions between GA livers and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &gt; FSR_level &amp; GA_BHpvalue[,2] &lt; FSR_level &amp; GA_BHpvalue[,3] &gt; FSR_level &amp; GA_BHpvalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 19</code></pre>
<pre class="r"><code># Interactions between GA lungs and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &gt; FSR_level &amp; GA_BHpvalue[,2] &gt; FSR_level &amp; GA_BHpvalue[,3] &lt; FSR_level &amp; GA_BHpvalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code># Interactions between GA kidneys and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] &gt; FSR_level &amp; GA_BHpvalue[,2] &gt; FSR_level &amp; GA_BHpvalue[,3] &gt; FSR_level &amp; GA_BHpvalue[,4] &lt; FSR_level), ])</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code># 57, 19, 2, 8</code></pre>
<pre class="r"><code># Find tissue specific

FSR_level &lt;- 0.005
# bind s values together

NH_svalue &lt;- cbind(NH_HG_svalue$svalue, NH_LiG_svalue$svalue, NH_LuG_svalue$svalue, NH_KG_svalue$svalue)
rownames(NH_svalue) &lt;- rownames(NH_HG_svalue)
# Heart tissue specific
nrow(NH_svalue[which(NH_svalue[,1] &lt; FSR_level &amp; NH_svalue[,2] &gt; FSR_level &amp; NH_svalue[,3] &gt; FSR_level &amp; NH_svalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 117</code></pre>
<pre class="r"><code>NH_svalue[which(NH_svalue[,1] &lt; FSR_level &amp; NH_svalue[,2] &gt; FSR_level &amp; NH_svalue[,3] &gt; FSR_level &amp; NH_svalue[,4] &gt; FSR_level), ]</code></pre>
<pre><code>                        [,1]        [,2]       [,3]      [,4]
ENSG00000004897 1.341246e-03 0.507172895 0.14059397 0.9966071
ENSG00000009307 5.868660e-05 0.555130872 0.41799702 0.9952206
ENSG00000011304 4.045729e-03 0.478251889 0.52323510 0.9947140
ENSG00000031823 3.708979e-03 0.724277353 0.05548814 0.9964124
ENSG00000040341 4.993823e-03 0.540691386 0.47178264 0.9951596
ENSG00000048028 2.044478e-03 0.602286567 0.51078280 0.9924441
ENSG00000048991 2.231198e-03 0.186690638 0.52890582 0.9953879
ENSG00000056097 3.169971e-05 0.231419720 0.33891932 0.9959395
ENSG00000062485 1.997902e-03 0.476627386 0.46273676 0.9911144
ENSG00000064726 5.483875e-04 0.686847443 0.31538152 0.9935962
ENSG00000079785 3.899840e-03 0.708271377 0.29034016 0.9921186
ENSG00000085377 4.528282e-03 0.678029749 0.21931079 0.9890439
ENSG00000086015 3.852273e-03 0.625661973 0.43041084 0.9933651
ENSG00000086200 2.709648e-03 0.710595310 0.09597739 0.9965366
ENSG00000088247 2.285030e-03 0.514968996 0.41862845 0.9963722
ENSG00000101019 1.585913e-03 0.652720615 0.40929762 0.9850649
ENSG00000101473 3.313639e-03 0.431560495 0.41343189 0.9961908
ENSG00000102054 4.361721e-03 0.681119821 0.38016422 0.9854476
ENSG00000102893 2.514721e-06 0.515217239 0.32263520 0.9806187
ENSG00000103266 1.762963e-03 0.613068978 0.31524433 0.9962136
ENSG00000104320 1.697212e-04 0.685532808 0.10173457 0.9925004
ENSG00000104671 1.061569e-03 0.575111446 0.26343109 0.9952701
ENSG00000105063 2.443249e-04 0.339276103 0.50712774 0.9933312
ENSG00000107771 3.757141e-03 0.594267557 0.38855539 0.9941164
ENSG00000108094 1.196013e-03 0.623693886 0.35757272 0.9927910
ENSG00000108424 3.693528e-04 0.720028801 0.02016526 0.9953650
ENSG00000108515 4.198067e-05 0.601381446 0.49052561 0.9694115
ENSG00000110429 3.805106e-03 0.178377442 0.48580380 0.9963002
ENSG00000110841 2.858999e-04 0.561990360 0.39636053 0.9895970
ENSG00000111142 1.902312e-03 0.464468467 0.32935537 0.9911431
ENSG00000113312 4.097825e-03 0.690910413 0.25875360 0.9959819
ENSG00000114316 3.514574e-03 0.571119244 0.40757257 0.9864773
ENSG00000115234 4.757561e-03 0.495809951 0.36203997 0.9965209
ENSG00000123106 2.655684e-03 0.668893541 0.40732120 0.9852532
ENSG00000126226 3.612264e-03 0.325855678 0.45761600 0.9959751
ENSG00000126457 3.947124e-03 0.708642873 0.40662053 0.9910563
ENSG00000126581 3.042041e-03 0.631091811 0.37674696 0.9918797
ENSG00000126777 1.107683e-03 0.617796816 0.32316283 0.9948236
ENSG00000126858 1.972982e-04 0.623054866 0.28913379 0.9944271
ENSG00000130559 2.600588e-03 0.440825215 0.25594258 0.9956841
ENSG00000130638 4.149389e-03 0.723420447 0.17394497 0.9938724
ENSG00000130703 5.366004e-06 0.715987132 0.31989072 0.9353969
ENSG00000131966 8.394079e-04 0.722500178 0.44618647 0.9543174
ENSG00000132359 4.643179e-03 0.613517100 0.44412865 0.9847224
ENSG00000132912 2.931106e-03 0.178822167 0.34295070 0.9965603
ENSG00000132953 9.426814e-05 0.554252424 0.32763963 0.9953224
ENSG00000135040 1.807343e-03 0.498128943 0.31574675 0.9930538
ENSG00000136504 7.901077e-04 0.494814414 0.38164270 0.9931961
ENSG00000136758 1.071643e-04 0.710448126 0.04679796 0.9856911
ENSG00000137101 4.307869e-03 0.641436890 0.16869480 0.9962712
ENSG00000138032 1.442455e-03 0.713654754 0.03848889 0.9961397
ENSG00000138138 1.853602e-03 0.630830494 0.14494873 0.9876115
ENSG00000138463 2.764208e-03 0.690922862 0.23704874 0.9909775
ENSG00000140497 4.202613e-03 0.674798103 0.44301175 0.9856036
ENSG00000141556 4.584347e-04 0.686258452 0.06547097 0.9857452
ENSG00000144566 9.322166e-04 0.719531764 0.11639597 0.9911221
ENSG00000144579 5.940818e-04 0.439388136 0.32944080 0.9955035
ENSG00000144744 1.339149e-05 0.638489454 0.33328323 0.9936756
ENSG00000144746 1.539682e-03 0.601694402 0.41622096 0.9859728
ENSG00000145349 3.206743e-03 0.629826663 0.44140330 0.9912985
ENSG00000146833 3.097016e-03 0.563680238 0.47302669 0.9907044
ENSG00000147439 1.719830e-03 0.660032125 0.37324247 0.9949257
ENSG00000153113 7.392560e-04 0.594733635 0.45655750 0.9787782
ENSG00000154832 4.934800e-03 0.720362595 0.07493833 0.9964307
ENSG00000155313 9.753288e-04 0.706588766 0.30565703 0.9719086
ENSG00000155324 2.338379e-03 0.711466315 0.20721886 0.9836711
ENSG00000155363 5.020473e-04 0.580278931 0.32091188 0.9940371
ENSG00000156471 1.675534e-03 0.624354084 0.43954889 0.9920873
ENSG00000157500 4.875319e-03 0.675216055 0.19900384 0.9959635
ENSG00000157540 2.494591e-03 0.360919088 0.51559322 0.9917852
ENSG00000160299 1.950298e-03 0.718821372 0.09217259 0.9951023
ENSG00000160948 3.260817e-03 0.620195082 0.17767630 0.9966241
ENSG00000161217 2.875137e-03 0.673926396 0.23630501 0.9942285
ENSG00000162889 5.095893e-05 0.651130985 0.42226789 0.9838360
ENSG00000163681 4.816949e-03 0.624581607 0.42666447 0.9937149
ENSG00000163806 1.555952e-06 0.646333516 0.24340068 0.9937672
ENSG00000164597 2.547752e-03 0.601350096 0.35468041 0.9947415
ENSG00000164823 1.151693e-03 0.588589997 0.06071917 0.9961934
ENSG00000164975 8.870023e-04 0.480969529 0.38716090 0.9944621
ENSG00000167110 3.660569e-03 0.454829726 0.06121773 0.9840041
ENSG00000169851 3.243831e-04 0.683550423 0.44848994 0.9782405
ENSG00000170448 6.209123e-11 0.595396771 0.24808748 0.9105664
ENSG00000171853 4.255379e-03 0.432531895 0.44840099 0.9959920
ENSG00000172531 6.886823e-04 0.699515073 0.01952882 0.9966303
ENSG00000172939 1.239203e-03 0.245800170 0.52356906 0.9937058
ENSG00000174132 4.699691e-03 0.536858681 0.42308996 0.9918495
ENSG00000174446 4.586116e-03 0.302369236 0.39956268 0.9960705
ENSG00000175582 2.818576e-03 0.644590933 0.09691484 0.9963284
ENSG00000176978 1.825165e-04 0.689631530 0.33589870 0.8933580
ENSG00000178950 2.986819e-03 0.637872178 0.50331326 0.9833152
ENSG00000178982 7.124343e-05 0.414398003 0.03195681 0.9934498
ENSG00000179912 2.184183e-03 0.008864768 0.48775550 0.9966476
ENSG00000180228 3.415878e-03 0.704073056 0.27811527 0.9958077
ENSG00000182923 3.151716e-03 0.564965972 0.31140175 0.9947963
ENSG00000183814 1.558773e-04 0.705496925 0.20737423 0.9951248
ENSG00000184162 8.284712e-05 0.635157923 0.03770730 0.9964181
ENSG00000184792 1.016652e-03 0.697712239 0.43938567 0.9745661
ENSG00000184900 1.630808e-03 0.637161667 0.14648244 0.9738488
ENSG00000185122 1.491719e-03 0.702863146 0.23555871 0.9946086
ENSG00000185418 3.465362e-03 0.628837147 0.47180215 0.9715309
ENSG00000185928 4.414797e-03 0.611656422 0.52216065 0.9888837
ENSG00000198898 2.084722e-05 0.468395889 0.16071523 0.9953082
ENSG00000205560 4.139112e-04 0.618441478 0.50278391 0.9890922
ENSG00000213079 1.244557e-04 0.322831725 0.04397226 0.9965800
ENSG00000213639 1.392033e-03 0.472396241 0.35954752 0.9937832
ENSG00000234431 4.472038e-03 0.645558091 0.42362737 0.9854870
ENSG00000235883 1.291178e-03 0.665459480 0.33210852 0.9942840
ENSG00000235945 3.365326e-03 0.687237880 0.07372435 0.9953636
ENSG00000237298 3.563128e-03 0.683333324 0.47504790 0.9892401
ENSG00000238045 3.995015e-03 0.471891016 0.22017976 0.9962796
ENSG00000254413 2.390302e-03 0.599739126 0.49214430 0.9893667
ENSG00000254690 1.408091e-04 0.575855897 0.24681581 0.9924549
ENSG00000255302 2.137707e-03 0.474069925 0.43856734 0.9943384
ENSG00000256439 2.443173e-03 0.581558601 0.09313728 0.9930493
ENSG00000257270 2.108045e-04 0.589184691 0.20087198 0.9959733
ENSG00000258199 6.413490e-04 0.714441540 0.01608618 0.9964633
ENSG00000258830 2.091641e-03 0.719443956 0.11320860 0.9960886</code></pre>
<pre class="r"><code># Liver tissue specific
nrow(NH_svalue[which(NH_svalue[,1] &gt; FSR_level &amp; NH_svalue[,2] &lt; FSR_level &amp; NH_svalue[,3] &gt; FSR_level &amp; NH_svalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>[1] 19</code></pre>
<pre class="r"><code># Lung tissue specific
nrow(NH_svalue[which(NH_svalue[,1] &gt; FSR_level &amp; NH_svalue[,2] &gt; FSR_level &amp; NH_svalue[,3] &lt; FSR_level &amp; NH_svalue[,4] &gt; FSR_level), ])</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code># Kidney tissue specific
nrow(NH_svalue[which(NH_svalue[,1] &gt; FSR_level &amp; NH_svalue[,2] &gt; FSR_level &amp; NH_svalue[,3] &gt; FSR_level &amp; NH_svalue[,4] &lt; FSR_level), ])</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code># 199, 28, 5, 5</code></pre>
<pre class="r"><code># Great ape by tissue
exp &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000196943&quot;, rownames(cpm_12184)), ]))
make_exp_df &lt;- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
#make_exp_df &lt;- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] &lt;- as.numeric(make_exp_df[,1])
make_exp_df[,2] &lt;- as.character(make_exp_df[,2])

#levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus macaque&quot;)

make_exp_df$species[make_exp_df$species ==&quot;1&quot;] &lt;- &quot;Chimpanzee&quot;
make_exp_df$species[make_exp_df$species ==&quot;2&quot;] &lt;- &quot;Human&quot;
make_exp_df$species[make_exp_df$species ==&quot;3&quot;] &lt;-  &quot;Rh. macaque&quot;



colnames(make_exp_df) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

p2 &lt;- ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized \n Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;H&quot;,&quot;K&quot;,&quot;Li&quot;, &quot;Lu&quot;)) +  theme_bw() + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;SYPL (ENSG00000008282)&quot;) 

#ENSG00000009307


#ENSG00000147416

# Great ape by tissue
exp1 &lt;- t(as.data.frame(cpm_12184[grepl(&quot;ENSG00000147416&quot;, rownames(cpm_12184)), ]))
make_exp_df1 &lt;- as.data.frame(cbind(exp1, species, tissue), stringsAsFactors = F)
#make_exp_df &lt;- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df1[,1] &lt;- as.numeric(make_exp_df1[,1])
make_exp_df1[,2] &lt;- as.character(make_exp_df1[,2])

#levels(make_exp_df$species) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;, &quot;Rhesus macaque&quot;)

make_exp_df1$species[make_exp_df1$species ==&quot;1&quot;] &lt;- &quot;Chimpanzee&quot;
make_exp_df1$species[make_exp_df1$species ==&quot;2&quot;] &lt;- &quot;Human&quot;
make_exp_df1$species[make_exp_df1$species ==&quot;3&quot;] &lt;-  &quot;Rh. macaque&quot;


colnames(make_exp_df1) &lt;- c(&quot;Normalized Gene Expression&quot;, &quot;Species&quot;, &quot;Tissue&quot;)

p1 &lt;- ggplot(make_exp_df1, aes(x=as.factor(Tissue), y=make_exp_df1[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab(&quot;Tissue&quot;) + ylab(&quot;Normalized \n Gene Expression&quot;) + scale_x_discrete(labels=c(&quot;H&quot;,&quot;K&quot;,&quot;Li&quot;, &quot;Lu&quot;)) +  theme_bw() + bjp + theme(legend.position=&quot;none&quot;) + ggtitle(&quot;CDC27 (ENSG00000004897)&quot;) 



make_fig3 &lt;- plot_grid(p2, p1, labels = c(&quot;3A.&quot;, &quot;3B.&quot;), ncol = 1)

save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Figures/heart_figures.pdf&quot;, make_fig3,
          ncol = 1, # we&#39;re saving a grid plot of 2 columns
          nrow = 2, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 1.3
          )</code></pre>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
