---
title: "ASH_DE_RIN"
author: "Lauren Blake"
date: "June 7, 2017"
output: html_document
editor_options: 
  chunk_output_type: console
---



```{r chunk-options, include=FALSE}
source("chunk-options.R")
```


```{r}
# Load libraries

library("gplots")
library("ggplot2")
library("RColorBrewer")
library("scales")
library("edgeR")
library("R.utils")
library("plyr")
library("limma")
library("gridExtra")
library("VennDiagram")
source("functions.R")
library(ashr)
library(cowplot)

# Add function for making ggplot2 figures look good (from Bryan Pavlovic)

bjp<-
theme(
  panel.border = element_rect(colour = "black", fill = NA, size = 2),
  plot.title = element_text(size = 16, face = "bold", hjust = 0.5),
  axis.text.y =  element_text(size = 14,face = "bold",color = "black"),
  axis.text.x =  element_text(size = 14,face = "bold",color = "black"),
  axis.title.y = element_text(size = 14,face = "bold"),
  axis.title.x = element_text(size = 14,face = "bold"),
  legend.text = element_text(size = 14,face = "bold"),
  legend.title = element_text(size = 14,face = "bold"),
  strip.text.x = element_text(size = 14,face = "bold"),
  strip.text.y = element_text(size = 14,face = "bold"),
  strip.background = element_rect(colour = "black", size = 2))

# Set directory to save the data

data_dir <- "../data"

# Load colors 

colors <- colorRampPalette(c(brewer.pal(9, "Blues")[1],brewer.pal(9, "Blues")[9]))(100)
pal <- c(brewer.pal(9, "Set1"), brewer.pal(8, "Set2"), brewer.pal(12, "Set3"))

# Retrieve RIN score for each sample
RNA_seq_info <- read.csv("../../../Reg_Evo_Primates/data/RNA_seq_info.csv")
RIN <- as.data.frame(RNA_seq_info[,22])
RIN <- as.matrix(RIN)
#RIN <- as.data.frame(RIN[-1,])
#RIN <- as.matrix(RIN)
colnames(RIN) <- c("RIN")

# Retrieve sample information
samples <- read.delim("../../../Reg_Evo_Primates/data/Sample_info_RNAseq_limma.txt")

# Eliminate H1H
samples <- samples[-17,]
#samples <- samples[-1,]
dim(samples)

# Make labels

labels <- paste(samples$Species, samples$Tissue, sep=".")
```

## Pairwise comparisons


```{r}
# Load count data

counts_genes_in_cutoff <- read.delim("../../../Reg_Evo_Primates/data/counts_12184.txt")
cpm_12184 <- read.delim("../../../Reg_Evo_Primates/data/cpm_12184.txt")

# TMM 

dge_in_cutoff <- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff <- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff <- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)
hist(cpm_in_cutoff, xlab = "Log2(CPM)", main = "Log2(CPM) values for genes meeting the filtering criteria", breaks = 100 )

# Make a design matrix

tissue <- samples$Tissue
species <- samples$Species
design <- model.matrix(~ tissue*species + RIN)
colnames(design)[1] <- "Intercept"

# Rename columns of the contrast matrix

colnames(design)[2] <- "kidney"
colnames(design)[3] <- "liver"
colnames(design)[4] <- "lung"
colnames(design)[5] <- "human"
colnames(design)[6] <- "rhesus"

colnames(design)[8] <- "kidneyHuman"
colnames(design)[9] <- "liverHuman"
colnames(design)[10] <- "lungHuman"
colnames(design)[11] <- "kidneyRhesus"
colnames(design)[12] <- "liverRhesus"
colnames(design)[13] <- "lungRhesus"



# Look at the number of samples in each column 
colSums(design)

# Voom with individual as a random variable

#cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=T)

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus <- 0.2197275

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit.consensus)

```







### Fit the linear model

```{r}
fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit.consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# MA Plots
## If 'MA' is an 'MArrayLM' object, then the plot is a fitted model MA-plot in which the estimated coefficient is on the y-axis and the average A-value is on the x-axis.                

#limma::plotMA(fit.cyclic.norm, array=1, xlab="average coefficient", ylab="estimated coefficient")

## - Potential caveat: variances could be different between human, chimp and rhesus (see Gordon Smyth email, 7 June 2013).                                                               
##  We look at the standard error for each condition                                                    
hist(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma, breaks=100)
hist(log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma), breaks=100)
boxplot(log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma))
## This seems to be pretty comparable between conditions. The human heart is higher, probably because of H1H missing and H3H with a bit strange behavior                                     
stderror <- log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma)
boxplot(list(stderror[,1:4], stderror[,5:8], stderror[,9:12]))
## A bit higher for human, and a bit lower for rhesus                                                                                                                                    
boxplot(list(stderror[,2:4], stderror[,6:8], stderror[,8:12])) ## excluding heart samples  
```

```{r}
# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm1 <- makeContrasts(HvC_Liver = human + liverHuman, 
                     HvC_Lung = human + lungHuman, 
                     HvC_Heart = human, 
                     HvC_Kidney = human + kidneyHuman, 
                     HvR_Liver = human + liverHuman - (rhesus + liverRhesus), 
                     HvR_Lung = human + lungHuman - (rhesus + lungRhesus), 
                     HvR_Heart = human - rhesus, 
                     HvR_Kidney = human + kidneyHuman - (rhesus + kidneyRhesus), 
                     CvR_Liver = -(rhesus + liverRhesus),  
                     CvR_Lung = -(rhesus + lungRhesus), 
                     CvR_Heart = -(rhesus), 
                     CvR_Kidney = -(rhesus + kidneyRhesus), 
                     H_HeartvLi = (human) - (human + liver + liverHuman), 
                     H_HeartvLu = (human) - (human + lung + lungHuman), 
                     H_HeartvK = (human) - (human + kidney + kidneyHuman), 
                     H_LivLu = (human + liver + liverHuman) - (human + lung + lungHuman), 
                     H_LivK = (human + liver + liverHuman) - (human + kidney + kidneyHuman),  
                     H_LuvK = (human + lung + lungHuman) - (human + kidney + kidneyHuman), 
                     C_HeartvLi = -(liver), 
                     C_HeartvLu = -(lung), 
                     C_HeartvK = -(kidney), 
                     C_LivLu = liver - lung, 
                     C_LivK = liver - kidney,  
                     C_LuvK = lung - kidney,  
                     R_HeartvLi = (rhesus) - (rhesus + liver + liverRhesus), 
                     R_HeartvLu = (rhesus) - (rhesus + lung + lungRhesus), 
                     R_HeartvK = (rhesus) - (rhesus + kidney + kidneyRhesus), 
                     R_LivLu = (rhesus + liver + liverRhesus) - (rhesus + lung + lungRhesus), 
                     R_LivK = (rhesus + liver + liverRhesus) - (rhesus + kidney + kidneyRhesus),  
                     R_LuvK = (rhesus + lung + lungRhesus) - (rhesus + kidney + kidneyRhesus), 
                     levels = design)

# Implement contrasts

contrasts_each_species <- contrasts.fit(fit.cyclic.norm, cm1)
fit1 <- eBayes(contrasts_each_species)

top3 <- list(HvC_Liver =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none"), 
             HvC_Lung =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none"),  
             HvC_Heart =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none"),  
             HvC_Kidney =topTable(fit1, coef=4, adjust="BH", number=Inf, sort.by="none"), 
             
             HvR_Liver =topTable(fit1, coef=5, adjust="BH", number=Inf, sort.by="none"), 
             HvR_Lung =topTable(fit1, coef=6, adjust="BH", number=Inf, sort.by="none"),
             HvR_Heart =topTable(fit1, coef=7, adjust="BH", number=Inf, sort.by="none"), 
             HvR_Kidney =topTable(fit1, coef=8, adjust="BH", number=Inf, sort.by="none"),  
             
             CvR_Liver =topTable(fit1, coef=9, adjust="BH", number=Inf, sort.by="none"),  
             CvR_Lung =topTable(fit1, coef=10, adjust="BH", number=Inf, sort.by="none"), 
             CvR_Heart =topTable(fit1, coef=11, adjust="BH", number=Inf, sort.by="none"), 
             CvR_Kidney =topTable(fit1, coef=12, adjust="BH", number=Inf, sort.by="none"),
             
             H_HeartvLi =topTable(fit1, coef=13, adjust="BH", number=Inf, sort.by="none"), 
             H_HeartvLu =topTable(fit1, coef=14, adjust="BH", number=Inf, sort.by="none"),  
             H_HeartvK =topTable(fit1, coef=15, adjust="BH", number=Inf, sort.by="none"),  
             H_LivLu =topTable(fit1, coef=16, adjust="BH", number=Inf, sort.by="none"), 
             H_LivK =topTable(fit1, coef=17, adjust="BH", number=Inf, sort.by="none"), 
             H_LuvK =topTable(fit1, coef=18, adjust="BH", number=Inf, sort.by="none"),
             
             C_HeartvLi =topTable(fit1, coef=19, adjust="BH", number=Inf, sort.by="none"), 
             C_HeartvLu =topTable(fit1, coef=20, adjust="BH", number=Inf, sort.by="none"),  
             C_HeartvK =topTable(fit1, coef=21, adjust="BH", number=Inf, sort.by="none"),  
             C_LivLu =topTable(fit1, coef=22, adjust="BH", number=Inf, sort.by="none"), 
             C_LivK =topTable(fit1, coef=23, adjust="BH", number=Inf, sort.by="none"), 
             C_LuvK =topTable(fit1, coef=24, adjust="BH", number=Inf, sort.by="none"),
             
             R_HeartvLi =topTable(fit1, coef=25, adjust="BH", number=Inf, sort.by="none"), 
             R_HeartvLu =topTable(fit1, coef=26, adjust="BH", number=Inf, sort.by="none"),  
             R_HeartvK =topTable(fit1, coef=27, adjust="BH", number=Inf, sort.by="none"),  
             R_LivLu =topTable(fit1, coef=28, adjust="BH", number=Inf, sort.by="none"), 
             R_LivK =topTable(fit1, coef=29, adjust="BH", number=Inf, sort.by="none"), 
             R_LuvK =topTable(fit1, coef=30, adjust="BH", number=Inf, sort.by="none") )


# Set FDR level at 1% 

FDR_level <- 0.01

## DE between HvC

HvC_Liver =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none")
dim(HvC_Liver[which(HvC_Liver$adj.P.Val < FDR_level), ]) 

HvC_Lung =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
dim(HvC_Lung[which(HvC_Lung$adj.P.Val < FDR_level), ]) 

HvC_Heart =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none") 
dim(HvC_Heart[which(HvC_Heart$adj.P.Val < FDR_level), ]) 

HvC_Kidney =topTable(fit1, coef=4, adjust="BH", number=Inf, sort.by="none") 
dim(HvC_Kidney[which(HvC_Kidney$adj.P.Val < FDR_level), ]) 



mylist <- list()
mylist[["Liver"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val < FDR_level]
mylist[["Lung"]] <-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val < FDR_level]
mylist[["Heart"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val < FDR_level]
mylist[["Kidney"]] <- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val < FDR_level]


# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Humans and Chimps (FDR 1%)", cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold", cat.cex = 1.2, main.cex = 1.5)
grid.draw(Four_comp)

## DE between HvR

HvR_Liver =topTable(fit1, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(HvR_Liver[which(HvR_Liver$adj.P.Val < FDR_level), ]) 

HvR_Lung =topTable(fit1, coef=6, adjust="BH", number=Inf, sort.by="none")
dim(HvR_Lung[which(HvR_Lung$adj.P.Val < FDR_level), ]) 
             
HvR_Heart =topTable(fit1, coef=7, adjust="BH", number=Inf, sort.by="none") 
dim(HvR_Heart[which(HvR_Heart$adj.P.Val < FDR_level), ]) 
             
HvR_Kidney =topTable(fit1, coef=8, adjust="BH", number=Inf, sort.by="none")  
dim(HvR_Kidney[which(HvR_Kidney$adj.P.Val < FDR_level), ]) 

mylist <- list()
mylist[["Liver"]] <- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val < FDR_level]
mylist[["Lung"]] <-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val < FDR_level]
mylist[["Heart"]] <- row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val < FDR_level]
mylist[["Kidney"]] <- row.names(top3[[names(top3)[8]]])[top3[[names(top3)[8]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Humans and Rhesus (FDR 1%)", cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between CvR

CvR_Liver =topTable(fit1, coef=9, adjust="BH", number=Inf, sort.by="none")
dim(CvR_Liver[which(CvR_Liver$adj.P.Val < FDR_level), ]) 

CvR_Lung =topTable(fit1, coef=10, adjust="BH", number=Inf, sort.by="none")
dim(CvR_Lung[which(CvR_Lung$adj.P.Val < FDR_level), ]) 
             
CvR_Heart =topTable(fit1, coef=11, adjust="BH", number=Inf, sort.by="none") 
dim(CvR_Heart[which(CvR_Heart$adj.P.Val < FDR_level), ]) 
              
CvR_Kidney =topTable(fit1, coef=12, adjust="BH", number=Inf, sort.by="none")
dim(CvR_Kidney[which(CvR_Kidney$adj.P.Val < FDR_level), ]) 


mylist <- list()
mylist[["Liver"]] <- row.names(top3[[names(top3)[9]]])[top3[[names(top3)[9]]]$adj.P.Val < FDR_level]
mylist[["Lung"]] <-  row.names(top3[[names(top3)[10]]])[top3[[names(top3)[10]]]$adj.P.Val < FDR_level]
mylist[["Heart"]] <- row.names(top3[[names(top3)[11]]])[top3[[names(top3)[11]]]$adj.P.Val < FDR_level]
mylist[["Kidney"]] <- row.names(top3[[names(top3)[12]]])[top3[[names(top3)[12]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Chimps and Rhesus (FDR 1%)", cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

# DE between tissues within a species

  # Within humans
H_HeartvLi =topTable(fit1, coef=13, adjust="BH", number=Inf, sort.by="none") 
dim(H_HeartvLi[which(H_HeartvLi$adj.P.Val < FDR_level), ]) 
            
H_HeartvLu =topTable(fit1, coef=14, adjust="BH", number=Inf, sort.by="none")  
dim(H_HeartvLu[which(H_HeartvLu$adj.P.Val < FDR_level), ]) 
               
H_HeartvK =topTable(fit1, coef=15, adjust="BH", number=Inf, sort.by="none")  
dim(H_HeartvK[which(H_HeartvK$adj.P.Val < FDR_level), ]) 
               
H_LivLu =topTable(fit1, coef=16, adjust="BH", number=Inf, sort.by="none")
dim(H_LivLu[which(H_LivLu$adj.P.Val < FDR_level), ]) 
               
H_LivK =topTable(fit1, coef=17, adjust="BH", number=Inf, sort.by="none") 
dim(H_LivK[which(H_LivK$adj.P.Val < FDR_level), ]) 
               
H_LuvK =topTable(fit1, coef=18, adjust="BH", number=Inf, sort.by="none")
dim(H_LuvK[which(H_LuvK$adj.P.Val < FDR_level), ]) 
   
  # Within chimps

C_HeartvLi =topTable(fit1, coef=19, adjust="BH", number=Inf, sort.by="none") 
dim(C_HeartvLi[which(C_HeartvLi$adj.P.Val < FDR_level), ]) 
             
C_HeartvLu =topTable(fit1, coef=20, adjust="BH", number=Inf, sort.by="none")  
dim(C_HeartvLu[which(C_HeartvLu$adj.P.Val < FDR_level), ]) 
              
C_HeartvK =topTable(fit1, coef=21, adjust="BH", number=Inf, sort.by="none")  
dim(C_HeartvK[which(C_HeartvK$adj.P.Val < FDR_level), ]) 
          
C_LivLu =topTable(fit1, coef=22, adjust="BH", number=Inf, sort.by="none")
dim(C_LivLu[which(C_LivLu$adj.P.Val < FDR_level), ]) 
             
C_LivK =topTable(fit1, coef=23, adjust="BH", number=Inf, sort.by="none") 
dim(C_LivK[which(C_LivK$adj.P.Val < FDR_level), ]) 
             
C_LuvK =topTable(fit1, coef=24, adjust="BH", number=Inf, sort.by="none")
dim(C_LuvK[which(C_LuvK$adj.P.Val < FDR_level), ]) 

  # Within rhesus

R_HeartvLi =topTable(fit1, coef=25, adjust="BH", number=Inf, sort.by="none")
dim(R_HeartvLi[which(R_HeartvLi$adj.P.Val < FDR_level), ]) 
             
R_HeartvLu =topTable(fit1, coef=26, adjust="BH", number=Inf, sort.by="none") 
dim(R_HeartvLu[which(R_HeartvLu$adj.P.Val < FDR_level), ]) 
             
R_HeartvK =topTable(fit1, coef=27, adjust="BH", number=Inf, sort.by="none")  
dim(R_HeartvK[which(R_HeartvK$adj.P.Val < FDR_level), ]) 
              
R_LivLu =topTable(fit1, coef=28, adjust="BH", number=Inf, sort.by="none")
dim(R_LivLu[which(R_LivLu$adj.P.Val < FDR_level), ]) 
             
R_LivK =topTable(fit1, coef=29, adjust="BH", number=Inf, sort.by="none") 
dim(R_LivK[which(R_LivK$adj.P.Val < FDR_level), ]) 
             
R_LuvK =topTable(fit1, coef=30, adjust="BH", number=Inf, sort.by="none")
dim(R_LuvK[which(R_LuvK$adj.P.Val < FDR_level), ]) 


## DE between Heart and Liver

mylist <- list()
mylist[["Human"]] <- row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val < FDR_level]
mylist[["Chimp"]] <-  row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val < FDR_level]
mylist[["Rhesus"]] <- row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Heart and Liver (FDR 1%)", cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Heart and Lung

mylist <- list()
mylist[["Human"]] <- row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val < FDR_level]
mylist[["Chimp"]] <-  row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val < FDR_level]
mylist[["Rhesus"]] <- row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Heart and Lung (FDR 1%)", cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## DE between Heart and Kidney

mylist <- list()
mylist[["Human"]] <- row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val < FDR_level]
mylist[["Chimp"]] <-  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val < FDR_level]
mylist[["Rhesus"]] <- row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Heart and Kidney (FDR 1%)", cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## DE between Liver and Lung

mylist <- list()
mylist[["Human"]] <- row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val < FDR_level]
mylist[["Chimp"]] <-  row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val < FDR_level]
mylist[["Rhesus"]] <- row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Liver and Lung (FDR 1%)", cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Liver and Kidney

mylist <- list()
mylist[["Human"]] <- row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val < FDR_level]
mylist[["Chimp"]] <-  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val < FDR_level]
mylist[["Rhesus"]] <- row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Liver and Kidney (FDR 1%)", cex=1.5, lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Lung and Kidney

mylist <- list()
mylist[["Human"]] <- row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val < FDR_level]
mylist[["Chimp"]] <-  row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val < FDR_level]
mylist[["Rhesus"]] <- row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val < FDR_level]

# Make 
dev.off()
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between Lung and Kidney (FDR 1%)", cex=1.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

```

## Make Venn Diagrams for directionality using ASH

In this section, we are going to detect genes that have differences in the same direction (false sign rate = 1%). 

### Run ASH

```{r}
# Prepare the data for ASH
tests <- colnames(fit1$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Perform multiple testing correction with adaptive shrinkage (ASH) 
 #
 # x - object MArrayLM from eBayes output
 # coef - coefficient tested by eBayes

run_ash <- function(x, coef){
  #stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients),
  #             length(unique(x$df.total) == 1))
  result <- ash(betahat = x$coefficients[, coef], sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post), df = x$df.total[1])
  return(result)
}

get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = test)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}

# Save results from analysis with limma and ash.
saveRDS(results, file.path(data_dir, "combined-limma.rds"))

# Get the FSR for all genes in all comparisons

#results <- readRDS("../data/combined-limma.rds")
#write.csv(results, file = "../data/pairwise_FSR.csv")
```




### Find where the genes are upregulated between tissues

```{r}
# Set false sign rate
FDR_level <- 0.05
FSR_level <- 0.005

# FDR upregulated / downregulated

test <- "HvC_Liver"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvC_Liver"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "HvC_Lung"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvC_Lung"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "HvC_Heart"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvC_Heart"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "HvC_Kidney"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvC_Kidney"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])




test <- "HvR_Liver"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvR_Liver"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "HvR_Lung"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvR_Lung"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "HvR_Heart"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvR_Heart"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "HvR_Kidney"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "HvR_Kidney"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

  
  

test <- "CvR_Liver"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "CvR_Liver"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "CvR_Lung"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "CvR_Lung"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "CvR_Heart"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "CvR_Heart"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "CvR_Kidney"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "CvR_Kidney"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

  


               
 
test <- "H_HeartvLi"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "H_HeartvLi"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "H_HeartvLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "H_HeartvLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "H_HeartvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "H_HeartvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "H_LivLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "H_LivLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "H_LivK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "H_LivK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "H_LuvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "H_LuvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])



test <- "C_HeartvLi"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "C_HeartvLi"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "C_HeartvLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "C_HeartvLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "C_HeartvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "C_HeartvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "C_LivLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "C_LivLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "C_LivK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "C_LivK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "C_LuvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "C_LuvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])



test <- "R_HeartvLi"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "R_HeartvLi"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "R_HeartvLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "R_HeartvLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "R_HeartvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "R_HeartvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "R_LivLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "R_LivLu"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "R_LivK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "R_LivK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])

test <- "R_LuvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC > 0])

test <- "R_LuvK"
length((results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$logFC < 0])



# FSR

FSR_level <- 0.01


test <- "HvC_Liver"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvC_Liver"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "HvC_Lung"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvC_Lung"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "HvC_Heart"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvC_Heart"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "HvC_Kidney"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvC_Kidney"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])




test <- "HvR_Liver"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvR_Liver"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "HvR_Lung"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvR_Lung"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "HvR_Heart"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvR_Heart"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "HvR_Kidney"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "HvR_Kidney"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

  
  

test <- "CvR_Liver"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "CvR_Liver"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "CvR_Lung"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "CvR_Lung"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "CvR_Heart"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "CvR_Heart"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "CvR_Kidney"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "CvR_Kidney"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

  


               
 
test <- "H_HeartvLi"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "H_HeartvLi"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "H_HeartvLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "H_HeartvLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "H_HeartvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "H_HeartvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "H_LivLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "H_LivLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "H_LivK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "H_LivK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "H_LuvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "H_LuvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])



test <- "C_HeartvLi"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "C_HeartvLi"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "C_HeartvLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "C_HeartvLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "C_HeartvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "C_HeartvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "C_LivLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "C_LivLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "C_LivK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "C_LivK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "C_LuvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "C_LuvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])



test <- "R_HeartvLi"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "R_HeartvLi"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "R_HeartvLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "R_HeartvLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "R_HeartvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "R_HeartvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "R_LivLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "R_LivLu"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "R_LivK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "R_LivK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])

test <- "R_LuvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0])

test <- "R_LuvK"
length((results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0])







# Heart versus kidney FSR

mylist <- list()

test <- "H_HeartvK"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_HeartvK"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_HeartvK"
mylist[["Rhesus Macaque"]] <-  (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]


# Find ones that are DE in hearts and kidneys in chimpanzees and rhesus but not humans

setdiff(intersect(mylist[["Rhesus Macaque"]], mylist[["Chimpanzee"]]), mylist[["Human"]])





# Make Venn Diagram
dev.off()
Heart_kidney <- venn.diagram(mylist, filename= NULL, main="Upregulated in Heart compared to Kidney (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_kidney)

# ENSG00000004468

big_list <- union(mylist[["Chimpanzee"]], mylist[["Rhesus Macaque"]])
mylist[["Human"]][1:200] %in% big_list

exp <- t(as.data.frame(cpm_12184[grepl(mylist[["Human"]][176], rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
make_exp_df <- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
make_exp_df[,2] <- as.character(make_exp_df[,2])

#levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus macaque")

make_exp_df$species[make_exp_df$species =="1"] <- "Chimpanzee"
make_exp_df$species[make_exp_df$species =="2"] <- "Human"
make_exp_df$species[make_exp_df$species =="3"] <-  "Rhesus macaque"


colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized Gene Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) +  theme_bw() + bjp + theme(legend.position="none") + ggtitle("PQBP1 (ENSG00000102103)") 




exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000006740", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
#make_exp_df <- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
make_exp_df[,2] <- as.character(make_exp_df[,2])

#levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus macaque")

make_exp_df$species[make_exp_df$species =="1"] <- "Chimpanzee"
make_exp_df$species[make_exp_df$species =="2"] <- "Human"
make_exp_df$species[make_exp_df$species =="3"] <-  "Rhesus macaque"


colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized Gene Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) +  theme_bw() + bjp + theme(legend.position="none") + ggtitle("TNNT2 (ENSG00000118194)") 





exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000118194", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
#make_exp_df <- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
make_exp_df[,2] <- as.character(make_exp_df[,2])

#levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus macaque")

make_exp_df$species[make_exp_df$species =="1"] <- "Chimpanzee"
make_exp_df$species[make_exp_df$species =="2"] <- "Human"
make_exp_df$species[make_exp_df$species =="3"] <-  "Rhesus macaque"


colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized Gene Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) +  theme_bw() + bjp + theme(legend.position="none") + ggtitle("TNNT2 (ENSG00000118194)") 




# Heart versus lung

mylist <- list()

test <- "H_HeartvLu"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_HeartvLu"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_HeartvLu"
mylist[["Rhesus Macaque"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Heart_lung <- venn.diagram(mylist, filename= NULL, main="Upregulated in Heart compared to Lung (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_lung)


# Liver versus lung

mylist <- list()

test <- "H_LivLu"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_LivLu"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_LivLu"
mylist[["Rhesus Macaque"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Liver_lung <- venn.diagram(mylist, filename= NULL, main="Upregulated in Liver compared to Lung (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_lung)


# Liver versus kidney

mylist <- list()

test <- "H_LivK"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_LivK"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_LivK"
mylist[["Rhesus Macaque"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Liver_kidney <- venn.diagram(mylist, filename= NULL, main="Upregulated in Liver compared to Kidney (FSR 1%)",  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_kidney)


# Lung versus Kidney

mylist <- list()

test <- "H_LuvK"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_LuvK"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_LuvK"
mylist[["Rhesus Macaque"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Lung_kidney <- venn.diagram(mylist, filename= NULL, main="Upregulated in Lung compared to Kidney (FSR 1%)",  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Lung_kidney)


```

## Upregulated and downregulated pairwise figure (Supplementary Figure)

```{r}
upreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]
  return(upreg_two)
}

FSR_level <- 0.01
upreg_human_heart_kidney <- upreg_pairwise_comp("H_HeartvK", FSR_level)
upreg_chimp_heart_kidney <- upreg_pairwise_comp("C_HeartvK", FSR_level)
upreg_rhesus_heart_kidney <- upreg_pairwise_comp("R_HeartvK", FSR_level)

upreg_human_heart_liver <- upreg_pairwise_comp("H_HeartvLi", FSR_level)
upreg_chimp_heart_liver <- upreg_pairwise_comp("C_HeartvLi", FSR_level)
upreg_rhesus_heart_liver <- upreg_pairwise_comp("R_HeartvLi", FSR_level)

upreg_human_heart_lung <- upreg_pairwise_comp("H_HeartvLu", FSR_level)
upreg_chimp_heart_lung <- upreg_pairwise_comp("C_HeartvLu", FSR_level)
upreg_rhesus_heart_lung <- upreg_pairwise_comp("R_HeartvLu", FSR_level)

upreg_human_liver_lung <- upreg_pairwise_comp("H_LivLu", FSR_level)
upreg_chimp_liver_lung <- upreg_pairwise_comp("C_LivLu", FSR_level)
upreg_rhesus_liver_lung <- upreg_pairwise_comp("R_LivLu", FSR_level)

upreg_human_liver_kidney <- upreg_pairwise_comp("H_LivK", FSR_level)
upreg_chimp_liver_kidney <- upreg_pairwise_comp("C_LivK", FSR_level)
upreg_rhesus_liver_kidney <- upreg_pairwise_comp("R_LivK", FSR_level)

upreg_human_lung_kidney <- upreg_pairwise_comp("H_LuvK", FSR_level)
upreg_chimp_lung_kidney <- upreg_pairwise_comp("C_LuvK", FSR_level)
upreg_rhesus_lung_kidney <- upreg_pairwise_comp("R_LuvK", FSR_level)


downreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]
  return(upreg_two)
}


downreg_human_heart_kidney <- downreg_pairwise_comp("H_HeartvK", FSR_level)
downreg_chimp_heart_kidney <- downreg_pairwise_comp("C_HeartvK", FSR_level)
downreg_rhesus_heart_kidney <- downreg_pairwise_comp("R_HeartvK", FSR_level)

downreg_human_heart_liver <- downreg_pairwise_comp("H_HeartvLi", FSR_level)
downreg_chimp_heart_liver <- downreg_pairwise_comp("C_HeartvLi", FSR_level)
downreg_rhesus_heart_liver <- downreg_pairwise_comp("R_HeartvLi", FSR_level)

downreg_human_heart_lung <- downreg_pairwise_comp("H_HeartvLu", FSR_level)
downreg_chimp_heart_lung <- downreg_pairwise_comp("C_HeartvLu", FSR_level)
downreg_rhesus_heart_lung <- downreg_pairwise_comp("R_HeartvLu", FSR_level)

downreg_human_liver_lung <- downreg_pairwise_comp("H_LivLu", FSR_level)
downreg_chimp_liver_lung <- downreg_pairwise_comp("C_LivLu", FSR_level)
downreg_rhesus_liver_lung <- downreg_pairwise_comp("R_LivLu", FSR_level)

downreg_human_liver_kidney <- downreg_pairwise_comp("H_LivK", FSR_level)
downreg_chimp_liver_kidney <- downreg_pairwise_comp("C_LivK", FSR_level)
downreg_rhesus_liver_kidney <- downreg_pairwise_comp("R_LivK", FSR_level)

downreg_human_lung_kidney <- downreg_pairwise_comp("H_LuvK", FSR_level)
downreg_chimp_lung_kidney <- downreg_pairwise_comp("C_LuvK", FSR_level)
downreg_rhesus_lung_kidney <- downreg_pairwise_comp("R_LuvK", FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram <- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist <- list()
mylist[["Hu"]] <- upreg_human_tissue
mylist[["Ch"]] <- upreg_chimp_tissue
mylist[["Rh"]] <- upreg_rhesus_tissue


# Make Venn Diagram
#dev.off()
Heart_specific_up <- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold", cat.dist = 0.05)
grid.draw(Heart_specific_up)

}



svg("../data/Upregulated_heart_liver.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_liver, upreg_chimp_heart_liver, upreg_rhesus_heart_liver, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Upregulated_heart_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_lung, upreg_chimp_heart_lung, upreg_rhesus_heart_lung, NULL)
dev.off()

svg("../data/Upregulated_heart_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_kidney, upreg_chimp_heart_kidney, upreg_rhesus_heart_kidney, NULL)
dev.off()


svg("../data/Upregulated_liver_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_lung, upreg_chimp_liver_lung, upreg_rhesus_liver_lung, NULL)
dev.off()

svg("../data/Upregulated_liver_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_kidney, upreg_chimp_liver_kidney, upreg_rhesus_liver_kidney, NULL)
dev.off()

svg("../data/Upregulated_lung_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung_kidney, upreg_chimp_lung_kidney, upreg_rhesus_lung_kidney, NULL)
dev.off()



svg("../data/Downregulated_heart_liver.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_liver, downreg_chimp_heart_liver, downreg_rhesus_heart_liver, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Downregulated_heart_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_lung, downreg_chimp_heart_lung, downreg_rhesus_heart_lung, NULL)
dev.off()

svg("../data/Downregulated_heart_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_kidney, downreg_chimp_heart_kidney, downreg_rhesus_heart_kidney, NULL)
dev.off()


svg("../data/Downregulated_liver_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_lung, downreg_chimp_liver_lung, downreg_rhesus_liver_lung, NULL)
dev.off()

svg("../data/Downregulated_liver_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_kidney, downreg_chimp_liver_kidney, downreg_rhesus_liver_kidney, NULL)
dev.off()

svg("../data/Downregulated_lung_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung_kidney, downreg_chimp_lung_kidney, downreg_rhesus_lung_kidney, NULL)
dev.off()

#title <- ggdraw() + draw_label("Pairwise DE genes (FSR 1%)", fontface='bold', size = 20)
p1 <- ggdraw()+draw_image("../data/Upregulated_heart_liver.svg")+draw_figure_label("4A.", size = 16, fontface = "bold")
p2 <- ggdraw()+draw_image("../data/Downregulated_heart_liver.svg")+draw_figure_label("4B.", size = 16, fontface = "bold")

p3 <- ggdraw()+draw_image("../data/Upregulated_heart_lung.svg")+draw_figure_label("4C.", size = 16, fontface = "bold")
p4 <- ggdraw()+draw_image("../data/Downregulated_heart_lung.svg")+draw_figure_label("4D.", size = 16, fontface = "bold")

p5 <- ggdraw()+draw_image("../data/Upregulated_heart_kidney.svg")+draw_figure_label("4E.", size = 16, fontface = "bold")
p6 <- ggdraw()+draw_image("../data/Downregulated_heart_kidney.svg")+draw_figure_label("4F.", size = 16, fontface = "bold")

p7 <- ggdraw()+draw_image("../data/Upregulated_liver_lung.svg")+draw_figure_label("4G.", size = 16, fontface = "bold")
p8 <- ggdraw()+draw_image("../data/Downregulated_liver_lung.svg")+draw_figure_label("4H.", size = 16, fontface = "bold")

p9 <- ggdraw()+draw_image("../data/Upregulated_liver_kidney.svg")+draw_figure_label("4I.", size = 16, fontface = "bold")
p10 <- ggdraw()+draw_image("../data/Downregulated_liver_kidney.svg")+draw_figure_label("4J.", size = 16, fontface = "bold")

p11 <- ggdraw()+draw_image("../data/Upregulated_lung_kidney.svg")+draw_figure_label("4K.", size = 16, fontface = "bold")
p12 <- ggdraw()+draw_image("../data/Downregulated_lung_kidney.svg")+draw_figure_label("4L.", size = 16, fontface = "bold")


#p2 <- ggdraw()+draw_image("Methyl_heatmap.tiff")
eight_plots <- plot_grid(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, ncol = 4)
plot_supp_fig <- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot("../data/test.png", plot_supp_fig,
          ncol = 4, # we're saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

#save_plot("/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Supp_figures/Supplementary_Figure_4_common_pairwise.png", plot_supp_fig,
#          ncol = 4, # we're saving a grid plot of 2 columns
#          nrow = 3, # and 2 rows
#          # each individual subplot should have an aspect ratio of 1.3
#          base_aspect_ratio = 0.75
#          )

upreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FSR_level & results[[test]]$logFC > 0]
  return(upreg_two)
}

FSR_level <- 0.01
upreg_human_heart_kidney <- upreg_pairwise_comp("H_HeartvK", FSR_level)
upreg_chimp_heart_kidney <- upreg_pairwise_comp("C_HeartvK", FSR_level)
upreg_rhesus_heart_kidney <- upreg_pairwise_comp("R_HeartvK", FSR_level)

upreg_human_heart_liver <- upreg_pairwise_comp("H_HeartvLi", FSR_level)
upreg_chimp_heart_liver <- upreg_pairwise_comp("C_HeartvLi", FSR_level)
upreg_rhesus_heart_liver <- upreg_pairwise_comp("R_HeartvLi", FSR_level)

upreg_human_heart_lung <- upreg_pairwise_comp("H_HeartvLu", FSR_level)
upreg_chimp_heart_lung <- upreg_pairwise_comp("C_HeartvLu", FSR_level)
upreg_rhesus_heart_lung <- upreg_pairwise_comp("R_HeartvLu", FSR_level)

upreg_human_liver_lung <- upreg_pairwise_comp("H_LivLu", FSR_level)
upreg_chimp_liver_lung <- upreg_pairwise_comp("C_LivLu", FSR_level)
upreg_rhesus_liver_lung <- upreg_pairwise_comp("R_LivLu", FSR_level)

upreg_human_liver_kidney <- upreg_pairwise_comp("H_LivK", FSR_level)
upreg_chimp_liver_kidney <- upreg_pairwise_comp("C_LivK", FSR_level)
upreg_rhesus_liver_kidney <- upreg_pairwise_comp("R_LivK", FSR_level)

upreg_human_lung_kidney <- upreg_pairwise_comp("H_LuvK", FSR_level)
upreg_chimp_lung_kidney <- upreg_pairwise_comp("C_LuvK", FSR_level)
upreg_rhesus_lung_kidney <- upreg_pairwise_comp("R_LuvK", FSR_level)


downreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FSR_level & results[[test]]$logFC < 0]
  return(upreg_two)
}


downreg_human_heart_kidney <- downreg_pairwise_comp("H_HeartvK", FSR_level)
downreg_chimp_heart_kidney <- downreg_pairwise_comp("C_HeartvK", FSR_level)
downreg_rhesus_heart_kidney <- downreg_pairwise_comp("R_HeartvK", FSR_level)

downreg_human_heart_liver <- downreg_pairwise_comp("H_HeartvLi", FSR_level)
downreg_chimp_heart_liver <- downreg_pairwise_comp("C_HeartvLi", FSR_level)
downreg_rhesus_heart_liver <- downreg_pairwise_comp("R_HeartvLi", FSR_level)

downreg_human_heart_lung <- downreg_pairwise_comp("H_HeartvLu", FSR_level)
downreg_chimp_heart_lung <- downreg_pairwise_comp("C_HeartvLu", FSR_level)
downreg_rhesus_heart_lung <- downreg_pairwise_comp("R_HeartvLu", FSR_level)

downreg_human_liver_lung <- downreg_pairwise_comp("H_LivLu", FSR_level)
downreg_chimp_liver_lung <- downreg_pairwise_comp("C_LivLu", FSR_level)
downreg_rhesus_liver_lung <- downreg_pairwise_comp("R_LivLu", FSR_level)

downreg_human_liver_kidney <- downreg_pairwise_comp("H_LivK", FSR_level)
downreg_chimp_liver_kidney <- downreg_pairwise_comp("C_LivK", FSR_level)
downreg_rhesus_liver_kidney <- downreg_pairwise_comp("R_LivK", FSR_level)

downreg_human_lung_kidney <- downreg_pairwise_comp("H_LuvK", FSR_level)
downreg_chimp_lung_kidney <- downreg_pairwise_comp("C_LuvK", FSR_level)
downreg_rhesus_lung_kidney <- downreg_pairwise_comp("R_LuvK", FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram <- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist <- list()
mylist[["Hu"]] <- upreg_human_tissue
mylist[["Ch"]] <- upreg_chimp_tissue
mylist[["Rh"]] <- upreg_rhesus_tissue


# Make Venn Diagram
#dev.off()
Heart_specific_up <- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold", cat.dist = 0.05)
grid.draw(Heart_specific_up)

}



svg("../data/Upregulated_heart_liver.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_liver, upreg_chimp_heart_liver, upreg_rhesus_heart_liver, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Upregulated_heart_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_lung, upreg_chimp_heart_lung, upreg_rhesus_heart_lung, NULL)
dev.off()

svg("../data/Upregulated_heart_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart_kidney, upreg_chimp_heart_kidney, upreg_rhesus_heart_kidney, NULL)
dev.off()


svg("../data/Upregulated_liver_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_lung, upreg_chimp_liver_lung, upreg_rhesus_liver_lung, NULL)
dev.off()

svg("../data/Upregulated_liver_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver_kidney, upreg_chimp_liver_kidney, upreg_rhesus_liver_kidney, NULL)
dev.off()

svg("../data/Upregulated_lung_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung_kidney, upreg_chimp_lung_kidney, upreg_rhesus_lung_kidney, NULL)
dev.off()



svg("../data/Downregulated_heart_liver.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_liver, downreg_chimp_heart_liver, downreg_rhesus_heart_liver, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Downregulated_heart_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_lung, downreg_chimp_heart_lung, downreg_rhesus_heart_lung, NULL)
dev.off()

svg("../data/Downregulated_heart_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart_kidney, downreg_chimp_heart_kidney, downreg_rhesus_heart_kidney, NULL)
dev.off()


svg("../data/Downregulated_liver_lung.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_lung, downreg_chimp_liver_lung, downreg_rhesus_liver_lung, NULL)
dev.off()

svg("../data/Downregulated_liver_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver_kidney, downreg_chimp_liver_kidney, downreg_rhesus_liver_kidney, NULL)
dev.off()

svg("../data/Downregulated_lung_kidney.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung_kidney, downreg_chimp_lung_kidney, downreg_rhesus_lung_kidney, NULL)
dev.off()

#title <- ggdraw() + draw_label("Pairwise DE genes (FSR 1%)", fontface='bold', size = 20)
p1 <- ggdraw()+draw_image("../data/Upregulated_heart_liver.svg")+draw_figure_label("4A.", size = 16, fontface = "bold")
p2 <- ggdraw()+draw_image("../data/Downregulated_heart_liver.svg")+draw_figure_label("4B.", size = 16, fontface = "bold")

p3 <- ggdraw()+draw_image("../data/Upregulated_heart_lung.svg")+draw_figure_label("4C.", size = 16, fontface = "bold")
p4 <- ggdraw()+draw_image("../data/Downregulated_heart_lung.svg")+draw_figure_label("4D.", size = 16, fontface = "bold")

p5 <- ggdraw()+draw_image("../data/Upregulated_heart_kidney.svg")+draw_figure_label("4E.", size = 16, fontface = "bold")
p6 <- ggdraw()+draw_image("../data/Downregulated_heart_kidney.svg")+draw_figure_label("4F.", size = 16, fontface = "bold")

p7 <- ggdraw()+draw_image("../data/Upregulated_liver_lung.svg")+draw_figure_label("4G.", size = 16, fontface = "bold")
p8 <- ggdraw()+draw_image("../data/Downregulated_liver_lung.svg")+draw_figure_label("4H.", size = 16, fontface = "bold")

p9 <- ggdraw()+draw_image("../data/Upregulated_liver_kidney.svg")+draw_figure_label("4I.", size = 16, fontface = "bold")
p10 <- ggdraw()+draw_image("../data/Downregulated_liver_kidney.svg")+draw_figure_label("4J.", size = 16, fontface = "bold")

p11 <- ggdraw()+draw_image("../data/Upregulated_lung_kidney.svg")+draw_figure_label("4K.", size = 16, fontface = "bold")
p12 <- ggdraw()+draw_image("../data/Downregulated_lung_kidney.svg")+draw_figure_label("4L.", size = 16, fontface = "bold")


#p2 <- ggdraw()+draw_image("Methyl_heatmap.tiff")
eight_plots <- plot_grid(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, ncol = 4)
plot_supp_fig <- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot("../data/test.png", plot_supp_fig,
          ncol = 4, # we're saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

save_plot("/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Supp_figures/Supplementary_Figure_4_common_pairwise.png", plot_supp_fig,
          ncol = 4, # we're saving a grid plot of 2 columns
         nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

```


## Upregulated and downregulated pairwise species figure (Supplementary Figure)

```{r}
# With FDR
upreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FSR_level & results[[test]]$logFC > 0]
  return(upreg_two)
}

FSR_level <- 0.05
upreg_hc_heart <- upreg_pairwise_comp("HvC_Heart", FSR_level)
upreg_hr_heart <- upreg_pairwise_comp("HvR_Heart", FSR_level)
upreg_cr_heart <- upreg_pairwise_comp("CvR_Heart", FSR_level)

upreg_hc_kidney <- upreg_pairwise_comp("HvC_Kidney", FSR_level)
upreg_hr_kidney <- upreg_pairwise_comp("HvR_Kidney", FSR_level)
upreg_cr_kidney <- upreg_pairwise_comp("CvR_Kidney", FSR_level)

upreg_hc_liver <- upreg_pairwise_comp("HvC_Liver", FSR_level)
upreg_hr_liver <- upreg_pairwise_comp("HvR_Liver", FSR_level)
upreg_cr_liver <- upreg_pairwise_comp("CvR_Liver", FSR_level)

upreg_hc_lung <- upreg_pairwise_comp("HvC_Lung", FSR_level)
upreg_hr_lung <- upreg_pairwise_comp("HvR_Lung", FSR_level)
upreg_cr_lung <- upreg_pairwise_comp("CvR_Lung", FSR_level)


downreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FSR_level & results[[test]]$logFC < 0]
  return(upreg_two)
}


downreg_hc_heart <- downreg_pairwise_comp("HvC_Heart", FSR_level)
downreg_hr_heart <- downreg_pairwise_comp("HvR_Heart", FSR_level)
downreg_cr_heart <- downreg_pairwise_comp("CvR_Heart", FSR_level)

downreg_hc_kidney <- downreg_pairwise_comp("HvC_Kidney", FSR_level)
downreg_hr_kidney <- downreg_pairwise_comp("HvR_Kidney", FSR_level)
downreg_cr_kidney <- downreg_pairwise_comp("CvR_Kidney", FSR_level)

downreg_hc_liver <- downreg_pairwise_comp("HvC_Liver", FSR_level)
downreg_hr_liver <- downreg_pairwise_comp("HvR_Liver", FSR_level)
downreg_cr_liver <- downreg_pairwise_comp("CvR_Liver", FSR_level)

downreg_hc_lung <- downreg_pairwise_comp("HvC_Lung", FSR_level)
downreg_hr_lung <- downreg_pairwise_comp("HvR_Lung", FSR_level)
downreg_cr_lung <- downreg_pairwise_comp("CvR_Lung", FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram <- function(heart, liver, lung, kidney, name_file){
mylist <- list()
mylist[["Heart"]] <- heart
mylist[["Liver"]] <- liver
mylist[["Lung"]] <- lung
mylist[["Kidney"]] <- kidney

# Make Venn Diagram
#dev.off()
Heart_specific_up <- venn.diagram(mylist,  filename = NULL, fill = pal[1:4], main.cex = 3, cat.cex = 2.7, cex=2.7, lty=1, height=2000, width=2000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold")
grid.draw(Heart_specific_up)

}



svg("../data/Upregulated_hc.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_hc_heart, upreg_hc_liver, upreg_hc_lung, upreg_hc_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Upregulated_hr.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_hr_heart, upreg_hr_liver, upreg_hr_lung, upreg_hr_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Upregulated_cr.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_cr_heart, upreg_cr_liver, upreg_cr_lung, upreg_cr_kidney, "Upregulated heart rel. to liver")
dev.off()


svg("../data/Downregulated_hc.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_hc_heart, downreg_hc_liver, downreg_hc_lung, downreg_hc_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Downregulated_hr.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_hr_heart, downreg_hr_liver, downreg_hr_lung, downreg_hr_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Downregulated_cr.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_cr_heart, downreg_cr_liver, downreg_cr_lung, downreg_cr_kidney, "Upregulated heart rel. to liver")
dev.off()



#title <- ggdraw() + draw_label("Pairwise DE genes (FSR 1%)", fontface='bold', size = 20)
p1 <- ggdraw()+draw_image("../data/Upregulated_hc.svg")+draw_figure_label("5A.", size = 15, fontface = "bold", position = "top.left")
p2 <- ggdraw()+draw_image("../data/Downregulated_hc.svg")+draw_figure_label("5B.", size = 15, fontface = "bold", position = "top.left")

p3 <- ggdraw()+draw_image("../data/Upregulated_hr.svg")+draw_figure_label("5C.", size = 15, fontface = "bold", position = "top.left")
p4 <- ggdraw()+draw_image("../data/Downregulated_hr.svg")+draw_figure_label("5D.", size = 15, fontface = "bold", position = "top.left")

p5 <- ggdraw()+draw_image("../data/Upregulated_cr.svg")+draw_figure_label("5E.", size = 15, fontface = "bold", position = "top.left")
p6 <- ggdraw()+draw_image("../data/Downregulated_cr.svg")+draw_figure_label("5F.", size = 15, fontface = "bold", position = "top.left")



#p2 <- ggdraw()+draw_image("Methyl_heatmap.tiff")
eight_plots <- plot_grid(p1, p2, p3, p4, p5, p6, ncol = 2)
plot_supp_fig <- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot("../data/test.png", plot_supp_fig,
          ncol = 3, # we're saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

save_plot("/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Supp_figures/Supplementary_Figure_5_species_pairwise.png", plot_supp_fig,
          ncol = 4, # we're saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )

# With FSR
upreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]
  return(upreg_two)
}

FSR_level <- 0.05
upreg_hc_heart <- upreg_pairwise_comp("HvC_Heart", FSR_level)
upreg_hr_heart <- upreg_pairwise_comp("HvR_Heart", FSR_level)
upreg_cr_heart <- upreg_pairwise_comp("CvR_Heart", FSR_level)

upreg_hc_kidney <- upreg_pairwise_comp("HvC_Kidney", FSR_level)
upreg_hr_kidney <- upreg_pairwise_comp("HvR_Kidney", FSR_level)
upreg_cr_kidney <- upreg_pairwise_comp("CvR_Kidney", FSR_level)

upreg_hc_liver <- upreg_pairwise_comp("HvC_Liver", FSR_level)
upreg_hr_liver <- upreg_pairwise_comp("HvR_Liver", FSR_level)
upreg_cr_liver <- upreg_pairwise_comp("CvR_Liver", FSR_level)

upreg_hc_lung <- upreg_pairwise_comp("HvC_Lung", FSR_level)
upreg_hr_lung <- upreg_pairwise_comp("HvR_Lung", FSR_level)
upreg_cr_lung <- upreg_pairwise_comp("CvR_Lung", FSR_level)


downreg_pairwise_comp <- function(testname, FSR_level){
  test <- testname
  upreg_two <- (results[[test]]$genes)[results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]
  return(upreg_two)
}


downreg_hc_heart <- downreg_pairwise_comp("HvC_Heart", FSR_level)
downreg_hr_heart <- downreg_pairwise_comp("HvR_Heart", FSR_level)
downreg_cr_heart <- downreg_pairwise_comp("CvR_Heart", FSR_level)

downreg_hc_kidney <- downreg_pairwise_comp("HvC_Kidney", FSR_level)
downreg_hr_kidney <- downreg_pairwise_comp("HvR_Kidney", FSR_level)
downreg_cr_kidney <- downreg_pairwise_comp("CvR_Kidney", FSR_level)

downreg_hc_liver <- downreg_pairwise_comp("HvC_Liver", FSR_level)
downreg_hr_liver <- downreg_pairwise_comp("HvR_Liver", FSR_level)
downreg_cr_liver <- downreg_pairwise_comp("CvR_Liver", FSR_level)

downreg_hc_lung <- downreg_pairwise_comp("HvC_Lung", FSR_level)
downreg_hr_lung <- downreg_pairwise_comp("HvR_Lung", FSR_level)
downreg_cr_lung <- downreg_pairwise_comp("CvR_Lung", FSR_level)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram <- function(heart, liver, lung, kidney, name_file){
mylist <- list()
mylist[["Heart"]] <- heart
mylist[["Liver"]] <- liver
mylist[["Lung"]] <- lung
mylist[["Kidney"]] <- kidney

# Make Venn Diagram
#dev.off()
Heart_specific_up <- venn.diagram(mylist,  filename = NULL, fill = pal[1:4], main.cex = 3, cat.cex = 2.7, cex=2.7, lty=1, height=2000, width=2000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold")
grid.draw(Heart_specific_up)

}



svg("../data/Upregulated_hc.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_hc_heart, upreg_hc_liver, upreg_hc_lung, upreg_hc_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Upregulated_hr.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_hr_heart, upreg_hr_liver, upreg_hr_lung, upreg_hr_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Upregulated_cr.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_cr_heart, upreg_cr_liver, upreg_cr_lung, upreg_cr_kidney, "Upregulated heart rel. to liver")
dev.off()


svg("../data/Downregulated_hc.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_hc_heart, downreg_hc_liver, downreg_hc_lung, downreg_hc_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Downregulated_hr.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_hr_heart, downreg_hr_liver, downreg_hr_lung, downreg_hr_kidney, "Upregulated heart rel. to liver")
dev.off()

svg("../data/Downregulated_cr.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_cr_heart, downreg_cr_liver, downreg_cr_lung, downreg_cr_kidney, "Upregulated heart rel. to liver")
dev.off()



#title <- ggdraw() + draw_label("Pairwise DE genes (FSR 1%)", fontface='bold', size = 20)
p1 <- ggdraw()+draw_image("../data/Upregulated_hc.svg")+draw_figure_label("A. Upregulated Human v Chimpanzee", size = 15, fontface = "bold", position = "top.left")
p2 <- ggdraw()+draw_image("../data/Downregulated_hc.svg")+draw_figure_label("B. Downregulated Human v Chimpanzee", size = 15, fontface = "bold", position = "top.left")

p3 <- ggdraw()+draw_image("../data/Upregulated_hr.svg")+draw_figure_label("C. Upregulated Human v Rhesus", size = 15, fontface = "bold", position = "top.left")
p4 <- ggdraw()+draw_image("../data/Downregulated_hr.svg")+draw_figure_label("D. Downregulated Human v Rhesus", size = 15, fontface = "bold", position = "top.left")

p5 <- ggdraw()+draw_image("../data/Upregulated_cr.svg")+draw_figure_label("E. Upregulated Chimpanzee v Rhesus", size = 15, fontface = "bold", position = "top.left")
p6 <- ggdraw()+draw_image("../data/Downregulated_cr.svg")+draw_figure_label("F. Downregulated Chimpanzee v Rhesus ", size = 15, fontface = "bold", position = "top.left")



#p2 <- ggdraw()+draw_image("Methyl_heatmap.tiff")
eight_plots <- plot_grid(p1, p2, p3, p4, p5, p6, ncol = 2)
plot_supp_fig <- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))


save_plot("../data/test.png", plot_supp_fig,
          ncol = 3, # we're saving a grid plot of 2 columns
          nrow = 3, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
          )
```


### Downregulated

```{r}
# Set false sign rate
FDR_level <- 0.01
FSR_level <- 0.01

# Heart versus kidney

mylist <- list()

test <- "H_HeartvK"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "C_HeartvK"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "R_HeartvK"
mylist[["Rhesus Macaque"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

# Make Venn Diagram
dev.off()
Heart_kidney <- venn.diagram(mylist, filename= NULL, main="Downregulated in Heart compared to Kidney (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_kidney)


# Heart versus liver

mylist <- list()

test <- "H_HeartvLi"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_HeartvLi"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_HeartvLi"
mylist[["Rhesus Macaque"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Heart_liver <- venn.diagram(mylist, filename= NULL, main="Upregulated in Heart compared to Liver (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_liver)


# Heart versus lung

mylist <- list()

test <- "H_HeartvLu"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_HeartvLu"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_HeartvLu"
mylist[["Rhesus Macaque"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Heart_lung <- venn.diagram(mylist, filename= NULL, main="Upregulated in Heart compared to Lung (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Heart_lung)


# Liver versus lung

mylist <- list()

test <- "H_LivLu"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_LivLu"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_LivLu"
mylist[["Rhesus Macaque"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Liver_lung <- venn.diagram(mylist, filename= NULL, main="Upregulated in Liver compared to Lung (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_lung)


# Liver versus kidney

mylist <- list()

test <- "H_LivK"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_LivK"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_LivK"
mylist[["Rhesus Macaque"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Liver_kidney <- venn.diagram(mylist, filename= NULL, main="Upregulated in Liver compared to Kidney (FSR 1%)",  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Liver_kidney)


# Lung versus Kidney

mylist <- list()

test <- "H_LuvK"
mylist[["Human"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "C_LuvK"
mylist[["Chimpanzee"]] <-   (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "R_LuvK"
mylist[["Rhesus Macaque"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Lung_kidney <- venn.diagram(mylist, filename= NULL, main="Upregulated in Lung compared to Kidney (FSR 1%)",  main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Lung_kidney)


```




```{r}
exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000000938", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species <- as.factor(make_exp_df$species)

keep <- c(1,2,5,6,9,10,13,14,17,20,21,24,25,28,29, 32, 33, 36, 37, 40, 41, 44, 45)
make_exp_df <- make_exp_df[keep,]

levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus")
colnames(make_exp_df) <- c("CPM", "Species", "Tissue")
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none") + ggtitle("TNNT2 (ENSG00000104951)") 

```



### Find where the genes are upregulated between species

```{r}
# Set false sign rate

FSR_level <- 0.01

  # DE between humans and chimpanzees

mylist <- list()

test <- "HvC_Liver"
mylist[["Liver"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "HvC_Lung"
mylist[["Lung"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "HvC_Heart"
mylist[["Heart"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "HvC_Kidney"
mylist[["Kidney"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Human_chimp_up <- venn.diagram(mylist, filename= NULL, main="Upregulated in Humans compared to Chimps (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_chimp_up)


  # DE between humans and rhesus
mylist <- list()
test <- "HvR_Liver"
mylist[["Liver"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "HvR_Lung"
mylist[["Lung"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "HvR_Heart"
mylist[["Heart"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "HvR_Kidney"
mylist[["Kidney"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Human_rhesus_up <- venn.diagram(mylist, filename= NULL, main="Upregulated in Humans compared to Rhesus (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_rhesus_up)



  # DE between chimpanzees and rhesus
mylist <- list()
test <- "CvR_Liver"
mylist[["Liver"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "CvR_Lung"
mylist[["Lung"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "CvR_Heart"
mylist[["Heart"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

test <- "CvR_Kidney"
mylist[["Kidney"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est > 0]

# Make Venn Diagram
dev.off()
Chimp_rhesus_up <- venn.diagram(mylist, filename= NULL, main="Upregulated in Chimps relative to Rhesus (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Chimp_rhesus_up)


```



### Find where the genes are downregulated between species

```{r}
# Set false sign rate

FSR_level <- 0.01

  # DE between humans and chimpanzees

mylist <- list()

test <- "HvC_Liver"
mylist[["Liver"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "HvC_Lung"
mylist[["Lung"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "HvC_Heart"
mylist[["Heart"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "HvC_Kidney"
mylist[["Kidney"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

# Make Venn Diagram
dev.off()
Human_chimp_down <- venn.diagram(mylist, filename= NULL, main="Downregulated in Humans compared to Chimps (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_chimp_down)


  # DE between humans and rhesus
mylist <- list()
test <- "HvR_Liver"
mylist[["Liver"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "HvR_Lung"
mylist[["Lung"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "HvR_Heart"
mylist[["Heart"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "HvR_Kidney"
mylist[["Kidney"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

# Make Venn Diagram
dev.off()
Human_rhesus_down <- venn.diagram(mylist, filename= NULL, main="Downregulated in Humans compared to Rhesus (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Human_rhesus_down)



  # DE between chimpanzees and rhesus
mylist <- list()
test <- "CvR_Liver"
mylist[["Liver"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "CvR_Lung"
mylist[["Lung"]] <-  (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "CvR_Heart"
mylist[["Heart"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

test <- "CvR_Kidney"
mylist[["Kidney"]] <- (results[[test]]$genes)[results[[test]]$adj.P.Val < FDR_level & results[[test]]$svalue < FSR_level & results[[test]]$beta_est < 0]

# Make Venn Diagram
dev.off()
Chimp_rhesus_down <- venn.diagram(mylist, filename= NULL, main="Downregulated in Chimps relative to Rhesus (FSR 1%)", main.cex = 2, cat.cex = 1.5, cex=2.5, fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Chimp_rhesus_down)

```


## Find genes with a tissue-specific direction (significant and upregulated in 1 versus 2, 3, 4 but not significant in 2 v 3, 4 and 3 v 4) with FSR (supplementary tables)

```{r}
# Combine the human ASH results
combined_human <- cbind(results[["H_HeartvLi"]]$svalue, results[["H_HeartvLi"]]$beta_est, results[["H_HeartvLu"]]$svalue, results[["H_HeartvLu"]]$beta_est, results[["H_HeartvK"]]$svalue, results[["H_HeartvK"]]$beta_est, results[["H_LivLu"]]$svalue, results[["H_LivLu"]]$beta_est, results[["H_LivK"]]$svalue, results[["H_LivK"]]$beta_est, results[["H_LuvK"]]$svalue, results[["H_LuvK"]]$beta_est)


#combined_human <- cbind(results[["H_HeartvLi"]]$qvalue, results[["H_HeartvLi"]]$beta_est, results[["H_HeartvLu"]]$qvalue, results[["H_HeartvLu"]]$beta_est, results[["H_HeartvK"]]$qvalue, results[["H_HeartvK"]]$beta_est, results[["H_LivLu"]]$qvalue, results[["H_LivLu"]]$beta_est, results[["H_LivK"]]$qvalue, results[["H_LivK"]]$beta_est, results[["H_LuvK"]]$qvalue, results[["H_LuvK"]]$beta_est)


dim(combined_human)
rownames(combined_human) <- rownames(cpm_12184)
  
# Combine the chimp ASH results
combined_chimp <- cbind(results[["C_HeartvLi"]]$svalue, results[["C_HeartvLi"]]$beta_est, results[["C_HeartvLu"]]$svalue, results[["C_HeartvLu"]]$beta_est, results[["C_HeartvK"]]$svalue, results[["C_HeartvK"]]$beta_est, results[["C_LivLu"]]$svalue, results[["C_LivLu"]]$beta_est, results[["C_LivK"]]$svalue, results[["C_LivK"]]$beta_est, results[["C_LuvK"]]$svalue, results[["C_LuvK"]]$beta_est)

#combined_chimp <- cbind(results[["C_HeartvLi"]]$qvalue, results[["C_HeartvLi"]]$beta_est, results[["C_HeartvLu"]]$qvalue, results[["C_HeartvLu"]]$beta_est, results[["C_HeartvK"]]$qvalue, results[["C_HeartvK"]]$beta_est, results[["C_LivLu"]]$qvalue, results[["C_LivLu"]]$beta_est, results[["C_LivK"]]$qvalue, results[["C_LivK"]]$beta_est, results[["C_LuvK"]]$qvalue, results[["C_LuvK"]]$beta_est)

dim(combined_chimp)
rownames(combined_chimp) <- rownames(cpm_12184)

# Combine the rhesus ASH results
combined_rhesus <- cbind(results[["R_HeartvLi"]]$svalue, results[["R_HeartvLi"]]$beta_est, results[["R_HeartvLu"]]$svalue, results[["R_HeartvLu"]]$beta_est, results[["R_HeartvK"]]$svalue, results[["R_HeartvK"]]$beta_est, results[["R_LivLu"]]$svalue, results[["R_LivLu"]]$beta_est, results[["R_LivK"]]$svalue, results[["R_LivK"]]$beta_est, results[["R_LuvK"]]$svalue, results[["R_LuvK"]]$beta_est)

#combined_rhesus <- cbind(results[["R_HeartvLi"]]$qvalue, results[["R_HeartvLi"]]$beta_est, results[["R_HeartvLu"]]$qvalue, results[["R_HeartvLu"]]$beta_est, results[["R_HeartvK"]]$qvalue, results[["R_HeartvK"]]$beta_est, results[["R_LivLu"]]$qvalue, results[["R_LivLu"]]$beta_est, results[["R_LivK"]]$qvalue, results[["R_LivK"]]$beta_est, results[["R_LuvK"]]$qvalue, results[["R_LuvK"]]$beta_est)

dim(combined_rhesus)
rownames(combined_rhesus) <- rownames(cpm_12184)

FSR_level <- 0.01

# Upregulated consistently in one human tissue

upreg_human_heart <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] > 0 & combined_human[,3] < FSR_level & combined_human[,4] > 0 & combined_human[,5] < FSR_level & combined_human[,6] > 0 & combined_human[,7] > FSR_level & combined_human[,9] > FSR_level &  combined_human[,11] > FSR_level),])

upreg_human_kidney <- as.data.frame(combined_human[which(combined_human[,5] < FSR_level & combined_human[,6] < 0 & combined_human[,9] < FSR_level & combined_human[,10] < 0 & combined_human[,11] < FSR_level & combined_human[,12] < 0 & combined_human[,1] > FSR_level & combined_human[,3] > FSR_level & combined_human[,7] > FSR_level),])

upreg_human_liver <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] < 0 & combined_human[,7] < FSR_level & combined_human[,8] > 0 & combined_human[,9] < FSR_level & combined_human[,10] > 0 & combined_human[,3] > FSR_level & combined_human[,5] > FSR_level & combined_human[,11] > FSR_level),])

upreg_human_lung <- as.data.frame(combined_human[which(combined_human[,3] < FSR_level & combined_human[,4] < 0 & combined_human[,7] < FSR_level & combined_human[,8] < 0 & combined_human[,11] < FSR_level & combined_human[,12] > 0 & combined_human[,1] > FSR_level & combined_human[,5] > FSR_level & combined_human[,9] > FSR_level),])

# Upregulated in only one human tissue

upreg_human_heart_only <- setdiff(upreg_human_heart, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_lung)))
upreg_human_kidney_only <- setdiff(upreg_human_kidney, (union(union(upreg_human_heart, upreg_human_liver), upreg_human_lung)))
upreg_human_liver_only <- setdiff(upreg_human_liver, (union(union(upreg_human_kidney, upreg_human_heart), upreg_human_lung)))
upreg_human_lung_only <- setdiff(upreg_human_lung, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_heart)))

nrow(upreg_human_heart)
nrow(upreg_human_kidney)
nrow(upreg_human_liver)
nrow(upreg_human_lung)

nrow(upreg_human_heart_only)
nrow(upreg_human_kidney_only)
nrow(upreg_human_liver_only)
nrow(upreg_human_lung_only)

# Upregulated consistently in one chimp tissue

upreg_chimp_heart <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] > 0 & combined_chimp[,3] < FSR_level & combined_chimp[,4] > 0 & combined_chimp[,5] < FSR_level & combined_chimp[,6] > 0 & combined_chimp[,7] > FSR_level & combined_chimp[,9] > FSR_level &  combined_chimp[,11] > FSR_level),])

upreg_chimp_kidney <- as.data.frame(combined_chimp[which(combined_chimp[,5] < FSR_level & combined_chimp[,6] < 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] < 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] < 0 & combined_chimp[,1] > FSR_level & combined_chimp[,3] > FSR_level &  combined_chimp[,7] > FSR_level),])

upreg_chimp_liver <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] < 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] > 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] > 0 & combined_chimp[,3] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,11] > FSR_level),])

upreg_chimp_lung <- as.data.frame(combined_chimp[which(combined_chimp[,3] < FSR_level & combined_chimp[,4] < 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] < 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] > 0 & combined_chimp[,1] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,9] > FSR_level),])

nrow(upreg_chimp_heart)
nrow(upreg_chimp_kidney)
nrow(upreg_chimp_liver)
nrow(upreg_chimp_lung)

# Upregulated consistently in one rhesus tissue

upreg_rhesus_heart <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] > 0 & combined_rhesus[,3] < FSR_level & combined_rhesus[,4] > 0 & combined_rhesus[,5] < FSR_level & combined_rhesus[,6] > 0 & combined_rhesus[,7] > FSR_level & combined_rhesus[,9] > FSR_level &  combined_rhesus[,11] > FSR_level),])

upreg_rhesus_kidney <- as.data.frame(combined_rhesus[which(combined_rhesus[,5] < FSR_level & combined_rhesus[,6] < 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] < 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] < 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,3] > FSR_level &  combined_rhesus[,7] > FSR_level),])

upreg_rhesus_liver <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] < 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] > 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] > 0 & combined_rhesus[,3] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,11] > FSR_level),])

upreg_rhesus_lung <- as.data.frame(combined_rhesus[which(combined_rhesus[,3] < FSR_level & combined_rhesus[,4] < 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] < 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] > 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,9] > FSR_level),])

nrow(upreg_rhesus_heart)
nrow(upreg_rhesus_kidney)
nrow(upreg_rhesus_liver)
nrow(upreg_rhesus_lung)


# Downregulated consistently in one human tissue

downreg_human_heart <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] < 0 & combined_human[,3] < FSR_level & combined_human[,4] < 0 & combined_human[,5] < FSR_level & combined_human[,6] < 0 &  combined_human[,7] > FSR_level & combined_human[,9] > FSR_level &  combined_human[,11] > FSR_level),])

downreg_human_kidney <- as.data.frame(combined_human[which(combined_human[,5] < FSR_level & combined_human[,6] > 0 & combined_human[,9] < FSR_level & combined_human[,10] > 0 & combined_human[,11] < FSR_level & combined_human[,12] > 0 & combined_human[,1] > FSR_level & combined_human[,3] > FSR_level &  combined_human[,7] > FSR_level),])

downreg_human_liver <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] > 0 & combined_human[,7] < FSR_level & combined_human[,8] < 0 & combined_human[,9] < FSR_level & combined_human[,10] < 0 & combined_human[,3] > FSR_level & combined_human[,5] > FSR_level &  combined_human[,11] > FSR_level),])

downreg_human_lung <- as.data.frame(combined_human[which(combined_human[,3] < FSR_level & combined_human[,4] > 0 & combined_human[,7] < FSR_level & combined_human[,8] > 0 & combined_human[,11] < FSR_level & combined_human[,12] < 0 & combined_human[,1] > FSR_level & combined_human[,5] > FSR_level &  combined_human[,9] > FSR_level),])

# Downregulated in only one human tissue

nrow(downreg_human_heart)
nrow(downreg_human_kidney)
nrow(downreg_human_liver)
nrow(downreg_human_lung)

# Downregulated consistently in one chimp tissue

downreg_chimp_heart <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] < 0 & combined_chimp[,3] < FSR_level & combined_chimp[,4] < 0 & combined_chimp[,5] < FSR_level & combined_chimp[,6] < 0 & combined_chimp[,7] > FSR_level & combined_chimp[,9] > FSR_level &  combined_chimp[,11] > FSR_level),])

downreg_chimp_kidney <- as.data.frame(combined_chimp[which(combined_chimp[,5] < FSR_level & combined_chimp[,6] > 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] > 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] > 0 & combined_chimp[,1] > FSR_level & combined_chimp[,3] > FSR_level &  combined_chimp[,7] > FSR_level),])

downreg_chimp_liver <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] > 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] < 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] < 0 & combined_chimp[,3] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,11] > FSR_level),])

downreg_chimp_lung <- as.data.frame(combined_chimp[which(combined_chimp[,3] < FSR_level & combined_chimp[,4] > 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] > 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] < 0 & combined_chimp[,1] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,9] > FSR_level),])

nrow(downreg_chimp_heart)
nrow(downreg_chimp_kidney)
nrow(downreg_chimp_liver)
nrow(downreg_chimp_lung)

# Downregulated consistently in one rhesus tissue

downreg_rhesus_heart <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] < 0 & combined_rhesus[,3] < FSR_level & combined_rhesus[,4] < 0 & combined_rhesus[,5] < FSR_level & combined_rhesus[,6] < 0 & combined_rhesus[,7] > FSR_level & combined_rhesus[,9] > FSR_level &  combined_rhesus[,11] > FSR_level),])

downreg_rhesus_kidney <- as.data.frame(combined_rhesus[which(combined_rhesus[,5] < FSR_level & combined_rhesus[,6] > 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] > 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] > 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,3] > FSR_level &  combined_rhesus[,7] > FSR_level),])

downreg_rhesus_liver <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] > 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] < 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] < 0 & combined_rhesus[,3] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,11] > FSR_level),])

downreg_rhesus_lung <- as.data.frame(combined_rhesus[which(combined_rhesus[,3] < FSR_level & combined_rhesus[,4] > 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] > 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] < 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,9] > FSR_level),])

nrow(downreg_rhesus_heart)
nrow(downreg_rhesus_kidney)
nrow(downreg_rhesus_liver)
nrow(downreg_rhesus_lung)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram <- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist <- list()
mylist[["Hu"]] <- rownames(upreg_human_tissue)
mylist[["Ch"]] <- rownames(upreg_chimp_tissue)
mylist[["Rh"]] <- rownames(upreg_rhesus_tissue)


# Make Venn Diagram
#dev.off()
Heart_specific_up <- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold", cat.dist = 0.05)
grid.draw(Heart_specific_up)
}

svg("Upregulated_heart_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart, "Upregulated in heart only")
dev.off()

intersect(intersect(upreg_human_heart, upreg_chimp_heart), upreg_rhesus_heart)


#dev.off()
#draw.triple.venn(323+183+626+158, 521+183+626+618, 158+626+618+1182, 183+626, 618+626, 158+626, 626,  main= "Heart #versus kidney (hypomethylated)",  category = c("Human", "Chimp", "Rhesus macaque"), height=3000, width=3000, cex=2 , #fill = pal[1:3], lty=1, cat.cex = 2, main.cex = 2)



svg("Upregulated_kidney_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney, NULL)
dev.off()

svg("Upregulated_liver_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver, NULL)
dev.off()

svg("Upregulated_lung_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung, NULL)
dev.off()

svg("Downregulated_heart_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart, NULL)
dev.off()

svg("Downregulated_kidney_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney, NULL)
dev.off()

svg("Downregulated_liver_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver, NULL)
dev.off()

svg("Downregulated_lung_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung, NULL)
dev.off()

#title <- ggdraw() + draw_label("Genes regulated in a tissue specific direction (FSR 1%)", fontface='bold')
p1 <- ggdraw()+draw_image("Upregulated_heart_specific.svg") 
p2 <- ggdraw()+draw_image("Downregulated_heart_specific.svg") 
p3 <- ggdraw()+draw_image("Upregulated_liver_specific.svg")  
p4 <- ggdraw()+draw_image("Downregulated_liver_specific.svg") 

p5 <- ggdraw()+draw_image("Upregulated_lung_specific.svg")  
p6 <- ggdraw()+draw_image("Downregulated_lung_specific.svg")  
p7 <- ggdraw()+draw_image("Upregulated_kidney_specific.svg")  
p8 <- ggdraw()+draw_image("Downregulated_kidney_specific.svg")  

eight_plots <- plot_grid(p1, p2, p3, p4, p5, p6, p7, p8, labels = c("6A.", "6B.", "6C.", "6D.", "6E.", "6F.", "6G.", "6H."), ncol = 2)
plot_fig2 <- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))

save_plot("../data/test.png", plot_fig2,
           ncol = 2, # we're saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
           # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 0.75
           )

save_plot("/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Figures/Figure_6_Tissue_specific.png", plot_fig2,
           ncol = 2, # we're saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
         # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 0.75
           )



# Find all conserved up/downregulated tissue specific

conserved_tissue_specific <- function(human_tissue_name, chimp_tissue_name, rhesus_tissue_name){

  upreg_human_tissue <- rownames(human_tissue_name)
  upreg_chimp_tissue <- rownames(chimp_tissue_name)
  upreg_rhesus_tissue <- rownames(rhesus_tissue_name)
  
all_three <- intersect(intersect(upreg_human_tissue, upreg_chimp_tissue), upreg_rhesus_tissue)

return(all_three)
}

heart_upregulated <- conserved_tissue_specific(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart)
write.csv(heart_upregulated, "../data/specific_upreg_heart_genes.csv", row.names = FALSE)
heart_downregulated <- conserved_tissue_specific(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart)
write.csv(heart_downregulated, "../data/specific_downreg_heart_genes.csv", row.names = FALSE)

liver_upregulated <- conserved_tissue_specific(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver)
write.csv(liver_upregulated, "../data/specific_upreg_liver_genes.csv", row.names = FALSE)
liver_downregulated <- conserved_tissue_specific(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver)
write.csv(liver_downregulated, "../data/specific_downreg_liver_genes.csv", row.names = FALSE)

lung_upregulated <- conserved_tissue_specific(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung)
write.csv(lung_upregulated, "../data/specific_upreg_lung_genes.csv", row.names = FALSE)
lung_downregulated <- conserved_tissue_specific(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung)
write.csv(lung_downregulated, "../data/specific_downreg_lung_genes.csv", row.names = FALSE)

kidney_upregulated <- conserved_tissue_specific(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney)
write.csv(kidney_upregulated, "../data/specific_upreg_kidney_genes.csv", row.names = FALSE)
kidney_downregulated <- conserved_tissue_specific(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney)
write.csv(kidney_downregulated, "../data/specific_downreg_kidney_genes.csv", row.names = FALSE)

```

## Find genes with a tissue-specific direction (significant and upregulated in 1 versus 2, 3, 4 but not significant in 2 v 3, 4 and 3 v 4) with FDR

```{r}
# Combine the human ASH results
combined_human <- cbind(results[["H_HeartvLi"]]$adj.P.Val, results[["H_HeartvLi"]]$logFC, results[["H_HeartvLu"]]$adj.P.Val, results[["H_HeartvLu"]]$logFC, results[["H_HeartvK"]]$adj.P.Val, results[["H_HeartvK"]]$logFC, results[["H_LivLu"]]$adj.P.Val, results[["H_LivLu"]]$logFC, results[["H_LivK"]]$adj.P.Val, results[["H_LivK"]]$logFC, results[["H_LuvK"]]$adj.P.Val, results[["H_LuvK"]]$logFC)

dim(combined_human)
rownames(combined_human) <- rownames(cpm_12184)
  
# Combine the chimp ASH results
combined_chimp <- cbind(results[["C_HeartvLi"]]$adj.P.Val, results[["C_HeartvLi"]]$logFC, results[["C_HeartvLu"]]$adj.P.Val, results[["C_HeartvLu"]]$logFC, results[["C_HeartvK"]]$adj.P.Val, results[["C_HeartvK"]]$logFC, results[["C_LivLu"]]$adj.P.Val, results[["C_LivLu"]]$logFC, results[["C_LivK"]]$adj.P.Val, results[["C_LivK"]]$logFC, results[["C_LuvK"]]$adj.P.Val, results[["C_LuvK"]]$logFC)

dim(combined_chimp)
rownames(combined_chimp) <- rownames(cpm_12184)

# Combine the rhesus ASH results
combined_rhesus <- cbind(results[["R_HeartvLi"]]$adj.P.Val, results[["R_HeartvLi"]]$logFC, results[["R_HeartvLu"]]$adj.P.Val, results[["R_HeartvLu"]]$logFC, results[["R_HeartvK"]]$adj.P.Val, results[["R_HeartvK"]]$logFC, results[["R_LivLu"]]$adj.P.Val, results[["R_LivLu"]]$logFC, results[["R_LivK"]]$adj.P.Val, results[["R_LivK"]]$logFC, results[["R_LuvK"]]$adj.P.Val, results[["R_LuvK"]]$logFC)

dim(combined_rhesus)
rownames(combined_rhesus) <- rownames(cpm_12184)


FSR_level <-  0.01

# Upregulated consistently in one human tissue

upreg_human_heart <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] > 0 & combined_human[,3] < FSR_level & combined_human[,4] > 0 & combined_human[,5] < FSR_level & combined_human[,6] > 0 & combined_human[,7] > FSR_level & combined_human[,9] > FSR_level &  combined_human[,11] > FSR_level),])

upreg_human_kidney <- as.data.frame(combined_human[which(combined_human[,5] < FSR_level & combined_human[,6] < 0 & combined_human[,9] < FSR_level & combined_human[,10] < 0 & combined_human[,11] < FSR_level & combined_human[,12] < 0 & combined_human[,1] > FSR_level & combined_human[,3] > FSR_level & combined_human[,7] > FSR_level),])

upreg_human_liver <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] < 0 & combined_human[,7] < FSR_level & combined_human[,8] > 0 & combined_human[,9] < FSR_level & combined_human[,10] > 0 & combined_human[,3] > FSR_level & combined_human[,5] > FSR_level & combined_human[,11] > FSR_level),])

upreg_human_lung <- as.data.frame(combined_human[which(combined_human[,3] < FSR_level & combined_human[,4] < 0 & combined_human[,7] < FSR_level & combined_human[,8] < 0 & combined_human[,11] < FSR_level & combined_human[,12] > 0 & combined_human[,1] > FSR_level & combined_human[,5] > FSR_level & combined_human[,9] > FSR_level),])

# Upregulated in only one human tissue

upreg_human_heart_only <- setdiff(upreg_human_heart, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_lung)))
upreg_human_kidney_only <- setdiff(upreg_human_kidney, (union(union(upreg_human_heart, upreg_human_liver), upreg_human_lung)))
upreg_human_liver_only <- setdiff(upreg_human_liver, (union(union(upreg_human_kidney, upreg_human_heart), upreg_human_lung)))
upreg_human_lung_only <- setdiff(upreg_human_lung, (union(union(upreg_human_kidney, upreg_human_liver), upreg_human_heart)))

nrow(upreg_human_heart)
nrow(upreg_human_kidney)
nrow(upreg_human_liver)
nrow(upreg_human_lung)

nrow(upreg_human_heart_only)
nrow(upreg_human_kidney_only)
nrow(upreg_human_liver_only)
nrow(upreg_human_lung_only)

# Upregulated consistently in one chimp tissue

upreg_chimp_heart <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] > 0 & combined_chimp[,3] < FSR_level & combined_chimp[,4] > 0 & combined_chimp[,5] < FSR_level & combined_chimp[,6] > 0 & combined_chimp[,7] > FSR_level & combined_chimp[,9] > FSR_level &  combined_chimp[,11] > FSR_level),])

upreg_chimp_kidney <- as.data.frame(combined_chimp[which(combined_chimp[,5] < FSR_level & combined_chimp[,6] < 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] < 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] < 0 & combined_chimp[,1] > FSR_level & combined_chimp[,3] > FSR_level &  combined_chimp[,7] > FSR_level),])

upreg_chimp_liver <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] < 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] > 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] > 0 & combined_chimp[,3] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,11] > FSR_level),])

upreg_chimp_lung <- as.data.frame(combined_chimp[which(combined_chimp[,3] < FSR_level & combined_chimp[,4] < 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] < 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] > 0 & combined_chimp[,1] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,9] > FSR_level),])

nrow(upreg_chimp_heart)
nrow(upreg_chimp_kidney)
nrow(upreg_chimp_liver)
nrow(upreg_chimp_lung)

# Upregulated consistently in one rhesus tissue

upreg_rhesus_heart <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] > 0 & combined_rhesus[,3] < FSR_level & combined_rhesus[,4] > 0 & combined_rhesus[,5] < FSR_level & combined_rhesus[,6] > 0 & combined_rhesus[,7] > FSR_level & combined_rhesus[,9] > FSR_level &  combined_rhesus[,11] > FSR_level),])

upreg_rhesus_kidney <- as.data.frame(combined_rhesus[which(combined_rhesus[,5] < FSR_level & combined_rhesus[,6] < 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] < 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] < 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,3] > FSR_level &  combined_rhesus[,7] > FSR_level),])

upreg_rhesus_liver <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] < 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] > 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] > 0 & combined_rhesus[,3] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,11] > FSR_level),])

upreg_rhesus_lung <- as.data.frame(combined_rhesus[which(combined_rhesus[,3] < FSR_level & combined_rhesus[,4] < 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] < 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] > 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,9] > FSR_level),])

nrow(upreg_rhesus_heart)
nrow(upreg_rhesus_kidney)
nrow(upreg_rhesus_liver)
nrow(upreg_rhesus_lung)


# Downregulated consistently in one human tissue

downreg_human_heart <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] < 0 & combined_human[,3] < FSR_level & combined_human[,4] < 0 & combined_human[,5] < FSR_level & combined_human[,6] < 0 &  combined_human[,7] > FSR_level & combined_human[,9] > FSR_level &  combined_human[,11] > FSR_level),])

downreg_human_kidney <- as.data.frame(combined_human[which(combined_human[,5] < FSR_level & combined_human[,6] > 0 & combined_human[,9] < FSR_level & combined_human[,10] > 0 & combined_human[,11] < FSR_level & combined_human[,12] > 0 & combined_human[,1] > FSR_level & combined_human[,3] > FSR_level &  combined_human[,7] > FSR_level),])

downreg_human_liver <- as.data.frame(combined_human[which(combined_human[,1] < FSR_level & combined_human[,2] > 0 & combined_human[,7] < FSR_level & combined_human[,8] < 0 & combined_human[,9] < FSR_level & combined_human[,10] < 0 & combined_human[,3] > FSR_level & combined_human[,5] > FSR_level &  combined_human[,11] > FSR_level),])

downreg_human_lung <- as.data.frame(combined_human[which(combined_human[,3] < FSR_level & combined_human[,4] > 0 & combined_human[,7] < FSR_level & combined_human[,8] > 0 & combined_human[,11] < FSR_level & combined_human[,12] < 0 & combined_human[,1] > FSR_level & combined_human[,5] > FSR_level &  combined_human[,9] > FSR_level),])

# Downregulated in only one human tissue

nrow(downreg_human_heart)
nrow(downreg_human_kidney)
nrow(downreg_human_liver)
nrow(downreg_human_lung)

# Downregulated consistently in one chimp tissue

downreg_chimp_heart <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] < 0 & combined_chimp[,3] < FSR_level & combined_chimp[,4] < 0 & combined_chimp[,5] < FSR_level & combined_chimp[,6] < 0 & combined_chimp[,7] > FSR_level & combined_chimp[,9] > FSR_level &  combined_chimp[,11] > FSR_level),])

downreg_chimp_kidney <- as.data.frame(combined_chimp[which(combined_chimp[,5] < FSR_level & combined_chimp[,6] > 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] > 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] > 0 & combined_chimp[,1] > FSR_level & combined_chimp[,3] > FSR_level &  combined_chimp[,7] > FSR_level),])

downreg_chimp_liver <- as.data.frame(combined_chimp[which(combined_chimp[,1] < FSR_level & combined_chimp[,2] > 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] < 0 & combined_chimp[,9] < FSR_level & combined_chimp[,10] < 0 & combined_chimp[,3] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,11] > FSR_level),])

downreg_chimp_lung <- as.data.frame(combined_chimp[which(combined_chimp[,3] < FSR_level & combined_chimp[,4] > 0 & combined_chimp[,7] < FSR_level & combined_chimp[,8] > 0 & combined_chimp[,11] < FSR_level & combined_chimp[,12] < 0 & combined_chimp[,1] > FSR_level & combined_chimp[,5] > FSR_level &  combined_chimp[,9] > FSR_level),])

nrow(downreg_chimp_heart)
nrow(downreg_chimp_kidney)
nrow(downreg_chimp_liver)
nrow(downreg_chimp_lung)

# Downregulated consistently in one rhesus tissue

downreg_rhesus_heart <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] < 0 & combined_rhesus[,3] < FSR_level & combined_rhesus[,4] < 0 & combined_rhesus[,5] < FSR_level & combined_rhesus[,6] < 0 & combined_rhesus[,7] > FSR_level & combined_rhesus[,9] > FSR_level &  combined_rhesus[,11] > FSR_level),])

downreg_rhesus_kidney <- as.data.frame(combined_rhesus[which(combined_rhesus[,5] < FSR_level & combined_rhesus[,6] > 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] > 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] > 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,3] > FSR_level &  combined_rhesus[,7] > FSR_level),])

downreg_rhesus_liver <- as.data.frame(combined_rhesus[which(combined_rhesus[,1] < FSR_level & combined_rhesus[,2] > 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] < 0 & combined_rhesus[,9] < FSR_level & combined_rhesus[,10] < 0 & combined_rhesus[,3] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,11] > FSR_level),])

downreg_rhesus_lung <- as.data.frame(combined_rhesus[which(combined_rhesus[,3] < FSR_level & combined_rhesus[,4] > 0 & combined_rhesus[,7] < FSR_level & combined_rhesus[,8] > 0 & combined_rhesus[,11] < FSR_level & combined_rhesus[,12] < 0 & combined_rhesus[,1] > FSR_level & combined_rhesus[,5] > FSR_level &  combined_rhesus[,9] > FSR_level),])

nrow(downreg_rhesus_heart)
nrow(downreg_rhesus_kidney)
nrow(downreg_rhesus_liver)
nrow(downreg_rhesus_lung)


# Make a function that makes a list and then outputs the Venn Diagram

make_venn_diagram <- function(upreg_human_tissue, upreg_chimp_tissue, upreg_rhesus_tissue, name_file){
mylist <- list()
mylist[["Hu"]] <- rownames(upreg_human_tissue)
mylist[["Ch"]] <- rownames(upreg_chimp_tissue)
mylist[["Rh"]] <- rownames(upreg_rhesus_tissue)


# Make Venn Diagram
#dev.off()
Heart_specific_up <- venn.diagram(mylist,  filename = NULL, fill = pal[1:3], main.cex = 3, cat.cex = 3, cex=3.5, lty=1, height=2000, width=2000, fontfamily = "sans", cat.fontfamily = "sans", main.fontfamily = "sans", cat.fontface = "bold", main.fontface = "bold", cat.dist = 0.05)
grid.draw(Heart_specific_up)
}

svg("Upregulated_heart_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart, "Upregulated in heart only")
dev.off()

svg("Upregulated_kidney_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney, "Upregulated in kidney only")
dev.off()

svg("Upregulated_liver_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver, "Upregulated in liver only")
dev.off()

svg("Upregulated_lung_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung, "Upregulated in lung only")
dev.off()

svg("Downregulated_heart_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart, "Downregulated in heart only")
dev.off()

svg("Downregulated_kidney_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney, "Downregulated in kidney only")
dev.off()

svg("Downregulated_liver_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver, "Downregulated in liver only")
dev.off()

svg("Downregulated_lung_specific.svg", width = 7.2, height = 7.2)
make_venn_diagram(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung, "Downregulated in lung only")
dev.off()

#title <- ggdraw() + draw_label("Genes regulated in a tissue specific direction (FDR 1%)", fontface='bold')
p1 <- ggdraw()+draw_image("Upregulated_heart_specific.svg")
p2 <- ggdraw()+draw_image("Downregulated_heart_specific.svg")
p3 <- ggdraw()+draw_image("Upregulated_kidney_specific.svg")
p4 <- ggdraw()+draw_image("Downregulated_kidney_specific.svg")

p5 <- ggdraw()+draw_image("Upregulated_liver_specific.svg")
p6 <- ggdraw()+draw_image("Downregulated_liver_specific.svg")
p7 <- ggdraw()+draw_image("Upregulated_lung_specific.svg")
p8 <- ggdraw()+draw_image("Downregulated_lung_specific.svg")

eight_plots <- plot_grid(p1, p2,  p5, p6, p7, p8, p3, p4, labels = c("2A.", "2B.", "2C.", "2D.", "2E.", "2F.", "2G.", "2H."), ncol = 2)
plot_fig2 <- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))

save_plot("../data/test.png", plot_fig2,
           ncol = 2, # we're saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
           # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 0.75
           )

save_plot("/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Figures/Figure_2_Tissue_specific.png", plot_fig2,
           ncol = 2, # we're saving a grid plot of 2 columns
           nrow = 4, # and 2 rows
       # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 0.75
           )

heart_upregulated <- conserved_tissue_specific(upreg_human_heart, upreg_chimp_heart, upreg_rhesus_heart)
#write.csv(heart_upregulated, "../data/specific_upreg_heart_genes.csv", row.names = FALSE)
heart_downregulated <- conserved_tissue_specific(downreg_human_heart, downreg_chimp_heart, downreg_rhesus_heart)
#write.csv(heart_downregulated, "../data/specific_downreg_heart_genes.csv", row.names = FALSE)

liver_upregulated <- conserved_tissue_specific(upreg_human_liver, upreg_chimp_liver, upreg_rhesus_liver)
#write.csv(liver_upregulated, "../data/specific_upreg_liver_genes.csv", row.names = FALSE)
liver_downregulated <- conserved_tissue_specific(downreg_human_liver, downreg_chimp_liver, downreg_rhesus_liver)
#write.csv(liver_downregulated, "../data/specific_downreg_liver_genes.csv", row.names = FALSE)

lung_upregulated <- conserved_tissue_specific(upreg_human_lung, upreg_chimp_lung, upreg_rhesus_lung)
#write.csv(lung_upregulated, "../data/specific_upreg_lung_genes.csv", row.names = FALSE)
lung_downregulated <- conserved_tissue_specific(downreg_human_lung, downreg_chimp_lung, downreg_rhesus_lung)
#write.csv(lung_downregulated, "../data/specific_downreg_lung_genes.csv", row.names = FALSE)

kidney_upregulated <- conserved_tissue_specific(upreg_human_kidney, upreg_chimp_kidney, upreg_rhesus_kidney)
#write.csv(kidney_upregulated, "../data/specific_upreg_kidney_genes.csv", row.names = FALSE)
kidney_downregulated <- conserved_tissue_specific(downreg_human_kidney, downreg_chimp_kidney, downreg_rhesus_kidney)
#write.csv(kidney_downregulated, "../data/specific_downreg_kidney_genes.csv", row.names = FALSE)



```

## Species specific 1 direction (supplementary table)

```{r}
FDR_level <- 0.01
combined_hcr <- cbind(results[["HvC_Heart"]]$adj.P.Val, results[["HvC_Heart"]]$logFC, results[["HvR_Heart"]]$adj.P.Val, results[["HvR_Heart"]]$logFC, results[["CvR_Heart"]]$adj.P.Val, results[["CvR_Heart"]]$logFC, results[["HvC_Kidney"]]$adj.P.Val, results[["HvC_Kidney"]]$logFC, results[["HvR_Kidney"]]$adj.P.Val, results[["HvR_Kidney"]]$logFC, results[["CvR_Kidney"]]$adj.P.Val, results[["CvR_Kidney"]]$logFC, results[["HvC_Liver"]]$adj.P.Val, results[["HvC_Liver"]]$logFC, results[["HvR_Liver"]]$adj.P.Val, results[["HvR_Liver"]]$logFC, results[["CvR_Liver"]]$adj.P.Val, results[["CvR_Liver"]]$logFC, results[["HvC_Lung"]]$adj.P.Val, results[["HvC_Lung"]]$logFC, results[["HvR_Lung"]]$adj.P.Val, results[["HvR_Lung"]]$logFC, results[["CvR_Lung"]]$adj.P.Val, results[["CvR_Lung"]]$logFC)

dim(combined_hcr)
rownames(combined_hcr) <- rownames(cpm_12184)

# Human specific and upregulated

combined_human_upregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] > 0 & combined_hcr[,3] < FDR_level & combined_hcr[,4] > 0 & combined_hcr[,5] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] > 0 & combined_hcr[,9] < FDR_level & combined_hcr[,10] > 0 & combined_hcr[,11] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] > 0 & combined_hcr[,15] < FDR_level & combined_hcr[,16] > 0 & combined_hcr[,17] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] > 0 & combined_hcr[,21] < FDR_level & combined_hcr[,22] > 0 & combined_hcr[,23] > FDR_level),]


# Human specific and downregulated

combined_human_downregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] < 0 & combined_hcr[,3] < FDR_level & combined_hcr[,4] < 0 & combined_hcr[,5] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] < 0 & combined_hcr[,9] < FDR_level & combined_hcr[,10] < 0 & combined_hcr[,11] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] < 0 & combined_hcr[,15] < FDR_level & combined_hcr[,16] < 0 & combined_hcr[,17] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] < 0 & combined_hcr[,21] < FDR_level & combined_hcr[,22] < 0 & combined_hcr[,23] > FDR_level),]

# Chimp specific and upregulated

combined_chimp_upregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] < 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] > 0 & combined_hcr[,3] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] < 0 & combined_hcr[,10] < FDR_level & combined_hcr[,11] > 0 & combined_hcr[,9] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] < 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] > 0 & combined_hcr[,15] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] < 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] > 0 & combined_hcr[,21] > FDR_level),]


# Chimp specific and downregulated

combined_chimp_downregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] > 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] < 0 & combined_hcr[,3] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] > 0 & combined_hcr[,10] < FDR_level & combined_hcr[,11] < 0 & combined_hcr[,9] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] > 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] < 0 & combined_hcr[,15] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] > 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] < 0 & combined_hcr[,21] > FDR_level),]

# Rhesus specific and upregulated

combined_rhesus_upregulated <- combined_hcr[which(combined_hcr[,3] < FDR_level & combined_hcr[,4] < 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] < 0 & combined_hcr[,1] > FDR_level & combined_hcr[,9] < FDR_level & combined_hcr[,10] < 0 & combined_hcr[,11] < FDR_level & combined_hcr[,12] < 0 & combined_hcr[,7] > FDR_level & combined_hcr[,15] < FDR_level & combined_hcr[,16] < 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] < 0 & combined_hcr[,13] > FDR_level & combined_hcr[,21] < FDR_level & combined_hcr[,22] < 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] < 0 & combined_hcr[,19] > FDR_level),]

combined_rhesus_downregulated <- combined_hcr[which(combined_hcr[,3] < FDR_level & combined_hcr[,4] > 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] > 0 & combined_hcr[,1] > FDR_level & combined_hcr[,9] < FDR_level & combined_hcr[,10] > 0 & combined_hcr[,11] < FDR_level & combined_hcr[,12] > 0 & combined_hcr[,7] > FDR_level & combined_hcr[,15] < FDR_level & combined_hcr[,16] > 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] > 0 & combined_hcr[,13] > FDR_level & combined_hcr[,21] < FDR_level & combined_hcr[,22] > 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] > 0 & combined_hcr[,19] > FDR_level),]


length(combined_human_upregulated)
length(combined_human_downregulated)
length(combined_chimp_upregulated)
length(combined_chimp_downregulated)
length(combined_rhesus_upregulated)
length(combined_rhesus_downregulated)


# Use adaptive shrinkage/FSR instead of FDR

FDR_level <- 0.05
combined_hcr <- cbind(results[["HvC_Heart"]]$svalue, results[["HvC_Heart"]]$beta_est, results[["HvR_Heart"]]$svalue, results[["HvR_Heart"]]$beta_est, results[["CvR_Heart"]]$svalue, results[["CvR_Heart"]]$beta_est, results[["HvC_Kidney"]]$svalue, results[["HvC_Kidney"]]$beta_est, results[["HvR_Kidney"]]$svalue, results[["HvR_Kidney"]]$beta_est, results[["CvR_Kidney"]]$svalue, results[["CvR_Kidney"]]$beta_est, results[["HvC_Liver"]]$svalue, results[["HvC_Liver"]]$beta_est, results[["HvR_Liver"]]$svalue, results[["HvR_Liver"]]$beta_est, results[["CvR_Liver"]]$svalue, results[["CvR_Liver"]]$beta_est, results[["HvC_Lung"]]$svalue, results[["HvC_Lung"]]$beta_est, results[["HvR_Lung"]]$svalue, results[["HvR_Lung"]]$beta_est, results[["CvR_Lung"]]$svalue, results[["CvR_Lung"]]$beta_est)

dim(combined_hcr)
rownames(combined_hcr) <- rownames(cpm_12184)

# Human specific and upregulated

combined_human_upregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] > 0 & combined_hcr[,3] < FDR_level & combined_hcr[,4] > 0 & combined_hcr[,5] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] > 0 & combined_hcr[,9] < FDR_level & combined_hcr[,10] > 0 & combined_hcr[,11] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] > 0 & combined_hcr[,15] < FDR_level & combined_hcr[,16] > 0 & combined_hcr[,17] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] > 0 & combined_hcr[,21] < FDR_level & combined_hcr[,22] > 0 & combined_hcr[,23] > FDR_level),]


# Human specific and downregulated

combined_human_downregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] < 0 & combined_hcr[,3] < FDR_level & combined_hcr[,4] < 0 & combined_hcr[,5] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] < 0 & combined_hcr[,9] < FDR_level & combined_hcr[,10] < 0 & combined_hcr[,11] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] < 0 & combined_hcr[,15] < FDR_level & combined_hcr[,16] < 0 & combined_hcr[,17] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] < 0 & combined_hcr[,21] < FDR_level & combined_hcr[,22] < 0 & combined_hcr[,23] > FDR_level),]

# Chimp specific and upregulated

combined_chimp_upregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] < 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] > 0 & combined_hcr[,3] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] < 0 & combined_hcr[,10] < FDR_level & combined_hcr[,11] > 0 & combined_hcr[,9] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] < 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] > 0 & combined_hcr[,15] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] < 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] > 0 & combined_hcr[,21] > FDR_level),]


# Chimp specific and downregulated

combined_chimp_downregulated <- combined_hcr[which(combined_hcr[,1] < FDR_level & combined_hcr[,2] > 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] < 0 & combined_hcr[,3] > FDR_level & combined_hcr[,7] < FDR_level & combined_hcr[,8] > 0 & combined_hcr[,10] < FDR_level & combined_hcr[,11] < 0 & combined_hcr[,9] > FDR_level & combined_hcr[,13] < FDR_level & combined_hcr[,14] > 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] < 0 & combined_hcr[,15] > FDR_level & combined_hcr[,19] < FDR_level & combined_hcr[,20] > 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] < 0 & combined_hcr[,21] > FDR_level),]

# Rhesus specific and upregulated

combined_rhesus_upregulated <- combined_hcr[which(combined_hcr[,3] < FDR_level & combined_hcr[,4] < 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] < 0 & combined_hcr[,1] > FDR_level & combined_hcr[,9] < FDR_level & combined_hcr[,10] < 0 & combined_hcr[,11] < FDR_level & combined_hcr[,12] < 0 & combined_hcr[,7] > FDR_level & combined_hcr[,15] < FDR_level & combined_hcr[,16] < 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] < 0 & combined_hcr[,13] > FDR_level & combined_hcr[,21] < FDR_level & combined_hcr[,22] < 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] < 0 & combined_hcr[,19] > FDR_level),]

combined_rhesus_downregulated <- combined_hcr[which(combined_hcr[,3] < FDR_level & combined_hcr[,4] > 0 & combined_hcr[,5] < FDR_level & combined_hcr[,6] > 0 & combined_hcr[,1] > FDR_level & combined_hcr[,9] < FDR_level & combined_hcr[,10] > 0 & combined_hcr[,11] < FDR_level & combined_hcr[,12] > 0 & combined_hcr[,7] > FDR_level & combined_hcr[,15] < FDR_level & combined_hcr[,16] > 0 & combined_hcr[,17] < FDR_level & combined_hcr[,18] > 0 & combined_hcr[,13] > FDR_level & combined_hcr[,21] < FDR_level & combined_hcr[,22] > 0 & combined_hcr[,23] < FDR_level & combined_hcr[,24] > 0 & combined_hcr[,19] > FDR_level),]


length(combined_human_upregulated)
length(combined_human_downregulated)
length(combined_chimp_upregulated)
length(combined_chimp_downregulated)
length(combined_rhesus_upregulated)
length(combined_rhesus_downregulated)
```





## Experimenting with interactions (3 species, 1 tissue versus a group of 3 tissues)

There are many possible interactions (e.g. Human and Chimp heart versus liver). To minimize the number of comparisons, we are going to compare 1 tissue to a group of the other 3 tissues (e.g.  Human and Chimp heart versus non-heart). 

### Interactions in all 3 species' hearts versus all other tissues

```{r}
# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 
tissue_group <- gsub("lung", "Grouped", tissue_group) 

## Make the contrast matrix and rename columns of the contrast matrix

design <- model.matrix(~ species*tissue_group + RIN)

# Rename columns of the contrast matrix
colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Human"
colnames(design)[3] <- "Rhesus"
colnames(design)[4] <- "tissue_heart" # Heart versus non-heart
colnames(design)[5] <- "RIN"
colnames(design)[6] <- "Human_heart"
colnames(design)[7] <- "Rhesus_heart"


# Look at the number of samples in each column 
colSums(design)

# Run linear model with new design matrix
cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <- -0.1009512
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)



fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm2 <- makeContrasts(HC_HG = Human_heart - Human, 
                     HR_HG = (Human_heart - Rhesus_heart) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_heart,
                     levels = design)
# Implement contrasts

contrasts_interactions <- contrasts.fit(fit.cyclic.norm, cm2)
fit1 <- eBayes(contrasts_interactions)



# Pull interactions

top3 <- list(HC_HG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none"), 
             HR_HG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none"),  
             CR_HG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none"))

# Determine the number of statistically signficant interactions

HC_HG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none")
dim(HC_HG[which(HC_HG$adj.P.Val < FDR_level), ]) 

HR_HG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
dim(HR_HG[which(HR_HG$adj.P.Val < FDR_level), ]) 

CR_HG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none")
dim(CR_HG[which(CR_HG$adj.P.Val < FDR_level), ]) 

```

### Adaptive shrinkage on interactions in all 3 species' hearts versus all other tissues

```{r}
# Prepare the data for ASH
tests <- colnames(fit1$coefficients)
tests
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = test)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}

# Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, "results-limma-voom-ash-interactions.rds"))

heart_combined_interactions <- as.data.frame(cbind(results[["HC_HG"]]$svalue, 
                               results[["HR_HG"]]$svalue, 
                               results[["CR_HG"]]$svalue))

dim(heart_combined_interactions)
colnames(heart_combined_interactions) <- c("HC_HG", "HR_HG", "CR_HG")
```



### Interactions in all 3 species' livers versus all other tissues

```{r}
# Reassign tissues to be liver versus non-livers

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("lung", "Grouped", tissue_group) 


# Rename columns of the contrast matrix

## Make the contrast matrix and rename columns of the contrast matrix

design <- model.matrix(~ species*tissue_group + RIN)

# Rename columns of the contrast matrix
colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Human"
colnames(design)[3] <- "Rhesus"
colnames(design)[4] <- "tissue_livers" # Heart versus non-livers
colnames(design)[5] <- "RIN"
colnames(design)[6] <- "Human_liver"
colnames(design)[7] <- "Rhesus_liver"


# Look at the number of samples in each column 
colSums(design)

# Run linear model with new design matrix
cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <- -0.0912331
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)



fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm2 <- makeContrasts(HC_HG = Human_liver - Human, 
                     HR_HG = (Human_liver - Rhesus_liver) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_liver,
                     levels = design)

# Implement contrasts

contrasts_interactions <- contrasts.fit(fit.cyclic.norm, cm2)
fit1 <- eBayes(contrasts_interactions)

# Pull interactions

top3 <- list(HC_LiG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none"), 
             HR_LiG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none"),  
             CR_LiG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none"))

# Determine the number of statistically signficant interactions

HC_LiG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none")
dim(HC_LiG[which(HC_LiG$adj.P.Val < FDR_level), ]) 

HR_LiG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
dim(HR_LiG[which(HR_LiG$adj.P.Val < FDR_level), ]) 

CR_LiG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none")
dim(CR_LiG[which(CR_LiG$adj.P.Val < FDR_level), ]) 

```

### Adaptive shrinkage on interactions in all 3 species' livers versus all other tissues

```{r}
# Prepare the data for ASH
tests <- colnames(fit1$coefficients)
tests
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = test)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}

# Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, "results-limma-voom-ash-interactions.rds"))

liver_combined_interactions <- as.data.frame(cbind(results[["HC_HG"]]$svalue, 
                               results[["HR_HG"]]$svalue, 
                               results[["CR_HG"]]$svalue))

dim(liver_combined_interactions)
colnames(liver_combined_interactions) <- c("HC_LiG", "HR_LiG", "CR_LiG")
```


### Interactions in all 3 species' lungs versus all other tissues

```{r}
# Reassign tissues to be lung versus non-lung

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 

## Make the contrast matrix and rename columns of the contrast matrix

design <- model.matrix(~ species*tissue_group + RIN)

# Rename columns of the contrast matrix
colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Human"
colnames(design)[3] <- "Rhesus"
colnames(design)[4] <- "tissue_lungs" # Heart versus non-livers
colnames(design)[5] <- "RIN"
colnames(design)[6] <- "Human_lung"
colnames(design)[7] <- "Rhesus_lung"


# Look at the number of samples in each column 
colSums(design)

# Run linear model with new design matrix
cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.09711687
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)



fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm4 <-  makeContrasts(HC_HG = Human_lung - Human, 
                     HR_HG = (Human_lung - Rhesus_lung) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_lung,
                     levels = design)

# Implement contrasts

contrasts_interactions <- contrasts.fit(fit.cyclic.norm, cm4)
fit1 <- eBayes(contrasts_interactions)

# Pull interactions

top3 <- list(HC_LuG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none"), 
             HR_LuG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none"),  
             CR_LuG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none"))

# Determine the number of statistically signficant interactions

HC_LuG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none")
dim(HC_LuG[which(HC_LuG$adj.P.Val < FDR_level), ]) 

HR_LuG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
dim(HR_LuG[which(HR_LuG$adj.P.Val < FDR_level), ]) 

CR_LuG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none")
dim(CR_LuG[which(CR_LuG$adj.P.Val < FDR_level), ]) 

```

### Adaptive shrinkage on interactions in all 3 species' lungs versus all other tissues

```{r}
# Prepare the data for ASH
tests <- colnames(fit1$coefficients)
tests
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = test)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}

# Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, "results-limma-voom-ash-interactions.rds"))

lung_combined_interactions <- as.data.frame(cbind(results[["HC_HG"]]$svalue, 
                               results[["HR_HG"]]$svalue, 
                               results[["CR_HG"]]$svalue))

dim(lung_combined_interactions)
colnames(lung_combined_interactions) <- c("HC_LuG", "HR_LuG", "CR_LuG")

#check_lung_combined_interactions <- as.data.frame(cbind(results[["HC_LuG"]]$adj.P.Val, results[["HC_LuG"]]$svalue))
#colnames(check_lung_combined_interactions) <- c("BH-adj P value", "s value")
#dim(check_lung_combined_interactions[which(check_lung_combined_interactions[,1] < 0.01) ,])
#check_lung_combined_interactions[which(check_lung_combined_interactions[,1] < 0.01) ,]
#dim(check_lung_combined_interactions[which(check_lung_combined_interactions[,2] < 0.01) ,])
```

### Interactions in all 3 species' kidneys versus all other tissues

```{r}
# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("lung", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 

design <- model.matrix(~ species*tissue_group + RIN)

## Make the contrast matrix and rename columns of the contrast matrix

# Rename columns of the contrast matrix
colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Human"
colnames(design)[3] <- "Rhesus"
colnames(design)[4] <- "tissue_kidneys" # Heart versus non-livers
colnames(design)[5] <- "RIN"
colnames(design)[6] <- "Human_kidney"
colnames(design)[7] <- "Rhesus_kidney"


# Implement contrasts

# Look at the number of samples in each column 
colSums(design)

# Run linear model with new design matrix
cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.128052
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)



fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm5 <- makeContrasts(HC_HG = Human_kidney - Human, 
                     HR_HG = (Human_kidney - Rhesus_kidney) - (Human - Rhesus),
                     CR_HG = Rhesus - Rhesus_kidney,
                     levels = design)

# Implement contrasts

contrasts_interactions <- contrasts.fit(fit.cyclic.norm, cm5)
fit1 <- eBayes(contrasts_interactions)

# Pull interactions

top3 <- list(HC_KG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none"), 
             HR_KG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none"),  
             CR_KG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none"))

# Determine the number of statistically signficant interactions

HC_KG =topTable(fit1, coef=1, adjust="BH", number=Inf, sort.by="none")
dim(HC_KG[which(HC_KG$adj.P.Val < FDR_level), ]) 

HR_KG =topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
dim(HR_KG[which(HR_KG$adj.P.Val < FDR_level), ]) 

CR_KG =topTable(fit1, coef=3, adjust="BH", number=Inf, sort.by="none")
dim(CR_KG[which(CR_KG$adj.P.Val < FDR_level), ]) 

```

### Tissue-specific interactions (FDR)

```{r}
# Set FDR level

FDR_level <- 0.01

# Interactions between 3 species hearts and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val < FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val < FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val < FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val > FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val > FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val > FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val > FDR_level), ]))

# Interactions between 3 species livers and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val < FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val < FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val < FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val > FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val > FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val > FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val > FDR_level), ]))

# Interactions between 3 species lungs and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val < FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val < FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val < FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val > FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val > FDR_level), ]))

# Interactions between 3 species kidneus and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val > FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val > FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val < FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val < FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val < FDR_level), ]))


# Interactions between 3 species hearts and other tissues (Human/Chimp versus Rhesus)

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val < FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val < FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val > FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val > FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val > FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val > FDR_level), ]))

# Interactions between 3 species livers and other tissues (Human/Chimp versus Rhesus)

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val < FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val < FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val > FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val > FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val > FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val > FDR_level), ]))

# Interactions between 3 species lungs and other tissues (Human/Chimp versus Rhesus)

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val < FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val < FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val > FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val > FDR_level), ]))

# Interactions between 3 species kidneys and other tissues

intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(rownames(HC_HG[which(HC_HG$adj.P.Val > FDR_level), ]), rownames(HR_HG[which(HR_HG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                                                      rownames(HC_LiG[which(HC_LiG$adj.P.Val > FDR_level), ])), rownames(HR_LiG[which(HR_LiG$adj.P.Val > FDR_level), ])), rownames(CR_HG[which(CR_HG$adj.P.Val > FDR_level), ])), 
                                                                                                                        
            rownames(HC_LuG[which(HC_LuG$adj.P.Val > FDR_level), ])), rownames(HR_LuG[which(HR_LuG$adj.P.Val > FDR_level), ])), rownames(CR_LuG[which(CR_LuG$adj.P.Val > FDR_level), ])), 
            
            rownames(HC_KG[which(HC_KG$adj.P.Val > FDR_level), ])), rownames(HR_KG[which(HR_KG$adj.P.Val < FDR_level), ])), rownames(CR_KG[which(CR_KG$adj.P.Val < FDR_level), ]))

```


### Adaptive shrinkage on interactions in all 3 species' kidneys versus all other tissues

```{r}
# Prepare the data for ASH
tests <- colnames(fit1$coefficients)
tests
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit1, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = test)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}

# Save results from analysis with limma and ash.
#saveRDS(results, file.path(data_dir, "results-limma-voom-ash-interactions.rds"))

kidney_combined_interactions <- as.data.frame(cbind(results[["HC_HG"]]$svalue, 
                               results[["HR_HG"]]$svalue, 
                               results[["CR_HG"]]$svalue))

dim(kidney_combined_interactions)
colnames(kidney_combined_interactions) <- c("HC_KG", "HR_KG", "CR_KG")
```

### Number of interactions using FSR

```{r}
# Heart
# Human versus Chimp

dim(heart_combined_interactions[which(heart_combined_interactions[,1] < FSR_level), ]) 

# Human versus Rhesus
dim(heart_combined_interactions[which(heart_combined_interactions[,2] < FSR_level), ]) 

# Chimp versus Rhesus
dim(heart_combined_interactions[which(heart_combined_interactions[,3] < FSR_level), ]) 

# Liver
# Human versus Chimp

dim(liver_combined_interactions[which(liver_combined_interactions[,1] < FSR_level), ]) 

# Human versus Rhesus
dim(liver_combined_interactions[which(liver_combined_interactions[,2] < FSR_level), ]) 

# Chimp versus Rhesus
dim(liver_combined_interactions[which(liver_combined_interactions[,3] < FSR_level), ]) 

# Lung
# Human versus Chimp

dim(lung_combined_interactions[which(lung_combined_interactions[,1] < FSR_level), ]) 

# Human versus Rhesus
dim(lung_combined_interactions[which(lung_combined_interactions[,2] < FSR_level), ]) 

# Chimp versus Rhesus
dim(lung_combined_interactions[which(lung_combined_interactions[,3] < FSR_level), ]) 

#  Kidney
# Human versus Chimp

dim(kidney_combined_interactions[which(kidney_combined_interactions[,1] < FSR_level), ]) 

# Human versus Rhesus
dim(kidney_combined_interactions[which(kidney_combined_interactions[,2] < FSR_level), ]) 

# Chimp versus Rhesus
dim(kidney_combined_interactions[which(kidney_combined_interactions[,3] < FSR_level), ]) 

```


```{r}
# Combine the svalue from interactions
three_species_all_tissue_interactions <- cbind(heart_combined_interactions, liver_combined_interactions, lung_combined_interactions, kidney_combined_interactions)

rownames(three_species_all_tissue_interactions) <- rownames(cpm.voom.cyclic$E)
dim(three_species_all_tissue_interactions)
```



```{r}
# Set FDR

FDR_level <- 0.01
FSR_level <- 0.01

# Interactions between 3 species hearts and other tissues

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] < FSR_level & three_species_all_tissue_interactions[,2] < FSR_level & three_species_all_tissue_interactions[,3] < FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] > FSR_level & three_species_all_tissue_interactions[,6] > FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] > FSR_level & three_species_all_tissue_interactions[,9] > FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] > FSR_level & three_species_all_tissue_interactions[,12] > FSR_level), ])

exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000182923", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species <- as.factor(make_exp_df$species)
levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus")
colnames(make_exp_df) <- c("CPM", "Species", "Tissue")
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species) + xlab("Tissue") + ylab("Normalized Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none")

# Interactions between 3 species livers and other tissues

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] > FSR_level & three_species_all_tissue_interactions[,3] > FSR_level & three_species_all_tissue_interactions[,4] < FSR_level & three_species_all_tissue_interactions[,5] < FSR_level & three_species_all_tissue_interactions[,6] < FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] > FSR_level & three_species_all_tissue_interactions[,9] > FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] > FSR_level & three_species_all_tissue_interactions[,12] > FSR_level), ])


exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000183077", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species <- as.factor(make_exp_df$species)
levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus")
colnames(make_exp_df) <- c("CPM", "Species", "Tissue")
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species) + xlab("Tissue") + ylab("Normalized Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none")

# Interactions between 3 species lungs and other tissues


nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] > FSR_level & three_species_all_tissue_interactions[,3] > FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] > FSR_level & three_species_all_tissue_interactions[,6] > FSR_level & three_species_all_tissue_interactions[,7] < FSR_level & three_species_all_tissue_interactions[,8] < FSR_level & three_species_all_tissue_interactions[,9] < FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] > FSR_level & three_species_all_tissue_interactions[,12] > FSR_level), ])


# Interactions between 3 species kidneys and other tissues


nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] > FSR_level & three_species_all_tissue_interactions[,3] > FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] > FSR_level & three_species_all_tissue_interactions[,6] > FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] > FSR_level & three_species_all_tissue_interactions[,9] > FSR_level & three_species_all_tissue_interactions[,10] < FSR_level & three_species_all_tissue_interactions[,11] < FSR_level & three_species_all_tissue_interactions[,12] < FSR_level), ])

# Interactions between 3 species hearts and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] < FSR_level & three_species_all_tissue_interactions[,3] < FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] > FSR_level & three_species_all_tissue_interactions[,6] > FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] > FSR_level & three_species_all_tissue_interactions[,9] > FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] > FSR_level & three_species_all_tissue_interactions[,12] > FSR_level), ])

exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000004487", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue))
make_exp_df$species <- as.factor(make_exp_df$species)
levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus")
colnames(make_exp_df) <- c("CPM", "Species", "Tissue")
ggplot(make_exp_df, aes(x=as.factor(Tissue), y=CPM, fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species) + xlab("Tissue") + ylab("Normalized Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none")

# Interactions between 3 species livers and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] > FSR_level & three_species_all_tissue_interactions[,3] > FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] < FSR_level & three_species_all_tissue_interactions[,6] < FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] > FSR_level & three_species_all_tissue_interactions[,9] > FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] > FSR_level & three_species_all_tissue_interactions[,12] > FSR_level), ])

# Interactions between 3 species lungs and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] > FSR_level & three_species_all_tissue_interactions[,3] > FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] > FSR_level & three_species_all_tissue_interactions[,6] > FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] < FSR_level & three_species_all_tissue_interactions[,9] < FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] > FSR_level & three_species_all_tissue_interactions[,12] > FSR_level), ])


# Interactions between 3 species kidneys and other tissues (Human/Chimp versus Rhesus)

nrow(three_species_all_tissue_interactions[which(three_species_all_tissue_interactions[,1] > FSR_level & three_species_all_tissue_interactions[,2] > FSR_level & three_species_all_tissue_interactions[,3] > FSR_level & three_species_all_tissue_interactions[,4] > FSR_level & three_species_all_tissue_interactions[,5] > FSR_level & three_species_all_tissue_interactions[,6] > FSR_level & three_species_all_tissue_interactions[,7] > FSR_level & three_species_all_tissue_interactions[,8] > FSR_level & three_species_all_tissue_interactions[,9] > FSR_level & three_species_all_tissue_interactions[,10] > FSR_level & three_species_all_tissue_interactions[,11] < FSR_level & three_species_all_tissue_interactions[,12] < FSR_level), ])

```

## Combining humans and chimps to have a Great Ape versus Rhesus tissues set of interactions (Figure 3)

### Hearts

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Human", "GA", GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 
tissue_group <- gsub("lung", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Heart"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_heart"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.02032154
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

GA_HG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(GA_HG[which(GA_HG$adj.P.Val < FDR_level), ]) 

dim(GA_HG[which(GA_HG$adj.P.Val < 0.005), ]) 
dim(GA_HG[which(GA_HG$adj.P.Val < 0.01), ]) 
dim(GA_HG[which(GA_HG$adj.P.Val < 0.05), ]) 
exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000007384", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
#levels(make_exp_df$Species) <- c("Chimpanzee", "Human", "Rhesus")
colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~make_exp_df[,2])  + xlab("Tissue") + ylab("Normalized Gene Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none") 
```


```{r}
  # Extract limma results
  results[["GA_Heart"]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[["GA_Heart"]] <- cbind(results[["GA_Heart"]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

GA_HG_svalue <- results[["GA_Heart"]]
nrow(GA_HG_svalue[which(GA_HG_svalue$svalue < FSR_level),])

nrow(GA_HG_svalue[which(GA_HG_svalue$svalue < 0.005),])
nrow(GA_HG_svalue[which(GA_HG_svalue$svalue < 0.01),])
nrow(GA_HG_svalue[which(GA_HG_svalue$svalue < 0.05),])


write.csv(results[["GA_Heart"]], "../data/GA_heart.csv")
```


### Livers

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Human", "GA", GA_species) 

# Reassign tissues to be liver versus non-liver

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("lung", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Liver"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_liver"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.008945556
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)


# Determine the number of statistically signficant interactions

GA_LiG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(GA_LiG[which(GA_LiG$adj.P.Val < FDR_level), ]) 

dim(GA_LiG[which(GA_LiG$adj.P.Val < 0.005), ]) 
dim(GA_LiG[which(GA_LiG$adj.P.Val < 0.01), ]) 
dim(GA_LiG[which(GA_LiG$adj.P.Val < 0.05), ]) 
```


```{r}
  # Extract limma results
  results[["GA_Liver"]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[["GA_Liver"]] <- cbind(results[["GA_Liver"]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


GA_LiG_svalue <- results[["GA_Liver"]]
nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue < FSR_level),])

nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue < 0.005),])
nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue < 0.01),])
nrow(GA_LiG_svalue[which(GA_LiG_svalue$svalue < 0.05),])
```


### Lungs

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Human", "GA", GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Lung"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_lung"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.01548285
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)


# Determine the number of statistically signficant interactions

GA_LuG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(GA_LuG[which(GA_LuG$adj.P.Val < FDR_level), ]) 

dim(GA_LuG[which(GA_LuG$adj.P.Val < 0.005), ]) 
dim(GA_LuG[which(GA_LuG$adj.P.Val < 0.01), ]) 
dim(GA_LuG[which(GA_LuG$adj.P.Val < 0.05), ]) 
```


```{r}
  # Extract limma results
  results[["GA_Lung"]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[["GA_Lung"]] <- cbind(results[["GA_Lung"]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

GA_LuG_svalue <- results[["GA_Lung"]]
nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue < FSR_level),])

nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue < 0.005),])
nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue < 0.01),])
nrow(GA_LuG_svalue[which(GA_LuG_svalue$svalue < 0.05),])
```

### Kidneys

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Human", "GA", GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("lung", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Kidney"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_kidney"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.05037098
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

GA_KG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(GA_KG[which(GA_KG$adj.P.Val < FDR_level), ]) 

dim(GA_KG[which(GA_KG$adj.P.Val < 0.005), ]) 
dim(GA_KG[which(GA_KG$adj.P.Val < 0.01), ]) 
dim(GA_KG[which(GA_KG$adj.P.Val < 0.05), ]) 
```

```{r}

  # Extract limma results
  results[["GA_Kidney"]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[["GA_Kidney"]] <- cbind(results[["GA_Kidney"]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

GA_KG_svalue <- results[["GA_Kidney"]]

nrow(GA_KG_svalue[which(GA_KG_svalue$svalue < FSR_level),])

nrow(GA_KG_svalue[which(GA_KG_svalue$svalue < 0.005),])
nrow(GA_KG_svalue[which(GA_KG_svalue$svalue < 0.01),])
nrow(GA_KG_svalue[which(GA_KG_svalue$svalue < 0.05),])

GA_values <- cbind(GA_HG_svalue, GA_LiG_svalue, GA_LuG_svalue, GA_KG_svalue)
```

```{r}
# Find tissue specific

# bind s values together

GA_svalue <- cbind(GA_HG_svalue$svalue, GA_LiG_svalue$svalue, GA_LuG_svalue$svalue, GA_KG_svalue$svalue)
rownames(GA_svalue) <- rownames(cpm_12184)
# Heart tissue specific
nrow(GA_svalue[which(GA_svalue[,1] < FSR_level & GA_svalue[,2] > FSR_level & GA_svalue[,3] > FSR_level & GA_svalue[,4] > FSR_level), ])

# Liver tissue specific
nrow(GA_svalue[which(GA_svalue[,1] > FSR_level & GA_svalue[,2] < FSR_level & GA_svalue[,3] > FSR_level & GA_svalue[,4] > FSR_level), ])

# Lung tissue specific
nrow(GA_svalue[which(GA_svalue[,1] > FSR_level & GA_svalue[,2] > FSR_level & GA_svalue[,3] < FSR_level & GA_svalue[,4] > FSR_level), ])

# Kidney tissue specific
nrow(GA_svalue[which(GA_svalue[,1] > FSR_level & GA_svalue[,2] > FSR_level & GA_svalue[,3] > FSR_level & GA_svalue[,4] < FSR_level), ])

# 377, 283, 161, 57

exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000001561", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
#levels(make_exp_df$species) <- c("Great Ape", "Rhesus")
colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~make_exp_df[,2])  + xlab("Tissue") + ylab("Normalized Gene Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none") 


```

### Great Ape Tissue Specific (using FDR)

```{r}
# Set FDR level

FDR_level <- 0.05

GA_BHpvalue <- cbind(GA_HG[,6], GA_LiG[,6], GA_LuG[,6], GA_KG[,6])

# Interactions between GA hearts and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] < FDR_level & GA_BHpvalue[,2] > FDR_level & GA_BHpvalue[,3] > FDR_level & GA_BHpvalue[,4] > FDR_level), ])


# Interactions between GA livers and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] > FDR_level & GA_BHpvalue[,2] < FDR_level & GA_BHpvalue[,3] > FDR_level & GA_BHpvalue[,4] > FDR_level), ])

# Interactions between GA lungs and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] > FDR_level & GA_BHpvalue[,2] > FDR_level & GA_BHpvalue[,3] < FDR_level & GA_BHpvalue[,4] > FDR_level), ])

# Interactions between GA kidneys and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] > FDR_level & GA_BHpvalue[,2] > FDR_level & GA_BHpvalue[,3] > FDR_level & GA_BHpvalue[,4] < FDR_level), ])

```




## Combining chimps and rhesus to have a human versus non-human tissues set of interactions

### Hearts

```{r}
# Reassign species to be human versus non-human (GA)

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Rhesus", "GA", GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 
tissue_group <- gsub("lung", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Heart"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_heart"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.01741244
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

NH_HG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(NH_HG[which(NH_HG$adj.P.Val < FDR_level), ]) 

dim(NH_HG[which(NH_HG$adj.P.Val < 0.005), ]) 
dim(NH_HG[which(NH_HG$adj.P.Val < 0.01), ]) 
dim(NH_HG[which(NH_HG$adj.P.Val < 0.05), ]) 
```


```{r}

  # Extract limma results
  results[[test]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

NH_HG_svalue <- results[[test]]
nrow(NH_HG_svalue[which(NH_HG_svalue$svalue < FSR_level),])

nrow(NH_HG_svalue[which(NH_HG_svalue$svalue < 0.005),])
nrow(NH_HG_svalue[which(NH_HG_svalue$svalue < 0.01),])
nrow(NH_HG_svalue[which(NH_HG_svalue$svalue < 0.05),])
```


### Livers

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Rhesus", "GA", GA_species) 

# Reassign tissues to be heart versus non-heart

# Reassign tissues to be liver versus non-liver

tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("lung", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Liver"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_liver"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <-  -0.01239498
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)


# Determine the number of statistically signficant interactions

NH_LiG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(NH_LiG[which(NH_LiG$adj.P.Val < FDR_level), ]) 

dim(NH_LiG[which(NH_LiG$adj.P.Val < 0.005), ]) 
dim(NH_LiG[which(NH_LiG$adj.P.Val < 0.01), ]) 
dim(NH_LiG[which(NH_LiG$adj.P.Val < 0.05), ]) 
```


```{r}

  # Extract limma results
  results[[test]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


NH_LiG_svalue <- results[[test]]
nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue < FSR_level),])

nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue < 0.005),])
nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue < 0.01),])
nrow(NH_LiG_svalue[which(NH_LiG_svalue$svalue < 0.05),])
```


### Lungs

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "GA", species) 
GA_species <- gsub("Rhesus", "GA", GA_species) 

# Reassign tissues to be heart versus non-heart


tissue_group <- gsub("kidney", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Lung"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_lung"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
#corfit$consensus <-  -0.5891937
#corfit$consensus <- 0.2177817

# Final voom on filtered data

#cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

NH_LuG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(NH_LuG[which(NH_LuG$adj.P.Val < FDR_level), ]) 

dim(NH_LuG[which(NH_LuG$adj.P.Val < 0.005), ]) 
dim(NH_LuG[which(NH_LuG$adj.P.Val < 0.01), ]) 
dim(NH_LuG[which(NH_LuG$adj.P.Val < 0.05), ]) 
```


```{r}

  results[[test]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

NH_LuG_svalue <- results[[test]]
nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue < FSR_level),])

nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue < 0.005),])
nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue < 0.01),])
nrow(NH_LuG_svalue[which(NH_LuG_svalue$svalue < 0.05),])
```

### Kidneys

```{r}
# Reassign species to be GA versus not GA

GA_species <- gsub("Chimp", "Non human", species) 
GA_species <- gsub("Rhesus", "Non human", GA_species) 

# Reassign tissues to be heart versus non-heart

tissue_group <- gsub("lung", "Grouped", tissue) 
tissue_group <- gsub("heart", "Grouped", tissue_group) 
tissue_group <- gsub("liver", "Grouped", tissue_group) 


## Make the contrast matrix and rename columns of the contrast matrix
design <- model.matrix(~ GA_species*tissue_group + RIN)

# Rename columns of the contrast matrix

colnames(design)[1] <- "Intercept"
colnames(design)[2] <- "Rhesus"
colnames(design)[3] <- "Kidney"
colnames(design)[4] <- "RIN"
colnames(design)[5] <- "Rhesus_kidney"

# Look at the number of samples in each column 
colSums(design)

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess")

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus <- -0.05638723
#corfit$consensus <- 0.2177817

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit$consensus)

# Run linear model with new design matrix

fit.cyclic.norm <- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm <- eBayes(fit.cyclic.norm)

# Determine the number of statistically signficant interactions

NH_KG =topTable(fit.cyclic.norm, coef=5, adjust="BH", number=Inf, sort.by="none")
dim(NH_KG[which(NH_KG$adj.P.Val < FDR_level), ]) 

dim(NH_KG[which(NH_KG$adj.P.Val < 0.005), ]) 
dim(NH_KG[which(NH_KG$adj.P.Val < 0.01), ]) 
dim(NH_KG[which(NH_KG$adj.P.Val < 0.05), ]) 
```

```{r}

  # Extract limma results
  results[[test]] <- get_results(fit.cyclic.norm, coef = 5)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit.cyclic.norm, coef = 5)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

NH_KG_svalue <- results[[test]]

nrow(NH_KG_svalue[which(NH_KG_svalue$svalue < FSR_level),])
nrow(NH_KG_svalue[which(NH_KG_svalue$svalue < 0.005),])
nrow(NH_KG_svalue[which(NH_KG_svalue$svalue < 0.01),])
nrow(NH_KG_svalue[which(NH_KG_svalue$svalue < 0.05),])


exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000240230", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, GA_species, tissue), stringsAsFactors = F)
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
levels(make_exp_df$GA_species) <- c("Human", "Non-Human")
colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized Gene Expression") + scale_x_discrete(labels=c("Heart","Kidney","Liver", "Lung")) + bjp + theme(legend.position="none") + ggtitle("COX19 (ENSG00000240230)") 
```

### Human/Non-Human Tissue Specific (using FDR)

```{r}
# Set FDR level

FDR_level <- 0.05
FSR_level <- 0.05

GA_BHpvalue <- cbind(NH_HG[,6], NH_LiG[,6], NH_LuG[,6], NH_KG[,6])

# Interactions between GA hearts and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] < FSR_level & GA_BHpvalue[,2] > FSR_level & GA_BHpvalue[,3] > FSR_level & GA_BHpvalue[,4] > FSR_level), ])


# Interactions between GA livers and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] > FSR_level & GA_BHpvalue[,2] < FSR_level & GA_BHpvalue[,3] > FSR_level & GA_BHpvalue[,4] > FSR_level), ])

# Interactions between GA lungs and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] > FSR_level & GA_BHpvalue[,2] > FSR_level & GA_BHpvalue[,3] < FSR_level & GA_BHpvalue[,4] > FSR_level), ])

# Interactions between GA kidneys and other tissues

nrow(GA_BHpvalue[which(GA_BHpvalue[,1] > FSR_level & GA_BHpvalue[,2] > FSR_level & GA_BHpvalue[,3] > FSR_level & GA_BHpvalue[,4] < FSR_level), ])

# 57, 19, 2, 8
```



```{r}
# Find tissue specific

FSR_level <- 0.005
# bind s values together

NH_svalue <- cbind(NH_HG_svalue$svalue, NH_LiG_svalue$svalue, NH_LuG_svalue$svalue, NH_KG_svalue$svalue)
rownames(NH_svalue) <- rownames(NH_HG_svalue)
# Heart tissue specific
nrow(NH_svalue[which(NH_svalue[,1] < FSR_level & NH_svalue[,2] > FSR_level & NH_svalue[,3] > FSR_level & NH_svalue[,4] > FSR_level), ])

NH_svalue[which(NH_svalue[,1] < FSR_level & NH_svalue[,2] > FSR_level & NH_svalue[,3] > FSR_level & NH_svalue[,4] > FSR_level), ]

# Liver tissue specific
nrow(NH_svalue[which(NH_svalue[,1] > FSR_level & NH_svalue[,2] < FSR_level & NH_svalue[,3] > FSR_level & NH_svalue[,4] > FSR_level), ])

# Lung tissue specific
nrow(NH_svalue[which(NH_svalue[,1] > FSR_level & NH_svalue[,2] > FSR_level & NH_svalue[,3] < FSR_level & NH_svalue[,4] > FSR_level), ])

# Kidney tissue specific
nrow(NH_svalue[which(NH_svalue[,1] > FSR_level & NH_svalue[,2] > FSR_level & NH_svalue[,3] > FSR_level & NH_svalue[,4] < FSR_level), ])

# 199, 28, 5, 5
```

```{r}
# Great ape by tissue
exp <- t(as.data.frame(cpm_12184[grepl("ENSG00000008282", rownames(cpm_12184)), ]))
make_exp_df <- as.data.frame(cbind(exp, species, tissue), stringsAsFactors = F)
#make_exp_df <- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df[,1] <- as.numeric(make_exp_df[,1])
make_exp_df[,2] <- as.character(make_exp_df[,2])

#levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus macaque")

make_exp_df$species[make_exp_df$species =="1"] <- "Chimpanzee"
make_exp_df$species[make_exp_df$species =="2"] <- "Human"
make_exp_df$species[make_exp_df$species =="3"] <-  "Rh. macaque"



colnames(make_exp_df) <- c("Normalized Gene Expression", "Species", "Tissue")

p2 <- ggplot(make_exp_df, aes(x=as.factor(Tissue), y=make_exp_df[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized \n Gene Expression") + scale_x_discrete(labels=c("H","K","Li", "Lu")) +  theme_bw() + bjp + theme(legend.position="none") + ggtitle("SYPL (ENSG00000008282)") 



# Great ape by tissue
exp1 <- t(as.data.frame(cpm_12184[grepl("ENSG00000004897", rownames(cpm_12184)), ]))
make_exp_df1 <- as.data.frame(cbind(exp1, species, tissue), stringsAsFactors = F)
#make_exp_df <- make_exp_df[which(make_exp_df$tissue == 1 | make_exp_df$tissue == 2),]
make_exp_df1[,1] <- as.numeric(make_exp_df1[,1])
make_exp_df1[,2] <- as.character(make_exp_df1[,2])

#levels(make_exp_df$species) <- c("Chimpanzee", "Human", "Rhesus macaque")

make_exp_df1$species[make_exp_df1$species =="1"] <- "Chimpanzee"
make_exp_df1$species[make_exp_df1$species =="2"] <- "Human"
make_exp_df1$species[make_exp_df1$species =="3"] <-  "Rh. macaque"


colnames(make_exp_df1) <- c("Normalized Gene Expression", "Species", "Tissue")

p1 <- ggplot(make_exp_df1, aes(x=as.factor(Tissue), y=make_exp_df1[,1], fill=as.factor(Tissue))) + geom_boxplot() + facet_wrap(~Species)  + xlab("Tissue") + ylab("Normalized \n Gene Expression") + scale_x_discrete(labels=c("H","K","Li", "Lu")) +  theme_bw() + bjp + theme(legend.position="none") + ggtitle("CDC27 (ENSG00000004897)") 



make_fig3 <- plot_grid(p2, p1, labels = c("3A.", "3B."), ncol = 1)

save_plot("/Users/laurenblake/Dropbox/Tissue_paper/Supplement/Figures/fig3.pdf", make_fig3,
          ncol = 1, # we're saving a grid plot of 2 columns
          nrow = 2, # and 2 rows
          # each individual subplot should have an aspect ratio of 1.3
          base_aspect_ratio = 1.3
          )
```

