<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />


<title>Figure_Supplementary11</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Figure_Supplementary11</h1>
<h4 class="author">Lauren Blake</h4>
<h4 class="date">December 2, 2018</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#load-library-and-data">Load library and data</a></li>
<li><a href="#human-rhesus-heart--de-genes">Human-rhesus heart- DE genes</a><ul>
<li><a href="#human-versus-rhesus-heart--de-genes">Human versus rhesus heart- DE genes</a></li>
<li><a href="#human-versus-rhesus-heart--non-de-genes">Human versus rhesus heart- non DE genes</a></li>
</ul></li>
<li><a href="#human-rhesus-kidney--de-genes">Human-rhesus kidney- DE genes</a><ul>
<li><a href="#human-versus-rhesus-kidney--de-genes">Human versus rhesus kidney- DE genes</a></li>
<li><a href="#human-versus-rhesus-kidney--non-de-genes">Human versus rhesus kidney- non DE genes</a></li>
</ul></li>
<li><a href="#liver">Liver</a><ul>
<li><a href="#human-versus-rhesus-liver--de-genes">Human versus rhesus liver- DE genes</a></li>
<li><a href="#human-versus-rhesus-liver--non-de-genes">Human versus rhesus liver- non DE genes</a></li>
</ul></li>
<li><a href="#lung">Lung</a><ul>
<li><a href="#human-versus-rhesus-lung--de-genes">Human versus rhesus lung- DE genes</a></li>
<li><a href="#human-versus-rhesus-lung--non-de-genes">Human versus rhesus lung- non DE genes</a></li>
</ul></li>
<li><a href="#make-into-figure">Make into figure</a></li>
</ul>
</div>

<div id="load-library-and-data" class="section level1">
<h1>Load library and data</h1>
<pre class="r"><code># Load libraries
library(&quot;edgeR&quot;)</code></pre>
<pre><code>## Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;limma&quot;)
library(&quot;plyr&quot;)
library(&quot;ashr&quot;)</code></pre>
<pre><code>## Warning: package &#39;ashr&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>library(&quot;cowplot&quot;)</code></pre>
<pre><code>## Warning: package &#39;cowplot&#39; was built under R version 3.4.4</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## 
## Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggplot2&#39;:
## 
##     ggsave</code></pre>
<pre class="r"><code>library(&quot;vashr&quot;)</code></pre>
<pre><code>## Loading required package: SQUAREM</code></pre>
<pre><code>## Loading required package: qvalue</code></pre>
<pre class="r"><code>library(&quot;devtools&quot;)
#devtools::install_github(&quot;jhsiao999/mediation/pkg&quot;)
library(&quot;medinome&quot;)

FDR_level = 0.05</code></pre>
<pre class="r"><code># import sample labels
samples &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/Sample_info_RNAseq_RIN.txt&quot;)
RIN &lt;- samples$RIN
# expression
exprs &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/250_exp_avg_methyl_hcr_4155_genes.txt&quot;, sep=&quot;&quot;)


# Normalized gene expression data
cpm.voom.cyclic &lt;- readRDS(&quot;../../../Reg_Evo_Primates/data/human_chimp_orth_cpm_voom_cyclic.rds&quot;)</code></pre>
</div>
<div id="human-rhesus-heart--de-genes" class="section level1">
<h1>Human-rhesus heart- DE genes</h1>
<pre class="r"><code># Check parameters

human_chimp_heart &lt;- c(32, 36, 40, 44, 20, 24, 28)

# Methylation
hc_exprs &lt;- exprs[,2:48]
exprs_methyl &lt;- exprs[,49:95]

# HC heart only


hc_exprs_heart &lt;- hc_exprs[,human_chimp_heart]
rownames(hc_exprs_heart) &lt;- exprs[,1]

methyl &lt;- exprs_methyl[,human_chimp_heart]
rownames(methyl) &lt;- exprs[,1]


Y &lt;- hc_exprs_heart
two_species &lt;- samples$Species[human_chimp_heart]
X &lt;- droplevels.factor(two_species)
M &lt;- methyl
RIN_subset &lt;- RIN[human_chimp_heart]



check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)</code></pre>
<pre><code>## Warning: package &#39;assertthat&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash_normal &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)
summary(fit_ash_normal$result$svalue &lt; 0.05)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical    3889     266</code></pre>
<div id="human-versus-rhesus-heart--de-genes" class="section level2">
<h2>Human versus rhesus heart- DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.3378995</code></pre>
<pre class="r"><code>ind_sigg &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigg[ind_sigg == TRUE] &lt;- &quot;red&quot;
ind_sigg[ind_sigg == FALSE] &lt;- &quot;black&quot;

combine_datag &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigg), stringsAsFactors = FALSE)

combine_datag[,1] &lt;- as.numeric(combine_datag[,1])
combine_datag[,2] &lt;- as.numeric(combine_datag[,2])
colnames(combine_datag) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigg&quot;)

hr_deg &lt;- ggplot(combine_datag, aes(combine_datag$Effect, combine_datag$SE)) + geom_point(color = c(combine_datag$ind_sigg)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
<div id="human-versus-rhesus-heart--non-de-genes" class="section level2">
<h2>Human versus rhesus heart- non DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &gt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.005228758</code></pre>
<pre class="r"><code>ind_sigh &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigh[ind_sigh == TRUE] &lt;- &quot;red&quot;
ind_sigh[ind_sigh == FALSE] &lt;- &quot;black&quot;

combine_datah &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigh), stringsAsFactors = FALSE)

combine_datah[,1] &lt;- as.numeric(combine_datah[,1])
combine_datah[,2] &lt;- as.numeric(combine_datah[,2])
colnames(combine_datah) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigh&quot;)

hr_deh &lt;- ggplot(combine_datah, aes(combine_datah$Effect, combine_datah$SE)) + geom_point(color = c(combine_datah$ind_sigh)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
</div>
<div id="human-rhesus-kidney--de-genes" class="section level1">
<h1>Human-rhesus kidney- DE genes</h1>
<pre class="r"><code># Check parameters

human_chimp_heart &lt;- c(33, 37, 41, 45, 17, 21, 25, 29)

# Methylation
hc_exprs &lt;- exprs[,2:48]
exprs_methyl &lt;- exprs[,49:95]

# HC heart only


hc_exprs_heart &lt;- hc_exprs[,human_chimp_heart]
rownames(hc_exprs_heart) &lt;- exprs[,1]

methyl &lt;- exprs_methyl[,human_chimp_heart]
rownames(methyl) &lt;- exprs[,1]


Y &lt;- hc_exprs_heart
two_species &lt;- samples$Species[human_chimp_heart]
X &lt;- droplevels.factor(two_species)
M &lt;- methyl
RIN_subset &lt;- RIN[human_chimp_heart]



check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)
fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash_normal &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)
summary(fit_ash_normal$result$svalue &lt; 0.05)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical    4073      82</code></pre>
<div id="human-versus-rhesus-kidney--de-genes" class="section level2">
<h2>Human versus rhesus kidney- DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.3333333</code></pre>
<pre class="r"><code>ind_sig &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sig[ind_sig == TRUE] &lt;- &quot;red&quot;
ind_sig[ind_sig == FALSE] &lt;- &quot;black&quot;

combine_dataa &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sig), stringsAsFactors = FALSE)

combine_dataa[,1] &lt;- as.numeric(combine_dataa[,1])
combine_dataa[,2] &lt;- as.numeric(combine_dataa[,2])
colnames(combine_dataa) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigc&quot;)

hr_dea &lt;- ggplot(combine_dataa, aes(combine_dataa$Effect, combine_dataa$SE)) + geom_point(color = c(combine_dataa$ind_sigc)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
<div id="human-versus-rhesus-kidney--non-de-genes" class="section level2">
<h2>Human versus rhesus kidney- non DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &gt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.005989583</code></pre>
<pre class="r"><code>ind_sigd &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigd[ind_sigd == TRUE] &lt;- &quot;red&quot;
ind_sigd[ind_sigd == FALSE] &lt;- &quot;black&quot;

combine_datab &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigd), stringsAsFactors = FALSE)

combine_datab[,1] &lt;- as.numeric(combine_datab[,1])
combine_datab[,2] &lt;- as.numeric(combine_datab[,2])
colnames(combine_datab) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigd&quot;)

hr_deb &lt;- ggplot(combine_datab, aes(combine_datab$Effect, combine_datab$SE)) + geom_point(color = c(combine_datab$ind_sigd)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
</div>
<div id="liver" class="section level1">
<h1>Liver</h1>
<pre class="r"><code># Check parameters

human_chimp_heart &lt;- c(34, 38, 42, 46, 18, 22, 26, 30)

# Methylation
hc_exprs &lt;- exprs[,2:48]
exprs_methyl &lt;- exprs[,49:95]

# HC heart only


hc_exprs_heart &lt;- hc_exprs[,human_chimp_heart]
rownames(hc_exprs_heart) &lt;- exprs[,1]

methyl &lt;- exprs_methyl[,human_chimp_heart]
rownames(methyl) &lt;- exprs[,1]


Y &lt;- hc_exprs_heart
two_species &lt;- samples$Species[human_chimp_heart]
X &lt;- droplevels.factor(two_species)
M &lt;- methyl
RIN_subset &lt;- RIN[human_chimp_heart]



check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)
fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash_normal &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)
summary(fit_ash_normal$result$svalue &lt; 0.05)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical    4057      98</code></pre>
<div id="human-versus-rhesus-liver--de-genes" class="section level2">
<h2>Human versus rhesus liver- DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.4021352</code></pre>
<pre class="r"><code>ind_sig &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sig[ind_sig == TRUE] &lt;- &quot;red&quot;
ind_sig[ind_sig == FALSE] &lt;- &quot;black&quot;

combine_datac &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sig), stringsAsFactors = FALSE)

combine_datac[,1] &lt;- as.numeric(combine_datac[,1])
combine_datac[,2] &lt;- as.numeric(combine_datac[,2])
colnames(combine_datac) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigc&quot;)

hr_dec &lt;- ggplot(combine_datac, aes(combine_datac$Effect, combine_datac$SE)) + geom_point(color = c(combine_datac$ind_sigc)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
<div id="human-versus-rhesus-liver--non-de-genes" class="section level2">
<h2>Human versus rhesus liver- non DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &gt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.008776458</code></pre>
<pre class="r"><code>ind_sigd &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigd[ind_sigd == TRUE] &lt;- &quot;red&quot;
ind_sigd[ind_sigd == FALSE] &lt;- &quot;black&quot;

combine_datad &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigd), stringsAsFactors = FALSE)

combine_datad[,1] &lt;- as.numeric(combine_datad[,1])
combine_datad[,2] &lt;- as.numeric(combine_datad[,2])
colnames(combine_datad) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigd&quot;)

hr_ded &lt;- ggplot(combine_datad, aes(combine_datad$Effect, combine_datad$SE)) + geom_point(color = c(combine_datad$ind_sigd)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
</div>
<div id="lung" class="section level1">
<h1>Lung</h1>
<pre class="r"><code># Check parameters

human_chimp_heart &lt;- c(35, 39, 43, 47, 19, 23, 27, 31)

# Methylation
hc_exprs &lt;- exprs[,2:48]
exprs_methyl &lt;- exprs[,49:95]

# HC heart only


hc_exprs_heart &lt;- hc_exprs[,human_chimp_heart]
rownames(hc_exprs_heart) &lt;- exprs[,1]

methyl &lt;- exprs_methyl[,human_chimp_heart]
rownames(methyl) &lt;- exprs[,1]


Y &lt;- hc_exprs_heart
two_species &lt;- samples$Species[human_chimp_heart]
X &lt;- droplevels.factor(two_species)
M &lt;- methyl
RIN_subset &lt;- RIN[human_chimp_heart]



check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)
fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash_normal &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)
summary(fit_ash_normal$result$svalue &lt; 0.05)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical    4061      94</code></pre>
<div id="human-versus-rhesus-lung--de-genes" class="section level2">
<h2>Human versus rhesus lung- DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.2151394</code></pre>
<pre class="r"><code>ind_sig &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sig[ind_sig == TRUE] &lt;- &quot;red&quot;
ind_sig[ind_sig == FALSE] &lt;- &quot;black&quot;

combine_datae &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sig), stringsAsFactors = FALSE)

combine_datae[,1] &lt;- as.numeric(combine_datae[,1])
combine_datae[,2] &lt;- as.numeric(combine_datae[,2])
colnames(combine_datae) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigc&quot;)

hr_dee &lt;- ggplot(combine_datae, aes(combine_datae$Effect, combine_datae$SE)) + geom_point(color = c(combine_datae$ind_sigc)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
<div id="human-versus-rhesus-lung--non-de-genes" class="section level2">
<h2>Human versus rhesus lung- non DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &gt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:8]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=6, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)

calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.0002939447</code></pre>
<pre class="r"><code>ind_sigd &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigd[ind_sigd == TRUE] &lt;- &quot;red&quot;
ind_sigd[ind_sigd == FALSE] &lt;- &quot;black&quot;

combine_dataf &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigd), stringsAsFactors = FALSE)

combine_dataf[,1] &lt;- as.numeric(combine_dataf[,1])
combine_dataf[,2] &lt;- as.numeric(combine_dataf[,2])
colnames(combine_dataf) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigd&quot;)

hr_def &lt;- ggplot(combine_dataf, aes(combine_dataf$Effect, combine_dataf$SE)) + geom_point(color = c(combine_dataf$ind_sigd)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
</div>
<div id="make-into-figure" class="section level1">
<h1>Make into figure</h1>
<pre class="r"><code>eight_plots &lt;- plot_grid(hr_deg, hr_deh, hr_dea, hr_deb, hr_dec, hr_ded, hr_dee, hr_def, labels = c(&quot;11A. Heart DE Genes&quot;, &quot;11B. Heart non-DE genes&quot;, &quot;11C. Kidney DE Genes&quot;, &quot;11D. Kidney non-DE genes&quot;, &quot;11E. Liver DE genes&quot;, &quot;11F. Liver non-DE genes&quot;, &quot;11G. Lung DE genes&quot;, &quot;11H. Lung non-DE genes&quot;, label_y = 0), ncol = 3, nrow = 5)</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code>plot_fig4 &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))

save_plot(&quot;/Users/laurenblake/Desktop/tissue_paper/test11b.pdf&quot;, plot_fig4,
           ncol = 3, # we&#39;re saving a grid plot of 2 columns
           nrow = 5, # and 2 rows
           # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 1
           )</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
