---
title: "Figures_integration"
author: "Lauren Blake"
date: "April 24, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

This file is to generate figures and tables for the gene expression and DNA methyation integration. 

```{r cars}
summary(cars)
```

# Interspecies differences in methylation (using FDR)

Using this approach, we estimated that interspecies differences in methylation might underlie as much as X% of differences in gene expression levels between humans and chimpanzees (FDR X%, X P value, based on 1000 permutations of the DNA methylation levels, see Methods; Figure X) and X% of the differences in gene expression levels between humans and rhesus macaques (FDR X, P value). 

```{r}

```

## Interspecies differences in methylation (using FSR)

```{r}

```

# Intertissue differences in methylation (using FDR)

Utilizing intertissue DE genes, we found that the percentage of methylation that may underlie human and chimpanzee DE genes is between X% and X% in humans and chimpanzees, and X% and X% in humans and rhesus macaques.

```{r}

```

## Intertissue differences in methylation (using FSR)

```{r}

```

# Species-specific versus conserved in humans and chimpanzees. 

Indeed, we found that this percentage is higher in conserved intertissue DE genes than those that are species-specific in humans and chimpanzees (X% versus X%, P in difference in proportions test). 

```{r}

```

## Robustness in species-specific versus conserved in humans and chimpanzees. 

Multiple p value cutoff. 

```{r}

```

# Species-specific versus conserved in humans and rhesus macaques

We also see this trend when comparing the two classes of DE genes in humans and rhesus (X% versus X%, but there were only X genes that were human-specific DE). 

# Using ASHR with effect size

To address this concern, we used adaptive shrinkage to estimate this difference in the effect of tissue for each intertissue comparison. When we separated these estimates by DE status, the DE genes had significantly more genes with large effect size differences than non-DE genes (X% versus X% at FSR 1%, Figure X, Additional File X). 

```{r}

```

# Robustness of effect size differences

This result is robust with respect to FSR cutoff (Additional Figure X). 

```{r}

```


