<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2018-04-24" />

<title>Figures_integration</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Figures_integration</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>April 24, 2018</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#interspecies-differences-in-methylation-using-fdr">Interspecies differences in methylation (using FDR)</a><ul>
<li><a href="#interspecies-differences-in-methylation-using-fsr">Interspecies differences in methylation (using FSR)</a></li>
</ul></li>
<li><a href="#intertissue-differences-in-methylation-using-fdr">Intertissue differences in methylation (using FDR)</a><ul>
<li><a href="#intertissue-differences-in-methylation-using-fsr">Intertissue differences in methylation (using FSR)</a></li>
</ul></li>
<li><a href="#species-specific-versus-conserved-in-humans-and-chimpanzees.">Species-specific versus conserved in humans and chimpanzees.</a><ul>
<li><a href="#robustness-in-species-specific-versus-conserved-in-humans-and-chimpanzees.">Robustness in species-specific versus conserved in humans and chimpanzees.</a></li>
</ul></li>
<li><a href="#species-specific-versus-conserved-in-humans-and-rhesus-macaques">Species-specific versus conserved in humans and rhesus macaques</a></li>
<li><a href="#using-ashr-with-effect-size">Using ASHR with effect size</a></li>
<li><a href="#robustness-of-effect-size-differences">Robustness of effect size differences</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This file is to generate figures and tables for the gene expression and DNA methyation integration.</p>
<pre class="r"><code>summary(cars)</code></pre>
<pre><code>##      speed           dist       
##  Min.   : 4.0   Min.   :  2.00  
##  1st Qu.:12.0   1st Qu.: 26.00  
##  Median :15.0   Median : 36.00  
##  Mean   :15.4   Mean   : 42.98  
##  3rd Qu.:19.0   3rd Qu.: 56.00  
##  Max.   :25.0   Max.   :120.00</code></pre>
</div>
<div id="interspecies-differences-in-methylation-using-fdr" class="section level1">
<h1>Interspecies differences in methylation (using FDR)</h1>
<p>Using this approach, we estimated that interspecies differences in methylation might underlie as much as X% of differences in gene expression levels between humans and chimpanzees (FDR X%, X P value, based on 1000 permutations of the DNA methylation levels, see Methods; Figure X) and X% of the differences in gene expression levels between humans and rhesus macaques (FDR X, P value).</p>
<div id="interspecies-differences-in-methylation-using-fsr" class="section level2">
<h2>Interspecies differences in methylation (using FSR)</h2>
</div>
</div>
<div id="intertissue-differences-in-methylation-using-fdr" class="section level1">
<h1>Intertissue differences in methylation (using FDR)</h1>
<p>Utilizing intertissue DE genes, we found that the percentage of methylation that may underlie human and chimpanzee DE genes is between X% and X% in humans and chimpanzees, and X% and X% in humans and rhesus macaques.</p>
<div id="intertissue-differences-in-methylation-using-fsr" class="section level2">
<h2>Intertissue differences in methylation (using FSR)</h2>
</div>
</div>
<div id="species-specific-versus-conserved-in-humans-and-chimpanzees." class="section level1">
<h1>Species-specific versus conserved in humans and chimpanzees.</h1>
<p>Indeed, we found that this percentage is higher in conserved intertissue DE genes than those that are species-specific in humans and chimpanzees (X% versus X%, P in difference in proportions test).</p>
<div id="robustness-in-species-specific-versus-conserved-in-humans-and-chimpanzees." class="section level2">
<h2>Robustness in species-specific versus conserved in humans and chimpanzees.</h2>
<p>Multiple p value cutoff.</p>
</div>
</div>
<div id="species-specific-versus-conserved-in-humans-and-rhesus-macaques" class="section level1">
<h1>Species-specific versus conserved in humans and rhesus macaques</h1>
<p>We also see this trend when comparing the two classes of DE genes in humans and rhesus (X% versus X%, but there were only X genes that were human-specific DE).</p>
</div>
<div id="using-ashr-with-effect-size" class="section level1">
<h1>Using ASHR with effect size</h1>
<p>To address this concern, we used adaptive shrinkage to estimate this difference in the effect of tissue for each intertissue comparison. When we separated these estimates by DE status, the DE genes had significantly more genes with large effect size differences than non-DE genes (X% versus X% at FSR 1%, Figure X, Additional File X).</p>
</div>
<div id="robustness-of-effect-size-differences" class="section level1">
<h1>Robustness of effect size differences</h1>
<p>This result is robust with respect to FSR cutoff (Additional Figure X).</p>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
