<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2018-10-11" />

<title>Final_effect_size_figure_4</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Final_effect_size_figure_4</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>October 11, 2018</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#human-chimp-heart--de-genes">Human-chimp heart- DE genes</a></li>
<li><a href="#set-fdr-levelsvalue">Set FDR level/svalue</a></li>
<li><a href="#human-chimp-heart">Human-chimp heart</a></li>
<li><a href="#human-chimp-heart--non-de-genes">Human-chimp heart- non DE genes</a></li>
<li><a href="#human-rhesus-heart--de-genes">Human-rhesus heart- DE genes</a><ul>
<li><a href="#human-versus-rhesus-heart--non-de-genes">Human versus rhesus heart- non DE genes</a></li>
</ul></li>
<li><a href="#human-chimp-percents">Human-chimp percents</a></li>
<li><a href="#human-rhesus-percentages">Human-rhesus percentages</a></li>
<li><a href="#make-into-figure">Make into figure</a></li>
</ul>
</div>

<pre class="r"><code># Load libraries
library(&quot;edgeR&quot;)</code></pre>
<pre><code>## Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;limma&quot;)
library(&quot;plyr&quot;)
library(&quot;ashr&quot;)
library(&quot;cowplot&quot;)</code></pre>
<pre><code>## Warning: package &#39;cowplot&#39; was built under R version 3.4.4</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 3.4.4</code></pre>
<pre><code>## 
## Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggplot2&#39;:
## 
##     ggsave</code></pre>
<pre class="r"><code>library(&quot;vashr&quot;)</code></pre>
<pre><code>## Loading required package: SQUAREM</code></pre>
<pre><code>## Loading required package: qvalue</code></pre>
<pre class="r"><code>library(&quot;devtools&quot;)</code></pre>
<pre><code>## Warning: package &#39;devtools&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>#devtools::install_github(&quot;jhsiao999/mediation/pkg&quot;)
library(&quot;medinome&quot;)</code></pre>
<div id="human-chimp-heart--de-genes" class="section level1">
<h1>Human-chimp heart- DE genes</h1>
<pre class="r"><code># import sample labels
samples &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/Sample_info_RNAseq_RIN.txt&quot;)
RIN &lt;- samples$RIN
# expression
exprs &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/human_chimp_orth_exp_methyl_7725_hum.txt&quot;, sep=&quot;&quot;)
# methylation data
methyl &lt;- read.csv(&quot;../../../Reg_Evo_Primates/data/chimp_human_orth_7725_avg_methyl_per_ts_gene.txt&quot;, sep=&quot;&quot;)

# Normalized gene expression data
cpm.voom.cyclic &lt;- readRDS(&quot;../../../Reg_Evo_Primates/data/human_chimp_orth_cpm_voom_cyclic.rds&quot;)</code></pre>
</div>
<div id="set-fdr-levelsvalue" class="section level1">
<h1>Set FDR level/svalue</h1>
<pre class="r"><code>FDR_level &lt;- 0.05</code></pre>
</div>
<div id="human-chimp-heart" class="section level1">
<h1>Human-chimp heart</h1>
<pre class="r"><code># Check parameters

human_chimp_heart &lt;- c(1, 5, 9, 13, 20, 24, 28)

# Methylation
hc_exprs &lt;- exprs[,2:48]
exprs_methyl &lt;- exprs[,49:79]

# HC heart only


hc_exprs_heart &lt;- hc_exprs[,human_chimp_heart]


methyl &lt;- exprs_methyl[,human_chimp_heart]



Y &lt;- hc_exprs_heart
two_species &lt;- samples$Species[human_chimp_heart]
X &lt;- droplevels.factor(two_species)
M &lt;- methyl
RIN_subset &lt;- RIN[human_chimp_heart]



check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)
fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash_normal &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>summary(fit_ash_normal$result$svalue &lt; 0.05)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical    7688      37</code></pre>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.1141498</code></pre>
<pre class="r"><code>ind_siga &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_siga[ind_siga == TRUE] &lt;- &quot;red&quot;
ind_siga[ind_siga == FALSE] &lt;- &quot;black&quot;

combine_dataa &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_siga), stringsAsFactors = FALSE)

combine_dataa[,1] &lt;- as.numeric(combine_dataa[,1])
combine_dataa[,2] &lt;- as.numeric(combine_dataa[,2])
colnames(combine_dataa) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_siga&quot;)

hr_dea &lt;- ggplot(combine_dataa, aes(combine_dataa$Effect, combine_dataa$SE)) + geom_point(color = c(combine_dataa$ind_siga)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
<div id="human-chimp-heart--non-de-genes" class="section level1">
<h1>Human-chimp heart- non DE genes</h1>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &gt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0</code></pre>
<pre class="r"><code>ind_sigb &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigb[ind_sigb == TRUE] &lt;- &quot;red&quot;
ind_sigb[ind_sigb == FALSE] &lt;- &quot;black&quot;

combine_datab &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigb), stringsAsFactors = FALSE)

combine_datab[,1] &lt;- as.numeric(combine_datab[,1])
combine_datab[,2] &lt;- as.numeric(combine_datab[,2])
colnames(combine_datab) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigb&quot;)

hr_deb &lt;- ggplot(combine_datab, aes(combine_datab$Effect, combine_datab$SE)) + geom_point(color = c(combine_datab$ind_sigb)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
<div id="human-rhesus-heart--de-genes" class="section level1">
<h1>Human-rhesus heart- DE genes</h1>
<pre class="r"><code># import sample labels
samples &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/Sample_info_RNAseq_RIN.txt&quot;)
RIN &lt;- samples$RIN
# expression
exprs &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/250_exp_avg_methyl_hcr_4155_genes.txt&quot;, sep=&quot;&quot;)


# Normalized gene expression data
cpm.voom.cyclic &lt;- readRDS(&quot;../../../Reg_Evo_Primates/data/human_chimp_orth_cpm_voom_cyclic.rds&quot;)</code></pre>
<pre class="r"><code># Check parameters

human_chimp_heart &lt;- c(32, 36, 40, 44, 20, 24, 28)

# Methylation
hc_exprs &lt;- exprs[,2:48]
exprs_methyl &lt;- exprs[,49:95]

# HC heart only


hc_exprs_heart &lt;- hc_exprs[,human_chimp_heart]
rownames(hc_exprs_heart) &lt;- exprs[,1]

methyl &lt;- exprs_methyl[,human_chimp_heart]
rownames(methyl) &lt;- exprs[,1]


Y &lt;- hc_exprs_heart
two_species &lt;- samples$Species[human_chimp_heart]
X &lt;- droplevels.factor(two_species)
M &lt;- methyl
RIN_subset &lt;- RIN[human_chimp_heart]



check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)
fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash_normal &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>summary(fit_ash_normal$result$svalue &lt; 0.05)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical    3889     266</code></pre>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.3378995</code></pre>
<pre class="r"><code>ind_sig &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sig[ind_sig == TRUE] &lt;- &quot;red&quot;
ind_sig[ind_sig == FALSE] &lt;- &quot;black&quot;

combine_data &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sig), stringsAsFactors = FALSE)

combine_data[,1] &lt;- as.numeric(combine_data[,1])
combine_data[,2] &lt;- as.numeric(combine_data[,2])
colnames(combine_data) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigc&quot;)

hr_dec &lt;- ggplot(combine_data, aes(combine_data$Effect, combine_data$SE)) + geom_point(color = c(combine_data$ind_sigc)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
<div id="human-versus-rhesus-heart--non-de-genes" class="section level2">
<h2>Human versus rhesus heart- non DE genes</h2>
<pre class="r"><code># Run the linear model in limma
  design_1 &lt;- model.matrix(~ X + RIN_subset)
  fit_all &lt;- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all &lt;- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &gt; FDR_level), ]
  
 human_chimp_heart1 &lt;-  rownames(methyl) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(methyl, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
M &lt;- counts_human_chimp_heart_subset

human_chimp_heart1 &lt;-  rownames(hc_exprs_heart) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 &lt;- as.data.frame(human_chimp_heart1)
counts_genes_in &lt;- cbind(hc_exprs_heart, human_chimp_heart1)
counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart1 == &quot;TRUE&quot;)
counts_human_chimp_heart_subset &lt;- counts_genes_in_cutoff[,1:7]
Y &lt;- counts_human_chimp_heart_subset

check_values &lt;- mediate.test.regressing(Y, X, M, RIN_subset)

fit &lt;- vash(check_values$d_se,df=5, singlecomp = T, unimodal = &quot;auto&quot;) 
fit_ash &lt;- ash(as.vector(check_values$d), fit$sd.post, mode = 0, mixcompdist = &quot;normal&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>calc &lt;- as.array(summary(fit_ash$result$svalue &lt; FDR_level))
calc2 &lt;- as.numeric(calc[[2]])

(nrow(Y)-calc2)/nrow(Y)</code></pre>
<pre><code>## [1] 0.005228758</code></pre>
<pre class="r"><code>ind_sigd &lt;- (fit_ash$result$svalue &lt; 0.05)
ind_sigd[ind_sigd == TRUE] &lt;- &quot;red&quot;
ind_sigd[ind_sigd == FALSE] &lt;- &quot;black&quot;

combine_datad &lt;- as.data.frame(cbind(fit_ash$result$betahat, fit_ash$result$sebetahat, ind_sigd), stringsAsFactors = FALSE)

combine_datad[,1] &lt;- as.numeric(combine_datad[,1])
combine_datad[,2] &lt;- as.numeric(combine_datad[,2])
colnames(combine_datad) &lt;- c(&quot;Effect&quot;, &quot;SE&quot;, &quot;ind_sigd&quot;)

hr_ded &lt;- ggplot(combine_datad, aes(combine_datad$Effect, combine_datad$SE)) + geom_point(color = c(combine_datad$ind_sigd)) + ylab(&quot;Standard error&quot;) + xlab(&quot;Effect size difference&quot;) + ylim(0, 6.5) + xlim(-8, 8)</code></pre>
</div>
</div>
<div id="human-chimp-percents" class="section level1">
<h1>Human-chimp percents</h1>
<pre class="r"><code># Make a table



comparison = data.frame(c(&quot;Heart&quot;, &quot;Heart&quot;, &quot;Kidney&quot;, &quot;Kidney&quot;,  &quot;Liver&quot;, &quot;Liver&quot;, &quot;Lung&quot;, &quot;Lung&quot;))
genes =  data.frame(c(&quot;DE&quot;, &quot;non-DE&quot;, &quot;DE&quot;, &quot;non-DE&quot;, &quot;DE&quot;, &quot;non-DE&quot;, &quot;DE&quot;, &quot;non-DE&quot;))

percentage = data.frame(c(0.114, 0, 0.122, 0, 0.256, 0, 0.1155, 0))



df &lt;- cbind(comparison, genes, percentage)
colnames(df) &lt;- c(&quot;Tissue&quot;, &quot;Gene_type&quot;, &quot;Percentage&quot;)
#df$Tissue_comparison &lt;- factor(df$Tissue, levels = df$Tissue )


plot_interspecies_hc &lt;- ggplot(data=df, aes(x=Gene_type, y=Percentage)) +
  geom_bar(stat=&quot;identity&quot;, fill = as.numeric(df$Gene_type)) + facet_wrap(~ df$Tissue) + xlab(&quot;Human-chimpanzee comparison&quot;) + ylab(&quot;% of genes for which \n DNAm could underlie \n gene expression&quot;) + theme(strip.text.x = element_text(face = &quot;bold&quot;, size = 5.5)) + ylim(0, 0.42)</code></pre>
</div>
<div id="human-rhesus-percentages" class="section level1">
<h1>Human-rhesus percentages</h1>
<pre class="r"><code># Make a table



comparison1 = data.frame(c(&quot;Heart&quot;, &quot;Heart&quot;, &quot;Kidney&quot;, &quot;Kidney&quot;,  &quot;Liver&quot;, &quot;Liver&quot;, &quot;Lung&quot;, &quot;Lung&quot;))
genes1 =  data.frame(c(&quot;DE&quot;, &quot;non-DE&quot;, &quot;DE&quot;, &quot;non-DE&quot;, &quot;DE&quot;, &quot;non-DE&quot;, &quot;DE&quot;, &quot;non-DE&quot;))

percentage1 = data.frame(c(0.337, 0.005, 0.333, 0.0059, 0.402, 0.00877, 0.215, 0.00029))



df1 &lt;- cbind(comparison1, genes1, percentage1)
colnames(df1) &lt;- c(&quot;Tissue&quot;, &quot;Gene_type&quot;, &quot;Percentage&quot;)
#df$Tissue_comparison &lt;- factor(df$Tissue, levels = df$Tissue )


plot_interspecies_hr &lt;- ggplot(data=df1, aes(x=Gene_type, y=Percentage)) +
  geom_bar(stat=&quot;identity&quot;, fill = as.numeric(df1$Gene_type)) + facet_wrap(~ df1$Tissue) + xlab(&quot;Human-rhesus comparison&quot;) + ylab(&quot;% of genes for which \n DNAm could underlie \n gene expression&quot;) + theme(strip.text.x = element_text(size = 5.5, face = &quot;bold&quot;)) + ylim(0, 0.42)</code></pre>
</div>
<div id="make-into-figure" class="section level1">
<h1>Make into figure</h1>
<pre class="r"><code>eight_plots &lt;- plot_grid(hr_dea, hr_dec, hr_deb, hr_ded, plot_interspecies_hc, plot_interspecies_hr, labels = c(&quot;4A. Human-chimpanzee&quot;, &quot;4B. Human-rhesus macaque&quot;, &quot;4C.&quot;, &quot;4D.&quot;, &quot;4E.&quot;, &quot;4F.&quot;, label_y = 0), ncol = 2)
plot_fig4 &lt;- plot_grid(eight_plots, ncol = 1, rel_heights=c(0.1, 1))

save_plot(&quot;/Users/laurenblake/Dropbox/Tissue_paper/Draft_4/Figure_4.png&quot;, plot_fig4,
           ncol = 2, # we&#39;re saving a grid plot of 2 columns
           nrow = 3, # and 2 rows
           # each individual subplot should have an aspect ratio of 1.3
           base_aspect_ratio = 1
           )</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
