---
title: "Identifying DMRs"
author: "Lauren Blake"
date: "January 5, 2018"
output: html_document
editor_options: 
  chunk_output_type: console
---

Since identifying DMRs is computationally intensive using the BSmooth package, we needed to run this on the cluster (/mnt/gluster/home/leblake/Methylation/tissue\_DMRs and /mnt/gluster/home/leblake/Methylation/species\_DMRs)

First, we will examine the results of the tissue DMRs


```{r chunk-options, include=FALSE}
source("chunk-options.R")
```


```{r}
# Import libraries

library("ggplot2")
library("VennDiagram")
library(plyr)


# Human tDMRs
humans_heart_kidney_DMRs <- read.delim("../data/humans_heart_kidney_DMRs.txt")
humans_heart_liver_DMRs <- read.delim("../data/humans_heart_liver_DMRs.txt")
humans_heart_lung_DMRs <- read.delim("../data/humans_heart_lung_DMRs.txt")
humans_liver_lung_DMRs <- read.delim("../data/humans_liver_lung_DMRs.txt")
humans_liver_kidney_DMRs <- read.delim("../data/humans_liver_kidney_DMRs.txt")
humans_lung_kidney_DMRs <- read.delim("../data/humans_lung_kidney_DMRs.txt")

levels(factor(humans_lung_kidney_DMRs[,1]))
count(humans_lung_kidney_DMRs[,1])

# Chimp tDMRs
chimps_heart_kidney_DMRs <- read.delim("../data/chimps_heart_kidney_DMRs.txt")
chimps_heart_liver_DMRs <- read.delim("../data/chimps_heart_liver_DMRs.txt")
chimps_heart_lung_DMRs <- read.delim("../data/chimps_heart_lung_DMRs.txt")
chimps_liver_lung_DMRs <- read.delim("../data/chimps_liver_lung_DMRs.txt")
chimps_liver_kidney_DMRs <- read.delim("../data/chimps_liver_kidney_DMRs.txt")
chimps_lung_kidney_DMRs <- read.delim("../data/chimps_lung_kidney_DMRs.txt")

levels(factor(chimps_lung_kidney_DMRs[,1]))
count(chimps_lung_kidney_DMRs[,1])

# Rhesus tDMRs

rhesus_heart_kidney_DMRs <- read.delim("../data/rhesus_heart_kidney_DMRs.txt")
rhesus_heart_liver_DMRs <- read.delim("../data/rhesus_heart_liver_DMRs.txt")
rhesus_heart_lung_DMRs <- read.delim("../data/rhesus_heart_lung_DMRs.txt")
rhesus_liver_lung_DMRs <- read.delim("../data/rhesus_liver_lung_DMRs.txt")
rhesus_liver_kidney_DMRs <- read.delim("../data/rhesus_liver_kidney_DMRs.txt")
rhesus_lung_kidney_DMRs <- read.delim("../data/rhesus_lung_kidney_DMRs.txt")

levels(factor(rhesus_heart_kidney_DMRs[,1]))
count(rhesus_lung_kidney_DMRs[,1])
```

# How many regions overlap eachother?

```{r}
library("bedr")

# columns to pull
pull_col <- c(1:3, 16)

# make bed style format
humans_heart_liver_DMRs_bed <- humans_heart_liver_DMRs[,pull_col]

# Get DMRs on the autosomal chromosomes only
humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed[which(humans_heart_liver_DMRs_bed[,1] != "chrX"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])

summary(humans_heart_liver_DMRs_bed_noX[,3] - humans_heart_liver_DMRs_bed_noX[,2])

sort_humans_heart_liver_DMRs_bed <- bedr.sort.region(humans_heart_liver_DMRs_bed_noX)

# make bed style format
humans_heart_lung_DMRs_bed <- humans_heart_lung_DMRs[,pull_col]
humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed[which(humans_heart_lung_DMRs_bed[,1] != "chrX"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])
sort_humans_heart_lung_DMRs_bed <- bedr.sort.region(humans_heart_lung_DMRs_bed_noX)

summary(humans_heart_lung_DMRs_bed_noX[,3] - humans_heart_lung_DMRs_bed_noX[,2])

heart_lung <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5")

summary(heart_lung[,3] - heart_lung[,2])
summary(as.integer(heart_lung[,9]))
summary(as.integer(heart_lung[,9]) > 0)

lung_heart <- bedr(input = list(a = sort_humans_heart_lung_DMRs_bed, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5")

summary(as.integer(lung_heart[,9]))
summary(as.integer(lung_heart[,9]) > 0)

# Make a Venn diagram based on the results

summary(as.integer(heart_lung[,9]) > 0)
summary(as.integer(lung_heart[,9]) > 0)

grid.newpage()
draw.pairwise.venn(area1 = 22561, area2 = 14208, cross.area = 3858, category = c("Heart v. Liver", "Heart v. Lung"))
```

# How does the overlap change based on % bp?

```{r}
# 1 bp overlap
heart_liver_1 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 1E-9 -F 1E-9")

summary(as.integer(heart_liver_1[,9]) > 0)

# 10% overlap
heart_liver_10 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.10 -F 0.10")

summary(as.integer(heart_liver_10[,9]) > 0)

# 20% overlap
heart_liver_20 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.20 -F 0.20")

summary(as.integer(heart_liver_20[,9]) > 0)

# 25% overlap
heart_liver_25 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.25 -F 0.25")

summary(as.integer(heart_liver_25[,9]) > 0)

# 30% overlap
heart_liver_30 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.30 -F 0.30")

summary(as.integer(heart_liver_30[,9]) > 0)

#40% overlap
heart_liver_40 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.4 -F 0.4")

summary(as.integer(heart_liver_40[,9]) > 0)

# 50% overlap
heart_liver_50 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.50 -F 0.50")

summary(as.integer(heart_liver_50[,9]) > 0)

#60% overlap
heart_liver_60 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.6 -F 0.6")

summary(as.integer(heart_liver_60[,9]) > 0)

# 70% overlap
heart_liver_70 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.70 -F 0.70")

summary(as.integer(heart_liver_70[,9]) > 0)

#75% overlap
heart_liver_75 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.75 -F 0.75")

summary(as.integer(heart_liver_75[,9]) > 0)

# 80% overlap
heart_liver_80 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.80 -F 0.80")

summary(as.integer(heart_liver_80[,9]) > 0)

#90% overlap
heart_liver_90 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.9 -F 0.9")

summary(as.integer(heart_liver_90[,9]) > 0)

#100% overlap
heart_liver_100 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 1 -F 1")

summary(as.integer(heart_liver_100[,9]) > 0)
```

## Plot the overlap

```{r}
# Find the number of tDMRs depending on the bp overlap
summary(as.integer(heart_liver_1[,9]) > 0)
summary(as.integer(heart_liver_10[,9]) > 0)
summary(as.integer(heart_liver_20[,9]) > 0)
summary(as.integer(heart_liver_25[,9]) > 0)
summary(as.integer(heart_liver_30[,9]) > 0)
summary(as.integer(heart_liver_40[,9]) > 0)
summary(as.integer(heart_liver_50[,9]) > 0)
summary(as.integer(heart_liver_60[,9]) > 0)
summary(as.integer(heart_liver_70[,9]) > 0)
summary(as.integer(heart_liver_75[,9]) > 0)
summary(as.integer(heart_liver_80[,9]) > 0)
summary(as.integer(heart_liver_90[,9]) > 0)
summary(as.integer(heart_liver_100[,9]) > 0)

# Plot the number of tDMRs
conserved_tdmr_heart_liver_heart_lung <- c(5607, 5472, 5189, 5007, 4824, 4400, 3858, 3275, 2636, 2305,1964,1338, 974)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung, xlab = "Percentage of overlap (bp)", ylab = "Number of human tDMRs (HvLi and HvLu)",  pch = 19)

# Investigate the size of the tDMRs depending on the percentage overlap

check_1 <- heart_liver_10[which((as.integer(heart_liver_1[,9]) > 0) == TRUE),]
check_10 <- heart_liver_10[which((as.integer(heart_liver_10[,9]) > 0) == TRUE),]
check_20 <- heart_liver_20[which((as.integer(heart_liver_20[,9]) > 0) == TRUE),]
check_25 <- heart_liver_25[which((as.integer(heart_liver_25[,9]) > 0) == TRUE),]
check_30 <- heart_liver_30[which((as.integer(heart_liver_30[,9]) > 0) == TRUE),]
check_40 <- heart_liver_40[which((as.integer(heart_liver_40[,9]) > 0) == TRUE),]
check_50 <- heart_liver_50[which((as.integer(heart_liver_50[,9]) > 0) == TRUE),]
check_60 <- heart_liver_60[which((as.integer(heart_liver_60[,9]) > 0) == TRUE),]
check_70 <- heart_liver_70[which((as.integer(heart_liver_70[,9]) > 0) == TRUE),]
check_75 <- heart_liver_75[which((as.integer(heart_liver_75[,9]) > 0) == TRUE),]
check_80 <- heart_liver_80[which((as.integer(heart_liver_80[,9]) > 0) == TRUE),]
check_90 <- heart_liver_90[which((as.integer(heart_liver_90[,9]) > 0) == TRUE),]
check_100 <- heart_liver_100[which((as.integer(heart_liver_100[,9]) > 0) == TRUE),]

summary(as.integer(check_1[,9]))
summary(as.integer(check_10[,9]))
summary(as.integer(check_20[,9]))
summary(as.integer(check_25[,9]))
summary(as.integer(check_30[,9]))
summary(as.integer(check_40[,9]))
summary(as.integer(check_50[,9]))
summary(as.integer(check_60[,9]))
summary(as.integer(check_70[,9]))
summary(as.integer(check_75[,9]))
summary(as.integer(check_80[,9]))
summary(as.integer(check_90[,9]))
summary(as.integer(check_100[,9]))

# Plot the median tDMR length
conserved_tdmr_heart_liver_heart_lung_bp <- c(120, 419, 440, 454, 470, 500, 533, 574, 601, 604,599,522, 410.5)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung_bp, xlab = "Percentage of overlap (bp)", ylab = "Median length of overlap (bp)", pch = 19, main = "Median length of overlap between human tDMRs (HvLi and HvLu)")

```

# Are the overlap results robust with respect to tissue?

Look at human heart kidney versus human liver lung. 

```{r}
# make bed style format: change name from human to rhesus
humans_heart_liver_DMRs_bed <- humans_heart_kidney_DMRs[,pull_col]

# Get DMRs on the autosomal chromosomes only
humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed[which(humans_heart_liver_DMRs_bed[,1] != "chrX"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])

summary(humans_heart_liver_DMRs_bed_noX[,3] - humans_heart_liver_DMRs_bed_noX[,2])

sort_humans_heart_liver_DMRs_bed <- bedr.sort.region(humans_heart_liver_DMRs_bed_noX, check.chr = FALSE)

# make bed style format
humans_heart_lung_DMRs_bed <- humans_liver_lung_DMRs[,pull_col]
humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed[which(humans_heart_lung_DMRs_bed[,1] != "chrX"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])
sort_humans_heart_lung_DMRs_bed <- bedr.sort.region(humans_heart_lung_DMRs_bed_noX, check.chr = FALSE)

summary(humans_heart_lung_DMRs_bed_noX[,3] - humans_heart_lung_DMRs_bed_noX[,2])

heart_lung <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5", check.chr = FALSE)

summary(heart_lung[,3] - heart_lung[,2])
summary(as.integer(heart_lung[,9]))
summary(as.integer(heart_lung[,9]) > 0)

lung_heart <- bedr(input = list(a = sort_humans_heart_lung_DMRs_bed, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5", check.chr = FALSE)

summary(as.integer(lung_heart[,9]))
summary(as.integer(lung_heart[,9]) > 0)

# Make a Venn diagram based on the results

summary(as.integer(heart_lung[,9]) > 0)
summary(as.integer(lung_heart[,9]) > 0)

grid.newpage()
draw.pairwise.venn(area1 = (29136+1155), area2 = (11687+1155), cross.area = 1155, category = c("Heart v. Liver", "Heart v. Lung"))
```


```{r}
# 1 bp overlap
heart_liver_1 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.00000001 -F 0.00000001")

# 10% overlap
heart_liver_10 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.10 -F 0.10", check.chr = FALSE)

summary(as.integer(heart_liver_10[,9]) > 0)

# 20% overlap
heart_liver_20 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.20 -F 0.20", check.chr = FALSE)

summary(as.integer(heart_liver_20[,9]) > 0)

# 25% overlap
heart_liver_25 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.25 -F 0.25", check.chr = FALSE)

summary(as.integer(heart_liver_25[,9]) > 0)

# 30% overlap
heart_liver_30 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.30 -F 0.30", check.chr = FALSE)

summary(as.integer(heart_liver_30[,9]) > 0)

#40% overlap
heart_liver_40 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.4 -F 0.4", check.chr = FALSE)

summary(as.integer(heart_liver_40[,9]) > 0)

# 50% overlap
heart_liver_50 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.50 -F 0.50", check.chr = FALSE)

summary(as.integer(heart_liver_50[,9]) > 0)

#60% overlap
heart_liver_60 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.6 -F 0.6", check.chr = FALSE)

summary(as.integer(heart_liver_60[,9]) > 0)

# 70% overlap
heart_liver_70 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.70 -F 0.70", check.chr = FALSE)

summary(as.integer(heart_liver_70[,9]) > 0)

#75% overlap
heart_liver_75 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.75 -F 0.75", check.chr = FALSE)

summary(as.integer(heart_liver_75[,9]) > 0)

# 80% overlap
heart_liver_80 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.80 -F 0.80", check.chr = FALSE)

summary(as.integer(heart_liver_80[,9]) > 0)

#90% overlap
heart_liver_90 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.9 -F 0.9", check.chr = FALSE)

summary(as.integer(heart_liver_90[,9]) > 0)

#100% overlap
heart_liver_100 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 1 -F 1", check.chr = FALSE)

summary(as.integer(heart_liver_100[,9]) > 0)

# Find the number of tDMRs depending on the bp overlap
summary(as.integer(heart_liver_1[,9]) > 0)
summary(as.integer(heart_liver_10[,9]) > 0)
summary(as.integer(heart_liver_20[,9]) > 0)
summary(as.integer(heart_liver_25[,9]) > 0)
summary(as.integer(heart_liver_30[,9]) > 0)
summary(as.integer(heart_liver_40[,9]) > 0)
summary(as.integer(heart_liver_50[,9]) > 0)
summary(as.integer(heart_liver_60[,9]) > 0)
summary(as.integer(heart_liver_70[,9]) > 0)
summary(as.integer(heart_liver_75[,9]) > 0)
summary(as.integer(heart_liver_80[,9]) > 0)
summary(as.integer(heart_liver_90[,9]) > 0)
summary(as.integer(heart_liver_100[,9]) > 0)

# Plot the number of tDMRs
conserved_tdmr_heart_liver_heart_lung <- c(2440, 2275, 1998, 1870, 1739, 1451, 1155, 866, 632, 507,426,294, 234)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung, xlab = "Percentage of overlap (bp)", ylab = "Number of overlapping chimp tDMRs",  pch = 19, main = "Overlapping Human tDMRs (HvK and LivLu)")

# Investigate the size of the tDMRs depending on the percentage overlap
check_1 <- heart_liver_1[which((as.integer(heart_liver_1[,9]) > 0) == TRUE),]
check_10 <- heart_liver_10[which((as.integer(heart_liver_10[,9]) > 0) == TRUE),]
check_20 <- heart_liver_20[which((as.integer(heart_liver_20[,9]) > 0) == TRUE),]
check_25 <- heart_liver_25[which((as.integer(heart_liver_25[,9]) > 0) == TRUE),]
check_30 <- heart_liver_30[which((as.integer(heart_liver_30[,9]) > 0) == TRUE),]
check_40 <- heart_liver_40[which((as.integer(heart_liver_40[,9]) > 0) == TRUE),]
check_50 <- heart_liver_50[which((as.integer(heart_liver_50[,9]) > 0) == TRUE),]
check_60 <- heart_liver_60[which((as.integer(heart_liver_60[,9]) > 0) == TRUE),]
check_70 <- heart_liver_70[which((as.integer(heart_liver_70[,9]) > 0) == TRUE),]
check_75 <- heart_liver_75[which((as.integer(heart_liver_75[,9]) > 0) == TRUE),]
check_80 <- heart_liver_80[which((as.integer(heart_liver_80[,9]) > 0) == TRUE),]
check_90 <- heart_liver_90[which((as.integer(heart_liver_90[,9]) > 0) == TRUE),]
check_100 <- heart_liver_100[which((as.integer(heart_liver_100[,9]) > 0) == TRUE),]

summary(as.integer(check_1[,9]))
summary(as.integer(check_10[,9]))
summary(as.integer(check_20[,9]))
summary(as.integer(check_25[,9]))
summary(as.integer(check_30[,9]))
summary(as.integer(check_40[,9]))
summary(as.integer(check_50[,9]))
summary(as.integer(check_60[,9]))
summary(as.integer(check_70[,9]))
summary(as.integer(check_75[,9]))
summary(as.integer(check_80[,9]))
summary(as.integer(check_90[,9]))
summary(as.integer(check_100[,9]))

# Plot the median tDMR length
conserved_tdmr_heart_liver_heart_lung_bp <- c(301, 321, 357, 376, 394, 429, 467, 498.5, 503.5, 514, 475.5, 390.5, 340)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung_bp, xlab = "Percentage of overlap (bp)", ylab = "Median length of overlap (bp)", pch = 19, main = "Median length of overlap between human tDMRs (HvK and LivLu)")

```


# Are the overlap % results robust with respect to species?

## Chimp heart versus liver and chimp heart versus lung
```{r}
# columns to pull
pull_col <- c(1:3, 16)

# make bed style format: change name from human to rhesus
humans_heart_liver_DMRs_bed <- chimps_heart_liver_DMRs[,pull_col]

# Get DMRs on the autosomal chromosomes only
humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed[which(humans_heart_liver_DMRs_bed[,1] != "chrX"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])

summary(humans_heart_liver_DMRs_bed_noX[,3] - humans_heart_liver_DMRs_bed_noX[,2])

sort_humans_heart_liver_DMRs_bed <- bedr.sort.region(humans_heart_liver_DMRs_bed_noX, check.chr = FALSE)

# make bed style format
humans_heart_lung_DMRs_bed <- chimps_heart_lung_DMRs[,pull_col]
humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed[which(humans_heart_lung_DMRs_bed[,1] != "chrX"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])
sort_humans_heart_lung_DMRs_bed <- bedr.sort.region(humans_heart_lung_DMRs_bed_noX, check.chr = FALSE)

summary(humans_heart_lung_DMRs_bed_noX[,3] - humans_heart_lung_DMRs_bed_noX[,2])

heart_lung <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5", check.chr = FALSE)

summary(heart_lung[,3] - heart_lung[,2])
summary(as.integer(heart_lung[,9]))
summary(as.integer(heart_lung[,9]) > 0)

lung_heart <- bedr(input = list(a = sort_humans_heart_lung_DMRs_bed, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5", check.chr = FALSE)

summary(as.integer(lung_heart[,9]))
summary(as.integer(lung_heart[,9]) > 0)

# Make a Venn diagram based on the results

summary(as.integer(heart_lung[,9]) > 0)
summary(as.integer(lung_heart[,9]) > 0)

grid.newpage()
draw.pairwise.venn(area1 = (26162+2607), area2 = (5919+2607), cross.area = 2607, category = c("Heart v. Liver", "Heart v. Lung"))
```

```{r}
# 1 bp overlap
heart_liver_1 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.00000001 -F 0.00000001", check.chr = FALSE)

# 10% overlap
heart_liver_10 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.10 -F 0.10", check.chr = FALSE)

summary(as.integer(heart_liver_10[,9]) > 0)

# 20% overlap
heart_liver_20 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.20 -F 0.20", check.chr = FALSE)

summary(as.integer(heart_liver_20[,9]) > 0)

# 25% overlap
heart_liver_25 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.25 -F 0.25", check.chr = FALSE)

summary(as.integer(heart_liver_25[,9]) > 0)

# 30% overlap
heart_liver_30 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.30 -F 0.30", check.chr = FALSE)

summary(as.integer(heart_liver_30[,9]) > 0)

#40% overlap
heart_liver_40 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.4 -F 0.4", check.chr = FALSE)

summary(as.integer(heart_liver_40[,9]) > 0)

# 50% overlap
heart_liver_50 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.50 -F 0.50", check.chr = FALSE)

summary(as.integer(heart_liver_50[,9]) > 0)

#60% overlap
heart_liver_60 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.6 -F 0.6", check.chr = FALSE)

summary(as.integer(heart_liver_60[,9]) > 0)

# 70% overlap
heart_liver_70 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.70 -F 0.70", check.chr = FALSE)

summary(as.integer(heart_liver_70[,9]) > 0)

#75% overlap
heart_liver_75 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.75 -F 0.75", check.chr = FALSE)

summary(as.integer(heart_liver_75[,9]) > 0)

# 80% overlap
heart_liver_80 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.80 -F 0.80", check.chr = FALSE)

summary(as.integer(heart_liver_80[,9]) > 0)

#90% overlap
heart_liver_90 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.9 -F 0.9", check.chr = FALSE)

summary(as.integer(heart_liver_90[,9]) > 0)

#100% overlap
heart_liver_100 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 1 -F 1", check.chr = FALSE)

summary(as.integer(heart_liver_100[,9]) > 0)

# Find the number of tDMRs depending on the bp overlap
summary(as.integer(heart_liver_1[,9]) > 0)
summary(as.integer(heart_liver_10[,9]) > 0)
summary(as.integer(heart_liver_20[,9]) > 0)
summary(as.integer(heart_liver_25[,9]) > 0)
summary(as.integer(heart_liver_30[,9]) > 0)
summary(as.integer(heart_liver_40[,9]) > 0)
summary(as.integer(heart_liver_50[,9]) > 0)
summary(as.integer(heart_liver_60[,9]) > 0)
summary(as.integer(heart_liver_70[,9]) > 0)
summary(as.integer(heart_liver_75[,9]) > 0)
summary(as.integer(heart_liver_80[,9]) > 0)
summary(as.integer(heart_liver_90[,9]) > 0)
summary(as.integer(heart_liver_100[,9]) > 0)

# Plot the number of tDMRs
conserved_tdmr_heart_liver_heart_lung <- c(621, 3837, 3614, 3472, 3332, 2974, 2607, 2180, 1740, 1502,1254,854, 621)
perc_overlap <- c(0.00000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung, xlab = "Percentage of overlap (bp)", ylab = "Number of overlapping chimp tDMRs",  pch = 19, main = "Overlapping Chimp tDMRs (HvLi and HvLu)")

# Investigate the size of the tDMRs depending on the percentage overlap

check_1 <- heart_liver_1[which((as.integer(heart_liver_1[,9]) > 0) == TRUE),]
check_10 <- heart_liver_10[which((as.integer(heart_liver_10[,9]) > 0) == TRUE),]
check_20 <- heart_liver_20[which((as.integer(heart_liver_20[,9]) > 0) == TRUE),]
check_25 <- heart_liver_25[which((as.integer(heart_liver_25[,9]) > 0) == TRUE),]
check_30 <- heart_liver_30[which((as.integer(heart_liver_30[,9]) > 0) == TRUE),]
check_40 <- heart_liver_40[which((as.integer(heart_liver_40[,9]) > 0) == TRUE),]
check_50 <- heart_liver_50[which((as.integer(heart_liver_50[,9]) > 0) == TRUE),]
check_60 <- heart_liver_60[which((as.integer(heart_liver_60[,9]) > 0) == TRUE),]
check_70 <- heart_liver_70[which((as.integer(heart_liver_70[,9]) > 0) == TRUE),]
check_75 <- heart_liver_75[which((as.integer(heart_liver_75[,9]) > 0) == TRUE),]
check_80 <- heart_liver_80[which((as.integer(heart_liver_80[,9]) > 0) == TRUE),]
check_90 <- heart_liver_90[which((as.integer(heart_liver_90[,9]) > 0) == TRUE),]
check_100 <- heart_liver_100[which((as.integer(heart_liver_100[,9]) > 0) == TRUE),]

summary(as.integer(check_1[,9]))
summary(as.integer(check_10[,9]))
summary(as.integer(check_20[,9]))
summary(as.integer(check_25[,9]))
summary(as.integer(check_30[,9]))
summary(as.integer(check_40[,9]))
summary(as.integer(check_50[,9]))
summary(as.integer(check_60[,9]))
summary(as.integer(check_70[,9]))
summary(as.integer(check_75[,9]))
summary(as.integer(check_80[,9]))
summary(as.integer(check_90[,9]))
summary(as.integer(check_100[,9]))

# Plot the median tDMR length
conserved_tdmr_heart_liver_heart_lung_bp <- c(301, 433, 459.5, 477.5, 491, 533.5, 560, 597, 608, 612,589,537.5, 428)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung_bp, xlab = "Percentage of overlap (bp)", ylab = "Median length of overlap (bp)", pch = 19, main = "Median length of overlap between chimp tDMRs (HvLi and HvLu)")

```


## Rhesus heart versus liver and rhesus heart versus lung
```{r}

# Heart versus liver

# columns to pull
pull_col <- c(1:3, 16)

# make bed style format: change name from human to rhesus
humans_heart_liver_DMRs_bed <- rhesus_heart_liver_DMRs[,pull_col]

# Remove chr17_gl000204_random
humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed[which(humans_heart_liver_DMRs_bed[,1] != "chr17_gl000204_random"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])

# Remove chr11_gl000202_random

humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed_noX[which(humans_heart_liver_DMRs_bed_noX[,1] != "chr11_gl000202_random"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])


# Get DMRs on the autosomal chromosomes only
humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed_noX[which(humans_heart_liver_DMRs_bed_noX[,1] != "chrX"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])

summary(humans_heart_liver_DMRs_bed_noX[,3] - humans_heart_liver_DMRs_bed_noX[,2])

sort_humans_heart_liver_DMRs_bed <- bedr.sort.region(humans_heart_liver_DMRs_bed_noX, check.chr = FALSE)

# Heart versus lung
humans_heart_lung_DMRs_bed <- rhesus_heart_lung_DMRs[,pull_col]

humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed[which(humans_heart_lung_DMRs_bed[,1] != "chr17_gl000204_random"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])

humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed_noX[which(humans_heart_lung_DMRs_bed_noX[,1] != "chr11_gl000202_random"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])



humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed_noX[which(humans_heart_lung_DMRs_bed_noX[,1] != "chrX"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])
sort_humans_heart_lung_DMRs_bed <- bedr.sort.region(humans_heart_lung_DMRs_bed_noX, check.chr = FALSE)

summary(humans_heart_lung_DMRs_bed_noX[,3] - humans_heart_lung_DMRs_bed_noX[,2])





heart_lung <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5", check.chr = FALSE)

summary(heart_lung[,3] - heart_lung[,2])
summary(as.integer(heart_lung[,9]))
summary(as.integer(heart_lung[,9]) > 0)

lung_heart <- bedr(input = list(a = sort_humans_heart_lung_DMRs_bed, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.5 -F 0.5", check.chr = FALSE)

summary(as.integer(lung_heart[,9]))
summary(as.integer(lung_heart[,9]) > 0)

# Make a Venn diagram based on the results

summary(as.integer(heart_lung[,9]) > 0)
summary(as.integer(lung_heart[,9]) > 0)

grid.newpage()
draw.pairwise.venn(area1 = (39057+2229), area2 = (4798+2229), cross.area = 2229, category = c("Heart v. Liver", "Heart v. Lung"))
```

```{r}
# 1 bp overlap
heart_liver_1 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.00000001 -F 0.00000001")

summary(as.integer(heart_liver_1[,9]) > 0)

# 10% overlap
heart_liver_10 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.10 -F 0.10", check.chr = FALSE)

summary(as.integer(heart_liver_10[,9]) > 0)

# 20% overlap
heart_liver_20 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.20 -F 0.20", check.chr = FALSE)

summary(as.integer(heart_liver_20[,9]) > 0)

# 25% overlap
heart_liver_25 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.25 -F 0.25", check.chr = FALSE)

summary(as.integer(heart_liver_25[,9]) > 0)

# 30% overlap
heart_liver_30 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.30 -F 0.30", check.chr = FALSE)

summary(as.integer(heart_liver_30[,9]) > 0)

#40% overlap
heart_liver_40 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.4 -F 0.4", check.chr = FALSE)

summary(as.integer(heart_liver_40[,9]) > 0)

# 50% overlap
heart_liver_50 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.50 -F 0.50", check.chr = FALSE)

summary(as.integer(heart_liver_50[,9]) > 0)

#60% overlap
heart_liver_60 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.6 -F 0.6", check.chr = FALSE)

summary(as.integer(heart_liver_60[,9]) > 0)

# 70% overlap
heart_liver_70 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.70 -F 0.70", check.chr = FALSE)

summary(as.integer(heart_liver_70[,9]) > 0)

#75% overlap
heart_liver_75 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.75 -F 0.75", check.chr = FALSE)

summary(as.integer(heart_liver_75[,9]) > 0)

# 80% overlap
heart_liver_80 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.80 -F 0.80", check.chr = FALSE)

summary(as.integer(heart_liver_80[,9]) > 0)

#90% overlap
heart_liver_90 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.9 -F 0.9", check.chr = FALSE)

summary(as.integer(heart_liver_90[,9]) > 0)

#100% overlap
heart_liver_100 <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 1 -F 1", check.chr = FALSE)

summary(as.integer(heart_liver_100[,9]) > 0)

# Find the number of tDMRs depending on the bp overlap
summary(as.integer(heart_liver_1[,9]) > 0)
summary(as.integer(heart_liver_10[,9]) > 0)
summary(as.integer(heart_liver_20[,9]) > 0)
summary(as.integer(heart_liver_25[,9]) > 0)
summary(as.integer(heart_liver_30[,9]) > 0)
summary(as.integer(heart_liver_40[,9]) > 0)
summary(as.integer(heart_liver_50[,9]) > 0)
summary(as.integer(heart_liver_60[,9]) > 0)
summary(as.integer(heart_liver_70[,9]) > 0)
summary(as.integer(heart_liver_75[,9]) > 0)
summary(as.integer(heart_liver_80[,9]) > 0)
summary(as.integer(heart_liver_90[,9]) > 0)
summary(as.integer(heart_liver_100[,9]) > 0)

# Plot the number of tDMRs
conserved_tdmr_heart_liver_heart_lung <- c(3654, 3520, 3245, 3472, 3332, 2974, 2607, 2180, 1740, 1502,1254,854, 621)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung, xlab = "Percentage of overlap (bp)", ylab = "Number of overlapping rhesus tDMRs (HvLi and HvLu)",  pch = 19)

# Investigate the size of the tDMRs depending on the percentage overlap

check_1 <- heart_liver_1[which((as.integer(heart_liver_1[,9]) > 0) == TRUE),]
check_10 <- heart_liver_10[which((as.integer(heart_liver_10[,9]) > 0) == TRUE),]
check_20 <- heart_liver_20[which((as.integer(heart_liver_20[,9]) > 0) == TRUE),]
check_25 <- heart_liver_25[which((as.integer(heart_liver_25[,9]) > 0) == TRUE),]
check_30 <- heart_liver_30[which((as.integer(heart_liver_30[,9]) > 0) == TRUE),]
check_40 <- heart_liver_40[which((as.integer(heart_liver_40[,9]) > 0) == TRUE),]
check_50 <- heart_liver_50[which((as.integer(heart_liver_50[,9]) > 0) == TRUE),]
check_60 <- heart_liver_60[which((as.integer(heart_liver_60[,9]) > 0) == TRUE),]
check_70 <- heart_liver_70[which((as.integer(heart_liver_70[,9]) > 0) == TRUE),]
check_75 <- heart_liver_75[which((as.integer(heart_liver_75[,9]) > 0) == TRUE),]
check_80 <- heart_liver_80[which((as.integer(heart_liver_80[,9]) > 0) == TRUE),]
check_90 <- heart_liver_90[which((as.integer(heart_liver_90[,9]) > 0) == TRUE),]
check_100 <- heart_liver_100[which((as.integer(heart_liver_100[,9]) > 0) == TRUE),]

summary(as.integer(check_1[,9]))
summary(as.integer(check_10[,9]))
summary(as.integer(check_20[,9]))
summary(as.integer(check_25[,9]))
summary(as.integer(check_30[,9]))
summary(as.integer(check_40[,9]))
summary(as.integer(check_50[,9]))
summary(as.integer(check_60[,9]))
summary(as.integer(check_70[,9]))
summary(as.integer(check_75[,9]))
summary(as.integer(check_80[,9]))
summary(as.integer(check_90[,9]))
summary(as.integer(check_100[,9]))

# Plot the median tDMR length
conserved_tdmr_heart_liver_heart_lung_bp <- c(404, 418, 450.5, 473, 497, 533, 564, 590, 619.5, 621,602,501, 378)
perc_overlap <- c(0.000000001, 0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.80, 0.90, 1)

plot(perc_overlap, conserved_tdmr_heart_liver_heart_lung_bp, xlab = "Percentage of overlap (bp)", ylab = "Median length of overlap (bp)", pch = 19, main = "Median length of overlap between rhesus tDMRs (HvLi and HvLu)")

```

An overlap of 0.70-0.80 maximizes the median length of overlap between tDMRs. This result appears robust to the tissues and species. 





# Pairwise tDMRs

```{r}
within_species_tDMR_overlap <- function(filename1, filename2){
  
# make bed style format: change name from human to rhesus
humans_heart_liver_DMRs_bed <- filename1[,pull_col]

# Get DMRs on the autosomal chromosomes only
humans_heart_liver_DMRs_bed_noX <- humans_heart_liver_DMRs_bed[which(humans_heart_liver_DMRs_bed[,1] != "chrX"),]

humans_heart_liver_DMRs_bed_noX[,1] <- as.character(humans_heart_liver_DMRs_bed_noX[,1])
humans_heart_liver_DMRs_bed_noX[,2] <- as.integer(humans_heart_liver_DMRs_bed_noX[,2])
humans_heart_liver_DMRs_bed_noX[,3] <- as.integer(humans_heart_liver_DMRs_bed_noX[,3])

summary(humans_heart_liver_DMRs_bed_noX[,3] - humans_heart_liver_DMRs_bed_noX[,2])

sort_humans_heart_liver_DMRs_bed <- bedr.sort.region(humans_heart_liver_DMRs_bed_noX, check.chr = FALSE)

# make bed style format
humans_heart_lung_DMRs_bed <- filename2[,pull_col]
humans_heart_lung_DMRs_bed_noX <- humans_heart_lung_DMRs_bed[which(humans_heart_lung_DMRs_bed[,1] != "chrX"),]

humans_heart_lung_DMRs_bed_noX[,1] <- as.character(humans_heart_lung_DMRs_bed_noX[,1])
humans_heart_lung_DMRs_bed_noX[,2] <- as.integer(humans_heart_lung_DMRs_bed_noX[,2])
humans_heart_lung_DMRs_bed_noX[,3] <- as.integer(humans_heart_lung_DMRs_bed_noX[,3])
sort_humans_heart_lung_DMRs_bed <- bedr.sort.region(humans_heart_lung_DMRs_bed_noX, check.chr = FALSE)

summary(humans_heart_lung_DMRs_bed_noX[,3] - humans_heart_lung_DMRs_bed_noX[,2])

heart_lung <- bedr(input = list(a = sort_humans_heart_liver_DMRs_bed, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 0.75 -F 0.75", check.chr = FALSE)

summary(heart_lung[,3] - heart_lung[,2])
summary(as.integer(heart_lung[,9]))
summary(as.integer(heart_lung[,9]) > 0)

lung_heart <- bedr(input = list(a = sort_humans_heart_lung_DMRs_bed, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.75 -F 0.75", check.chr = FALSE)

summary(as.integer(lung_heart[,9]))
summary(as.integer(lung_heart[,9]) > 0)

# Make a Venn diagram based on the results

stats_filename1 <- as.array(summary(as.integer(heart_lung[,9]) > 0))
stats_filename2 <- as.array(summary(as.integer(lung_heart[,9]) > 0))

stats_filename12 <- as.data.frame(cbind(stats_filename1[2], stats_filename1[3], stats_filename1[4], stats_filename2[2], stats_filename2[3], stats_filename2[4]))

#stats_filename12[,1] <- as.integer(stats_filename12[,1])
#stats_filename12[,2] <- as.integer(stats_filename12[,2])
#stats_filename12[,3] <- as.integer(stats_filename12[,3])
#stats_filename12[,4] <- as.integer(stats_filename12[,4])
#stats_filename12[,5] <- as.integer(stats_filename12[,5])
#stats_filename12[,6] <- as.integer(stats_filename12[,6])


return(stats_filename12)

}

```


## Heart versus liver and heart versus lung

```{r}
within_species_tDMR_overlap(humans_heart_liver_DMRs, humans_heart_lung_DMRs)

grid.newpage()
draw.pairwise.venn(area1 = (20256+2305), area2 = (11903+2305), cross.area = 2305, category = c("Human Heart v. Liver", "Heart v. Lung"))

within_species_tDMR_overlap(chimps_heart_liver_DMRs, chimps_heart_lung_DMRs)

grid.newpage()
draw.pairwise.venn(area1 = (27267+1502), area2 = (7024+1502), cross.area = 1502, category = c("Chimp Heart v. Liver", "Heart v. Lung"))

within_species_tDMR_overlap(rhesus_heart_liver_DMRs, rhesus_heart_lung_DMRs)

grid.newpage()
draw.pairwise.venn(area1 = (39998+1288), area2 = (5739+1288), cross.area = 1288, category = c("Chimp Heart v. Liver", "Heart v. Lung"))

```


# Which overlap promoters?

```{r}
# Import data about orthologous promoters 
boundaries_within_250_humans <- read.delim("~/Desktop/Reg_Evo_Primates/data/250_complete_interval_humans.bed")
boundaries_within_250_humans[,1] <- as.character(boundaries_within_250_humans[,1])
  
# How many tDMRs cover various percentages of the orthologous promoters
sort_boundaries_within_250_humans <- bedr.sort.region(boundaries_within_250_humans, check.chr = FALSE)
sort_boundaries_within_250_humans <- bedr.merge.region(boundaries_within_250_humans, check.chr = FALSE)

# Various percentages 

heart_liver_promoter_10 <- bedr(input = list(a = boundaries_within_250_humans, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.1", check.chr = FALSE)

summary(heart_liver_promoter_10[,11] > 0)

heart_liver_promoter_50 <- bedr(input = list(a = boundaries_within_250_humans, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 0.5", check.chr = FALSE)

summary(heart_liver_promoter_50[,11] > 0)

heart_liver_promoter_100 <- bedr(input = list(a = boundaries_within_250_humans, b = sort_humans_heart_liver_DMRs_bed), method = "intersect",  params = "-wao -f 1", check.chr = FALSE)

summary(heart_liver_promoter_100[,11] > 0)

# What if we look at 100% coverage in the human hearts and lungs?

heart_lung_promoter_100 <- bedr(input = list(a = boundaries_within_250_humans, b = sort_humans_heart_lung_DMRs_bed), method = "intersect",  params = "-wao -f 1", check.chr = FALSE)

summary(heart_lung_promoter_100[,11] > 0)
```

