---
title: "Home"
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

## Project Introduction

## Sample Information and Record of Technical Variables

*  [Technical variables documented/tested (sample information, RNA-seq, BS-seq, QC, processing, etc.)](../docs/List_technical_variables.docx)

## RNA-seq Data

### Pre-processing

*  [Unmapped and Mapped Reads](../analysis/Plots_unmapped_mapped_RNAseq_reads.html)

### Normalization

*  [Raw counts](../analysis/Normalization_plots.html)
*  [Filtering](../analysis/Filtering_analysis.html)
*  [GC content normalization](../analysis/GC_content_normalization_CHT.html)
*  [Library size adjustment and decoupling the relationship between expression mean and variance](../analysis/TMM_voom_on_GC_normalized_data.html)

### Analysis with Technical Factors

*  [Testing for confounders](../analysis/Tech_factor_analysis1_gene_exp.html)
*  [Further analysis with potential confounders](../analysis/Analysis_of_confounders2_gene_exp.html)
*  [Best set of technical factors](../analysis/GLMnet_gene_exp.html)
*  [Correlation between technical factors in the best sets and expression](../analysis/Follow_up_best_set_tech_factors.html)

### Final filtering, normalizations, and analysis with technical factors (12,184 genes)

*  [Processing using filtering with lowly expressed genes](../analysis/Correlation_bet_tech_factors_in_best_set_and_expression_stringent_filtering.html)

### Univariate analysis on gene expression

* [Test for DE genes with ASH](../analysis/DE_RIN_Plus_ASH.html)
* [Significance of conservation in tissue specific gene expression](../analysis/Tissue_specific_overlap.html)
* [Comparison of conserved tissue specific genes to other data](../analysis/Comparison_to_GTEx.html)

## Methylation Data

*  [Human-chimpanzee orthologous promoters](../analysis/Hg19_PanTro3_TSS.html)
*  [Human-chimpanzee-rhesus orthologous promoters](../analysis/hg19_to_rhesus_and_chimp_coor.html)
*  Clustering after testing for technical variables 

## Differentiatlly methylated regions

*  [Pairwise tDMRs I](../analysis/Finding_DMRs.html)
*  [Pairwise tDMRs II](../analysis/Pairwise_tDMRs.html)
*  [Finding conserved tDMRs](../analysis/Finding_conserved_DMRs.html)
*  [Overlap of tDMRs with H3K27ac and DHS](../analysis/Overlap_reg_features.html)


## Integration of gene expression levels and methylatiom levels: humans and chimpanzees

* [Inter-species and inter-tissue example with FSR](../analysis/All_cpgs_for_orth_genes.html)

## Effect size estimation

* [Effect size of DE and non-DE genes: humans and chimpanzees](../analysis/Effect_size_2lms.html)

* [Effect size of DE and non-DE genes: humans and rhesus](../analysis/Effect_size_2lms_human_rhesus.html)

## Additional figure/table generation

* [Global surveys of RNA-seq and BS-seq data](../analysis/gene_exp_corr.html)
* [About samples](../analysis/Figures_samples_info.html)
* [Final figure](../analysis/Figures_integration_analysis.html)



## Notes

*  [RNA-seq Normalization Notes](../docs/Normalization_Notes.docx)




