<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#project-introduction">Project Introduction</a></li>
<li><a href="#sample-information-and-record-of-technical-variables">Sample Information and Record of Technical Variables</a></li>
<li><a href="#rna-seq-data">RNA-seq Data</a><ul>
<li><a href="#pre-processing">Pre-processing</a></li>
<li><a href="#normalization">Normalization</a></li>
<li><a href="#analysis-with-technical-factors">Analysis with Technical Factors</a></li>
<li><a href="#final-filtering-normalizations-and-analysis-with-technical-factors-12184-genes">Final filtering, normalizations, and analysis with technical factors (12,184 genes)</a></li>
<li><a href="#univariate-analysis-on-gene-expression">Univariate analysis on gene expression</a></li>
</ul></li>
<li><a href="#methylation-data">Methylation Data</a></li>
<li><a href="#differentially-methylated-regions">Differentially methylated regions</a></li>
<li><a href="#integration-of-gene-expression-levels-and-methylation-levels--interspecies-de-genes">Integration of gene expression levels and methylation levels- interspecies DE genes</a></li>
<li><a href="#integration-of-gene-expression-levels-and-methylation-levels--intertissue-de-genes">Integration of gene expression levels and methylation levels- intertissue DE genes</a></li>
<li><a href="#associated-analyses">Associated analyses</a></li>
<li><a href="#additional-figuretable-generation">Additional figure/table generation</a></li>
<li><a href="#revisions">Revisions</a></li>
<li><a href="#notes">Notes</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2019-07-25</p>
<p><strong>Code version:</strong> cbe7dd2181c4b0621f7ac9e39431d269c52c352f</p>
<div id="project-introduction" class="section level2">
<h2>Project Introduction</h2>
</div>
<div id="sample-information-and-record-of-technical-variables" class="section level2">
<h2>Sample Information and Record of Technical Variables</h2>
<ul>
<li><a href="../docs/List_technical_variables.docx">Technical variables documented/tested (sample information, RNA-seq, BS-seq, QC, processing, etc.)</a></li>
</ul>
</div>
<div id="rna-seq-data" class="section level2">
<h2>RNA-seq Data</h2>
<div id="pre-processing" class="section level3">
<h3>Pre-processing</h3>
<ul>
<li><a href="../analysis/Plots_unmapped_mapped_RNAseq_reads.html">Unmapped and Mapped Reads</a></li>
</ul>
</div>
<div id="normalization" class="section level3">
<h3>Normalization</h3>
<ul>
<li><a href="../analysis/Normalization_plots.html">Raw counts</a></li>
<li><a href="../analysis/Filtering_analysis.html">Filtering</a></li>
<li><a href="../analysis/GC_content_normalization_CHT.html">GC content normalization</a></li>
<li><a href="../analysis/TMM_voom_on_GC_normalized_data.html">Library size adjustment and decoupling the relationship between expression mean and variance</a></li>
</ul>
</div>
<div id="analysis-with-technical-factors" class="section level3">
<h3>Analysis with Technical Factors</h3>
<ul>
<li><a href="../analysis/Tech_factor_analysis1_gene_exp.html">Testing for confounders</a></li>
<li><a href="../analysis/Technical_factor_figures.html">Technical factor figures</a></li>
<li><a href="../analysis/Analysis_of_confounders2_gene_exp.html">Further analysis with potential confounders</a></li>
<li><a href="../analysis/GLMnet_gene_exp.html">Best set of technical factors</a></li>
<li><a href="../analysis/Follow_up_best_set_tech_factors.html">Correlation between technical factors in the best sets and expression</a></li>
</ul>
</div>
<div id="final-filtering-normalizations-and-analysis-with-technical-factors-12184-genes" class="section level3">
<h3>Final filtering, normalizations, and analysis with technical factors (12,184 genes)</h3>
<ul>
<li><a href="../analysis/PC_RNA_seq.html">Technical factor analysis</a></li>
<li><a href="../analysis/Correlation_bet_tech_factors_in_best_set_and_expression_stringent_filtering.html">Processing using filtering with lowly expressed genes</a></li>
</ul>
</div>
<div id="univariate-analysis-on-gene-expression" class="section level3">
<h3>Univariate analysis on gene expression</h3>
<ul>
<li><a href="../analysis/DE_RIN_Plus_ASH.html">Test for DE genes with ASH</a></li>
<li><a href="../analysis/DE_RIN_Interactions_Plus_ASH.html">Test for signficant interactions</a></li>
<li><a href="../analysis/Recapitulate_Pai_2011.html">Overlap analysis</a></li>
<li><a href="../analysis/More_lenient_tissue_specific.html">More lenient tissue specific</a></li>
<li><a href="../analysis/Tissue_specific_overlap.html">Significance of conservation in tissue specific gene expression</a></li>
<li><a href="../analysis/Comparison_to_GTEx.html">Comparison of conserved tissue specific genes to other data</a></li>
<li><a href="../analysis/Human_chimp_var.html">Variance of gene expression between tissues</a></li>
</ul>
</div>
</div>
<div id="methylation-data" class="section level2">
<h2>Methylation Data</h2>
<ul>
<li><a href="../analysis/Hg19_PanTro3_TSS.html">Human-chimpanzee orthologous promoters</a></li>
<li><a href="../analysis/hg19_to_rhesus_and_chimp_coor.html">Human-chimpanzee-rhesus orthologous promoters</a></li>
</ul>
</div>
<div id="differentially-methylated-regions" class="section level2">
<h2>Differentially methylated regions</h2>
<ul>
<li><a href="../analysis/Compare_methylated_regions.html">Initial analysis- Size of DMRs</a></li>
<li><a href="../analysis/Finding_DMRs.html">Pairwise tDMRs I</a></li>
<li><a href="../analysis/Pairwise_tDMRs.html">Pairwise tDMRs II</a></li>
<li><a href="../analysis/tissue_specific_tDMRs.html">Tissue specific tDMRs</a></li>
<li><a href="../analysis/Finding_conserved_DMRs.html">Finding conserved tDMRs I</a></li>
<li>[List of conserved tDMRs] conserved tDMRs(../Conserved_tDMR_list.html)</li>
<li><a href="../analysis/Pairwise_tDMR_overlap.html">Pairwise tDMR overlap</a></li>
<li><a href="Overlap_H3K27ac.html">Overlap of tDMRs with H3K27ac and DHS</a></li>
<li><a href="../analysis/Overlap_reg_features.html">Exploring the parameters for this overlap analysis</a></li>
<li><a href="../analysis/Pairwise_SDMRs.html">Pairwise sDMRs</a></li>
</ul>
</div>
<div id="integration-of-gene-expression-levels-and-methylation-levels--interspecies-de-genes" class="section level2">
<h2>Integration of gene expression levels and methylation levels- interspecies DE genes</h2>
<ul>
<li><a href="../analysis/Final_effect_size_human_chimp.html">Human-chimpanzee with 7125 genes</a></li>
<li><a href="../analysis/Final_effect_size_human_chimp_4155.html">Human-chimpanzee with 4155 genes</a></li>
<li><a href="../analysis/Final_effect_size_human_rhesus_4155.html">Human-rhesus macaque with 4155 genes</a></li>
</ul>
</div>
<div id="integration-of-gene-expression-levels-and-methylation-levels--intertissue-de-genes" class="section level2">
<h2>Integration of gene expression levels and methylation levels- intertissue DE genes</h2>
<ul>
<li><a href="../analysis/Final_effect_size_human_tissues.html">Intertissue humans</a></li>
<li><a href="../analysis/Final_effect_size_chimp_tissues.html">Intertissue chimps</a></li>
<li><a href="../analysis/Final_effect_size_rhesus_tissues_4155.html">Intertissue rhesus macaques</a></li>
<li><a href="../analysis/Final_effect_size_tissue_specific.html">Tissue specific DE genes</a></li>
</ul>
</div>
<div id="associated-analyses" class="section level2">
<h2>Associated analyses</h2>
<ul>
<li><a href="../analysis/Compare_effect_sizes_DE_pairs.html">Compare effect sizes</a></li>
<li><a href="../analysis/Conserved_nonconserved_correlation.html">Conserved versus nonconserved DE genes- humans and chimpanzees</a></li>
<li><a href="../analysis/Conserved_nonconserved_correlation_rhesus.html">Conserved versus nonconserved DE genes- humans and rhesus macaques</a></li>
</ul>
</div>
<div id="additional-figuretable-generation" class="section level2">
<h2>Additional figure/table generation</h2>
<ul>
<li><a href="../analysis/gene_exp_corr.html">Global surveys of RNA-seq and BS-seq data</a></li>
<li><p><a href="../analysis/Figures_samples_info.html">About samples</a></p></li>
<li><a href="../analysis/Figure_1.html">Figure 1</a></li>
<li><a href="../analysis/Figures_methylation.html">Figure 3</a></li>
<li><a href="../analysis/Final_effect_size_figure_4.html">Figure 4</a></li>
<li><a href="../analysis/Final_effect_size_figure_5.html">Figure 5</a></li>
<li><a href="../analysis/Supplementary_figures_clustering.html">Figures S7-9</a></li>
<li><a href="../analysis/Figure_Supplementary10.html">Figure S10</a></li>
<li><p><a href="../analysis/Figure_Supplementary11.html">Figure S11</a></p></li>
</ul>
</div>
<div id="revisions" class="section level2">
<h2>Revisions</h2>
<p>Many thanks to the editor and 3 reviewers who provided helpful comments on this manuscript.</p>
<ul>
<li>[Matched versus non matched individuals]</li>
<li><a href="../analysis/Revisions_H3K27ac.html">Futher H3K27ac analysis</a></li>
<li><a href="../analysis/Revisions_genomes.html">HG19 versus HG38 genome builds</a></li>
<li><a href="../analysis/Supplementary_figures_clustering.html">Clustering</a></li>
<li><a href="../analysis/revisions_networks.html">Network analysis</a></li>
<li><a href="../analysis/revisions_misc.html">Number of genes per tissue</a></li>
</ul>
</div>
<div id="notes" class="section level2">
<h2>Notes</h2>
<ul>
<li><a href="../docs/Normalization_Notes.docx">RNA-seq Normalization Notes</a></li>
</ul>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
