<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#project-introduction">Project Introduction</a></li>
<li><a href="#sample-information-and-record-of-technical-variables">Sample Information and Record of Technical Variables</a></li>
<li><a href="#rna-seq-data">RNA-seq Data</a><ul>
<li><a href="#pre-processing">Pre-processing</a></li>
<li><a href="#normalization">Normalization</a></li>
<li><a href="#testing-the-relationship-between-a-large-number-of-technical-factors-recorded-throughout-our-experiments-and-the-biological-variables-of-interest-in-the-study">Testing the relationship between a large number of technical factors recorded throughout our experiments and the biological variables of interest in the study</a></li>
<li><a href="#final-filtering-normalizations-and-analysis-with-technical-factors-12184-genes">Final filtering, normalizations, and analysis with technical factors (12,184 genes)</a></li>
<li><a href="#univariate-analysis-on-gene-expression-levels">Univariate analysis on gene expression levels</a></li>
</ul></li>
<li><a href="#methylation-data">Methylation Data</a></li>
<li><a href="#differentially-methylated-regions-dmrs">Differentially methylated regions (DMRs)</a></li>
<li><a href="#integration-of-gene-expression-levels-and-methylation-levels--interspecies-de-genes">Integration of gene expression levels and methylation levels- interspecies DE genes</a></li>
<li><a href="#integration-of-gene-expression-levels-and-methylation-levels--intertissue-de-genes">Integration of gene expression levels and methylation levels- intertissue DE genes</a></li>
<li><a href="#integration-of-gene-expression-levels-and-methylation-levels--conserved-versus-non-conserved-intertissue-de-genes">Integration of gene expression levels and methylation levels- conserved versus non-conserved intertissue DE genes</a></li>
<li><a href="#additional-figuretable-generation">Additional figure/table generation</a></li>
<li><a href="#revisions">Revisions</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2021-12-27</p>
<p><strong>Code version:</strong> f339be92373f0bba81d2d6c636d1156c79a269c3</p>
<div id="project-introduction" class="section level2">
<h2>Project Introduction</h2>
<p>The project website for Blake, Roux, et al., “A comparison of gene expression and DNA methylation patterns across tissues and species”. Many of the main conclusions from each section are quoted directly from the <a href="https://genome.cshlp.org/content/30/2/250.long">publication</a>.</p>
</div>
<div id="sample-information-and-record-of-technical-variables" class="section level2">
<h2>Sample Information and Record of Technical Variables</h2>
<ul>
<li><a href="../docs/List_technical_variables.docx">Technical variables documented/tested (sample information, RNA-seq, BS-seq, QC, processing, etc.).</a><br />
Note: Clicking on this link will lead to the automatic download of a .docx file. For more information about the samples, see <a href="https://genome.cshlp.org/content/30/2/250/suppl/DC1">Supplemental_Table_S1.xls</a>.</li>
<li>Main conclusion from this section: All sample processing steps should be recorded in a sample history file that includes anything that happened to any sample. This documentation can help provide evidence that a phenomenon is driven by biological rather than technical factors.</li>
</ul>
</div>
<div id="rna-seq-data" class="section level2">
<h2>RNA-seq Data</h2>
<div id="pre-processing" class="section level3">
<h3>Pre-processing</h3>
<ul>
<li><a href="../analysis/Plots_unmapped_mapped_RNAseq_reads.html">Exploratory analysis of unmapped and mapped reads.</a></li>
<li>Conclusion from this section: There is a sufficient number of (orthologous) reads per sample to proceed with the analysis.</li>
</ul>
</div>
<div id="normalization" class="section level3">
<h3>Normalization</h3>
<ul>
<li><a href="../analysis/Normalization_plots.html">Visualize raw gene expression counts.</a></li>
<li><a href="../analysis/Filtering_analysis.html">Establishing filtering criteria for multispecies RNA-seq data.</a></li>
<li><a href="../analysis/GC_content_normalization_CHT.html">Should the data be normalized for GC content?</a><br />
</li>
<li><a href="../analysis/TMM_voom_on_GC_normalized_data.html">Library size adjustment and decoupling the relationship between expression mean and variance with edgeR+voom.</a></li>
<li>Conclusion from this section: For each tissue-species pair, the adjusted raw counts and the actual raw counts were highly correlated (&gt; 0.98). Therefore, I did not adjust for read depth or GC content in the RNA-sequencing data.</li>
</ul>
</div>
<div id="testing-the-relationship-between-a-large-number-of-technical-factors-recorded-throughout-our-experiments-and-the-biological-variables-of-interest-in-the-study" class="section level3">
<h3>Testing the relationship between a large number of technical factors recorded throughout our experiments and the biological variables of interest in the study</h3>
<ul>
<li><a href="../analysis/Tech_factor_analysis1_gene_exp.html">Part 1: Identifying variables potentially confounded with biological variables of interest.</a>. As described in <a href="https://f1000research.com/articles/4-121">Gilad and Mizrahi-Man et al 2015</a>, study design is imperative in primate comparative genomics. In this study, I took extra care to record and analyze non-biological variables of interest.</li>
<li><a href="../analysis/Analysis_of_confounders2_gene_exp.html">Part 2: Relationships between potential confounders and biological variables of interest.</a></li>
<li><a href="../analysis/GLMnet_gene_exp.html">Using glmnet to find the “best sets” of technical factors.</a></li>
<li><a href="../analysis/Follow_up_best_set_tech_factors.html">Correlation between technical factors in the best sets and gene expression levels.</a></li>
<li>Conclusion from this section: There were no recorded technical confounders with tissue but two technical factors were confounded with species: time postmortem until collection and RNA extraction date. Because of the opportunistic nature of sample collection, these confounders are practically impossible to avoid in comparative studies of primates (especially apes). We discuss possible implications of these confounders throughout the <a href="https://genome.cshlp.org/content/30/2/250.long">publication</a>.</li>
</ul>
</div>
<div id="final-filtering-normalizations-and-analysis-with-technical-factors-12184-genes" class="section level3">
<h3>Final filtering, normalizations, and analysis with technical factors (12,184 genes)</h3>
<ul>
<li><a href="../analysis/Correlation_bet_tech_factors_in_best_set_and_expression_stringent_filtering.html">Processing using filtering with lowly expressed genes.</a></li>
<li>Conclusion from this section: I estimated gene expression levels using an approach designed to prevent biases driven by sequence divergence across species and excluded data from genes that were lowly expressed in more than half of the samples (as well as data from one human heart sample that was an obvious outlier, probably owing to a sample swap). I normalized the distribution of gene expression levels to remove systematic expression differences between species (maximizing the number of genes with invariant expression levels across species corresponds to our null hypothesis).</li>
</ul>
</div>
<div id="univariate-analysis-on-gene-expression-levels" class="section level3">
<h3>Univariate analysis on gene expression levels</h3>
<ul>
<li><a href="../analysis/DE_RIN_Plus_ASH.html">Characterize differentially expressed (DE) genes with ASH.</a></li>
<li><a href="../analysis/DE_RIN_Interactions_Plus_ASH.html">Test for signficant interactions and perform Gene Ontology (GO) enrichments.</a></li>
<li><a href="../analysis/Recapitulate_Pai_2011.html">Code for testing the enrichment of the number of overlapping genes using the hypergeometric distribution.</a></li>
<li><a href="../analysis/More_lenient_tissue_specific.html">Number of genes consistent with a tissue specific pattern using a more lenient definition of “tissue specific”.</a> Since there are only 4 tissues in this analysis, it is not possible to separate whether the gene expression level in a tissue is truly tissue specific (versus all tissues) or just the other 3 tissues included in the analysis.<br />
</li>
<li><a href="../analysis/Tissue_specific_overlap.html">Significance of conservation in tissue specific gene expression patterns using the hypergeometric distribution.</a></li>
<li><a href="../analysis/Comparison_to_GTEx.html">Comparison of conserved tissue specific genes to other data.</a> The goal of this script is to compare the order of the gene expression in our tissues to the order in a well established biobank, GTEx.</li>
<li>Conclusions from this section: Overall, the patterns of inter-tissue differences in gene expression levels are similar in the three species, significantly more so than expected by chance alone (P &lt; 10−16, hypergeometric distribution). A range of 17%–26% of inter-tissue DE genes have conserved inter-tissue expression patterns in all three species. These genes are also highly enriched with functional annotations that are relevant to the corresponding tissue, which is suggestive of functional importance.</li>
</ul>
</div>
</div>
<div id="methylation-data" class="section level2">
<h2>Methylation Data</h2>
<ul>
<li><a href="../analysis/PC_RNA_seq.html">Identifying variables related to methylation processing that are potentially confounded with biological variables of interest.</a></li>
<li><a href="../analysis/Hg19_PanTro3_TSS.html">Identifying genes for which promoters are orthologous across humans and chimpanzees.</a></li>
<li><a href="../analysis/hg19_to_rhesus_and_chimp_coor.html">Identifying genes for which promoters are orthologous across humans, chimpanzees, and rhesus macaques.</a></li>
<li>Conclusion from this section: I identified 7,725 orthologous genes with expression data and corresponding promoter DNA methylation data in humans and chimpanzees, and 4155 orthologous genes with the same information for all three species.</li>
</ul>
</div>
<div id="differentially-methylated-regions-dmrs" class="section level2">
<h2>Differentially methylated regions (DMRs)</h2>
<p>Identifying DMRs was computationally intensive using the BSmooth package. Therefore, tissue and species DMRs were identified using the orthologous CpGs on a HPC. The output of each file was either (1) a list of sDMRs where the same tissue was analyzed across species or (2) a list of tDMRs where different tissues were analyzed within a species.</p>
<ul>
<li><a href="../analysis/Compare_methylated_regions.html">Obtain statistics about the sizes of DMRs.</a></li>
<li><a href="../analysis/Finding_DMRs.html">Defining DMRs based on the base pair (bp) overlap.</a></li>
<li><a href="../analysis/Pairwise_tDMRs.html">Identifying human-specific and conserved DMRs using the package bedr.</a></li>
<li><a href="../analysis/tissue_specific_tDMRs.html">Identifying DMRs with a pattern consistent with tissue specificity.</a> Again, the methylation pattern in 1 tissue is being compared to methylation in only 3 other tissues.</li>
<li><a href="../analysis/Pairwise_tDMR_overlap.html">Enrichment of tDMR overlap using the hypergeometric distribution.</a></li>
<li><a href="../analysis/Overlap_reg_features.html">Overlap between DMRs and H3K27ac, an epigenetic mark often associated with active gene expression.</a></li>
<li>Conclusions from this section: Inter-tissue DNA methylation differences within a species tend to be conserved, consistent with the observations of previous studies. In all four tissues, &gt;59% of conserved DMRs are hypomethylated in a tissue-specific manner. We evaluated the overlap between tissue-specific DMRs and genomic regions marked with H3K27ac, a mark often associated with active gene expression (The ENCODE Project Consortium 2012). We found that conserved hypomethylated tissue-specific DMRs were annotated with H3K27ac more frequently than tissue-specific DMRs identified only in humans (P &lt; 0.001, difference of proportions test).</li>
</ul>
</div>
<div id="integration-of-gene-expression-levels-and-methylation-levels--interspecies-de-genes" class="section level2">
<h2>Integration of gene expression levels and methylation levels- interspecies DE genes</h2>
<p>What percentage of interspecies gene expression level differences can be explained by variation in DNA methylation levels?</p>
<ul>
<li><a href="../analysis/Final_effect_size_human_chimp.html">In humans and chimpanzees with 7125 orthologous genes.</a></li>
<li><a href="../analysis/Final_effect_size_human_chimp_4155.html">In human and chimpanzees with 4155 human-chimp-rhesus orthologous genes.</a></li>
<li><a href="../analysis/Final_effect_size_human_rhesus_4155.html">In humans and rhesus macaques with 4155 orthologous genes.</a></li>
<li>Conclusion from this section: When considering DE genes between humans and chimpanzees (in at least one tissue), I found that between 11% and 25% of genes (depending on tissue) showed a difference in the effect of species on gene expression level.</li>
</ul>
</div>
<div id="integration-of-gene-expression-levels-and-methylation-levels--intertissue-de-genes" class="section level2">
<h2>Integration of gene expression levels and methylation levels- intertissue DE genes</h2>
<p>Within a species, what percentage of intertissue gene expression level differences can be explained by variation in DNA methylation levels?</p>
<ul>
<li><a href="../analysis/Final_effect_size_human_tissues.html">Intertissue in humans.</a></li>
<li><a href="../analysis/Final_effect_size_chimp_tissues.html">Intertissue in chimpanzees.</a></li>
<li><a href="../analysis/Final_effect_size_rhesus_tissues_4155.html">Intertissue in rhesus macaques.</a></li>
<li><a href="../analysis/Final_effect_size_tissue_specific.html">In tissue specific DE genes.</a></li>
<li>Conclusion from this section: 9%–17% of inter-tissue gene expression differences could potentially be explained by DNA methylation differences across tissues (FSR 5%).</li>
</ul>
</div>
<div id="integration-of-gene-expression-levels-and-methylation-levels--conserved-versus-non-conserved-intertissue-de-genes" class="section level2">
<h2>Integration of gene expression levels and methylation levels- conserved versus non-conserved intertissue DE genes</h2>
<p>What percentage of conserved intertissue gene expression level differences can be explained by variation in DNA methylation levels? How does this percentage compare to the percentage of non-conserved intertissue gene expression level differences that can be explained by variation in DNA methylation levels?</p>
<ul>
<li><a href="../analysis/Conserved_nonconserved_correlation.html">Conserved versus nonconserved DE genes- humans and chimpanzees.</a></li>
<li><a href="../analysis/Conserved_nonconserved_correlation_rhesus.html">Conserved versus nonconserved DE genes- humans and rhesus macaques.</a></li>
<li>Conclusion from this section: When focusing on regulatory patterns that are most likely to be functional—namely, conserved inter-tissue gene regulatory differences, these differences were more likely to be explained by variation in DNA methylation levels than nonconserved inter-tissue gene expression differences (minimum difference is 7%, P &lt; 0.005 for all comparisons; at FDR &lt;5% and FSR &lt;5%).</li>
</ul>
</div>
<div id="additional-figuretable-generation" class="section level2">
<h2>Additional figure/table generation</h2>
<ul>
<li><a href="../analysis/gene_exp_corr.html">Correlations of gene expression levels in tissues from matched versus unmatched individuals.</a></li>
<li><a href="../analysis/Figure_1.html">Figure 1.</a></li>
<li><a href="../analysis/Figures_methylation.html">Figure 3.</a></li>
<li><a href="../analysis/Final_effect_size_figure_4.html">Figure 4.</a></li>
<li><a href="../analysis/Final_effect_size_figure_5.html">Figure 5.</a></li>
<li><a href="../analysis/Supplementary_figures_clustering.html">Figures S7-9.</a></li>
<li><a href="../analysis/Figure_Supplementary10.html">Figure S10.</a></li>
<li><a href="../analysis/Figure_Supplementary11.html">Figure S11.</a></li>
</ul>
</div>
<div id="revisions" class="section level2">
<h2>Revisions</h2>
<p>Many thanks to the editor and 3 reviewers who provided helpful comments on this manuscript.</p>
<ul>
<li><a href="../analysis/Revisions_H3K27ac.html">Additional H3K27ac analysis, including GO enrichment.</a></li>
<li><a href="../analysis/Revisions_genomes.html">Comparing gene expression levels with reads mapped using the HG19 versus HG38 genome builds.</a></li>
<li><a href="../analysis/revisions_networks.html">Network analysis.</a></li>
<li><a href="../analysis/revisions_misc.html">Obtain the number of genes per tissue.</a></li>
<li>Conclusion from this section: I found that genes with conserved tissue-specific expression patterns are more likely to appear as nodes in the networks than genes without tissue-specific expression patterns or genes whose tissue-specific expression patterns are not conserved (P &lt; 10−5).</li>
</ul>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
