<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>R Notebook</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Notebook</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#load-librariesdata">Load libraries/data</a></li>
<li><a href="#find-de-genes-fdr-1">Find DE genes (FDR = 1%)</a></li>
<li><a href="#run-the-mediation-analysis-on-unscaled-data">Run the mediation analysis on unscaled data</a></li>
<li><a href="#run-the-mediation-analysis-on-scaled-data">Run the mediation analysis on scaled data</a></li>
<li><a href="#permute-the-methylation-values-scale-the-methylation-values-and-then">Permute the methylation values, scale the methylation values, and then</a><ul>
<li><a href="#permute-the-scaled-values-and-then-run-the-mediation-analysis">Permute the scaled values and then run the mediation analysis</a></li>
</ul></li>
</ul>
</div>

<p>This is a test of the mediation analysis.</p>
<div id="load-librariesdata" class="section level1">
<h1>Load libraries/data</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.4.3 (2017-11-30)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: OS X El Capitan 10.11.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] compiler_3.4.3  backports_1.1.2 magrittr_1.5    rprojroot_1.3-2
##  [5] tools_3.4.3     htmltools_0.3.6 yaml_2.1.16     Rcpp_0.12.15   
##  [9] stringi_1.1.6   rmarkdown_1.8   knitr_1.18      stringr_1.2.0  
## [13] digest_0.6.14   evaluate_0.10.1</code></pre>
<pre class="r"><code># Load Joyce&#39;s package
devtools::install_github(&quot;jhsiao999/mediation/pkg&quot;)</code></pre>
<pre><code>## Downloading GitHub repo jhsiao999/mediation@master
## from URL https://api.github.com/repos/jhsiao999/mediation/zipball/master</code></pre>
<pre><code>## Installing medinome</code></pre>
<pre><code>## &#39;/Library/Frameworks/R.framework/Resources/bin/R&#39; --no-site-file  \
##   --no-environ --no-save --no-restore --quiet CMD INSTALL  \
##   &#39;/private/var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T/RtmpB8snJH/devtools154ea6661a0f0/jhsiao999-mediation-79aa5ab/pkg&#39;  \
##   --library=&#39;/Library/Frameworks/R.framework/Versions/3.4/Resources/library&#39;  \
##   --install-tests</code></pre>
<pre><code>## </code></pre>
<pre class="r"><code>library(medinome)</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre class="r"><code>library(limma)
library(ashr)
getwd()</code></pre>
<pre><code>## [1] &quot;/Users/laurenblake/Desktop/Regulatory_Evol/ashlar-trial/analysis&quot;</code></pre>
<pre class="r"><code># Load data
exp_methyl &lt;- read.csv(&quot;../../../Reg_Evo_Primates/data/human_chimp_orth_exp_methyl_7725_hum.txt&quot;, sep=&quot;&quot;)
sample_info_RNAseq &lt;- read.csv(&quot;../../../Reg_Evo_Primates/data/Sample_info_RNAseq.csv&quot;)
samples &lt;- sample_info_RNAseq[-17,]

species &lt;- as.data.frame(samples[,3])
tissue &lt;- as.data.frame(samples[,4])
RIN &lt;- as.data.frame(samples[,5])
cpm.voom.cyclic &lt;- readRDS(&quot;../../../Reg_Evo_Primates/data/human_chimp_orth_cpm_voom_cyclic.rds&quot;)

chimp_human_heart &lt;- c(17,20,21,24,25,28,29)</code></pre>
</div>
<div id="find-de-genes-fdr-1" class="section level1">
<h1>Find DE genes (FDR = 1%)</h1>
<pre class="r"><code># Set FSR and FDR_level

FSR_level &lt;- 0.01
FDR_level &lt;- 0.01

# Find DE genes

tissue &lt;- as.data.frame(samples[,4])
  tissue &lt;- tissue[chimp_human_heart,]
  tissue_no_extra &lt;- droplevels.factor(tissue)


  design &lt;- model.matrix(~ as.factor(tissue_no_extra) + RIN[chimp_human_heart,])
  fit_all &lt;- lmFit(cpm.voom.cyclic[,chimp_human_heart], design)
  fit_all &lt;- eBayes(fit_all)
  
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust=&quot;BH&quot;, number=Inf,
sort.by=&quot;none&quot;)
  HvC_Heart_fit_all_5perc &lt;-HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val &lt; FDR_level), ]


  # For the DE genes, find which ones we have methylation values for                                                                                                                                                              
#  rownames(exp_methyl) &lt;- row.names(cpm.voom.cyclic)                             

  human_chimp_heart &lt;-  rownames(exp_methyl) %in%
HvC_Heart_fit_all_5perc$genes
  human_chimp_heart &lt;- as.data.frame(human_chimp_heart)
  counts_genes_in &lt;- cbind(exp_methyl, human_chimp_heart)
  counts_genes_in_cutoff &lt;- subset(counts_genes_in, human_chimp_heart ==
&quot;TRUE&quot;)
  counts_genes_in_cutoff &lt;- counts_genes_in_cutoff[,1:79]
  expression_values_only &lt;- counts_genes_in_cutoff[,2:32]
  methylation_values_only &lt;- counts_genes_in_cutoff[,49:79]
  dim(expression_values_only)</code></pre>
<pre><code>## [1] 623  31</code></pre>
</div>
<div id="run-the-mediation-analysis-on-unscaled-data" class="section level1">
<h1>Run the mediation analysis on unscaled data</h1>
<pre class="r"><code># Run on non-scaled data
  
     # Prepare for mediation analysis                                                               
Y &lt;- expression_values_only[,chimp_human_heart]
X &lt;- as.integer(tissue_no_extra)
M &lt;- methylation_values_only[,chimp_human_heart]

dim(Y)</code></pre>
<pre><code>## [1] 623   7</code></pre>
<pre class="r"><code>dim(X)</code></pre>
<pre><code>## NULL</code></pre>
<pre class="r"><code>dim(M)</code></pre>
<pre><code>## [1] 623   7</code></pre>
<pre class="r"><code>nsam &lt;- ncol(Y)

fit.voom &lt;- mediate.test.voom(Y, X, M)

fit.ash.tau.fs &lt;- ash(fit.voom$d, fit.voom$se.fs, df=nsam-2, mixcompdist = &quot;halfuniform&quot;, method = &quot;fdr&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>svalue_table &lt;- fit.ash.tau.fs$result$svalue &lt; FSR_level
svalue_true &lt;- length(which(svalue_table == TRUE))
svalue_total &lt;- length(svalue_table)

svalue_true/svalue_total</code></pre>
<pre><code>## [1] 1</code></pre>
</div>
<div id="run-the-mediation-analysis-on-scaled-data" class="section level1">
<h1>Run the mediation analysis on scaled data</h1>
<pre class="r"><code># Transpose by gene

Y &lt;- t(scale(t(expression_values_only[,chimp_human_heart])))
M &lt;- t(scale(t(methylation_values_only[,chimp_human_heart])))

fit.voom &lt;- mediate.test.voom(Y, X, M)

fit.ash.tau.fs &lt;- ash(fit.voom$d, fit.voom$se.fs, df=nsam-2, mixcompdist = &quot;halfuniform&quot;, method = &quot;fdr&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>svalue_table &lt;- fit.ash.tau.fs$result$svalue &lt; FSR_level
svalue_true &lt;- length(which(svalue_table == TRUE))
svalue_total &lt;- length(svalue_table)

svalue_true/svalue_total</code></pre>
<pre><code>## [1] 0.5971108</code></pre>
</div>
<div id="permute-the-methylation-values-scale-the-methylation-values-and-then" class="section level1">
<h1>Permute the methylation values, scale the methylation values, and then</h1>
<pre class="r"><code># Permute the methylation values and then run

run1 &lt;- array(0, dim = c(nrow(expression_values_only), length(chimp_human_heart)))

for (i in 1:nrow(expression_values_only)){
  df &lt;- as.data.frame(methylation_values_only[i,chimp_human_heart])
  new_df &lt;- as.data.frame(sample(df))
  run1[i,] &lt;- t(new_df)
}


M_perm &lt;- t(scale(t(run1)))

fit.voom &lt;- mediate.test.voom(Y, X, M_perm)

fit.ash.tau.fs &lt;- ash(fit.voom$d, fit.voom$se.fs, df=nsam-2, mixcompdist = &quot;halfuniform&quot;, method = &quot;fdr&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>svalue_table &lt;- fit.ash.tau.fs$result$svalue &lt; FSR_level
svalue_true &lt;- length(which(svalue_table == TRUE))
svalue_total &lt;- length(svalue_table)

svalue_true/svalue_total</code></pre>
<pre><code>## [1] 0.7142857</code></pre>
<div id="permute-the-scaled-values-and-then-run-the-mediation-analysis" class="section level2">
<h2>Permute the scaled values and then run the mediation analysis</h2>
<pre class="r"><code># Another way to permute
M &lt;- t(scale(t(methylation_values_only[,chimp_human_heart])))
run1 &lt;- array(0, dim = c(nrow(expression_values_only), length(chimp_human_heart)))

for (i in 1:nrow(expression_values_only)){
  new_df &lt;- sample(M[i,])
  run1[i,] &lt;- t(new_df)
}

fit.voom &lt;- mediate.test.voom(Y, X, run1)

fit.ash.tau.fs &lt;- ash(fit.voom$d, fit.voom$se.fs, df=nsam-2, mixcompdist = &quot;halfuniform&quot;, method = &quot;fdr&quot;)</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre class="r"><code>svalue_table &lt;- fit.ash.tau.fs$result$svalue &lt; FSR_level
svalue_true &lt;- length(which(svalue_table == TRUE))
svalue_total &lt;- length(svalue_table)

svalue_true/svalue_total</code></pre>
<pre><code>## [1] 0.7560193</code></pre>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
