<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2019-06-24" />

<title>Revisions_H3K27ac</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Revisions_H3K27ac</h1>
<h4 class="author">Lauren Blake</h4>
<h4 class="date">June 24, 2019</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#open-files">Open files</a><ul>
<li><a href="#kidney">Kidney</a></li>
<li><a href="#liver">Liver</a></li>
<li><a href="#lung">Lung</a></li>
</ul></li>
<li><a href="#separate-into-hypermethylated-and-overlapping-h3k27ac">Separate into hypermethylated and overlapping H3K27ac</a></li>
<li><a href="#heart">Heart</a><ul>
<li><a href="#kidney-1">Kidney</a></li>
<li><a href="#liver-1">Liver</a></li>
<li><a href="#lung-1">Lung</a></li>
</ul></li>
<li><a href="#run-clusterprofiler">Run clusterProfiler</a></li>
</ul>
</div>

<div id="open-files" class="section level1">
<h1>Open files</h1>
<p>Note: due to pairwise ordering from another file (tissue_specific_tDMRs), tDMRs labeled as “hypermethylated” are actually hypomethylated in the lung and kidney.</p>
<pre class="r"><code># Library
library(bedr)</code></pre>
<pre><code>## Warning: package &#39;bedr&#39; was built under R version 3.4.4</code></pre>
<pre><code>## 
## 
## ######################
## #### bedr v1.0.7 ####
## ######################
## 
## checking binary availability...
##   * Checking path for bedtools... PASS
##     /usr/local/bin/bedtools
##   * Checking path for bedops... FAIL
##   * Checking path for tabix... FAIL
## tests and examples will be skipped on R CMD check if binaries are missing</code></pre>
<pre class="r"><code>library(&quot;topGO&quot;)</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, cbind, colMeans,
##     colnames, colSums, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>## Loading required package: graph</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Loading required package: GO.db</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## </code></pre>
<pre><code>## Loading required package: SparseM</code></pre>
<pre><code>## 
## Attaching package: &#39;SparseM&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     backsolve</code></pre>
<pre><code>## 
## groupGOTerms:    GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
<pre><code>## 
## Attaching package: &#39;topGO&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     members</code></pre>
<pre class="r"><code>#library(&quot;biomaRt&quot;)
library(&quot;clusterProfiler&quot;)</code></pre>
<pre><code>## Loading required package: DOSE</code></pre>
<pre><code>## DOSE v3.4.0  For help: https://guangchuangyu.github.io/DOSE
## 
## If you use DOSE in published research, please cite:
## Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
<pre><code>## clusterProfiler v3.6.0  For help: https://guangchuangyu.github.io/clusterProfiler
## 
## If you use clusterProfiler in published research, please cite:
## Guangchuang Yu., Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.</code></pre>
<pre class="r"><code>library(&quot;org.Hs.eg.db&quot;)</code></pre>
<pre><code>## </code></pre>
<pre class="r"><code>human_chimp_heart_specific_25_conserved_H3K27_HEART &lt;- read.delim(&quot;~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_heart_specific_25_conserved_H3K27_HEART.bed&quot;, header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_HEART$V4)</code></pre>
<pre><code>##    Length     Class      Mode 
##       455 character character</code></pre>
<pre class="r"><code>413/(413+42)</code></pre>
<pre><code>## [1] 0.9076923</code></pre>
<pre class="r"><code>human_chimp_heart_specific_25_conserved_H3K27_KIDNEY &lt;- read.delim(&quot;~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_kidney_specific_25_conserved_H3K27_KIDNEY.bed&quot;, header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_KIDNEY$V4)</code></pre>
<pre><code>##    Length     Class      Mode 
##       897 character character</code></pre>
<pre class="r"><code>858/(858+39)</code></pre>
<pre><code>## [1] 0.9565217</code></pre>
<pre class="r"><code>human_chimp_heart_specific_25_conserved_H3K27_LIVER &lt;- read.delim(&quot;~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_liver_specific_25_conserved_H3K27_LIVER.bed&quot;, header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_LIVER$V4)</code></pre>
<pre><code>##    Length     Class      Mode 
##      1333 character character</code></pre>
<pre class="r"><code>1081/(1081+252)</code></pre>
<pre><code>## [1] 0.8109527</code></pre>
<pre class="r"><code>human_chimp_heart_specific_25_conserved_H3K27_LUNG &lt;- read.delim(&quot;~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_lung_specific_25_conserved_H3K27_LUNG.bed&quot;, header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_LUNG$V4)</code></pre>
<pre><code>##    Length     Class      Mode 
##        51 character character</code></pre>
<pre class="r"><code>30/(30+21)</code></pre>
<pre><code>## [1] 0.5882353</code></pre>
<pre class="r"><code>refGene_hg19_TSS &lt;- read.delim(&quot;~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/refGene_hg19_TSS.R&quot;, header=FALSE, stringsAsFactors = FALSE)

# Adjust so that bedtools will accept it
refGene_hg19_TSS[,3] &lt;- refGene_hg19_TSS[,3] + 1 

# Now sort

human_chimp_heart_cons &lt;- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_HEART[,1:3])</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Overlapping regions can cause unexpected results.</code></pre>
<pre class="r"><code>refGene_hg19_TSS &lt;- bedr.sort.region(refGene_hg19_TSS)</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS</code></pre>
<pre class="r"><code>closest_heart &lt;- bedr(input = list(a = human_chimp_heart_specific_25_conserved_H3K27_HEART[,1:3], b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... PASS
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc557225e1f3.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc5530e100bf.bed</code></pre>
<pre class="r"><code># Sort based on upsteam/downstream

closest_heart_upstream &lt;- closest_heart[which(closest_heart$V9 == &quot;+&quot;),]
closest_heart_downstream &lt;- closest_heart[which(closest_heart$V9 == &quot;-&quot;),]
distance_heart_upstream &lt;- as.numeric(closest_heart_upstream$V3) - as.numeric(closest_heart_upstream$V5)
distance_heart_downstream &lt;- as.numeric(closest_heart_downstream$V5) - as.numeric(closest_heart_downstream$V3)

length(which(distance_heart_upstream &gt; 0))/length(distance_heart_upstream)</code></pre>
<pre><code>## [1] 0.6037152</code></pre>
<pre class="r"><code>length(which(distance_heart_downstream &gt; 0))/length(distance_heart_downstream)</code></pre>
<pre><code>## [1] 0.6140351</code></pre>
<pre class="r"><code>summary(distance_heart_downstream)</code></pre>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
## -751049  -12592    3472   15480   50862  386993</code></pre>
<div id="kidney" class="section level2">
<h2>Kidney</h2>
<pre class="r"><code># Now sort

human_chimp_heart_cons &lt;- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_KIDNEY[,1:3])</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS</code></pre>
<pre class="r"><code>refGene_hg19_TSS &lt;- bedr.sort.region(refGene_hg19_TSS)</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS</code></pre>
<pre class="r"><code>closest_heart &lt;- bedr(input = list(a = human_chimp_heart_cons, b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc553d991179.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc551a874cb1.bed</code></pre>
<pre class="r"><code># Sort based on upsteam/downstream

closest_heart_upstream &lt;- closest_heart[which(closest_heart$V9 == &quot;+&quot;),]
closest_heart_downstream &lt;- closest_heart[which(closest_heart$V9 == &quot;-&quot;),]
distance_heart_upstream &lt;- as.numeric(closest_heart_upstream$V3) - as.numeric(closest_heart_upstream$V5)
distance_heart_downstream &lt;- as.numeric(closest_heart_downstream$V5) - as.numeric(closest_heart_downstream$V3)

length(which(distance_heart_upstream &gt; 0))/length(distance_heart_upstream)</code></pre>
<pre><code>## [1] 0.5685358</code></pre>
<pre class="r"><code>length(which(distance_heart_downstream &gt; 0))/length(distance_heart_downstream)</code></pre>
<pre><code>## [1] 0.5628227</code></pre>
</div>
<div id="liver" class="section level2">
<h2>Liver</h2>
<pre class="r"><code># Now sort

human_chimp_heart_cons &lt;- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_LIVER[,1:3])</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS</code></pre>
<pre class="r"><code>refGene_hg19_TSS &lt;- bedr.sort.region(refGene_hg19_TSS)</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS</code></pre>
<pre class="r"><code>closest_heart &lt;- bedr(input = list(a = human_chimp_heart_cons, b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc551ef7a8be.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc551c85df81.bed</code></pre>
<pre class="r"><code># Sort based on upsteam/downstream

closest_liver_upstream &lt;- closest_heart[which(closest_heart$V9 == &quot;+&quot;),]
closest_liver_downstream &lt;- closest_heart[which(closest_heart$V9 == &quot;-&quot;),]
distance_liver_upstream &lt;- as.numeric(closest_liver_upstream$V3) - as.numeric(closest_liver_upstream$V5)
distance_liver_downstream &lt;- as.numeric(closest_liver_downstream$V5) - as.numeric(closest_liver_downstream$V3)

length(which(distance_liver_upstream &gt; 0))/length(distance_liver_upstream)</code></pre>
<pre><code>## [1] 0.6512043</code></pre>
<pre class="r"><code>length(which(distance_liver_downstream &gt; 0))/length(distance_liver_downstream)</code></pre>
<pre><code>## [1] 0.6230032</code></pre>
<pre class="r"><code>all_distance &lt;- c(distance_heart_upstream, distance_heart_downstream, distance_liver_upstream, distance_liver_downstream)
summary(all_distance)</code></pre>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
## -805181   -7548    1690   13079   31168  999526</code></pre>
</div>
<div id="lung" class="section level2">
<h2>Lung</h2>
<pre class="r"><code># Now sort

human_chimp_heart_cons &lt;- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_LUNG[,1:3])</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Overlapping regions can cause unexpected results.</code></pre>
<pre class="r"><code>refGene_hg19_TSS &lt;- bedr.sort.region(refGene_hg19_TSS)</code></pre>
<pre><code>## SORTING
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS</code></pre>
<pre class="r"><code>closest_heart &lt;- bedr(input = list(a = human_chimp_heart_cons, b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... PASS
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc55282f46b3.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc551be5509c.bed</code></pre>
<pre class="r"><code># Sort based on upsteam/downstream

closest_heart_upstream &lt;- closest_heart[which(closest_heart$V9 == &quot;+&quot;),]
closest_heart_downstream &lt;- closest_heart[which(closest_heart$V9 == &quot;-&quot;),]
distance_heart_upstream &lt;- as.numeric(closest_heart_upstream$V3) - as.numeric(closest_heart_upstream$V5)
distance_heart_downstream &lt;- as.numeric(closest_heart_downstream$V5) - as.numeric(closest_heart_downstream$V3)

length(which(distance_heart_upstream &gt; 0))/length(distance_heart_upstream)</code></pre>
<pre><code>## [1] 0.673913</code></pre>
<pre class="r"><code>length(which(distance_heart_downstream &gt; 0))/length(distance_heart_downstream)</code></pre>
<pre><code>## [1] 0.5714286</code></pre>
</div>
</div>
<div id="separate-into-hypermethylated-and-overlapping-h3k27ac" class="section level1">
<h1>Separate into hypermethylated and overlapping H3K27ac</h1>
</div>
<div id="heart" class="section level1">
<h1>Heart</h1>
<pre class="r"><code># Find the heart hypo
heart_hypo &lt;- human_chimp_heart_specific_25_conserved_H3K27_HEART[which(human_chimp_heart_specific_25_conserved_H3K27_LIVER$V4 == &quot;hyper&quot; &amp; human_chimp_heart_specific_25_conserved_H3K27_LIVER$V6 != &quot;-1&quot;),]

heart_hypo &lt;- heart_hypo[complete.cases(heart_hypo), ]

# Find the closest gene

closest_heart &lt;- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... PASS
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc55367de709.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc55788e817d.bed</code></pre>
<pre class="r"><code># Convert the gene name to ensg

gene_id &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv&quot;, stringsAsFactors = FALSE, header=TRUE, sep = &quot;,&quot;)

comb_kidney &lt;- merge(closest_heart, gene_id, by.x = c(&quot;V8&quot;), by.y = c(&quot;Gene&quot;))

comb_kidney$ensg</code></pre>
<pre><code>##  [1] &quot;ENSG00000196839&quot; &quot;ENSG00000163050&quot; &quot;ENSG00000178878&quot;
##  [4] &quot;ENSG00000130762&quot; &quot;ENSG00000168646&quot; &quot;ENSG00000110092&quot;
##  [7] &quot;ENSG00000106554&quot; &quot;ENSG00000134871&quot; &quot;ENSG00000244274&quot;
## [10] &quot;ENSG00000244274&quot; &quot;ENSG00000135636&quot; &quot;ENSG00000135636&quot;
## [13] &quot;ENSG00000135636&quot; &quot;ENSG00000135636&quot; &quot;ENSG00000135636&quot;
## [16] &quot;ENSG00000135636&quot; &quot;ENSG00000135636&quot; &quot;ENSG00000108001&quot;
## [19] &quot;ENSG00000187773&quot; &quot;ENSG00000139132&quot; &quot;ENSG00000153303&quot;
## [22] &quot;ENSG00000143641&quot; &quot;ENSG00000144596&quot; &quot;ENSG00000115756&quot;
## [25] &quot;ENSG00000197081&quot; &quot;ENSG00000185507&quot; &quot;ENSG00000123104&quot;
## [28] &quot;ENSG00000127528&quot; &quot;ENSG00000106003&quot; &quot;ENSG00000107902&quot;
## [31] &quot;ENSG00000107902&quot; &quot;ENSG00000183060&quot; &quot;ENSG00000068305&quot;
## [34] &quot;ENSG00000068305&quot; &quot;ENSG00000068305&quot; &quot;ENSG00000068305&quot;
## [37] &quot;ENSG00000120254&quot; &quot;ENSG00000120254&quot; &quot;ENSG00000120254&quot;
## [40] &quot;ENSG00000133454&quot; &quot;ENSG00000124785&quot; &quot;ENSG00000162407&quot;
## [43] &quot;ENSG00000107263&quot; &quot;ENSG00000159788&quot; &quot;ENSG00000138835&quot;
## [46] &quot;ENSG00000064932&quot; &quot;ENSG00000104969&quot; &quot;ENSG00000130147&quot;
## [49] &quot;ENSG00000156222&quot; &quot;ENSG00000156222&quot; &quot;ENSG00000156222&quot;
## [52] &quot;ENSG00000156222&quot; &quot;ENSG00000095637&quot; &quot;ENSG00000095637&quot;
## [55] &quot;ENSG00000185594&quot; &quot;ENSG00000065882&quot; &quot;ENSG00000109927&quot;
## [58] &quot;ENSG00000100726&quot; &quot;ENSG00000185361&quot; &quot;ENSG00000198795&quot;
## [61] &quot;ENSG00000198597&quot; &quot;ENSG00000141579&quot; &quot;ENSG00000070476&quot;
## [64] &quot;ENSG00000070476&quot;</code></pre>
<pre class="r"><code># Convert the TSS to ensg as well 

ref_gene_hg19_ensg &lt;- merge(refGene_hg19_TSS, gene_id, by.x = c(&quot;V5&quot;), by.y = c(&quot;Gene&quot;))

all_ref_gene_hg19_ensg &lt;- unique(ref_gene_hg19_ensg$ensg)

heart_ref_gene &lt;- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene &lt;- as.numeric(as.vector(heart_ref_gene))
names(test_gene) &lt;-  all_ref_gene_hg19_ensg

# Run topGO
go_data &lt;- new(&quot;topGOdata&quot;,
                   ontology = &quot;BP&quot;,
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore &gt; 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = &quot;org.Hs.eg.db&quot;,
                   ID = &quot;ensembl&quot;)</code></pre>
<pre><code>## 
## Building most specific GOs .....</code></pre>
<pre><code>##  ( 11450 GO terms found. )</code></pre>
<pre><code>## 
## Build GO DAG topology ..........</code></pre>
<pre><code>##  ( 15456 GO terms and 36153 relations. )</code></pre>
<pre><code>## 
## Annotating nodes ...............</code></pre>
<pre><code>##  ( 14505 genes annotated to the GO terms. )</code></pre>
<pre class="r"><code># Perform enrichment test
go_test &lt;- runTest(go_data, algorithm = &quot;weight01&quot;, statistic = &quot;fisher&quot;)</code></pre>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 1568 nontrivial nodes
##       parameters: 
##           test statistic: fisher</code></pre>
<pre><code>## 
##   Level 15:  1 nodes to be scored    (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 14:  11 nodes to be scored   (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 13:  29 nodes to be scored   (103 eliminated genes)</code></pre>
<pre><code>## 
##   Level 12:  50 nodes to be scored   (298 eliminated genes)</code></pre>
<pre><code>## 
##   Level 11:  72 nodes to be scored   (2501 eliminated genes)</code></pre>
<pre><code>## 
##   Level 10:  107 nodes to be scored  (3937 eliminated genes)</code></pre>
<pre><code>## 
##   Level 9:   161 nodes to be scored  (5541 eliminated genes)</code></pre>
<pre><code>## 
##   Level 8:   201 nodes to be scored  (7251 eliminated genes)</code></pre>
<pre><code>## 
##   Level 7:   245 nodes to be scored  (8558 eliminated genes)</code></pre>
<pre><code>## 
##   Level 6:   265 nodes to be scored  (10615 eliminated genes)</code></pre>
<pre><code>## 
##   Level 5:   217 nodes to be scored  (12035 eliminated genes)</code></pre>
<pre><code>## 
##   Level 4:   127 nodes to be scored  (13480 eliminated genes)</code></pre>
<pre><code>## 
##   Level 3:   63 nodes to be scored   (13985 eliminated genes)</code></pre>
<pre><code>## 
##   Level 2:   18 nodes to be scored   (14177 eliminated genes)</code></pre>
<pre><code>## 
##   Level 1:   1 nodes to be scored    (14301 eliminated genes)</code></pre>
<pre class="r"><code>go_table &lt;- GenTable(go_data, weightFisher = go_test,
                         orderBy = &quot;weightFisher&quot;, ranksOf = &quot;weightFisher&quot;,
                         topNodes = sum(score(go_test) &lt; .05))

go_table</code></pre>
<pre><code>##          GO.ID                                        Term Annotated
## 1   GO:0009113      purine nucleobase biosynthetic process        12
## 2   GO:0033197                       response to vitamin E        12
## 3   GO:0006351                transcription, DNA-templated      3277
## 4   GO:0032007        negative regulation of TOR signaling        40
## 5   GO:0043547      positive regulation of GTPase activity       354
## 6   GO:0060070             canonical Wnt signaling pathway       274
## 7   GO:1903078 positive regulation of protein localizat...        46
## 8   GO:0071320                   cellular response to cAMP        47
## 9   GO:0006501               C-terminal protein lipidation        62
## 10  GO:0007386           compartment pattern specification         5
## 11  GO:0070375                                ERK5 cascade         5
## 12  GO:0042321 negative regulation of circadian sleep/w...         5
## 13  GO:0009146 purine nucleoside triphosphate catabolic...         5
## 14  GO:0060509           Type I pneumocyte differentiation         5
## 15  GO:0046060                      dATP metabolic process         5
## 16  GO:0009256 10-formyltetrahydrofolate metabolic proc...         5
## 17  GO:0048541                   Peyer&#39;s patch development         5
## 18  GO:0009168 purine ribonucleoside monophosphate bios...        66
## 19  GO:0006863                 purine nucleobase transport         6
## 20  GO:0015942                   formate metabolic process         6
## 21  GO:0018242 protein O-linked glycosylation via serin...         6
## 22  GO:0001957                intramembranous ossification         6
## 23  GO:0060368 regulation of Fc receptor mediated stimu...         6
## 24  GO:0070141                            response to UV-A         6
## 25  GO:0038203                             TORC2 signaling         6
## 26  GO:0033632 regulation of cell-cell adhesion mediate...         6
## 27  GO:0060405               regulation of penile erection         6
## 28  GO:0009204 deoxyribonucleoside triphosphate catabol...         6
## 29  GO:0070244 negative regulation of thymocyte apoptot...         6
## 30  GO:0090630               activation of GTPase activity        76
## 31  GO:0002819      regulation of adaptive immune response       127
## 32  GO:0032261                   purine nucleotide salvage         7
## 33  GO:0018243 protein O-linked glycosylation via threo...         7
## 34  GO:0001778                      plasma membrane repair         7
## 35  GO:1903071 positive regulation of ER-associated ubi...         7
## 36  GO:0046654       tetrahydrofolate biosynthetic process         7
## 37  GO:0009120       deoxyribonucleoside metabolic process         7
## 38  GO:0019043              establishment of viral latency         7
## 39  GO:0048702        embryonic neurocranium morphogenesis         7
## 40  GO:1901642          nucleoside transmembrane transport         7
## 41  GO:2000109 regulation of macrophage apoptotic proce...         7
## 42  GO:0046130     purine ribonucleoside catabolic process         7
## 43  GO:0033601 positive regulation of mammary gland epi...         7
## 44  GO:0071499 cellular response to laminar fluid shear...         7
## 45  GO:0008277 regulation of G-protein coupled receptor...       120
## 46  GO:1904263      positive regulation of TORC1 signaling         8
## 47  GO:0038180       nerve growth factor signaling pathway         8
## 48  GO:0038063 collagen-activated tyrosine kinase recep...         8
## 49  GO:0060340 positive regulation of type I interferon...         8
## 50  GO:0072530 purine-containing compound transmembrane...         8
## 51  GO:0048703      embryonic viscerocranium morphogenesis         8
## 52  GO:0034127 regulation of MyD88-independent toll-lik...         8
## 53  GO:0008354                         germ cell migration         8
## 54  GO:0014807                 regulation of somitogenesis         8
## 55  GO:0045188 regulation of circadian sleep/wake cycle...         8
## 56  GO:0048387 negative regulation of retinoic acid rec...         8
## 57  GO:0002313 mature B cell differentiation involved i...        18
## 58  GO:0033089 positive regulation of T cell differenti...         9
## 59  GO:0002315        marginal zone B cell differentiation         9
## 60  GO:0035999            tetrahydrofolate interconversion         9
## 61  GO:0001821                         histamine secretion         9
## 62  GO:0003139         secondary heart field specification         9
## 63  GO:0039530                     MDA-5 signaling pathway         9
## 64  GO:0002634     regulation of germinal center formation         9
## 65  GO:0070255               regulation of mucus secretion         9
## 66  GO:0001889                           liver development       124
## 67  GO:0030282                         bone mineralization        93
## 68  GO:0002903 negative regulation of B cell apoptotic ...        10
## 69  GO:0070601       centromeric sister chromatid cohesion        10
## 70  GO:0046085                 adenosine metabolic process        10
## 71  GO:0042118                 endothelial cell activation        10
## 72  GO:0035414 negative regulation of catenin import in...        10
## 73  GO:0033033 negative regulation of myeloid cell apop...        11
## 74  GO:0001973        adenosine receptor signaling pathway        11
## 75  GO:0003413 chondrocyte differentiation involved in ...        11
## 76  GO:0033327                 Leydig cell differentiation        11
## 77  GO:0043249                      erythrocyte maturation        11
## 78  GO:1904293         negative regulation of ERAD pathway        12
## 79  GO:0045351      type I interferon biosynthetic process        12
## 80  GO:0055015 ventricular cardiac muscle cell developm...        12
## 81  GO:2000059 negative regulation of protein ubiquitin...        12
## 82  GO:0050862 positive regulation of T cell receptor s...        12
## 83  GO:0048311                  mitochondrion distribution        12
## 84  GO:0002281 macrophage activation involved in immune...        12
## 85  GO:0061430                bone trabecula morphogenesis        13
## 86  GO:0043650      dicarboxylic acid biosynthetic process        13
## 87  GO:0071361                cellular response to ethanol        13
## 88  GO:0032486             Rap protein signal transduction        13
## 89  GO:2000107 negative regulation of leukocyte apoptot...        39
## 90  GO:0036066               protein O-linked fucosylation        14
## 91  GO:0045725 positive regulation of glycogen biosynth...        15
## 92  GO:0060749          mammary gland alveolus development        15
## 93  GO:0001829        trophectodermal cell differentiation        15
## 94  GO:0006744             ubiquinone biosynthetic process        15
## 95  GO:0035023 regulation of Rho protein signal transdu...       118
## 96  GO:0051412                  response to corticosterone        16
## 97  GO:0051446 positive regulation of meiotic cell cycl...        16
## 98  GO:0072531 pyrimidine-containing compound transmemb...        16
## 99  GO:0055003                  cardiac myofibril assembly        16
## 100 GO:0009264       deoxyribonucleotide catabolic process        16
## 101 GO:0000122 negative regulation of transcription fro...       727
## 102 GO:0030111         regulation of Wnt signaling pathway       295
## 103 GO:0046386     deoxyribose phosphate catabolic process        17
## 104 GO:0045655      regulation of monocyte differentiation        17
## 105 GO:0071157    negative regulation of cell cycle arrest        18
## 106 GO:0032026                   response to magnesium ion        18
## 107 GO:0046655                folic acid metabolic process        18
## 108 GO:0007063     regulation of sister chromatid cohesion        19
## 109 GO:2000178 negative regulation of neural precursor ...        19
## 110 GO:0006139 nucleobase-containing compound metabolic...      5072
##     Significant Expected weightFisher
## 1             2     0.03      0.00046
## 2             2     0.03      0.00046
## 3            14     8.81      0.00408
## 4             2     0.11      0.00515
## 5             6     0.95      0.00540
## 6             4     0.74      0.00656
## 7             2     0.12      0.00677
## 8             2     0.13      0.00706
## 9             2     0.17      0.01203
## 10            1     0.01      0.01337
## 11            1     0.01      0.01337
## 12            1     0.01      0.01337
## 13            1     0.01      0.01337
## 14            1     0.01      0.01337
## 15            1     0.01      0.01337
## 16            1     0.01      0.01337
## 17            1     0.01      0.01337
## 18            2     0.18      0.01356
## 19            1     0.02      0.01603
## 20            1     0.02      0.01603
## 21            1     0.02      0.01603
## 22            1     0.02      0.01603
## 23            1     0.02      0.01603
## 24            1     0.02      0.01603
## 25            1     0.02      0.01603
## 26            1     0.02      0.01603
## 27            1     0.02      0.01603
## 28            1     0.02      0.01603
## 29            1     0.02      0.01603
## 30            2     0.20      0.01771
## 31            2     0.34      0.01835
## 32            1     0.02      0.01867
## 33            1     0.02      0.01867
## 34            1     0.02      0.01867
## 35            1     0.02      0.01867
## 36            1     0.02      0.01867
## 37            1     0.02      0.01867
## 38            1     0.02      0.01867
## 39            1     0.02      0.01867
## 40            1     0.02      0.01867
## 41            1     0.02      0.01867
## 42            1     0.02      0.01867
## 43            1     0.02      0.01867
## 44            1     0.02      0.01867
## 45            3     0.32      0.02091
## 46            1     0.02      0.02131
## 47            1     0.02      0.02131
## 48            1     0.02      0.02131
## 49            1     0.02      0.02131
## 50            1     0.02      0.02131
## 51            1     0.02      0.02131
## 52            1     0.02      0.02131
## 53            1     0.02      0.02131
## 54            1     0.02      0.02131
## 55            1     0.02      0.02131
## 56            1     0.02      0.02131
## 57            2     0.05      0.02335
## 58            1     0.02      0.02395
## 59            1     0.02      0.02395
## 60            1     0.02      0.02395
## 61            1     0.02      0.02395
## 62            1     0.02      0.02395
## 63            1     0.02      0.02395
## 64            1     0.02      0.02395
## 65            1     0.02      0.02395
## 66            3     0.33      0.02423
## 67            2     0.25      0.02584
## 68            1     0.03      0.02657
## 69            1     0.03      0.02657
## 70            1     0.03      0.02657
## 71            1     0.03      0.02657
## 72            1     0.03      0.02657
## 73            1     0.03      0.02919
## 74            1     0.03      0.02919
## 75            1     0.03      0.02919
## 76            1     0.03      0.02919
## 77            1     0.03      0.02919
## 78            1     0.03      0.03180
## 79            1     0.03      0.03180
## 80            1     0.03      0.03180
## 81            1     0.03      0.03180
## 82            1     0.03      0.03180
## 83            1     0.03      0.03180
## 84            1     0.03      0.03180
## 85            1     0.03      0.03441
## 86            1     0.03      0.03441
## 87            1     0.03      0.03441
## 88            1     0.03      0.03441
## 89            2     0.10      0.03614
## 90            1     0.04      0.03701
## 91            1     0.04      0.03960
## 92            1     0.04      0.03960
## 93            1     0.04      0.03960
## 94            1     0.04      0.03960
## 95            2     0.32      0.03999
## 96            1     0.04      0.04218
## 97            1     0.04      0.04218
## 98            1     0.04      0.04218
## 99            1     0.04      0.04218
## 100           1     0.04      0.04218
## 101           5     1.95      0.04391
## 102           4     0.79      0.04460
## 103           1     0.05      0.04476
## 104           1     0.05      0.04476
## 105           1     0.05      0.04733
## 106           1     0.05      0.04733
## 107           1     0.05      0.04733
## 108           1     0.05      0.04990
## 109           1     0.05      0.04990
## 110          19    13.64      0.04991</code></pre>
<pre class="r"><code>go_table_heart &lt;- as.data.frame(cbind(go_table$GO.ID, go_table$weightFisher))

write.table(go_table_heart, &quot;../data/go_table_heart.txt&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)

# Get names of kidney genes
sig.genes &lt;- sigGenes(go_data)
goresults &lt;- sapply(go_table$GO.ID, function(x)
    {
      genes&lt;-genesInTerm(go_data, x) 
      genes[[1]][genes[[1]] %in% sig.genes]
    })

# cardiac myofibril assembly, positive regulation of heart rate, regulation of monocyte differentiation
goresults[&quot;GO:0055003&quot;] </code></pre>
<pre><code>## $`GO:0055003`
## [1] &quot;ENSG00000068305&quot;</code></pre>
<pre class="r"><code>goresults[&quot;GO:0010460&quot;]</code></pre>
<pre><code>## $&lt;NA&gt;
## NULL</code></pre>
<pre class="r"><code>goresults[&quot;GO:0045655&quot;]</code></pre>
<pre><code>## $`GO:0045655`
## [1] &quot;ENSG00000185507&quot;</code></pre>
<div id="kidney-1" class="section level2">
<h2>Kidney</h2>
<pre class="r"><code># Find the heart hypo
heart_hypo &lt;- human_chimp_heart_specific_25_conserved_H3K27_KIDNEY[which(human_chimp_heart_specific_25_conserved_H3K27_KIDNEY$V4 == &quot;hyper&quot; &amp; human_chimp_heart_specific_25_conserved_H3K27_KIDNEY$V6 != &quot;-1&quot;),]

heart_hypo &lt;- heart_hypo[complete.cases(heart_hypo), ]
# Find the closest gene

closest_heart &lt;- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc55c1f3227.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc553f2d81a4.bed</code></pre>
<pre class="r"><code># Convert the gene name to ensg

gene_id &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv&quot;, stringsAsFactors = FALSE, header=TRUE, sep = &quot;,&quot;)

comb_kidney &lt;- merge(closest_heart, gene_id, by.x = c(&quot;V8&quot;), by.y = c(&quot;Gene&quot;))

comb_kidney$ensg</code></pre>
<pre><code>##   [1] &quot;ENSG00000081760&quot; &quot;ENSG00000198691&quot; &quot;ENSG00000100997&quot;
##   [4] &quot;ENSG00000100997&quot; &quot;ENSG00000117148&quot; &quot;ENSG00000169717&quot;
##   [7] &quot;ENSG00000169717&quot; &quot;ENSG00000139567&quot; &quot;ENSG00000145536&quot;
##  [10] &quot;ENSG00000134917&quot; &quot;ENSG00000105963&quot; &quot;ENSG00000162104&quot;
##  [13] &quot;ENSG00000164252&quot; &quot;ENSG00000063438&quot; &quot;ENSG00000063438&quot;
##  [16] &quot;ENSG00000110711&quot; &quot;ENSG00000196581&quot; &quot;ENSG00000196581&quot;
##  [19] &quot;ENSG00000179841&quot; &quot;ENSG00000053371&quot; &quot;ENSG00000151360&quot;
##  [22] &quot;ENSG00000178038&quot; &quot;ENSG00000178038&quot; &quot;ENSG00000106927&quot;
##  [25] &quot;ENSG00000159461&quot; &quot;ENSG00000089053&quot; &quot;ENSG00000138613&quot;
##  [28] &quot;ENSG00000138613&quot; &quot;ENSG00000198576&quot; &quot;ENSG00000147799&quot;
##  [31] &quot;ENSG00000116017&quot; &quot;ENSG00000029153&quot; &quot;ENSG00000029153&quot;
##  [34] &quot;ENSG00000029153&quot; &quot;ENSG00000029153&quot; &quot;ENSG00000029153&quot;
##  [37] &quot;ENSG00000117407&quot; &quot;ENSG00000117407&quot; &quot;ENSG00000117407&quot;
##  [40] &quot;ENSG00000126775&quot; &quot;ENSG00000126775&quot; &quot;ENSG00000068650&quot;
##  [43] &quot;ENSG00000068650&quot; &quot;ENSG00000068650&quot; &quot;ENSG00000068650&quot;
##  [46] &quot;ENSG00000241837&quot; &quot;ENSG00000241837&quot; &quot;ENSG00000158321&quot;
##  [49] &quot;ENSG00000158321&quot; &quot;ENSG00000183778&quot; &quot;ENSG00000183778&quot;
##  [52] &quot;ENSG00000183778&quot; &quot;ENSG00000183778&quot; &quot;ENSG00000183778&quot;
##  [55] &quot;ENSG00000128298&quot; &quot;ENSG00000125492&quot; &quot;ENSG00000125492&quot;
##  [58] &quot;ENSG00000125124&quot; &quot;ENSG00000050820&quot; &quot;ENSG00000127152&quot;
##  [61] &quot;ENSG00000127152&quot; &quot;ENSG00000110987&quot; &quot;ENSG00000110987&quot;
##  [64] &quot;ENSG00000116128&quot; &quot;ENSG00000162373&quot; &quot;ENSG00000165626&quot;
##  [67] &quot;ENSG00000165626&quot; &quot;ENSG00000122870&quot; &quot;ENSG00000122870&quot;
##  [70] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [73] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [76] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [79] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [82] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [85] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [88] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000125845&quot;
##  [91] &quot;ENSG00000101144&quot; &quot;ENSG00000104221&quot; &quot;ENSG00000165863&quot;
##  [94] &quot;ENSG00000109944&quot; &quot;ENSG00000109944&quot; &quot;ENSG00000173064&quot;
##  [97] &quot;ENSG00000235162&quot; &quot;ENSG00000089916&quot; &quot;ENSG00000089916&quot;
## [100] &quot;ENSG00000184601&quot; &quot;ENSG00000180336&quot; &quot;ENSG00000131943&quot;
## [103] &quot;ENSG00000175262&quot; &quot;ENSG00000128346&quot; &quot;ENSG00000185056&quot;
## [106] &quot;ENSG00000146521&quot; &quot;ENSG00000074410&quot; &quot;ENSG00000074410&quot;
## [109] &quot;ENSG00000158966&quot; &quot;ENSG00000198286&quot; &quot;ENSG00000130940&quot;
## [112] &quot;ENSG00000130940&quot; &quot;ENSG00000160200&quot; &quot;ENSG00000135736&quot;
## [115] &quot;ENSG00000168491&quot; &quot;ENSG00000168491&quot; &quot;ENSG00000109881&quot;
## [118] &quot;ENSG00000109881&quot; &quot;ENSG00000135127&quot; &quot;ENSG00000135127&quot;
## [121] &quot;ENSG00000135127&quot; &quot;ENSG00000110092&quot; &quot;ENSG00000138764&quot;
## [124] &quot;ENSG00000126353&quot; &quot;ENSG00000146731&quot; &quot;ENSG00000146731&quot;
## [127] &quot;ENSG00000112149&quot; &quot;ENSG00000112149&quot; &quot;ENSG00000198752&quot;
## [130] &quot;ENSG00000184661&quot; &quot;ENSG00000166589&quot; &quot;ENSG00000166589&quot;
## [133] &quot;ENSG00000166589&quot; &quot;ENSG00000166589&quot; &quot;ENSG00000179242&quot;
## [136] &quot;ENSG00000179242&quot; &quot;ENSG00000113100&quot; &quot;ENSG00000227767&quot;
## [139] &quot;ENSG00000153046&quot; &quot;ENSG00000101489&quot; &quot;ENSG00000101489&quot;
## [142] &quot;ENSG00000101489&quot; &quot;ENSG00000101489&quot; &quot;ENSG00000134873&quot;
## [145] &quot;ENSG00000134873&quot; &quot;ENSG00000159261&quot; &quot;ENSG00000169583&quot;
## [148] &quot;ENSG00000120885&quot; &quot;ENSG00000148842&quot; &quot;ENSG00000148842&quot;
## [151] &quot;ENSG00000148842&quot; &quot;ENSG00000149972&quot; &quot;ENSG00000149972&quot;
## [154] &quot;ENSG00000149972&quot; &quot;ENSG00000134871&quot; &quot;ENSG00000130635&quot;
## [157] &quot;ENSG00000165644&quot; &quot;ENSG00000198612&quot; &quot;ENSG00000198612&quot;
## [160] &quot;ENSG00000091704&quot; &quot;ENSG00000091704&quot; &quot;ENSG00000169372&quot;
## [163] &quot;ENSG00000146592&quot; &quot;ENSG00000146592&quot; &quot;ENSG00000160202&quot;
## [166] &quot;ENSG00000060069&quot; &quot;ENSG00000060069&quot; &quot;ENSG00000134030&quot;
## [169] &quot;ENSG00000134030&quot; &quot;ENSG00000153015&quot; &quot;ENSG00000153015&quot;
## [172] &quot;ENSG00000153015&quot; &quot;ENSG00000019186&quot; &quot;ENSG00000019186&quot;
## [175] &quot;ENSG00000132437&quot; &quot;ENSG00000079785&quot; &quot;ENSG00000141141&quot;
## [178] &quot;ENSG00000160049&quot; &quot;ENSG00000160049&quot; &quot;ENSG00000109016&quot;
## [181] &quot;ENSG00000100697&quot; &quot;ENSG00000151240&quot; &quot;ENSG00000150672&quot;
## [184] &quot;ENSG00000151208&quot; &quot;ENSG00000080845&quot; &quot;ENSG00000119689&quot;
## [187] &quot;ENSG00000119689&quot; &quot;ENSG00000119689&quot; &quot;ENSG00000119689&quot;
## [190] &quot;ENSG00000143006&quot; &quot;ENSG00000079805&quot; &quot;ENSG00000079805&quot;
## [193] &quot;ENSG00000079805&quot; &quot;ENSG00000079805&quot; &quot;ENSG00000079805&quot;
## [196] &quot;ENSG00000130226&quot; &quot;ENSG00000156162&quot; &quot;ENSG00000149636&quot;
## [199] &quot;ENSG00000149636&quot; &quot;ENSG00000149636&quot; &quot;ENSG00000047579&quot;
## [202] &quot;ENSG00000047579&quot; &quot;ENSG00000047579&quot; &quot;ENSG00000143507&quot;
## [205] &quot;ENSG00000143507&quot; &quot;ENSG00000229847&quot; &quot;ENSG00000229847&quot;
## [208] &quot;ENSG00000229847&quot; &quot;ENSG00000164778&quot; &quot;ENSG00000167280&quot;
## [211] &quot;ENSG00000104714&quot; &quot;ENSG00000119715&quot; &quot;ENSG00000119715&quot;
## [214] &quot;ENSG00000072840&quot; &quot;ENSG00000064655&quot; &quot;ENSG00000064655&quot;
## [217] &quot;ENSG00000088926&quot; &quot;ENSG00000168309&quot; &quot;ENSG00000168309&quot;
## [220] &quot;ENSG00000184731&quot; &quot;ENSG00000196814&quot; &quot;ENSG00000196814&quot;
## [223] &quot;ENSG00000154153&quot; &quot;ENSG00000189292&quot; &quot;ENSG00000182183&quot;
## [226] &quot;ENSG00000115363&quot; &quot;ENSG00000047662&quot; &quot;ENSG00000219438&quot;
## [229] &quot;ENSG00000168672&quot; &quot;ENSG00000083857&quot; &quot;ENSG00000083857&quot;
## [232] &quot;ENSG00000165140&quot; &quot;ENSG00000112787&quot; &quot;ENSG00000112787&quot;
## [235] &quot;ENSG00000085265&quot; &quot;ENSG00000126266&quot; &quot;ENSG00000066468&quot;
## [238] &quot;ENSG00000066468&quot; &quot;ENSG00000066468&quot; &quot;ENSG00000126500&quot;
## [241] &quot;ENSG00000075426&quot; &quot;ENSG00000164916&quot; &quot;ENSG00000114861&quot;
## [244] &quot;ENSG00000150893&quot; &quot;ENSG00000033170&quot; &quot;ENSG00000166206&quot;
## [247] &quot;ENSG00000116717&quot; &quot;ENSG00000116717&quot; &quot;ENSG00000116717&quot;
## [250] &quot;ENSG00000116717&quot; &quot;ENSG00000116717&quot; &quot;ENSG00000116717&quot;
## [253] &quot;ENSG00000143641&quot; &quot;ENSG00000182870&quot; &quot;ENSG00000130700&quot;
## [256] &quot;ENSG00000168505&quot; &quot;ENSG00000165702&quot; &quot;ENSG00000165702&quot;
## [259] &quot;ENSG00000167741&quot; &quot;ENSG00000167741&quot; &quot;ENSG00000074047&quot;
## [262] &quot;ENSG00000107249&quot; &quot;ENSG00000151948&quot; &quot;ENSG00000151948&quot;
## [265] &quot;ENSG00000137198&quot; &quot;ENSG00000156049&quot; &quot;ENSG00000197177&quot;
## [268] &quot;ENSG00000077585&quot; &quot;ENSG00000158292&quot; &quot;ENSG00000119714&quot;
## [271] &quot;ENSG00000178075&quot; &quot;ENSG00000125651&quot; &quot;ENSG00000077235&quot;
## [274] &quot;ENSG00000069812&quot; &quot;ENSG00000166135&quot; &quot;ENSG00000180448&quot;
## [277] &quot;ENSG00000108753&quot; &quot;ENSG00000108753&quot; &quot;ENSG00000108753&quot;
## [280] &quot;ENSG00000108753&quot; &quot;ENSG00000136720&quot; &quot;ENSG00000165868&quot;
## [283] &quot;ENSG00000165868&quot; &quot;ENSG00000003147&quot; &quot;ENSG00000003147&quot;
## [286] &quot;ENSG00000003147&quot; &quot;ENSG00000160223&quot; &quot;ENSG00000172201&quot;
## [289] &quot;ENSG00000172201&quot; &quot;ENSG00000134049&quot; &quot;ENSG00000117154&quot;
## [292] &quot;ENSG00000117154&quot; &quot;ENSG00000185950&quot; &quot;ENSG00000113430&quot;
## [295] &quot;ENSG00000113430&quot; &quot;ENSG00000105655&quot; &quot;ENSG00000105655&quot;
## [298] &quot;ENSG00000105655&quot; &quot;ENSG00000188385&quot; &quot;ENSG00000154118&quot;
## [301] &quot;ENSG00000107104&quot; &quot;ENSG00000102781&quot; &quot;ENSG00000189337&quot;
## [304] &quot;ENSG00000115041&quot; &quot;ENSG00000169427&quot; &quot;ENSG00000053918&quot;
## [307] &quot;ENSG00000134504&quot; &quot;ENSG00000167977&quot; &quot;ENSG00000127663&quot;
## [310] &quot;ENSG00000127663&quot; &quot;ENSG00000119537&quot; &quot;ENSG00000131149&quot;
## [313] &quot;ENSG00000136051&quot; &quot;ENSG00000257093&quot; &quot;ENSG00000122778&quot;
## [316] &quot;ENSG00000122778&quot; &quot;ENSG00000140950&quot; &quot;ENSG00000173214&quot;
## [319] &quot;ENSG00000130294&quot; &quot;ENSG00000066735&quot; &quot;ENSG00000066735&quot;
## [322] &quot;ENSG00000067082&quot; &quot;ENSG00000067082&quot; &quot;ENSG00000067082&quot;
## [325] &quot;ENSG00000067082&quot; &quot;ENSG00000150361&quot; &quot;ENSG00000197705&quot;
## [328] &quot;ENSG00000053747&quot; &quot;ENSG00000053747&quot; &quot;ENSG00000091136&quot;
## [331] &quot;ENSG00000168961&quot; &quot;ENSG00000168961&quot; &quot;ENSG00000168961&quot;
## [334] &quot;ENSG00000132130&quot; &quot;ENSG00000073350&quot; &quot;ENSG00000073350&quot;
## [337] &quot;ENSG00000073350&quot; &quot;ENSG00000103227&quot; &quot;ENSG00000103227&quot;
## [340] &quot;ENSG00000113368&quot; &quot;ENSG00000163380&quot; &quot;ENSG00000136944&quot;
## [343] &quot;ENSG00000136944&quot; &quot;ENSG00000136944&quot; &quot;ENSG00000167210&quot;
## [346] &quot;ENSG00000153395&quot; &quot;ENSG00000136141&quot; &quot;ENSG00000136141&quot;
## [349] &quot;ENSG00000136141&quot; &quot;ENSG00000146006&quot; &quot;ENSG00000146006&quot;
## [352] &quot;ENSG00000166159&quot; &quot;ENSG00000166159&quot; &quot;ENSG00000166159&quot;
## [355] &quot;ENSG00000166159&quot; &quot;ENSG00000166159&quot; &quot;ENSG00000166159&quot;
## [358] &quot;ENSG00000119681&quot; &quot;ENSG00000002822&quot; &quot;ENSG00000002822&quot;
## [361] &quot;ENSG00000002822&quot; &quot;ENSG00000178573&quot; &quot;ENSG00000178573&quot;
## [364] &quot;ENSG00000185022&quot; &quot;ENSG00000172469&quot; &quot;ENSG00000135525&quot;
## [367] &quot;ENSG00000135525&quot; &quot;ENSG00000135525&quot; &quot;ENSG00000138834&quot;
## [370] &quot;ENSG00000138834&quot; &quot;ENSG00000137337&quot; &quot;ENSG00000085871&quot;
## [373] &quot;ENSG00000085871&quot; &quot;ENSG00000085871&quot; &quot;ENSG00000085871&quot;
## [376] &quot;ENSG00000249567&quot; &quot;ENSG00000221323&quot; &quot;ENSG00000207588&quot;
## [379] &quot;ENSG00000105926&quot; &quot;ENSG00000158186&quot; &quot;ENSG00000182170&quot;
## [382] &quot;ENSG00000153944&quot; &quot;ENSG00000138823&quot; &quot;ENSG00000129422&quot;
## [385] &quot;ENSG00000129422&quot; &quot;ENSG00000132938&quot; &quot;ENSG00000133055&quot;
## [388] &quot;ENSG00000169994&quot; &quot;ENSG00000187556&quot; &quot;ENSG00000196498&quot;
## [391] &quot;ENSG00000196498&quot; &quot;ENSG00000196498&quot; &quot;ENSG00000196498&quot;
## [394] &quot;ENSG00000196498&quot; &quot;ENSG00000196498&quot; &quot;ENSG00000104419&quot;
## [397] &quot;ENSG00000104419&quot; &quot;ENSG00000104419&quot; &quot;ENSG00000104419&quot;
## [400] &quot;ENSG00000115286&quot; &quot;ENSG00000103154&quot; &quot;ENSG00000109320&quot;
## [403] &quot;ENSG00000109320&quot; &quot;ENSG00000148826&quot; &quot;ENSG00000148826&quot;
## [406] &quot;ENSG00000106410&quot; &quot;ENSG00000106410&quot; &quot;ENSG00000074771&quot;
## [409] &quot;ENSG00000170485&quot; &quot;ENSG00000214285&quot; &quot;ENSG00000105954&quot;
## [412] &quot;ENSG00000182667&quot; &quot;ENSG00000182667&quot; &quot;ENSG00000182667&quot;
## [415] &quot;ENSG00000115758&quot; &quot;ENSG00000205927&quot; &quot;ENSG00000183715&quot;
## [418] &quot;ENSG00000115947&quot; &quot;ENSG00000178602&quot; &quot;ENSG00000172818&quot;
## [421] &quot;ENSG00000125850&quot; &quot;ENSG00000099864&quot; &quot;ENSG00000099864&quot;
## [424] &quot;ENSG00000116183&quot; &quot;ENSG00000116183&quot; &quot;ENSG00000137819&quot;
## [427] &quot;ENSG00000102699&quot; &quot;ENSG00000242265&quot; &quot;ENSG00000242265&quot;
## [430] &quot;ENSG00000142655&quot; &quot;ENSG00000142655&quot; &quot;ENSG00000067057&quot;
## [433] &quot;ENSG00000144824&quot; &quot;ENSG00000165443&quot; &quot;ENSG00000105229&quot;
## [436] &quot;ENSG00000126822&quot; &quot;ENSG00000171680&quot; &quot;ENSG00000114554&quot;
## [439] &quot;ENSG00000114554&quot; &quot;ENSG00000114554&quot; &quot;ENSG00000141682&quot;
## [442] &quot;ENSG00000124225&quot; &quot;ENSG00000122512&quot; &quot;ENSG00000122512&quot;
## [445] &quot;ENSG00000168081&quot; &quot;ENSG00000014138&quot; &quot;ENSG00000166169&quot;
## [448] &quot;ENSG00000166169&quot; &quot;ENSG00000166169&quot; &quot;ENSG00000166169&quot;
## [451] &quot;ENSG00000106536&quot; &quot;ENSG00000106536&quot; &quot;ENSG00000162407&quot;
## [454] &quot;ENSG00000147535&quot; &quot;ENSG00000147535&quot; &quot;ENSG00000147535&quot;
## [457] &quot;ENSG00000154001&quot; &quot;ENSG00000170325&quot; &quot;ENSG00000170325&quot;
## [460] &quot;ENSG00000110851&quot; &quot;ENSG00000175785&quot; &quot;ENSG00000111725&quot;
## [463] &quot;ENSG00000106617&quot; &quot;ENSG00000065675&quot; &quot;ENSG00000065675&quot;
## [466] &quot;ENSG00000111218&quot; &quot;ENSG00000111218&quot; &quot;ENSG00000155066&quot;
## [469] &quot;ENSG00000155066&quot; &quot;ENSG00000155066&quot; &quot;ENSG00000205352&quot;
## [472] &quot;ENSG00000205352&quot; &quot;ENSG00000205352&quot; &quot;ENSG00000205352&quot;
## [475] &quot;ENSG00000167157&quot; &quot;ENSG00000163636&quot; &quot;ENSG00000169398&quot;
## [478] &quot;ENSG00000169398&quot; &quot;ENSG00000169398&quot; &quot;ENSG00000134644&quot;
## [481] &quot;ENSG00000134644&quot; &quot;ENSG00000112531&quot; &quot;ENSG00000112531&quot;
## [484] &quot;ENSG00000112531&quot; &quot;ENSG00000112531&quot; &quot;ENSG00000107560&quot;
## [487] &quot;ENSG00000168461&quot; &quot;ENSG00000141542&quot; &quot;ENSG00000017797&quot;
## [490] &quot;ENSG00000136828&quot; &quot;ENSG00000136828&quot; &quot;ENSG00000136828&quot;
## [493] &quot;ENSG00000185989&quot; &quot;ENSG00000165105&quot; &quot;ENSG00000122035&quot;
## [496] &quot;ENSG00000122965&quot; &quot;ENSG00000122965&quot; &quot;ENSG00000122965&quot;
## [499] &quot;ENSG00000160957&quot; &quot;ENSG00000143954&quot; &quot;ENSG00000143954&quot;
## [502] &quot;ENSG00000143954&quot; &quot;ENSG00000159788&quot; &quot;ENSG00000091844&quot;
## [505] &quot;ENSG00000158106&quot; &quot;ENSG00000183421&quot; &quot;ENSG00000139797&quot;
## [508] &quot;ENSG00000113269&quot; &quot;ENSG00000069667&quot; &quot;ENSG00000100784&quot;
## [511] &quot;ENSG00000100784&quot; &quot;ENSG00000141564&quot; &quot;ENSG00000141564&quot;
## [514] &quot;ENSG00000165526&quot; &quot;ENSG00000165526&quot; &quot;ENSG00000182010&quot;
## [517] &quot;ENSG00000198838&quot; &quot;ENSG00000198838&quot; &quot;ENSG00000256463&quot;
## [520] &quot;ENSG00000256463&quot; &quot;ENSG00000256463&quot; &quot;ENSG00000149021&quot;
## [523] &quot;ENSG00000136546&quot; &quot;ENSG00000136546&quot; &quot;ENSG00000146555&quot;
## [526] &quot;ENSG00000146555&quot; &quot;ENSG00000187764&quot; &quot;ENSG00000187764&quot;
## [529] &quot;ENSG00000163904&quot; &quot;ENSG00000152217&quot; &quot;ENSG00000183576&quot;
## [532] &quot;ENSG00000183576&quot; &quot;ENSG00000225383&quot; &quot;ENSG00000164690&quot;
## [535] &quot;ENSG00000142178&quot; &quot;ENSG00000159263&quot; &quot;ENSG00000159263&quot;
## [538] &quot;ENSG00000159263&quot; &quot;ENSG00000159263&quot; &quot;ENSG00000113504&quot;
## [541] &quot;ENSG00000152779&quot; &quot;ENSG00000162241&quot; &quot;ENSG00000162241&quot;
## [544] &quot;ENSG00000173262&quot; &quot;ENSG00000157765&quot; &quot;ENSG00000157765&quot;
## [547] &quot;ENSG00000160190&quot; &quot;ENSG00000143036&quot; &quot;ENSG00000143036&quot;
## [550] &quot;ENSG00000143036&quot; &quot;ENSG00000162426&quot; &quot;ENSG00000066230&quot;
## [553] &quot;ENSG00000173930&quot; &quot;ENSG00000166949&quot; &quot;ENSG00000127616&quot;
## [556] &quot;ENSG00000127616&quot; &quot;ENSG00000127616&quot; &quot;ENSG00000127616&quot;
## [559] &quot;ENSG00000188176&quot; &quot;ENSG00000048471&quot; &quot;ENSG00000172803&quot;
## [562] &quot;ENSG00000184557&quot; &quot;ENSG00000125398&quot; &quot;ENSG00000125398&quot;
## [565] &quot;ENSG00000185594&quot; &quot;ENSG00000187678&quot; &quot;ENSG00000187678&quot;
## [568] &quot;ENSG00000124783&quot; &quot;ENSG00000149418&quot; &quot;ENSG00000133121&quot;
## [571] &quot;ENSG00000133121&quot; &quot;ENSG00000102572&quot; &quot;ENSG00000152953&quot;
## [574] &quot;ENSG00000015592&quot; &quot;ENSG00000165730&quot; &quot;ENSG00000165730&quot;
## [577] &quot;ENSG00000165730&quot; &quot;ENSG00000165730&quot; &quot;ENSG00000198203&quot;
## [580] &quot;ENSG00000198203&quot; &quot;ENSG00000117614&quot; &quot;ENSG00000117614&quot;
## [583] &quot;ENSG00000131018&quot; &quot;ENSG00000131018&quot; &quot;ENSG00000078269&quot;
## [586] &quot;ENSG00000143028&quot; &quot;ENSG00000130699&quot; &quot;ENSG00000164691&quot;
## [589] &quot;ENSG00000164691&quot; &quot;ENSG00000164691&quot; &quot;ENSG00000167291&quot;
## [592] &quot;ENSG00000095383&quot; &quot;ENSG00000095383&quot; &quot;ENSG00000095383&quot;
## [595] &quot;ENSG00000095383&quot; &quot;ENSG00000095383&quot; &quot;ENSG00000095383&quot;
## [598] &quot;ENSG00000006638&quot; &quot;ENSG00000006638&quot; &quot;ENSG00000187621&quot;
## [601] &quot;ENSG00000160180&quot; &quot;ENSG00000163235&quot; &quot;ENSG00000163235&quot;
## [604] &quot;ENSG00000041988&quot; &quot;ENSG00000041988&quot; &quot;ENSG00000146426&quot;
## [607] &quot;ENSG00000106829&quot; &quot;ENSG00000155957&quot; &quot;ENSG00000144120&quot;
## [610] &quot;ENSG00000144120&quot; &quot;ENSG00000144120&quot; &quot;ENSG00000151353&quot;
## [613] &quot;ENSG00000151353&quot; &quot;ENSG00000214128&quot; &quot;ENSG00000160055&quot;
## [616] &quot;ENSG00000136205&quot; &quot;ENSG00000162341&quot; &quot;ENSG00000115705&quot;
## [619] &quot;ENSG00000115705&quot; &quot;ENSG00000126602&quot; &quot;ENSG00000167632&quot;
## [622] &quot;ENSG00000071575&quot; &quot;ENSG00000071575&quot; &quot;ENSG00000137699&quot;
## [625] &quot;ENSG00000100815&quot; &quot;ENSG00000103671&quot; &quot;ENSG00000119121&quot;
## [628] &quot;ENSG00000154743&quot; &quot;ENSG00000154743&quot; &quot;ENSG00000179981&quot;
## [631] &quot;ENSG00000182463&quot; &quot;ENSG00000121297&quot; &quot;ENSG00000214063&quot;
## [634] &quot;ENSG00000032389&quot; &quot;ENSG00000123607&quot; &quot;ENSG00000105948&quot;
## [637] &quot;ENSG00000105948&quot; &quot;ENSG00000105948&quot; &quot;ENSG00000100154&quot;
## [640] &quot;ENSG00000143367&quot; &quot;ENSG00000143367&quot; &quot;ENSG00000233608&quot;
## [643] &quot;ENSG00000184787&quot; &quot;ENSG00000184787&quot; &quot;ENSG00000184787&quot;
## [646] &quot;ENSG00000178473&quot; &quot;ENSG00000115446&quot; &quot;ENSG00000113763&quot;
## [649] &quot;ENSG00000101558&quot; &quot;ENSG00000101558&quot; &quot;ENSG00000129003&quot;
## [652] &quot;ENSG00000129003&quot; &quot;ENSG00000129003&quot; &quot;ENSG00000129003&quot;
## [655] &quot;ENSG00000176428&quot; &quot;ENSG00000185274&quot; &quot;ENSG00000185274&quot;
## [658] &quot;ENSG00000185274&quot; &quot;ENSG00000065268&quot; &quot;ENSG00000160193&quot;
## [661] &quot;ENSG00000103175&quot; &quot;ENSG00000198373&quot; &quot;ENSG00000076924&quot;
## [664] &quot;ENSG00000196584&quot; &quot;ENSG00000130733&quot; &quot;ENSG00000184828&quot;
## [667] &quot;ENSG00000213588&quot; &quot;ENSG00000014164&quot; &quot;ENSG00000175048&quot;
## [670] &quot;ENSG00000175048&quot; &quot;ENSG00000153786&quot; &quot;ENSG00000153786&quot;
## [673] &quot;ENSG00000156639&quot; &quot;ENSG00000156639&quot; &quot;ENSG00000133858&quot;
## [676] &quot;ENSG00000139800&quot; &quot;ENSG00000015171&quot; &quot;ENSG00000015171&quot;
## [679] &quot;ENSG00000102935&quot; &quot;ENSG00000102935&quot; &quot;ENSG00000183621&quot;
## [682] &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot;
## [685] &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot;
## [688] &quot;ENSG00000197857&quot; &quot;ENSG00000197857&quot; &quot;ENSG00000225614&quot;
## [691] &quot;ENSG00000225614&quot; &quot;ENSG00000171425&quot; &quot;ENSG00000157657&quot;
## [694] &quot;ENSG00000157657&quot; &quot;ENSG00000183779&quot; &quot;ENSG00000183779&quot;</code></pre>
<pre class="r"><code># Convert the TSS to ensg as well 

ref_gene_hg19_ensg &lt;- merge(refGene_hg19_TSS, gene_id, by.x = c(&quot;V5&quot;), by.y = c(&quot;Gene&quot;))

all_ref_gene_hg19_ensg &lt;- unique(ref_gene_hg19_ensg$ensg)

kidney_ref_gene &lt;- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene &lt;- as.numeric(as.vector(heart_ref_gene))
names(test_gene) &lt;-  all_ref_gene_hg19_ensg

# Run topGO
go_data &lt;- new(&quot;topGOdata&quot;,
                   ontology = &quot;BP&quot;,
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore &gt; 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = &quot;org.Hs.eg.db&quot;,
                   ID = &quot;ensembl&quot;)</code></pre>
<pre><code>## 
## Building most specific GOs .....</code></pre>
<pre><code>##  ( 11450 GO terms found. )</code></pre>
<pre><code>## 
## Build GO DAG topology ..........</code></pre>
<pre><code>##  ( 15456 GO terms and 36153 relations. )</code></pre>
<pre><code>## 
## Annotating nodes ...............</code></pre>
<pre><code>##  ( 14505 genes annotated to the GO terms. )</code></pre>
<pre class="r"><code># Perform enrichment test
go_test &lt;- runTest(go_data, algorithm = &quot;weight01&quot;, statistic = &quot;fisher&quot;)</code></pre>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 1568 nontrivial nodes
##       parameters: 
##           test statistic: fisher</code></pre>
<pre><code>## 
##   Level 15:  1 nodes to be scored    (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 14:  11 nodes to be scored   (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 13:  29 nodes to be scored   (103 eliminated genes)</code></pre>
<pre><code>## 
##   Level 12:  50 nodes to be scored   (298 eliminated genes)</code></pre>
<pre><code>## 
##   Level 11:  72 nodes to be scored   (2501 eliminated genes)</code></pre>
<pre><code>## 
##   Level 10:  107 nodes to be scored  (3937 eliminated genes)</code></pre>
<pre><code>## 
##   Level 9:   161 nodes to be scored  (5541 eliminated genes)</code></pre>
<pre><code>## 
##   Level 8:   201 nodes to be scored  (7251 eliminated genes)</code></pre>
<pre><code>## 
##   Level 7:   245 nodes to be scored  (8558 eliminated genes)</code></pre>
<pre><code>## 
##   Level 6:   265 nodes to be scored  (10615 eliminated genes)</code></pre>
<pre><code>## 
##   Level 5:   217 nodes to be scored  (12035 eliminated genes)</code></pre>
<pre><code>## 
##   Level 4:   127 nodes to be scored  (13480 eliminated genes)</code></pre>
<pre><code>## 
##   Level 3:   63 nodes to be scored   (13985 eliminated genes)</code></pre>
<pre><code>## 
##   Level 2:   18 nodes to be scored   (14177 eliminated genes)</code></pre>
<pre><code>## 
##   Level 1:   1 nodes to be scored    (14301 eliminated genes)</code></pre>
<pre class="r"><code>go_table &lt;- GenTable(go_data, weightFisher = go_test,
                         orderBy = &quot;weightFisher&quot;, ranksOf = &quot;weightFisher&quot;,
                         topNodes = sum(score(go_test) &lt; .01))

go_table</code></pre>
<pre><code>##        GO.ID                                        Term Annotated
## 1 GO:0009113      purine nucleobase biosynthetic process        12
## 2 GO:0033197                       response to vitamin E        12
## 3 GO:0006351                transcription, DNA-templated      3277
## 4 GO:0032007        negative regulation of TOR signaling        40
## 5 GO:0043547      positive regulation of GTPase activity       354
## 6 GO:0060070             canonical Wnt signaling pathway       274
## 7 GO:1903078 positive regulation of protein localizat...        46
## 8 GO:0071320                   cellular response to cAMP        47
##   Significant Expected weightFisher
## 1           2     0.03      0.00046
## 2           2     0.03      0.00046
## 3          14     8.81      0.00408
## 4           2     0.11      0.00515
## 5           6     0.95      0.00540
## 6           4     0.74      0.00656
## 7           2     0.12      0.00677
## 8           2     0.13      0.00706</code></pre>
<pre class="r"><code>goresults[&quot;GO:0098719&quot;] </code></pre>
<pre><code>## $&lt;NA&gt;
## NULL</code></pre>
</div>
<div id="liver-1" class="section level2">
<h2>Liver</h2>
<pre class="r"><code># Find the heart hypo
heart_hypo &lt;- human_chimp_heart_specific_25_conserved_H3K27_LIVER[which(human_chimp_heart_specific_25_conserved_H3K27_LIVER$V4 == &quot;hypo&quot; &amp; human_chimp_heart_specific_25_conserved_H3K27_LIVER$V6 != &quot;-1&quot;),]

heart_hypo &lt;- heart_hypo[complete.cases(heart_hypo), ]

# Find the closest gene

closest_heart &lt;- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc556d78f000.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc556f7b3abb.bed</code></pre>
<pre class="r"><code># Convert the gene name to ensg

gene_id &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv&quot;, stringsAsFactors = FALSE, header=TRUE, sep = &quot;,&quot;)

comb_kidney &lt;- merge(closest_heart, gene_id, by.x = c(&quot;V8&quot;), by.y = c(&quot;Gene&quot;))

comb_kidney$ensg</code></pre>
<pre><code>##    [1] &quot;ENSG00000121410&quot; &quot;ENSG00000114771&quot; &quot;ENSG00000114771&quot;
##    [4] &quot;ENSG00000141338&quot; &quot;ENSG00000121270&quot; &quot;ENSG00000108846&quot;
##    [7] &quot;ENSG00000108846&quot; &quot;ENSG00000108846&quot; &quot;ENSG00000108846&quot;
##   [10] &quot;ENSG00000138075&quot; &quot;ENSG00000076555&quot; &quot;ENSG00000076555&quot;
##   [13] &quot;ENSG00000181513&quot; &quot;ENSG00000181513&quot; &quot;ENSG00000181513&quot;
##   [16] &quot;ENSG00000181513&quot; &quot;ENSG00000122729&quot; &quot;ENSG00000197142&quot;
##   [19] &quot;ENSG00000183549&quot; &quot;ENSG00000143199&quot; &quot;ENSG00000143199&quot;
##   [22] &quot;ENSG00000143199&quot; &quot;ENSG00000143199&quot; &quot;ENSG00000162104&quot;
##   [25] &quot;ENSG00000183077&quot; &quot;ENSG00000183077&quot; &quot;ENSG00000183077&quot;
##   [28] &quot;ENSG00000157985&quot; &quot;ENSG00000157985&quot; &quot;ENSG00000157985&quot;
##   [31] &quot;ENSG00000165923&quot; &quot;ENSG00000165923&quot; &quot;ENSG00000165923&quot;
##   [34] &quot;ENSG00000144891&quot; &quot;ENSG00000144891&quot; &quot;ENSG00000144891&quot;
##   [37] &quot;ENSG00000144891&quot; &quot;ENSG00000172482&quot; &quot;ENSG00000113492&quot;
##   [40] &quot;ENSG00000042286&quot; &quot;ENSG00000042286&quot; &quot;ENSG00000011243&quot;
##   [43] &quot;ENSG00000106948&quot; &quot;ENSG00000198610&quot; &quot;ENSG00000198610&quot;
##   [46] &quot;ENSG00000142208&quot; &quot;ENSG00000023330&quot; &quot;ENSG00000023330&quot;
##   [49] &quot;ENSG00000170017&quot; &quot;ENSG00000170017&quot; &quot;ENSG00000170017&quot;
##   [52] &quot;ENSG00000170017&quot; &quot;ENSG00000170017&quot; &quot;ENSG00000170017&quot;
##   [55] &quot;ENSG00000170017&quot; &quot;ENSG00000170017&quot; &quot;ENSG00000033011&quot;
##   [58] &quot;ENSG00000198796&quot; &quot;ENSG00000136383&quot; &quot;ENSG00000136383&quot;
##   [61] &quot;ENSG00000162551&quot; &quot;ENSG00000162551&quot; &quot;ENSG00000162551&quot;
##   [64] &quot;ENSG00000106927&quot; &quot;ENSG00000139344&quot; &quot;ENSG00000166126&quot;
##   [67] &quot;ENSG00000166025&quot; &quot;ENSG00000166025&quot; &quot;ENSG00000116337&quot;
##   [70] &quot;ENSG00000116337&quot; &quot;ENSG00000174945&quot; &quot;ENSG00000116194&quot;
##   [73] &quot;ENSG00000116194&quot; &quot;ENSG00000198483&quot; &quot;ENSG00000154945&quot;
##   [76] &quot;ENSG00000131620&quot; &quot;ENSG00000131620&quot; &quot;ENSG00000074855&quot;
##   [79] &quot;ENSG00000042753&quot; &quot;ENSG00000042753&quot; &quot;ENSG00000132703&quot;
##   [82] &quot;ENSG00000110243&quot; &quot;ENSG00000110243&quot; &quot;ENSG00000084674&quot;
##   [85] &quot;ENSG00000175336&quot; &quot;ENSG00000118520&quot; &quot;ENSG00000118520&quot;
##   [88] &quot;ENSG00000118520&quot; &quot;ENSG00000118520&quot; &quot;ENSG00000196843&quot;
##   [91] &quot;ENSG00000133794&quot; &quot;ENSG00000133794&quot; &quot;ENSG00000133794&quot;
##   [94] &quot;ENSG00000136950&quot; &quot;ENSG00000111339&quot; &quot;ENSG00000111339&quot;
##   [97] &quot;ENSG00000100325&quot; &quot;ENSG00000100325&quot; &quot;ENSG00000141505&quot;
##  [100] &quot;ENSG00000141505&quot; &quot;ENSG00000141505&quot; &quot;ENSG00000141505&quot;
##  [103] &quot;ENSG00000161944&quot; &quot;ENSG00000161944&quot; &quot;ENSG00000161944&quot;
##  [106] &quot;ENSG00000161944&quot; &quot;ENSG00000161944&quot; &quot;ENSG00000126522&quot;
##  [109] &quot;ENSG00000126522&quot; &quot;ENSG00000126522&quot; &quot;ENSG00000169696&quot;
##  [112] &quot;ENSG00000169696&quot; &quot;ENSG00000169696&quot; &quot;ENSG00000169136&quot;
##  [115] &quot;ENSG00000169136&quot; &quot;ENSG00000157087&quot; &quot;ENSG00000157087&quot;
##  [118] &quot;ENSG00000186009&quot; &quot;ENSG00000107518&quot; &quot;ENSG00000204842&quot;
##  [121] &quot;ENSG00000158321&quot; &quot;ENSG00000119986&quot; &quot;ENSG00000109956&quot;
##  [124] &quot;ENSG00000109956&quot; &quot;ENSG00000182272&quot; &quot;ENSG00000117411&quot;
##  [127] &quot;ENSG00000156273&quot; &quot;ENSG00000175866&quot; &quot;ENSG00000175866&quot;
##  [130] &quot;ENSG00000175866&quot; &quot;ENSG00000175866&quot; &quot;ENSG00000255056&quot;
##  [133] &quot;ENSG00000172530&quot; &quot;ENSG00000172530&quot; &quot;ENSG00000172530&quot;
##  [136] &quot;ENSG00000172530&quot; &quot;ENSG00000172530&quot; &quot;ENSG00000114200&quot;
##  [139] &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot;
##  [142] &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot;
##  [145] &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot;
##  [148] &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot;
##  [151] &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot;
##  [154] &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot; &quot;ENSG00000153094&quot;
##  [157] &quot;ENSG00000121380&quot; &quot;ENSG00000121380&quot; &quot;ENSG00000100739&quot;
##  [160] &quot;ENSG00000015475&quot; &quot;ENSG00000015475&quot; &quot;ENSG00000136717&quot;
##  [163] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [166] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [169] &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot; &quot;ENSG00000136717&quot;
##  [172] &quot;ENSG00000115760&quot; &quot;ENSG00000197299&quot; &quot;ENSG00000168487&quot;
##  [175] &quot;ENSG00000168487&quot; &quot;ENSG00000168487&quot; &quot;ENSG00000168487&quot;
##  [178] &quot;ENSG00000140299&quot; &quot;ENSG00000145919&quot; &quot;ENSG00000145919&quot;
##  [181] &quot;ENSG00000176720&quot; &quot;ENSG00000234235&quot; &quot;ENSG00000214688&quot;
##  [184] &quot;ENSG00000110696&quot; &quot;ENSG00000111412&quot; &quot;ENSG00000173064&quot;
##  [187] &quot;ENSG00000214900&quot; &quot;ENSG00000128944&quot; &quot;ENSG00000128944&quot;
##  [190] &quot;ENSG00000128944&quot; &quot;ENSG00000166780&quot; &quot;ENSG00000125149&quot;
##  [193] &quot;ENSG00000125149&quot; &quot;ENSG00000174109&quot; &quot;ENSG00000185504&quot;
##  [196] &quot;ENSG00000185504&quot; &quot;ENSG00000168675&quot; &quot;ENSG00000168675&quot;
##  [199] &quot;ENSG00000168675&quot; &quot;ENSG00000168675&quot; &quot;ENSG00000168675&quot;
##  [202] &quot;ENSG00000168675&quot; &quot;ENSG00000130813&quot; &quot;ENSG00000130813&quot;
##  [205] &quot;ENSG00000130173&quot; &quot;ENSG00000168275&quot; &quot;ENSG00000159403&quot;
##  [208] &quot;ENSG00000159403&quot; &quot;ENSG00000182326&quot; &quot;ENSG00000182326&quot;
##  [211] &quot;ENSG00000166278&quot; &quot;ENSG00000166278&quot; &quot;ENSG00000166278&quot;
##  [214] &quot;ENSG00000166278&quot; &quot;ENSG00000101474&quot; &quot;ENSG00000128254&quot;
##  [217] &quot;ENSG00000215012&quot; &quot;ENSG00000100249&quot; &quot;ENSG00000115998&quot;
##  [220] &quot;ENSG00000204128&quot; &quot;ENSG00000215217&quot; &quot;ENSG00000039537&quot;
##  [223] &quot;ENSG00000146521&quot; &quot;ENSG00000244291&quot; &quot;ENSG00000170279&quot;
##  [226] &quot;ENSG00000157131&quot; &quot;ENSG00000021852&quot; &quot;ENSG00000021852&quot;
##  [229] &quot;ENSG00000021852&quot; &quot;ENSG00000176919&quot; &quot;ENSG00000113600&quot;
##  [232] &quot;ENSG00000134508&quot; &quot;ENSG00000134508&quot; &quot;ENSG00000154040&quot;
##  [235] &quot;ENSG00000154040&quot; &quot;ENSG00000154040&quot; &quot;ENSG00000154040&quot;
##  [238] &quot;ENSG00000154040&quot; &quot;ENSG00000171735&quot; &quot;ENSG00000171735&quot;
##  [241] &quot;ENSG00000171735&quot; &quot;ENSG00000171735&quot; &quot;ENSG00000042493&quot;
##  [244] &quot;ENSG00000110888&quot; &quot;ENSG00000110888&quot; &quot;ENSG00000110888&quot;
##  [247] &quot;ENSG00000110888&quot; &quot;ENSG00000110888&quot; &quot;ENSG00000213995&quot;
##  [250] &quot;ENSG00000213995&quot; &quot;ENSG00000142273&quot; &quot;ENSG00000142273&quot;
##  [253] &quot;ENSG00000139899&quot; &quot;ENSG00000139899&quot; &quot;ENSG00000160200&quot;
##  [256] &quot;ENSG00000160200&quot; &quot;ENSG00000144648&quot; &quot;ENSG00000005059&quot;
##  [259] &quot;ENSG00000160799&quot; &quot;ENSG00000144395&quot; &quot;ENSG00000165972&quot;
##  [262] &quot;ENSG00000161573&quot; &quot;ENSG00000110092&quot; &quot;ENSG00000110092&quot;
##  [265] &quot;ENSG00000177575&quot; &quot;ENSG00000177575&quot; &quot;ENSG00000177575&quot;
##  [268] &quot;ENSG00000177575&quot; &quot;ENSG00000134061&quot; &quot;ENSG00000178562&quot;
##  [271] &quot;ENSG00000178562&quot; &quot;ENSG00000178562&quot; &quot;ENSG00000110651&quot;
##  [274] &quot;ENSG00000151465&quot; &quot;ENSG00000099804&quot; &quot;ENSG00000198752&quot;
##  [277] &quot;ENSG00000149798&quot; &quot;ENSG00000170779&quot; &quot;ENSG00000170779&quot;
##  [280] &quot;ENSG00000136807&quot; &quot;ENSG00000166446&quot; &quot;ENSG00000153879&quot;
##  [283] &quot;ENSG00000153879&quot; &quot;ENSG00000241832&quot; &quot;ENSG00000149187&quot;
##  [286] &quot;ENSG00000011523&quot; &quot;ENSG00000011523&quot; &quot;ENSG00000159398&quot;
##  [289] &quot;ENSG00000159398&quot; &quot;ENSG00000087237&quot; &quot;ENSG00000243649&quot;
##  [292] &quot;ENSG00000165410&quot; &quot;ENSG00000165410&quot; &quot;ENSG00000165410&quot;
##  [295] &quot;ENSG00000016391&quot; &quot;ENSG00000016391&quot; &quot;ENSG00000085872&quot;
##  [298] &quot;ENSG00000177830&quot; &quot;ENSG00000177830&quot; &quot;ENSG00000177830&quot;
##  [301] &quot;ENSG00000177830&quot; &quot;ENSG00000177830&quot; &quot;ENSG00000110721&quot;
##  [304] &quot;ENSG00000110721&quot; &quot;ENSG00000104879&quot; &quot;ENSG00000159261&quot;
##  [307] &quot;ENSG00000159261&quot; &quot;ENSG00000125246&quot; &quot;ENSG00000091317&quot;
##  [310] &quot;ENSG00000091317&quot; &quot;ENSG00000170293&quot; &quot;ENSG00000150656&quot;
##  [313] &quot;ENSG00000133313&quot; &quot;ENSG00000168763&quot; &quot;ENSG00000168763&quot;
##  [316] &quot;ENSG00000158158&quot; &quot;ENSG00000184144&quot; &quot;ENSG00000106078&quot;
##  [319] &quot;ENSG00000196739&quot; &quot;ENSG00000118004&quot; &quot;ENSG00000118004&quot;
##  [322] &quot;ENSG00000118004&quot; &quot;ENSG00000118004&quot; &quot;ENSG00000118004&quot;
##  [325] &quot;ENSG00000118004&quot; &quot;ENSG00000131143&quot; &quot;ENSG00000178772&quot;
##  [328] &quot;ENSG00000178772&quot; &quot;ENSG00000178772&quot; &quot;ENSG00000100884&quot;
##  [331] &quot;ENSG00000157184&quot; &quot;ENSG00000167193&quot; &quot;ENSG00000167193&quot;
##  [334] &quot;ENSG00000178585&quot; &quot;ENSG00000178585&quot; &quot;ENSG00000040531&quot;
##  [337] &quot;ENSG00000040531&quot; &quot;ENSG00000111249&quot; &quot;ENSG00000111249&quot;
##  [340] &quot;ENSG00000154639&quot; &quot;ENSG00000154639&quot; &quot;ENSG00000154639&quot;
##  [343] &quot;ENSG00000154639&quot; &quot;ENSG00000154639&quot; &quot;ENSG00000166347&quot;
##  [346] &quot;ENSG00000166347&quot; &quot;ENSG00000166347&quot; &quot;ENSG00000166347&quot;
##  [349] &quot;ENSG00000166347&quot; &quot;ENSG00000166347&quot; &quot;ENSG00000166394&quot;
##  [352] &quot;ENSG00000140505&quot; &quot;ENSG00000135929&quot; &quot;ENSG00000135929&quot;
##  [355] &quot;ENSG00000135929&quot; &quot;ENSG00000138115&quot; &quot;ENSG00000138115&quot;
##  [358] &quot;ENSG00000138115&quot; &quot;ENSG00000138115&quot; &quot;ENSG00000106258&quot;
##  [361] &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot;
##  [364] &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot;
##  [367] &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot;
##  [370] &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot;
##  [373] &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot; &quot;ENSG00000106258&quot;
##  [376] &quot;ENSG00000186115&quot; &quot;ENSG00000171954&quot; &quot;ENSG00000171954&quot;
##  [379] &quot;ENSG00000186529&quot; &quot;ENSG00000186529&quot; &quot;ENSG00000186529&quot;
##  [382] &quot;ENSG00000164488&quot; &quot;ENSG00000112977&quot; &quot;ENSG00000172992&quot;
##  [385] &quot;ENSG00000043093&quot; &quot;ENSG00000132437&quot; &quot;ENSG00000100201&quot;
##  [388] &quot;ENSG00000100201&quot; &quot;ENSG00000141141&quot; &quot;ENSG00000100150&quot;
##  [391] &quot;ENSG00000100150&quot; &quot;ENSG00000077044&quot; &quot;ENSG00000102967&quot;
##  [394] &quot;ENSG00000181192&quot; &quot;ENSG00000181192&quot; &quot;ENSG00000067596&quot;
##  [397] &quot;ENSG00000184047&quot; &quot;ENSG00000184047&quot; &quot;ENSG00000184047&quot;
##  [400] &quot;ENSG00000184047&quot; &quot;ENSG00000258498&quot; &quot;ENSG00000258498&quot;
##  [403] &quot;ENSG00000258498&quot; &quot;ENSG00000066084&quot; &quot;ENSG00000108176&quot;
##  [406] &quot;ENSG00000108176&quot; &quot;ENSG00000163687&quot; &quot;ENSG00000163687&quot;
##  [409] &quot;ENSG00000167130&quot; &quot;ENSG00000167130&quot; &quot;ENSG00000167130&quot;
##  [412] &quot;ENSG00000184845&quot; &quot;ENSG00000143507&quot; &quot;ENSG00000161326&quot;
##  [415] &quot;ENSG00000108861&quot; &quot;ENSG00000107404&quot; &quot;ENSG00000105204&quot;
##  [418] &quot;ENSG00000105204&quot; &quot;ENSG00000105204&quot; &quot;ENSG00000123179&quot;
##  [421] &quot;ENSG00000134463&quot; &quot;ENSG00000229715&quot; &quot;ENSG00000229715&quot;
##  [424] &quot;ENSG00000142634&quot; &quot;ENSG00000169242&quot; &quot;ENSG00000169242&quot;
##  [427] &quot;ENSG00000099617&quot; &quot;ENSG00000172889&quot; &quot;ENSG00000172889&quot;
##  [430] &quot;ENSG00000146648&quot; &quot;ENSG00000146648&quot; &quot;ENSG00000146648&quot;
##  [433] &quot;ENSG00000146648&quot; &quot;ENSG00000171570&quot; &quot;ENSG00000173812&quot;
##  [436] &quot;ENSG00000106263&quot; &quot;ENSG00000106263&quot; &quot;ENSG00000066044&quot;
##  [439] &quot;ENSG00000196361&quot; &quot;ENSG00000196361&quot; &quot;ENSG00000155849&quot;
##  [442] &quot;ENSG00000155849&quot; &quot;ENSG00000155849&quot; &quot;ENSG00000134759&quot;
##  [445] &quot;ENSG00000134759&quot; &quot;ENSG00000134759&quot; &quot;ENSG00000134759&quot;
##  [448] &quot;ENSG00000134759&quot; &quot;ENSG00000134759&quot; &quot;ENSG00000134759&quot;
##  [451] &quot;ENSG00000160963&quot; &quot;ENSG00000167136&quot; &quot;ENSG00000116016&quot;
##  [454] &quot;ENSG00000182580&quot; &quot;ENSG00000196411&quot; &quot;ENSG00000196411&quot;
##  [457] &quot;ENSG00000130427&quot; &quot;ENSG00000113719&quot; &quot;ENSG00000086619&quot;
##  [460] &quot;ENSG00000139684&quot; &quot;ENSG00000196405&quot; &quot;ENSG00000205436&quot;
##  [463] &quot;ENSG00000107371&quot; &quot;ENSG00000107371&quot; &quot;ENSG00000182197&quot;
##  [466] &quot;ENSG00000092820&quot; &quot;ENSG00000126218&quot; &quot;ENSG00000180210&quot;
##  [469] &quot;ENSG00000159784&quot; &quot;ENSG00000109794&quot; &quot;ENSG00000152102&quot;
##  [472] &quot;ENSG00000185442&quot; &quot;ENSG00000188916&quot; &quot;ENSG00000189319&quot;
##  [475] &quot;ENSG00000133477&quot; &quot;ENSG00000188522&quot; &quot;ENSG00000176853&quot;
##  [478] &quot;ENSG00000169710&quot; &quot;ENSG00000177294&quot; &quot;ENSG00000100225&quot;
##  [481] &quot;ENSG00000100225&quot; &quot;ENSG00000171557&quot; &quot;ENSG00000171557&quot;
##  [484] &quot;ENSG00000115226&quot; &quot;ENSG00000160097&quot; &quot;ENSG00000160097&quot;
##  [487] &quot;ENSG00000170802&quot; &quot;ENSG00000106701&quot; &quot;ENSG00000106701&quot;
##  [490] &quot;ENSG00000070404&quot; &quot;ENSG00000160282&quot; &quot;ENSG00000160282&quot;
##  [493] &quot;ENSG00000107164&quot; &quot;ENSG00000165060&quot; &quot;ENSG00000165060&quot;
##  [496] &quot;ENSG00000165060&quot; &quot;ENSG00000163251&quot; &quot;ENSG00000131482&quot;
##  [499] &quot;ENSG00000131482&quot; &quot;ENSG00000154252&quot; &quot;ENSG00000108479&quot;
##  [502] &quot;ENSG00000106648&quot; &quot;ENSG00000106648&quot; &quot;ENSG00000128310&quot;
##  [505] &quot;ENSG00000130005&quot; &quot;ENSG00000130005&quot; &quot;ENSG00000111640&quot;
##  [508] &quot;ENSG00000213512&quot; &quot;ENSG00000213512&quot; &quot;ENSG00000106633&quot;
##  [511] &quot;ENSG00000106633&quot; &quot;ENSG00000084734&quot; &quot;ENSG00000023909&quot;
##  [514] &quot;ENSG00000178795&quot; &quot;ENSG00000141098&quot; &quot;ENSG00000141098&quot;
##  [517] &quot;ENSG00000141098&quot; &quot;ENSG00000151892&quot; &quot;ENSG00000115486&quot;
##  [520] &quot;ENSG00000115486&quot; &quot;ENSG00000157017&quot; &quot;ENSG00000157017&quot;
##  [523] &quot;ENSG00000157017&quot; &quot;ENSG00000157017&quot; &quot;ENSG00000157017&quot;
##  [526] &quot;ENSG00000157017&quot; &quot;ENSG00000157017&quot; &quot;ENSG00000157017&quot;
##  [529] &quot;ENSG00000157017&quot; &quot;ENSG00000157017&quot; &quot;ENSG00000137960&quot;
##  [532] &quot;ENSG00000151948&quot; &quot;ENSG00000016864&quot; &quot;ENSG00000016864&quot;
##  [535] &quot;ENSG00000203972&quot; &quot;ENSG00000196743&quot; &quot;ENSG00000196743&quot;
##  [538] &quot;ENSG00000112699&quot; &quot;ENSG00000114349&quot; &quot;ENSG00000114349&quot;
##  [541] &quot;ENSG00000100522&quot; &quot;ENSG00000185245&quot; &quot;ENSG00000119927&quot;
##  [544] &quot;ENSG00000119927&quot; &quot;ENSG00000119927&quot; &quot;ENSG00000119927&quot;
##  [547] &quot;ENSG00000112293&quot; &quot;ENSG00000132975&quot; &quot;ENSG00000180758&quot;
##  [550] &quot;ENSG00000166123&quot; &quot;ENSG00000176153&quot; &quot;ENSG00000176153&quot;
##  [553] &quot;ENSG00000176153&quot; &quot;ENSG00000075240&quot; &quot;ENSG00000083307&quot;
##  [556] &quot;ENSG00000139835&quot; &quot;ENSG00000100577&quot; &quot;ENSG00000077809&quot;
##  [559] &quot;ENSG00000077809&quot; &quot;ENSG00000077809&quot; &quot;ENSG00000077809&quot;
##  [562] &quot;ENSG00000077809&quot; &quot;ENSG00000101181&quot; &quot;ENSG00000169919&quot;
##  [565] &quot;ENSG00000130600&quot; &quot;ENSG00000148702&quot; &quot;ENSG00000148702&quot;
##  [568] &quot;ENSG00000148702&quot; &quot;ENSG00000130956&quot; &quot;ENSG00000084110&quot;
##  [571] &quot;ENSG00000084110&quot; &quot;ENSG00000084110&quot; &quot;ENSG00000115677&quot;
##  [574] &quot;ENSG00000115677&quot; &quot;ENSG00000122557&quot; &quot;ENSG00000114315&quot;
##  [577] &quot;ENSG00000114315&quot; &quot;ENSG00000168509&quot; &quot;ENSG00000168509&quot;
##  [580] &quot;ENSG00000168509&quot; &quot;ENSG00000168509&quot; &quot;ENSG00000109758&quot;
##  [583] &quot;ENSG00000137133&quot; &quot;ENSG00000100084&quot; &quot;ENSG00000108924&quot;
##  [586] &quot;ENSG00000117305&quot; &quot;ENSG00000117305&quot; &quot;ENSG00000115756&quot;
##  [589] &quot;ENSG00000158104&quot; &quot;ENSG00000110169&quot; &quot;ENSG00000110169&quot;
##  [592] &quot;ENSG00000113905&quot; &quot;ENSG00000113905&quot; &quot;ENSG00000005700&quot;
##  [595] &quot;ENSG00000115738&quot; &quot;ENSG00000142166&quot; &quot;ENSG00000142166&quot;
##  [598] &quot;ENSG00000115457&quot; &quot;ENSG00000243646&quot; &quot;ENSG00000115590&quot;
##  [601] &quot;ENSG00000115590&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [604] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [607] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [610] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [613] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [616] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [619] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [622] &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot; &quot;ENSG00000196083&quot;
##  [625] &quot;ENSG00000196083&quot; &quot;ENSG00000136689&quot; &quot;ENSG00000103522&quot;
##  [628] &quot;ENSG00000103522&quot; &quot;ENSG00000100385&quot; &quot;ENSG00000203485&quot;
##  [631] &quot;ENSG00000203485&quot; &quot;ENSG00000203485&quot; &quot;ENSG00000122641&quot;
##  [634] &quot;ENSG00000148384&quot; &quot;ENSG00000254647&quot; &quot;ENSG00000254647&quot;
##  [637] &quot;ENSG00000254647&quot; &quot;ENSG00000129965&quot; &quot;ENSG00000129965&quot;
##  [640] &quot;ENSG00000186480&quot; &quot;ENSG00000186480&quot; &quot;ENSG00000186480&quot;
##  [643] &quot;ENSG00000186480&quot; &quot;ENSG00000186480&quot; &quot;ENSG00000186480&quot;
##  [646] &quot;ENSG00000125629&quot; &quot;ENSG00000134070&quot; &quot;ENSG00000134070&quot;
##  [649] &quot;ENSG00000172183&quot; &quot;ENSG00000078747&quot; &quot;ENSG00000078747&quot;
##  [652] &quot;ENSG00000078747&quot; &quot;ENSG00000083457&quot; &quot;ENSG00000083457&quot;
##  [655] &quot;ENSG00000162267&quot; &quot;ENSG00000055955&quot; &quot;ENSG00000055955&quot;
##  [658] &quot;ENSG00000100605&quot; &quot;ENSG00000100605&quot; &quot;ENSG00000100605&quot;
##  [661] &quot;ENSG00000205726&quot; &quot;ENSG00000205726&quot; &quot;ENSG00000163166&quot;
##  [664] &quot;ENSG00000009765&quot; &quot;ENSG00000009765&quot; &quot;ENSG00000009765&quot;
##  [667] &quot;ENSG00000140044&quot; &quot;ENSG00000154118&quot; &quot;ENSG00000102781&quot;
##  [670] &quot;ENSG00000176407&quot; &quot;ENSG00000069424&quot; &quot;ENSG00000158445&quot;
##  [673] &quot;ENSG00000124780&quot; &quot;ENSG00000169427&quot; &quot;ENSG00000188997&quot;
##  [676] &quot;ENSG00000089094&quot; &quot;ENSG00000089094&quot; &quot;ENSG00000107077&quot;
##  [679] &quot;ENSG00000107077&quot; &quot;ENSG00000166783&quot; &quot;ENSG00000166783&quot;
##  [682] &quot;ENSG00000166783&quot; &quot;ENSG00000176542&quot; &quot;ENSG00000125337&quot;
##  [685] &quot;ENSG00000125337&quot; &quot;ENSG00000118922&quot; &quot;ENSG00000129911&quot;
##  [688] &quot;ENSG00000128607&quot; &quot;ENSG00000128607&quot; &quot;ENSG00000025800&quot;
##  [691] &quot;ENSG00000185896&quot; &quot;ENSG00000002549&quot; &quot;ENSG00000107929&quot;
##  [694] &quot;ENSG00000086730&quot; &quot;ENSG00000086730&quot; &quot;ENSG00000086730&quot;
##  [697] &quot;ENSG00000213398&quot; &quot;ENSG00000164406&quot; &quot;ENSG00000164406&quot;
##  [700] &quot;ENSG00000145826&quot; &quot;ENSG00000116977&quot; &quot;ENSG00000072163&quot;
##  [703] &quot;ENSG00000166035&quot; &quot;ENSG00000166035&quot; &quot;ENSG00000166035&quot;
##  [706] &quot;ENSG00000101670&quot; &quot;ENSG00000189067&quot; &quot;ENSG00000189067&quot;
##  [709] &quot;ENSG00000162761&quot; &quot;ENSG00000102910&quot; &quot;ENSG00000134324&quot;
##  [712] &quot;ENSG00000134324&quot; &quot;ENSG00000070018&quot; &quot;ENSG00000175489&quot;
##  [715] &quot;ENSG00000160233&quot; &quot;ENSG00000124831&quot; &quot;ENSG00000124831&quot;
##  [718] &quot;ENSG00000124831&quot; &quot;ENSG00000124831&quot; &quot;ENSG00000143669&quot;
##  [721] &quot;ENSG00000197063&quot; &quot;ENSG00000145050&quot; &quot;ENSG00000101460&quot;
##  [724] &quot;ENSG00000076984&quot; &quot;ENSG00000119487&quot; &quot;ENSG00000119487&quot;
##  [727] &quot;ENSG00000119487&quot; &quot;ENSG00000119487&quot; &quot;ENSG00000119487&quot;
##  [730] &quot;ENSG00000119487&quot; &quot;ENSG00000099785&quot; &quot;ENSG00000099785&quot;
##  [733] &quot;ENSG00000099785&quot; &quot;ENSG00000009724&quot; &quot;ENSG00000009724&quot;
##  [736] &quot;ENSG00000197971&quot; &quot;ENSG00000197971&quot; &quot;ENSG00000197971&quot;
##  [739] &quot;ENSG00000197971&quot; &quot;ENSG00000126217&quot; &quot;ENSG00000180398&quot;
##  [742] &quot;ENSG00000180398&quot; &quot;ENSG00000128285&quot; &quot;ENSG00000090674&quot;
##  [745] &quot;ENSG00000135679&quot; &quot;ENSG00000135679&quot; &quot;ENSG00000135679&quot;
##  [748] &quot;ENSG00000135679&quot; &quot;ENSG00000108510&quot; &quot;ENSG00000152127&quot;
##  [751] &quot;ENSG00000170430&quot; &quot;ENSG00000199065&quot; &quot;ENSG00000199065&quot;
##  [754] &quot;ENSG00000199065&quot; &quot;ENSG00000221039&quot; &quot;ENSG00000221063&quot;
##  [757] &quot;ENSG00000208017&quot; &quot;ENSG00000199075&quot; &quot;ENSG00000207727&quot;
##  [760] &quot;ENSG00000207875&quot; &quot;ENSG00000207875&quot; &quot;ENSG00000130382&quot;
##  [763] &quot;ENSG00000115648&quot; &quot;ENSG00000115648&quot; &quot;ENSG00000009950&quot;
##  [766] &quot;ENSG00000009950&quot; &quot;ENSG00000009950&quot; &quot;ENSG00000009950&quot;
##  [769] &quot;ENSG00000108960&quot; &quot;ENSG00000173269&quot; &quot;ENSG00000075643&quot;
##  [772] &quot;ENSG00000166391&quot; &quot;ENSG00000166391&quot; &quot;ENSG00000107186&quot;
##  [775] &quot;ENSG00000107186&quot; &quot;ENSG00000107186&quot; &quot;ENSG00000214026&quot;
##  [778] &quot;ENSG00000173867&quot; &quot;ENSG00000173531&quot; &quot;ENSG00000149480&quot;
##  [781] &quot;ENSG00000122085&quot; &quot;ENSG00000122085&quot; &quot;ENSG00000122085&quot;
##  [784] &quot;ENSG00000122085&quot; &quot;ENSG00000132613&quot; &quot;ENSG00000110921&quot;
##  [787] &quot;ENSG00000110921&quot; &quot;ENSG00000172927&quot; &quot;ENSG00000172927&quot;
##  [790] &quot;ENSG00000106436&quot; &quot;ENSG00000065534&quot; &quot;ENSG00000065534&quot;
##  [793] &quot;ENSG00000065534&quot; &quot;ENSG00000065534&quot; &quot;ENSG00000091536&quot;
##  [796] &quot;ENSG00000091536&quot; &quot;ENSG00000236242&quot; &quot;ENSG00000139597&quot;
##  [799] &quot;ENSG00000139597&quot; &quot;ENSG00000008130&quot; &quot;ENSG00000008130&quot;
##  [802] &quot;ENSG00000152620&quot; &quot;ENSG00000141562&quot; &quot;ENSG00000141562&quot;
##  [805] &quot;ENSG00000141562&quot; &quot;ENSG00000166833&quot; &quot;ENSG00000255043&quot;
##  [808] &quot;ENSG00000255043&quot; &quot;ENSG00000255043&quot; &quot;ENSG00000196498&quot;
##  [811] &quot;ENSG00000196498&quot; &quot;ENSG00000196498&quot; &quot;ENSG00000166579&quot;
##  [814] &quot;ENSG00000166579&quot; &quot;ENSG00000151366&quot; &quot;ENSG00000151366&quot;
##  [817] &quot;ENSG00000151366&quot; &quot;ENSG00000165802&quot; &quot;ENSG00000165802&quot;
##  [820] &quot;ENSG00000165802&quot; &quot;ENSG00000165802&quot; &quot;ENSG00000165802&quot;
##  [823] &quot;ENSG00000173848&quot; &quot;ENSG00000107954&quot; &quot;ENSG00000100906&quot;
##  [826] &quot;ENSG00000104825&quot; &quot;ENSG00000104825&quot; &quot;ENSG00000101004&quot;
##  [829] &quot;ENSG00000163293&quot; &quot;ENSG00000172548&quot; &quot;ENSG00000172548&quot;
##  [832] &quot;ENSG00000136783&quot; &quot;ENSG00000169251&quot; &quot;ENSG00000169251&quot;
##  [835] &quot;ENSG00000166741&quot; &quot;ENSG00000015520&quot; &quot;ENSG00000015520&quot;
##  [838] &quot;ENSG00000015520&quot; &quot;ENSG00000015520&quot; &quot;ENSG00000148734&quot;
##  [841] &quot;ENSG00000242349&quot; &quot;ENSG00000131910&quot; &quot;ENSG00000012504&quot;
##  [844] &quot;ENSG00000012504&quot; &quot;ENSG00000012504&quot; &quot;ENSG00000012504&quot;
##  [847] &quot;ENSG00000012504&quot; &quot;ENSG00000012504&quot; &quot;ENSG00000012504&quot;
##  [850] &quot;ENSG00000012504&quot; &quot;ENSG00000144852&quot; &quot;ENSG00000144852&quot;
##  [853] &quot;ENSG00000144852&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [856] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [859] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [862] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [865] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [868] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [871] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [874] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [877] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [880] &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot; &quot;ENSG00000143257&quot;
##  [883] &quot;ENSG00000143257&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [886] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [889] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [892] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [895] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [898] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [901] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot;
##  [904] &quot;ENSG00000113580&quot; &quot;ENSG00000113580&quot; &quot;ENSG00000116833&quot;
##  [907] &quot;ENSG00000116833&quot; &quot;ENSG00000116833&quot; &quot;ENSG00000116833&quot;
##  [910] &quot;ENSG00000116833&quot; &quot;ENSG00000116833&quot; &quot;ENSG00000180530&quot;
##  [913] &quot;ENSG00000171119&quot; &quot;ENSG00000151413&quot; &quot;ENSG00000090273&quot;
##  [916] &quot;ENSG00000168101&quot; &quot;ENSG00000168101&quot; &quot;ENSG00000168101&quot;
##  [919] &quot;ENSG00000168101&quot; &quot;ENSG00000132182&quot; &quot;ENSG00000125450&quot;
##  [922] &quot;ENSG00000176046&quot; &quot;ENSG00000176046&quot; &quot;ENSG00000065154&quot;
##  [925] &quot;ENSG00000065154&quot; &quot;ENSG00000180304&quot; &quot;ENSG00000111325&quot;
##  [928] &quot;ENSG00000060491&quot; &quot;ENSG00000119547&quot; &quot;ENSG00000119547&quot;
##  [931] &quot;ENSG00000224855&quot; &quot;ENSG00000175619&quot; &quot;ENSG00000140961&quot;
##  [934] &quot;ENSG00000185624&quot; &quot;ENSG00000179364&quot; &quot;ENSG00000100266&quot;
##  [937] &quot;ENSG00000006712&quot; &quot;ENSG00000006712&quot; &quot;ENSG00000006712&quot;
##  [940] &quot;ENSG00000140694&quot; &quot;ENSG00000140694&quot; &quot;ENSG00000140694&quot;
##  [943] &quot;ENSG00000140694&quot; &quot;ENSG00000140694&quot; &quot;ENSG00000140694&quot;
##  [946] &quot;ENSG00000140694&quot; &quot;ENSG00000140694&quot; &quot;ENSG00000140694&quot;
##  [949] &quot;ENSG00000111224&quot; &quot;ENSG00000185630&quot; &quot;ENSG00000185630&quot;
##  [952] &quot;ENSG00000185630&quot; &quot;ENSG00000167081&quot; &quot;ENSG00000173599&quot;
##  [955] &quot;ENSG00000166228&quot; &quot;ENSG00000175198&quot; &quot;ENSG00000175198&quot;
##  [958] &quot;ENSG00000175198&quot; &quot;ENSG00000114054&quot; &quot;ENSG00000114054&quot;
##  [961] &quot;ENSG00000140479&quot; &quot;ENSG00000140479&quot; &quot;ENSG00000140479&quot;
##  [964] &quot;ENSG00000140479&quot; &quot;ENSG00000140479&quot; &quot;ENSG00000140479&quot;
##  [967] &quot;ENSG00000140479&quot; &quot;ENSG00000140479&quot; &quot;ENSG00000169174&quot;
##  [970] &quot;ENSG00000116005&quot; &quot;ENSG00000106244&quot; &quot;ENSG00000113721&quot;
##  [973] &quot;ENSG00000113721&quot; &quot;ENSG00000155660&quot; &quot;ENSG00000107438&quot;
##  [976] &quot;ENSG00000133027&quot; &quot;ENSG00000133027&quot; &quot;ENSG00000124299&quot;
##  [979] &quot;ENSG00000124299&quot; &quot;ENSG00000124299&quot; &quot;ENSG00000124299&quot;
##  [982] &quot;ENSG00000124299&quot; &quot;ENSG00000124299&quot; &quot;ENSG00000161031&quot;
##  [985] &quot;ENSG00000107537&quot; &quot;ENSG00000060642&quot; &quot;ENSG00000155629&quot;
##  [988] &quot;ENSG00000155629&quot; &quot;ENSG00000174238&quot; &quot;ENSG00000090975&quot;
##  [991] &quot;ENSG00000178385&quot; &quot;ENSG00000076356&quot; &quot;ENSG00000115946&quot;
##  [994] &quot;ENSG00000101346&quot; &quot;ENSG00000101346&quot; &quot;ENSG00000127948&quot;
##  [997] &quot;ENSG00000168938&quot; &quot;ENSG00000168938&quot; &quot;ENSG00000108179&quot;
## [1000] &quot;ENSG00000118898&quot; &quot;ENSG00000219607&quot; &quot;ENSG00000137713&quot;
## [1003] &quot;ENSG00000137713&quot; &quot;ENSG00000137713&quot; &quot;ENSG00000137713&quot;
## [1006] &quot;ENSG00000137713&quot; &quot;ENSG00000066027&quot; &quot;ENSG00000066027&quot;
## [1009] &quot;ENSG00000066027&quot; &quot;ENSG00000100239&quot; &quot;ENSG00000100239&quot;
## [1012] &quot;ENSG00000100239&quot; &quot;ENSG00000100239&quot; &quot;ENSG00000110075&quot;
## [1015] &quot;ENSG00000110075&quot; &quot;ENSG00000110075&quot; &quot;ENSG00000110075&quot;
## [1018] &quot;ENSG00000110075&quot; &quot;ENSG00000110075&quot; &quot;ENSG00000147596&quot;
## [1021] &quot;ENSG00000116690&quot; &quot;ENSG00000116690&quot; &quot;ENSG00000116690&quot;
## [1024] &quot;ENSG00000116690&quot; &quot;ENSG00000131791&quot; &quot;ENSG00000154229&quot;
## [1027] &quot;ENSG00000067606&quot; &quot;ENSG00000101000&quot; &quot;ENSG00000117707&quot;
## [1030] &quot;ENSG00000117707&quot; &quot;ENSG00000126231&quot; &quot;ENSG00000126231&quot;
## [1033] &quot;ENSG00000161542&quot; &quot;ENSG00000161542&quot; &quot;ENSG00000248405&quot;
## [1036] &quot;ENSG00000080815&quot; &quot;ENSG00000080815&quot; &quot;ENSG00000159352&quot;
## [1039] &quot;ENSG00000180822&quot; &quot;ENSG00000180822&quot; &quot;ENSG00000180822&quot;
## [1042] &quot;ENSG00000124212&quot; &quot;ENSG00000206527&quot; &quot;ENSG00000070159&quot;
## [1045] &quot;ENSG00000070159&quot; &quot;ENSG00000070159&quot; &quot;ENSG00000070159&quot;
## [1048] &quot;ENSG00000060656&quot; &quot;ENSG00000060656&quot; &quot;ENSG00000060656&quot;
## [1051] &quot;ENSG00000060656&quot; &quot;ENSG00000060656&quot; &quot;ENSG00000060656&quot;
## [1054] &quot;ENSG00000060656&quot; &quot;ENSG00000060656&quot; &quot;ENSG00000136045&quot;
## [1057] &quot;ENSG00000136045&quot; &quot;ENSG00000165661&quot; &quot;ENSG00000213339&quot;
## [1060] &quot;ENSG00000172007&quot; &quot;ENSG00000172007&quot; &quot;ENSG00000172007&quot;
## [1063] &quot;ENSG00000160271&quot; &quot;ENSG00000108961&quot; &quot;ENSG00000108961&quot;
## [1066] &quot;ENSG00000108961&quot; &quot;ENSG00000173166&quot; &quot;ENSG00000173166&quot;
## [1069] &quot;ENSG00000105122&quot; &quot;ENSG00000108551&quot; &quot;ENSG00000108551&quot;
## [1072] &quot;ENSG00000165105&quot; &quot;ENSG00000100320&quot; &quot;ENSG00000100320&quot;
## [1075] &quot;ENSG00000184863&quot; &quot;ENSG00000138207&quot; &quot;ENSG00000139547&quot;
## [1078] &quot;ENSG00000139547&quot; &quot;ENSG00000174236&quot; &quot;ENSG00000173156&quot;
## [1081] &quot;ENSG00000111785&quot; &quot;ENSG00000167705&quot; &quot;ENSG00000108830&quot;
## [1084] &quot;ENSG00000108830&quot; &quot;ENSG00000178828&quot; &quot;ENSG00000132383&quot;
## [1087] &quot;ENSG00000125844&quot; &quot;ENSG00000125844&quot; &quot;ENSG00000160214&quot;
## [1090] &quot;ENSG00000163825&quot; &quot;ENSG00000163825&quot; &quot;ENSG00000163602&quot;
## [1093] &quot;ENSG00000020577&quot; &quot;ENSG00000020577&quot; &quot;ENSG00000101347&quot;
## [1096] &quot;ENSG00000123453&quot; &quot;ENSG00000168077&quot; &quot;ENSG00000168077&quot;
## [1099] &quot;ENSG00000073060&quot; &quot;ENSG00000073060&quot; &quot;ENSG00000073060&quot;
## [1102] &quot;ENSG00000073060&quot; &quot;ENSG00000073060&quot; &quot;ENSG00000073060&quot;
## [1105] &quot;ENSG00000143653&quot; &quot;ENSG00000100012&quot; &quot;ENSG00000100012&quot;
## [1108] &quot;ENSG00000100012&quot; &quot;ENSG00000100012&quot; &quot;ENSG00000103184&quot;
## [1111] &quot;ENSG00000254154&quot; &quot;ENSG00000254154&quot; &quot;ENSG00000065665&quot;
## [1114] &quot;ENSG00000065665&quot; &quot;ENSG00000065665&quot; &quot;ENSG00000065665&quot;
## [1117] &quot;ENSG00000065665&quot; &quot;ENSG00000025796&quot; &quot;ENSG00000187742&quot;
## [1120] &quot;ENSG00000187742&quot; &quot;ENSG00000179918&quot; &quot;ENSG00000184640&quot;
## [1123] &quot;ENSG00000197249&quot; &quot;ENSG00000196136&quot; &quot;ENSG00000100665&quot;
## [1126] &quot;ENSG00000117601&quot; &quot;ENSG00000117601&quot; &quot;ENSG00000183576&quot;
## [1129] &quot;ENSG00000183576&quot; &quot;ENSG00000144040&quot; &quot;ENSG00000144040&quot;
## [1132] &quot;ENSG00000144040&quot; &quot;ENSG00000104205&quot; &quot;ENSG00000104205&quot;
## [1135] &quot;ENSG00000130147&quot; &quot;ENSG00000107957&quot; &quot;ENSG00000105251&quot;
## [1138] &quot;ENSG00000105251&quot; &quot;ENSG00000138606&quot; &quot;ENSG00000164690&quot;
## [1141] &quot;ENSG00000164690&quot; &quot;ENSG00000138771&quot; &quot;ENSG00000112246&quot;
## [1144] &quot;ENSG00000169375&quot; &quot;ENSG00000096717&quot; &quot;ENSG00000124523&quot;
## [1147] &quot;ENSG00000141526&quot; &quot;ENSG00000170190&quot; &quot;ENSG00000112337&quot;
## [1150] &quot;ENSG00000112337&quot; &quot;ENSG00000112337&quot; &quot;ENSG00000146039&quot;
## [1153] &quot;ENSG00000110436&quot; &quot;ENSG00000168575&quot; &quot;ENSG00000175003&quot;
## [1156] &quot;ENSG00000175003&quot; &quot;ENSG00000175003&quot; &quot;ENSG00000175003&quot;
## [1159] &quot;ENSG00000149742&quot; &quot;ENSG00000148339&quot; &quot;ENSG00000171612&quot;
## [1162] &quot;ENSG00000171612&quot; &quot;ENSG00000171612&quot; &quot;ENSG00000181035&quot;
## [1165] &quot;ENSG00000140107&quot; &quot;ENSG00000140284&quot; &quot;ENSG00000140284&quot;
## [1168] &quot;ENSG00000083807&quot; &quot;ENSG00000197496&quot; &quot;ENSG00000173262&quot;
## [1171] &quot;ENSG00000196660&quot; &quot;ENSG00000196660&quot; &quot;ENSG00000196660&quot;
## [1174] &quot;ENSG00000196660&quot; &quot;ENSG00000130958&quot; &quot;ENSG00000127526&quot;
## [1177] &quot;ENSG00000138449&quot; &quot;ENSG00000162426&quot; &quot;ENSG00000022567&quot;
## [1180] &quot;ENSG00000022567&quot; &quot;ENSG00000076351&quot; &quot;ENSG00000076351&quot;
## [1183] &quot;ENSG00000139508&quot; &quot;ENSG00000139508&quot; &quot;ENSG00000111181&quot;
## [1186] &quot;ENSG00000111181&quot; &quot;ENSG00000111181&quot; &quot;ENSG00000130876&quot;
## [1189] &quot;ENSG00000130876&quot; &quot;ENSG00000103257&quot; &quot;ENSG00000137834&quot;
## [1192] &quot;ENSG00000137834&quot; &quot;ENSG00000137834&quot; &quot;ENSG00000137834&quot;
## [1195] &quot;ENSG00000103056&quot; &quot;ENSG00000103056&quot; &quot;ENSG00000198742&quot;
## [1198] &quot;ENSG00000198742&quot; &quot;ENSG00000198742&quot; &quot;ENSG00000165684&quot;
## [1201] &quot;ENSG00000201902&quot; &quot;ENSG00000104852&quot; &quot;ENSG00000147481&quot;
## [1204] &quot;ENSG00000142168&quot; &quot;ENSG00000142168&quot; &quot;ENSG00000184985&quot;
## [1207] &quot;ENSG00000184985&quot; &quot;ENSG00000137642&quot; &quot;ENSG00000100146&quot;
## [1210] &quot;ENSG00000153498&quot; &quot;ENSG00000227213&quot; &quot;ENSG00000006282&quot;
## [1213] &quot;ENSG00000006282&quot; &quot;ENSG00000006282&quot; &quot;ENSG00000186583&quot;
## [1216] &quot;ENSG00000186583&quot; &quot;ENSG00000204710&quot; &quot;ENSG00000100014&quot;
## [1219] &quot;ENSG00000072080&quot; &quot;ENSG00000136158&quot; &quot;ENSG00000162032&quot;
## [1222] &quot;ENSG00000173898&quot; &quot;ENSG00000137877&quot; &quot;ENSG00000049319&quot;
## [1225] &quot;ENSG00000182934&quot; &quot;ENSG00000182934&quot; &quot;ENSG00000147488&quot;
## [1228] &quot;ENSG00000166444&quot; &quot;ENSG00000073849&quot; &quot;ENSG00000070731&quot;
## [1231] &quot;ENSG00000133121&quot; &quot;ENSG00000166888&quot; &quot;ENSG00000115107&quot;
## [1234] &quot;ENSG00000115107&quot; &quot;ENSG00000115107&quot; &quot;ENSG00000115107&quot;
## [1237] &quot;ENSG00000115107&quot; &quot;ENSG00000115107&quot; &quot;ENSG00000118046&quot;
## [1240] &quot;ENSG00000204344&quot; &quot;ENSG00000116030&quot; &quot;ENSG00000116030&quot;
## [1243] &quot;ENSG00000116030&quot; &quot;ENSG00000116030&quot; &quot;ENSG00000116030&quot;
## [1246] &quot;ENSG00000116030&quot; &quot;ENSG00000100242&quot; &quot;ENSG00000196235&quot;
## [1249] &quot;ENSG00000173227&quot; &quot;ENSG00000177156&quot; &quot;ENSG00000143374&quot;
## [1252] &quot;ENSG00000135111&quot; &quot;ENSG00000135111&quot; &quot;ENSG00000135111&quot;
## [1255] &quot;ENSG00000135111&quot; &quot;ENSG00000135111&quot; &quot;ENSG00000135111&quot;
## [1258] &quot;ENSG00000011007&quot; &quot;ENSG00000163792&quot; &quot;ENSG00000163792&quot;
## [1261] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1264] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1267] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1270] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1273] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1276] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1279] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1282] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1285] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1288] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1291] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1294] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1297] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1300] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1303] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1306] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1309] &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot; &quot;ENSG00000148737&quot;
## [1312] &quot;ENSG00000148737&quot; &quot;ENSG00000101190&quot; &quot;ENSG00000100109&quot;
## [1315] &quot;ENSG00000100109&quot; &quot;ENSG00000106327&quot; &quot;ENSG00000072274&quot;
## [1318] &quot;ENSG00000125780&quot; &quot;ENSG00000125780&quot; &quot;ENSG00000101158&quot;
## [1321] &quot;ENSG00000079134&quot; &quot;ENSG00000172009&quot; &quot;ENSG00000090534&quot;
## [1324] &quot;ENSG00000090534&quot; &quot;ENSG00000090534&quot; &quot;ENSG00000187720&quot;
## [1327] &quot;ENSG00000221995&quot; &quot;ENSG00000104980&quot; &quot;ENSG00000100234&quot;
## [1330] &quot;ENSG00000150455&quot; &quot;ENSG00000150455&quot; &quot;ENSG00000196781&quot;
## [1333] &quot;ENSG00000140332&quot; &quot;ENSG00000140332&quot; &quot;ENSG00000140332&quot;
## [1336] &quot;ENSG00000135926&quot; &quot;ENSG00000134291&quot; &quot;ENSG00000134291&quot;
## [1339] &quot;ENSG00000134291&quot; &quot;ENSG00000134291&quot; &quot;ENSG00000168936&quot;
## [1342] &quot;ENSG00000168936&quot; &quot;ENSG00000064545&quot; &quot;ENSG00000064545&quot;
## [1345] &quot;ENSG00000106565&quot; &quot;ENSG00000106565&quot; &quot;ENSG00000106565&quot;
## [1348] &quot;ENSG00000187824&quot; &quot;ENSG00000204278&quot; &quot;ENSG00000204278&quot;
## [1351] &quot;ENSG00000204278&quot; &quot;ENSG00000149582&quot; &quot;ENSG00000149582&quot;
## [1354] &quot;ENSG00000149582&quot; &quot;ENSG00000149582&quot; &quot;ENSG00000149582&quot;
## [1357] &quot;ENSG00000149582&quot; &quot;ENSG00000109133&quot; &quot;ENSG00000109133&quot;
## [1360] &quot;ENSG00000177042&quot; &quot;ENSG00000177042&quot; &quot;ENSG00000109084&quot;
## [1363] &quot;ENSG00000187045&quot; &quot;ENSG00000109079&quot; &quot;ENSG00000185361&quot;
## [1366] &quot;ENSG00000028137&quot; &quot;ENSG00000141232&quot; &quot;ENSG00000141232&quot;
## [1369] &quot;ENSG00000136816&quot; &quot;ENSG00000188001&quot; &quot;ENSG00000056558&quot;
## [1372] &quot;ENSG00000131653&quot; &quot;ENSG00000131653&quot; &quot;ENSG00000173334&quot;
## [1375] &quot;ENSG00000173334&quot; &quot;ENSG00000173334&quot; &quot;ENSG00000101255&quot;
## [1378] &quot;ENSG00000234127&quot; &quot;ENSG00000234127&quot; &quot;ENSG00000169871&quot;
## [1381] &quot;ENSG00000144481&quot; &quot;ENSG00000144481&quot; &quot;ENSG00000011105&quot;
## [1384] &quot;ENSG00000011105&quot; &quot;ENSG00000235217&quot; &quot;ENSG00000172425&quot;
## [1387] &quot;ENSG00000100304&quot; &quot;ENSG00000120440&quot; &quot;ENSG00000124120&quot;
## [1390] &quot;ENSG00000124120&quot; &quot;ENSG00000124120&quot; &quot;ENSG00000118271&quot;
## [1393] &quot;ENSG00000136295&quot; &quot;ENSG00000198431&quot; &quot;ENSG00000143569&quot;
## [1396] &quot;ENSG00000078140&quot; &quot;ENSG00000078140&quot; &quot;ENSG00000078140&quot;
## [1399] &quot;ENSG00000169139&quot; &quot;ENSG00000168246&quot; &quot;ENSG00000162543&quot;
## [1402] &quot;ENSG00000162543&quot; &quot;ENSG00000241635&quot; &quot;ENSG00000241635&quot;
## [1405] &quot;ENSG00000105668&quot; &quot;ENSG00000105698&quot; &quot;ENSG00000105698&quot;
## [1408] &quot;ENSG00000140455&quot; &quot;ENSG00000140455&quot; &quot;ENSG00000140455&quot;
## [1411] &quot;ENSG00000140455&quot; &quot;ENSG00000103404&quot; &quot;ENSG00000148429&quot;
## [1414] &quot;ENSG00000132952&quot; &quot;ENSG00000152818&quot; &quot;ENSG00000132612&quot;
## [1417] &quot;ENSG00000141252&quot; &quot;ENSG00000141252&quot; &quot;ENSG00000109072&quot;
## [1420] &quot;ENSG00000109072&quot; &quot;ENSG00000110799&quot; &quot;ENSG00000186153&quot;
## [1423] &quot;ENSG00000186153&quot; &quot;ENSG00000186153&quot; &quot;ENSG00000114127&quot;
## [1426] &quot;ENSG00000114127&quot; &quot;ENSG00000175155&quot; &quot;ENSG00000205189&quot;
## [1429] &quot;ENSG00000205189&quot; &quot;ENSG00000169155&quot; &quot;ENSG00000169155&quot;
## [1432] &quot;ENSG00000184828&quot; &quot;ENSG00000141664&quot; &quot;ENSG00000153786&quot;
## [1435] &quot;ENSG00000153786&quot; &quot;ENSG00000099904&quot; &quot;ENSG00000099904&quot;
## [1438] &quot;ENSG00000140836&quot; &quot;ENSG00000185650&quot; &quot;ENSG00000185650&quot;
## [1441] &quot;ENSG00000100711&quot; &quot;ENSG00000100711&quot; &quot;ENSG00000108175&quot;
## [1444] &quot;ENSG00000122515&quot; &quot;ENSG00000085644&quot; &quot;ENSG00000171940&quot;
## [1447] &quot;ENSG00000168813&quot; &quot;ENSG00000168813&quot; &quot;ENSG00000185252&quot;
## [1450] &quot;ENSG00000185252&quot; &quot;ENSG00000185252&quot; &quot;ENSG00000185252&quot;
## [1453] &quot;ENSG00000185252&quot; &quot;ENSG00000188372&quot; &quot;ENSG00000070476&quot;
## [1456] &quot;ENSG00000070476&quot;</code></pre>
<pre class="r"><code># Convert the TSS to ensg as well 

ref_gene_hg19_ensg &lt;- merge(refGene_hg19_TSS, gene_id, by.x = c(&quot;V5&quot;), by.y = c(&quot;Gene&quot;))

all_ref_gene_hg19_ensg &lt;- unique(ref_gene_hg19_ensg$ensg)

liver_ref_gene &lt;- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene &lt;- as.numeric(as.vector(liver_ref_gene))
names(test_gene) &lt;-  all_ref_gene_hg19_ensg

# Run topGO
go_data &lt;- new(&quot;topGOdata&quot;,
                   ontology = &quot;BP&quot;,
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore &gt; 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = &quot;org.Hs.eg.db&quot;,
                   ID = &quot;ensembl&quot;)</code></pre>
<pre><code>## 
## Building most specific GOs .....</code></pre>
<pre><code>##  ( 11450 GO terms found. )</code></pre>
<pre><code>## 
## Build GO DAG topology ..........</code></pre>
<pre><code>##  ( 15456 GO terms and 36153 relations. )</code></pre>
<pre><code>## 
## Annotating nodes ...............</code></pre>
<pre><code>##  ( 14505 genes annotated to the GO terms. )</code></pre>
<pre class="r"><code># Perform enrichment test
go_test &lt;- runTest(go_data, algorithm = &quot;weight01&quot;, statistic = &quot;fisher&quot;)</code></pre>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 6020 nontrivial nodes
##       parameters: 
##           test statistic: fisher</code></pre>
<pre><code>## 
##   Level 18:  1 nodes to be scored    (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 17:  4 nodes to be scored    (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 16:  20 nodes to be scored   (8 eliminated genes)</code></pre>
<pre><code>## 
##   Level 15:  57 nodes to be scored   (71 eliminated genes)</code></pre>
<pre><code>## 
##   Level 14:  116 nodes to be scored  (316 eliminated genes)</code></pre>
<pre><code>## 
##   Level 13:  208 nodes to be scored  (911 eliminated genes)</code></pre>
<pre><code>## 
##   Level 12:  308 nodes to be scored  (1979 eliminated genes)</code></pre>
<pre><code>## 
##   Level 11:  502 nodes to be scored  (4148 eliminated genes)</code></pre>
<pre><code>## 
##   Level 10:  677 nodes to be scored  (5660 eliminated genes)</code></pre>
<pre><code>## 
##   Level 9:   819 nodes to be scored  (7855 eliminated genes)</code></pre>
<pre><code>## 
##   Level 8:   848 nodes to be scored  (9971 eliminated genes)</code></pre>
<pre><code>## 
##   Level 7:   873 nodes to be scored  (11436 eliminated genes)</code></pre>
<pre><code>## 
##   Level 6:   739 nodes to be scored  (12657 eliminated genes)</code></pre>
<pre><code>## 
##   Level 5:   470 nodes to be scored  (13331 eliminated genes)</code></pre>
<pre><code>## 
##   Level 4:   250 nodes to be scored  (13846 eliminated genes)</code></pre>
<pre><code>## 
##   Level 3:   105 nodes to be scored  (14079 eliminated genes)</code></pre>
<pre><code>## 
##   Level 2:   22 nodes to be scored   (14207 eliminated genes)</code></pre>
<pre><code>## 
##   Level 1:   1 nodes to be scored    (14324 eliminated genes)</code></pre>
<pre class="r"><code>go_table &lt;- GenTable(go_data, weightFisher = go_test,
                         orderBy = &quot;weightFisher&quot;, ranksOf = &quot;weightFisher&quot;,
                         topNodes = sum(score(go_test) &lt; .05))

go_table</code></pre>
<pre><code>##          GO.ID                                        Term Annotated
## 1   GO:0030449         regulation of complement activation        41
## 2   GO:0042632                     cholesterol homeostasis        68
## 3   GO:0006953                        acute-phase response        42
## 4   GO:0006958    complement activation, classical pathway        28
## 5   GO:0070328                    triglyceride homeostasis        29
## 6   GO:0009820                  alkaloid metabolic process         5
## 7   GO:0002576                      platelet degranulation       120
## 8   GO:0043691               reverse cholesterol transport        16
## 9   GO:0034384 high-density lipoprotein particle cleara...        11
## 10  GO:0006641              triglyceride metabolic process        90
## 11  GO:0055089                      fatty acid homeostasis        12
## 12  GO:0046628 positive regulation of insulin receptor ...        13
## 13  GO:0006879               cellular iron ion homeostasis        57
## 14  GO:0006957 complement activation, alternative pathw...        14
## 15  GO:0034375 high-density lipoprotein particle remode...        15
## 16  GO:0008203               cholesterol metabolic process       121
## 17  GO:0090277 positive regulation of peptide hormone s...        79
## 18  GO:0006898               receptor-mediated endocytosis       264
## 19  GO:0008202                   steroid metabolic process       265
## 20  GO:0006768                    biotin metabolic process        10
## 21  GO:0034372 very-low-density lipoprotein particle re...        10
## 22  GO:0019835                                   cytolysis        35
## 23  GO:0007597        blood coagulation, intrinsic pathway        17
## 24  GO:1904179 positive regulation of adipose tissue de...         5
## 25  GO:0001907           killing by symbiont of host cells         5
## 26  GO:0006524                   alanine catabolic process         5
## 27  GO:0042738            exogenous drug catabolic process         5
## 28  GO:0001889                           liver development       124
## 29  GO:0007584                        response to nutrient       168
## 30  GO:0010951 negative regulation of endopeptidase act...       213
## 31  GO:0017187        peptidyl-glutamic acid carboxylation        11
## 32  GO:0050892                       intestinal absorption        34
## 33  GO:1904683 regulation of metalloendopeptidase activ...         6
## 34  GO:0015942                   formate metabolic process         6
## 35  GO:0070141                            response to UV-A         6
## 36  GO:0097267              omega-hydroxylase P450 pathway         6
## 37  GO:0006536                 glutamate metabolic process        30
## 38  GO:0061138     morphogenesis of a branching epithelium       168
## 39  GO:0045944 positive regulation of transcription fro...      1010
## 40  GO:0034374 low-density lipoprotein particle remodel...        13
## 41  GO:0043433 negative regulation of DNA binding trans...       130
## 42  GO:0070301      cellular response to hydrogen peroxide        74
## 43  GO:0048538                          thymus development        41
## 44  GO:0060059 embryonic retina morphogenesis in camera...         7
## 45  GO:0045620 negative regulation of lymphocyte differ...        40
## 46  GO:0045717 negative regulation of fatty acid biosyn...        14
## 47  GO:0002244 hematopoietic progenitor cell differenti...       155
## 48  GO:0045540 regulation of cholesterol biosynthetic p...        38
## 49  GO:0034380 high-density lipoprotein particle assemb...        15
## 50  GO:0045725 positive regulation of glycogen biosynth...        15
## 51  GO:0071276            cellular response to cadmium ion        24
## 52  GO:0046620                  regulation of organ growth        89
## 53  GO:1903427 negative regulation of reactive oxygen s...        23
## 54  GO:0010918 positive regulation of mitochondrial mem...         8
## 55  GO:0015886                              heme transport         8
## 56  GO:0051715                 cytolysis in other organism         8
## 57  GO:0006548                 histidine catabolic process         8
## 58  GO:0019216       regulation of lipid metabolic process       332
## 59  GO:0071559 response to transforming growth factor b...       203
## 60  GO:0019695                   choline metabolic process         9
## 61  GO:0010984 regulation of lipoprotein particle clear...        17
## 62  GO:0090218 positive regulation of lipid kinase acti...        31
## 63  GO:0010894 negative regulation of steroid biosynthe...        22
## 64  GO:0046329          negative regulation of JNK cascade        30
## 65  GO:0006739                      NADP metabolic process        33
## 66  GO:0016310                             phosphorylation      2076
## 67  GO:0000098         sulfur amino acid catabolic process         9
## 68  GO:0042908                        xenobiotic transport         9
## 69  GO:0070989                     oxidative demethylation         9
## 70  GO:0030917     midbrain-hindbrain boundary development         9
## 71  GO:0046185                  aldehyde catabolic process         9
## 72  GO:0070857 regulation of bile acid biosynthetic pro...         9
## 73  GO:0055091                    phospholipid homeostasis         9
## 74  GO:0033344                          cholesterol efflux        42
## 75  GO:0071157    negative regulation of cell cycle arrest        18
## 76  GO:0042593                         glucose homeostasis       200
## 77  GO:0006869                             lipid transport       311
## 78  GO:0051004 regulation of lipoprotein lipase activit...        18
## 79  GO:0050665      hydrogen peroxide biosynthetic process        10
## 80  GO:0051712 positive regulation of killing of cells ...        10
## 81  GO:0006853                           carnitine shuttle        10
## 82  GO:1902237 positive regulation of endoplasmic retic...        10
## 83  GO:0001558                   regulation of cell growth       356
## 84  GO:0050730 regulation of peptidyl-tyrosine phosphor...       210
## 85  GO:0042157               lipoprotein metabolic process       164
## 86  GO:0051081                nuclear envelope disassembly        43
## 87  GO:0019359 nicotinamide nucleotide biosynthetic pro...        27
## 88  GO:0034340               response to type I interferon        68
## 89  GO:0051647                        nucleus localization        22
## 90  GO:0001867       complement activation, lectin pathway        11
## 91  GO:0019373                    epoxygenase P450 pathway        11
## 92  GO:0071372 cellular response to follicle-stimulatin...        11
## 93  GO:0006559           L-phenylalanine catabolic process        11
## 94  GO:0045739           positive regulation of DNA repair        44
## 95  GO:0071560 cellular response to transforming growth...       200
## 96  GO:0006855                drug transmembrane transport        21
## 97  GO:0006368 transcription elongation from RNA polyme...        94
## 98  GO:0097421                          liver regeneration        32
## 99  GO:0030514 negative regulation of BMP signaling pat...        44
## 100 GO:0006629                     lipid metabolic process      1267
## 101 GO:0006520       cellular amino acid metabolic process       337
## 102 GO:0043903 regulation of symbiosis, encompassing mu...       197
## 103 GO:0042730                                fibrinolysis        25
## 104 GO:0045830    positive regulation of isotype switching        22
## 105 GO:0099133 ATP hydrolysis coupled anion transmembra...        12
## 106 GO:1900119 positive regulation of execution phase o...        12
## 107 GO:2001241 positive regulation of extrinsic apoptot...        12
## 108 GO:0045721      negative regulation of gluconeogenesis        12
## 109 GO:0006695            cholesterol biosynthetic process        60
## 110 GO:0021915                     neural tube development       153
## 111 GO:0010646            regulation of cell communication      3042
## 112 GO:0042177 negative regulation of protein catabolic...       102
## 113 GO:0033572                       transferrin transport        34
## 114 GO:0022900                    electron transport chain       143
## 115 GO:0042158            lipoprotein biosynthetic process       135
## 116 GO:2000646 positive regulation of receptor cataboli...         5
## 117 GO:0001887         selenium compound metabolic process         5
## 118 GO:0021894 cerebral cortex GABAergic interneuron de...         5
## 119 GO:1901523                 icosanoid catabolic process         5
## 120 GO:0021814 cell motility involved in cerebral corte...         5
## 121 GO:0032430 positive regulation of phospholipase A2 ...         5
## 122 GO:0010991 negative regulation of SMAD protein comp...         5
## 123 GO:0019740                        nitrogen utilization         5
## 124 GO:0060437                                 lung growth         5
## 125 GO:0048625                    myoblast fate commitment         5
## 126 GO:0019626    short-chain fatty acid catabolic process         5
## 127 GO:1903753      negative regulation of p38MAPK cascade         5
## 128 GO:0010040                    response to iron(II) ion         5
## 129 GO:0015911 plasma membrane long-chain fatty acid tr...         5
## 130 GO:0006572                  tyrosine catabolic process         5
## 131 GO:0090324 negative regulation of oxidative phospho...         5
## 132 GO:0031999 negative regulation of fatty acid beta-o...         5
## 133 GO:0090118 receptor-mediated endocytosis involved i...         5
## 134 GO:0072049             comma-shaped body morphogenesis         5
## 135 GO:0010269                    response to selenium ion         5
## 136 GO:0051642                     centrosome localization        23
## 137 GO:0051147 regulation of muscle cell differentiatio...       179
## 138 GO:0034142      toll-like receptor 4 signaling pathway        28
## 139 GO:1900016 negative regulation of cytokine producti...        13
## 140 GO:0071397            cellular response to cholesterol        13
## 141 GO:0016540                      protein autoprocessing        13
## 142 GO:0060712        spongiotrophoblast layer development        13
## 143 GO:0032436 positive regulation of proteasomal ubiqu...        75
## 144 GO:0055114                 oxidation-reduction process       852
## 145 GO:0034504             protein localization to nucleus       341
## 146 GO:0006699              bile acid biosynthetic process        33
## 147 GO:0046718                  viral entry into host cell       102
## 148 GO:2000463 positive regulation of excitatory postsy...        24
## 149 GO:0007411                               axon guidance       219
## 150 GO:0042127            regulation of cell proliferation      1447
## 151 GO:0031325 positive regulation of cellular metaboli...      2748
## 152 GO:0002922 positive regulation of humoral immune re...        14
## 153 GO:0034356 NAD biosynthesis via nicotinamide ribosi...        14
## 154 GO:0042104 positive regulation of activated T cell ...        25
## 155 GO:0032270 positive regulation of cellular protein ...      1352
## 156 GO:0090335 regulation of brown fat cell differentia...        13
## 157 GO:0034763 negative regulation of transmembrane tra...        85
## 158 GO:0071156             regulation of cell cycle arrest        98
## 159 GO:0032375 negative regulation of cholesterol trans...        14
## 160 GO:2000659 regulation of interleukin-1-mediated sig...         6
## 161 GO:1990535               neuron projection maintenance         6
## 162 GO:1902946      protein localization to early endosome         6
## 163 GO:0042940                      D-amino acid transport         6
## 164 GO:0033860      regulation of NAD(P)H oxidase activity         6
## 165 GO:1900107       regulation of nodal signaling pathway         6
## 166 GO:0046498    S-adenosylhomocysteine metabolic process         6
## 167 GO:1905049 negative regulation of metallopeptidase ...         6
## 168 GO:1903896 positive regulation of IRE1-mediated unf...         6
## 169 GO:1902459 positive regulation of stem cell populat...         6
## 170 GO:0042426                   choline catabolic process         6
## 171 GO:0031659 positive regulation of cyclin-dependent ...         6
## 172 GO:0034382               chylomicron remnant clearance         6
## 173 GO:2000343 positive regulation of chemokine (C-X-C ...         6
## 174 GO:0006545                glycine biosynthetic process         6
## 175 GO:0010793      regulation of mRNA export from nucleus         6
## 176 GO:1905668 positive regulation of protein localizat...         6
## 177 GO:1903364 positive regulation of cellular protein ...       186
## 178 GO:0021879            forebrain neuron differentiation        49
## 179 GO:0019432           triglyceride biosynthetic process        31
## 180 GO:0006461                    protein complex assembly      1257
## 181 GO:0021542                   dentate gyrus development        15
## 182 GO:0090201 negative regulation of release of cytoch...        15
## 183 GO:0000060 protein import into nucleus, translocati...        45
## 184 GO:0046326       positive regulation of glucose import        39
## 185 GO:0035774 positive regulation of insulin secretion...        26
## 186 GO:0007628                      adult walking behavior        26
## 187 GO:0032088 negative regulation of NF-kappaB transcr...        66
## 188 GO:0008152                           metabolic process      9852
## 189 GO:0048013           ephrin receptor signaling pathway        81
## 190 GO:0043392          negative regulation of DNA binding        47
## 191 GO:0009887                  animal organ morphogenesis       919
## 192 GO:0032094                            response to food        34
## 193 GO:0021756                        striatum development        16
## 194 GO:0060644 mammary gland epithelial cell differenti...        16
## 195 GO:1903830       magnesium ion transmembrane transport        16
## 196 GO:0007595                                   lactation        40
## 197 GO:0000122 negative regulation of transcription fro...       727
## 198 GO:0097190                 apoptotic signaling pathway       543
## 199 GO:0031663 lipopolysaccharide-mediated signaling pa...        48
## 200 GO:0030178 negative regulation of Wnt signaling pat...       186
## 201 GO:0030195    negative regulation of blood coagulation        46
## 202 GO:0019674                       NAD metabolic process        64
## 203 GO:1902373 negative regulation of mRNA catabolic pr...        40
## 204 GO:0032786 positive regulation of DNA-templated tra...        20
## 205 GO:0033598 mammary gland epithelial cell proliferat...        22
## 206 GO:0034638       phosphatidylcholine catabolic process         7
## 207 GO:0010898 positive regulation of triglyceride cata...         7
## 208 GO:0070544                histone H3-K36 demethylation         7
## 209 GO:0010886 positive regulation of cholesterol stora...         7
## 210 GO:0060052     neurofilament cytoskeleton organization         7
## 211 GO:0021830 interneuron migration from the subpalliu...         7
## 212 GO:1901679          nucleotide transmembrane transport         7
## 213 GO:0000042                  protein targeting to Golgi         7
## 214 GO:0051045 negative regulation of membrane protein ...         7
## 215 GO:1904468 negative regulation of tumor necrosis fa...         7
## 216 GO:0044332 Wnt signaling pathway involved in dorsal...         7
## 217 GO:0035356           cellular triglyceride homeostasis         7
## 218 GO:0032287 peripheral nervous system myelin mainten...         7
## 219 GO:0060463                     lung lobe morphogenesis         7
## 220 GO:0046439                L-cysteine metabolic process         7
## 221 GO:1903874        ferrous iron transmembrane transport         7
## 222 GO:0031666 positive regulation of lipopolysaccharid...         7
## 223 GO:0042167                      heme catabolic process         7
## 224 GO:2000192 negative regulation of fatty acid transp...         7
## 225 GO:0090336 positive regulation of brown fat cell di...         7
## 226 GO:0042758     long-chain fatty acid catabolic process         7
## 227 GO:0030300 regulation of intestinal cholesterol abs...         7
## 228 GO:0006600                  creatine metabolic process         7
## 229 GO:2000271 positive regulation of fibroblast apopto...         7
## 230 GO:0008090                 retrograde axonal transport         7
## 231 GO:0002679 respiratory burst involved in defense re...         7
## 232 GO:0032071 regulation of endodeoxyribonuclease acti...         7
## 233 GO:0043627                        response to estrogen        66
## 234 GO:0051346 negative regulation of hydrolase activit...       362
## 235 GO:0061045        negative regulation of wound healing        63
## 236 GO:0017144                      drug metabolic process        24
## 237 GO:0007041                         lysosomal transport        86
## 238 GO:0001938 positive regulation of endothelial cell ...        70
## 239 GO:0051000 positive regulation of nitric-oxide synt...        17
## 240 GO:1902176 negative regulation of oxidative stress-...        17
## 241 GO:0031100                   animal organ regeneration        74
## 242 GO:0032869       cellular response to insulin stimulus       176
## 243 GO:0032007        negative regulation of TOR signaling        40
## 244 GO:0018279 protein N-linked glycosylation via aspar...        42
## 245 GO:0045599 negative regulation of fat cell differen...        42
## 246 GO:0032148     activation of protein kinase B activity        29
## 247 GO:0001503                                ossification       343
## 248 GO:1902680 positive regulation of RNA biosynthetic ...      1299
## 249 GO:0030324                            lung development       164
## 250 GO:0060021                          palate development        87
## 251 GO:2000257    regulation of protein activation cascade        42
## 252 GO:0042493                            response to drug       373
## 253 GO:0001568                    blood vessel development       570
## 254 GO:0031016                        pancreas development        67
## 255 GO:0006778 porphyrin-containing compound metabolic ...        34
## 256 GO:1900076 regulation of cellular response to insul...        60
## 257 GO:0090239        regulation of histone H4 acetylation        10
## 258 GO:0019627                      urea metabolic process        11
## 259 GO:0015865                 purine nucleotide transport        12
## 260 GO:0061620 glycolytic process through glucose-6-pho...        24
## 261 GO:0050820          positive regulation of coagulation        24
## 262 GO:0006476                       protein deacetylation        92
## 263 GO:0046631                alpha-beta T cell activation       114
## 264 GO:0098902 regulation of membrane depolarization du...         6
## 265 GO:1902959 regulation of aspartic-type endopeptidas...         6
## 266 GO:0052433 modulation by organism of apoptotic proc...         6
## 267 GO:0070943 neutrophil mediated killing of symbiont ...         6
## 268 GO:0046884      follicle-stimulating hormone secretion         6
## 269 GO:0052040 modulation by symbiont of host programme...         6
## 270 GO:0051340               regulation of ligase activity         7
## 271 GO:0035733            hepatic stellate cell activation         7
## 272 GO:0046838 phosphorylated carbohydrate dephosphoryl...        10
## 273 GO:1901160    primary amino compound metabolic process        13
## 274 GO:0036475 neuron death in response to oxidative st...        21
## 275 GO:0042773    ATP synthesis coupled electron transport        66
## 276 GO:0010039                        response to iron ion        33
## 277 GO:0006505                GPI anchor metabolic process        28
## 278 GO:1900182 positive regulation of protein localizat...       124
## 279 GO:0052652 cyclic purine nucleotide metabolic proce...       145
## 280 GO:0035909                         aorta morphogenesis        30
## 281 GO:0014911 positive regulation of smooth muscle cel...        30
## 282 GO:0035435       phosphate ion transmembrane transport        18
## 283 GO:0006465                   signal peptide processing        18
## 284 GO:0061003 positive regulation of dendritic spine m...        18
## 285 GO:0035264               multicellular organism growth       137
## 286 GO:0014068 positive regulation of phosphatidylinosi...        58
## 287 GO:0043405           regulation of MAP kinase activity       303
## 288 GO:0008595 anterior/posterior axis specification, e...        13
## 289 GO:0050746 regulation of lipoprotein metabolic proc...        13
## 290 GO:0045657 positive regulation of monocyte differen...         8
## 291 GO:0010873 positive regulation of cholesterol ester...         8
## 292 GO:0001561                  fatty acid alpha-oxidation         8
## 293 GO:0071073 positive regulation of phospholipid bios...         8
## 294 GO:1900364 negative regulation of mRNA polyadenylat...         8
## 295 GO:1900112 regulation of histone H3-K9 trimethylati...         8
## 296 GO:0008635 activation of cysteine-type endopeptidas...         8
## 297 GO:0007262            STAT protein import into nucleus         8
## 298 GO:0006390 transcription from mitochondrial promote...         8
## 299 GO:0015866                               ADP transport         8
## 300 GO:0046487                glyoxylate metabolic process         8
## 301 GO:0051573 negative regulation of histone H3-K9 met...         8
## 302 GO:0032926 negative regulation of activin receptor ...         8
## 303 GO:0032933                     SREBP signaling pathway         8
## 304 GO:0006591                 ornithine metabolic process         8
## 305 GO:1990314 cellular response to insulin-like growth...         8
## 306 GO:0031937 positive regulation of chromatin silenci...         8
## 307 GO:0090400         stress-induced premature senescence         8
## 308 GO:0051897 positive regulation of protein kinase B ...       137
##     Significant Expected weightFisher
## 1            13     1.87      1.8e-08
## 2            15     3.10      1.4e-07
## 3            11     1.91      6.7e-07
## 4             9     1.28      2.5e-06
## 5            10     1.32      3.2e-06
## 6             4     0.23      2.1e-05
## 7            17     5.47      3.0e-05
## 8             6     0.73      4.7e-05
## 9             5     0.50      7.1e-05
## 10           16     4.10      8.7e-05
## 11            5     0.55      0.00012
## 12            5     0.59      0.00018
## 13           10     2.60      0.00022
## 14            5     0.64      0.00027
## 15            5     0.68      0.00040
## 16           21     5.51      0.00041
## 17           12     3.60      0.00044
## 18           25    12.03      0.00044
## 19           41    12.08      0.00047
## 20            4     0.46      0.00072
## 21            4     0.46      0.00072
## 22            9     1.59      0.00074
## 23            5     0.77      0.00076
## 24            3     0.23      0.00088
## 25            3     0.23      0.00088
## 26            3     0.23      0.00088
## 27            3     0.23      0.00088
## 28           17     5.65      0.00091
## 29           21     7.66      0.00100
## 30           18     9.71      0.00103
## 31            4     0.50      0.00109
## 32            8     1.55      0.00155
## 33            3     0.27      0.00170
## 34            3     0.27      0.00170
## 35            3     0.27      0.00170
## 36            3     0.27      0.00170
## 37            6     1.37      0.00204
## 38            9     7.66      0.00208
## 39           67    46.03      0.00213
## 40            4     0.59      0.00220
## 41           16     5.92      0.00225
## 42            9     3.37      0.00227
## 43            7     1.87      0.00230
## 44            3     0.32      0.00287
## 45            4     1.82      0.00288
## 46            4     0.64      0.00297
## 47           12     7.06      0.00328
## 48            7     1.73      0.00336
## 49            4     0.68      0.00390
## 50            4     0.68      0.00390
## 51            5     1.09      0.00400
## 52            8     4.06      0.00442
## 53            4     1.05      0.00444
## 54            3     0.36      0.00444
## 55            3     0.36      0.00444
## 56            3     0.36      0.00444
## 57            3     0.36      0.00444
## 58           39    15.13      0.00523
## 59           15     9.25      0.00596
## 60            4     0.41      0.00600
## 61            4     0.77      0.00601
## 62            3     1.41      0.00604
## 63            5     1.00      0.00630
## 64            5     1.37      0.00631
## 65            5     1.50      0.00632
## 66          116    94.60      0.00636
## 67            3     0.41      0.00644
## 68            3     0.41      0.00644
## 69            3     0.41      0.00644
## 70            3     0.41      0.00644
## 71            3     0.41      0.00644
## 72            3     0.41      0.00644
## 73            3     0.41      0.00644
## 74            8     1.91      0.00772
## 75            4     0.82      0.00784
## 76           20     9.11      0.00840
## 77           38    14.17      0.00841
## 78            4     0.82      0.00887
## 79            3     0.46      0.00889
## 80            3     0.46      0.00889
## 81            3     0.46      0.00889
## 82            3     0.46      0.00889
## 83           28    16.22      0.00937
## 84           18     9.57      0.01152
## 85           14     7.47      0.01162
## 86            6     1.96      0.01162
## 87            5     1.23      0.01163
## 88            6     3.10      0.01166
## 89            3     1.00      0.01169
## 90            3     0.50      0.01181
## 91            3     0.50      0.01181
## 92            3     0.50      0.01181
## 93            3     0.50      0.01181
## 94            6     2.01      0.01226
## 95           13     9.11      0.01269
## 96            4     0.96      0.01377
## 97           10     4.28      0.01386
## 98            5     1.46      0.01404
## 99            6     2.01      0.01413
## 100          99    57.74      0.01441
## 101          36    15.36      0.01480
## 102          17     8.98      0.01488
## 103           5     1.14      0.01514
## 104           4     1.00      0.01519
## 105           3     0.55      0.01522
## 106           3     0.55      0.01522
## 107           3     0.55      0.01522
## 108           3     0.55      0.01522
## 109          11     2.73      0.01581
## 110          13     6.97      0.01599
## 111         165   138.63      0.01725
## 112          11     4.65      0.01770
## 113           5     1.55      0.01801
## 114          12     6.52      0.01863
## 115           8     6.15      0.01891
## 116           2     0.23      0.01892
## 117           2     0.23      0.01892
## 118           2     0.23      0.01892
## 119           2     0.23      0.01892
## 120           2     0.23      0.01892
## 121           2     0.23      0.01892
## 122           2     0.23      0.01892
## 123           2     0.23      0.01892
## 124           2     0.23      0.01892
## 125           2     0.23      0.01892
## 126           2     0.23      0.01892
## 127           2     0.23      0.01892
## 128           2     0.23      0.01892
## 129           2     0.23      0.01892
## 130           2     0.23      0.01892
## 131           2     0.23      0.01892
## 132           2     0.23      0.01892
## 133           2     0.23      0.01892
## 134           2     0.23      0.01892
## 135           2     0.23      0.01892
## 136           4     1.05      0.01896
## 137           9     8.16      0.01897
## 138           4     1.28      0.01911
## 139           3     0.59      0.01913
## 140           3     0.59      0.01913
## 141           3     0.59      0.01913
## 142           3     0.59      0.01913
## 143           8     3.42      0.02051
## 144          59    38.83      0.02063
## 145          27    15.54      0.02102
## 146           7     1.50      0.02168
## 147          11     4.65      0.02172
## 148           4     1.09      0.02196
## 149          17     9.98      0.02232
## 150          79    65.94      0.02293
## 151         172   125.23      0.02332
## 152           3     0.64      0.02354
## 153           3     0.64      0.02354
## 154           4     1.14      0.02522
## 155          81    61.61      0.02602
## 156           4     0.59      0.02739
## 157           7     3.87      0.02741
## 158           7     4.47      0.02746
## 159           3     0.64      0.02747
## 160           2     0.27      0.02752
## 161           2     0.27      0.02752
## 162           2     0.27      0.02752
## 163           2     0.27      0.02752
## 164           2     0.27      0.02752
## 165           2     0.27      0.02752
## 166           2     0.27      0.02752
## 167           2     0.27      0.02752
## 168           2     0.27      0.02752
## 169           2     0.27      0.02752
## 170           2     0.27      0.02752
## 171           2     0.27      0.02752
## 172           2     0.27      0.02752
## 173           2     0.27      0.02752
## 174           2     0.27      0.02752
## 175           2     0.27      0.02752
## 176           2     0.27      0.02752
## 177          13     8.48      0.02820
## 178           6     2.23      0.02829
## 179           5     1.41      0.02831
## 180          73    57.28      0.02845
## 181           3     0.68      0.02845
## 182           3     0.68      0.02845
## 183           6     2.05      0.02861
## 184           5     1.78      0.02871
## 185           4     1.18      0.02877
## 186           4     1.18      0.02877
## 187           7     3.01      0.03008
## 188         492   448.96      0.03079
## 189           8     3.69      0.03092
## 190           6     2.14      0.03242
## 191          51    41.88      0.03332
## 192           4     1.55      0.03384
## 193           3     0.73      0.03386
## 194           3     0.73      0.03386
## 195           3     0.73      0.03386
## 196           5     1.82      0.03412
## 197          45    33.13      0.03431
## 198          39    24.74      0.03565
## 199           7     2.19      0.03633
## 200          11     8.48      0.03673
## 201           7     2.10      0.03705
## 202           7     2.92      0.03713
## 203           4     1.82      0.03733
## 204           3     0.91      0.03734
## 205           3     1.00      0.03735
## 206           2     0.32      0.03738
## 207           2     0.32      0.03738
## 208           2     0.32      0.03738
## 209           2     0.32      0.03738
## 210           2     0.32      0.03738
## 211           2     0.32      0.03738
## 212           2     0.32      0.03738
## 213           2     0.32      0.03738
## 214           2     0.32      0.03738
## 215           2     0.32      0.03738
## 216           2     0.32      0.03738
## 217           2     0.32      0.03738
## 218           2     0.32      0.03738
## 219           2     0.32      0.03738
## 220           2     0.32      0.03738
## 221           2     0.32      0.03738
## 222           2     0.32      0.03738
## 223           2     0.32      0.03738
## 224           2     0.32      0.03738
## 225           2     0.32      0.03738
## 226           2     0.32      0.03738
## 227           2     0.32      0.03738
## 228           2     0.32      0.03738
## 229           2     0.32      0.03738
## 230           2     0.32      0.03738
## 231           2     0.32      0.03738
## 232           2     0.32      0.03738
## 233           8     3.01      0.03791
## 234          26    16.50      0.03864
## 235          10     2.87      0.03902
## 236           7     1.09      0.03920
## 237           8     3.92      0.03948
## 238           7     3.19      0.03971
## 239           3     0.77      0.03976
## 240           3     0.77      0.03976
## 241          10     3.37      0.04019
## 242          15     8.02      0.04037
## 243           6     1.82      0.04080
## 244           5     1.91      0.04099
## 245           5     1.91      0.04099
## 246           4     1.32      0.04109
## 247          19    15.63      0.04129
## 248          85    59.20      0.04369
## 249          12     7.47      0.04431
## 250           8     3.96      0.04448
## 251          14     1.91      0.04480
## 252          32    17.00      0.04486
## 253          31    25.98      0.04523
## 254           8     3.05      0.04527
## 255           6     1.55      0.04533
## 256           6     2.73      0.04541
## 257           3     0.46      0.04546
## 258           3     0.50      0.04546
## 259           3     0.55      0.04547
## 260           3     1.09      0.04550
## 261           3     1.09      0.04550
## 262           6     4.19      0.04551
## 263           7     5.20      0.04551
## 264           2     0.27      0.04552
## 265           2     0.27      0.04552
## 266           2     0.27      0.04552
## 267           2     0.27      0.04552
## 268           2     0.27      0.04552
## 269           2     0.27      0.04552
## 270           2     0.32      0.04552
## 271           2     0.32      0.04552
## 272           2     0.46      0.04553
## 273           2     0.59      0.04554
## 274           2     0.96      0.04556
## 275           4     3.01      0.04557
## 276           7     1.50      0.04557
## 277           2     1.28      0.04559
## 278           9     5.65      0.04561
## 279           6     6.61      0.04568
## 280           4     1.37      0.04576
## 281           4     1.37      0.04607
## 282           3     0.82      0.04615
## 283           3     0.82      0.04615
## 284           3     0.82      0.04615
## 285          13     6.24      0.04712
## 286           6     2.64      0.04773
## 287          19    13.81      0.04788
## 288           3     0.59      0.04826
## 289           3     0.59      0.04826
## 290           2     0.36      0.04837
## 291           2     0.36      0.04837
## 292           2     0.36      0.04837
## 293           2     0.36      0.04837
## 294           2     0.36      0.04837
## 295           2     0.36      0.04837
## 296           2     0.36      0.04837
## 297           2     0.36      0.04837
## 298           2     0.36      0.04837
## 299           2     0.36      0.04837
## 300           2     0.36      0.04837
## 301           2     0.36      0.04837
## 302           2     0.36      0.04837
## 303           2     0.36      0.04837
## 304           2     0.36      0.04837
## 305           2     0.36      0.04837
## 306           2     0.36      0.04837
## 307           2     0.36      0.04837
## 308          11     6.24      0.04848</code></pre>
<pre class="r"><code> go_table_liver &lt;- as.data.frame(cbind(go_table$GO.ID, go_table$weightFisher))

write.table(go_table_liver, &quot;../data/go_table_liver.txt&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)</code></pre>
</div>
<div id="lung-1" class="section level2">
<h2>Lung</h2>
<pre class="r"><code># Find the heart hypo
heart_hypo &lt;- human_chimp_heart_specific_25_conserved_H3K27_LUNG[which(human_chimp_heart_specific_25_conserved_H3K27_LUNG$V4 == &quot;hyper&quot; &amp; human_chimp_heart_specific_25_conserved_H3K27_LUNG$V6 != &quot;-1&quot;),]

heart_hypo &lt;- heart_hypo[complete.cases(heart_hypo), ]

# Find the closest gene

closest_heart &lt;- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = &quot;closest&quot;, check.chr = FALSE)</code></pre>
<pre><code>##  * Processing input (1): a
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... PASS
##  * Checking for overlapping &#39;contiguous&#39; regions... PASS
##  * Processing input (2): b
## CONVERT TO BED
##  * Checking input type... PASS
##    Input seems to be in bed format but chr/start/end column names are missing
## VALIDATE REGIONS
##  * Check if index is a string... PASS
##  * Check index pattern... PASS
##  * Check for missing values... PASS
##  * Check for larger start position... PASS.
##  * Check if zero based... PASS
##  * Checking sort order... FAIL
##    The input for object is not *lexographically* ordered!
##    This can cause unexpected results for some set operations.
##    try: x &lt;- bedr.sort.region(x)
##  * Checking for overlapping &#39;contiguous&#39; regions... FAIL
##    The input for object has overlapping features!
##    This can cause unexpected results for some set operations.
##    i.e. x &lt;- bedr.merge.region(x)
##    bedtools closest -a /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/a_bc552850ec24.bed -b /var/folders/rf/qrcw6ncj05z1pc_pq9xzw3540000gn/T//RtmpXK3I56/b_bc557772e5c1.bed</code></pre>
<pre class="r"><code># Convert the gene name to ensg

gene_id &lt;- read.table(&quot;../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv&quot;, stringsAsFactors = FALSE, header=TRUE, sep = &quot;,&quot;)

comb_kidney &lt;- merge(closest_heart, gene_id, by.x = c(&quot;V8&quot;), by.y = c(&quot;Gene&quot;))

comb_kidney$ensg</code></pre>
<pre><code>##  [1] &quot;ENSG00000177674&quot; &quot;ENSG00000177674&quot; &quot;ENSG00000177674&quot;
##  [4] &quot;ENSG00000177674&quot; &quot;ENSG00000177674&quot; &quot;ENSG00000130762&quot;
##  [7] &quot;ENSG00000117411&quot; &quot;ENSG00000026508&quot; &quot;ENSG00000026508&quot;
## [10] &quot;ENSG00000026508&quot; &quot;ENSG00000026508&quot; &quot;ENSG00000026508&quot;
## [13] &quot;ENSG00000026508&quot; &quot;ENSG00000026508&quot; &quot;ENSG00000026508&quot;
## [16] &quot;ENSG00000114861&quot; &quot;ENSG00000114315&quot; &quot;ENSG00000131149&quot;
## [19] &quot;ENSG00000198951&quot; &quot;ENSG00000196498&quot; &quot;ENSG00000196498&quot;
## [22] &quot;ENSG00000196498&quot; &quot;ENSG00000170322&quot; &quot;ENSG00000100100&quot;
## [25] &quot;ENSG00000100100&quot; &quot;ENSG00000185917&quot; &quot;ENSG00000185917&quot;
## [28] &quot;ENSG00000167323&quot; &quot;ENSG00000143643&quot; &quot;ENSG00000143643&quot;
## [31] &quot;ENSG00000153786&quot; &quot;ENSG00000153786&quot; &quot;ENSG00000171940&quot;
## [34] &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot;
## [37] &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot; &quot;ENSG00000183621&quot;
## [40] &quot;ENSG00000183621&quot;</code></pre>
<pre class="r"><code># Convert the TSS to ensg as well 

ref_gene_hg19_ensg &lt;- merge(refGene_hg19_TSS, gene_id, by.x = c(&quot;V5&quot;), by.y = c(&quot;Gene&quot;))

all_ref_gene_hg19_ensg &lt;- unique(ref_gene_hg19_ensg$ensg)

heart_ref_gene &lt;- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene &lt;- as.numeric(as.vector(heart_ref_gene))
names(test_gene) &lt;-  all_ref_gene_hg19_ensg

# Run topGO
go_data &lt;- new(&quot;topGOdata&quot;,
                   ontology = &quot;BP&quot;,
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore &gt; 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = &quot;org.Hs.eg.db&quot;,
                   ID = &quot;ensembl&quot;)</code></pre>
<pre><code>## 
## Building most specific GOs .....</code></pre>
<pre><code>##  ( 11450 GO terms found. )</code></pre>
<pre><code>## 
## Build GO DAG topology ..........</code></pre>
<pre><code>##  ( 15456 GO terms and 36153 relations. )</code></pre>
<pre><code>## 
## Annotating nodes ...............</code></pre>
<pre><code>##  ( 14505 genes annotated to the GO terms. )</code></pre>
<pre class="r"><code># Perform enrichment test
go_test &lt;- runTest(go_data, algorithm = &quot;weight01&quot;, statistic = &quot;fisher&quot;)</code></pre>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 1110 nontrivial nodes
##       parameters: 
##           test statistic: fisher</code></pre>
<pre><code>## 
##   Level 15:  3 nodes to be scored    (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 14:  5 nodes to be scored    (0 eliminated genes)</code></pre>
<pre><code>## 
##   Level 13:  23 nodes to be scored   (86 eliminated genes)</code></pre>
<pre><code>## 
##   Level 12:  44 nodes to be scored   (192 eliminated genes)</code></pre>
<pre><code>## 
##   Level 11:  76 nodes to be scored   (1997 eliminated genes)</code></pre>
<pre><code>## 
##   Level 10:  103 nodes to be scored  (3134 eliminated genes)</code></pre>
<pre><code>## 
##   Level 9:   112 nodes to be scored  (5198 eliminated genes)</code></pre>
<pre><code>## 
##   Level 8:   124 nodes to be scored  (6736 eliminated genes)</code></pre>
<pre><code>## 
##   Level 7:   155 nodes to be scored  (7719 eliminated genes)</code></pre>
<pre><code>## 
##   Level 6:   166 nodes to be scored  (10145 eliminated genes)</code></pre>
<pre><code>## 
##   Level 5:   144 nodes to be scored  (11508 eliminated genes)</code></pre>
<pre><code>## 
##   Level 4:   90 nodes to be scored   (13129 eliminated genes)</code></pre>
<pre><code>## 
##   Level 3:   48 nodes to be scored   (13788 eliminated genes)</code></pre>
<pre><code>## 
##   Level 2:   16 nodes to be scored   (14144 eliminated genes)</code></pre>
<pre><code>## 
##   Level 1:   1 nodes to be scored    (14277 eliminated genes)</code></pre>
<pre class="r"><code>go_table &lt;- GenTable(go_data, weightFisher = go_test,
                         orderBy = &quot;weightFisher&quot;, ranksOf = &quot;weightFisher&quot;,
                         topNodes = sum(score(go_test) &lt; .01))

go_table</code></pre>
<pre><code>##         GO.ID                                        Term Annotated
## 1  GO:0035019    somatic stem cell population maintenance        61
## 2  GO:0061009                common bile duct development         5
## 3  GO:0032237 activation of store-operated calcium cha...         5
## 4  GO:0072049             comma-shaped body morphogenesis         5
## 5  GO:0003310           pancreatic A cell differentiation         5
## 6  GO:0009912      auditory receptor cell fate commitment         6
## 7  GO:1901533 negative regulation of hematopoietic pro...         6
## 8  GO:2001182      regulation of interleukin-12 secretion         6
## 9  GO:2000973    regulation of pro-B cell differentiation         6
## 10 GO:0021861 forebrain radial glial cell differentiat...         6
## 11 GO:0018230        peptidyl-L-cysteine S-palmitoylation         6
## 12 GO:1903039 positive regulation of leukocyte cell-ce...       195
## 13 GO:0046477          glycosylceramide catabolic process         7
## 14 GO:1903799 negative regulation of production of miR...         7
## 15 GO:0061101         neuroendocrine cell differentiation         7
## 16 GO:0072050                 S-shaped body morphogenesis         7
## 17 GO:0061626        pharyngeal arch artery morphogenesis         7
## 18 GO:0042117                         monocyte activation         7
## 19 GO:0048664                   neuron fate determination         7
## 20 GO:0045607 regulation of auditory receptor cell dif...         8
## 21 GO:0072282    metanephric nephron tubule morphogenesis         8
## 22 GO:0097084     vascular smooth muscle cell development         8
## 23 GO:0007262            STAT protein import into nucleus         8
## 24 GO:0016139                 glycoside catabolic process         8
## 25 GO:1903726 negative regulation of phospholipid meta...         8
## 26 GO:0061309 cardiac neural crest cell development in...         9
## 27 GO:0018026             peptidyl-lysine monomethylation         9
## 28 GO:0003266 regulation of secondary heart field card...         9
## 29 GO:0030917     midbrain-hindbrain boundary development         9
## 30 GO:0070486                       leukocyte aggregation         9
## 31 GO:0072567 chemokine (C-X-C motif) ligand 2 product...         9
##    Significant Expected weightFisher
## 1            2     0.06       0.0018
## 2            1     0.01       0.0052
## 3            1     0.01       0.0052
## 4            1     0.01       0.0052
## 5            1     0.01       0.0052
## 6            1     0.01       0.0062
## 7            1     0.01       0.0062
## 8            1     0.01       0.0062
## 9            1     0.01       0.0062
## 10           1     0.01       0.0062
## 11           1     0.01       0.0062
## 12           2     0.20       0.0068
## 13           1     0.01       0.0072
## 14           1     0.01       0.0072
## 15           1     0.01       0.0072
## 16           1     0.01       0.0072
## 17           1     0.01       0.0072
## 18           1     0.01       0.0072
## 19           1     0.01       0.0072
## 20           1     0.01       0.0082
## 21           1     0.01       0.0082
## 22           1     0.01       0.0082
## 23           1     0.01       0.0082
## 24           1     0.01       0.0082
## 25           1     0.01       0.0082
## 26           1     0.01       0.0093
## 27           1     0.01       0.0093
## 28           1     0.01       0.0093
## 29           1     0.01       0.0093
## 30           1     0.01       0.0093
## 31           1     0.01       0.0093</code></pre>
</div>
</div>
<div id="run-clusterprofiler" class="section level1">
<h1>Run clusterProfiler</h1>
<pre class="r"><code>#head(heart_ref_gene)
#length(heart_ref_gene)
#head(liver_ref_gene)
#length(liver_ref_gene)

#mydf &lt;- as.data.frame(cbind(all_ref_gene_hg19_ensg, heart_ref_gene, liver_ref_gene))

#colnames(mydf) &lt;- c(&quot;ensg&quot;, &quot;heart&quot;, &quot;liver&quot;)

#xx.formula.twogroups &lt;- compareCluster(mydf$heart, data = mydf, fun=&quot;enrichGO&quot;, universe = df$ensg,
#                OrgDb         = org.Hs.eg.db)

#dotplot(xx.formula.twogroups, showCategory=5, includeAll=FALSE)</code></pre>
<pre class="r"><code>## Reivigo

# If you don&#39;t have the ggplot2 package installed, uncomment the following line:
# install.packages( &quot;ggplot2&quot; );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don&#39;t have the scales package installed, uncomment the following line:
# install.packages( &quot;scales&quot; );
library( scales );</code></pre>
<pre><code>## Warning: package &#39;scales&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>library(ggrepel)

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names &lt;- c(&quot;term_ID&quot;,&quot;description&quot;,&quot;frequency_%&quot;,&quot;plot_X&quot;,&quot;plot_Y&quot;,&quot;plot_size&quot;,&quot;value&quot;,&quot;uniqueness&quot;,&quot;dispensability&quot;);
revigo.data &lt;- rbind(c(&quot;GO:0009113&quot;,&quot;purine nucleobase biosynthetic process&quot;, 0.124,-4.742,-3.192, 4.202, 0.0005,0.769,0.000),
c(&quot;GO:0033197&quot;,&quot;response to vitamin E&quot;, 0.000, 7.179,-3.273, 1.740, 0.0005,0.849,0.000),
c(&quot;GO:0019043&quot;,&quot;establishment of viral latency&quot;, 0.001,-5.135, 5.016, 1.959, 0.0187,0.945,0.018),
c(&quot;GO:0006501&quot;,&quot;C-terminal protein lipidation&quot;, 0.004, 0.003,-5.207, 2.693, 0.0120,0.913,0.100),
c(&quot;GO:0060509&quot;,&quot;Type I pneumocyte differentiation&quot;, 0.001,-0.420, 6.928, 2.121, 0.0134,0.698,0.103),
c(&quot;GO:1903078&quot;,&quot;positive regulation of protein localization to plasma membrane&quot;, 0.010, 1.621, 0.720, 3.123, 0.0068,0.688,0.118),
c(&quot;GO:0033632&quot;,&quot;regulation of cell-cell adhesion mediated by integrin&quot;, 0.002, 2.909, 0.449, 2.378, 0.0160,0.835,0.132),
c(&quot;GO:2000109&quot;,&quot;regulation of macrophage apoptotic process&quot;, 0.001, 2.079,-2.113, 2.188, 0.0187,0.768,0.132),
c(&quot;GO:0032007&quot;,&quot;negative regulation of TOR signaling&quot;, 0.016, 5.974, 0.117, 3.325, 0.0052,0.640,0.175),
c(&quot;GO:0043547&quot;,&quot;positive regulation of GTPase activity&quot;, 0.470, 4.687, 1.488, 4.780, 0.0054,0.838,0.179),
c(&quot;GO:0009256&quot;,&quot;10-formyltetrahydrofolate metabolic process&quot;, 0.002,-6.344,-1.530, 2.474, 0.0134,0.809,0.185),
c(&quot;GO:0070141&quot;,&quot;response to UV-A&quot;, 0.002, 6.586,-3.433, 2.369, 0.0160,0.883,0.188),
c(&quot;GO:0071499&quot;,&quot;cellular response to laminar fluid shear stress&quot;, 0.001, 6.586,-2.544, 2.143, 0.0187,0.852,0.211),
c(&quot;GO:0006139&quot;,&quot;nucleobase-containing compound metabolic process&quot;,26.547,-4.434,-2.184, 6.532, 0.0499,0.889,0.214),
c(&quot;GO:0018242&quot;,&quot;protein O-linked glycosylation via serine&quot;, 0.001,-1.029,-5.988, 2.117, 0.0160,0.818,0.218),
c(&quot;GO:0015942&quot;,&quot;formate metabolic process&quot;, 0.065,-5.564,-1.813, 3.921, 0.0160,0.815,0.246),
c(&quot;GO:0070601&quot;,&quot;centromeric sister chromatid cohesion&quot;, 0.004,-1.575,-1.033, 2.749, 0.0266,0.829,0.253),
c(&quot;GO:0006863&quot;,&quot;purine nucleobase transport&quot;, 0.002,-4.632, 3.161, 2.464, 0.0160,0.877,0.268),
c(&quot;GO:0032026&quot;,&quot;response to magnesium ion&quot;, 0.002, 6.932,-3.864, 2.389, 0.0473,0.863,0.290),
c(&quot;GO:0070375&quot;,&quot;ERK5 cascade&quot;, 0.001, 3.865,-3.823, 2.041, 0.0134,0.729,0.303),
c(&quot;GO:0060070&quot;,&quot;canonical Wnt signaling pathway&quot;, 0.061, 4.991,-1.442, 3.893, 0.0066,0.678,0.315),
c(&quot;GO:0032486&quot;,&quot;Rap protein signal transduction&quot;, 0.002, 6.347,-0.924, 2.458, 0.0344,0.763,0.318),
c(&quot;GO:0061430&quot;,&quot;bone trabecula morphogenesis&quot;, 0.003,-2.302, 6.815, 2.553, 0.0344,0.798,0.326),
c(&quot;GO:0001957&quot;,&quot;intramembranous ossification&quot;, 0.001, 0.332, 7.419, 2.167, 0.0160,0.766,0.343),
c(&quot;GO:0009120&quot;,&quot;deoxyribonucleoside metabolic process&quot;, 0.006,-4.177,-4.617, 2.908, 0.0187,0.800,0.350),
c(&quot;GO:0007386&quot;,&quot;compartment pattern specification&quot;, 0.001, 0.048, 7.245, 2.111, 0.0134,0.749,0.356),
c(&quot;GO:0045351&quot;,&quot;type I interferon biosynthetic process&quot;, 0.002, 0.347, 6.406, 2.505, 0.0318,0.743,0.357),
c(&quot;GO:0001821&quot;,&quot;histamine secretion&quot;, 0.001,-3.314, 2.812, 2.143, 0.0240,0.877,0.358),
c(&quot;GO:0042321&quot;,&quot;negative regulation of circadian sleep/wake cycle, sleep&quot;, 0.001, 2.446, 5.266, 1.845, 0.0134,0.663,0.369),
c(&quot;GO:0006744&quot;,&quot;ubiquinone biosynthetic process&quot;, 0.136,-5.223,-2.418, 4.242, 0.0396,0.786,0.389),
c(&quot;GO:0009146&quot;,&quot;purine nucleoside triphosphate catabolic process&quot;, 0.035,-3.242,-4.729, 3.647, 0.0134,0.749,0.391),
c(&quot;GO:0046085&quot;,&quot;adenosine metabolic process&quot;, 0.003,-4.584,-4.405, 2.524, 0.0266,0.804,0.409),
c(&quot;GO:0060405&quot;,&quot;regulation of penile erection&quot;, 0.000, 1.798, 6.090, 1.716, 0.0160,0.701,0.410),
c(&quot;GO:0045725&quot;,&quot;positive regulation of glycogen biosynthetic process&quot;, 0.003, 0.849,-3.136, 2.547, 0.0396,0.723,0.414),
c(&quot;GO:1901642&quot;,&quot;nucleoside transmembrane transport&quot;, 0.047,-3.483, 2.842, 3.781, 0.0187,0.860,0.432),
c(&quot;GO:0048702&quot;,&quot;embryonic neurocranium morphogenesis&quot;, 0.002, 0.142, 6.947, 2.444, 0.0187,0.711,0.433),
c(&quot;GO:0048311&quot;,&quot;mitochondrion distribution&quot;, 0.025,-3.646, 1.645, 3.502, 0.0318,0.901,0.436),
c(&quot;GO:0046060&quot;,&quot;dATP metabolic process&quot;, 0.001,-3.497,-5.142, 2.207, 0.0134,0.789,0.437),
c(&quot;GO:1903071&quot;,&quot;positive regulation of ER-associated ubiquitin-dependent protein catabolic process&quot;, 0.001, 4.542,-2.446, 2.137, 0.0187,0.618,0.440),
c(&quot;GO:0033327&quot;,&quot;Leydig cell differentiation&quot;, 0.002,-0.252, 6.634, 2.407, 0.0292,0.686,0.450),
c(&quot;GO:0048541&quot;,&quot;Peyer&#39;s patch development&quot;, 0.002, 1.607, 6.672, 2.316, 0.0134,0.674,0.470),
c(&quot;GO:0060368&quot;,&quot;regulation of Fc receptor mediated stimulatory signaling pathway&quot;, 0.001, 6.210, 1.102, 1.919, 0.0160,0.646,0.480),
c(&quot;GO:0006351&quot;,&quot;transcription, DNA-templated&quot;,10.659,-3.806,-3.709, 6.136, 0.0041,0.852,0.480),
c(&quot;GO:0072530&quot;,&quot;purine-containing compound transmembrane transport&quot;, 0.016,-4.166, 2.833, 3.304, 0.0213,0.934,0.483),
c(&quot;GO:0072531&quot;,&quot;pyrimidine-containing compound transmembrane transport&quot;, 0.019,-4.073, 2.881, 3.391, 0.0422,0.934,0.489),
c(&quot;GO:0070255&quot;,&quot;regulation of mucus secretion&quot;, 0.002, 1.155, 4.966, 2.504, 0.0240,0.662,0.494),
c(&quot;GO:0038180&quot;,&quot;nerve growth factor signaling pathway&quot;, 0.001, 6.875,-1.582, 2.190, 0.0213,0.729,0.496),
c(&quot;GO:0009168&quot;,&quot;purine ribonucleoside monophosphate biosynthetic process&quot;, 1.043,-3.650,-3.225, 5.127, 0.0136,0.734,0.509),
c(&quot;GO:0033601&quot;,&quot;positive regulation of mammary gland epithelial cell proliferation&quot;, 0.002, 2.665, 4.905, 2.398, 0.0187,0.604,0.519),
c(&quot;GO:0000122&quot;,&quot;negative regulation of transcription from RNA polymerase II promoter&quot;, 0.199, 1.681,-2.816, 4.408, 0.0439,0.725,0.526),
c(&quot;GO:0001973&quot;,&quot;adenosine receptor signaling pathway&quot;, 0.005, 7.122,-0.404, 2.792, 0.0292,0.738,0.536),
c(&quot;GO:0038063&quot;,&quot;collagen-activated tyrosine kinase receptor signaling pathway&quot;, 0.005, 6.816,-0.631, 2.810, 0.0213,0.742,0.537),
c(&quot;GO:0008354&quot;,&quot;germ cell migration&quot;, 0.004,-0.730, 5.666, 2.729, 0.0213,0.700,0.552),
c(&quot;GO:0071320&quot;,&quot;cellular response to cAMP&quot;, 0.007, 7.142,-2.642, 2.937, 0.0071,0.810,0.553),
c(&quot;GO:0008277&quot;,&quot;regulation of G-protein coupled receptor protein signaling pathway&quot;, 0.024, 6.792, 0.170, 3.494, 0.0209,0.691,0.554),
c(&quot;GO:2000178&quot;,&quot;negative regulation of neural precursor cell proliferation&quot;, 0.005, 4.025, 1.444, 2.838, 0.0499,0.775,0.556),
c(&quot;GO:0071361&quot;,&quot;cellular response to ethanol&quot;, 0.003, 7.311,-2.794, 2.550, 0.0344,0.820,0.562),
c(&quot;GO:0043650&quot;,&quot;dicarboxylic acid biosynthetic process&quot;, 0.530,-5.261,-1.661, 4.833, 0.0344,0.767,0.571),
c(&quot;GO:0051412&quot;,&quot;response to corticosterone&quot;, 0.001, 6.485,-3.908, 2.053, 0.0422,0.848,0.572),
c(&quot;GO:0034127&quot;,&quot;regulation of MyD88-independent toll-like receptor signaling pathway&quot;, 0.000, 6.487, 1.509, 1.531, 0.0213,0.663,0.590),
c(&quot;GO:0002313&quot;,&quot;mature B cell differentiation involved in immune response&quot;, 0.003, 3.280, 4.691, 2.642, 0.0234,0.567,0.592),
c(&quot;GO:0032261&quot;,&quot;purine nucleotide salvage&quot;, 0.059,-4.165,-4.110, 3.878, 0.0187,0.768,0.593),
c(&quot;GO:0042118&quot;,&quot;endothelial cell activation&quot;, 0.002,-3.033, 0.342, 2.367, 0.0266,0.837,0.600),
c(&quot;GO:0003139&quot;,&quot;secondary heart field specification&quot;, 0.002, 0.004, 7.063, 2.358, 0.0240,0.711,0.600),
c(&quot;GO:0048387&quot;,&quot;negative regulation of retinoic acid receptor signaling pathway&quot;, 0.002, 6.096, 0.051, 2.394, 0.0213,0.681,0.601),
c(&quot;GO:0043249&quot;,&quot;erythrocyte maturation&quot;, 0.002, 2.358, 5.650, 2.461, 0.0292,0.610,0.607),
c(&quot;GO:0046130&quot;,&quot;purine ribonucleoside catabolic process&quot;, 0.036,-3.056,-4.878, 3.660, 0.0187,0.745,0.609),
c(&quot;GO:0001829&quot;,&quot;trophectodermal cell differentiation&quot;, 0.003,-0.657, 6.646, 2.521, 0.0396,0.689,0.620),
c(&quot;GO:0055015&quot;,&quot;ventricular cardiac muscle cell development&quot;, 0.003,-0.775, 6.758, 2.553, 0.0318,0.678,0.622),
c(&quot;GO:0002819&quot;,&quot;regulation of adaptive immune response&quot;, 0.025, 6.335, 1.438, 3.513, 0.0184,0.673,0.642),
c(&quot;GO:0051446&quot;,&quot;positive regulation of meiotic cell cycle&quot;, 0.007, 3.995, 2.576, 2.925, 0.0422,0.712,0.643),
c(&quot;GO:0014807&quot;,&quot;regulation of somitogenesis&quot;, 0.002, 1.564, 6.051, 2.342, 0.0213,0.651,0.660),
c(&quot;GO:0035414&quot;,&quot;negative regulation of catenin import into nucleus&quot;, 0.002, 1.162, 1.277, 2.497, 0.0266,0.750,0.669),
c(&quot;GO:0018243&quot;,&quot;protein O-linked glycosylation via threonine&quot;, 0.001,-1.054,-6.012, 2.201, 0.0187,0.817,0.676),
c(&quot;GO:0046654&quot;,&quot;tetrahydrofolate biosynthetic process&quot;, 0.101,-5.448,-2.197, 4.113, 0.0187,0.765,0.680),
c(&quot;GO:0060340&quot;,&quot;positive regulation of type I interferon-mediated signaling pathway&quot;, 0.002, 6.123, 0.385, 2.412, 0.0213,0.608,0.680),
c(&quot;GO:0046386&quot;,&quot;deoxyribose phosphate catabolic process&quot;, 0.023,-1.891,-6.345, 3.469, 0.0448,0.843,0.684),
c(&quot;GO:2000059&quot;,&quot;negative regulation of protein ubiquitination involved in ubiquitin-dependent protein catabolic process&quot;, 0.003, 2.146,-4.195, 2.603, 0.0318,0.743,0.686),
c(&quot;GO:0036066&quot;,&quot;protein O-linked fucosylation&quot;, 0.002,-0.988,-6.193, 2.307, 0.0370,0.815,0.689),
c(&quot;GO:0039530&quot;,&quot;MDA-5 signaling pathway&quot;, 0.002, 4.143, 3.197, 2.346, 0.0240,0.527,0.698),
c(&quot;GO:0071157&quot;,&quot;negative regulation of cell cycle arrest&quot;, 0.005, 3.360,-0.231, 2.804, 0.0473,0.687,0.699),
c(&quot;GO:0035023&quot;,&quot;regulation of Rho protein signal transduction&quot;, 0.125, 6.365, 0.101, 4.206, 0.0400,0.660,0.699));

bjp&lt;-
theme(
  panel.border = element_rect(colour = &quot;black&quot;, fill = NA, size = 2),
  plot.title = element_text(size = 16, face = &quot;bold&quot;),
  axis.text.y =  element_text(size = 14,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.text.x =  element_text(size = 14,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.title.y = element_text(size = 14,face = &quot;bold&quot;),
  axis.title.x = element_text(size = 14,face = &quot;bold&quot;),
  legend.text = element_text(size = 14,face = &quot;bold&quot;),
  legend.title = element_text(size = 14,face = &quot;bold&quot;),
  strip.text.x = element_text(size = 14,face = &quot;bold&quot;),
  strip.text.y = element_text(size = 14,face = &quot;bold&quot;),
  strip.background = element_rect(colour = &quot;black&quot;, size = 2))

one.data &lt;- data.frame(revigo.data);
names(one.data) &lt;- revigo.names;
one.data &lt;- one.data [(one.data$plot_X != &quot;null&quot; &amp; one.data$plot_Y != &quot;null&quot;), ];
one.data$plot_X &lt;- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y &lt;- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size &lt;- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value &lt;- as.numeric( as.character(one.data$value) );
one.data$frequency &lt;- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness &lt;- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability &lt;- as.numeric( as.character(one.data$dispensability) );
# --------------------------------------------------------------------------


# This is a table indicating which labels you would like to plot (all data will be plotted, but you need to indicate whether the label should be displayed), make this table manually.


label1 &lt;- c(&quot;GO:0070375&quot;,&quot;ERK5 cascade&quot;, 0.001, 3.865,-3.823, 2.041, 0.0134,0.729,0.303)

label2 &lt;- c(&quot;GO:0060070&quot;,&quot;canonical Wnt signaling pathway&quot;, 0.061, 4.991,-1.442, 3.893, 0.0066,0.678,0.315) 

label3 &lt;- c(&quot;GO:0055015&quot;,&quot;ventricular cardiac muscle cell development&quot;, 0.003,-0.775, 6.758, 2.553, 0.0318,0.678,0.622) 

label4 &lt;- c(&quot;GO:0003139&quot;,&quot;secondary heart field specification&quot;, 0.002, 0.004, 7.063, 2.358, 0.0240,0.711,0.600)

ex &lt;- as.data.frame(rbind(label1, label2, label3), stringsAsFactors = FALSE)
colnames(ex) &lt;- c(&quot;term_ID&quot;,&quot;description&quot;,&quot;frequency_%&quot;,&quot;plot_X&quot;,&quot;plot_Y&quot;,&quot;plot_size&quot;,&quot;value&quot;,&quot;uniqueness&quot;,&quot;dispensability&quot;)

ggplot(data = one.data)+
geom_point( aes( plot_X, plot_Y, fill = log10_p_value, size = plot_size),shape=21,color=&quot;black&quot;,stroke=0.7, alpha = 0.8) + guides(size=guide_legend(title=&quot;No. of genes&quot;)) + scale_fill_gradient2(low = muted(&quot;red&quot;), mid = &quot;white&quot;,
  high = muted(&quot;blue&quot;)) + guides(fill=guide_legend(title=&quot;P value&quot;), size=guide_legend(title=&quot;No. of genes&quot;)) +
#scale_size( range=c(5, 20)) +
  labs (y = &quot;Semantic space x&quot;, x = &quot;Semantic space y&quot;) + theme_bw()  + geom_label_repel(ex, mapping = aes(as.numeric(ex$plot_X), as.numeric(ex$plot_Y), label = description),
    fontface = &#39;bold&#39;, color = &#39;black&#39;,
    box.padding = 0.6, point.padding = 1.6,
        size = 3,
    # Width of the line segments.
    segment.size = 0,
    # Strength of the repulsion force.
    force = 1)</code></pre>
<p><img src="Revisions_H3K27ac_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<pre class="r"><code>## Revigo liver


revigo.names &lt;- c(&quot;term_ID&quot;,&quot;description&quot;,&quot;frequency_%&quot;,&quot;plot_X&quot;,&quot;plot_Y&quot;,&quot;plot_size&quot;,&quot;value&quot;,&quot;uniqueness&quot;,&quot;dispensability&quot;);
revigo.data &lt;- rbind(c(&quot;GO:0002576&quot;,&quot;platelet degranulation&quot;, 0.003, 0.496,-3.539, 2.519, 0.0000,0.842,0.000),
c(&quot;GO:0008152&quot;,&quot;metabolic process&quot;,75.387, 0.813,-0.564, 6.986, 0.0308,0.998,0.000),
c(&quot;GO:0009820&quot;,&quot;alkaloid metabolic process&quot;, 0.006, 4.043, 1.159, 2.869, 0.0000,0.947,0.000),
c(&quot;GO:0042632&quot;,&quot;cholesterol homeostasis&quot;, 0.019,-3.975, 2.639, 3.381, 0.0000,0.818,0.000),
c(&quot;GO:0019835&quot;,&quot;cytolysis&quot;, 0.044, 0.583,-0.452, 3.749, 0.0007,0.971,0.018),
c(&quot;GO:0001887&quot;,&quot;selenium compound metabolic process&quot;, 0.008, 0.543,-0.357, 2.990, 0.0189,0.963,0.019),
c(&quot;GO:0042157&quot;,&quot;lipoprotein metabolic process&quot;, 0.210, 0.683,-0.584, 4.431, 0.0116,0.957,0.028),
c(&quot;GO:0017144&quot;,&quot;drug metabolic process&quot;, 0.058, 0.564,-0.386, 3.868, 0.0392,0.959,0.035),
c(&quot;GO:0070989&quot;,&quot;oxidative demethylation&quot;, 0.011, 1.605, 0.368, 3.133, 0.0064,0.912,0.041),
c(&quot;GO:0019740&quot;,&quot;nitrogen utilization&quot;, 0.085, 0.842,-0.775, 4.038, 0.0189,0.932,0.050),
c(&quot;GO:0019695&quot;,&quot;choline metabolic process&quot;, 0.016, 0.978, 0.110, 3.301, 0.0060,0.962,0.063),
c(&quot;GO:0048625&quot;,&quot;myoblast fate commitment&quot;, 0.001, 0.704,-6.082, 2.100, 0.0189,0.822,0.085),
c(&quot;GO:0006368&quot;,&quot;transcription elongation from RNA polymerase II promoter&quot;, 0.082, 1.989, 1.876, 4.021, 0.0139,0.890,0.085),
c(&quot;GO:0051081&quot;,&quot;nuclear envelope disassembly&quot;, 0.002, 0.688,-0.443, 2.338, 0.0116,0.883,0.087),
c(&quot;GO:0006957&quot;,&quot;complement activation, alternative pathway&quot;, 0.000,-6.310, 2.907, 1.771, 0.0003,0.689,0.121),
c(&quot;GO:0017187&quot;,&quot;peptidyl-glutamic acid carboxylation&quot;, 0.006,-0.167, 1.374, 2.865, 0.0011,0.918,0.125),
c(&quot;GO:0006629&quot;,&quot;lipid metabolic process&quot;, 3.522, 3.786, 1.634, 5.655, 0.0144,0.876,0.126),
c(&quot;GO:0034384&quot;,&quot;high-density lipoprotein particle clearance&quot;, 0.002,-3.947,-4.331, 2.294, 0.0001,0.733,0.128),
c(&quot;GO:1904683&quot;,&quot;regulation of metalloendopeptidase activity&quot;, 0.001,-3.073, 3.701, 2.210, 0.0017,0.790,0.131),
c(&quot;GO:0006641&quot;,&quot;triglyceride metabolic process&quot;, 0.038, 3.327, 1.871, 3.687, 0.0001,0.832,0.140),
c(&quot;GO:0051647&quot;,&quot;nucleus localization&quot;, 0.012,-0.140,-3.521, 3.198, 0.0117,0.927,0.141),
c(&quot;GO:0042738&quot;,&quot;exogenous drug catabolic process&quot;, 0.001,-1.097, 3.996, 2.193, 0.0009,0.834,0.156),
c(&quot;GO:0070141&quot;,&quot;response to UV-A&quot;, 0.002,-3.392, 1.485, 2.369, 0.0017,0.918,0.166),
c(&quot;GO:0006768&quot;,&quot;biotin metabolic process&quot;, 0.081, 5.718, 3.455, 4.018, 0.0007,0.805,0.167),
c(&quot;GO:1901160&quot;,&quot;primary amino compound metabolic process&quot;, 0.015, 4.703, 1.279, 3.290, 0.0455,0.944,0.178),
c(&quot;GO:0043627&quot;,&quot;response to estrogen&quot;, 0.010,-3.637, 1.577, 3.128, 0.0379,0.912,0.184),
c(&quot;GO:0099133&quot;,&quot;ATP hydrolysis coupled anion transmembrane transport&quot;, 0.271,-0.029,-4.342, 4.541, 0.0152,0.919,0.184),
c(&quot;GO:0010646&quot;,&quot;regulation of cell communication&quot;, 0.929,-6.450, 3.995, 5.076, 0.0173,0.784,0.190),
c(&quot;GO:0007041&quot;,&quot;lysosomal transport&quot;, 0.017, 0.052,-3.377, 3.341, 0.0395,0.947,0.196),
c(&quot;GO:0042158&quot;,&quot;lipoprotein biosynthetic process&quot;, 0.192, 0.957, 0.864, 4.391, 0.0189,0.919,0.199),
c(&quot;GO:1990535&quot;,&quot;neuron projection maintenance&quot;, 0.000, 0.267,-0.657, 1.301, 0.0275,0.897,0.200),
c(&quot;GO:0052652&quot;,&quot;cyclic purine nucleotide metabolic process&quot;, 0.094, 4.152, 4.182, 4.082, 0.0457,0.804,0.234),
c(&quot;GO:0006390&quot;,&quot;transcription from mitochondrial promoter&quot;, 0.005, 0.802, 1.042, 2.794, 0.0484,0.915,0.235),
c(&quot;GO:0006600&quot;,&quot;creatine metabolic process&quot;, 0.002, 5.427, 2.495, 2.328, 0.0374,0.857,0.244),
c(&quot;GO:0042127&quot;,&quot;regulation of cell proliferation&quot;, 0.313,-5.699, 3.430, 4.603, 0.0229,0.785,0.259),
c(&quot;GO:0016540&quot;,&quot;protein autoprocessing&quot;, 0.011,-0.432, 1.323, 3.147, 0.0191,0.924,0.274),
c(&quot;GO:0006476&quot;,&quot;protein deacetylation&quot;, 0.072,-0.361, 1.247, 3.967, 0.0455,0.913,0.278),
c(&quot;GO:0055114&quot;,&quot;oxidation-reduction process&quot;,15.060, 4.858, 2.263, 6.286, 0.0206,0.872,0.300),
c(&quot;GO:0015886&quot;,&quot;heme transport&quot;, 0.066,-0.143,-4.692, 3.927, 0.0044,0.860,0.302),
c(&quot;GO:0006461&quot;,&quot;protein complex assembly&quot;, 0.960, 0.086,-0.705, 5.091, 0.0285,0.920,0.312),
c(&quot;GO:0060052&quot;,&quot;neurofilament cytoskeleton organization&quot;, 0.002, 1.098,-0.364, 2.501, 0.0374,0.883,0.319),
c(&quot;GO:0010918&quot;,&quot;positive regulation of mitochondrial membrane potential&quot;, 0.001,-2.305, 1.655, 2.164, 0.0044,0.851,0.327),
c(&quot;GO:0033572&quot;,&quot;transferrin transport&quot;, 0.002,-0.385,-3.961, 2.449, 0.0180,0.915,0.333),
c(&quot;GO:0006524&quot;,&quot;alanine catabolic process&quot;, 0.025, 5.584, 3.621, 3.509, 0.0009,0.800,0.338),
c(&quot;GO:0097267&quot;,&quot;omega-hydroxylase P450 pathway&quot;, 0.000, 4.727, 2.654, 1.000, 0.0017,0.851,0.339),
c(&quot;GO:0019627&quot;,&quot;urea metabolic process&quot;, 0.051, 4.765, 2.756, 3.820, 0.0455,0.846,0.339),
c(&quot;GO:1903830&quot;,&quot;magnesium ion transmembrane transport&quot;, 0.088, 0.021,-4.086, 4.053, 0.0339,0.938,0.340),
c(&quot;GO:1904179&quot;,&quot;positive regulation of adipose tissue development&quot;, 0.001,-5.562,-3.251, 1.996, 0.0009,0.657,0.344),
c(&quot;GO:0007584&quot;,&quot;response to nutrient&quot;, 0.026,-3.941, 1.599, 3.530, 0.0010,0.865,0.345),
c(&quot;GO:0090400&quot;,&quot;stress-induced premature senescence&quot;, 0.002,-4.185,-2.548, 2.350, 0.0484,0.752,0.346),
c(&quot;GO:0042908&quot;,&quot;xenobiotic transport&quot;, 0.028,-0.145,-4.010, 3.556, 0.0064,0.876,0.348),
c(&quot;GO:0060021&quot;,&quot;palate development&quot;, 0.018,-0.008,-5.732, 3.363, 0.0445,0.883,0.355),
c(&quot;GO:0033860&quot;,&quot;regulation of NAD(P)H oxidase activity&quot;, 0.001,-3.702, 4.403, 2.057, 0.0275,0.831,0.356),
c(&quot;GO:0045725&quot;,&quot;positive regulation of glycogen biosynthetic process&quot;, 0.003,-0.905, 5.340, 2.547, 0.0039,0.686,0.357),
c(&quot;GO:0001907&quot;,&quot;killing by symbiont of host cells&quot;, 0.008,-3.276, 2.257, 3.001, 0.0009,0.818,0.362),
c(&quot;GO:0016310&quot;,&quot;phosphorylation&quot;, 7.764, 2.492, 2.058, 5.998, 0.0064,0.912,0.371),
c(&quot;GO:0050892&quot;,&quot;intestinal absorption&quot;, 0.006,-0.941,-7.157, 2.913, 0.0016,0.782,0.379),
c(&quot;GO:1900107&quot;,&quot;regulation of nodal signaling pathway&quot;, 0.002,-6.525, 3.443, 2.316, 0.0275,0.740,0.380),
c(&quot;GO:0008090&quot;,&quot;retrograde axonal transport&quot;, 0.003, 0.413,-4.290, 2.521, 0.0374,0.827,0.390),
c(&quot;GO:0008203&quot;,&quot;cholesterol metabolic process&quot;, 0.028, 5.004, 2.844, 3.554, 0.0004,0.835,0.402),
c(&quot;GO:0070301&quot;,&quot;cellular response to hydrogen peroxide&quot;, 0.019,-5.078, 2.109, 3.397, 0.0023,0.823,0.402),
c(&quot;GO:0007628&quot;,&quot;adult walking behavior&quot;, 0.008,-0.864,-5.875, 3.000, 0.0288,0.814,0.411),
c(&quot;GO:0043433&quot;,&quot;negative regulation of sequence-specific DNA binding transcription factor activity&quot;, 0.031,-3.638, 4.535, 3.600, 0.0023,0.715,0.412),
c(&quot;GO:0051340&quot;,&quot;regulation of ligase activity&quot;, 0.002,-2.775, 3.103, 2.391, 0.0455,0.849,0.421),
c(&quot;GO:0097190&quot;,&quot;apoptotic signaling pathway&quot;, 0.117,-4.388, 4.477, 4.177, 0.0357,0.702,0.427),
c(&quot;GO:0006869&quot;,&quot;lipid transport&quot;, 0.270,-0.359,-4.931, 4.539, 0.0084,0.825,0.430),
c(&quot;GO:0045717&quot;,&quot;negative regulation of fatty acid biosynthetic process&quot;, 0.007, 0.731, 5.462, 2.935, 0.0030,0.667,0.434),
c(&quot;GO:0051004&quot;,&quot;regulation of lipoprotein lipase activity&quot;, 0.003,-2.788, 2.773, 2.587, 0.0089,0.840,0.435),
c(&quot;GO:0006536&quot;,&quot;glutamate metabolic process&quot;, 0.201, 5.539, 3.254, 4.412, 0.0020,0.816,0.441),
c(&quot;GO:0018279&quot;,&quot;protein N-linked glycosylation via asparagine&quot;, 0.015, 1.838, 3.560, 3.284, 0.0410,0.833,0.446),
c(&quot;GO:0046185&quot;,&quot;aldehyde catabolic process&quot;, 0.048, 4.214, 2.777, 3.790, 0.0064,0.832,0.451),
c(&quot;GO:0032071&quot;,&quot;regulation of endodeoxyribonuclease activity&quot;, 0.003,-3.421, 4.017, 2.537, 0.0374,0.791,0.451),
c(&quot;GO:0008202&quot;,&quot;steroid metabolic process&quot;, 0.161, 3.336, 1.762, 4.315, 0.0005,0.851,0.451),
c(&quot;GO:0071156&quot;,&quot;regulation of cell cycle arrest&quot;, 0.013,-2.547, 3.556, 3.230, 0.0275,0.745,0.452),
c(&quot;GO:0071372&quot;,&quot;cellular response to follicle-stimulating hormone stimulus&quot;, 0.001,-4.403, 1.848, 2.220, 0.0118,0.859,0.460),
c(&quot;GO:0042940&quot;,&quot;D-amino acid transport&quot;, 0.003,-0.063,-4.288, 2.562, 0.0275,0.873,0.471),
c(&quot;GO:1903427&quot;,&quot;negative regulation of reactive oxygen species biosynthetic process&quot;, 0.002,-3.996, 3.252, 2.486, 0.0044,0.777,0.477),
c(&quot;GO:0042493&quot;,&quot;response to drug&quot;, 0.266,-5.108, 2.172, 4.534, 0.0449,0.863,0.477),
c(&quot;GO:0001889&quot;,&quot;liver development&quot;, 0.023,-1.101,-6.479, 3.471, 0.0009,0.766,0.478),
c(&quot;GO:0006545&quot;,&quot;glycine biosynthetic process&quot;, 0.079, 5.573, 3.375, 4.005, 0.0275,0.814,0.479),
c(&quot;GO:0019216&quot;,&quot;regulation of lipid metabolic process&quot;, 0.095,-2.218, 5.684, 4.086, 0.0052,0.745,0.482),
c(&quot;GO:0006591&quot;,&quot;ornithine metabolic process&quot;, 0.088, 5.478, 3.118, 4.052, 0.0484,0.825,0.483),
c(&quot;GO:0046838&quot;,&quot;phosphorylated carbohydrate dephosphorylation&quot;, 0.064, 1.908, 2.032, 3.915, 0.0455,0.926,0.484),
c(&quot;GO:0006739&quot;,&quot;NADP metabolic process&quot;, 0.357, 4.701, 4.332, 4.661, 0.0063,0.793,0.485),
c(&quot;GO:0006520&quot;,&quot;cellular amino acid metabolic process&quot;, 5.591, 5.069, 3.403, 5.856, 0.0148,0.779,0.487),
c(&quot;GO:2000343&quot;,&quot;positive regulation of chemokine (C-X-C motif) ligand 2 production&quot;, 0.001,-5.534,-2.610, 1.959, 0.0275,0.680,0.489),
c(&quot;GO:0021814&quot;,&quot;cell motility involved in cerebral cortex radial glia guided migration&quot;, 0.001,-0.848,-6.605, 2.238, 0.0189,0.729,0.492),
c(&quot;GO:0022900&quot;,&quot;electron transport chain&quot;, 0.564, 3.324, 2.185, 4.860, 0.0186,0.848,0.493),
c(&quot;GO:0090277&quot;,&quot;positive regulation of peptide hormone secretion&quot;, 0.013,-4.678, 0.230, 3.226, 0.0004,0.601,0.496),
c(&quot;GO:0015942&quot;,&quot;formate metabolic process&quot;, 0.065, 5.321, 3.235, 3.921, 0.0017,0.832,0.499),
c(&quot;GO:0046329&quot;,&quot;negative regulation of JNK cascade&quot;, 0.007,-4.599, 5.445, 2.940, 0.0063,0.605,0.500),
c(&quot;GO:0001503&quot;,&quot;ossification&quot;, 0.074,-0.710,-6.573, 3.979, 0.0413,0.791,0.504),
c(&quot;GO:0006898&quot;,&quot;receptor-mediated endocytosis&quot;, 0.095, 0.278,-4.169, 4.086, 0.0004,0.935,0.511),
c(&quot;GO:0090239&quot;,&quot;regulation of histone H4 acetylation&quot;, 0.003,-3.175, 3.088, 2.531, 0.0455,0.769,0.511),
c(&quot;GO:0060059&quot;,&quot;embryonic retina morphogenesis in camera-type eye&quot;, 0.003,-0.909,-6.515, 2.573, 0.0029,0.790,0.512),
c(&quot;GO:1902459&quot;,&quot;positive regulation of stem cell population maintenance&quot;, 0.001,-4.906,-2.513, 2.236, 0.0275,0.663,0.517),
c(&quot;GO:1902959&quot;,&quot;regulation of aspartic-type endopeptidase activity involved in amyloid precursor protein catabolic process&quot;, 0.000,-2.854, 3.852, 1.756, 0.0455,0.802,0.517),
c(&quot;GO:0031100&quot;,&quot;animal organ regeneration&quot;, 0.005,-1.364,-6.064, 2.792, 0.0402,0.788,0.526),
c(&quot;GO:0061620&quot;,&quot;glycolytic process through glucose-6-phosphate&quot;, 0.002, 5.004, 4.281, 2.290, 0.0455,0.798,0.529),
c(&quot;GO:0002679&quot;,&quot;respiratory burst involved in defense response&quot;, 0.001,-3.703, 1.450, 2.258, 0.0374,0.816,0.535),
c(&quot;GO:0034504&quot;,&quot;protein localization to nucleus&quot;, 0.129,-0.313,-4.647, 4.218, 0.0210,0.899,0.539),
c(&quot;GO:0014911&quot;,&quot;positive regulation of smooth muscle cell migration&quot;, 0.003,-5.348,-0.209, 2.620, 0.0461,0.695,0.540),
c(&quot;GO:0061138&quot;,&quot;morphogenesis of a branching epithelium&quot;, 0.042,-1.117,-6.445, 3.730, 0.0021,0.775,0.541),
c(&quot;GO:0010793&quot;,&quot;regulation of mRNA export from nucleus&quot;, 0.003,-3.669,-1.160, 2.625, 0.0275,0.749,0.541),
c(&quot;GO:0072049&quot;,&quot;comma-shaped body morphogenesis&quot;, 0.001,-1.363,-5.960, 2.173, 0.0189,0.801,0.543),
c(&quot;GO:0046498&quot;,&quot;S-adenosylhomocysteine metabolic process&quot;, 0.066, 5.725, 3.823, 3.925, 0.0275,0.808,0.544),
c(&quot;GO:0045620&quot;,&quot;negative regulation of lymphocyte differentiation&quot;, 0.008,-3.807,-2.279, 3.018, 0.0029,0.590,0.549),
c(&quot;GO:0019626&quot;,&quot;short-chain fatty acid catabolic process&quot;, 0.024, 5.002, 3.731, 3.490, 0.0189,0.779,0.556),
c(&quot;GO:0034356&quot;,&quot;NAD biosynthesis via nicotinamide riboside salvage pathway&quot;, 0.004, 4.983, 4.281, 2.691, 0.0235,0.828,0.560),
c(&quot;GO:0046620&quot;,&quot;regulation of organ growth&quot;, 0.020,-4.335,-3.406, 3.403, 0.0044,0.688,0.561),
c(&quot;GO:0006778&quot;,&quot;porphyrin-containing compound metabolic process&quot;, 0.457, 5.545, 2.444, 4.768, 0.0453,0.896,0.561),
c(&quot;GO:0010269&quot;,&quot;response to selenium ion&quot;, 0.001,-3.417, 1.446, 1.991, 0.0189,0.892,0.564),
c(&quot;GO:0090335&quot;,&quot;regulation of brown fat cell differentiation&quot;, 0.003,-3.134,-2.585, 2.531, 0.0274,0.735,0.566),
c(&quot;GO:0000098&quot;,&quot;sulfur amino acid catabolic process&quot;, 0.008, 5.398, 3.462, 3.030, 0.0064,0.815,0.566),
c(&quot;GO:1905668&quot;,&quot;positive regulation of protein localization to endosome&quot;, 0.001,-4.713,-0.861, 2.079, 0.0275,0.732,0.569),
c(&quot;GO:0031016&quot;,&quot;pancreas development&quot;, 0.021,-1.275,-6.229, 3.438, 0.0453,0.774,0.573),
c(&quot;GO:0071397&quot;,&quot;cellular response to cholesterol&quot;, 0.001,-4.311, 1.853, 2.272, 0.0191,0.857,0.574),
c(&quot;GO:0060712&quot;,&quot;spongiotrophoblast layer development&quot;, 0.003,-1.057,-6.380, 2.533, 0.0191,0.790,0.578),
c(&quot;GO:0031659&quot;,&quot;positive regulation of cyclin-dependent protein serine/threonine kinase activity involved in G1/S transition of mitotic cell cycle&quot;, 0.002,-3.538, 4.926, 2.350, 0.0275,0.670,0.578),
c(&quot;GO:1903896&quot;,&quot;positive regulation of IRE1-mediated unfolded protein response&quot;, 0.001,-6.708, 3.143, 1.857, 0.0275,0.671,0.580),
c(&quot;GO:0071560&quot;,&quot;cellular response to transforming growth factor beta stimulus&quot;, 0.050,-4.466, 1.856, 3.804, 0.0127,0.832,0.582),
c(&quot;GO:0035733&quot;,&quot;hepatic stellate cell activation&quot;, 0.000, 0.492,-0.383, 1.748, 0.0455,0.898,0.584),
c(&quot;GO:0034763&quot;,&quot;negative regulation of transmembrane transport&quot;, 0.014,-4.429,-0.887, 3.251, 0.0274,0.743,0.585),
c(&quot;GO:0042167&quot;,&quot;heme catabolic process&quot;, 0.002, 5.014, 2.712, 2.408, 0.0374,0.834,0.586),
c(&quot;GO:0006853&quot;,&quot;carnitine shuttle&quot;, 0.000,-0.241,-4.610, 1.114, 0.0089,0.860,0.587),
c(&quot;GO:0007411&quot;,&quot;axon guidance&quot;, 0.066,-3.249,-3.116, 3.930, 0.0223,0.617,0.587),
c(&quot;GO:0048013&quot;,&quot;ephrin receptor signaling pathway&quot;, 0.016,-6.783, 3.560, 3.308, 0.0309,0.752,0.593),
c(&quot;GO:0034340&quot;,&quot;response to type I interferon&quot;, 0.008,-5.227, 1.865, 3.007, 0.0117,0.821,0.598),
c(&quot;GO:0031325&quot;,&quot;positive regulation of cellular metabolic process&quot;, 1.004,-6.818, 4.034, 5.110, 0.0233,0.668,0.602),
c(&quot;GO:0042177&quot;,&quot;negative regulation of protein catabolic process&quot;, 0.025,-2.311, 5.307, 3.513, 0.0177,0.698,0.606),
c(&quot;GO:1900016&quot;,&quot;negative regulation of cytokine production involved in inflammatory response&quot;, 0.002,-5.942,-1.181, 2.391, 0.0191,0.603,0.606),
c(&quot;GO:2000463&quot;,&quot;positive regulation of excitatory postsynaptic potential&quot;, 0.003,-6.002, 3.548, 2.606, 0.0220,0.638,0.607),
c(&quot;GO:0098902&quot;,&quot;regulation of membrane depolarization during action potential&quot;, 0.001,-3.098, 2.354, 2.188, 0.0455,0.827,0.608),
c(&quot;GO:1902237&quot;,&quot;positive regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway&quot;, 0.002,-6.892, 3.889, 2.324, 0.0089,0.647,0.617),
c(&quot;GO:0008635&quot;,&quot;activation of cysteine-type endopeptidase activity involved in apoptotic process by cytochrome c&quot;, 0.001,-5.340, 4.602, 2.279, 0.0484,0.665,0.619),
c(&quot;GO:2000271&quot;,&quot;positive regulation of fibroblast apoptotic process&quot;, 0.002,-6.128, 4.110, 2.320, 0.0374,0.726,0.621),
c(&quot;GO:0050730&quot;,&quot;regulation of peptidyl-tyrosine phosphorylation&quot;, 0.045,-2.927, 5.206, 3.759, 0.0115,0.746,0.627),
c(&quot;GO:0014068&quot;,&quot;positive regulation of phosphatidylinositol 3-kinase signaling&quot;, 0.013,-6.835, 3.378, 3.216, 0.0477,0.656,0.627),
c(&quot;GO:0032148&quot;,&quot;activation of protein kinase B activity&quot;, 0.004,-3.889, 4.818, 2.738, 0.0411,0.689,0.628),
c(&quot;GO:0000122&quot;,&quot;negative regulation of transcription from RNA polymerase II promoter&quot;, 0.199,-3.286, 4.570, 4.408, 0.0343,0.704,0.628),
c(&quot;GO:2000659&quot;,&quot;regulation of interleukin-1-mediated signaling pathway&quot;, 0.001,-6.610, 3.398, 2.117, 0.0275,0.727,0.629),
c(&quot;GO:0043392&quot;,&quot;negative regulation of DNA binding&quot;, 0.014,-3.451, 2.990, 3.256, 0.0324,0.830,0.631),
c(&quot;GO:0042426&quot;,&quot;choline catabolic process&quot;, 0.001, 4.091, 2.217, 2.230, 0.0275,0.898,0.632),
c(&quot;GO:1900119&quot;,&quot;positive regulation of execution phase of apoptosis&quot;, 0.003,-5.254, 3.939, 2.528, 0.0152,0.720,0.635),
c(&quot;GO:0046487&quot;,&quot;glyoxylate metabolic process&quot;, 0.049, 5.512, 3.224, 3.794, 0.0484,0.832,0.635),
c(&quot;GO:0030917&quot;,&quot;midbrain-hindbrain boundary development&quot;, 0.003,-0.916,-6.603, 2.576, 0.0064,0.776,0.636),
c(&quot;GO:0031663&quot;,&quot;lipopolysaccharide-mediated signaling pathway&quot;, 0.010,-6.114, 3.001, 3.128, 0.0363,0.713,0.638),
c(&quot;GO:1901679&quot;,&quot;nucleotide transmembrane transport&quot;, 0.011,-0.193,-4.465, 3.131, 0.0374,0.862,0.638),
c(&quot;GO:0032933&quot;,&quot;SREBP signaling pathway&quot;, 0.008,-6.008, 4.192, 2.989, 0.0484,0.656,0.640),
c(&quot;GO:2000646&quot;,&quot;positive regulation of receptor catabolic process&quot;, 0.001,-2.717, 5.596, 1.881, 0.0189,0.731,0.642),
c(&quot;GO:0071276&quot;,&quot;cellular response to cadmium ion&quot;, 0.009,-4.925, 2.123, 3.077, 0.0040,0.848,0.644),
c(&quot;GO:0000060&quot;,&quot;protein import into nucleus, translocation&quot;, 0.012,-0.358,-4.229, 3.175, 0.0286,0.853,0.645),
c(&quot;GO:0006548&quot;,&quot;histidine catabolic process&quot;, 0.052, 5.484, 3.853, 3.828, 0.0044,0.789,0.649),
c(&quot;GO:0034638&quot;,&quot;phosphatidylcholine catabolic process&quot;, 0.002, 4.403, 3.693, 2.328, 0.0374,0.793,0.652),
c(&quot;GO:1901523&quot;,&quot;icosanoid catabolic process&quot;, 0.000, 5.088, 3.154, 1.732, 0.0189,0.841,0.653),
c(&quot;GO:0006559&quot;,&quot;L-phenylalanine catabolic process&quot;, 0.031, 5.469, 3.690, 3.596, 0.0118,0.794,0.657),
c(&quot;GO:0051642&quot;,&quot;centrosome localization&quot;, 0.007,-0.128,-3.483, 2.973, 0.0190,0.928,0.658),
c(&quot;GO:0010040&quot;,&quot;response to iron(II) ion&quot;, 0.002,-4.034, 1.687, 2.407, 0.0189,0.886,0.658),
c(&quot;GO:0000042&quot;,&quot;protein targeting to Golgi&quot;, 0.021,-0.296,-4.424, 3.422, 0.0374,0.900,0.670),
c(&quot;GO:0090201&quot;,&quot;negative regulation of release of cytochrome c from mitochondria&quot;, 0.003,-5.536, 3.408, 2.591, 0.0285,0.651,0.672),
c(&quot;GO:0006855&quot;,&quot;drug transmembrane transport&quot;, 0.189,-4.897,-2.034, 4.384, 0.0138,0.752,0.672),
c(&quot;GO:0009887&quot;,&quot;animal organ morphogenesis&quot;, 0.264,-1.342,-6.260, 4.530, 0.0333,0.742,0.673),
c(&quot;GO:0006879&quot;,&quot;cellular iron ion homeostasis&quot;, 0.110,-1.935, 4.780, 4.148, 0.0002,0.750,0.674),
c(&quot;GO:0030178&quot;,&quot;negative regulation of Wnt signaling pathway&quot;, 0.037,-5.011, 4.290, 3.681, 0.0367,0.636,0.676),
c(&quot;GO:0007262&quot;,&quot;STAT protein import into nucleus&quot;, 0.001,-5.800,-0.845, 2.223, 0.0484,0.729,0.677),
c(&quot;GO:0045739&quot;,&quot;positive regulation of DNA repair&quot;, 0.013,-5.991, 3.701, 3.205, 0.0123,0.635,0.678),
c(&quot;GO:0034142&quot;,&quot;toll-like receptor 4 signaling pathway&quot;, 0.006,-6.777, 3.183, 2.892, 0.0191,0.642,0.679),
c(&quot;GO:0046326&quot;,&quot;positive regulation of glucose import&quot;, 0.007,-6.120,-0.943, 2.979, 0.0287,0.694,0.680),
c(&quot;GO:0042773&quot;,&quot;ATP synthesis coupled electron transport&quot;, 0.221, 4.104, 4.074, 4.452, 0.0456,0.794,0.685),
c(&quot;GO:0010886&quot;,&quot;positive regulation of cholesterol storage&quot;, 0.002,-5.084,-0.414, 2.320, 0.0374,0.711,0.685),
c(&quot;GO:0001568&quot;,&quot;blood vessel development&quot;, 0.136,-1.287,-6.290, 4.243, 0.0452,0.758,0.689),
c(&quot;GO:0006505&quot;,&quot;GPI anchor metabolic process&quot;, 0.102, 3.442, 3.385, 4.115, 0.0456,0.805,0.690),
c(&quot;GO:0046439&quot;,&quot;L-cysteine metabolic process&quot;, 0.039, 6.129, 3.195, 3.699, 0.0374,0.827,0.692),
c(&quot;GO:0051897&quot;,&quot;positive regulation of protein kinase B signaling&quot;, 0.016,-6.848, 3.394, 3.304, 0.0485,0.653,0.692),
c(&quot;GO:0032287&quot;,&quot;peripheral nervous system myelin maintenance&quot;, 0.002,-0.055,-6.738, 2.468, 0.0374,0.750,0.693),
c(&quot;GO:1904468&quot;,&quot;negative regulation of tumor necrosis factor secretion&quot;, 0.001,-3.544,-4.697, 1.982, 0.0374,0.625,0.695),
c(&quot;GO:1903874&quot;,&quot;ferrous iron transmembrane transport&quot;, 0.028, 0.089,-4.022, 3.563, 0.0374,0.931,0.700));



one.data &lt;- data.frame(revigo.data);
names(one.data) &lt;- revigo.names;
one.data &lt;- one.data [(one.data$plot_X != &quot;null&quot; &amp; one.data$plot_Y != &quot;null&quot;), ];
one.data$plot_X &lt;- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y &lt;- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size &lt;- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value &lt;- as.numeric( as.character(one.data$value) );
one.data$frequency &lt;- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness &lt;- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability &lt;- as.numeric( as.character(one.data$dispensability) );
# --------------------------------------------------------------------------


# This is a table indicating which labels you would like to plot (all data will be plotted, but you need to indicate whether the label should be displayed), make this table manually.


label1 &lt;- c(&quot;GO:0042632&quot;,&quot;cholesterol homeostasis&quot;, 0.019,-3.975, 2.639, 3.381, 0.0000,0.818,0.000)

label2 &lt;- c(&quot;GO:0017144&quot;,&quot;drug metabolic process&quot;, 0.058, 0.564,-0.386, 3.868, 0.0392,0.959,0.035) 

label3 &lt;- c(&quot;GO:0006629&quot;,&quot;lipid metabolic process&quot;, 3.522, 3.786, 1.634, 5.655, 0.0144,0.876,0.126)



ex &lt;- as.data.frame(rbind(label1, label2, label3), stringsAsFactors = FALSE)
colnames(ex) &lt;- c(&quot;term_ID&quot;,&quot;description&quot;,&quot;frequency_%&quot;,&quot;plot_X&quot;,&quot;plot_Y&quot;,&quot;plot_size&quot;,&quot;value&quot;,&quot;uniqueness&quot;,&quot;dispensability&quot;)

ggplot(data = one.data)+
geom_point( aes( plot_X, plot_Y, fill = log10_p_value, size = plot_size),shape=21,color=&quot;black&quot;,stroke=0.7, alpha = 0.8) + guides(size=guide_legend(title=&quot;No. of genes&quot;)) + scale_fill_gradient2(low = muted(&quot;red&quot;), mid = &quot;white&quot;,
  high = muted(&quot;blue&quot;)) + guides(fill=guide_legend(title=&quot;P value&quot;), size=guide_legend(title=&quot;No. of genes&quot;)) +
#scale_size( range=c(5, 20)) +
  labs (y = &quot;Semantic space x&quot;, x = &quot;Semantic space y&quot;) + theme_bw()  + geom_label_repel(ex, mapping = aes(as.numeric(ex$plot_X), as.numeric(ex$plot_Y), label = description),
    fontface = &#39;bold&#39;, color = &#39;black&#39;,
    box.padding = 0.6, point.padding = 1.6,
        size = 3,
    # Width of the line segments.
    segment.size = 0,
    # Strength of the repulsion force.
    force = 1)</code></pre>
<p><img src="Revisions_H3K27ac_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
