---
title: "Revisions_H3K27ac"
author: "Lauren Blake"
date: "June 24, 2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Open files

Note: due to pairwise ordering from another file (tissue_specific_tDMRs), tDMRs labeled as "hypermethylated" are actually hypomethylated in the lung and kidney.

```{r}
# Library
library(bedr)
library("topGO")
#library("biomaRt")
library("clusterProfiler")
library("org.Hs.eg.db")

human_chimp_heart_specific_25_conserved_H3K27_HEART <- read.delim("~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_heart_specific_25_conserved_H3K27_HEART.bed", header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_HEART$V4)

413/(413+42)



human_chimp_heart_specific_25_conserved_H3K27_KIDNEY <- read.delim("~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_kidney_specific_25_conserved_H3K27_KIDNEY.bed", header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_KIDNEY$V4)

858/(858+39)

human_chimp_heart_specific_25_conserved_H3K27_LIVER <- read.delim("~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_liver_specific_25_conserved_H3K27_LIVER.bed", header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_LIVER$V4)

1081/(1081+252)

human_chimp_heart_specific_25_conserved_H3K27_LUNG <- read.delim("~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/human_chimp_lung_specific_25_conserved_H3K27_LUNG.bed", header=FALSE, stringsAsFactors = FALSE)

summary(human_chimp_heart_specific_25_conserved_H3K27_LUNG$V4)

30/(30+21)
```



```{r}
refGene_hg19_TSS <- read.delim("~/Desktop/Regulatory_Evol/ashlar-trial/data/overlap_reg/overlap_25/refGene_hg19_TSS.R", header=FALSE, stringsAsFactors = FALSE)

# Adjust so that bedtools will accept it
refGene_hg19_TSS[,3] <- refGene_hg19_TSS[,3] + 1 

# Now sort

human_chimp_heart_cons <- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_HEART[,1:3])

refGene_hg19_TSS <- bedr.sort.region(refGene_hg19_TSS)

closest_heart <- bedr(input = list(a = human_chimp_heart_specific_25_conserved_H3K27_HEART[,1:3], b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Sort based on upsteam/downstream

closest_heart_upstream <- closest_heart[which(closest_heart$V9 == "+"),]
closest_heart_downstream <- closest_heart[which(closest_heart$V9 == "-"),]
distance_heart_upstream <- as.numeric(closest_heart_upstream$V3) - as.numeric(closest_heart_upstream$V5)
distance_heart_downstream <- as.numeric(closest_heart_downstream$V5) - as.numeric(closest_heart_downstream$V3)

length(which(distance_heart_upstream > 0))/length(distance_heart_upstream)
length(which(distance_heart_downstream > 0))/length(distance_heart_downstream)

summary(distance_heart_downstream)
```

## Kidney

```{r}
# Now sort

human_chimp_heart_cons <- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_KIDNEY[,1:3])

refGene_hg19_TSS <- bedr.sort.region(refGene_hg19_TSS)

closest_heart <- bedr(input = list(a = human_chimp_heart_cons, b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Sort based on upsteam/downstream

closest_heart_upstream <- closest_heart[which(closest_heart$V9 == "+"),]
closest_heart_downstream <- closest_heart[which(closest_heart$V9 == "-"),]
distance_heart_upstream <- as.numeric(closest_heart_upstream$V3) - as.numeric(closest_heart_upstream$V5)
distance_heart_downstream <- as.numeric(closest_heart_downstream$V5) - as.numeric(closest_heart_downstream$V3)

length(which(distance_heart_upstream > 0))/length(distance_heart_upstream)
length(which(distance_heart_downstream > 0))/length(distance_heart_downstream)

```

## Liver

```{r}
# Now sort

human_chimp_heart_cons <- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_LIVER[,1:3])

refGene_hg19_TSS <- bedr.sort.region(refGene_hg19_TSS)

closest_heart <- bedr(input = list(a = human_chimp_heart_cons, b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Sort based on upsteam/downstream

closest_liver_upstream <- closest_heart[which(closest_heart$V9 == "+"),]
closest_liver_downstream <- closest_heart[which(closest_heart$V9 == "-"),]
distance_liver_upstream <- as.numeric(closest_liver_upstream$V3) - as.numeric(closest_liver_upstream$V5)
distance_liver_downstream <- as.numeric(closest_liver_downstream$V5) - as.numeric(closest_liver_downstream$V3)

length(which(distance_liver_upstream > 0))/length(distance_liver_upstream)
length(which(distance_liver_downstream > 0))/length(distance_liver_downstream)

all_distance <- c(distance_heart_upstream, distance_heart_downstream, distance_liver_upstream, distance_liver_downstream)
summary(all_distance)
```

## Lung

```{r}
# Now sort

human_chimp_heart_cons <- bedr.sort.region(human_chimp_heart_specific_25_conserved_H3K27_LUNG[,1:3])

refGene_hg19_TSS <- bedr.sort.region(refGene_hg19_TSS)

closest_heart <- bedr(input = list(a = human_chimp_heart_cons, b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Sort based on upsteam/downstream

closest_heart_upstream <- closest_heart[which(closest_heart$V9 == "+"),]
closest_heart_downstream <- closest_heart[which(closest_heart$V9 == "-"),]
distance_heart_upstream <- as.numeric(closest_heart_upstream$V3) - as.numeric(closest_heart_upstream$V5)
distance_heart_downstream <- as.numeric(closest_heart_downstream$V5) - as.numeric(closest_heart_downstream$V3)

length(which(distance_heart_upstream > 0))/length(distance_heart_upstream)
length(which(distance_heart_downstream > 0))/length(distance_heart_downstream)

```


# Separate into hypermethylated and overlapping H3K27ac

# Heart
```{r}
# Find the heart hypo
heart_hypo <- human_chimp_heart_specific_25_conserved_H3K27_HEART[which(human_chimp_heart_specific_25_conserved_H3K27_LIVER$V4 == "hyper" & human_chimp_heart_specific_25_conserved_H3K27_LIVER$V6 != "-1"),]

heart_hypo <- heart_hypo[complete.cases(heart_hypo), ]

# Find the closest gene

closest_heart <- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Convert the gene name to ensg

gene_id <- read.table("../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv", stringsAsFactors = FALSE, header=TRUE, sep = ",")

comb_kidney <- merge(closest_heart, gene_id, by.x = c("V8"), by.y = c("Gene"))

comb_kidney$ensg

# Convert the TSS to ensg as well 

ref_gene_hg19_ensg <- merge(refGene_hg19_TSS, gene_id, by.x = c("V5"), by.y = c("Gene"))

all_ref_gene_hg19_ensg <- unique(ref_gene_hg19_ensg$ensg)

heart_ref_gene <- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene <- as.numeric(as.vector(heart_ref_gene))
names(test_gene) <-  all_ref_gene_hg19_ensg

# Run topGO
go_data <- new("topGOdata",
                   ontology = "BP",
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore > 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = "org.Hs.eg.db",
                   ID = "ensembl")

# Perform enrichment test
go_test <- runTest(go_data, algorithm = "weight01", statistic = "fisher")

go_table <- GenTable(go_data, weightFisher = go_test,
                         orderBy = "weightFisher", ranksOf = "weightFisher",
                         topNodes = sum(score(go_test) < .05))

go_table

go_table_heart <- as.data.frame(cbind(go_table$GO.ID, go_table$weightFisher))

write.table(go_table_heart, "../data/go_table_heart.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)

# Get names of kidney genes
sig.genes <- sigGenes(go_data)
goresults <- sapply(go_table$GO.ID, function(x)
    {
      genes<-genesInTerm(go_data, x) 
      genes[[1]][genes[[1]] %in% sig.genes]
    })

# cardiac myofibril assembly, positive regulation of heart rate, regulation of monocyte differentiation
goresults["GO:0055003"] 
goresults["GO:0010460"]
goresults["GO:0045655"]
```



## Kidney

```{r}
# Find the heart hypo
heart_hypo <- human_chimp_heart_specific_25_conserved_H3K27_KIDNEY[which(human_chimp_heart_specific_25_conserved_H3K27_KIDNEY$V4 == "hyper" & human_chimp_heart_specific_25_conserved_H3K27_KIDNEY$V6 != "-1"),]

heart_hypo <- heart_hypo[complete.cases(heart_hypo), ]
# Find the closest gene

closest_heart <- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Convert the gene name to ensg

gene_id <- read.table("../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv", stringsAsFactors = FALSE, header=TRUE, sep = ",")

comb_kidney <- merge(closest_heart, gene_id, by.x = c("V8"), by.y = c("Gene"))

comb_kidney$ensg

# Convert the TSS to ensg as well 

ref_gene_hg19_ensg <- merge(refGene_hg19_TSS, gene_id, by.x = c("V5"), by.y = c("Gene"))

all_ref_gene_hg19_ensg <- unique(ref_gene_hg19_ensg$ensg)

kidney_ref_gene <- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene <- as.numeric(as.vector(heart_ref_gene))
names(test_gene) <-  all_ref_gene_hg19_ensg

# Run topGO
go_data <- new("topGOdata",
                   ontology = "BP",
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore > 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = "org.Hs.eg.db",
                   ID = "ensembl")

# Perform enrichment test
go_test <- runTest(go_data, algorithm = "weight01", statistic = "fisher")

go_table <- GenTable(go_data, weightFisher = go_test,
                         orderBy = "weightFisher", ranksOf = "weightFisher",
                         topNodes = sum(score(go_test) < .01))

go_table


goresults["GO:0098719"] 


```

## Liver

```{r}
# Find the heart hypo
heart_hypo <- human_chimp_heart_specific_25_conserved_H3K27_LIVER[which(human_chimp_heart_specific_25_conserved_H3K27_LIVER$V4 == "hypo" & human_chimp_heart_specific_25_conserved_H3K27_LIVER$V6 != "-1"),]

heart_hypo <- heart_hypo[complete.cases(heart_hypo), ]

# Find the closest gene

closest_heart <- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Convert the gene name to ensg

gene_id <- read.table("../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv", stringsAsFactors = FALSE, header=TRUE, sep = ",")

comb_kidney <- merge(closest_heart, gene_id, by.x = c("V8"), by.y = c("Gene"))

comb_kidney$ensg

# Convert the TSS to ensg as well 

ref_gene_hg19_ensg <- merge(refGene_hg19_TSS, gene_id, by.x = c("V5"), by.y = c("Gene"))

all_ref_gene_hg19_ensg <- unique(ref_gene_hg19_ensg$ensg)

liver_ref_gene <- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene <- as.numeric(as.vector(liver_ref_gene))
names(test_gene) <-  all_ref_gene_hg19_ensg

# Run topGO
go_data <- new("topGOdata",
                   ontology = "BP",
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore > 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = "org.Hs.eg.db",
                   ID = "ensembl")

# Perform enrichment test
go_test <- runTest(go_data, algorithm = "weight01", statistic = "fisher")

go_table <- GenTable(go_data, weightFisher = go_test,
                         orderBy = "weightFisher", ranksOf = "weightFisher",
                         topNodes = sum(score(go_test) < .05))

go_table


 go_table_liver <- as.data.frame(cbind(go_table$GO.ID, go_table$weightFisher))

write.table(go_table_liver, "../data/go_table_liver.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)



```

## Lung

```{r}
# Find the heart hypo
heart_hypo <- human_chimp_heart_specific_25_conserved_H3K27_LUNG[which(human_chimp_heart_specific_25_conserved_H3K27_LUNG$V4 == "hyper" & human_chimp_heart_specific_25_conserved_H3K27_LUNG$V6 != "-1"),]

heart_hypo <- heart_hypo[complete.cases(heart_hypo), ]

# Find the closest gene

closest_heart <- bedr(input = list(a = heart_hypo[,1:3], b = refGene_hg19_TSS), method = "closest", check.chr = FALSE)

# Convert the gene name to ensg

gene_id <- read.table("../../../Reg_Evo_Primates/data/ENSG_GENE_HG19.csv", stringsAsFactors = FALSE, header=TRUE, sep = ",")

comb_kidney <- merge(closest_heart, gene_id, by.x = c("V8"), by.y = c("Gene"))

comb_kidney$ensg

# Convert the TSS to ensg as well 

ref_gene_hg19_ensg <- merge(refGene_hg19_TSS, gene_id, by.x = c("V5"), by.y = c("Gene"))

all_ref_gene_hg19_ensg <- unique(ref_gene_hg19_ensg$ensg)

heart_ref_gene <- all_ref_gene_hg19_ensg %in% comb_kidney$ensg

# Revisions- run GO
# Merge ENSG with true/false

test_gene <- as.numeric(as.vector(heart_ref_gene))
names(test_gene) <-  all_ref_gene_hg19_ensg

# Run topGO
go_data <- new("topGOdata",
                   ontology = "BP",
                   allGenes = test_gene, 
                    geneSel = function(allScore){
    return(allScore > 0)
},
                   nodeSize = 5,
                   annotationFun = annFUN.org,
                   mapping = "org.Hs.eg.db",
                   ID = "ensembl")

# Perform enrichment test
go_test <- runTest(go_data, algorithm = "weight01", statistic = "fisher")

go_table <- GenTable(go_data, weightFisher = go_test,
                         orderBy = "weightFisher", ranksOf = "weightFisher",
                         topNodes = sum(score(go_test) < .01))

go_table


 

```

# Run clusterProfiler

```{r}
#head(heart_ref_gene)
#length(heart_ref_gene)
#head(liver_ref_gene)
#length(liver_ref_gene)

#mydf <- as.data.frame(cbind(all_ref_gene_hg19_ensg, heart_ref_gene, liver_ref_gene))

#colnames(mydf) <- c("ensg", "heart", "liver")

#xx.formula.twogroups <- compareCluster(mydf$heart, data = mydf, fun="enrichGO", universe = df$ensg,
#                OrgDb         = org.Hs.eg.db)

#dotplot(xx.formula.twogroups, showCategory=5, includeAll=FALSE)
```

```{r}
## Reivigo

# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );
library(ggrepel)

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0009113","purine nucleobase biosynthetic process", 0.124,-4.742,-3.192, 4.202, 0.0005,0.769,0.000),
c("GO:0033197","response to vitamin E", 0.000, 7.179,-3.273, 1.740, 0.0005,0.849,0.000),
c("GO:0019043","establishment of viral latency", 0.001,-5.135, 5.016, 1.959, 0.0187,0.945,0.018),
c("GO:0006501","C-terminal protein lipidation", 0.004, 0.003,-5.207, 2.693, 0.0120,0.913,0.100),
c("GO:0060509","Type I pneumocyte differentiation", 0.001,-0.420, 6.928, 2.121, 0.0134,0.698,0.103),
c("GO:1903078","positive regulation of protein localization to plasma membrane", 0.010, 1.621, 0.720, 3.123, 0.0068,0.688,0.118),
c("GO:0033632","regulation of cell-cell adhesion mediated by integrin", 0.002, 2.909, 0.449, 2.378, 0.0160,0.835,0.132),
c("GO:2000109","regulation of macrophage apoptotic process", 0.001, 2.079,-2.113, 2.188, 0.0187,0.768,0.132),
c("GO:0032007","negative regulation of TOR signaling", 0.016, 5.974, 0.117, 3.325, 0.0052,0.640,0.175),
c("GO:0043547","positive regulation of GTPase activity", 0.470, 4.687, 1.488, 4.780, 0.0054,0.838,0.179),
c("GO:0009256","10-formyltetrahydrofolate metabolic process", 0.002,-6.344,-1.530, 2.474, 0.0134,0.809,0.185),
c("GO:0070141","response to UV-A", 0.002, 6.586,-3.433, 2.369, 0.0160,0.883,0.188),
c("GO:0071499","cellular response to laminar fluid shear stress", 0.001, 6.586,-2.544, 2.143, 0.0187,0.852,0.211),
c("GO:0006139","nucleobase-containing compound metabolic process",26.547,-4.434,-2.184, 6.532, 0.0499,0.889,0.214),
c("GO:0018242","protein O-linked glycosylation via serine", 0.001,-1.029,-5.988, 2.117, 0.0160,0.818,0.218),
c("GO:0015942","formate metabolic process", 0.065,-5.564,-1.813, 3.921, 0.0160,0.815,0.246),
c("GO:0070601","centromeric sister chromatid cohesion", 0.004,-1.575,-1.033, 2.749, 0.0266,0.829,0.253),
c("GO:0006863","purine nucleobase transport", 0.002,-4.632, 3.161, 2.464, 0.0160,0.877,0.268),
c("GO:0032026","response to magnesium ion", 0.002, 6.932,-3.864, 2.389, 0.0473,0.863,0.290),
c("GO:0070375","ERK5 cascade", 0.001, 3.865,-3.823, 2.041, 0.0134,0.729,0.303),
c("GO:0060070","canonical Wnt signaling pathway", 0.061, 4.991,-1.442, 3.893, 0.0066,0.678,0.315),
c("GO:0032486","Rap protein signal transduction", 0.002, 6.347,-0.924, 2.458, 0.0344,0.763,0.318),
c("GO:0061430","bone trabecula morphogenesis", 0.003,-2.302, 6.815, 2.553, 0.0344,0.798,0.326),
c("GO:0001957","intramembranous ossification", 0.001, 0.332, 7.419, 2.167, 0.0160,0.766,0.343),
c("GO:0009120","deoxyribonucleoside metabolic process", 0.006,-4.177,-4.617, 2.908, 0.0187,0.800,0.350),
c("GO:0007386","compartment pattern specification", 0.001, 0.048, 7.245, 2.111, 0.0134,0.749,0.356),
c("GO:0045351","type I interferon biosynthetic process", 0.002, 0.347, 6.406, 2.505, 0.0318,0.743,0.357),
c("GO:0001821","histamine secretion", 0.001,-3.314, 2.812, 2.143, 0.0240,0.877,0.358),
c("GO:0042321","negative regulation of circadian sleep/wake cycle, sleep", 0.001, 2.446, 5.266, 1.845, 0.0134,0.663,0.369),
c("GO:0006744","ubiquinone biosynthetic process", 0.136,-5.223,-2.418, 4.242, 0.0396,0.786,0.389),
c("GO:0009146","purine nucleoside triphosphate catabolic process", 0.035,-3.242,-4.729, 3.647, 0.0134,0.749,0.391),
c("GO:0046085","adenosine metabolic process", 0.003,-4.584,-4.405, 2.524, 0.0266,0.804,0.409),
c("GO:0060405","regulation of penile erection", 0.000, 1.798, 6.090, 1.716, 0.0160,0.701,0.410),
c("GO:0045725","positive regulation of glycogen biosynthetic process", 0.003, 0.849,-3.136, 2.547, 0.0396,0.723,0.414),
c("GO:1901642","nucleoside transmembrane transport", 0.047,-3.483, 2.842, 3.781, 0.0187,0.860,0.432),
c("GO:0048702","embryonic neurocranium morphogenesis", 0.002, 0.142, 6.947, 2.444, 0.0187,0.711,0.433),
c("GO:0048311","mitochondrion distribution", 0.025,-3.646, 1.645, 3.502, 0.0318,0.901,0.436),
c("GO:0046060","dATP metabolic process", 0.001,-3.497,-5.142, 2.207, 0.0134,0.789,0.437),
c("GO:1903071","positive regulation of ER-associated ubiquitin-dependent protein catabolic process", 0.001, 4.542,-2.446, 2.137, 0.0187,0.618,0.440),
c("GO:0033327","Leydig cell differentiation", 0.002,-0.252, 6.634, 2.407, 0.0292,0.686,0.450),
c("GO:0048541","Peyer's patch development", 0.002, 1.607, 6.672, 2.316, 0.0134,0.674,0.470),
c("GO:0060368","regulation of Fc receptor mediated stimulatory signaling pathway", 0.001, 6.210, 1.102, 1.919, 0.0160,0.646,0.480),
c("GO:0006351","transcription, DNA-templated",10.659,-3.806,-3.709, 6.136, 0.0041,0.852,0.480),
c("GO:0072530","purine-containing compound transmembrane transport", 0.016,-4.166, 2.833, 3.304, 0.0213,0.934,0.483),
c("GO:0072531","pyrimidine-containing compound transmembrane transport", 0.019,-4.073, 2.881, 3.391, 0.0422,0.934,0.489),
c("GO:0070255","regulation of mucus secretion", 0.002, 1.155, 4.966, 2.504, 0.0240,0.662,0.494),
c("GO:0038180","nerve growth factor signaling pathway", 0.001, 6.875,-1.582, 2.190, 0.0213,0.729,0.496),
c("GO:0009168","purine ribonucleoside monophosphate biosynthetic process", 1.043,-3.650,-3.225, 5.127, 0.0136,0.734,0.509),
c("GO:0033601","positive regulation of mammary gland epithelial cell proliferation", 0.002, 2.665, 4.905, 2.398, 0.0187,0.604,0.519),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 0.199, 1.681,-2.816, 4.408, 0.0439,0.725,0.526),
c("GO:0001973","adenosine receptor signaling pathway", 0.005, 7.122,-0.404, 2.792, 0.0292,0.738,0.536),
c("GO:0038063","collagen-activated tyrosine kinase receptor signaling pathway", 0.005, 6.816,-0.631, 2.810, 0.0213,0.742,0.537),
c("GO:0008354","germ cell migration", 0.004,-0.730, 5.666, 2.729, 0.0213,0.700,0.552),
c("GO:0071320","cellular response to cAMP", 0.007, 7.142,-2.642, 2.937, 0.0071,0.810,0.553),
c("GO:0008277","regulation of G-protein coupled receptor protein signaling pathway", 0.024, 6.792, 0.170, 3.494, 0.0209,0.691,0.554),
c("GO:2000178","negative regulation of neural precursor cell proliferation", 0.005, 4.025, 1.444, 2.838, 0.0499,0.775,0.556),
c("GO:0071361","cellular response to ethanol", 0.003, 7.311,-2.794, 2.550, 0.0344,0.820,0.562),
c("GO:0043650","dicarboxylic acid biosynthetic process", 0.530,-5.261,-1.661, 4.833, 0.0344,0.767,0.571),
c("GO:0051412","response to corticosterone", 0.001, 6.485,-3.908, 2.053, 0.0422,0.848,0.572),
c("GO:0034127","regulation of MyD88-independent toll-like receptor signaling pathway", 0.000, 6.487, 1.509, 1.531, 0.0213,0.663,0.590),
c("GO:0002313","mature B cell differentiation involved in immune response", 0.003, 3.280, 4.691, 2.642, 0.0234,0.567,0.592),
c("GO:0032261","purine nucleotide salvage", 0.059,-4.165,-4.110, 3.878, 0.0187,0.768,0.593),
c("GO:0042118","endothelial cell activation", 0.002,-3.033, 0.342, 2.367, 0.0266,0.837,0.600),
c("GO:0003139","secondary heart field specification", 0.002, 0.004, 7.063, 2.358, 0.0240,0.711,0.600),
c("GO:0048387","negative regulation of retinoic acid receptor signaling pathway", 0.002, 6.096, 0.051, 2.394, 0.0213,0.681,0.601),
c("GO:0043249","erythrocyte maturation", 0.002, 2.358, 5.650, 2.461, 0.0292,0.610,0.607),
c("GO:0046130","purine ribonucleoside catabolic process", 0.036,-3.056,-4.878, 3.660, 0.0187,0.745,0.609),
c("GO:0001829","trophectodermal cell differentiation", 0.003,-0.657, 6.646, 2.521, 0.0396,0.689,0.620),
c("GO:0055015","ventricular cardiac muscle cell development", 0.003,-0.775, 6.758, 2.553, 0.0318,0.678,0.622),
c("GO:0002819","regulation of adaptive immune response", 0.025, 6.335, 1.438, 3.513, 0.0184,0.673,0.642),
c("GO:0051446","positive regulation of meiotic cell cycle", 0.007, 3.995, 2.576, 2.925, 0.0422,0.712,0.643),
c("GO:0014807","regulation of somitogenesis", 0.002, 1.564, 6.051, 2.342, 0.0213,0.651,0.660),
c("GO:0035414","negative regulation of catenin import into nucleus", 0.002, 1.162, 1.277, 2.497, 0.0266,0.750,0.669),
c("GO:0018243","protein O-linked glycosylation via threonine", 0.001,-1.054,-6.012, 2.201, 0.0187,0.817,0.676),
c("GO:0046654","tetrahydrofolate biosynthetic process", 0.101,-5.448,-2.197, 4.113, 0.0187,0.765,0.680),
c("GO:0060340","positive regulation of type I interferon-mediated signaling pathway", 0.002, 6.123, 0.385, 2.412, 0.0213,0.608,0.680),
c("GO:0046386","deoxyribose phosphate catabolic process", 0.023,-1.891,-6.345, 3.469, 0.0448,0.843,0.684),
c("GO:2000059","negative regulation of protein ubiquitination involved in ubiquitin-dependent protein catabolic process", 0.003, 2.146,-4.195, 2.603, 0.0318,0.743,0.686),
c("GO:0036066","protein O-linked fucosylation", 0.002,-0.988,-6.193, 2.307, 0.0370,0.815,0.689),
c("GO:0039530","MDA-5 signaling pathway", 0.002, 4.143, 3.197, 2.346, 0.0240,0.527,0.698),
c("GO:0071157","negative regulation of cell cycle arrest", 0.005, 3.360,-0.231, 2.804, 0.0473,0.687,0.699),
c("GO:0035023","regulation of Rho protein signal transduction", 0.125, 6.365, 0.101, 4.206, 0.0400,0.660,0.699));

bjp<-
theme(
  panel.border = element_rect(colour = "black", fill = NA, size = 2),
  plot.title = element_text(size = 16, face = "bold"),
  axis.text.y =  element_text(size = 14,face = "bold",color = "black"),
  axis.text.x =  element_text(size = 14,face = "bold",color = "black"),
  axis.title.y = element_text(size = 14,face = "bold"),
  axis.title.x = element_text(size = 14,face = "bold"),
  legend.text = element_text(size = 14,face = "bold"),
  legend.title = element_text(size = 14,face = "bold"),
  strip.text.x = element_text(size = 14,face = "bold"),
  strip.text.y = element_text(size = 14,face = "bold"),
  strip.background = element_rect(colour = "black", size = 2))

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
# --------------------------------------------------------------------------


# This is a table indicating which labels you would like to plot (all data will be plotted, but you need to indicate whether the label should be displayed), make this table manually.


label1 <- c("GO:0070375","ERK5 cascade", 0.001, 3.865,-3.823, 2.041, 0.0134,0.729,0.303)

label2 <- c("GO:0060070","canonical Wnt signaling pathway", 0.061, 4.991,-1.442, 3.893, 0.0066,0.678,0.315) 

label3 <- c("GO:0055015","ventricular cardiac muscle cell development", 0.003,-0.775, 6.758, 2.553, 0.0318,0.678,0.622) 

label4 <- c("GO:0003139","secondary heart field specification", 0.002, 0.004, 7.063, 2.358, 0.0240,0.711,0.600)

ex <- as.data.frame(rbind(label1, label2, label3), stringsAsFactors = FALSE)
colnames(ex) <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","value","uniqueness","dispensability")

ggplot(data = one.data)+
geom_point( aes( plot_X, plot_Y, fill = log10_p_value, size = plot_size),shape=21,color="black",stroke=0.7, alpha = 0.8) + guides(size=guide_legend(title="No. of genes")) + scale_fill_gradient2(low = muted("red"), mid = "white",
  high = muted("blue")) + guides(fill=guide_legend(title="P value"), size=guide_legend(title="No. of genes")) +
#scale_size( range=c(5, 20)) +
  labs (y = "Semantic space x", x = "Semantic space y") + theme_bw()  + geom_label_repel(ex, mapping = aes(as.numeric(ex$plot_X), as.numeric(ex$plot_Y), label = description),
    fontface = 'bold', color = 'black',
    box.padding = 0.6, point.padding = 1.6,
		size = 3,
    # Width of the line segments.
    segment.size = 0,
    # Strength of the repulsion force.
    force = 1)
```

```{r}
## Revigo liver


revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0002576","platelet degranulation", 0.003, 0.496,-3.539, 2.519, 0.0000,0.842,0.000),
c("GO:0008152","metabolic process",75.387, 0.813,-0.564, 6.986, 0.0308,0.998,0.000),
c("GO:0009820","alkaloid metabolic process", 0.006, 4.043, 1.159, 2.869, 0.0000,0.947,0.000),
c("GO:0042632","cholesterol homeostasis", 0.019,-3.975, 2.639, 3.381, 0.0000,0.818,0.000),
c("GO:0019835","cytolysis", 0.044, 0.583,-0.452, 3.749, 0.0007,0.971,0.018),
c("GO:0001887","selenium compound metabolic process", 0.008, 0.543,-0.357, 2.990, 0.0189,0.963,0.019),
c("GO:0042157","lipoprotein metabolic process", 0.210, 0.683,-0.584, 4.431, 0.0116,0.957,0.028),
c("GO:0017144","drug metabolic process", 0.058, 0.564,-0.386, 3.868, 0.0392,0.959,0.035),
c("GO:0070989","oxidative demethylation", 0.011, 1.605, 0.368, 3.133, 0.0064,0.912,0.041),
c("GO:0019740","nitrogen utilization", 0.085, 0.842,-0.775, 4.038, 0.0189,0.932,0.050),
c("GO:0019695","choline metabolic process", 0.016, 0.978, 0.110, 3.301, 0.0060,0.962,0.063),
c("GO:0048625","myoblast fate commitment", 0.001, 0.704,-6.082, 2.100, 0.0189,0.822,0.085),
c("GO:0006368","transcription elongation from RNA polymerase II promoter", 0.082, 1.989, 1.876, 4.021, 0.0139,0.890,0.085),
c("GO:0051081","nuclear envelope disassembly", 0.002, 0.688,-0.443, 2.338, 0.0116,0.883,0.087),
c("GO:0006957","complement activation, alternative pathway", 0.000,-6.310, 2.907, 1.771, 0.0003,0.689,0.121),
c("GO:0017187","peptidyl-glutamic acid carboxylation", 0.006,-0.167, 1.374, 2.865, 0.0011,0.918,0.125),
c("GO:0006629","lipid metabolic process", 3.522, 3.786, 1.634, 5.655, 0.0144,0.876,0.126),
c("GO:0034384","high-density lipoprotein particle clearance", 0.002,-3.947,-4.331, 2.294, 0.0001,0.733,0.128),
c("GO:1904683","regulation of metalloendopeptidase activity", 0.001,-3.073, 3.701, 2.210, 0.0017,0.790,0.131),
c("GO:0006641","triglyceride metabolic process", 0.038, 3.327, 1.871, 3.687, 0.0001,0.832,0.140),
c("GO:0051647","nucleus localization", 0.012,-0.140,-3.521, 3.198, 0.0117,0.927,0.141),
c("GO:0042738","exogenous drug catabolic process", 0.001,-1.097, 3.996, 2.193, 0.0009,0.834,0.156),
c("GO:0070141","response to UV-A", 0.002,-3.392, 1.485, 2.369, 0.0017,0.918,0.166),
c("GO:0006768","biotin metabolic process", 0.081, 5.718, 3.455, 4.018, 0.0007,0.805,0.167),
c("GO:1901160","primary amino compound metabolic process", 0.015, 4.703, 1.279, 3.290, 0.0455,0.944,0.178),
c("GO:0043627","response to estrogen", 0.010,-3.637, 1.577, 3.128, 0.0379,0.912,0.184),
c("GO:0099133","ATP hydrolysis coupled anion transmembrane transport", 0.271,-0.029,-4.342, 4.541, 0.0152,0.919,0.184),
c("GO:0010646","regulation of cell communication", 0.929,-6.450, 3.995, 5.076, 0.0173,0.784,0.190),
c("GO:0007041","lysosomal transport", 0.017, 0.052,-3.377, 3.341, 0.0395,0.947,0.196),
c("GO:0042158","lipoprotein biosynthetic process", 0.192, 0.957, 0.864, 4.391, 0.0189,0.919,0.199),
c("GO:1990535","neuron projection maintenance", 0.000, 0.267,-0.657, 1.301, 0.0275,0.897,0.200),
c("GO:0052652","cyclic purine nucleotide metabolic process", 0.094, 4.152, 4.182, 4.082, 0.0457,0.804,0.234),
c("GO:0006390","transcription from mitochondrial promoter", 0.005, 0.802, 1.042, 2.794, 0.0484,0.915,0.235),
c("GO:0006600","creatine metabolic process", 0.002, 5.427, 2.495, 2.328, 0.0374,0.857,0.244),
c("GO:0042127","regulation of cell proliferation", 0.313,-5.699, 3.430, 4.603, 0.0229,0.785,0.259),
c("GO:0016540","protein autoprocessing", 0.011,-0.432, 1.323, 3.147, 0.0191,0.924,0.274),
c("GO:0006476","protein deacetylation", 0.072,-0.361, 1.247, 3.967, 0.0455,0.913,0.278),
c("GO:0055114","oxidation-reduction process",15.060, 4.858, 2.263, 6.286, 0.0206,0.872,0.300),
c("GO:0015886","heme transport", 0.066,-0.143,-4.692, 3.927, 0.0044,0.860,0.302),
c("GO:0006461","protein complex assembly", 0.960, 0.086,-0.705, 5.091, 0.0285,0.920,0.312),
c("GO:0060052","neurofilament cytoskeleton organization", 0.002, 1.098,-0.364, 2.501, 0.0374,0.883,0.319),
c("GO:0010918","positive regulation of mitochondrial membrane potential", 0.001,-2.305, 1.655, 2.164, 0.0044,0.851,0.327),
c("GO:0033572","transferrin transport", 0.002,-0.385,-3.961, 2.449, 0.0180,0.915,0.333),
c("GO:0006524","alanine catabolic process", 0.025, 5.584, 3.621, 3.509, 0.0009,0.800,0.338),
c("GO:0097267","omega-hydroxylase P450 pathway", 0.000, 4.727, 2.654, 1.000, 0.0017,0.851,0.339),
c("GO:0019627","urea metabolic process", 0.051, 4.765, 2.756, 3.820, 0.0455,0.846,0.339),
c("GO:1903830","magnesium ion transmembrane transport", 0.088, 0.021,-4.086, 4.053, 0.0339,0.938,0.340),
c("GO:1904179","positive regulation of adipose tissue development", 0.001,-5.562,-3.251, 1.996, 0.0009,0.657,0.344),
c("GO:0007584","response to nutrient", 0.026,-3.941, 1.599, 3.530, 0.0010,0.865,0.345),
c("GO:0090400","stress-induced premature senescence", 0.002,-4.185,-2.548, 2.350, 0.0484,0.752,0.346),
c("GO:0042908","xenobiotic transport", 0.028,-0.145,-4.010, 3.556, 0.0064,0.876,0.348),
c("GO:0060021","palate development", 0.018,-0.008,-5.732, 3.363, 0.0445,0.883,0.355),
c("GO:0033860","regulation of NAD(P)H oxidase activity", 0.001,-3.702, 4.403, 2.057, 0.0275,0.831,0.356),
c("GO:0045725","positive regulation of glycogen biosynthetic process", 0.003,-0.905, 5.340, 2.547, 0.0039,0.686,0.357),
c("GO:0001907","killing by symbiont of host cells", 0.008,-3.276, 2.257, 3.001, 0.0009,0.818,0.362),
c("GO:0016310","phosphorylation", 7.764, 2.492, 2.058, 5.998, 0.0064,0.912,0.371),
c("GO:0050892","intestinal absorption", 0.006,-0.941,-7.157, 2.913, 0.0016,0.782,0.379),
c("GO:1900107","regulation of nodal signaling pathway", 0.002,-6.525, 3.443, 2.316, 0.0275,0.740,0.380),
c("GO:0008090","retrograde axonal transport", 0.003, 0.413,-4.290, 2.521, 0.0374,0.827,0.390),
c("GO:0008203","cholesterol metabolic process", 0.028, 5.004, 2.844, 3.554, 0.0004,0.835,0.402),
c("GO:0070301","cellular response to hydrogen peroxide", 0.019,-5.078, 2.109, 3.397, 0.0023,0.823,0.402),
c("GO:0007628","adult walking behavior", 0.008,-0.864,-5.875, 3.000, 0.0288,0.814,0.411),
c("GO:0043433","negative regulation of sequence-specific DNA binding transcription factor activity", 0.031,-3.638, 4.535, 3.600, 0.0023,0.715,0.412),
c("GO:0051340","regulation of ligase activity", 0.002,-2.775, 3.103, 2.391, 0.0455,0.849,0.421),
c("GO:0097190","apoptotic signaling pathway", 0.117,-4.388, 4.477, 4.177, 0.0357,0.702,0.427),
c("GO:0006869","lipid transport", 0.270,-0.359,-4.931, 4.539, 0.0084,0.825,0.430),
c("GO:0045717","negative regulation of fatty acid biosynthetic process", 0.007, 0.731, 5.462, 2.935, 0.0030,0.667,0.434),
c("GO:0051004","regulation of lipoprotein lipase activity", 0.003,-2.788, 2.773, 2.587, 0.0089,0.840,0.435),
c("GO:0006536","glutamate metabolic process", 0.201, 5.539, 3.254, 4.412, 0.0020,0.816,0.441),
c("GO:0018279","protein N-linked glycosylation via asparagine", 0.015, 1.838, 3.560, 3.284, 0.0410,0.833,0.446),
c("GO:0046185","aldehyde catabolic process", 0.048, 4.214, 2.777, 3.790, 0.0064,0.832,0.451),
c("GO:0032071","regulation of endodeoxyribonuclease activity", 0.003,-3.421, 4.017, 2.537, 0.0374,0.791,0.451),
c("GO:0008202","steroid metabolic process", 0.161, 3.336, 1.762, 4.315, 0.0005,0.851,0.451),
c("GO:0071156","regulation of cell cycle arrest", 0.013,-2.547, 3.556, 3.230, 0.0275,0.745,0.452),
c("GO:0071372","cellular response to follicle-stimulating hormone stimulus", 0.001,-4.403, 1.848, 2.220, 0.0118,0.859,0.460),
c("GO:0042940","D-amino acid transport", 0.003,-0.063,-4.288, 2.562, 0.0275,0.873,0.471),
c("GO:1903427","negative regulation of reactive oxygen species biosynthetic process", 0.002,-3.996, 3.252, 2.486, 0.0044,0.777,0.477),
c("GO:0042493","response to drug", 0.266,-5.108, 2.172, 4.534, 0.0449,0.863,0.477),
c("GO:0001889","liver development", 0.023,-1.101,-6.479, 3.471, 0.0009,0.766,0.478),
c("GO:0006545","glycine biosynthetic process", 0.079, 5.573, 3.375, 4.005, 0.0275,0.814,0.479),
c("GO:0019216","regulation of lipid metabolic process", 0.095,-2.218, 5.684, 4.086, 0.0052,0.745,0.482),
c("GO:0006591","ornithine metabolic process", 0.088, 5.478, 3.118, 4.052, 0.0484,0.825,0.483),
c("GO:0046838","phosphorylated carbohydrate dephosphorylation", 0.064, 1.908, 2.032, 3.915, 0.0455,0.926,0.484),
c("GO:0006739","NADP metabolic process", 0.357, 4.701, 4.332, 4.661, 0.0063,0.793,0.485),
c("GO:0006520","cellular amino acid metabolic process", 5.591, 5.069, 3.403, 5.856, 0.0148,0.779,0.487),
c("GO:2000343","positive regulation of chemokine (C-X-C motif) ligand 2 production", 0.001,-5.534,-2.610, 1.959, 0.0275,0.680,0.489),
c("GO:0021814","cell motility involved in cerebral cortex radial glia guided migration", 0.001,-0.848,-6.605, 2.238, 0.0189,0.729,0.492),
c("GO:0022900","electron transport chain", 0.564, 3.324, 2.185, 4.860, 0.0186,0.848,0.493),
c("GO:0090277","positive regulation of peptide hormone secretion", 0.013,-4.678, 0.230, 3.226, 0.0004,0.601,0.496),
c("GO:0015942","formate metabolic process", 0.065, 5.321, 3.235, 3.921, 0.0017,0.832,0.499),
c("GO:0046329","negative regulation of JNK cascade", 0.007,-4.599, 5.445, 2.940, 0.0063,0.605,0.500),
c("GO:0001503","ossification", 0.074,-0.710,-6.573, 3.979, 0.0413,0.791,0.504),
c("GO:0006898","receptor-mediated endocytosis", 0.095, 0.278,-4.169, 4.086, 0.0004,0.935,0.511),
c("GO:0090239","regulation of histone H4 acetylation", 0.003,-3.175, 3.088, 2.531, 0.0455,0.769,0.511),
c("GO:0060059","embryonic retina morphogenesis in camera-type eye", 0.003,-0.909,-6.515, 2.573, 0.0029,0.790,0.512),
c("GO:1902459","positive regulation of stem cell population maintenance", 0.001,-4.906,-2.513, 2.236, 0.0275,0.663,0.517),
c("GO:1902959","regulation of aspartic-type endopeptidase activity involved in amyloid precursor protein catabolic process", 0.000,-2.854, 3.852, 1.756, 0.0455,0.802,0.517),
c("GO:0031100","animal organ regeneration", 0.005,-1.364,-6.064, 2.792, 0.0402,0.788,0.526),
c("GO:0061620","glycolytic process through glucose-6-phosphate", 0.002, 5.004, 4.281, 2.290, 0.0455,0.798,0.529),
c("GO:0002679","respiratory burst involved in defense response", 0.001,-3.703, 1.450, 2.258, 0.0374,0.816,0.535),
c("GO:0034504","protein localization to nucleus", 0.129,-0.313,-4.647, 4.218, 0.0210,0.899,0.539),
c("GO:0014911","positive regulation of smooth muscle cell migration", 0.003,-5.348,-0.209, 2.620, 0.0461,0.695,0.540),
c("GO:0061138","morphogenesis of a branching epithelium", 0.042,-1.117,-6.445, 3.730, 0.0021,0.775,0.541),
c("GO:0010793","regulation of mRNA export from nucleus", 0.003,-3.669,-1.160, 2.625, 0.0275,0.749,0.541),
c("GO:0072049","comma-shaped body morphogenesis", 0.001,-1.363,-5.960, 2.173, 0.0189,0.801,0.543),
c("GO:0046498","S-adenosylhomocysteine metabolic process", 0.066, 5.725, 3.823, 3.925, 0.0275,0.808,0.544),
c("GO:0045620","negative regulation of lymphocyte differentiation", 0.008,-3.807,-2.279, 3.018, 0.0029,0.590,0.549),
c("GO:0019626","short-chain fatty acid catabolic process", 0.024, 5.002, 3.731, 3.490, 0.0189,0.779,0.556),
c("GO:0034356","NAD biosynthesis via nicotinamide riboside salvage pathway", 0.004, 4.983, 4.281, 2.691, 0.0235,0.828,0.560),
c("GO:0046620","regulation of organ growth", 0.020,-4.335,-3.406, 3.403, 0.0044,0.688,0.561),
c("GO:0006778","porphyrin-containing compound metabolic process", 0.457, 5.545, 2.444, 4.768, 0.0453,0.896,0.561),
c("GO:0010269","response to selenium ion", 0.001,-3.417, 1.446, 1.991, 0.0189,0.892,0.564),
c("GO:0090335","regulation of brown fat cell differentiation", 0.003,-3.134,-2.585, 2.531, 0.0274,0.735,0.566),
c("GO:0000098","sulfur amino acid catabolic process", 0.008, 5.398, 3.462, 3.030, 0.0064,0.815,0.566),
c("GO:1905668","positive regulation of protein localization to endosome", 0.001,-4.713,-0.861, 2.079, 0.0275,0.732,0.569),
c("GO:0031016","pancreas development", 0.021,-1.275,-6.229, 3.438, 0.0453,0.774,0.573),
c("GO:0071397","cellular response to cholesterol", 0.001,-4.311, 1.853, 2.272, 0.0191,0.857,0.574),
c("GO:0060712","spongiotrophoblast layer development", 0.003,-1.057,-6.380, 2.533, 0.0191,0.790,0.578),
c("GO:0031659","positive regulation of cyclin-dependent protein serine/threonine kinase activity involved in G1/S transition of mitotic cell cycle", 0.002,-3.538, 4.926, 2.350, 0.0275,0.670,0.578),
c("GO:1903896","positive regulation of IRE1-mediated unfolded protein response", 0.001,-6.708, 3.143, 1.857, 0.0275,0.671,0.580),
c("GO:0071560","cellular response to transforming growth factor beta stimulus", 0.050,-4.466, 1.856, 3.804, 0.0127,0.832,0.582),
c("GO:0035733","hepatic stellate cell activation", 0.000, 0.492,-0.383, 1.748, 0.0455,0.898,0.584),
c("GO:0034763","negative regulation of transmembrane transport", 0.014,-4.429,-0.887, 3.251, 0.0274,0.743,0.585),
c("GO:0042167","heme catabolic process", 0.002, 5.014, 2.712, 2.408, 0.0374,0.834,0.586),
c("GO:0006853","carnitine shuttle", 0.000,-0.241,-4.610, 1.114, 0.0089,0.860,0.587),
c("GO:0007411","axon guidance", 0.066,-3.249,-3.116, 3.930, 0.0223,0.617,0.587),
c("GO:0048013","ephrin receptor signaling pathway", 0.016,-6.783, 3.560, 3.308, 0.0309,0.752,0.593),
c("GO:0034340","response to type I interferon", 0.008,-5.227, 1.865, 3.007, 0.0117,0.821,0.598),
c("GO:0031325","positive regulation of cellular metabolic process", 1.004,-6.818, 4.034, 5.110, 0.0233,0.668,0.602),
c("GO:0042177","negative regulation of protein catabolic process", 0.025,-2.311, 5.307, 3.513, 0.0177,0.698,0.606),
c("GO:1900016","negative regulation of cytokine production involved in inflammatory response", 0.002,-5.942,-1.181, 2.391, 0.0191,0.603,0.606),
c("GO:2000463","positive regulation of excitatory postsynaptic potential", 0.003,-6.002, 3.548, 2.606, 0.0220,0.638,0.607),
c("GO:0098902","regulation of membrane depolarization during action potential", 0.001,-3.098, 2.354, 2.188, 0.0455,0.827,0.608),
c("GO:1902237","positive regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway", 0.002,-6.892, 3.889, 2.324, 0.0089,0.647,0.617),
c("GO:0008635","activation of cysteine-type endopeptidase activity involved in apoptotic process by cytochrome c", 0.001,-5.340, 4.602, 2.279, 0.0484,0.665,0.619),
c("GO:2000271","positive regulation of fibroblast apoptotic process", 0.002,-6.128, 4.110, 2.320, 0.0374,0.726,0.621),
c("GO:0050730","regulation of peptidyl-tyrosine phosphorylation", 0.045,-2.927, 5.206, 3.759, 0.0115,0.746,0.627),
c("GO:0014068","positive regulation of phosphatidylinositol 3-kinase signaling", 0.013,-6.835, 3.378, 3.216, 0.0477,0.656,0.627),
c("GO:0032148","activation of protein kinase B activity", 0.004,-3.889, 4.818, 2.738, 0.0411,0.689,0.628),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 0.199,-3.286, 4.570, 4.408, 0.0343,0.704,0.628),
c("GO:2000659","regulation of interleukin-1-mediated signaling pathway", 0.001,-6.610, 3.398, 2.117, 0.0275,0.727,0.629),
c("GO:0043392","negative regulation of DNA binding", 0.014,-3.451, 2.990, 3.256, 0.0324,0.830,0.631),
c("GO:0042426","choline catabolic process", 0.001, 4.091, 2.217, 2.230, 0.0275,0.898,0.632),
c("GO:1900119","positive regulation of execution phase of apoptosis", 0.003,-5.254, 3.939, 2.528, 0.0152,0.720,0.635),
c("GO:0046487","glyoxylate metabolic process", 0.049, 5.512, 3.224, 3.794, 0.0484,0.832,0.635),
c("GO:0030917","midbrain-hindbrain boundary development", 0.003,-0.916,-6.603, 2.576, 0.0064,0.776,0.636),
c("GO:0031663","lipopolysaccharide-mediated signaling pathway", 0.010,-6.114, 3.001, 3.128, 0.0363,0.713,0.638),
c("GO:1901679","nucleotide transmembrane transport", 0.011,-0.193,-4.465, 3.131, 0.0374,0.862,0.638),
c("GO:0032933","SREBP signaling pathway", 0.008,-6.008, 4.192, 2.989, 0.0484,0.656,0.640),
c("GO:2000646","positive regulation of receptor catabolic process", 0.001,-2.717, 5.596, 1.881, 0.0189,0.731,0.642),
c("GO:0071276","cellular response to cadmium ion", 0.009,-4.925, 2.123, 3.077, 0.0040,0.848,0.644),
c("GO:0000060","protein import into nucleus, translocation", 0.012,-0.358,-4.229, 3.175, 0.0286,0.853,0.645),
c("GO:0006548","histidine catabolic process", 0.052, 5.484, 3.853, 3.828, 0.0044,0.789,0.649),
c("GO:0034638","phosphatidylcholine catabolic process", 0.002, 4.403, 3.693, 2.328, 0.0374,0.793,0.652),
c("GO:1901523","icosanoid catabolic process", 0.000, 5.088, 3.154, 1.732, 0.0189,0.841,0.653),
c("GO:0006559","L-phenylalanine catabolic process", 0.031, 5.469, 3.690, 3.596, 0.0118,0.794,0.657),
c("GO:0051642","centrosome localization", 0.007,-0.128,-3.483, 2.973, 0.0190,0.928,0.658),
c("GO:0010040","response to iron(II) ion", 0.002,-4.034, 1.687, 2.407, 0.0189,0.886,0.658),
c("GO:0000042","protein targeting to Golgi", 0.021,-0.296,-4.424, 3.422, 0.0374,0.900,0.670),
c("GO:0090201","negative regulation of release of cytochrome c from mitochondria", 0.003,-5.536, 3.408, 2.591, 0.0285,0.651,0.672),
c("GO:0006855","drug transmembrane transport", 0.189,-4.897,-2.034, 4.384, 0.0138,0.752,0.672),
c("GO:0009887","animal organ morphogenesis", 0.264,-1.342,-6.260, 4.530, 0.0333,0.742,0.673),
c("GO:0006879","cellular iron ion homeostasis", 0.110,-1.935, 4.780, 4.148, 0.0002,0.750,0.674),
c("GO:0030178","negative regulation of Wnt signaling pathway", 0.037,-5.011, 4.290, 3.681, 0.0367,0.636,0.676),
c("GO:0007262","STAT protein import into nucleus", 0.001,-5.800,-0.845, 2.223, 0.0484,0.729,0.677),
c("GO:0045739","positive regulation of DNA repair", 0.013,-5.991, 3.701, 3.205, 0.0123,0.635,0.678),
c("GO:0034142","toll-like receptor 4 signaling pathway", 0.006,-6.777, 3.183, 2.892, 0.0191,0.642,0.679),
c("GO:0046326","positive regulation of glucose import", 0.007,-6.120,-0.943, 2.979, 0.0287,0.694,0.680),
c("GO:0042773","ATP synthesis coupled electron transport", 0.221, 4.104, 4.074, 4.452, 0.0456,0.794,0.685),
c("GO:0010886","positive regulation of cholesterol storage", 0.002,-5.084,-0.414, 2.320, 0.0374,0.711,0.685),
c("GO:0001568","blood vessel development", 0.136,-1.287,-6.290, 4.243, 0.0452,0.758,0.689),
c("GO:0006505","GPI anchor metabolic process", 0.102, 3.442, 3.385, 4.115, 0.0456,0.805,0.690),
c("GO:0046439","L-cysteine metabolic process", 0.039, 6.129, 3.195, 3.699, 0.0374,0.827,0.692),
c("GO:0051897","positive regulation of protein kinase B signaling", 0.016,-6.848, 3.394, 3.304, 0.0485,0.653,0.692),
c("GO:0032287","peripheral nervous system myelin maintenance", 0.002,-0.055,-6.738, 2.468, 0.0374,0.750,0.693),
c("GO:1904468","negative regulation of tumor necrosis factor secretion", 0.001,-3.544,-4.697, 1.982, 0.0374,0.625,0.695),
c("GO:1903874","ferrous iron transmembrane transport", 0.028, 0.089,-4.022, 3.563, 0.0374,0.931,0.700));



one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
# --------------------------------------------------------------------------


# This is a table indicating which labels you would like to plot (all data will be plotted, but you need to indicate whether the label should be displayed), make this table manually.


label1 <- c("GO:0042632","cholesterol homeostasis", 0.019,-3.975, 2.639, 3.381, 0.0000,0.818,0.000)

label2 <- c("GO:0017144","drug metabolic process", 0.058, 0.564,-0.386, 3.868, 0.0392,0.959,0.035) 

label3 <- c("GO:0006629","lipid metabolic process", 3.522, 3.786, 1.634, 5.655, 0.0144,0.876,0.126)



ex <- as.data.frame(rbind(label1, label2, label3), stringsAsFactors = FALSE)
colnames(ex) <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","value","uniqueness","dispensability")

ggplot(data = one.data)+
geom_point( aes( plot_X, plot_Y, fill = log10_p_value, size = plot_size),shape=21,color="black",stroke=0.7, alpha = 0.8) + guides(size=guide_legend(title="No. of genes")) + scale_fill_gradient2(low = muted("red"), mid = "white",
  high = muted("blue")) + guides(fill=guide_legend(title="P value"), size=guide_legend(title="No. of genes")) +
#scale_size( range=c(5, 20)) +
  labs (y = "Semantic space x", x = "Semantic space y") + theme_bw()  + geom_label_repel(ex, mapping = aes(as.numeric(ex$plot_X), as.numeric(ex$plot_Y), label = description),
    fontface = 'bold', color = 'black',
    box.padding = 0.6, point.padding = 1.6,
		size = 3,
    # Width of the line segments.
    segment.size = 0,
    # Strength of the repulsion force.
    force = 1)
```

