<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2016-07-20" />

<title>Analysis_of_potential_confounders_gene_expression</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis_of_potential_confounders_gene_expression</h1>
<h4 class="author">Lauren Blake</h4>
<h4 class="date">July 20, 2016</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes">PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes</a><ul>
<li><a href="#find-gene-pcs">Find gene PCs</a><ul>
<li><a href="#relationship-between-pc1">Relationship between PC1</a></li>
<li><a href="#part-two-for-the-variables-that-correlate-see-if-these-segregate-with-either-species-or-tissue">PART TWO: For the variable(s) that correlate, see if these segregate with either species or tissue</a></li>
<li><a href="#part-three-which-variables-to-put-in-the-model">PART THREE: Which variables to put in the model?</a></li>
</ul></li>
</ul></li>
</ul>
</div>

<div id="part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes" class="section level1">
<h1>PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes</h1>
<div id="find-gene-pcs" class="section level2">
<h2>Find gene PCs</h2>
<pre class="r"><code># Load gene expression levels

cpm_12184 &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/cpm_12184.txt&quot;)

dim(cpm_12184)</code></pre>
<pre><code>## [1] 12184    47</code></pre>
<pre class="r"><code># Take gene PCs

pca_genes &lt;- prcomp(t(cpm_12184), scale = T)
scores &lt;- pca_genes$x
gene_pcs &lt;- scores </code></pre>
<div id="load-libraries-and-data" class="section level4">
<h4>Load libraries and data</h4>
<pre class="r"><code># Load libraries

# install.packages(&quot;polycor&quot;)
library(&quot;polycor&quot;)

# Load list of technical factors
RNA_seq_info &lt;- read.csv(&quot;/Users/laurenblake/Dropbox/primate_BS-seq_project/Correlation with PCs - Lauren/RNA_seq_info_testing.csv&quot;)
RNA_seq_info &lt;- as.data.frame(RNA_seq_info)

dim(RNA_seq_info)</code></pre>
<pre><code>## [1] 47 31</code></pre>
<pre class="r"><code>#Create plots for each of the possible confounders versus PCs 1-5

#pdf(&#39;../data/VarVsGenePCs.pdf&#39;)

#for (i in 3:length(RNA_seq_info)) { 
#  par(mfrow=c(1,5))
#  plot(RNA_seq_info[,i], gene_pcs[,1],  ylab = &quot;PC1&quot;, xlab = &quot; &quot;)
#  plot(RNA_seq_info[,i], gene_pcs[,2],  ylab = &quot;PC2&quot;, xlab = &quot; &quot;)
#  plot(RNA_seq_info[,i], gene_pcs[,3],  ylab = &quot;PC3&quot;, xlab = &quot; &quot;)
#  plot(RNA_seq_info[,i], gene_pcs[,4],  ylab = &quot;PC4&quot;, xlab = &quot; &quot;)
#  plot(RNA_seq_info[,i], gene_pcs[,5],  ylab = &quot;PC5&quot;, xlab = &quot; &quot;)
#  title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
#  mtext(substitute(paste(&#39;PCs vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
  
#}
#dev.off()</code></pre>
</div>
<div id="relationship-between-pc1" class="section level3">
<h3>Relationship between PC1</h3>
<pre class="r"><code>checkPC1 &lt;- lm(gene_pcs[,1] ~ as.factor(RNA_seq_info$Tissue))

summary(checkPC1)</code></pre>
<pre><code>## 
## Call:
## lm(formula = gene_pcs[, 1] ~ as.factor(RNA_seq_info$Tissue))
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -57.900 -12.990  -2.963  16.862  39.728 
## 
## Coefficients:
##                                      Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)                           -29.033      7.043  -4.122 0.000168
## as.factor(RNA_seq_info$Tissue)kidney   36.593      9.751   3.753 0.000519
## as.factor(RNA_seq_info$Tissue)liver    98.649      9.751  10.117 6.06e-13
## as.factor(RNA_seq_info$Tissue)lung    -21.530      9.751  -2.208 0.032625
##                                         
## (Intercept)                          ***
## as.factor(RNA_seq_info$Tissue)kidney ***
## as.factor(RNA_seq_info$Tissue)liver  ***
## as.factor(RNA_seq_info$Tissue)lung   *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 23.36 on 43 degrees of freedom
## Multiple R-squared:  0.8081, Adjusted R-squared:  0.7947 
## F-statistic: 60.35 on 3 and 43 DF,  p-value: 1.859e-15</code></pre>
<pre class="r"><code>checkPC1 &lt;- lm(gene_pcs[,2] ~ as.factor(RNA_seq_info$Tissue))

summary(checkPC1)</code></pre>
<pre><code>## 
## Call:
## lm(formula = gene_pcs[, 2] ~ as.factor(RNA_seq_info$Tissue))
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -48.175 -22.203   7.079  14.446  52.474 
## 
## Coefficients:
##                                      Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)                           -65.964      7.823  -8.432 1.17e-10
## as.factor(RNA_seq_info$Tissue)kidney   68.287     10.831   6.305 1.32e-07
## as.factor(RNA_seq_info$Tissue)liver    90.714     10.831   8.375 1.40e-10
## as.factor(RNA_seq_info$Tissue)lung     99.357     10.831   9.173 1.11e-11
##                                         
## (Intercept)                          ***
## as.factor(RNA_seq_info$Tissue)kidney ***
## as.factor(RNA_seq_info$Tissue)liver  ***
## as.factor(RNA_seq_info$Tissue)lung   ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 25.95 on 43 degrees of freedom
## Multiple R-squared:  0.7034, Adjusted R-squared:  0.6827 
## F-statistic: 33.99 on 3 and 43 DF,  p-value: 2.019e-11</code></pre>
<pre class="r"><code>checkPC2 &lt;- lm(gene_pcs[,2] ~ as.factor(RNA_seq_info$Species))

summary(checkPC2)</code></pre>
<pre><code>## 
## Call:
## lm(formula = gene_pcs[, 2] ~ as.factor(RNA_seq_info$Species))
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -82.400 -11.763   9.765  27.362  47.606 
## 
## Coefficients:
##                                               Estimate Std. Error t value
## (Intercept)                                      8.849     10.064   0.879
## as.factor(RNA_seq_info$Species)human            15.567     14.467   1.076
## as.factor(RNA_seq_info$Species)rhesus macaque  -40.588     14.232  -2.852
##                                               Pr(&gt;|t|)   
## (Intercept)                                     0.3840   
## as.factor(RNA_seq_info$Species)human            0.2878   
## as.factor(RNA_seq_info$Species)rhesus macaque   0.0066 **
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 40.25 on 44 degrees of freedom
## Multiple R-squared:  0.2696, Adjusted R-squared:  0.2364 
## F-statistic: 8.119 on 2 and 44 DF,  p-value: 0.0009972</code></pre>
<div id="testing-association-between-a-particular-variable-and-pcs-with-a-linear-model" class="section level4">
<h4>Testing association between a particular variable and PCs with a linear model</h4>
<pre class="r"><code>#Make an empty matrix to put all of the data in

pvaluesandr2 = matrix(data = NA, nrow = 5, ncol = 28*2, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Species&quot;, &quot;Species R^2&quot;, &quot;Tissue&quot;, &quot;Tissue R^2&quot;, &quot;Individual&quot;, &quot;Ind. R^2&quot;, &quot;RNA extraction date&quot;, &quot;Extract. date R^2&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multi. index R^2&quot;, &quot;Multiplexing mix codes&quot;, &quot;Multi. mix R^2&quot;, &quot;Sequencing location: at UCGF&quot;, &quot;Location R^2&quot;, &quot;Total number of reads sequenced&quot;, &quot;Read Seq R^2&quot;, &quot;Percentage of bps trimmed (adapters)&quot;, &quot;Trimmed R^2&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Removed R^2&quot;, &quot;Maximum read length after trimming&quot;, &quot;Max. length R^2&quot;, &quot;Total number of reads processed in tophat&quot;, &quot;Reads processed R^2&quot;, &quot;Total number of mapped reads&quot;, &quot;Mapped reads R^2&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Perc. junction R^2&quot;, &quot;Number of junctions&quot;, &quot;Num. junctions R^2&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;OrthoExons R^2&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Ortho exons read R^2&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;Ortho genes read R^2&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RNA conc. R^2&quot;, &quot;RIN score&quot;, &quot;RIN R^2&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Lib. conc. R^2&quot;, &quot;Library fragments size (bp)&quot;, &quot;Lib. frag. R^2&quot;,  &quot;Sex&quot;, &quot;R^2&quot;, &quot;Age quantile&quot;, &quot;R^2&quot;, &quot;Hours postmortem&quot;, &quot;R^2&quot;, &quot;Cause of death&quot;, &quot;R^2&quot;, &quot;Possible conditions affecting tissues&quot;, &quot;R^2&quot;, &quot;Source&quot;, &quot;R^2&quot;)))

#Check lm to see how well the variables containing ordinal data are correlated with a PC

#For PCs 1-5
j=1
for (i in 3:9){
  for (j in 1:5){
  
  checkPC1 &lt;- lm(gene_pcs[,j] ~ as.factor(RNA_seq_info[,i]))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

#Put the summary statistics into the matrix w

  pvaluesandr2[j, (2*i-5)] &lt;- p_fstat
  pvaluesandr2[j, (2*i-4)] &lt;- r2_value
  
  }
}

#Check lm to see how well the variables containing numerical data are correlated with a PC

j=1
for (j in 1:5){
  
  for (i in 10:24){
      
  checkPC &lt;- lm(gene_pcs[,j] ~ RNA_seq_info[,i])
      
  #Get the summary statistics from it
  summary(checkPC)
  
  #Get the p-value of the F-statistic
  summary(checkPC)$fstatistic
  
  fstat &lt;- as.data.frame(summary(checkPC)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
  #Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC)$r.squared
  
  #Put the summary statistics into the matrix a
  
  pvaluesandr2[j, 2*i-5] &lt;- p_fstat
  pvaluesandr2[j, 2*i-4] &lt;- r2_value
  
  i = i+1
  }
  j=j+1
}

j=1
for (i in 25:30){
  for (j in 1:5){
  
  checkPC1 &lt;- lm(gene_pcs[,j] ~ as.factor(RNA_seq_info[,i]))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

#Put the summary statistics into the matrix w

  pvaluesandr2[j, (2*i-5)] &lt;- p_fstat
  pvaluesandr2[j, (2*i-4)] &lt;- r2_value
  
  }
}</code></pre>
</div>
<div id="test-for-potential-violations-of-the-assumptions-of-the-linear-model" class="section level4">
<h4>Test for potential violations of the assumptions of the linear model</h4>
<pre class="r"><code>#Plot the residuals to look for violations of the assumptions of lm

pdf(&#39;../data/Residuals_vs_var_GenePCs.pdf&#39;)

i = 3
  for (i in 3:9){
    par(mfrow=c(1,5))
   
    checkPC1 &lt;- lm(gene_pcs[,1] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)

    checkPC2 &lt;- lm(gene_pcs[,2] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC2),  ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
    
    checkPC3 &lt;- lm(gene_pcs[,3] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC3),  ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
    
    checkPC4 &lt;- lm(gene_pcs[,4] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC4),  ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
    
    checkPC5 &lt;- lm(gene_pcs[,5] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC5),  ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
    
    title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
    mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
    
  }

#Need to divide this up because #22 has an NA in it 
i = 10
  for (i in 10:21){
    
    par(mfrow=c(1,5))
    
    checkPC1 &lt;- lm(gene_pcs[,1] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)
    
    checkPC2 &lt;- lm(gene_pcs[,2] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC2), ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
    
    checkPC3 &lt;- lm(gene_pcs[,3] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC3), ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
    
    checkPC4 &lt;- lm(gene_pcs[,4] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC4), ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
    
    checkPC5 &lt;- lm(gene_pcs[,5] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC5), ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
    
    title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
    mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
    
  }

#Skip #22 because there is an NA in it

for (i in 23:24){
  
  par(mfrow=c(1,5))
  
  checkPC1 &lt;- lm(gene_pcs[,1] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)
  
  checkPC2 &lt;- lm(gene_pcs[,2] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC2), ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
  
  checkPC3 &lt;- lm(gene_pcs[,3] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC3), ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
  
  checkPC4 &lt;- lm(gene_pcs[,4] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC4), ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
  
  checkPC5 &lt;- lm(gene_pcs[,5] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC5), ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
  
  title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
  mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
  
}</code></pre>
</div>
<div id="look-at-significant-p-values" class="section level4">
<h4>Look at significant p-values</h4>
<pre class="r"><code>#Get rid of the R^2 values
all_col &lt;- 1:ncol(pvaluesandr2)
matrix_pval &lt;- pvaluesandr2[ , all_col%%2==1 ]

#Find which variables/PC combinations are p-value &lt; 0.05
TorF_matrix &lt;- matrix_pval &lt;=0.05
sum(matrix_pval &lt;= 0.05/(28*5))</code></pre>
<pre><code>## [1] 21</code></pre>
<pre class="r"><code>#Find which variables/PC combinations are p-value &lt; 0.05 not including species or tissue (because those explain PC1 and PC2)
matrix_pval_no_tissue_or_species = matrix_pval[ , 3:28]
matrix_pval_no_tissue_or_species</code></pre>
<pre><code>##       Individual RNA extraction date Multiplexing index sequence
## PC1 8.577332e-01        1.933096e-01                3.202755e-05
## PC2 3.089533e-01        8.775476e-03                2.494550e-03
## PC3 4.210981e-03        6.385467e-06                1.665273e-01
## PC4 9.971820e-01        6.167683e-01                1.541532e-09
## PC5 4.996004e-15        0.000000e+00                9.542053e-01
##     Multiplexing mix codes Sequencing location: at UCGF
## PC1           3.828374e-01                   0.40556453
## PC2           8.784915e-03                   0.85463260
## PC3           3.806629e-05                   0.51996073
## PC4           1.927352e-01                   0.62874566
## PC5           3.820715e-01                   0.07672979
##     Total number of reads sequenced Percentage of bps trimmed (adapters)
## PC1                       0.6623602                           0.43484612
## PC2                       0.8367218                           0.03093471
## PC3                       0.9498725                           0.56268886
## PC4                       0.1491588                           0.55118204
## PC5                       0.6813582                           0.03486093
##     Number of reads shorter than 20bp removed
## PC1                                0.32598741
## PC2                                0.02745845
## PC3                                0.68306619
## PC4                                0.47137991
## PC5                                0.04640738
##     Maximum read length after trimming
## PC1                         0.33637965
## PC2                         0.92113549
## PC3                         0.75683691
## PC4                         0.59778768
## PC5                         0.05145475
##     Total number of reads processed in tophat Total number of mapped reads
## PC1                                 0.7051527                    0.8420494
## PC2                                 0.9463660                    0.0369636
## PC3                                 0.9269870                    0.3729374
## PC4                                 0.1578441                    0.0930245
## PC5                                 0.5848903                    0.7499138
##     Percentage of mapped reads overlapping a junction Number of junctions
## PC1                                      1.630216e-06         0.002629987
## PC2                                      5.446252e-01         0.003137455
## PC3                                      6.527941e-01         0.243584612
## PC4                                      1.540735e-02         0.179619438
## PC5                                      6.196622e-01         0.089573389
##     Number of reads mapped on orthologous exons
## PC1                                  0.12500946
## PC2                                  0.12756466
## PC3                                  0.02997633
## PC4                                  0.01379170
## PC5                                  0.08964825
##     Number of orthologous exons with at least 1 mapped read
## PC1                                             0.004114919
## PC2                                             0.059787325
## PC3                                             0.000255304
## PC4                                             0.078055122
## PC5                                             0.077428146
##     Number of orthologous genes with at least 1 mapped read
## PC1                                             0.005239234
## PC2                                             0.008480924
## PC3                                             0.005243010
## PC4                                             0.055612828
## PC5                                             0.006306295
##     RNA concentration (ng/uL)    RIN score Library concentration (ng/uL)
## PC1              0.0148708408 3.193556e-01                    0.05256745
## PC2              0.0578165224 4.364496e-03                    0.01417761
## PC3              0.0342043520 3.532540e-06                    0.74147481
## PC4              0.3831733717 2.661127e-02                    0.79123942
## PC5              0.0001948362 4.790354e-01                    0.02505369
##     Library fragments size (bp)          Sex Age quantile Hours postmortem
## PC1                 0.944720057 8.577332e-01    0.4981614       0.31353986
## PC2                 0.060512968 3.089533e-01    0.3851076       0.05951400
## PC3                 0.400971333 4.210981e-03    0.3017935       0.00477844
## PC4                 0.124460815 9.971820e-01    0.7967752       0.73082727
## PC5                 0.001394967 4.996004e-15    0.0619115       0.10747192
##     Cause of death Possible conditions affecting tissues      Source
## PC1   2.915113e-01                           0.596620345 0.596620345
## PC2   1.971076e-02                           0.103215509 0.103215509
## PC3   2.234941e-05                           0.000418567 0.000418567
## PC4   8.075915e-01                           0.968399378 0.968399378
## PC5   0.000000e+00                           0.000000000 0.000000000</code></pre>
<pre class="r"><code>sum(matrix_pval_no_tissue_or_species &lt;= 0.05/(28*5))</code></pre>
<pre><code>## [1] 15</code></pre>
<pre class="r"><code>#Distribution of p-values adjusted by FDR not including species or tissue

fdr_val = p.adjust(matrix_pval_no_tissue_or_species, method = &quot;fdr&quot;, n = length(matrix_pval_no_tissue_or_species))
fdr_val_order = fdr_val[order(fdr_val)]
hist(fdr_val_order, ylab = &quot;BH-adjusted p-values&quot;, main = &quot;Distribution of Benjamini and Hochberg adjusted p-values&quot;, breaks = 10)</code></pre>
<p><img src="Tech_factor_analysis1_gene_exp_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>matrix_fdr_val = matrix(data = fdr_val, nrow = 5, ncol = 26, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Sequencing location: at UCGF&quot;, &quot;Total number of reads sequenced&quot;, &quot;Percentage of bps trimmed (adapters)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Maximum read length after trimming&quot;, &quot;Total number of reads processed in tophat&quot;, &quot;Total number of mapped reads&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;,  &quot;Library fragments size (bp)&quot;,   &quot;Sex&quot;,  &quot;Age quantile&quot;,  &quot;Hours postmortem&quot;,  &quot;Cause of death&quot;,  &quot;Possible conditions affecting tissues&quot;, &quot;Source&quot;))) 

matrix_fdr_val</code></pre>
<pre><code>##       Individual RNA extraction date Multiplexing index sequence
## PC1 9.292110e-01        3.490312e-01                3.469651e-04
## PC2 5.216095e-01        3.568872e-02                1.706798e-02
## PC3 2.269538e-02        8.301107e-05                3.137471e-01
## PC4 9.971820e-01        7.897656e-01                2.862845e-08
## PC5 1.082467e-13        0.000000e+00                9.835306e-01
##     Multiplexing mix codes Sequencing location: at UCGF
## PC1           0.5821393960                    0.5991294
## PC2           0.0356887174                    0.9292110
## PC3           0.0003806629                    0.7268268
## PC4           0.3490312399                    0.7935625
## PC5           0.5821393960                    0.1780205
##     Total number of reads sequenced Percentage of bps trimmed (adapters)
## PC1                       0.8200650                           0.63516849
## PC2                       0.9292110                           0.09575031
## PC3                       0.9835306                           0.76197450
## PC4                       0.2894126                           0.75424910
## PC5                       0.8298935                           0.10299819
##     Number of reads shorter than 20bp removed
## PC1                                0.52972955
## PC2                                0.08923997
## PC3                                0.82989350
## PC4                                0.68088209
## PC5                                0.13115129
##     Maximum read length after trimming
## PC1                          0.5398686
## PC2                          0.9835306
## PC3                          0.8784714
## PC4                          0.7771240
## PC5                          0.1423217
##     Total number of reads processed in tophat Total number of mapped reads
## PC1                                 0.8487949                    0.9292110
## PC2                                 0.9835306                    0.1067837
## PC3                                 0.9835306                    0.5821394
## PC4                                 0.3017608                    0.2015531
## PC5                                 0.7771240                    0.8782774
##     Percentage of mapped reads overlapping a junction Number of junctions
## PC1                                      2.649101e-05          0.01709491
## PC2                                      7.532051e-01          0.01942234
## PC3                                      8.159926e-01          0.43378082
## PC4                                      5.563764e-02          0.33357896
## PC5                                      7.897656e-01          0.19753005
##     Number of reads mapped on orthologous exons
## PC1                                  0.25001893
## PC2                                  0.25126373
## PC3                                  0.09504690
## PC4                                  0.05420852
## PC5                                  0.19753005
##     Number of orthologous exons with at least 1 mapped read
## PC1                                             0.022695382
## PC2                                             0.148428035
## PC3                                             0.002212635
## PC4                                             0.178020453
## PC5                                             0.178020453
##     Number of orthologous genes with at least 1 mapped read
## PC1                                              0.02434255
## PC2                                              0.03568872
## PC3                                              0.02434255
## PC4                                              0.14754424
## PC5                                              0.02826960
##     RNA concentration (ng/uL)    RIN score Library concentration (ng/uL)
## PC1               0.055234552 5.255219e-01                    0.14237018
## PC2               0.148428035 2.269538e-02                    0.05420852
## PC3               0.102998195 5.102557e-05                    0.87628841
## PC4               0.582139396 8.870424e-02                    0.90860325
## PC5               0.001809193 6.843363e-01                    0.08570998
##     Library fragments size (bp)          Sex Age quantile Hours postmortem
## PC1                  0.98353062 9.292110e-01    0.7039238        0.5225664
## PC2                  0.14842803 5.216095e-01    0.5821394        0.1484280
## PC3                  0.59912943 2.269538e-02    0.5216095        0.0238922
## PC4                  0.25001893 9.971820e-01    0.9086032        0.8716289
## PC5                  0.01007476 1.082467e-13    0.1490462        0.2217675
##     Cause of death Possible conditions affecting tissues      Source
## PC1   0.5121143899                           0.777123978 0.777123978
## PC2   0.0692540265                           0.216419615 0.216419615
## PC3   0.0002641294                           0.003200806 0.003200806
## PC4   0.9129295702                           0.983530618 0.983530618
## PC5   0.0000000000                           0.000000000 0.000000000</code></pre>
<pre class="r"><code>write.csv(matrix_fdr_val, &quot;../data/matrix_fdr_val_RNA.csv&quot;)


# Number of values significant at 10% FDR not including species or tissue

sum(matrix_fdr_val &lt;= 0.1)</code></pre>
<pre><code>## [1] 42</code></pre>
<pre class="r"><code>#Get the coordinates of which variables/PC combinations are significant at FDR 10%

TorF_matrix_fdr &lt;- matrix_fdr_val &lt;=0.1
coor_to_check &lt;- which(matrix_fdr_val &lt;= 0.1, arr.ind=T)
coor_to_check &lt;- as.data.frame(coor_to_check)

# Number of variables significant at 10% FDR not including species or tissue (note: off by 4 from column number in RNA_seq_info file; see notes in Part Two)

coor_to_check_col &lt;- coor_to_check$col
unique(coor_to_check_col)</code></pre>
<pre><code>##  [1]  1  2  3  4  7  8 12 13 14 15 16 17 18 19 20 21 23 24 25 26</code></pre>
<pre class="r"><code>length(unique(coor_to_check_col))</code></pre>
<pre><code>## [1] 20</code></pre>
<p>** Conclusions from Part I**</p>
<p>The following variables are associated with one of the PCs tested and will be investigated further in Part 2.</p>
<ul>
<li>Individual</li>
<li>RNA extraction date</li>
<li>Multiplexing index sequence</li>
<li>Multiplexing mix codes</li>
<li>Percentage of bps trimmed (adapters)</li>
<li>Number of reads shorter than 20 bp removed</li>
<li>Percentage of mapped reads overlapping a junction</li>
<li>Number of junctions</li>
<li>Number of reads mapped on orthologous exons</li>
<li>Number of orthologous exons with at least 1 mapped read</li>
<li>Number of orthologous genes with at least 1 mapped read</li>
<li>RNA concentration</li>
<li>RIN score</li>
<li>Library concentration</li>
<li>Library fragments size</li>
</ul>
</div>
</div>
<div id="part-two-for-the-variables-that-correlate-see-if-these-segregate-with-either-species-or-tissue" class="section level3">
<h3>PART TWO: For the variable(s) that correlate, see if these segregate with either species or tissue</h3>
<p>In coor_to_check_col, row is the PC # and col is the column # -4 that is associated with the PC Want to take the coor_to_check_col column # and add four. That is the variable that we should check to see if it correlates with tissue or species.</p>
<pre class="r"><code>var_numb = unique(coor_to_check_col) + 4
#Making the variable versus species and tissue graphs into a file called VarVsSpeciesTissue.pdf
var.numb &lt;- as.data.frame(var_numb)

#pdf(&#39;../data/Var_GenePCsVsSpeciesTissue.pdf&#39;)

#for (i in var.numb[1:nrow(var.numb),]) { 
#par(mfrow=c(1,2))
#plot(RNA_seq_info$Species, RNA_seq_info[,i], xlab = &quot;Species&quot;, ylab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])))

#plot(RNA_seq_info$Tissue, RNA_seq_info[,i], xlab = &quot;Tissue&quot;, ylab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])))

#mtext(substitute(paste(k, &#39; vs. Species and Tissue&#39;), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)

#}
#dev.off()

#Testing to see if differences across variable groups are statistically significant for species. If statistically significant, then we should investigate further whether this variable is confounded with species or tissue. 

#Make a matrix to store the p-values

pvalues_species = matrix(data = NA, nrow = 1, ncol = 19, dimnames = list(c(&quot;p-value&quot;), c(&quot;Individual&quot;, &quot;Age quantile&quot;, &quot;Hours postmortem&quot;, &quot;Possible conditions affecting tissues&quot;, &quot;Source&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))

cat_var &lt;- c(5, 27, 28, 29, 30, 6, 7, 8)
num_var &lt;- c(11, 12, 16, 17, 18, 19, 20, 21, 22, 23, 24)

#Performing this test of significance for variables that are categorical data (Using Pearson&#39;s chi-squared test)
j=1
for (i in cat_var) { 

pval_chi &lt;- chisq.test(as.factor(RNA_seq_info[,i]), as.factor(RNA_seq_info$Species))$p.value
pvalues_species[,j] &lt;- pval_chi  
j=j+1

}</code></pre>
<pre><code>## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect</code></pre>
<pre class="r"><code>#Performing this test of significance for variables that are numerical data (Using an ANOVA)
j = 9
for (i in num_var) {  
summary_anova = summary(aov(RNA_seq_info[,i]~ as.factor(RNA_seq_info$Species)))
data_summary_anova &lt;- as.data.frame(summary_anova[[1]]$`Pr(&gt;F)`)
p_val_anova &lt;- data_summary_anova[1,]
pvalues_species[,j] &lt;- p_val_anova  
j=j+1
}

#Testing to see if differences across variable groups are statistically significant for tissues. If statistically significant, then we should investigate further whether this variable is confounded with species or tissue. 

pvalues_tissue = matrix(data = NA, nrow = 1, ncol = 19, dimnames = list(c(&quot;p-value&quot;), c(&quot;Individual&quot;, &quot;Age quantile&quot;, &quot;Hours postmortem&quot;, &quot;Possible conditions affecting tissues&quot;, &quot;Source&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))


#Performing this test of significance for variables that are categorical data (Using Pearson&#39;s chi-squared test)
j=1
for (i in cat_var) { 

pval_chi &lt;- chisq.test(as.factor(RNA_seq_info[,i]), as.factor(RNA_seq_info$Tissue))$p.value
pvalues_tissue[,j] &lt;- pval_chi  
j=j+1

}</code></pre>
<pre><code>## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect</code></pre>
<pre><code>## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>#Performing this test of significance for variables that are numerical data (Using an ANOVA)
j = 9
for (i in num_var) {  
summary_anova = summary(aov(RNA_seq_info[,i]~ as.factor(RNA_seq_info$Tissue)))
data_summary_anova &lt;- as.data.frame(summary_anova[[1]]$`Pr(&gt;F)`)
p_val_anova &lt;- data_summary_anova[1,]
pvalues_tissue[,j] &lt;- p_val_anova  
j=j+1
}



collapse_table &lt;- rbind(pvalues_tissue, pvalues_species)

rownames(collapse_table) &lt;- c(&quot;Tissue&quot;, &quot;Species&quot;)

collapse_table</code></pre>
<pre><code>##           Individual Age quantile Hours postmortem
## Tissue  1.000000e+00 9.999429e-01     1.000000e+00
## Species 7.084425e-11 4.663809e-05     8.591612e-16
##         Possible conditions affecting tissues       Source
## Tissue                           1.000000e+00 1.000000e+00
## Species                          4.556773e-13 4.556773e-13
##         RNA extraction date Multiplexing index sequence
## Tissue         9.999624e-01                7.313637e-07
## Species        7.149979e-17                9.951634e-01
##         Multiplexing mix codes Percentage of bps trimmed (adaptors)
## Tissue            8.963210e-01                          0.273399046
## Species           4.166637e-07                          0.007347325
##         Number of reads shorter than 20bp removed
## Tissue                                0.229196235
## Species                               0.005859874
##         Percentage of mapped reads overlapping a junction
## Tissue                                       0.0001309289
## Species                                      0.2729677718
##         Number of junctions Number of reads mapped on orthologous exons
## Tissue         2.566692e-04                                  0.01821964
## Species        8.155021e-05                                  0.30290842
##         Number of orthologous exons with at least 1 mapped read
## Tissue                                             2.551611e-08
## Species                                            5.853704e-01
##         Number of orthologous genes with at least 1 mapped read
## Tissue                                             2.913679e-07
## Species                                            4.263982e-02
##         RNA concentration (ng/uL)    RIN score
## Tissue               0.0014034907 6.370173e-01
## Species              0.0008347635 2.263714e-08
##         Library concentration (ng/uL) Library fragments size (bp)
## Tissue                    0.283235201                0.2664092315
## Species                   0.002485755                0.0008550811</code></pre>
<pre class="r"><code>#Calculate q-values (FDR = 10%)

fdr_val = p.adjust(collapse_table, method = &quot;fdr&quot;, n = length(collapse_table)*2)
fdr_val_order = fdr_val[order(fdr_val)]
hist(fdr_val_order, ylab = &quot;BH-adjusted p-values&quot;, main = &quot;Distribution of Benjamini and Hochberg adjusted p-values&quot;, breaks = 10)</code></pre>
<p><img src="Tech_factor_analysis1_gene_exp_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>collapse_table_fdr_val = matrix(data = fdr_val, nrow = 2, ncol = 19, dimnames = list(c(&quot;Tissue&quot;, &quot;Species&quot;), c(&quot;Individual&quot;, &quot;Age quantile&quot;, &quot;Hours postmortem&quot;, &quot;Possible conditions affecting tissues&quot;, &quot;Source&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))

collapse_table_fdr_val</code></pre>
<pre><code>##           Individual Age quantile Hours postmortem
## Tissue  1.000000e+00 1.0000000000     1.000000e+00
## Species 1.076833e-09 0.0003222268     3.264812e-14
##         Possible conditions affecting tissues       Source
## Tissue                           1.000000e+00 1.000000e+00
## Species                          8.657869e-12 8.657869e-12
##         RNA extraction date Multiplexing index sequence
## Tissue         1.000000e+00                5.558364e-06
## Species        5.433984e-15                1.000000e+00
##         Multiplexing mix codes Percentage of bps trimmed (adaptors)
## Tissue            1.000000e+00                           0.79725464
## Species           3.518494e-06                           0.02791984
##         Number of reads shorter than 20bp removed
## Tissue                                  0.7573441
## Species                                 0.0234395
##         Percentage of mapped reads overlapping a junction
## Tissue                                       0.0007654306
## Species                                      0.7972546398
##         Number of junctions Number of reads mapped on orthologous exons
## Tissue         0.0013933472                                  0.06593773
## Species        0.0005164846                                  0.82218001
##         Number of orthologous exons with at least 1 mapped read
## Tissue                                              2.77032e-07
## Species                                             1.00000e+00
##         Number of orthologous genes with at least 1 mapped read
## Tissue                                             2.767995e-06
## Species                                            1.473012e-01
##         RNA concentration (ng/uL)   RIN score
## Tissue                0.006274429 1.00000e+00
## Species               0.004061635 2.77032e-07
##         Library concentration (ng/uL) Library fragments size (bp)
## Tissue                     0.79725464                 0.797254640
## Species                    0.01049541                 0.004061635</code></pre>
<pre class="r"><code>#Find number below it
sum(collapse_table_fdr_val &lt;= 0.1)</code></pre>
<pre><code>## [1] 21</code></pre>
<pre class="r"><code>#write.csv(collapse_table_fdr_val, &quot;../data/collapse_table_fdr_val_RNA2.csv&quot;)</code></pre>
<p><strong>Conclusions from Part 2:</strong></p>
<p>The following variables are associated with species: * Individual * RNA extraction date * Multiplexing mix codes * Percentage of bps trimmed (adaptors) * Number of reads shorter than 20bp removed * Number of junctions * RNA concentration * RIN score * Library concentration * Library fragments size</p>
<p>The following variables are associated with tissue: * Multiplexing index sequence * Percentage of mapped reads overlapping a junction * Number of junctions * Number of reads mapped on othologous exons * Number of orthologous exons with at least 1 mapped read * Number of orthologous genes with at least 1 mapped read * RNA concentration</p>
</div>
<div id="part-three-which-variables-to-put-in-the-model" class="section level3">
<h3>PART THREE: Which variables to put in the model?</h3>
<p>Check for degree of similarity between the values of the different variables (not including species or tissue) For ones that show high correlation (multi-colinearity), we will only want to incorporate one of these variables into the model Note: the first 4 variables are ordinal, the others are numerical.</p>
<p>There is one pair of correlated variables: the number of orthologous exons with at least 1 mapped read and number of orthologous genes with at least 1 mapped read.</p>
</div>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
