<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2016-07-20" />

<title>Analysis_of_potential_confounders_gene_expression</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis_of_potential_confounders_gene_expression</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>July 20, 2016</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes">PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes</a><ul>
<li><a href="#find-gene-pcs">Find gene PCs</a><ul>
<li><a href="#relationship-between-pc1">Relationship between PC1</a></li>
<li><a href="#part-two-for-the-variables-that-correlate-see-if-these-segregate-with-either-species-or-tissue">PART TWO: For the variable(s) that correlate, see if these segregate with either species or tissue</a></li>
<li><a href="#part-three-which-variables-to-put-in-the-model">PART THREE: Which variables to put in the model?</a></li>
</ul></li>
</ul></li>
</ul>
</div>

<div id="part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes" class="section level1">
<h1>PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes</h1>
<div id="find-gene-pcs" class="section level2">
<h2>Find gene PCs</h2>
<pre class="r"><code># Load gene expression levels

cpm_12184 &lt;- read.delim(&quot;../../../Reg_Evo_Primates/data/cpm_12184.txt&quot;)

dim(cpm_12184)</code></pre>
<pre><code>## [1] 12184    47</code></pre>
<pre class="r"><code># Take gene PCs

pca_genes &lt;- prcomp(t(cpm_12184), scale = T)
scores &lt;- pca_genes$x
gene_pcs &lt;- scores </code></pre>
<div id="load-libraries-and-data" class="section level4">
<h4>Load libraries and data</h4>
<pre class="r"><code># Load libraries

# install.packages(&quot;polycor&quot;)
library(&quot;polycor&quot;)

# Load list of technical factors
RNA_seq_info &lt;- read.csv(&quot;../../../Reg_Evo_Primates/data/RNA_seq_info.csv&quot;)
RNA_seq_info &lt;- as.data.frame(RNA_seq_info)

dim(RNA_seq_info)</code></pre>
<pre><code>## [1] 47 24</code></pre>
<pre class="r"><code>#Create plots for each of the possible confounders versus PCs 1-5

pdf(&#39;../data/VarVsGenePCs.pdf&#39;)

for (i in 3:length(RNA_seq_info)) { 
  par(mfrow=c(1,5))
  plot(RNA_seq_info[,i], gene_pcs[,1],  ylab = &quot;PC1&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], gene_pcs[,2],  ylab = &quot;PC2&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], gene_pcs[,3],  ylab = &quot;PC3&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], gene_pcs[,4],  ylab = &quot;PC4&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], gene_pcs[,5],  ylab = &quot;PC5&quot;, xlab = &quot; &quot;)
  title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
  mtext(substitute(paste(&#39;PCs vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
  
}
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
</div>
<div id="relationship-between-pc1" class="section level3">
<h3>Relationship between PC1</h3>
<pre class="r"><code>checkPC1 &lt;- lm(gene_pcs[,1] ~ as.factor(RNA_seq_info$Tissue))

summary(checkPC1)</code></pre>
<pre><code>## 
## Call:
## lm(formula = gene_pcs[, 1] ~ as.factor(RNA_seq_info$Tissue))
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -57.900 -12.990  -2.963  16.862  39.728 
## 
## Coefficients:
##                                      Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)                           -29.033      7.043  -4.122 0.000168
## as.factor(RNA_seq_info$Tissue)kidney   36.593      9.751   3.753 0.000519
## as.factor(RNA_seq_info$Tissue)liver    98.649      9.751  10.117 6.06e-13
## as.factor(RNA_seq_info$Tissue)lung    -21.530      9.751  -2.208 0.032625
##                                         
## (Intercept)                          ***
## as.factor(RNA_seq_info$Tissue)kidney ***
## as.factor(RNA_seq_info$Tissue)liver  ***
## as.factor(RNA_seq_info$Tissue)lung   *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 23.36 on 43 degrees of freedom
## Multiple R-squared:  0.8081, Adjusted R-squared:  0.7947 
## F-statistic: 60.35 on 3 and 43 DF,  p-value: 1.859e-15</code></pre>
<pre class="r"><code>checkPC1 &lt;- lm(gene_pcs[,2] ~ as.factor(RNA_seq_info$Tissue))

summary(checkPC1)</code></pre>
<pre><code>## 
## Call:
## lm(formula = gene_pcs[, 2] ~ as.factor(RNA_seq_info$Tissue))
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -48.175 -22.203   7.079  14.446  52.474 
## 
## Coefficients:
##                                      Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)                           -65.964      7.823  -8.432 1.17e-10
## as.factor(RNA_seq_info$Tissue)kidney   68.287     10.831   6.305 1.32e-07
## as.factor(RNA_seq_info$Tissue)liver    90.714     10.831   8.375 1.40e-10
## as.factor(RNA_seq_info$Tissue)lung     99.357     10.831   9.173 1.11e-11
##                                         
## (Intercept)                          ***
## as.factor(RNA_seq_info$Tissue)kidney ***
## as.factor(RNA_seq_info$Tissue)liver  ***
## as.factor(RNA_seq_info$Tissue)lung   ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 25.95 on 43 degrees of freedom
## Multiple R-squared:  0.7034, Adjusted R-squared:  0.6827 
## F-statistic: 33.99 on 3 and 43 DF,  p-value: 2.019e-11</code></pre>
<pre class="r"><code>checkPC2 &lt;- lm(gene_pcs[,2] ~ as.factor(RNA_seq_info$Species))

summary(checkPC2)</code></pre>
<pre><code>## 
## Call:
## lm(formula = gene_pcs[, 2] ~ as.factor(RNA_seq_info$Species))
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -82.400 -11.763   9.765  27.362  47.606 
## 
## Coefficients:
##                                               Estimate Std. Error t value
## (Intercept)                                      8.849     10.064   0.879
## as.factor(RNA_seq_info$Species)human            15.567     14.467   1.076
## as.factor(RNA_seq_info$Species)rhesus macaque  -40.588     14.232  -2.852
##                                               Pr(&gt;|t|)   
## (Intercept)                                     0.3840   
## as.factor(RNA_seq_info$Species)human            0.2878   
## as.factor(RNA_seq_info$Species)rhesus macaque   0.0066 **
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 40.25 on 44 degrees of freedom
## Multiple R-squared:  0.2696, Adjusted R-squared:  0.2364 
## F-statistic: 8.119 on 2 and 44 DF,  p-value: 0.0009972</code></pre>
<div id="testing-association-between-a-particular-variable-and-pcs-with-a-linear-model" class="section level4">
<h4>Testing association between a particular variable and PCs with a linear model</h4>
<pre class="r"><code>#Make an empty matrix to put all of the data in

pvaluesandr2 = matrix(data = NA, nrow = 5, ncol = 22*2, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Species&quot;, &quot;Species R^2&quot;, &quot;Tissue&quot;, &quot;Tissue R^2&quot;, &quot;Individual&quot;, &quot;Ind. R^2&quot;, &quot;RNA extraction date&quot;, &quot;Extract. date R^2&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multi. index R^2&quot;, &quot;Multiplexing mix codes&quot;, &quot;Multi. mix R^2&quot;, &quot;Sequencing location: at UCGF&quot;, &quot;Location R^2&quot;, &quot;Total number of reads sequenced&quot;, &quot;Read Seq R^2&quot;, &quot;Percentage of bps trimmed (adapters)&quot;, &quot;Trimmed R^2&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Removed R^2&quot;, &quot;Maximum read length after trimming&quot;, &quot;Max. length R^2&quot;, &quot;Total number of reads processed in tophat&quot;, &quot;Reads processed R^2&quot;, &quot;Total number of mapped reads&quot;, &quot;Mapped reads R^2&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Perc. junction R^2&quot;, &quot;Number of junctions&quot;, &quot;Num. junctions R^2&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;OrthoExons R^2&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Ortho exons read R^2&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;Ortho genes read R^2&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RNA conc. R^2&quot;, &quot;RIN score&quot;, &quot;RIN R^2&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Lib. conc. R^2&quot;, &quot;Library fragments size (bp)&quot;, &quot;Lib. frag. R^2&quot;)))

#Check lm to see how well the variables containing ordinal data are correlated with a PC

#For PCs 1-5
j=1
for (i in 3:9){
  for (j in 1:5){
  
  checkPC1 &lt;- lm(gene_pcs[,j] ~ as.factor(RNA_seq_info[,i]))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

#Put the summary statistics into the matrix w

  pvaluesandr2[j, (2*i-5)] &lt;- p_fstat
  pvaluesandr2[j, (2*i-4)] &lt;- r2_value
  
  }
}

#Check lm to see how well the variables containing numerical data are correlated with a PC

j=1
for (j in 1:5){
  
  for (i in 10:24){
      
  checkPC &lt;- lm(gene_pcs[,j] ~ RNA_seq_info[,i])
      
  #Get the summary statistics from it
  summary(checkPC)
  
  #Get the p-value of the F-statistic
  summary(checkPC)$fstatistic
  
  fstat &lt;- as.data.frame(summary(checkPC)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
  #Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC)$r.squared
  
  #Put the summary statistics into the matrix a
  
  pvaluesandr2[j, 2*i-5] &lt;- p_fstat
  pvaluesandr2[j, 2*i-4] &lt;- r2_value
  
  i = i+1
  }
  j=j+1
}</code></pre>
</div>
<div id="test-for-potential-violations-of-the-assumptions-of-the-linear-model" class="section level4">
<h4>Test for potential violations of the assumptions of the linear model</h4>
<pre class="r"><code>#Plot the residuals to look for violations of the assumptions of lm

pdf(&#39;../data/Residuals_vs_var_GenePCs.pdf&#39;)

i = 3
  for (i in 3:9){
    par(mfrow=c(1,5))
   
    checkPC1 &lt;- lm(gene_pcs[,1] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)

    checkPC2 &lt;- lm(gene_pcs[,2] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC2),  ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
    
    checkPC3 &lt;- lm(gene_pcs[,3] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC3),  ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
    
    checkPC4 &lt;- lm(gene_pcs[,4] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC4),  ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
    
    checkPC5 &lt;- lm(gene_pcs[,5] ~ as.factor(RNA_seq_info[,i]))
    plot(RNA_seq_info[,i], resid(checkPC5),  ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
    
    title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
    mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
    
  }

#Need to divide this up because #22 has an NA in it 
i = 10
  for (i in 10:21){
    
    par(mfrow=c(1,5))
    
    checkPC1 &lt;- lm(gene_pcs[,1] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)
    
    checkPC2 &lt;- lm(gene_pcs[,2] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC2), ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
    
    checkPC3 &lt;- lm(gene_pcs[,3] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC3), ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
    
    checkPC4 &lt;- lm(gene_pcs[,4] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC4), ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
    
    checkPC5 &lt;- lm(gene_pcs[,5] ~ RNA_seq_info[,i])
    plot(RNA_seq_info[,i], resid(checkPC5), ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
    
    title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
    mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
    
  }

#Skip #22 because there is an NA in it

for (i in 23:24){
  
  par(mfrow=c(1,5))
  
  checkPC1 &lt;- lm(gene_pcs[,1] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)
  
  checkPC2 &lt;- lm(gene_pcs[,2] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC2), ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
  
  checkPC3 &lt;- lm(gene_pcs[,3] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC3), ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
  
  checkPC4 &lt;- lm(gene_pcs[,4] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC4), ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
  
  checkPC5 &lt;- lm(gene_pcs[,5] ~ RNA_seq_info[,i])
  plot(RNA_seq_info[,i], resid(checkPC5), ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
  
  title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
  mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
  
}</code></pre>
</div>
<div id="look-at-significant-p-values" class="section level4">
<h4>Look at significant p-values</h4>
<pre class="r"><code>#Get rid of the R^2 values
all_col &lt;- 1:ncol(pvaluesandr2)
matrix_pval &lt;- pvaluesandr2[ , all_col%%2==1 ]

#Find which variables/PC combinations are p-value &lt; 0.05
TorF_matrix &lt;- matrix_pval &lt;=0.05
sum(matrix_pval &lt;= 0.05/(22*5))</code></pre>
<pre><code>## [1] 16</code></pre>
<pre class="r"><code>#Find which variables/PC combinations are p-value &lt; 0.05 not including species or tissue (because those explain PC1 and PC2)
matrix_pval_no_tissue_or_species = matrix_pval[ , 3:22]
matrix_pval_no_tissue_or_species</code></pre>
<pre><code>##       Individual RNA extraction date Multiplexing index sequence
## PC1 8.577332e-01        1.933096e-01                3.202755e-05
## PC2 3.089533e-01        8.775476e-03                2.494550e-03
## PC3 4.210981e-03        6.385467e-06                1.665273e-01
## PC4 9.971820e-01        6.167683e-01                1.541532e-09
## PC5 4.996004e-15        0.000000e+00                9.542053e-01
##     Multiplexing mix codes Sequencing location: at UCGF
## PC1           3.828374e-01                   0.40556453
## PC2           8.784915e-03                   0.85463260
## PC3           3.806629e-05                   0.51996073
## PC4           1.927352e-01                   0.62874566
## PC5           3.820715e-01                   0.07672979
##     Total number of reads sequenced Percentage of bps trimmed (adapters)
## PC1                       0.6623602                           0.43484612
## PC2                       0.8367218                           0.03093471
## PC3                       0.9498725                           0.56268886
## PC4                       0.1491588                           0.55118204
## PC5                       0.6813582                           0.03486093
##     Number of reads shorter than 20bp removed
## PC1                                0.32598741
## PC2                                0.02745845
## PC3                                0.68306619
## PC4                                0.47137991
## PC5                                0.04640738
##     Maximum read length after trimming
## PC1                         0.33637965
## PC2                         0.92113549
## PC3                         0.75683691
## PC4                         0.59778768
## PC5                         0.05145475
##     Total number of reads processed in tophat Total number of mapped reads
## PC1                                 0.7051527                    0.8420494
## PC2                                 0.9463660                    0.0369636
## PC3                                 0.9269870                    0.3729374
## PC4                                 0.1578441                    0.0930245
## PC5                                 0.5848903                    0.7499138
##     Percentage of mapped reads overlapping a junction Number of junctions
## PC1                                      1.630216e-06         0.002629987
## PC2                                      5.446252e-01         0.003137455
## PC3                                      6.527941e-01         0.243584612
## PC4                                      1.540735e-02         0.179619438
## PC5                                      6.196622e-01         0.089573389
##     Number of reads mapped on orthologous exons
## PC1                                  0.12500946
## PC2                                  0.12756466
## PC3                                  0.02997633
## PC4                                  0.01379170
## PC5                                  0.08964825
##     Number of orthologous exons with at least 1 mapped read
## PC1                                             0.004114919
## PC2                                             0.059787325
## PC3                                             0.000255304
## PC4                                             0.078055122
## PC5                                             0.077428146
##     Number of orthologous genes with at least 1 mapped read
## PC1                                             0.005239234
## PC2                                             0.008480924
## PC3                                             0.005243010
## PC4                                             0.055612828
## PC5                                             0.006306295
##     RNA concentration (ng/uL)    RIN score Library concentration (ng/uL)
## PC1              0.0148708408 3.197387e-01                    0.05256745
## PC2              0.0578165224 4.172052e-03                    0.01417761
## PC3              0.0342043520 2.938711e-06                    0.74147481
## PC4              0.3831733717 2.597738e-02                    0.79123942
## PC5              0.0001948362 4.736250e-01                    0.02505369
##     Library fragments size (bp)
## PC1                 0.944720057
## PC2                 0.060512968
## PC3                 0.400971333
## PC4                 0.124460815
## PC5                 0.001394967</code></pre>
<pre class="r"><code>sum(matrix_pval_no_tissue_or_species &lt;= 0.05/(20*5))</code></pre>
<pre><code>## [1] 10</code></pre>
<pre class="r"><code>#Distribution of p-values adjusted by FDR not including species or tissue

fdr_val = p.adjust(matrix_pval_no_tissue_or_species, method = &quot;fdr&quot;, n = length(matrix_pval_no_tissue_or_species))
fdr_val_order = fdr_val[order(fdr_val)]
hist(fdr_val_order, ylab = &quot;BH-adjusted p-values&quot;, main = &quot;Distribution of Benjamini and Hochberg adjusted p-values&quot;, breaks = 10)</code></pre>
<p><img src="Tech_factor_analysis1_gene_exp_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>matrix_fdr_val = matrix(data = fdr_val, nrow = 5, ncol = 20, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Sequencing location: at UCGF&quot;, &quot;Total number of reads sequenced&quot;, &quot;Percentage of bps trimmed (adapters)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Maximum read length after trimming&quot;, &quot;Total number of reads processed in tophat&quot;, &quot;Total number of mapped reads&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;,  &quot;Library fragments size (bp)&quot;))) 

matrix_fdr_val</code></pre>
<pre><code>##       Individual RNA extraction date Multiplexing index sequence
## PC1 9.222938e-01        0.3391396663                4.575364e-04
## PC2 5.236497e-01        0.0381952828                2.023067e-02
## PC3 2.477048e-02        0.0001064245                3.083839e-01
## PC4 9.971820e-01        0.7843825501                5.138440e-08
## PC5 2.498002e-13        0.0000000000                9.638437e-01
##     Multiplexing mix codes Sequencing location: at UCGF
## PC1           0.5805657147                    0.5964184
## PC2           0.0381952828                    0.9222938
## PC3           0.0004758287                    0.7221677
## PC4           0.3391396663                    0.7859321
## PC5           0.5805657147                    0.1734558
##     Total number of reads sequenced Percentage of bps trimmed (adapters)
## PC1                       0.8077563                           0.63021176
## PC2                       0.9222938                           0.09667098
## PC3                       0.9638437                           0.75025182
## PC4                       0.2868438                           0.74484059
## PC5                       0.8131740                           0.10253214
##     Number of reads shorter than 20bp removed
## PC1                                0.53440560
## PC2                                0.09152817
## PC3                                0.81317404
## PC4                                0.66707751
## PC5                                0.12890938
##     Maximum read length after trimming
## PC1                          0.5425478
## PC2                          0.9638437
## PC3                          0.8600419
## PC4                          0.7763476
## PC5                          0.1383354
##     Total number of reads processed in tophat Total number of mapped reads
## PC1                                 0.8295914                    0.9222938
## PC2                                 0.9638437                    0.1056103
## PC3                                 0.9638437                    0.5805657
## PC4                                 0.2978190                    0.1938010
## PC5                                 0.7695925                    0.8600419
##     Percentage of mapped reads overlapping a junction Number of junctions
## PC1                                      0.0000407554          0.02023067
## PC2                                      0.7448405903          0.02241039
## PC3                                      0.8059186205          0.41997347
## PC4                                      0.0570642482          0.32658080
## PC5                                      0.7843825501          0.19074096
##     Number of reads mapped on orthologous exons
## PC1                                  0.25001893
## PC2                                  0.25012679
## PC3                                  0.09667098
## PC4                                  0.05671045
## PC5                                  0.19074096
##     Number of orthologous exons with at least 1 mapped read
## PC1                                              0.02477048
## PC2                                              0.14407850
## PC3                                              0.00255304
## PC4                                              0.17345583
## PC5                                              0.17345583
##     Number of orthologous genes with at least 1 mapped read
## PC1                                              0.02759479
## PC2                                              0.03819528
## PC3                                              0.02759479
## PC4                                              0.14259700
## PC5                                              0.03153147
##     RNA concentration (ng/uL)    RIN score Library concentration (ng/uL)
## PC1               0.057064248 5.328978e-01                    0.13833540
## PC2               0.144078495 2.477048e-02                    0.05671045
## PC3               0.102532140 5.877421e-05                    0.86004194
## PC4               0.580565715 8.957717e-02                    0.88903305
## PC5               0.002164847 6.670775e-01                    0.08947745
##     Library fragments size (bp)
## PC1                  0.96384373
## PC2                  0.14407850
## PC3                  0.59641843
## PC4                  0.25001893
## PC5                  0.01268152</code></pre>
<pre class="r"><code># Number of values significant at 10% FDR not including species or tissue

sum(matrix_fdr_val &lt;= 0.1)</code></pre>
<pre><code>## [1] 32</code></pre>
<pre class="r"><code>#Get the coordinates of which variables/PC combinations are significant at FDR 10%

TorF_matrix_fdr &lt;- matrix_fdr_val &lt;=0.1
coor_to_check &lt;- which(matrix_fdr_val &lt;= 0.1, arr.ind=T)
coor_to_check &lt;- as.data.frame(coor_to_check)

# Number of variables significant at 10% FDR not including species or tissue (note: off by 4 from column number in RNA_seq_info file; see notes in Part Two)

coor_to_check_col &lt;- coor_to_check$col
unique(coor_to_check_col)</code></pre>
<pre><code>##  [1]  1  2  3  4  7  8 12 13 14 15 16 17 18 19 20</code></pre>
<pre class="r"><code>length(unique(coor_to_check_col))</code></pre>
<pre><code>## [1] 15</code></pre>
<p>** Conclusions from Part I**</p>
<p>The following variables are associated with one of the PCs tested and will be investigated further in Part 2.</p>
<ul>
<li>Individual</li>
<li>RNA extraction date</li>
<li>Multiplexing index sequence</li>
<li>Multiplexing mix codes</li>
<li>Percentage of bps trimmed (adapters)</li>
<li>Number of reads shorter than 20 bp removed</li>
<li>Percentage of mapped reads overlapping a junction</li>
<li>Number of junctions</li>
<li>Number of reads mapped on orthologous exons</li>
<li>Number of orthologous exons with at least 1 mapped read</li>
<li>Number of orthologous genes with at least 1 mapped read</li>
<li>RNA concentration</li>
<li>RIN score</li>
<li>Library concentration</li>
<li>Library fragments size</li>
</ul>
</div>
</div>
<div id="part-two-for-the-variables-that-correlate-see-if-these-segregate-with-either-species-or-tissue" class="section level3">
<h3>PART TWO: For the variable(s) that correlate, see if these segregate with either species or tissue</h3>
<p>In coor_to_check_col, row is the PC # and col is the column # -4 that is associated with the PC Want to take the coor_to_check_col column # and add four. That is the variable that we should check to see if it correlates with tissue or species.</p>
<pre class="r"><code>var_numb = unique(coor_to_check_col) + 4
#Making the variable versus species and tissue graphs into a file called VarVsSpeciesTissue.pdf
var.numb &lt;- as.data.frame(var_numb)

pdf(&#39;../data/Var_GenePCsVsSpeciesTissue.pdf&#39;)

for (i in var.numb[1:nrow(var.numb),]) { 
par(mfrow=c(1,2))
plot(RNA_seq_info$Species, RNA_seq_info[,i], xlab = &quot;Species&quot;, ylab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])))

plot(RNA_seq_info$Tissue, RNA_seq_info[,i], xlab = &quot;Tissue&quot;, ylab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])))

mtext(substitute(paste(k, &#39; vs. Species and Tissue&#39;), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)

}
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<pre class="r"><code>#Testing to see if differences across variable groups are statistically significant for species. If statistically significant, then we should investigate further whether this variable is confounded with species or tissue. 

#Make a matrix to store the p-values

pvalues_species = matrix(data = NA, nrow = 1, ncol = nrow(var.numb), dimnames = list(c(&quot;p-value&quot;), c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))

#Performing this test of significance for variables that are categorical data (Using Pearson&#39;s chi-squared test)
j=1
for (i in var.numb[1:4,]) { 

pval_chi &lt;- chisq.test(as.factor(RNA_seq_info[,i]), as.factor(RNA_seq_info$Species))$p.value
pvalues_species[,j] &lt;- pval_chi  
j=j+1

}</code></pre>
<pre><code>## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Species)): Chi-squared approximation may be
## incorrect</code></pre>
<pre class="r"><code>#Performing this test of significance for variables that are numerical data (Using an ANOVA)
j=5
for (i in var.numb[5:nrow(var.numb),]) {  
summary_anova = summary(aov(RNA_seq_info[,i]~ as.factor(RNA_seq_info$Species)))
data_summary_anova &lt;- as.data.frame(summary_anova[[1]]$`Pr(&gt;F)`)
p_val_anova &lt;- data_summary_anova[1,]
pvalues_species[,j] &lt;- p_val_anova  
j=j+1
}

#Testing to see if differences across variable groups are statistically significant for tissues. If statistically significant, then we should investigate further whether this variable is confounded with species or tissue. 

#Make a matrix to store the p-values

pvalues_tissues = matrix(data = NA, nrow = 1, ncol = nrow(var.numb), dimnames = list(c(&quot;p-value&quot;), c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))

#Performing this test of significance for variables that are categorical data (Using Pearson&#39;s chi-squared test)
j=1
for (i in var.numb[1:4,]) { 
  
  pval_chi &lt;- chisq.test(as.factor(RNA_seq_info[,i]), as.factor(RNA_seq_info$Tissue))$p.value
  pvalues_tissues[,j] &lt;- pval_chi  
  j=j+1
  
}</code></pre>
<pre><code>## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect</code></pre>
<pre><code>## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect

## Warning in chisq.test(as.factor(RNA_seq_info[, i]),
## as.factor(RNA_seq_info$Tissue)): Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>#Performing this test of significance for variables that are numerical data (Using an ANOVA)
j=5
for (i in var.numb[5:nrow(var.numb),]) {  
  summary_anova = summary(aov(RNA_seq_info[,i]~ as.factor(RNA_seq_info$Tissue)))
  data_summary_anova &lt;- as.data.frame(summary_anova[[1]]$`Pr(&gt;F)`)
  p_val_anova &lt;- data_summary_anova[1,]
  pvalues_tissues[,j] &lt;- p_val_anova  
  j=j+1
}

collapse_table &lt;- rbind(pvalues_species, pvalues_tissues)
colnames(collapse_table) &lt;- c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)

rownames(collapse_table) &lt;- c(&quot;Species&quot;, &quot;Tissue&quot;)

collapse_table</code></pre>
<pre><code>##           Individual RNA extraction date Multiplexing index sequence
## Species 7.084425e-11        7.149979e-17                9.951634e-01
## Tissue  1.000000e+00        9.999624e-01                7.313637e-07
##         Multiplexing mix codes Percentage of bps trimmed (adaptors)
## Species           4.166637e-07                          0.007347325
## Tissue            8.963210e-01                          0.273399046
##         Number of reads shorter than 20bp removed
## Species                               0.005859874
## Tissue                                0.229196235
##         Percentage of mapped reads overlapping a junction
## Species                                      0.2729677718
## Tissue                                       0.0001309289
##         Number of junctions Number of reads mapped on orthologous exons
## Species        8.155021e-05                                  0.30290842
## Tissue         2.566692e-04                                  0.01821964
##         Number of orthologous exons with at least 1 mapped read
## Species                                            5.853704e-01
## Tissue                                             2.551611e-08
##         Number of orthologous genes with at least 1 mapped read
## Species                                            4.263982e-02
## Tissue                                             2.913679e-07
##         RNA concentration (ng/uL)    RIN score
## Species              0.0008347635 2.602041e-08
## Tissue               0.0014034907 6.437994e-01
##         Library concentration (ng/uL) Library fragments size (bp)
## Species                   0.002485755                0.0008550811
## Tissue                    0.283235201                0.2664092315</code></pre>
<pre class="r"><code>#Calculate q-values (FDR = 10%)

fdr_val = p.adjust(collapse_table, method = &quot;fdr&quot;, n = length(collapse_table)*2)
fdr_val_order = fdr_val[order(fdr_val)]
hist(fdr_val_order, ylab = &quot;BH-adjusted p-values&quot;, main = &quot;Distribution of Benjamini and Hochberg adjusted p-values&quot;, breaks = 10)</code></pre>
<p><img src="Tech_factor_analysis1_gene_exp_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>collapse_table_fdr_val = matrix(data = fdr_val, nrow = 2, ncol = nrow(var.numb), dimnames = list(c(&quot;Species&quot;, &quot;Tissue&quot;), c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))

collapse_table_fdr_val</code></pre>
<pre><code>##           Individual RNA extraction date Multiplexing index sequence
## Species 2.125327e-09        4.289987e-15                1.000000e+00
## Tissue  1.000000e+00        1.000000e+00                6.268832e-06
##         Multiplexing mix codes Percentage of bps trimmed (adaptors)
## Species           4.166637e-06                           0.02755247
## Tissue            1.000000e+00                           0.73887444
##         Number of reads shorter than 20bp removed
## Species                                 0.0234395
## Tissue                                  0.7237776
##         Percentage of mapped reads overlapping a junction
## Species                                      0.7388744374
## Tissue                                       0.0008728595
##         Number of junctions Number of reads mapped on orthologous exons
## Species        0.0006116265                                   0.7572711
## Tissue         0.0015400154                                   0.0643046
##         Number of orthologous exons with at least 1 mapped read
## Species                                            1.000000e+00
## Tissue                                             3.903061e-07
##         Number of orthologous genes with at least 1 mapped read
## Species                                            1.421327e-01
## Tissue                                             3.496415e-06
##         RNA concentration (ng/uL)    RIN score
## Species               0.004275406 3.903061e-07
## Tissue                0.006477650 1.000000e+00
##         Library concentration (ng/uL) Library fragments size (bp)
## Species                    0.01065324                 0.004275406
## Tissue                     0.73887444                 0.738874437</code></pre>
<pre class="r"><code>#Find number below it
sum(collapse_table_fdr_val &lt;= 0.1)</code></pre>
<pre><code>## [1] 17</code></pre>
<p><strong>Conclusions from Part 2:</strong></p>
<p>The following variables are associated with species: * Individual * RNA extraction date * Multiplexing mix codes * Percentage of bps trimmed (adaptors) * Number of reads shorter than 20bp removed * Number of junctions * RNA concentration * RIN score * Library concentration * Library fragments size</p>
<p>The following variables are associated with tissue: * Multiplexing index sequence * Percentage of mapped reads overlapping a junction * Number of junctions * Number of reads mapped on othologous exons * Number of orthologous exons with at least 1 mapped read * Number of orthologous genes with at least 1 mapped read * RNA concentration</p>
</div>
<div id="part-three-which-variables-to-put-in-the-model" class="section level3">
<h3>PART THREE: Which variables to put in the model?</h3>
<p>Check for degree of similarity between the values of the different variables (not including species or tissue) For ones that show high correlation (multi-colinearity), we will only want to incorporate one of these variables into the model Note: the first 4 variables are ordinal, the others are numerical.</p>
<pre class="r"><code>number_to_check &lt;- var.numb
number_to_check &lt;- as.data.frame(number_to_check)
number_to_check</code></pre>
<pre><code>##    var_numb
## 1         5
## 2         6
## 3         7
## 4         8
## 5        11
## 6        12
## 7        16
## 8        17
## 9        18
## 10       19
## 11       20
## 12       21
## 13       22
## 14       23
## 15       24</code></pre>
<pre class="r"><code>#Make the matrix

#Make the matrix
eval_multicolinearity = matrix(data = NA, nrow = nrow(var.numb), ncol = nrow(var.numb), dimnames = list(c(&quot;Individual&quot;, &quot;RNA extraction date&quot;, &quot;Multiplexing index sequence&quot;, &quot;Multiplexing mix codes&quot;, &quot;Percentage of bps trimmed (adaptors)&quot;, &quot;Number of reads shorter than 20bp removed&quot;, &quot;Percentage of mapped reads overlapping a junction&quot;, &quot;Number of junctions&quot;, &quot;Number of reads mapped on orthologous exons&quot;, &quot;Number of orthologous exons with at least 1 mapped read&quot;, &quot;Number of orthologous genes with at least 1 mapped read&quot;, &quot;RNA concentration (ng/uL)&quot;, &quot;RIN score&quot;, &quot;Library concentration (ng/uL)&quot;, &quot;Library fragments size (bp)&quot;)))

#Comparing two variables that are categorical (using the library polycor)

for (i in number_to_check[1:4,]){ 
  for (j in number_to_check[1:4,]){ 
    if (i==j) break
    corr_ord = polychor(RNA_seq_info[,i], RNA_seq_info[,j], ML=FALSE, std.err=T)
    corr_ord$rho
    eval_multicolinearity[i-4,j-4] &lt;- (corr_ord$rho)^2  
    
  }
  
}</code></pre>
<pre><code>## Warning in log(P): NaNs produced

## Warning in log(P): NaNs produced</code></pre>
<pre class="r"><code>for (j in number_to_check[1:4,]){ 
  for (i in number_to_check[1:4,]){ 
    if (i==j) eval_multicolinearity[i-4,j-4] = 1 
    if (i==j) break
    corr_ord = polychor(RNA_seq_info[,i], RNA_seq_info[,j], ML=FALSE, std.err=T)
    corr_ord$rho
    eval_multicolinearity[i-4,j-4] &lt;- (corr_ord$rho)^2  
    
  }
  
}</code></pre>
<pre><code>## Warning in log(P): NaNs produced

## Warning in log(P): NaNs produced</code></pre>
<pre class="r"><code>#Comparing one variable that is categorical with one that is numerical 

#For putting the percentage of bp trimmed into the correct location 

for (j in number_to_check[5:6,]){
  
  for (i in number_to_check[1:4,]){
    eval_multicolinearity[i-4,j-6] &lt;- (polyserial(RNA_seq_info[ ,j], as.factor(RNA_seq_info[ ,i])))^2
    
  }
}




for (i in number_to_check[1:4,]){
  for (j in number_to_check[5:6,]){
    eval_multicolinearity[j-6,i-4] &lt;- (polyserial(RNA_seq_info[ ,j], as.factor(RNA_seq_info[ ,i])))^2
    
  }
}


# All others

for (j in number_to_check[7:15,]){
  
  for (i in number_to_check[1:4,]){
    eval_multicolinearity[i-4,j-9] &lt;- (polyserial(RNA_seq_info[ ,j], as.factor(RNA_seq_info[ ,i])))^2
    
  }
}




for (i in number_to_check[1:4,]){
  for (j in number_to_check[7:15,]){
    eval_multicolinearity[j-9,i-4] &lt;- (polyserial(RNA_seq_info[ ,j], as.factor(RNA_seq_info[ ,i])))^2
    
  }
}


#Comparing two numerical variables

#Put the percentage of bps trimmed and number of reads shorter than 20bp in the correct location
eval_multicolinearity[5,5] = 1
eval_multicolinearity[6,6] = 1
for (j in number_to_check[5:6,]){
  
  for (i in number_to_check[7:15,]){

    checkPC1 &lt;- lm(RNA_seq_info[,j] ~ RNA_seq_info[,i])
    eval_multicolinearity[i-9,j-6] &lt;-  summary(checkPC1)$r.squared
    
  }
}

for (j in number_to_check[7:15,]){
  
  for (i in number_to_check[5:6,]){
    
    checkPC1 &lt;- lm(RNA_seq_info[,j] ~ RNA_seq_info[,i])
    eval_multicolinearity[i-6,j-9] &lt;-  summary(checkPC1)$r.squared
    
  }
}

# All others
for (j in number_to_check[7:15,]){
  
  for (i in number_to_check[7:15,]){
    
    checkPC1 &lt;- lm(RNA_seq_info[,j] ~ RNA_seq_info[,i])
    eval_multicolinearity[i-9,j-9] &lt;-  summary(checkPC1)$r.squared
    
  }
}</code></pre>
<pre><code>## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable</code></pre>
<pre><code>## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable

## Warning in summary.lm(checkPC1): essentially perfect fit: summary may be
## unreliable</code></pre>
<pre class="r"><code>colnames(eval_multicolinearity) &lt;- rownames(eval_multicolinearity)
eval_multicolinearity</code></pre>
<pre><code>##                                                          Individual
## Individual                                              1.000000000
## RNA extraction date                                     0.267352428
## Multiplexing index sequence                             0.102588597
## Multiplexing mix codes                                  0.142858085
## Percentage of bps trimmed (adaptors)                    0.004783876
## Number of reads shorter than 20bp removed               0.003454717
## Percentage of mapped reads overlapping a junction       0.020518088
## Number of junctions                                     0.026630340
## Number of reads mapped on orthologous exons             0.046521435
## Number of orthologous exons with at least 1 mapped read 0.001369256
## Number of orthologous genes with at least 1 mapped read 0.001511944
## RNA concentration (ng/uL)                               0.009799367
## RIN score                                               0.284228273
## Library concentration (ng/uL)                           0.053236253
## Library fragments size (bp)                             0.002049270
##                                                         RNA extraction date
## Individual                                                     0.2673524279
## RNA extraction date                                            1.0000000000
## Multiplexing index sequence                                    0.0202712581
## Multiplexing mix codes                                         0.8331814613
## Percentage of bps trimmed (adaptors)                           0.1047014493
## Number of reads shorter than 20bp removed                      0.1038001612
## Percentage of mapped reads overlapping a junction              0.0042038631
## Number of junctions                                            0.3227020152
## Number of reads mapped on orthologous exons                    0.0473247741
## Number of orthologous exons with at least 1 mapped read        0.0007208650
## Number of orthologous genes with at least 1 mapped read        0.0008020008
## RNA concentration (ng/uL)                                      0.1278047912
## RIN score                                                      0.4130470828
## Library concentration (ng/uL)                                  0.2413714601
## Library fragments size (bp)                                    0.1777435099
##                                                         Multiplexing index sequence
## Individual                                                              0.102588597
## RNA extraction date                                                     0.020271258
## Multiplexing index sequence                                             1.000000000
## Multiplexing mix codes                                                  0.001162681
## Percentage of bps trimmed (adaptors)                                    0.011869615
## Number of reads shorter than 20bp removed                               0.022765343
## Percentage of mapped reads overlapping a junction                       0.027728416
## Number of junctions                                                     0.087268913
## Number of reads mapped on orthologous exons                             0.019764270
## Number of orthologous exons with at least 1 mapped read                 0.074285825
## Number of orthologous genes with at least 1 mapped read                 0.041432371
## RNA concentration (ng/uL)                                               0.018576307
## RIN score                                                               0.009811559
## Library concentration (ng/uL)                                           0.003091644
## Library fragments size (bp)                                             0.009005200
##                                                         Multiplexing mix codes
## Individual                                                        0.1428580845
## RNA extraction date                                               0.8331814613
## Multiplexing index sequence                                       0.0011626810
## Multiplexing mix codes                                            1.0000000000
## Percentage of bps trimmed (adaptors)                              0.0913558356
## Number of reads shorter than 20bp removed                         0.0863753198
## Percentage of mapped reads overlapping a junction                 0.0338214186
## Number of junctions                                               0.1215597258
## Number of reads mapped on orthologous exons                       0.0871626617
## Number of orthologous exons with at least 1 mapped read           0.0005463795
## Number of orthologous genes with at least 1 mapped read           0.0202237529
## RNA concentration (ng/uL)                                         0.0452930855
## RIN score                                                         0.6375269697
## Library concentration (ng/uL)                                     0.0695981512
## Library fragments size (bp)                                       0.2849559738
##                                                         Percentage of bps trimmed (adaptors)
## Individual                                                                       0.004783876
## RNA extraction date                                                              0.104701449
## Multiplexing index sequence                                                      0.011869615
## Multiplexing mix codes                                                           0.091355836
## Percentage of bps trimmed (adaptors)                                             1.000000000
## Number of reads shorter than 20bp removed                                                 NA
## Percentage of mapped reads overlapping a junction                                0.021766883
## Number of junctions                                                              0.224265520
## Number of reads mapped on orthologous exons                                      0.015190795
## Number of orthologous exons with at least 1 mapped read                          0.036093820
## Number of orthologous genes with at least 1 mapped read                          0.020298557
## RNA concentration (ng/uL)                                                        0.027327992
## RIN score                                                                        0.006142403
## Library concentration (ng/uL)                                                    0.052360358
## Library fragments size (bp)                                                      0.152213741
##                                                         Number of reads shorter than 20bp removed
## Individual                                                                            0.003454717
## RNA extraction date                                                                   0.103800161
## Multiplexing index sequence                                                           0.022765343
## Multiplexing mix codes                                                                0.086375320
## Percentage of bps trimmed (adaptors)                                                           NA
## Number of reads shorter than 20bp removed                                             1.000000000
## Percentage of mapped reads overlapping a junction                                     0.022902651
## Number of junctions                                                                   0.284209579
## Number of reads mapped on orthologous exons                                           0.042197384
## Number of orthologous exons with at least 1 mapped read                               0.058867895
## Number of orthologous genes with at least 1 mapped read                               0.035770173
## RNA concentration (ng/uL)                                                             0.019462624
## RIN score                                                                             0.003281564
## Library concentration (ng/uL)                                                         0.051807120
## Library fragments size (bp)                                                           0.159896829
##                                                         Percentage of mapped reads overlapping a junction
## Individual                                                                                    0.020518088
## RNA extraction date                                                                           0.004203863
## Multiplexing index sequence                                                                   0.027728416
## Multiplexing mix codes                                                                        0.033821419
## Percentage of bps trimmed (adaptors)                                                          0.021766883
## Number of reads shorter than 20bp removed                                                     0.022902651
## Percentage of mapped reads overlapping a junction                                             1.000000000
## Number of junctions                                                                           0.002112492
## Number of reads mapped on orthologous exons                                                   0.349730195
## Number of orthologous exons with at least 1 mapped read                                       0.002093624
## Number of orthologous genes with at least 1 mapped read                                       0.018588016
## RNA concentration (ng/uL)                                                                     0.002535865
## RIN score                                                                                     0.192826139
## Library concentration (ng/uL)                                                                 0.019272798
## Library fragments size (bp)                                                                   0.009236824
##                                                         Number of junctions
## Individual                                                     0.0266303395
## RNA extraction date                                            0.3227020152
## Multiplexing index sequence                                    0.0872689135
## Multiplexing mix codes                                         0.1215597258
## Percentage of bps trimmed (adaptors)                           0.2242655202
## Number of reads shorter than 20bp removed                      0.2842095792
## Percentage of mapped reads overlapping a junction              0.0021124924
## Number of junctions                                            1.0000000000
## Number of reads mapped on orthologous exons                    0.1049525942
## Number of orthologous exons with at least 1 mapped read        0.5402598732
## Number of orthologous genes with at least 1 mapped read        0.3770659007
## RNA concentration (ng/uL)                                      0.0002780906
## RIN score                                                      0.0009298159
## Library concentration (ng/uL)                                  0.1203002326
## Library fragments size (bp)                                    0.1506713221
##                                                         Number of reads mapped on orthologous exons
## Individual                                                                               0.04652144
## RNA extraction date                                                                      0.04732477
## Multiplexing index sequence                                                              0.01976427
## Multiplexing mix codes                                                                   0.08716266
## Percentage of bps trimmed (adaptors)                                                     0.01519079
## Number of reads shorter than 20bp removed                                                0.04219738
## Percentage of mapped reads overlapping a junction                                        0.34973020
## Number of junctions                                                                      0.10495259
## Number of reads mapped on orthologous exons                                              1.00000000
## Number of orthologous exons with at least 1 mapped read                                  0.18515713
## Number of orthologous genes with at least 1 mapped read                                  0.15756143
## RNA concentration (ng/uL)                                                                0.03373064
## RIN score                                                                                0.19060564
## Library concentration (ng/uL)                                                            0.02418547
## Library fragments size (bp)                                                              0.02180704
##                                                         Number of orthologous exons with at least 1 mapped read
## Individual                                                                                         0.0013692563
## RNA extraction date                                                                                0.0007208650
## Multiplexing index sequence                                                                        0.0742858246
## Multiplexing mix codes                                                                             0.0005463795
## Percentage of bps trimmed (adaptors)                                                               0.0360938200
## Number of reads shorter than 20bp removed                                                          0.0588678946
## Percentage of mapped reads overlapping a junction                                                  0.0020936239
## Number of junctions                                                                                0.5402598732
## Number of reads mapped on orthologous exons                                                        0.1851571262
## Number of orthologous exons with at least 1 mapped read                                            1.0000000000
## Number of orthologous genes with at least 1 mapped read                                            0.9044032910
## RNA concentration (ng/uL)                                                                          0.1399002490
## RIN score                                                                                          0.0432731115
## Library concentration (ng/uL)                                                                      0.0011828522
## Library fragments size (bp)                                                                        0.0051684680
##                                                         Number of orthologous genes with at least 1 mapped read
## Individual                                                                                         1.511944e-03
## RNA extraction date                                                                                8.020008e-04
## Multiplexing index sequence                                                                        4.143237e-02
## Multiplexing mix codes                                                                             2.022375e-02
## Percentage of bps trimmed (adaptors)                                                               2.029856e-02
## Number of reads shorter than 20bp removed                                                          3.577017e-02
## Percentage of mapped reads overlapping a junction                                                  1.858802e-02
## Number of junctions                                                                                3.770659e-01
## Number of reads mapped on orthologous exons                                                        1.575614e-01
## Number of orthologous exons with at least 1 mapped read                                            9.044033e-01
## Number of orthologous genes with at least 1 mapped read                                            1.000000e+00
## RNA concentration (ng/uL)                                                                          1.258408e-01
## RIN score                                                                                          3.861981e-03
## Library concentration (ng/uL)                                                                      3.096126e-05
## Library fragments size (bp)                                                                        3.538886e-04
##                                                         RNA concentration (ng/uL)
## Individual                                                           0.0097993671
## RNA extraction date                                                  0.1278047912
## Multiplexing index sequence                                          0.0185763074
## Multiplexing mix codes                                               0.0452930855
## Percentage of bps trimmed (adaptors)                                 0.0273279919
## Number of reads shorter than 20bp removed                            0.0194626238
## Percentage of mapped reads overlapping a junction                    0.0025358651
## Number of junctions                                                  0.0002780906
## Number of reads mapped on orthologous exons                          0.0337306373
## Number of orthologous exons with at least 1 mapped read              0.1399002490
## Number of orthologous genes with at least 1 mapped read              0.1258408276
## RNA concentration (ng/uL)                                            1.0000000000
## RIN score                                                            0.2636530553
## Library concentration (ng/uL)                                        0.0293692747
## Library fragments size (bp)                                          0.0852433003
##                                                            RIN score
## Individual                                              0.2842282729
## RNA extraction date                                     0.4130470828
## Multiplexing index sequence                             0.0098115589
## Multiplexing mix codes                                  0.6375269697
## Percentage of bps trimmed (adaptors)                    0.0061424028
## Number of reads shorter than 20bp removed               0.0032815640
## Percentage of mapped reads overlapping a junction       0.1928261394
## Number of junctions                                     0.0009298159
## Number of reads mapped on orthologous exons             0.1906056428
## Number of orthologous exons with at least 1 mapped read 0.0432731115
## Number of orthologous genes with at least 1 mapped read 0.0038619814
## RNA concentration (ng/uL)                               0.2636530553
## RIN score                                               1.0000000000
## Library concentration (ng/uL)                           0.0505765177
## Library fragments size (bp)                             0.0800982308
##                                                         Library concentration (ng/uL)
## Individual                                                               5.323625e-02
## RNA extraction date                                                      2.413715e-01
## Multiplexing index sequence                                              3.091644e-03
## Multiplexing mix codes                                                   6.959815e-02
## Percentage of bps trimmed (adaptors)                                     5.236036e-02
## Number of reads shorter than 20bp removed                                5.180712e-02
## Percentage of mapped reads overlapping a junction                        1.927280e-02
## Number of junctions                                                      1.203002e-01
## Number of reads mapped on orthologous exons                              2.418547e-02
## Number of orthologous exons with at least 1 mapped read                  1.182852e-03
## Number of orthologous genes with at least 1 mapped read                  3.096126e-05
## RNA concentration (ng/uL)                                                2.936927e-02
## RIN score                                                                5.057652e-02
## Library concentration (ng/uL)                                            1.000000e+00
## Library fragments size (bp)                                              3.589062e-02
##                                                         Library fragments size (bp)
## Individual                                                             0.0020492697
## RNA extraction date                                                    0.1777435099
## Multiplexing index sequence                                            0.0090052000
## Multiplexing mix codes                                                 0.2849559738
## Percentage of bps trimmed (adaptors)                                   0.1522137406
## Number of reads shorter than 20bp removed                              0.1598968289
## Percentage of mapped reads overlapping a junction                      0.0092368242
## Number of junctions                                                    0.1506713221
## Number of reads mapped on orthologous exons                            0.0218070422
## Number of orthologous exons with at least 1 mapped read                0.0051684680
## Number of orthologous genes with at least 1 mapped read                0.0003538886
## RNA concentration (ng/uL)                                              0.0852433003
## RIN score                                                              0.0800982308
## Library concentration (ng/uL)                                          0.0358906236
## Library fragments size (bp)                                            1.0000000000</code></pre>
<pre class="r"><code>#Find how many pairs have a correlation greater than 0.9, not including variables paired with themselves 

length(eval_multicolinearity [eval_multicolinearity  &gt;= 0.9])</code></pre>
<pre><code>## [1] 19</code></pre>
<pre class="r"><code>#Find these pairs with correlation &gt;=0.9 not including variables compared to themselves

coor_to_check &lt;- which(eval_multicolinearity &gt;= 0.9, arr.ind=T)
coor_to_check &lt;- as.data.frame(coor_to_check)
new_coor &lt;- subset(coor_to_check, row != col, sel = c(row, col))

new_coor</code></pre>
<pre><code>##                                                         row col
## Number of orthologous genes with at least 1 mapped read  11  10
## Number of orthologous exons with at least 1 mapped read  10  11</code></pre>
<p><strong>Conclusions from Part 3:</strong></p>
<p>There is one pair of correlated variables: the number of orthologous exons with at least 1 mapped read and number of orthologous genes with at least 1 mapped read.</p>
</div>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
