<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />


<title>Test_DE_genes</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Test_DE_genes</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>August 3, 2016</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#pairwise-comparisons">Pairwise comparisons</a><ul>
<li><a href="#fit-the-linear-model">Fit the linear model</a></li>
</ul></li>
<li><a href="#find-tissue-specific-genes-found-in-each-species-first-and-then-combine-the-results-in-a-venn-diagram">Find tissue specific genes found in each species first and then combine the results in a Venn Diagram</a></li>
<li><a href="#looking-for-interesting-cases-for-all-of-the-tissue-specific-genes-see-which-have-a-species-specific-component">Looking for interesting cases: for all of the tissue-specific genes, see which have a species-specific component</a></li>
<li><a href="#species-specific-gene-expression-de-between-species-in-all-4-tissues">Species specific gene expression (DE between species in all 4 tissues)</a></li>
<li><a href="#what-happens-when-we-combine-groups-human-and-chimp-versus-rhesus">What happens when we combine groups? (Human and Chimp versus Rhesus)</a></li>
<li><a href="#what-happens-when-we-combine-groups-human-versus-chimp-and-rhesus">What happens when we combine groups? (Human versus Chimp and Rhesus)</a></li>
</ul>
</div>

<p>The goal of this script is to find a list of differentially expressed genes between tissues and between species.</p>
<pre class="r"><code># Load libraries

library(&quot;gplots&quot;)</code></pre>
<pre><code>Warning: package &#39;gplots&#39; was built under R version 3.2.4</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;RColorBrewer&quot;)
library(&quot;scales&quot;)</code></pre>
<pre><code>Warning: package &#39;scales&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;edgeR&quot;)</code></pre>
<pre><code>Warning: package &#39;edgeR&#39; was built under R version 3.2.4</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre><code>Warning: package &#39;limma&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code>library(&quot;R.utils&quot;)</code></pre>
<pre><code>Warning: package &#39;R.utils&#39; was built under R version 3.2.5</code></pre>
<pre><code>Loading required package: R.oo</code></pre>
<pre><code>Warning: package &#39;R.oo&#39; was built under R version 3.2.5</code></pre>
<pre><code>Loading required package: R.methodsS3</code></pre>
<pre><code>Warning: package &#39;R.methodsS3&#39; was built under R version 3.2.3</code></pre>
<pre><code>R.methodsS3 v1.7.1 (2016-02-15) successfully loaded. See ?R.methodsS3 for help.</code></pre>
<pre><code>R.oo v1.21.0 (2016-10-30) successfully loaded. See ?R.oo for help.</code></pre>
<pre><code>
Attaching package: &#39;R.oo&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:methods&#39;:

    getClasses, getMethods</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    attach, detach, gc, load, save</code></pre>
<pre><code>R.utils v2.5.0 (2016-11-07) successfully loaded. See ?R.utils for help.</code></pre>
<pre><code>
Attaching package: &#39;R.utils&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    timestamp</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    cat, commandArgs, getOption, inherits, isOpen, parse, warnings</code></pre>
<pre class="r"><code>library(&quot;plyr&quot;)</code></pre>
<pre><code>Warning: package &#39;plyr&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;limma&quot;)
library(&quot;VennDiagram&quot;)</code></pre>
<pre><code>Warning: package &#39;VennDiagram&#39; was built under R version 3.2.5</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>Loading required package: futile.logger</code></pre>
<pre><code>Warning: package &#39;futile.logger&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>source(&quot;functions.R&quot;)

# Load colors 

colors &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[1],brewer.pal(9, &quot;Blues&quot;)[9]))(100)
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))

# Retrieve RIN score for each sample
RNA_seq_info &lt;- read.csv(&quot;../data/RNA_seq_info.csv&quot;)
RIN &lt;- as.data.frame(RNA_seq_info[,22])
RIN &lt;- RIN[-31,]

# Retrieve sample information
samples &lt;- read.delim(&quot;../data/Sample_info_RNAseq_limma.txt&quot;)
samples &lt;- samples[-31,] # We don&#39;t have a RIN score for human 4 liver

# Eliminate H1H
samples &lt;- samples[-17,]
dim(samples)</code></pre>
<pre><code>[1] 46  4</code></pre>
<pre class="r"><code># Make labels

labels &lt;- paste(samples$Species, samples$Tissue, sep=&quot; &quot;)</code></pre>
<div id="pairwise-comparisons" class="section level2">
<h2>Pairwise comparisons</h2>
<pre class="r"><code>## Make the contrast matrix 

species &lt;- samples$Species
tissue &lt;- samples$Tissue

# Rename columns of the contrast matrix

design &lt;- model.matrix(~ species*tissue + RIN)
colnames(design)[1] &lt;- &quot;Intercept&quot;

colnames(design) &lt;- gsub(&quot;speciesHuman&quot;, &quot;Human&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;speciesRhesus&quot;, &quot;Rhesus&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissuekidney&quot;, &quot;Kidney&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissueliver&quot;, &quot;Liver&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissuelung&quot;, &quot;Lung&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;:&quot;, &quot;.&quot;, colnames(design))</code></pre>
<pre class="r"><code># Load count data

counts_genes_in_cutoff_47 &lt;- read.delim(&quot;../data/counts_12184.txt&quot;)
counts_genes_in_cutoff &lt;- counts_genes_in_cutoff_47[,-30]

# TMM 

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)</code></pre>
<pre><code>                     C1H      C1K      C1Li     C1Lu      C2H      C2K
ENSG00000000003 4.573568 6.488951  8.265202 5.486030 4.691103 6.081032
ENSG00000000419 5.846493 5.222441  5.941935 5.483014 5.685485 5.104872
ENSG00000000457 4.564597 5.219201  5.906964 4.977026 4.838499 5.293882
ENSG00000000460 1.511284 1.874333  2.084694 2.313438 1.665015 1.972697
ENSG00000000938 5.616253 3.824078  5.095620 7.555190 2.537590 4.182601
ENSG00000000971 6.881570 4.456291 11.372553 6.104651 6.140200 4.891851
                     C2Li     C2Lu      C3H      C3K      C3Li     C3Lu
ENSG00000000003  8.033942 4.568963 4.919846 6.410781  7.788836 5.880453
ENSG00000000419  5.817914 5.204323 5.680448 5.183887  6.418152 5.601179
ENSG00000000457  6.549741 4.990390 4.623124 5.208716  6.502523 5.173456
ENSG00000000460  2.329355 2.027982 1.584905 1.466073  2.348643 2.129148
ENSG00000000938  5.392928 8.087913 4.969616 4.227966  5.208902 7.164815
ENSG00000000971 11.391562 6.250982 5.611289 4.945529 11.424637 5.995247
                     C4H      C4K      C4Li     C4Lu      H1K      H1Li
ENSG00000000003 4.240221 6.508187  8.458198 5.434692 6.869131  6.580552
ENSG00000000419 5.789883 5.262406  5.886005 5.326251 5.592621  6.087467
ENSG00000000457 4.649760 5.027691  6.601969 5.268275 4.289473  4.958293
ENSG00000000460 1.461067 1.831232  2.211280 2.481119 2.771224  4.993804
ENSG00000000938 3.643416 3.625703  4.584843 7.722234 4.063809  4.484410
ENSG00000000971 6.849690 5.962307 11.335381 6.425887 6.590016 11.221112
                    H1Lu      H2H      H2K     H2Li     H2Lu         H3H
ENSG00000000003 5.103472 3.685552 7.210038 6.643414 4.185570  3.54806908
ENSG00000000419 5.815324 5.610795 5.466147 5.842913 5.317919  5.75537924
ENSG00000000457 4.506583 3.411145 4.162933 4.455307 4.206318  4.44509881
ENSG00000000460 3.322483 1.896662 1.982948 2.662376 2.557536 -0.07138695
ENSG00000000938 7.882637 5.564510 3.744777 5.994769 6.973363  4.10130761
ENSG00000000971 7.565878 6.367758 4.740910 9.413943 7.315285  6.22954405
                     H3K      H3Li     H3Lu      H4H      H4K     H4Lu
ENSG00000000003 7.096040  7.740415 5.294566 4.288642 6.376252 4.467923
ENSG00000000419 5.859410  6.221288 5.014314 6.248997 5.612557 5.354892
ENSG00000000457 4.727258  4.998187 4.003636 3.316831 4.091946 4.178113
ENSG00000000460 2.995062  3.242213 2.461715 1.634400 1.987589 2.608895
ENSG00000000938 2.647590  5.745535 6.917217 4.922960 3.825316 8.253473
ENSG00000000971 5.317724 10.350971 7.128721 6.931559 6.037082 8.281707
                     R1H      R1K       R1Li     R1Lu      R2H      R2K
ENSG00000000003 4.360836 6.937403  8.3387961 5.920017 4.629816 7.138654
ENSG00000000419 5.468551 5.396303  5.8304125 4.891525 5.299986 5.048299
ENSG00000000457 4.289677 5.029321  5.1480101 4.841841 4.316130 5.159301
ENSG00000000460 1.288792 1.559698 -0.1691694 2.485003 1.420245 1.324437
ENSG00000000938 1.841655 2.568666  3.8086285 6.519794 2.331513 2.686205
ENSG00000000971 4.032377 6.580489 12.1367355 7.450446 5.576154 6.783171
                     R2Li     R2Lu      R3H      R3K      R3Li     R3Lu
ENSG00000000003  8.644762 5.659132 4.473506 7.170518  8.206895 5.457959
ENSG00000000419  5.727589 4.885918 5.467427 5.371831  5.994773 4.960583
ENSG00000000457  5.515590 5.270085 4.245915 5.144261  5.364003 5.169900
ENSG00000000460  1.501379 2.399349 1.661056 1.878860  1.418164 2.581707
ENSG00000000938  3.676711 6.587845 2.875250 2.271666  3.640782 6.850353
ENSG00000000971 11.782724 7.236467 5.154956 6.058685 12.091712 7.262728
                     R4H      R4K      R4Li     R4Lu
ENSG00000000003 4.896983 7.098877  7.709806 5.365706
ENSG00000000419 5.431843 5.220175  5.468637 5.046226
ENSG00000000457 4.514151 5.148735  5.322679 4.832820
ENSG00000000460 1.092607 1.513856  1.538624 2.852017
ENSG00000000938 2.479232 2.315024  4.513771 6.839315
ENSG00000000971 6.413351 6.473508 11.699950 7.302673</code></pre>
<pre class="r"><code>hist(cpm_in_cutoff, xlab = &quot;Log2(CPM)&quot;, main = &quot;Log2(CPM) values for genes meeting the filtering criteria&quot;, breaks = 100 )</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Voom with individual as a random variable

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=T)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus &lt;- 0.221922

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit.consensus)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<div id="fit-the-linear-model" class="section level3">
<h3>Fit the linear model</h3>
<pre class="r"><code>fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)

# MA Plots
## If &#39;MA&#39; is an &#39;MArrayLM&#39; object, then the plot is a fitted model MA-plot in which the estimated coefficient is on the y-axis and the average A-value is on the x-axis.                

limma::plotMA(fit.cyclic.norm, array=1, xlab=&quot;average coefficient&quot;, ylab=&quot;estimated coefficient&quot;)</code></pre>
<pre><code>Warning in plot.window(...): &quot;array&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in plot.xy(xy, type, ...): &quot;array&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in axis(side = side, at = at, labels = labels, ...): &quot;array&quot; is not
a graphical parameter

Warning in axis(side = side, at = at, labels = labels, ...): &quot;array&quot; is not
a graphical parameter</code></pre>
<pre><code>Warning in box(...): &quot;array&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in title(...): &quot;array&quot; is not a graphical parameter</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## - Potential caveat: variances could be different between human, chimp and rhesus (see Gordon Smyth email, 7 June 2013).                                                               
##  We look at the standard error for each condition                                                    
hist(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma, breaks=100)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma), breaks=100)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma))</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## This seems to be pretty comparable between conditions. The human heart is higher, probably because of H1H missing and H3H with a bit strange behavior                                     
stderror &lt;- log2(fit.cyclic.norm$stdev.unscaled * fit.cyclic.norm$sigma)
boxplot(list(stderror[,1:4], stderror[,5:8], stderror[,9:12]))</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## A bit higher for human, and a bit lower for rhesus                                                                                                                                    
boxplot(list(stderror[,2:4], stderror[,6:8], stderror[,8:12])) ## excluding heart samples  </code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># In the contrast matrix, we have many comparisons for species and tissues individually
# Note: baseline is chimp heart

cm1 &lt;- makeContrasts(HvC_Liver = Human + Liver + Human.Liver - Liver, 
                     HvC_Lung = Human + Lung + Human.Lung - Lung, 
                     HvC_Heart = Human, 
                     HvC_Kidney = Human + Kidney + Human.Kidney - Kidney, 
                     HvR_Liver = Human + Liver + Human.Liver - (Rhesus + Liver + Rhesus.Liver), 
                     HvR_Lung = Human + Lung + Human.Lung - (Rhesus + Lung + Rhesus.Lung), 
                     HvR_Heart = Human - Rhesus, 
                     HvR_Kidney = Human + Kidney + Human.Kidney - (Rhesus + Kidney + Rhesus.Kidney), 
                     CvR_Liver = Liver - (Rhesus + Liver + Rhesus.Liver),  
                     CvR_Lung = Lung - (Rhesus + Lung + Rhesus.Lung), 
                     CvR_Heart = Rhesus, 
                     CvR_Kidney = Kidney - (Rhesus + Kidney + Rhesus.Kidney), 
                     H_HeartvLi = Human - (Human + Liver + Human.Liver), 
                     H_HeartvLu = Human - (Human + Lung + Human.Lung), 
                     H_HeartvK = Human - (Human + Kidney + Human.Kidney), 
                     H_LivLu = (Human + Liver + Human.Liver) - (Human + Lung + Human.Lung), 
                     H_LivK = (Human + Liver + Human.Liver) - (Human + Kidney + Human.Kidney),  
                     H_LuvK = (Human + Lung + Human.Lung) - (Human + Kidney + Human.Kidney), 
                     C_HeartvLi = Liver, 
                     C_HeartvLu = Lung, 
                     C_HeartvK = Kidney, 
                     C_LivLu = Liver - Lung, 
                     C_LivK = Liver - Kidney,  
                     C_LuvK = Lung - Kidney,  
                     R_HeartvLi = Rhesus - (Rhesus + Liver + Rhesus.Liver), 
                     R_HeartvLu = Rhesus - (Rhesus + Lung + Rhesus.Lung), 
                     R_HeartvK = Rhesus - (Rhesus + Kidney + Rhesus.Kidney), 
                     R_LivLu = (Rhesus + Liver + Rhesus.Liver) - (Rhesus + Lung + Rhesus.Lung), 
                     R_LivK = (Rhesus + Liver + Rhesus.Liver) - (Rhesus + Kidney + Rhesus.Kidney),  
                     R_LuvK = (Rhesus + Lung + Rhesus.Lung) - (Rhesus + Kidney + Rhesus.Kidney), 
                     Sig_HLi = Human.Liver, 
                     Sig_HLu = Human.Lung, 
                     Sig_HK = Human.Kidney, 
                     Sig_RLi = Rhesus.Liver, 
                     Sig_RLu = Rhesus.Lung, 
                     Sig_RK = Rhesus.Kidney, 
                     levels = design)




# Implement contrasts

contrasts_each_species &lt;- contrasts.fit(fit.cyclic.norm, cm1)
fit1 &lt;- eBayes(contrasts_each_species)

top3 &lt;- list(HvC_Liver =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HvC_Lung =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             HvC_Heart =topTable(fit1, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             HvC_Kidney =topTable(fit1, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             
             HvR_Liver =topTable(fit1, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HvR_Lung =topTable(fit1, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             HvR_Heart =topTable(fit1, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             HvR_Kidney =topTable(fit1, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             
             CvR_Liver =topTable(fit1, coef=9, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CvR_Lung =topTable(fit1, coef=10, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             CvR_Heart =topTable(fit1, coef=11, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             CvR_Kidney =topTable(fit1, coef=12, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             H_HeartvLi =topTable(fit1, coef=13, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             H_HeartvLu =topTable(fit1, coef=14, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             H_HeartvK =topTable(fit1, coef=15, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             H_LivLu =topTable(fit1, coef=16, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             H_LivK =topTable(fit1, coef=17, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             H_LuvK =topTable(fit1, coef=18, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             C_HeartvLi =topTable(fit1, coef=19, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             C_HeartvLu =topTable(fit1, coef=20, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             C_HeartvK =topTable(fit1, coef=21, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             C_LivLu =topTable(fit1, coef=22, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             C_LivK =topTable(fit1, coef=23, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             C_LuvK =topTable(fit1, coef=24, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             R_HeartvLi =topTable(fit1, coef=25, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             R_HeartvLu =topTable(fit1, coef=26, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             R_HeartvK =topTable(fit1, coef=27, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             R_LivLu =topTable(fit1, coef=28, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             R_LivK =topTable(fit1, coef=29, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             R_LuvK =topTable(fit1, coef=30, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             
             Sig_HLi =topTable(fit1, coef=31, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sig_HLu =topTable(fit1, coef=32, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             Sig_HK =topTable(fit1, coef=33, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             Sig_RLi =topTable(fit1, coef=34, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sig_RLu =topTable(fit1, coef=35, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sig_RK =topTable(fit1, coef=36, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;) )


# Set FDR level at 1% 

FDR_level &lt;- 0.01

## DE between HvC

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Humans and Chimps (FDR 1%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between HvR

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[8]]])[top3[[names(top3)[8]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Humans and Rhesus (FDR 1%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between CvR

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[9]]])[top3[[names(top3)[9]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[10]]])[top3[[names(top3)[10]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[11]]])[top3[[names(top3)[11]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[12]]])[top3[[names(top3)[12]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Chimps and Rhesus (FDR 1%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Heart and Liver

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Heart and Liver (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Heart and Lung

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Heart and Lung (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## DE between Heart and Kidney

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Heart and Kidney (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## DE between Liver and Lung

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Liver and Lung (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Liver and Kidney

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Liver and Kidney (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)

## DE between Lung and Kidney

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Chimp&quot;]] &lt;-  row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus&quot;]] &lt;- row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Lung and Kidney (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
</div>
</div>
<div id="find-tissue-specific-genes-found-in-each-species-first-and-then-combine-the-results-in-a-venn-diagram" class="section level2">
<h2>Find tissue specific genes found in each species first and then combine the results in a Venn Diagram</h2>
<pre class="r"><code># Set FDR level at 1%

FDR_level &lt;- 0.01

### Heart specific (DE in Heart v. Liver, Heart v. Lung, Heart v. Kidney but not Liver versus Lung, Liver versus Kidney, Lung v. Kidney)

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &gt; FDR_level])


mylist[[&quot;Chimp&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level])
  
mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &gt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Heart specific DE genes (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


# Set FDR level at 1%

FDR_level &lt;- 0.01

### Liver specific

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &gt; FDR_level])


mylist[[&quot;Chimp&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level])
  
mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &gt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Liver specific DE genes (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)




### Kidney specific 
mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &gt; FDR_level])


mylist[[&quot;Chimp&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &gt; FDR_level])
  
mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &gt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Kidney specific DE genes (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)



### Lung specific 
mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &gt; FDR_level])


mylist[[&quot;Chimp&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &gt; FDR_level])
  
mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &gt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Lung specific DE genes (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
</div>
<div id="looking-for-interesting-cases-for-all-of-the-tissue-specific-genes-see-which-have-a-species-specific-component" class="section level2">
<h2>Looking for interesting cases: for all of the tissue-specific genes, see which have a species-specific component</h2>
<pre class="r"><code># Set FDR level at 1%

FDR_level &lt;- 0.01

### Heart specific with interactions

mylist &lt;- list()

# Heart specific with human-by-heart interaction
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[32]]])[top3[[names(top3)[32]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[33]]])[top3[[names(top3)[33]]]$adj.P.Val &lt; FDR_level])

# Heart specific with chimp-by-tissue interactions

chimp_specific &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[32]]])[top3[[names(top3)[32]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[33]]])[top3[[names(top3)[33]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[35]]])[top3[[names(top3)[35]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[36]]])[top3[[names(top3)[36]]]$adj.P.Val &lt; FDR_level])

chimp_human &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[32]]])[top3[[names(top3)[32]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[33]]])[top3[[names(top3)[33]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[35]]])[top3[[names(top3)[35]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[36]]])[top3[[names(top3)[36]]]$adj.P.Val &gt; FDR_level])

chimp_rhesus &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[32]]])[top3[[names(top3)[32]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[33]]])[top3[[names(top3)[33]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[35]]])[top3[[names(top3)[35]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[36]]])[top3[[names(top3)[36]]]$adj.P.Val &lt; FDR_level])

mylist[[&quot;Chimp&quot;]] &lt;- union(union(chimp_specific, chimp_human), chimp_rhesus)

  
mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &gt; FDR_level]),  row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[35]]])[top3[[names(top3)[35]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[36]]])[top3[[names(top3)[36]]]$adj.P.Val &lt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Heart specific DE genes with species-by-tissue interactions (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


# Set FDR level at 1%

FDR_level &lt;- 0.01

### Liver specific with a human by liver interaction

mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[13]]])[top3[[names(top3)[13]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[16]]])[top3[[names(top3)[16]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[17]]])[top3[[names(top3)[17]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[14]]])[top3[[names(top3)[14]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[15]]])[top3[[names(top3)[15]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[18]]])[top3[[names(top3)[18]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[32]]])[top3[[names(top3)[32]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[33]]])[top3[[names(top3)[33]]]$adj.P.Val &gt; FDR_level])

mylist[[&quot;Chimp&quot;]] </code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>chimp_specific &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &lt; FDR_level])

chimp_human &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &gt; FDR_level])
  
chimp_rhesus &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[19]]])[top3[[names(top3)[19]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[22]]])[top3[[names(top3)[22]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[23]]])[top3[[names(top3)[23]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[20]]])[top3[[names(top3)[20]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[21]]])[top3[[names(top3)[21]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[24]]])[top3[[names(top3)[24]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[31]]])[top3[[names(top3)[31]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &lt; FDR_level])

mylist[[&quot;Chimp&quot;]] &lt;- union(union(chimp_specific, chimp_human), chimp_rhesus)

  
mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[25]]])[top3[[names(top3)[25]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[28]]])[top3[[names(top3)[28]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[29]]])[top3[[names(top3)[29]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[26]]])[top3[[names(top3)[26]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[27]]])[top3[[names(top3)[27]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[30]]])[top3[[names(top3)[30]]]$adj.P.Val &gt; FDR_level]),  row.names(top3[[names(top3)[34]]])[top3[[names(top3)[34]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[35]]])[top3[[names(top3)[35]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[36]]])[top3[[names(top3)[36]]]$adj.P.Val &gt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Liver specific DE genes with species-specific interactions (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
</div>
<div id="species-specific-gene-expression-de-between-species-in-all-4-tissues" class="section level2">
<h2>Species specific gene expression (DE between species in all 4 tissues)</h2>
<pre class="r"><code>### DE between Human and Chimp/Human and Rhesus but not DE between Chimp and Rhesus
mylist &lt;- list()
mylist[[&quot;Human&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &lt; FDR_level]),          row.names(top3[[names(top3)[8]]])[top3[[names(top3)[8]]]$adj.P.Val &lt; FDR_level]),          row.names(top3[[names(top3)[9]]])[top3[[names(top3)[9]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[10]]])[top3[[names(top3)[10]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[11]]])[top3[[names(top3)[11]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[12]]])[top3[[names(top3)[12]]]$adj.P.Val &gt; FDR_level])


mylist[[&quot;Chimp&quot;]]  &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[9]]])[top3[[names(top3)[9]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[10]]])[top3[[names(top3)[10]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[11]]])[top3[[names(top3)[11]]]$adj.P.Val &lt; FDR_level]),          row.names(top3[[names(top3)[12]]])[top3[[names(top3)[12]]]$adj.P.Val &lt; FDR_level]),          row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[8]]])[top3[[names(top3)[8]]]$adj.P.Val &gt; FDR_level])

mylist[[&quot;Rhesus&quot;]] &lt;- intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(intersect(row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level], row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[8]]])[top3[[names(top3)[8]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[9]]])[top3[[names(top3)[9]]]$adj.P.Val &lt; FDR_level]), row.names(top3[[names(top3)[10]]])[top3[[names(top3)[10]]]$adj.P.Val &lt; FDR_level]),  row.names(top3[[names(top3)[11]]])[top3[[names(top3)[11]]]$adj.P.Val &lt; FDR_level]),          row.names(top3[[names(top3)[12]]])[top3[[names(top3)[12]]]$adj.P.Val &lt; FDR_level]),          row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &gt; FDR_level]), row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &gt; FDR_level])
  

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Species specific DE genes (FDR 1%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
</div>
<div id="what-happens-when-we-combine-groups-human-and-chimp-versus-rhesus" class="section level2">
<h2>What happens when we combine groups? (Human and Chimp versus Rhesus)</h2>
<pre class="r"><code>## Make the contrast matrix 

species &lt;- samples$Species

# Replace Chimp and Human with &quot;GA&quot;

chimp_GA &lt;- gsub(&quot;Chimp&quot;, &quot;GA&quot;, species)
new_clade &lt;- gsub(&quot;Human&quot;, &quot;GA&quot;, chimp_GA)

tissue &lt;- samples$Tissue

# Rename columns of the contrast matrix

design &lt;- model.matrix(~ new_clade*tissue)
colnames(design)[1] &lt;- &quot;Intercept&quot;

colnames(design) &lt;- gsub(&quot;new_cladeRhesus&quot;, &quot;Rhesus&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissuekidney&quot;, &quot;Kidney&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissueliver&quot;, &quot;Liver&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissuelung&quot;, &quot;Lung&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;:&quot;, &quot;.&quot;, colnames(design))

# Voom with individual as a random variable

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=T)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;- 0.2631636

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fit the linear model
fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)


# Fit model

cm3 &lt;- makeContrasts(GAvR_Liver = Rhesus + Rhesus.Liver,
                     GAvR_Lung = Rhesus + Rhesus.Lung, 
                     GAvR_Heart = Rhesus, 
                     GAvR_Kidney = Rhesus + Rhesus.Kidney,
                     Sig_RLiver = Rhesus.Liver,
                     Sig_RLung = Rhesus.Lung,
                     Sig_RKidney = Rhesus.Kidney,
                     levels = design)

contrasts_GA &lt;- contrasts.fit(fit.cyclic.norm, cm3)
fit2 &lt;- eBayes(contrasts_GA)

top3 &lt;- list(GA_Liver = topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             GA_Lung = topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             GA_Heart = topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             GA_Kidney = topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sign_RLiver = topTable(fit2, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             Sign_RLung = topTable(fit2, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sign_RKidney =topTable(fit2, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Set FDR Level

FDR_level &lt;- 0.0001

## DE between GA and Rhesus

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Great Apes and Rhesus (0.01% FDR)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## Sign interaction

mylist &lt;- list()
mylist[[&quot;Rhesus by Liver&quot;]] &lt;- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus by Lung&quot;]] &lt;-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Rhesus by Kidney&quot;]] &lt;- row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Significant interaction in rhesus (0.01% FDR)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)



# Set FDR Level

FDR_level &lt;- 0.001

## DE between GA and Rhesus

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Great Apes and Rhesus (0.1% FDR)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


# Set FDR Level

FDR_level &lt;- 0.01

## DE between GA and Rhesus

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Great Apes and Rhesus (1% FDR)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
</div>
<div id="what-happens-when-we-combine-groups-human-versus-chimp-and-rhesus" class="section level2">
<h2>What happens when we combine groups? (Human versus Chimp and Rhesus)</h2>
<pre class="r"><code>## Make the contrast matrix 

species &lt;- samples$Species

# Replace Chimp and Rhesus with &quot;CR&quot;

chimp_GA &lt;- gsub(&quot;Chimp&quot;, &quot;CR&quot;, species)
new_clade &lt;- gsub(&quot;Rhesus&quot;, &quot;CR&quot;, chimp_GA)

tissue &lt;- samples$Tissue

# Rename columns of the contrast matrix

design &lt;- model.matrix(~ new_clade*tissue)
colnames(design)[1] &lt;- &quot;Intercept&quot;

colnames(design) &lt;- gsub(&quot;new_cladeHuman&quot;, &quot;Human&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissuekidney&quot;, &quot;Kidney&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissueliver&quot;, &quot;Liver&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;tissuelung&quot;, &quot;Lung&quot;, colnames(design))
colnames(design) &lt;- gsub(&quot;:&quot;, &quot;.&quot;, colnames(design))

# Voom with individual as a random variable

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=T)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit$consensus &lt;- 0.3073617

# Final voom on filtered data

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/Test_DE_genes_RIN.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fit the linear model
fit.cyclic.norm &lt;- lmFit(cpm.voom.cyclic, design, plot = TRUE, block=samples$Individual, correlation=corfit$consensus)
fit.cyclic.norm &lt;- eBayes(fit.cyclic.norm)


# Fit model

cm4 &lt;- makeContrasts(HvCR_Liver = Human + Human.Liver,
                     HvCR_Lung = Human + Human.Lung, 
                     HvCR_Heart = Human, 
                     HvCR_Kidney = Human + Human.Kidney,
                     Sig_HLiver = Human.Liver,
                     Sig_HLung = Human.Lung,
                     Sig_HKidney = Human.Kidney,
                     levels = design)

contrasts_CR &lt;- contrasts.fit(fit.cyclic.norm, cm4)
fit2 &lt;- eBayes(contrasts_CR)

top3 &lt;- list(CR_Liver = topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             CR_Lung = topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CR_Heart = topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  
             CR_Kidney = topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sign_HLiver = topTable(fit2, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),
             Sign_HLung = topTable(fit2, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), 
             Sign_HKidney =topTable(fit2, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Set FDR Level

FDR_level &lt;- 0.0001

## DE between Human and Chimp/Rhesus

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Human and Chimp/Rhesus (0.01% FDR)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


## Sign interaction

mylist &lt;- list()
mylist[[&quot;Human by Liver&quot;]] &lt;- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Human by Lung&quot;]] &lt;-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Human by Kidney&quot;]] &lt;- row.names(top3[[names(top3)[7]]])[top3[[names(top3)[7]]]$adj.P.Val &lt; FDR_level]

# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Significant interaction in humans (0.01% FDR)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


# Set FDR Level

FDR_level &lt;- 0.001

## DE between GA and Rhesus

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Great Apes and Rhesus (0.1% FDR)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)


# Set FDR Level

FDR_level &lt;- 0.01

## DE between GA and Rhesus

mylist &lt;- list()
mylist[[&quot;Liver&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Lung&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Heart&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]


# Make 
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between Humans versus Chimps/Rhesus (1% FDR)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
grid.draw(Four_comp)</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
